<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2023-10-24">
<title>Will’s Public NAO Notebook - Initial analysis of Project Runway protocol testing data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Initial analysis of Project Runway protocol testing data</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">October 24, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>On 2023-10-23, we received the first batch of sequencing data from our wastewater protocol optimization work. This notebook contains the output of initial QC and analysis for this data.</p>
<section id="background" class="level1"><h1>Background</h1>
<p>We carried out concentration and extraction on primary sludge samples from wastewater treatment plant A (WTP-A) using three different nucleic-acid extraction kits:</p>
<ul>
<li><p>Zymo quick-RNA kit</p></li>
<li><p>Zymo quick-DNA/RNA kit</p></li>
<li><p>QIAamp Viral RNA mini kit</p></li>
</ul>
<p>We <a href="https://data.securebio.org/wills-public-notebook/notebooks/2023-09-12_settled-solids-extraction-test.html">previously</a> looked at total NA yield, qPCR and Fragment Analyzer results for these extractions as well as several other kits. We selected these three kits as the most promising, and sent them for sequencing to further characterize the quality of the extractions. Two replicate extractions per kit were sent for sequencing; one (replicate A) underwent RNA sequencing only while the other (replicate C) underwent both DNA and RNA sequencing. We also sent two samples extracted in a previous experiment (using the QIAamp kit) for RNA sequencing.</p>
<p>So far, only the DNA sequencing data is available, corresponding to one replicate per kit, or three samples total.</p>
</section><section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The raw data from the BMC consists of sixteen FASTQ files, corresponding to paired-end data from eight sequencing libraries. This is rather more than the three libraries we were expecting; digging into this, each submitted sample seems to have been sequenced twice, and we also have two pairs of FASTQ files corresponding to “unmapped barcodes” (<code>231016_78783_6418E_L1_{1,2}_sequence_unmapped_barcodes.fastq</code> and <code>231016_78783_6418E_L2_{1,2}_sequence_unmapped_barcodes.fastq</code>).</p>
<p>Given that these have the same sequencing date and other annotations, and that (according to communication from the BMC) each submitted library has the same index barcode in both cases, my guess is that each sample was sequenced on both lanes of the AVITI flow cell. However, it would be useful to check with BMC and confirm this.</p>
<p>The data from the BMC helpfully comes with FASTQC files attached, which we can use to look at the quality of the raw data.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2023-10-24_pr-dna"</span></span>
<span><span class="va">outer_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc/raw"</span><span class="op">)</span></span>
<span><span class="va">qc_archives</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir</span>, pattern <span class="op">=</span> <span class="st">"*.zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Sample mapping</span></span>
<span><span class="va">sample_metadata</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"../data/2023-10-24_pr-dna/sample-metadata.csv"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute approximate number of sequence bases from sequence length distribution table</span></span>
<span><span class="va">compute_n_bases</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">sld</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">min_lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"-.*"</span>, <span class="st">""</span>, <span class="va">sld</span><span class="op">$</span><span class="va">Length</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span></span>
<span>  <span class="va">max_lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">".*-"</span>, <span class="st">""</span>, <span class="va">sld</span><span class="op">$</span><span class="va">Length</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span></span>
<span>  <span class="va">mean_lengths</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">min_lengths</span> <span class="op">+</span> <span class="va">max_lengths</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span></span>
<span>  <span class="va">sld2</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">sld</span>, Length_mean <span class="op">=</span> <span class="va">mean_lengths</span><span class="op">)</span></span>
<span>  <span class="va">n_bases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">sld2</span><span class="op">$</span><span class="va">Count</span> <span class="op">*</span> <span class="va">sld2</span><span class="op">$</span><span class="va">Length_mean</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># FASTQC processing functions</span></span>
<span><span class="va">extract_qc_data_base</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">qc</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">filename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">basic_statistics</span>, <span class="va">Measure</span> <span class="op">==</span> <span class="st">"Filename"</span><span class="op">)</span><span class="op">$</span><span class="va">Value</span></span>
<span>  <span class="va">n_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">basic_statistics</span>, <span class="va">Measure</span> <span class="op">==</span> <span class="st">"Total Sequences"</span><span class="op">)</span><span class="op">$</span><span class="va">Value</span> <span class="op">%&gt;%</span> <span class="va">as.integer</span></span>
<span>  <span class="va">n_bases</span> <span class="op">&lt;-</span> <span class="fu">compute_n_bases</span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">sequence_length_distribution</span><span class="op">)</span></span>
<span>  <span class="va">pc_dup</span> <span class="op">&lt;-</span> <span class="fl">100</span><span class="op">-</span><span class="va">qc</span><span class="op">$</span><span class="va">total_deduplicated_percentage</span></span>
<span>  <span class="va">per_base_quality</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Per base sequence quality"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">per_sequence_quality</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Per sequence quality scores"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">per_base_composition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Per base sequence content"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">per_sequence_composition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Per sequence GC content"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">per_base_n_content</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Per base N content"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">adapter_content</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qc</span><span class="op">$</span><span class="va">summary</span>, <span class="va">module</span> <span class="op">==</span> <span class="st">"Adapter Content"</span><span class="op">)</span><span class="op">$</span><span class="va">status</span></span>
<span>  <span class="va">tab_out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="va">n_reads</span>,</span>
<span>                    n_bases <span class="op">=</span> <span class="va">n_bases</span>,</span>
<span>                    pc_dup <span class="op">=</span> <span class="va">pc_dup</span>,</span>
<span>                    per_base_quality <span class="op">=</span> <span class="va">per_base_quality</span>,</span>
<span>                    per_sequence_quality <span class="op">=</span> <span class="va">per_sequence_quality</span>,</span>
<span>                    per_base_composition <span class="op">=</span> <span class="va">per_base_composition</span>,</span>
<span>                    per_sequence_composition <span class="op">=</span> <span class="va">per_sequence_composition</span>,</span>
<span>                    per_base_n_content <span class="op">=</span> <span class="va">per_base_n_content</span>,</span>
<span>                    adapter_content <span class="op">=</span> <span class="va">adapter_content</span>,</span>
<span>                    filename <span class="op">=</span> <span class="va">filename</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tab_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">extract_qc_data_paired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">qc</span>, <span class="va">id_pattern</span>, <span class="va">pair_pattern</span> <span class="op">=</span> <span class="st">".*_(\\d)[_.].*"</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">base</span> <span class="op">&lt;-</span> <span class="fu">extract_qc_data_base</span><span class="op">(</span><span class="va">qc</span><span class="op">)</span></span>
<span>  <span class="va">filename</span> <span class="op">&lt;-</span> <span class="va">base</span><span class="op">$</span><span class="va">filename</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="va">id_pattern</span>, <span class="st">""</span>, <span class="va">filename</span><span class="op">)</span></span>
<span>  <span class="va">read_pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="va">pair_pattern</span>, <span class="st">"\\1"</span>, <span class="va">filename</span><span class="op">)</span></span>
<span>  <span class="va">tab_out</span> <span class="op">&lt;-</span> <span class="va">base</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>sequencing_id <span class="op">=</span> <span class="va">prefix</span>, read_pair <span class="op">=</span> <span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">sequencing_id</span>, <span class="va">read_pair</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tab_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">extract_qc_data_unpaired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">qc</span>, <span class="va">id_pattern</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">base</span> <span class="op">&lt;-</span> <span class="fu">extract_qc_data_base</span><span class="op">(</span><span class="va">qc</span><span class="op">)</span></span>
<span>  <span class="va">filename</span> <span class="op">&lt;-</span> <span class="va">base</span><span class="op">$</span><span class="va">filename</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="va">id_pattern</span>, <span class="st">""</span>, <span class="va">filename</span><span class="op">)</span></span>
<span>  <span class="va">tab_out</span> <span class="op">&lt;-</span> <span class="va">base</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>sequencing_id <span class="op">=</span> <span class="va">prefix</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="va">sequencing_id</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">tab_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Generate and show sample table</span></span>
<span><span class="va">pattern_raw</span> <span class="op">&lt;-</span> <span class="st">"_\\d_sequence.fastq.*$"</span></span>
<span><span class="va">qc_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">extract_qc_data_paired</span><span class="op">(</span><span class="va">x</span>, <span class="va">pattern_raw</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">sample_metadata</span>, by <span class="op">=</span> <span class="st">"sequencing_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">qc_data</span><span class="op">)</span>  </span>
<span><span class="va">qc_tab</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["library_id"],"name":[2],"type":["chr"],"align":["left"]},{"label":["kit"],"name":[3],"type":["chr"],"align":["left"]},{"label":["sequencing_replicate"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["read_pair"],"name":[5],"type":["chr"],"align":["left"]},{"label":["sequencing_id"],"name":[6],"type":["chr"],"align":["left"]},{"label":["n_reads"],"name":[7],"type":["int"],"align":["right"]},{"label":["n_bases"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["pc_dup"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["per_base_quality"],"name":[10],"type":["chr"],"align":["left"]},{"label":["per_sequence_quality"],"name":[11],"type":["chr"],"align":["left"]},{"label":["per_base_composition"],"name":[12],"type":["chr"],"align":["left"]},{"label":["per_sequence_composition"],"name":[13],"type":["chr"],"align":["left"]},{"label":["per_base_n_content"],"name":[14],"type":["chr"],"align":["left"]},{"label":["adapter_content"],"name":[15],"type":["chr"],"align":["left"]},{"label":["filename"],"name":[16],"type":["chr"],"align":["left"]},{"label":["index"],"name":[17],"type":["chr"],"align":["left"]},{"label":["stage"],"name":[18],"type":["chr"],"align":["left"]}],"data":[{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"1","5":"1","6":"230926EsvA_D23-13404-1","7":"115069046","8":"17260356900","9":"9.48","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13404-1_1_sequence.fastq","17":"CCCAAGTT","18":"raw"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"1","5":"2","6":"230926EsvA_D23-13404-1","7":"115069046","8":"17260356900","9":"9.18","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13404-1_2_sequence.fastq","17":"CCCAAGTT","18":"raw"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"2","5":"1","6":"230926EsvA_D23-13404-2","7":"114841958","8":"17226293700","9":"9.43","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13404-2_1_sequence.fastq","17":"CCCAAGTT","18":"raw"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"2","5":"2","6":"230926EsvA_D23-13404-2","7":"114841958","8":"17226293700","9":"9.13","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13404-2_2_sequence.fastq","17":"CCCAAGTT","18":"raw"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"1","5":"1","6":"230926EsvA_D23-13405-1","7":"98396157","8":"14759423550","9":"8.34","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13405-1_1_sequence.fastq","17":"TCTAGGCT","18":"raw"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"1","5":"2","6":"230926EsvA_D23-13405-1","7":"98396157","8":"14759423550","9":"8.27","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13405-1_2_sequence.fastq","17":"TCTAGGCT","18":"raw"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"2","5":"1","6":"230926EsvA_D23-13405-2","7":"98159057","8":"14723858550","9":"8.45","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13405-2_1_sequence.fastq","17":"TCTAGGCT","18":"raw"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"2","5":"2","6":"230926EsvA_D23-13405-2","7":"98159057","8":"14723858550","9":"8.31","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13405-2_2_sequence.fastq","17":"TCTAGGCT","18":"raw"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"1","5":"1","6":"230926EsvA_D23-13406-1","7":"59522621","8":"8928393150","9":"5.00","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13406-1_1_sequence.fastq","17":"CCGCTATT","18":"raw"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"1","5":"2","6":"230926EsvA_D23-13406-1","7":"59522621","8":"8928393150","9":"4.14","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13406-1_2_sequence.fastq","17":"CCGCTATT","18":"raw"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"2","5":"1","6":"230926EsvA_D23-13406-2","7":"59462666","8":"8919399900","9":"5.11","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13406-2_1_sequence.fastq","17":"CCGCTATT","18":"raw"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"2","5":"2","6":"230926EsvA_D23-13406-2","7":"59462666","8":"8919399900","9":"4.45","10":"PASS","11":"PASS","12":"WARN","13":"PASS","14":"PASS","15":"FAIL","16":"230926EsvA_D23-13406-2_2_sequence.fastq","17":"CCGCTATT","18":"raw"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>In total, we obtained roughly 550M read pairs, corresponding to roughly 164 Gb of sequencing data. Reads were uniformly 150bp in length. The breakdown of reads was uneven, with the QIAamp kit receiving just over half as many reads – and thus half as many bases – as Zymo quick-RNA.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_1</span> <span class="op">&lt;-</span> <span class="va">qc_tab</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span>, n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">kit</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_bases_1</span> <span class="op">&lt;-</span> <span class="va">n_reads_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">theme_kit</span> <span class="op">&lt;-</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>  aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>  axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">g_reads_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_1</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">read_pair</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span><span class="va">g_bases_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_bases_1</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_bases</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="va">theme_base</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span></span>
<span><span class="va">g_reads_raw</span> <span class="op">+</span> <span class="va">g_bases_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Data quality appears good, with all libraries passing per-base and per-sequence quality checks as well as per-base N content checks. Sequence composition checks either pass or raise a warning; none fail. Duplication levels are low, with FASTQC predicting less than 10% duplicated sequences for all libraries. The only real issue with sequence quality is a high prevalence of adapter sequences, which is unsurprising for raw sequencing data.</p>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>To remove adapters and trim low-quality sequences, I ran the raw data through FASTP<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>:</p>
<pre><code>for p in $(cat prefixes.txt); do echo $p; fastp -i raw/${p}_1.fastq.gz -I raw/${p}_2.fastq.gz -o preproc/${p}_fastp_unmerged_1.fastq.gz -O preproc/${p}_fastp_unmerged_2.fastq.gz --merged_out preproc/${p}_fastp_merged.fastq.gz --failed_out preproc/${p}_fastp_failed.fastq.gz --cut_tail --correction --detect_adapter_for_pe --adapter_fasta adapters.fa --merge --trim_poly_x --thread 16; done</code></pre>
<p>With a single 16-thread process, this took a total of 30 minutes to process all six libraries: 12 minutes for Zymo quick-RNA data, 11 minutes for Zymo quick-DNA/RNA data, and 7 minutes for QIAamp Viral RNA mini kit data. All preprocessed libraries subsequently passed FASTQC’s adapter content checks<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process paired data</span></span>
<span><span class="va">outer_dir_preproc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc/preproc"</span><span class="op">)</span></span>
<span><span class="va">qc_archives_preproc_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_preproc</span>, pattern <span class="op">=</span> <span class="st">"unmerged.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_preproc_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_preproc_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_preproc</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_preproc_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_preproc_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_paired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_fastp_unmerged_\\d.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Import and process unpaired data</span></span>
<span><span class="va">qc_archives_preproc_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_preproc</span>, pattern <span class="op">=</span> <span class="st">"_merged.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_preproc_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_preproc_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_preproc</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_preproc_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_preproc_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_unpaired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_fastp_merged.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Combine</span></span>
<span><span class="va">qc_tab_preproc</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab_preproc_paired</span>, <span class="va">qc_tab_preproc_unpaired</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">sample_metadata</span>, by <span class="op">=</span> <span class="st">"sequencing_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span>,</span>
<span>         <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="st">"preprocessed"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span><span class="op">)</span></span>
<span><span class="va">qc_tab_preproc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["library_id"],"name":[2],"type":["chr"],"align":["left"]},{"label":["kit"],"name":[3],"type":["chr"],"align":["left"]},{"label":["sequencing_replicate"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["read_pair"],"name":[5],"type":["chr"],"align":["left"]},{"label":["sequencing_id"],"name":[6],"type":["chr"],"align":["left"]},{"label":["n_reads"],"name":[7],"type":["int"],"align":["right"]},{"label":["n_bases"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["pc_dup"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["per_base_quality"],"name":[10],"type":["chr"],"align":["left"]},{"label":["per_sequence_quality"],"name":[11],"type":["chr"],"align":["left"]},{"label":["per_base_composition"],"name":[12],"type":["chr"],"align":["left"]},{"label":["per_sequence_composition"],"name":[13],"type":["chr"],"align":["left"]},{"label":["per_base_n_content"],"name":[14],"type":["chr"],"align":["left"]},{"label":["adapter_content"],"name":[15],"type":["chr"],"align":["left"]},{"label":["filename"],"name":[16],"type":["chr"],"align":["left"]},{"label":["index"],"name":[17],"type":["chr"],"align":["left"]},{"label":["stage"],"name":[18],"type":["chr"],"align":["left"]}],"data":[{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"1","5":"1","6":"230926EsvA_D23-13404-1","7":"3741415","8":"400381265","9":"1.79","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-1_fastp_unmerged_1.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"1","5":"2","6":"230926EsvA_D23-13404-1","7":"3741415","8":"480493376","9":"2.30","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-1_fastp_unmerged_2.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"1","5":"NA","6":"230926EsvA_D23-13404-1","7":"110907603","8":"12338278194","9":"9.84","10":"PASS","11":"PASS","12":"WARN","13":"FAIL","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-1_fastp_merged.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"2","5":"1","6":"230926EsvA_D23-13404-2","7":"3760138","8":"402167668","9":"1.86","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-2_fastp_unmerged_1.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"2","5":"2","6":"230926EsvA_D23-13404-2","7":"3760138","8":"483433160","9":"2.31","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-2_fastp_unmerged_2.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"1C","2":"D23-13404","3":"Zymo quick-RNA","4":"2","5":"NA","6":"230926EsvA_D23-13404-2","7":"110666415","8":"12327488678","9":"9.73","10":"PASS","11":"PASS","12":"WARN","13":"FAIL","14":"PASS","15":"PASS","16":"230926EsvA_D23-13404-2_fastp_merged.fastq.gz","17":"CCCAAGTT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"1","5":"1","6":"230926EsvA_D23-13405-1","7":"3608892","8":"341526934","9":"1.19","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-1_fastp_unmerged_1.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"1","5":"2","6":"230926EsvA_D23-13405-1","7":"3608892","8":"446198588","9":"1.88","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-1_fastp_unmerged_2.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"1","5":"NA","6":"230926EsvA_D23-13405-1","7":"94359849","8":"9935360420","9":"8.74","10":"PASS","11":"PASS","12":"WARN","13":"FAIL","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-1_fastp_merged.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"2","5":"1","6":"230926EsvA_D23-13405-2","7":"3623282","8":"343214492","9":"1.26","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-2_fastp_unmerged_1.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"2","5":"2","6":"230926EsvA_D23-13405-2","7":"3623282","8":"448988408","9":"1.95","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-2_fastp_unmerged_2.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"2C","2":"D23-13405","3":"Zymo quick-DNA/RNA","4":"2","5":"NA","6":"230926EsvA_D23-13405-2","7":"94112377","8":"9922134626","9":"8.84","10":"PASS","11":"PASS","12":"WARN","13":"FAIL","14":"PASS","15":"PASS","16":"230926EsvA_D23-13405-2_fastp_merged.fastq.gz","17":"TCTAGGCT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"1","5":"1","6":"230926EsvA_D23-13406-1","7":"2362707","8":"208300642","9":"0.74","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-1_fastp_unmerged_1.fastq.gz","17":"CCGCTATT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"1","5":"2","6":"230926EsvA_D23-13406-1","7":"2362707","8":"283562504","9":"1.34","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-1_fastp_unmerged_2.fastq.gz","17":"CCGCTATT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"1","5":"NA","6":"230926EsvA_D23-13406-1","7":"56899143","8":"5740601194","9":"5.24","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-1_fastp_merged.fastq.gz","17":"CCGCTATT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"2","5":"1","6":"230926EsvA_D23-13406-2","7":"2371835","8":"209489496","9":"0.83","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-2_fastp_unmerged_1.fastq.gz","17":"CCGCTATT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"2","5":"2","6":"230926EsvA_D23-13406-2","7":"2371835","8":"285305707","9":"1.42","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-2_fastp_unmerged_2.fastq.gz","17":"CCGCTATT","18":"preprocessed"},{"1":"6C","2":"D23-13406","3":"QIAamp Viral RNA mini kit","4":"2","5":"NA","6":"230926EsvA_D23-13406-2","7":"56830597","8":"5741593286","9":"5.32","10":"PASS","11":"PASS","12":"WARN","13":"WARN","14":"PASS","15":"PASS","16":"230926EsvA_D23-13406-2_fastp_merged.fastq.gz","17":"CCGCTATT","18":"preprocessed"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Calculate per-stage read and base counts</span></span>
<span><span class="va">qc_tab_2</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab</span>, <span class="va">qc_tab_preproc</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span>, kit <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">kit</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_2</span> <span class="op">&lt;-</span> <span class="va">qc_tab_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">read_pair</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_bases_2</span> <span class="op">&lt;-</span> <span class="va">n_reads_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_reads_2_out</span> <span class="op">&lt;-</span> <span class="va">n_reads_2</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">read_pair</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_reads_preproc</span> <span class="op">&lt;-</span> <span class="va">n_reads_2_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_reads</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_bases_preproc</span> <span class="op">&lt;-</span> <span class="va">n_bases_2</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_bases</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_preproc_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_reads_preproc</span><span class="op">)</span></span>
<span><span class="va">g_reads_preproc_2</span> <span class="op">&lt;-</span> <span class="va">g_reads_preproc</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_reads_preproc_2</span> <span class="op">+</span> <span class="va">g_bases_preproc</span>, <span class="va">g_preproc_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.15</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In total, roughly 103 Gb of sequence data (63.1%) was lost during preprocessing. Per-kit losses ranged from 61.7% to 65.1%. Conversely, very few reads (&lt;1%) were discarded entirely.</p>
<p>I was sufficiently surprised by the magnitude of these losses that I checked the number of reads and bases in several files manually to confirm the FASTQC estimates were accurate – they are.</p>
<p>My current best explanation for the scale of the losses is that the fragments being sequenced are primarily small, such that a large number of bases are lost during collapsing of overlapping read pairs. This is consistent with the small fragment sizes that predominate in our Fragment Analyzer data, and the fact that much less material is lost when FASTP is run without collapsing overlapping read pairs (data not shown).</p>
<p>An upside of the large decrease in sequence information caused by collapsing redundant sequence across read pairs is that downstream steps should run substantially faster as a result.</p>
</section><section id="ribodetection" class="level1"><h1>Ribodetection</h1>
<p>As DNA sequencing reads, I expected these data to be low in ribosomal sequences. To test this, I ran the data through bbduk, using the same reference sequences described <a href="https://data.securebio.org/wills-public-notebook/notebooks/2023-10-13_rrna-removal.html">here</a>:</p>
<pre><code>for p in $(cat prefixes.txt); do echo $p; 
bbduk.sh in=preproc/${p}_fastp_unmerged_1.fastq.gz in2=preproc/${p}_fastp_unmerged_2.fastq.gz ref=../../riboseqs/SILVA_138.1_LSURef_NR99_tax_silva.fasta.gz,../../riboseqs/SILVA_138.1_SSURef_NR99_tax_silva.fasta.gz out=ribo/${p}_bbduk_unmerged_passed_1.fastq.gz out2=ribo/${p}_bbduk_unmerged_passed_2.fastq.gz outm=ribo/${p}_bbduk_unmerged_failed_1.fastq.gz outm2=ribo/${p}_bbduk_unmerged_failed_2.fastq.gz stats=ribo/${p}_bbduk_unmerged_stats.txt k=30 t=30;
bbduk.sh in=preproc/${p}_fastp_merged.fastq.gz ref=../../riboseqs/SILVA_138.1_LSURef_NR99_tax_silva.fasta.gz,../../riboseqs/SILVA_138.1_SSURef_NR99_tax_silva.fasta.gz out=ribo/${p}_bbduk_merged_passed.fastq.gz outm=ribo/${p}_bbduk_merged_failed.fastq.gz stats=ribo/${p}_bbduk_merged_stats.txt k=30 t=30; 
done</code></pre>
<p>In total, running ribodetection on all six libraries in series took 20 minutes: 8 minutes for Zymo quick-RNA data, 7 minutes for Zymo quick-DNA/RNA data, and 5 minutes for QIAamp Viral RNA mini kit data. As expected, the number of ribosomal reads detected was very low: in total, only 3.6M read pairs (0.67%) were removed during ribodetection, with levels for individual input samples ranging from 0.60% to 0.72%.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process paired data</span></span>
<span><span class="va">outer_dir_ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc/ribo"</span><span class="op">)</span></span>
<span><span class="va">qc_archives_ribo_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_ribo</span>, pattern <span class="op">=</span> <span class="st">"unmerged_passed.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_ribo_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_ribo_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_ribo</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_ribo_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_ribo_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_paired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_bbduk_unmerged_passed_\\d.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Import and process unpaired data</span></span>
<span><span class="va">qc_archives_ribo_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_ribo</span>, pattern <span class="op">=</span> <span class="st">"_merged_passed.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_ribo_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_ribo_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_ribo</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_ribo_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_ribo_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_unpaired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_bbduk_merged_passed.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Combine</span></span>
<span><span class="va">qc_tab_ribo</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab_ribo_paired</span>, <span class="va">qc_tab_ribo_unpaired</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">sample_metadata</span>, by <span class="op">=</span> <span class="st">"sequencing_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span>,</span>
<span>         <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="st">"ribodepleted"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate per-stage read and base counts</span></span>
<span><span class="va">qc_tab_3</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab</span>, <span class="va">qc_tab_preproc</span>, <span class="va">qc_tab_ribo</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span>, kit <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">kit</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_3</span> <span class="op">&lt;-</span> <span class="va">qc_tab_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">read_pair</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_bases_3</span> <span class="op">&lt;-</span> <span class="va">n_reads_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_reads_3_out</span> <span class="op">&lt;-</span> <span class="va">n_reads_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">read_pair</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_reads_ribo</span> <span class="op">&lt;-</span> <span class="va">n_reads_3_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_reads</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_bases_ribo</span> <span class="op">&lt;-</span> <span class="va">n_bases_3</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_bases</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_ribo_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_reads_ribo</span><span class="op">)</span></span>
<span><span class="va">g_reads_ribo_2</span> <span class="op">&lt;-</span> <span class="va">g_reads_ribo</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_reads_ribo_2</span> <span class="op">+</span> <span class="va">g_bases_ribo</span>, <span class="va">g_ribo_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.15</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="deduplication" class="level1"><h1>Deduplication</h1>
<p>Based on the initial FASTQC results, I expected duplication levels in the ribodepleted data to be low. To confirm this, I performed deduplication with Clumpify as described <a href="https://data.securebio.org/wills-public-notebook/notebooks/2023-10-19_deduplication.html">here</a><a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>:</p>
<pre><code>for p in $(cat prefixes.txt); do echo $p; 
clumpify.sh in=ribo/${p}_bbduk_unmerged_passed_1.fastq.gz in2=ribo/${p}_bbduk_unmerged_passed_2.fastq.gz out=dedup/${p}_clumpify_unmerged_1.fastq.gz out2=dedup/${p}_clumpify_unmerged_2.fastq.gz reorder dedupe t=30; 
clumpify.sh in=ribo/${p}_bbduk_merged_passed.fastq.gz out=dedup/${p}_clumpify_merged.fastq.gz reorder dedupe t=30; 
done</code></pre>
<p>In total, running deduplication on all six libraries in series took 22 minutes: 10 minutes for Zymo quick-RNA data, 8 minutes for Zymo quick-DNA/RNA data, and 4 minutes for QIAamp Viral RNA mini kit data.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process paired data</span></span>
<span><span class="va">outer_dir_dedup</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc/dedup"</span><span class="op">)</span></span>
<span><span class="va">qc_archives_dedup_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_dedup</span>, pattern <span class="op">=</span> <span class="st">"unmerged.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_dedup_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_dedup_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_dedup</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_dedup_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_dedup_paired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_paired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_clumpify_unmerged_\\d.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Import and process unpaired data</span></span>
<span><span class="va">qc_archives_dedup_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">outer_dir_dedup</span>, pattern <span class="op">=</span> <span class="st">"_merged.*zip"</span><span class="op">)</span></span>
<span><span class="va">qc_data_dedup_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_archives_dedup_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="fu">qc_read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outer_dir_dedup</span>, <span class="va">p</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">qc_tab_dedup_unpaired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">qc_data_dedup_unpaired</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>  <span class="fu">extract_qc_data_unpaired</span><span class="op">(</span><span class="va">x</span>, <span class="st">"_clumpify_merged.fastq.gz"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Combine</span></span>
<span><span class="va">qc_tab_dedup</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab_dedup_paired</span>, <span class="va">qc_tab_dedup_unpaired</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">sample_metadata</span>, by <span class="op">=</span> <span class="st">"sequencing_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span>,</span>
<span>         <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="st">"deduplicated"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">sequencing_replicate</span>, <span class="va">read_pair</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate per-stage read and base counts</span></span>
<span><span class="va">qc_tab_4</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">qc_tab</span>, <span class="va">qc_tab_preproc</span>, <span class="va">qc_tab_ribo</span>, <span class="va">qc_tab_dedup</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span>, kit <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">kit</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_4</span> <span class="op">&lt;-</span> <span class="va">qc_tab_4</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">read_pair</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_bases_4</span> <span class="op">&lt;-</span> <span class="va">n_reads_4</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_bases <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">n_reads_4_out</span> <span class="op">&lt;-</span> <span class="va">n_reads_4</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">read_pair</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_reads_dedup</span> <span class="op">&lt;-</span> <span class="va">n_reads_4_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_reads</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_bases_dedup</span> <span class="op">&lt;-</span> <span class="va">n_bases_4</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_bases</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_dedup_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_reads_dedup</span><span class="op">)</span></span>
<span><span class="va">g_reads_dedup_2</span> <span class="op">&lt;-</span> <span class="va">g_reads_dedup</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_reads_dedup_2</span> <span class="op">+</span> <span class="va">g_bases_dedup</span>, <span class="va">g_dedup_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.15</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>As expected, the number of duplicate reads was low: in total, 9.6M read pairs (1.8%) were removed during deduplication, with levels for individual input samples ranging from 1.4% to 2.1%. After deduplication using these parameters, the detected fraction of duplicates identified by FASTQC roughly halved for all samples:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_dups</span> <span class="op">&lt;-</span> <span class="va">qc_tab_4</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_dup <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span> <span class="op">*</span> <span class="va">pc_dup</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">g_dups</span> <span class="op">&lt;-</span> <span class="va">n_dups</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">pc_dup</span>, fill<span class="op">=</span><span class="va">stage</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Duplicate Sequences (FASTQC)"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">10</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>, axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dups</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="kraken2-initial-assignments" class="level1"><h1>Kraken2 – Initial Assignments</h1>
<p>Following deduplication, I ran the data through Kraken2 for taxonomic assignments, using the 16GB Standard database<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>:</p>
<pre><code>for p in $(cat prefixes.txt); do echo $p; 
kraken2 --db ~/kraken-db/ --use-names --paired --threads 30 --output kraken/${p}_unmerged.output --report kraken/${p}_unmerged.report dedup/${p}_clumpify_unmerged_[12].fastq.gz; 
kraken2 --db ~/kraken-db/ --use-names --threads 30 --output kraken/${p}_merged.output --report kraken/${p}_merged.report dedup/${p}_clumpify_merged.fastq.gz; cat kraken/${p}_unmerged.output kraken/${p}_merged.output &gt; kraken/${p}.output; 
done</code></pre>
<p>In all cases, most of the reads (&gt;91%) were unassigned, and most of the rest (&gt;7%) were bacterial. Assigned archaea, eukaryote, and virus reads all made up a very small fraction of the data for all samples.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">kraken_domains_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"kraken-domains.csv"</span><span class="op">)</span></span>
<span><span class="va">kraken_domains</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">kraken_domains_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">sample_metadata</span>, </span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"library_id"</span>, <span class="st">"sequencing_id"</span>, <span class="st">"sequencing_replicate"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">kraken_domains_gathered</span> <span class="op">&lt;-</span> <span class="va">kraken_domains</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">n_total</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">gather</span><span class="op">(</span><span class="va">domain</span>, <span class="va">n_reads</span>, <span class="op">-</span><span class="va">sample_id</span>, <span class="op">-</span><span class="va">library_id</span>, <span class="op">-</span><span class="va">kit</span>, <span class="op">-</span><span class="va">sequencing_replicate</span>, <span class="op">-</span><span class="va">sequencing_id</span>, <span class="op">-</span><span class="va">index</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>domain <span class="op">=</span> <span class="fu">str_to_sentence</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"n_"</span>, <span class="st">""</span>, <span class="va">domain</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         domain <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">domain</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">domain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_domains</span> <span class="op">&lt;-</span> <span class="va">kraken_domains_gathered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill <span class="op">=</span> <span class="va">domain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span>, name <span class="op">=</span> <span class="st">"Domain"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Deduplicated Reads"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_domains_minor</span> <span class="op">&lt;-</span> <span class="va">kraken_domains_gathered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">domain</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Unassigned"</span>, <span class="st">"Bacteria"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill <span class="op">=</span> <span class="va">domain</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/brewer_pal.html">brewer_pal</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span><span class="op">(</span><span class="fl">8</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                    name <span class="op">=</span> <span class="st">"Domain"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Deduplicated Reads"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.01</span>,<span class="fl">0.002</span><span class="op">)</span>,</span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_domains_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_domains</span><span class="op">)</span></span>
<span><span class="va">g_domains_2</span> <span class="op">&lt;-</span> <span class="va">g_domains</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_domains_2</span> <span class="op">+</span> <span class="va">g_domains_minor</span>, <span class="va">g_domains_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="kraken2-human-infecting-viruses" class="level1"><h1>Kraken2 – Human-Infecting Viruses</h1>
<p>Finally, I calculated the number of reads assigned to human-infecting viruses, using scripts based on the <code>cladecounts</code> and <code>allmatches</code> functions from <a href="https://github.com/naobservatory/mgs-pipeline/blob/main/run.py">the public pipeline</a>. The number of resulting read assignments varied from just over half as many as in the private dashboard, to slightly more than that number:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_counts_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv-counts.csv"</span><span class="op">)</span></span>
<span><span class="va">hv_counts</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">hv_counts_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">sample_metadata</span>, </span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"library_id"</span>, <span class="st">"sequencing_id"</span>, <span class="st">"sequencing_replicate"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">library_id</span>, <span class="va">kit</span>, <span class="va">sequencing_replicate</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_counts_gathered</span> <span class="op">&lt;-</span> <span class="va">hv_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">index</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">gather</span><span class="op">(</span><span class="va">pipeline</span>, <span class="va">n_hv</span>, <span class="op">-</span><span class="va">sample_id</span>, <span class="op">-</span><span class="va">library_id</span>, <span class="op">-</span><span class="va">kit</span>, <span class="op">-</span><span class="va">sequencing_replicate</span>, </span>
<span>         <span class="op">-</span><span class="va">sequencing_id</span>, <span class="op">-</span><span class="va">n_total</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>pipeline <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"hv_"</span>, <span class="st">""</span>, <span class="va">pipeline</span><span class="op">)</span>,</span>
<span>         pipeline <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">pipeline</span><span class="op">)</span>,</span>
<span>         p_hv <span class="op">=</span> <span class="va">n_hv</span><span class="op">/</span><span class="va">n_total</span><span class="op">)</span></span>
<span><span class="va">g_hv_n</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_counts_gathered</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sequencing_id</span>, y<span class="op">=</span><span class="va">n_hv</span>, fill<span class="op">=</span><span class="va">pipeline</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"# Human-infecting virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Pipeline"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_hv_p</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_counts_gathered</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sequencing_id</span>, y<span class="op">=</span><span class="va">p_hv</span>, fill<span class="op">=</span><span class="va">pipeline</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"RA(human-infecting virus reads)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Pipeline"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_hv_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_hv_n</span><span class="op">)</span></span>
<span><span class="va">g_hv_n_2</span> <span class="op">&lt;-</span> <span class="va">g_hv_n</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_hv_n_2</span> <span class="op">+</span> <span class="va">g_hv_p</span>, <span class="va">g_hv_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.15</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Summarizing over kits, the results are as follows:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_counts_summ</span> <span class="op">&lt;-</span> <span class="va">hv_counts_gathered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">kit</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">kit</span>, <span class="va">pipeline</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, n_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_hv</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_hv <span class="op">=</span> <span class="va">n_hv</span><span class="op">/</span><span class="va">n_total</span><span class="op">)</span></span>
<span><span class="va">g_hv_summ_n</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_counts_summ</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">n_hv</span>, fill<span class="op">=</span><span class="va">pipeline</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"# Human-infecting virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Pipeline"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_hv_summ_p</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_counts_summ</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">p_hv</span>, fill<span class="op">=</span><span class="va">pipeline</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"RA(human-infecting virus reads)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Pipeline"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_hv_summ_legend</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_hv_summ_n</span><span class="op">)</span></span>
<span><span class="va">g_hv_summ_n_2</span> <span class="op">&lt;-</span> <span class="va">g_hv_summ_n</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_hv_summ_n_2</span> <span class="op">+</span> <span class="va">g_hv_summ_p</span>, <span class="va">g_hv_summ_legend</span>, nrow <span class="op">=</span> <span class="fl">2</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.15</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2023-10-24_project-runway-initial_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I’m not sure what exactly is giving rise to the discrepancy between the two pipelines. While some degree of disagreement between two different pipelines isn’t surprising, it would be good to have a deeper understanding of what’s causing it, and which pipeline is giving results that are more “believable”. I’ll look more deeply into this in a subsequent notebook entry.</p>


<!-- -->

</section><div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>
<ol>
<li id="fn1"><p>I initially ran FASTP without collapsing read pairs (<code>--merge</code>), as this makes for easier and more elegant downstream processing. However, this led to erroneous overidentification of read taxa during the Kraken2 analysis step by spuriously duplicating overlapping sequences. As such, I grudgingly implemented read-pair collapsing here as in the original pipeline.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>I initially ran FASTP without using pre-specified adapter sequences (<code>–adapter_fasta adapters.fa</code>). In most cases, FASTP successfully auto-detected and removed adapters, resulting in very low adapter prevalence in the preprocessed files. However, adapter trimming failed for two files: <code>230926EsvA_D23-13406-1_fastp_2.fastq.gz</code> and <code>230926EsvA_D23-13406-2_fastp_2.fastq.gz</code>. Digging into the FASTP logs, it looks like it failed to identify an adapter sequence for these files, resulting in inadequate adapter trimming. Rerunning FASTP while explicitly passing the Illumina TruSeq adapter sequences (alongside automated adapter detection) solved this problem in this instance.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>After running this, I realized this is probably an underestimate of the level of duplication, as it’s not counting duplicate reads from the same library sequenced on different lanes. I’ll look into cross-lane duplicates as part of a deeper investigation of duplication levels that I plan to do a bit later.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>In the future, I plan to look at the effect of using different Kraken2 databases on read assignment, but for now I’m sticking with the database used in the main pipeline.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb15" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Initial analysis of Project Runway protocol testing data"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> ""</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2023-10-24</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>On 2023-10-23, we received the first batch of sequencing data from our wastewater protocol optimization work. This notebook contains the output of initial QC and analysis for this data.</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="fu"># Background</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>We carried out concentration and extraction on primary sludge samples from wastewater treatment plant A (WTP-A) using three different nucleic-acid extraction kits:</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Zymo quick-RNA kit</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Zymo quick-DNA/RNA kit</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>QIAamp Viral RNA mini kit</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>We <span class="co">[</span><span class="ot">previously</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2023-09-12_settled-solids-extraction-test.html)</span> looked at total NA yield, qPCR and Fragment Analyzer results for these extractions as well as several other kits. We selected these three kits as the most promising, and sent them for sequencing to further characterize the quality of the extractions. Two replicate extractions per kit were sent for sequencing; one (replicate A) underwent RNA sequencing only while the other (replicate C) underwent both DNA and RNA sequencing. We also sent two samples extracted in a previous experiment (using the QIAamp kit) for RNA sequencing.</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>So far, only the DNA sequencing data is available, corresponding to one replicate per kit, or three samples total.</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>The raw data from the BMC consists of sixteen FASTQ files, corresponding to paired-end data from eight sequencing libraries. This is rather more than the three libraries we were expecting; digging into this, each submitted sample seems to have been sequenced twice, and we also have two pairs of FASTQ files corresponding to "unmapped barcodes" (<span class="in">`231016_78783_6418E_L1_{1,2}_sequence_unmapped_barcodes.fastq`</span> and <span class="in">`231016_78783_6418E_L2_{1,2}_sequence_unmapped_barcodes.fastq`</span>).</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>Given that these have the same sequencing date and other annotations, and that (according to communication from the BMC) each submitted library has the same index barcode in both cases, my guess is that each sample was sequenced on both lanes of the AVITI flow cell. However, it would be useful to check with BMC and confirm this.</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>The data from the BMC helpfully comes with FASTQC files attached, which we can use to look at the quality of the raw data.</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2023-10-24_pr-dna"</span></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>outer_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc/raw"</span>)</span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>qc_archives <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir, <span class="at">pattern =</span> <span class="st">"*.zip"</span>)</span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>qc_data <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample mapping</span></span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../data/2023-10-24_pr-dna/sample-metadata.csv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute approximate number of sequence bases from sequence length distribution table</span></span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>compute_n_bases <span class="ot">&lt;-</span> <span class="cf">function</span>(sld){</span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>  min_lengths <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"-.*"</span>, <span class="st">""</span>, sld<span class="sc">$</span>Length) <span class="sc">%&gt;%</span> as.numeric</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>  max_lengths <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*-"</span>, <span class="st">""</span>, sld<span class="sc">$</span>Length) <span class="sc">%&gt;%</span> as.numeric</span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>  mean_lengths <span class="ot">&lt;-</span> (min_lengths <span class="sc">+</span> max_lengths)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>  sld2 <span class="ot">&lt;-</span> <span class="fu">mutate</span>(sld, <span class="at">Length_mean =</span> mean_lengths)</span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>  n_bases <span class="ot">&lt;-</span> <span class="fu">sum</span>(sld2<span class="sc">$</span>Count <span class="sc">*</span> sld2<span class="sc">$</span>Length_mean)</span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(n_bases)</span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a><span class="co"># FASTQC processing functions</span></span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>extract_qc_data_base <span class="ot">&lt;-</span> <span class="cf">function</span>(qc){</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>basic_statistics, Measure <span class="sc">==</span> <span class="st">"Filename"</span>)<span class="sc">$</span>Value</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>  n_reads <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>basic_statistics, Measure <span class="sc">==</span> <span class="st">"Total Sequences"</span>)<span class="sc">$</span>Value <span class="sc">%&gt;%</span> as.integer</span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>  n_bases <span class="ot">&lt;-</span> <span class="fu">compute_n_bases</span>(qc<span class="sc">$</span>sequence_length_distribution)</span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a>  pc_dup <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">-</span>qc<span class="sc">$</span>total_deduplicated_percentage</span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>  per_base_quality <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Per base sequence quality"</span>)<span class="sc">$</span>status</span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a>  per_sequence_quality <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Per sequence quality scores"</span>)<span class="sc">$</span>status</span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a>  per_base_composition <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Per base sequence content"</span>)<span class="sc">$</span>status</span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a>  per_sequence_composition <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Per sequence GC content"</span>)<span class="sc">$</span>status</span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a>  per_base_n_content <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Per base N content"</span>)<span class="sc">$</span>status</span>
<span id="cb15-84"><a href="#cb15-84" aria-hidden="true" tabindex="-1"></a>  adapter_content <span class="ot">&lt;-</span> <span class="fu">filter</span>(qc<span class="sc">$</span>summary, module <span class="sc">==</span> <span class="st">"Adapter Content"</span>)<span class="sc">$</span>status</span>
<span id="cb15-85"><a href="#cb15-85" aria-hidden="true" tabindex="-1"></a>  tab_out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">n_reads =</span> n_reads,</span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n_bases =</span> n_bases,</span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pc_dup =</span> pc_dup,</span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a>                    <span class="at">per_base_quality =</span> per_base_quality,</span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a>                    <span class="at">per_sequence_quality =</span> per_sequence_quality,</span>
<span id="cb15-90"><a href="#cb15-90" aria-hidden="true" tabindex="-1"></a>                    <span class="at">per_base_composition =</span> per_base_composition,</span>
<span id="cb15-91"><a href="#cb15-91" aria-hidden="true" tabindex="-1"></a>                    <span class="at">per_sequence_composition =</span> per_sequence_composition,</span>
<span id="cb15-92"><a href="#cb15-92" aria-hidden="true" tabindex="-1"></a>                    <span class="at">per_base_n_content =</span> per_base_n_content,</span>
<span id="cb15-93"><a href="#cb15-93" aria-hidden="true" tabindex="-1"></a>                    <span class="at">adapter_content =</span> adapter_content,</span>
<span id="cb15-94"><a href="#cb15-94" aria-hidden="true" tabindex="-1"></a>                    <span class="at">filename =</span> filename)</span>
<span id="cb15-95"><a href="#cb15-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tab_out)</span>
<span id="cb15-96"><a href="#cb15-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-97"><a href="#cb15-97" aria-hidden="true" tabindex="-1"></a>extract_qc_data_paired <span class="ot">&lt;-</span> <span class="cf">function</span>(qc, id_pattern, <span class="at">pair_pattern =</span> <span class="st">".*_(</span><span class="sc">\\</span><span class="st">d)[_.].*"</span>){</span>
<span id="cb15-98"><a href="#cb15-98" aria-hidden="true" tabindex="-1"></a>  base <span class="ot">&lt;-</span> <span class="fu">extract_qc_data_base</span>(qc)</span>
<span id="cb15-99"><a href="#cb15-99" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> base<span class="sc">$</span>filename[<span class="dv">1</span>]</span>
<span id="cb15-100"><a href="#cb15-100" aria-hidden="true" tabindex="-1"></a>  prefix <span class="ot">&lt;-</span> <span class="fu">sub</span>(id_pattern, <span class="st">""</span>, filename)</span>
<span id="cb15-101"><a href="#cb15-101" aria-hidden="true" tabindex="-1"></a>  read_pair <span class="ot">&lt;-</span> <span class="fu">sub</span>(pair_pattern, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, filename)</span>
<span id="cb15-102"><a href="#cb15-102" aria-hidden="true" tabindex="-1"></a>  tab_out <span class="ot">&lt;-</span> base <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sequencing_id =</span> prefix, <span class="at">read_pair =</span> read_pair) <span class="sc">%&gt;%</span></span>
<span id="cb15-103"><a href="#cb15-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sequencing_id, read_pair, <span class="fu">everything</span>())</span>
<span id="cb15-104"><a href="#cb15-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tab_out)</span>
<span id="cb15-105"><a href="#cb15-105" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-106"><a href="#cb15-106" aria-hidden="true" tabindex="-1"></a>extract_qc_data_unpaired <span class="ot">&lt;-</span> <span class="cf">function</span>(qc, id_pattern){</span>
<span id="cb15-107"><a href="#cb15-107" aria-hidden="true" tabindex="-1"></a>  base <span class="ot">&lt;-</span> <span class="fu">extract_qc_data_base</span>(qc)</span>
<span id="cb15-108"><a href="#cb15-108" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> base<span class="sc">$</span>filename[<span class="dv">1</span>]</span>
<span id="cb15-109"><a href="#cb15-109" aria-hidden="true" tabindex="-1"></a>  prefix <span class="ot">&lt;-</span> <span class="fu">sub</span>(id_pattern, <span class="st">""</span>, filename)</span>
<span id="cb15-110"><a href="#cb15-110" aria-hidden="true" tabindex="-1"></a>  tab_out <span class="ot">&lt;-</span> base <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sequencing_id =</span> prefix) <span class="sc">%&gt;%</span></span>
<span id="cb15-111"><a href="#cb15-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sequencing_id, <span class="fu">everything</span>())</span>
<span id="cb15-112"><a href="#cb15-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tab_out)</span>
<span id="cb15-113"><a href="#cb15-113" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-114"><a href="#cb15-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-115"><a href="#cb15-115" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate and show sample table</span></span>
<span id="cb15-116"><a href="#cb15-116" aria-hidden="true" tabindex="-1"></a>pattern_raw <span class="ot">&lt;-</span> <span class="st">"_</span><span class="sc">\\</span><span class="st">d_sequence.fastq.*$"</span></span>
<span id="cb15-117"><a href="#cb15-117" aria-hidden="true" tabindex="-1"></a>qc_tab <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data, <span class="cf">function</span>(x) <span class="fu">extract_qc_data_paired</span>(x, pattern_raw)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-118"><a href="#cb15-118" aria-hidden="true" tabindex="-1"></a>  bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb15-119"><a href="#cb15-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"sequencing_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-120"><a href="#cb15-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, read_pair, <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb15-121"><a href="#cb15-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="st">"raw"</span>)</span>
<span id="cb15-122"><a href="#cb15-122" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(qc_data)  </span>
<span id="cb15-123"><a href="#cb15-123" aria-hidden="true" tabindex="-1"></a>qc_tab</span>
<span id="cb15-124"><a href="#cb15-124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-125"><a href="#cb15-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-126"><a href="#cb15-126" aria-hidden="true" tabindex="-1"></a>In total, we obtained roughly 550M read pairs, corresponding to roughly 164 Gb of sequencing data. Reads were uniformly 150bp in length. The breakdown of reads was uneven, with the QIAamp kit receiving just over half as many reads -- and thus half as many bases -- as Zymo quick-RNA.</span>
<span id="cb15-127"><a href="#cb15-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-130"><a href="#cb15-130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-131"><a href="#cb15-131" aria-hidden="true" tabindex="-1"></a>n_reads_1 <span class="ot">&lt;-</span> qc_tab <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, read_pair) <span class="sc">%&gt;%</span> </span>
<span id="cb15-132"><a href="#cb15-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(<span class="fu">as.integer</span>(n_reads)), <span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-133"><a href="#cb15-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kit =</span> <span class="fu">fct_inorder</span>(kit))</span>
<span id="cb15-134"><a href="#cb15-134" aria-hidden="true" tabindex="-1"></a>n_bases_1 <span class="ot">&lt;-</span> n_reads_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-135"><a href="#cb15-135" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb15-136"><a href="#cb15-136" aria-hidden="true" tabindex="-1"></a>  <span class="at">aspect.ratio =</span> <span class="dv">1</span>,</span>
<span id="cb15-137"><a href="#cb15-137" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb15-138"><a href="#cb15-138" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb15-139"><a href="#cb15-139" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-140"><a href="#cb15-140" aria-hidden="true" tabindex="-1"></a>g_reads_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(read_pair <span class="sc">==</span> <span class="dv">1</span>), <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_reads)) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span> theme_kit</span>
<span id="cb15-141"><a href="#cb15-141" aria-hidden="true" tabindex="-1"></a>g_bases_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_bases_1, <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_bases)) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb15-142"><a href="#cb15-142" aria-hidden="true" tabindex="-1"></a>    theme_base <span class="sc">+</span> theme_kit</span>
<span id="cb15-143"><a href="#cb15-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-144"><a href="#cb15-144" aria-hidden="true" tabindex="-1"></a>g_reads_raw <span class="sc">+</span> g_bases_raw</span>
<span id="cb15-145"><a href="#cb15-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-146"><a href="#cb15-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-147"><a href="#cb15-147" aria-hidden="true" tabindex="-1"></a>Data quality appears good, with all libraries passing per-base and per-sequence quality checks as well as per-base N content checks. Sequence composition checks either pass or raise a warning; none fail. Duplication levels are low, with FASTQC predicting less than 10% duplicated sequences for all libraries. The only real issue with sequence quality is a high prevalence of adapter sequences, which is unsurprising for raw sequencing data.</span>
<span id="cb15-148"><a href="#cb15-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-149"><a href="#cb15-149" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb15-150"><a href="#cb15-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-151"><a href="#cb15-151" aria-hidden="true" tabindex="-1"></a>To remove adapters and trim low-quality sequences, I ran the raw data through FASTP<span class="ot">[^1]</span>:</span>
<span id="cb15-152"><a href="#cb15-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-153"><a href="#cb15-153" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>I initially ran FASTP without collapsing read pairs (<span class="in">`--merge`</span>), as this makes for easier and more elegant downstream processing. However, this led to erroneous overidentification of read taxa during the Kraken2 analysis step by spuriously duplicating overlapping sequences. As such, I grudgingly implemented read-pair collapsing here as in the original pipeline.</span>
<span id="cb15-154"><a href="#cb15-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-155"><a href="#cb15-155" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb15-156"><a href="#cb15-156" aria-hidden="true" tabindex="-1"></a><span class="in">for p in $(cat prefixes.txt); do echo $p; fastp -i raw/${p}_1.fastq.gz -I raw/${p}_2.fastq.gz -o preproc/${p}_fastp_unmerged_1.fastq.gz -O preproc/${p}_fastp_unmerged_2.fastq.gz --merged_out preproc/${p}_fastp_merged.fastq.gz --failed_out preproc/${p}_fastp_failed.fastq.gz --cut_tail --correction --detect_adapter_for_pe --adapter_fasta adapters.fa --merge --trim_poly_x --thread 16; done</span></span>
<span id="cb15-157"><a href="#cb15-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-158"><a href="#cb15-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-159"><a href="#cb15-159" aria-hidden="true" tabindex="-1"></a>With a single 16-thread process, this took a total of 30 minutes to process all six libraries: 12 minutes for Zymo quick-RNA data, 11 minutes for Zymo quick-DNA/RNA data, and 7 minutes for QIAamp Viral RNA mini kit data. All preprocessed libraries subsequently passed FASTQC's adapter content checks<span class="ot">[^2]</span>.</span>
<span id="cb15-160"><a href="#cb15-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-161"><a href="#cb15-161" aria-hidden="true" tabindex="-1"></a><span class="ot">[^2]: </span>I initially ran FASTP without using pre-specified adapter sequences (<span class="in">`–adapter_fasta adapters.fa`</span>). In most cases, FASTP successfully auto-detected and removed adapters, resulting in very low adapter prevalence in the preprocessed files. However, adapter trimming failed for two files: <span class="in">`230926EsvA_D23-13406-1_fastp_2.fastq.gz`</span> and <span class="in">`230926EsvA_D23-13406-2_fastp_2.fastq.gz`</span>. Digging into the FASTP logs, it looks like it failed to identify an adapter sequence for these files, resulting in inadequate adapter trimming. Rerunning FASTP while explicitly passing the Illumina TruSeq adapter sequences (alongside automated adapter detection) solved this problem in this instance.</span>
<span id="cb15-162"><a href="#cb15-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-165"><a href="#cb15-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-166"><a href="#cb15-166" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process paired data</span></span>
<span id="cb15-167"><a href="#cb15-167" aria-hidden="true" tabindex="-1"></a>outer_dir_preproc <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc/preproc"</span>)</span>
<span id="cb15-168"><a href="#cb15-168" aria-hidden="true" tabindex="-1"></a>qc_archives_preproc_paired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_preproc, <span class="at">pattern =</span> <span class="st">"unmerged.*zip"</span>)</span>
<span id="cb15-169"><a href="#cb15-169" aria-hidden="true" tabindex="-1"></a>qc_data_preproc_paired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_preproc_paired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_preproc, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-170"><a href="#cb15-170" aria-hidden="true" tabindex="-1"></a>qc_tab_preproc_paired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_preproc_paired, <span class="cf">function</span>(x)</span>
<span id="cb15-171"><a href="#cb15-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_paired</span>(x, <span class="st">"_fastp_unmerged_</span><span class="sc">\\</span><span class="st">d.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-172"><a href="#cb15-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-173"><a href="#cb15-173" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process unpaired data</span></span>
<span id="cb15-174"><a href="#cb15-174" aria-hidden="true" tabindex="-1"></a>qc_archives_preproc_unpaired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_preproc, <span class="at">pattern =</span> <span class="st">"_merged.*zip"</span>)</span>
<span id="cb15-175"><a href="#cb15-175" aria-hidden="true" tabindex="-1"></a>qc_data_preproc_unpaired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_preproc_unpaired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_preproc, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-176"><a href="#cb15-176" aria-hidden="true" tabindex="-1"></a>qc_tab_preproc_unpaired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_preproc_unpaired, <span class="cf">function</span>(x)</span>
<span id="cb15-177"><a href="#cb15-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_unpaired</span>(x, <span class="st">"_fastp_merged.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-178"><a href="#cb15-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-179"><a href="#cb15-179" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb15-180"><a href="#cb15-180" aria-hidden="true" tabindex="-1"></a>qc_tab_preproc <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab_preproc_paired, qc_tab_preproc_unpaired) <span class="sc">%&gt;%</span></span>
<span id="cb15-181"><a href="#cb15-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"sequencing_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-182"><a href="#cb15-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, read_pair,</span>
<span id="cb15-183"><a href="#cb15-183" aria-hidden="true" tabindex="-1"></a>         <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb15-184"><a href="#cb15-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="st">"preprocessed"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-185"><a href="#cb15-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample_id, sequencing_replicate, read_pair)</span>
<span id="cb15-186"><a href="#cb15-186" aria-hidden="true" tabindex="-1"></a>qc_tab_preproc</span>
<span id="cb15-187"><a href="#cb15-187" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-188"><a href="#cb15-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-191"><a href="#cb15-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-192"><a href="#cb15-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate per-stage read and base counts</span></span>
<span id="cb15-193"><a href="#cb15-193" aria-hidden="true" tabindex="-1"></a>qc_tab_2 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab, qc_tab_preproc) <span class="sc">%&gt;%</span></span>
<span id="cb15-194"><a href="#cb15-194" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">kit =</span> <span class="fu">fct_inorder</span>(kit))</span>
<span id="cb15-195"><a href="#cb15-195" aria-hidden="true" tabindex="-1"></a>n_reads_2 <span class="ot">&lt;-</span> qc_tab_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, read_pair, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-196"><a href="#cb15-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(<span class="fu">as.integer</span>(n_reads)),</span>
<span id="cb15-197"><a href="#cb15-197" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-198"><a href="#cb15-198" aria-hidden="true" tabindex="-1"></a>n_bases_2 <span class="ot">&lt;-</span> n_reads_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-199"><a href="#cb15-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-200"><a href="#cb15-200" aria-hidden="true" tabindex="-1"></a>n_reads_2_out <span class="ot">&lt;-</span> n_reads_2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(read_pair <span class="sc">==</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">is.na</span>(read_pair)) <span class="sc">%&gt;%</span></span>
<span id="cb15-201"><a href="#cb15-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-202"><a href="#cb15-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-203"><a href="#cb15-203" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb15-204"><a href="#cb15-204" aria-hidden="true" tabindex="-1"></a>g_reads_preproc <span class="ot">&lt;-</span> n_reads_2_out <span class="sc">%&gt;%</span></span>
<span id="cb15-205"><a href="#cb15-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_reads, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-206"><a href="#cb15-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-207"><a href="#cb15-207" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-208"><a href="#cb15-208" aria-hidden="true" tabindex="-1"></a>g_bases_preproc <span class="ot">&lt;-</span> n_bases_2 <span class="sc">%&gt;%</span></span>
<span id="cb15-209"><a href="#cb15-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_bases, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-210"><a href="#cb15-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-211"><a href="#cb15-211" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-212"><a href="#cb15-212" aria-hidden="true" tabindex="-1"></a>g_preproc_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_reads_preproc)</span>
<span id="cb15-213"><a href="#cb15-213" aria-hidden="true" tabindex="-1"></a>g_reads_preproc_2 <span class="ot">&lt;-</span> g_reads_preproc <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-214"><a href="#cb15-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-215"><a href="#cb15-215" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_reads_preproc_2 <span class="sc">+</span> g_bases_preproc, g_preproc_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.15</span>))</span>
<span id="cb15-216"><a href="#cb15-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-217"><a href="#cb15-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-218"><a href="#cb15-218" aria-hidden="true" tabindex="-1"></a>In total, roughly 103 Gb of sequence data (63.1%) was lost during preprocessing. Per-kit losses ranged from 61.7% to 65.1%. Conversely, very few reads (<span class="sc">\&lt;</span>1%) were discarded entirely.</span>
<span id="cb15-219"><a href="#cb15-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-220"><a href="#cb15-220" aria-hidden="true" tabindex="-1"></a>I was sufficiently surprised by the magnitude of these losses that I checked the number of reads and bases in several files manually to confirm the FASTQC estimates were accurate -- they are.</span>
<span id="cb15-221"><a href="#cb15-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-222"><a href="#cb15-222" aria-hidden="true" tabindex="-1"></a>My current best explanation for the scale of the losses is that the fragments being sequenced are primarily small, such that a large number of bases are lost during collapsing of overlapping read pairs. This is consistent with the small fragment sizes that predominate in our Fragment Analyzer data, and the fact that much less material is lost when FASTP is run without collapsing overlapping read pairs (data not shown).</span>
<span id="cb15-223"><a href="#cb15-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-224"><a href="#cb15-224" aria-hidden="true" tabindex="-1"></a>An upside of the large decrease in sequence information caused by collapsing redundant sequence across read pairs is that downstream steps should run substantially faster as a result.</span>
<span id="cb15-225"><a href="#cb15-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-226"><a href="#cb15-226" aria-hidden="true" tabindex="-1"></a><span class="fu"># Ribodetection</span></span>
<span id="cb15-227"><a href="#cb15-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-228"><a href="#cb15-228" aria-hidden="true" tabindex="-1"></a>As DNA sequencing reads, I expected these data to be low in ribosomal sequences. To test this, I ran the data through bbduk, using the same reference sequences described <span class="co">[</span><span class="ot">here</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2023-10-13_rrna-removal.html)</span>:</span>
<span id="cb15-229"><a href="#cb15-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-230"><a href="#cb15-230" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb15-231"><a href="#cb15-231" aria-hidden="true" tabindex="-1"></a><span class="in">for p in $(cat prefixes.txt); do echo $p; </span></span>
<span id="cb15-232"><a href="#cb15-232" aria-hidden="true" tabindex="-1"></a><span class="in">bbduk.sh in=preproc/${p}_fastp_unmerged_1.fastq.gz in2=preproc/${p}_fastp_unmerged_2.fastq.gz ref=../../riboseqs/SILVA_138.1_LSURef_NR99_tax_silva.fasta.gz,../../riboseqs/SILVA_138.1_SSURef_NR99_tax_silva.fasta.gz out=ribo/${p}_bbduk_unmerged_passed_1.fastq.gz out2=ribo/${p}_bbduk_unmerged_passed_2.fastq.gz outm=ribo/${p}_bbduk_unmerged_failed_1.fastq.gz outm2=ribo/${p}_bbduk_unmerged_failed_2.fastq.gz stats=ribo/${p}_bbduk_unmerged_stats.txt k=30 t=30;</span></span>
<span id="cb15-233"><a href="#cb15-233" aria-hidden="true" tabindex="-1"></a><span class="in">bbduk.sh in=preproc/${p}_fastp_merged.fastq.gz ref=../../riboseqs/SILVA_138.1_LSURef_NR99_tax_silva.fasta.gz,../../riboseqs/SILVA_138.1_SSURef_NR99_tax_silva.fasta.gz out=ribo/${p}_bbduk_merged_passed.fastq.gz outm=ribo/${p}_bbduk_merged_failed.fastq.gz stats=ribo/${p}_bbduk_merged_stats.txt k=30 t=30; </span></span>
<span id="cb15-234"><a href="#cb15-234" aria-hidden="true" tabindex="-1"></a><span class="in">done</span></span>
<span id="cb15-235"><a href="#cb15-235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-236"><a href="#cb15-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-237"><a href="#cb15-237" aria-hidden="true" tabindex="-1"></a>In total, running ribodetection on all six libraries in series took 20 minutes: 8 minutes for Zymo quick-RNA data, 7 minutes for Zymo quick-DNA/RNA data, and 5 minutes for QIAamp Viral RNA mini kit data. As expected, the number of ribosomal reads detected was very low: in total, only 3.6M read pairs (0.67%) were removed during ribodetection, with levels for individual input samples ranging from 0.60% to 0.72%.</span>
<span id="cb15-238"><a href="#cb15-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-241"><a href="#cb15-241" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-242"><a href="#cb15-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process paired data</span></span>
<span id="cb15-243"><a href="#cb15-243" aria-hidden="true" tabindex="-1"></a>outer_dir_ribo <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc/ribo"</span>)</span>
<span id="cb15-244"><a href="#cb15-244" aria-hidden="true" tabindex="-1"></a>qc_archives_ribo_paired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_ribo, <span class="at">pattern =</span> <span class="st">"unmerged_passed.*zip"</span>)</span>
<span id="cb15-245"><a href="#cb15-245" aria-hidden="true" tabindex="-1"></a>qc_data_ribo_paired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_ribo_paired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_ribo, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-246"><a href="#cb15-246" aria-hidden="true" tabindex="-1"></a>qc_tab_ribo_paired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_ribo_paired, <span class="cf">function</span>(x)</span>
<span id="cb15-247"><a href="#cb15-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_paired</span>(x, <span class="st">"_bbduk_unmerged_passed_</span><span class="sc">\\</span><span class="st">d.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-248"><a href="#cb15-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-249"><a href="#cb15-249" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process unpaired data</span></span>
<span id="cb15-250"><a href="#cb15-250" aria-hidden="true" tabindex="-1"></a>qc_archives_ribo_unpaired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_ribo, <span class="at">pattern =</span> <span class="st">"_merged_passed.*zip"</span>)</span>
<span id="cb15-251"><a href="#cb15-251" aria-hidden="true" tabindex="-1"></a>qc_data_ribo_unpaired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_ribo_unpaired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_ribo, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-252"><a href="#cb15-252" aria-hidden="true" tabindex="-1"></a>qc_tab_ribo_unpaired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_ribo_unpaired, <span class="cf">function</span>(x)</span>
<span id="cb15-253"><a href="#cb15-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_unpaired</span>(x, <span class="st">"_bbduk_merged_passed.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-254"><a href="#cb15-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-255"><a href="#cb15-255" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb15-256"><a href="#cb15-256" aria-hidden="true" tabindex="-1"></a>qc_tab_ribo <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab_ribo_paired, qc_tab_ribo_unpaired) <span class="sc">%&gt;%</span></span>
<span id="cb15-257"><a href="#cb15-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"sequencing_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-258"><a href="#cb15-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, read_pair,</span>
<span id="cb15-259"><a href="#cb15-259" aria-hidden="true" tabindex="-1"></a>         <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb15-260"><a href="#cb15-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="st">"ribodepleted"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-261"><a href="#cb15-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample_id, sequencing_replicate, read_pair)</span>
<span id="cb15-262"><a href="#cb15-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-263"><a href="#cb15-263" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate per-stage read and base counts</span></span>
<span id="cb15-264"><a href="#cb15-264" aria-hidden="true" tabindex="-1"></a>qc_tab_3 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab, qc_tab_preproc, qc_tab_ribo) <span class="sc">%&gt;%</span></span>
<span id="cb15-265"><a href="#cb15-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">kit =</span> <span class="fu">fct_inorder</span>(kit))</span>
<span id="cb15-266"><a href="#cb15-266" aria-hidden="true" tabindex="-1"></a>n_reads_3 <span class="ot">&lt;-</span> qc_tab_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, read_pair, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-267"><a href="#cb15-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(<span class="fu">as.integer</span>(n_reads)),</span>
<span id="cb15-268"><a href="#cb15-268" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-269"><a href="#cb15-269" aria-hidden="true" tabindex="-1"></a>n_bases_3 <span class="ot">&lt;-</span> n_reads_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-270"><a href="#cb15-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-271"><a href="#cb15-271" aria-hidden="true" tabindex="-1"></a>n_reads_3_out <span class="ot">&lt;-</span> n_reads_3 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(read_pair <span class="sc">==</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">is.na</span>(read_pair)) <span class="sc">%&gt;%</span></span>
<span id="cb15-272"><a href="#cb15-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-273"><a href="#cb15-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-274"><a href="#cb15-274" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb15-275"><a href="#cb15-275" aria-hidden="true" tabindex="-1"></a>g_reads_ribo <span class="ot">&lt;-</span> n_reads_3_out <span class="sc">%&gt;%</span></span>
<span id="cb15-276"><a href="#cb15-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_reads, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-277"><a href="#cb15-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-278"><a href="#cb15-278" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-279"><a href="#cb15-279" aria-hidden="true" tabindex="-1"></a>g_bases_ribo <span class="ot">&lt;-</span> n_bases_3 <span class="sc">%&gt;%</span></span>
<span id="cb15-280"><a href="#cb15-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_bases, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-281"><a href="#cb15-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-282"><a href="#cb15-282" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-283"><a href="#cb15-283" aria-hidden="true" tabindex="-1"></a>g_ribo_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_reads_ribo)</span>
<span id="cb15-284"><a href="#cb15-284" aria-hidden="true" tabindex="-1"></a>g_reads_ribo_2 <span class="ot">&lt;-</span> g_reads_ribo <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-285"><a href="#cb15-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-286"><a href="#cb15-286" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_reads_ribo_2 <span class="sc">+</span> g_bases_ribo, g_ribo_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.15</span>))</span>
<span id="cb15-287"><a href="#cb15-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-288"><a href="#cb15-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-289"><a href="#cb15-289" aria-hidden="true" tabindex="-1"></a><span class="fu"># Deduplication</span></span>
<span id="cb15-290"><a href="#cb15-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-291"><a href="#cb15-291" aria-hidden="true" tabindex="-1"></a>Based on the initial FASTQC results, I expected duplication levels in the ribodepleted data to be low. To confirm this, I performed deduplication with Clumpify as described <span class="co">[</span><span class="ot">here</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2023-10-19_deduplication.html)</span><span class="ot">[^3]</span>:</span>
<span id="cb15-292"><a href="#cb15-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-293"><a href="#cb15-293" aria-hidden="true" tabindex="-1"></a><span class="ot">[^3]: </span>After running this, I realized this is probably an underestimate of the level of duplication, as it's not counting duplicate reads from the same library sequenced on different lanes. I'll look into cross-lane duplicates as part of a deeper investigation of duplication levels that I plan to do a bit later.</span>
<span id="cb15-294"><a href="#cb15-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-295"><a href="#cb15-295" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb15-296"><a href="#cb15-296" aria-hidden="true" tabindex="-1"></a><span class="in">for p in $(cat prefixes.txt); do echo $p; </span></span>
<span id="cb15-297"><a href="#cb15-297" aria-hidden="true" tabindex="-1"></a><span class="in">clumpify.sh in=ribo/${p}_bbduk_unmerged_passed_1.fastq.gz in2=ribo/${p}_bbduk_unmerged_passed_2.fastq.gz out=dedup/${p}_clumpify_unmerged_1.fastq.gz out2=dedup/${p}_clumpify_unmerged_2.fastq.gz reorder dedupe t=30; </span></span>
<span id="cb15-298"><a href="#cb15-298" aria-hidden="true" tabindex="-1"></a><span class="in">clumpify.sh in=ribo/${p}_bbduk_merged_passed.fastq.gz out=dedup/${p}_clumpify_merged.fastq.gz reorder dedupe t=30; </span></span>
<span id="cb15-299"><a href="#cb15-299" aria-hidden="true" tabindex="-1"></a><span class="in">done</span></span>
<span id="cb15-300"><a href="#cb15-300" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-301"><a href="#cb15-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-302"><a href="#cb15-302" aria-hidden="true" tabindex="-1"></a>In total, running deduplication on all six libraries in series took 22 minutes: 10 minutes for Zymo quick-RNA data, 8 minutes for Zymo quick-DNA/RNA data, and 4 minutes for QIAamp Viral RNA mini kit data.</span>
<span id="cb15-303"><a href="#cb15-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-306"><a href="#cb15-306" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-307"><a href="#cb15-307" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process paired data</span></span>
<span id="cb15-308"><a href="#cb15-308" aria-hidden="true" tabindex="-1"></a>outer_dir_dedup <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc/dedup"</span>)</span>
<span id="cb15-309"><a href="#cb15-309" aria-hidden="true" tabindex="-1"></a>qc_archives_dedup_paired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_dedup, <span class="at">pattern =</span> <span class="st">"unmerged.*zip"</span>)</span>
<span id="cb15-310"><a href="#cb15-310" aria-hidden="true" tabindex="-1"></a>qc_data_dedup_paired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_dedup_paired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_dedup, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-311"><a href="#cb15-311" aria-hidden="true" tabindex="-1"></a>qc_tab_dedup_paired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_dedup_paired, <span class="cf">function</span>(x)</span>
<span id="cb15-312"><a href="#cb15-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_paired</span>(x, <span class="st">"_clumpify_unmerged_</span><span class="sc">\\</span><span class="st">d.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-313"><a href="#cb15-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-314"><a href="#cb15-314" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process unpaired data</span></span>
<span id="cb15-315"><a href="#cb15-315" aria-hidden="true" tabindex="-1"></a>qc_archives_dedup_unpaired <span class="ot">&lt;-</span> <span class="fu">list.files</span>(outer_dir_dedup, <span class="at">pattern =</span> <span class="st">"_merged.*zip"</span>)</span>
<span id="cb15-316"><a href="#cb15-316" aria-hidden="true" tabindex="-1"></a>qc_data_dedup_unpaired <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(qc_archives_dedup_unpaired, <span class="cf">function</span>(p) <span class="fu">qc_read</span>(<span class="fu">file.path</span>(outer_dir_dedup, p), <span class="at">verbose =</span> <span class="cn">FALSE</span>)))</span>
<span id="cb15-317"><a href="#cb15-317" aria-hidden="true" tabindex="-1"></a>qc_tab_dedup_unpaired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(qc_data_dedup_unpaired, <span class="cf">function</span>(x)</span>
<span id="cb15-318"><a href="#cb15-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract_qc_data_unpaired</span>(x, <span class="st">"_clumpify_merged.fastq.gz"</span>)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb15-319"><a href="#cb15-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-320"><a href="#cb15-320" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb15-321"><a href="#cb15-321" aria-hidden="true" tabindex="-1"></a>qc_tab_dedup <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab_dedup_paired, qc_tab_dedup_unpaired) <span class="sc">%&gt;%</span></span>
<span id="cb15-322"><a href="#cb15-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"sequencing_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-323"><a href="#cb15-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, read_pair,</span>
<span id="cb15-324"><a href="#cb15-324" aria-hidden="true" tabindex="-1"></a>         <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb15-325"><a href="#cb15-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="st">"deduplicated"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-326"><a href="#cb15-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample_id, sequencing_replicate, read_pair)</span>
<span id="cb15-327"><a href="#cb15-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-328"><a href="#cb15-328" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate per-stage read and base counts</span></span>
<span id="cb15-329"><a href="#cb15-329" aria-hidden="true" tabindex="-1"></a>qc_tab_4 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(qc_tab, qc_tab_preproc, qc_tab_ribo, qc_tab_dedup) <span class="sc">%&gt;%</span></span>
<span id="cb15-330"><a href="#cb15-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">fct_inorder</span>(stage), <span class="at">kit =</span> <span class="fu">fct_inorder</span>(kit))</span>
<span id="cb15-331"><a href="#cb15-331" aria-hidden="true" tabindex="-1"></a>n_reads_4 <span class="ot">&lt;-</span> qc_tab_4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, read_pair, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-332"><a href="#cb15-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(<span class="fu">as.integer</span>(n_reads)),</span>
<span id="cb15-333"><a href="#cb15-333" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-334"><a href="#cb15-334" aria-hidden="true" tabindex="-1"></a>n_bases_4 <span class="ot">&lt;-</span> n_reads_4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-335"><a href="#cb15-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_bases =</span> <span class="fu">sum</span>(n_bases), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-336"><a href="#cb15-336" aria-hidden="true" tabindex="-1"></a>n_reads_4_out <span class="ot">&lt;-</span> n_reads_4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(read_pair <span class="sc">==</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">is.na</span>(read_pair)) <span class="sc">%&gt;%</span></span>
<span id="cb15-337"><a href="#cb15-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-338"><a href="#cb15-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-339"><a href="#cb15-339" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb15-340"><a href="#cb15-340" aria-hidden="true" tabindex="-1"></a>g_reads_dedup <span class="ot">&lt;-</span> n_reads_4_out <span class="sc">%&gt;%</span></span>
<span id="cb15-341"><a href="#cb15-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_reads, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-342"><a href="#cb15-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-343"><a href="#cb15-343" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-344"><a href="#cb15-344" aria-hidden="true" tabindex="-1"></a>g_bases_dedup <span class="ot">&lt;-</span> n_bases_4 <span class="sc">%&gt;%</span></span>
<span id="cb15-345"><a href="#cb15-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_bases, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-346"><a href="#cb15-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-347"><a href="#cb15-347" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-348"><a href="#cb15-348" aria-hidden="true" tabindex="-1"></a>g_dedup_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_reads_dedup)</span>
<span id="cb15-349"><a href="#cb15-349" aria-hidden="true" tabindex="-1"></a>g_reads_dedup_2 <span class="ot">&lt;-</span> g_reads_dedup <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-350"><a href="#cb15-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-351"><a href="#cb15-351" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_reads_dedup_2 <span class="sc">+</span> g_bases_dedup, g_dedup_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.15</span>))</span>
<span id="cb15-352"><a href="#cb15-352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-353"><a href="#cb15-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-354"><a href="#cb15-354" aria-hidden="true" tabindex="-1"></a>As expected, the number of duplicate reads was low: in total, 9.6M read pairs (1.8%) were removed during deduplication, with levels for individual input samples ranging from 1.4% to 2.1%. After deduplication using these parameters, the detected fraction of duplicates identified by FASTQC roughly halved for all samples:</span>
<span id="cb15-355"><a href="#cb15-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-358"><a href="#cb15-358" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-359"><a href="#cb15-359" aria-hidden="true" tabindex="-1"></a>n_dups <span class="ot">&lt;-</span> qc_tab_4 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_id, kit, stage) <span class="sc">%&gt;%</span></span>
<span id="cb15-360"><a href="#cb15-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">as.integer</span>(n_reads)) <span class="sc">%&gt;%</span></span>
<span id="cb15-361"><a href="#cb15-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_dup =</span> <span class="fu">sum</span>(n_reads <span class="sc">*</span> pc_dup<span class="sc">/</span><span class="dv">100</span>)<span class="sc">/</span><span class="fu">sum</span>(n_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb15-362"><a href="#cb15-362" aria-hidden="true" tabindex="-1"></a>g_dups <span class="ot">&lt;-</span> n_dups <span class="sc">%&gt;%</span></span>
<span id="cb15-363"><a href="#cb15-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>pc_dup, <span class="at">fill=</span>stage)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-364"><a href="#cb15-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb15-365"><a href="#cb15-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Duplicate Sequences (FASTQC)"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb15-366"><a href="#cb15-366" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb15-367"><a href="#cb15-367" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span></span>
<span id="cb15-368"><a href="#cb15-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>, <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-369"><a href="#cb15-369" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>))</span>
<span id="cb15-370"><a href="#cb15-370" aria-hidden="true" tabindex="-1"></a>g_dups</span>
<span id="cb15-371"><a href="#cb15-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-372"><a href="#cb15-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-373"><a href="#cb15-373" aria-hidden="true" tabindex="-1"></a><span class="fu"># Kraken2 -- Initial Assignments</span></span>
<span id="cb15-374"><a href="#cb15-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-375"><a href="#cb15-375" aria-hidden="true" tabindex="-1"></a>Following deduplication, I ran the data through Kraken2 for taxonomic assignments, using the 16GB Standard database<span class="ot">[^4]</span>:</span>
<span id="cb15-376"><a href="#cb15-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-377"><a href="#cb15-377" aria-hidden="true" tabindex="-1"></a><span class="ot">[^4]: </span>In the future, I plan to look at the effect of using different Kraken2 databases on read assignment, but for now I'm sticking with the database used in the main pipeline.</span>
<span id="cb15-378"><a href="#cb15-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-379"><a href="#cb15-379" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb15-380"><a href="#cb15-380" aria-hidden="true" tabindex="-1"></a><span class="in">for p in $(cat prefixes.txt); do echo $p; </span></span>
<span id="cb15-381"><a href="#cb15-381" aria-hidden="true" tabindex="-1"></a><span class="in">kraken2 --db ~/kraken-db/ --use-names --paired --threads 30 --output kraken/${p}_unmerged.output --report kraken/${p}_unmerged.report dedup/${p}_clumpify_unmerged_[12].fastq.gz; </span></span>
<span id="cb15-382"><a href="#cb15-382" aria-hidden="true" tabindex="-1"></a><span class="in">kraken2 --db ~/kraken-db/ --use-names --threads 30 --output kraken/${p}_merged.output --report kraken/${p}_merged.report dedup/${p}_clumpify_merged.fastq.gz; cat kraken/${p}_unmerged.output kraken/${p}_merged.output &gt; kraken/${p}.output; </span></span>
<span id="cb15-383"><a href="#cb15-383" aria-hidden="true" tabindex="-1"></a><span class="in">done</span></span>
<span id="cb15-384"><a href="#cb15-384" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-385"><a href="#cb15-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-386"><a href="#cb15-386" aria-hidden="true" tabindex="-1"></a>In all cases, most of the reads (<span class="sc">\&gt;</span>91%) were unassigned, and most of the rest (<span class="sc">\&gt;</span>7%) were bacterial. Assigned archaea, eukaryote, and virus reads all made up a very small fraction of the data for all samples.</span>
<span id="cb15-387"><a href="#cb15-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-390"><a href="#cb15-390" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-391"><a href="#cb15-391" aria-hidden="true" tabindex="-1"></a>kraken_domains_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"kraken-domains.csv"</span>)</span>
<span id="cb15-392"><a href="#cb15-392" aria-hidden="true" tabindex="-1"></a>kraken_domains <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(kraken_domains_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-393"><a href="#cb15-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, </span>
<span id="cb15-394"><a href="#cb15-394" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"library_id"</span>, <span class="st">"sequencing_id"</span>, <span class="st">"sequencing_replicate"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-395"><a href="#cb15-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, <span class="fu">everything</span>())</span>
<span id="cb15-396"><a href="#cb15-396" aria-hidden="true" tabindex="-1"></a>kraken_domains_gathered <span class="ot">&lt;-</span> kraken_domains <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>n_total) <span class="sc">%&gt;%</span></span>
<span id="cb15-397"><a href="#cb15-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(domain, n_reads, <span class="sc">-</span>sample_id, <span class="sc">-</span>library_id, <span class="sc">-</span>kit, <span class="sc">-</span>sequencing_replicate, <span class="sc">-</span>sequencing_id, <span class="sc">-</span>index) <span class="sc">%&gt;%</span></span>
<span id="cb15-398"><a href="#cb15-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> <span class="fu">str_to_sentence</span>(<span class="fu">sub</span>(<span class="st">"n_"</span>, <span class="st">""</span>, domain)),</span>
<span id="cb15-399"><a href="#cb15-399" aria-hidden="true" tabindex="-1"></a>         <span class="at">domain =</span> <span class="fu">fct_inorder</span>(domain)) <span class="sc">%&gt;%</span></span>
<span id="cb15-400"><a href="#cb15-400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id, kit, domain) <span class="sc">%&gt;%</span></span>
<span id="cb15-401"><a href="#cb15-401" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-402"><a href="#cb15-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb15-403"><a href="#cb15-403" aria-hidden="true" tabindex="-1"></a>g_domains <span class="ot">&lt;-</span> kraken_domains_gathered <span class="sc">%&gt;%</span></span>
<span id="cb15-404"><a href="#cb15-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>p_reads, <span class="at">fill =</span> domain)) <span class="sc">+</span></span>
<span id="cb15-405"><a href="#cb15-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb15-406"><a href="#cb15-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>, <span class="at">name =</span> <span class="st">"Domain"</span>) <span class="sc">+</span></span>
<span id="cb15-407"><a href="#cb15-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Deduplicated Reads"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb15-408"><a href="#cb15-408" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb15-409"><a href="#cb15-409" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(y) y <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb15-410"><a href="#cb15-410" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-411"><a href="#cb15-411" aria-hidden="true" tabindex="-1"></a>g_domains_minor <span class="ot">&lt;-</span> kraken_domains_gathered <span class="sc">%&gt;%</span></span>
<span id="cb15-412"><a href="#cb15-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> domain <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Unassigned"</span>, <span class="st">"Bacteria"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-413"><a href="#cb15-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>p_reads, <span class="at">fill =</span> domain)) <span class="sc">+</span></span>
<span id="cb15-414"><a href="#cb15-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb15-415"><a href="#cb15-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> scales<span class="sc">::</span><span class="fu">brewer_pal</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>)(<span class="dv">8</span>)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)],</span>
<span id="cb15-416"><a href="#cb15-416" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">"Domain"</span>) <span class="sc">+</span></span>
<span id="cb15-417"><a href="#cb15-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Deduplicated Reads"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb15-418"><a href="#cb15-418" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">0.01</span>,<span class="fl">0.002</span>),</span>
<span id="cb15-419"><a href="#cb15-419" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(y) y <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb15-420"><a href="#cb15-420" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-421"><a href="#cb15-421" aria-hidden="true" tabindex="-1"></a>g_domains_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_domains)</span>
<span id="cb15-422"><a href="#cb15-422" aria-hidden="true" tabindex="-1"></a>g_domains_2 <span class="ot">&lt;-</span> g_domains <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-423"><a href="#cb15-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-424"><a href="#cb15-424" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_domains_2 <span class="sc">+</span> g_domains_minor, g_domains_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.2</span>))</span>
<span id="cb15-425"><a href="#cb15-425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-426"><a href="#cb15-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-427"><a href="#cb15-427" aria-hidden="true" tabindex="-1"></a><span class="fu"># Kraken2 -- Human-Infecting Viruses</span></span>
<span id="cb15-428"><a href="#cb15-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-429"><a href="#cb15-429" aria-hidden="true" tabindex="-1"></a>Finally, I calculated the number of reads assigned to human-infecting viruses, using scripts based on the <span class="in">`cladecounts`</span> and <span class="in">`allmatches`</span> functions from <span class="co">[</span><span class="ot">the public pipeline</span><span class="co">](https://github.com/naobservatory/mgs-pipeline/blob/main/run.py)</span>. The number of resulting read assignments varied from just over half as many as in the private dashboard, to slightly more than that number:</span>
<span id="cb15-430"><a href="#cb15-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-433"><a href="#cb15-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-434"><a href="#cb15-434" aria-hidden="true" tabindex="-1"></a>hv_counts_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv-counts.csv"</span>)</span>
<span id="cb15-435"><a href="#cb15-435" aria-hidden="true" tabindex="-1"></a>hv_counts <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(hv_counts_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-436"><a href="#cb15-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, </span>
<span id="cb15-437"><a href="#cb15-437" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"library_id"</span>, <span class="st">"sequencing_id"</span>, <span class="st">"sequencing_replicate"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-438"><a href="#cb15-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_id, library_id, kit, sequencing_replicate, <span class="fu">everything</span>())</span>
<span id="cb15-439"><a href="#cb15-439" aria-hidden="true" tabindex="-1"></a>hv_counts_gathered <span class="ot">&lt;-</span> hv_counts <span class="sc">%&gt;%</span> </span>
<span id="cb15-440"><a href="#cb15-440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>index) <span class="sc">%&gt;%</span></span>
<span id="cb15-441"><a href="#cb15-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(pipeline, n_hv, <span class="sc">-</span>sample_id, <span class="sc">-</span>library_id, <span class="sc">-</span>kit, <span class="sc">-</span>sequencing_replicate, </span>
<span id="cb15-442"><a href="#cb15-442" aria-hidden="true" tabindex="-1"></a>         <span class="sc">-</span>sequencing_id, <span class="sc">-</span>n_total) <span class="sc">%&gt;%</span></span>
<span id="cb15-443"><a href="#cb15-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pipeline =</span> <span class="fu">sub</span>(<span class="st">"hv_"</span>, <span class="st">""</span>, pipeline),</span>
<span id="cb15-444"><a href="#cb15-444" aria-hidden="true" tabindex="-1"></a>         <span class="at">pipeline =</span> <span class="fu">fct_inorder</span>(pipeline),</span>
<span id="cb15-445"><a href="#cb15-445" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_hv =</span> n_hv<span class="sc">/</span>n_total)</span>
<span id="cb15-446"><a href="#cb15-446" aria-hidden="true" tabindex="-1"></a>g_hv_n <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_counts_gathered, <span class="fu">aes</span>(<span class="at">x=</span>sequencing_id, <span class="at">y=</span>n_hv, <span class="at">fill=</span>pipeline)) <span class="sc">+</span></span>
<span id="cb15-447"><a href="#cb15-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-448"><a href="#cb15-448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"# Human-infecting virus reads"</span>) <span class="sc">+</span></span>
<span id="cb15-449"><a href="#cb15-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Pipeline"</span>) <span class="sc">+</span></span>
<span id="cb15-450"><a href="#cb15-450" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-451"><a href="#cb15-451" aria-hidden="true" tabindex="-1"></a>g_hv_p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_counts_gathered, <span class="fu">aes</span>(<span class="at">x=</span>sequencing_id, <span class="at">y=</span>p_hv, <span class="at">fill=</span>pipeline)) <span class="sc">+</span></span>
<span id="cb15-452"><a href="#cb15-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-453"><a href="#cb15-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"RA(human-infecting virus reads)"</span>) <span class="sc">+</span></span>
<span id="cb15-454"><a href="#cb15-454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Pipeline"</span>) <span class="sc">+</span></span>
<span id="cb15-455"><a href="#cb15-455" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-456"><a href="#cb15-456" aria-hidden="true" tabindex="-1"></a>g_hv_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_hv_n)</span>
<span id="cb15-457"><a href="#cb15-457" aria-hidden="true" tabindex="-1"></a>g_hv_n_2 <span class="ot">&lt;-</span> g_hv_n <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-458"><a href="#cb15-458" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_hv_n_2 <span class="sc">+</span> g_hv_p, g_hv_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.15</span>))</span>
<span id="cb15-459"><a href="#cb15-459" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-460"><a href="#cb15-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-461"><a href="#cb15-461" aria-hidden="true" tabindex="-1"></a>Summarizing over kits, the results are as follows:</span>
<span id="cb15-462"><a href="#cb15-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-465"><a href="#cb15-465" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-466"><a href="#cb15-466" aria-hidden="true" tabindex="-1"></a>hv_counts_summ <span class="ot">&lt;-</span> hv_counts_gathered <span class="sc">%&gt;%</span></span>
<span id="cb15-467"><a href="#cb15-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kit =</span> <span class="fu">fct_inorder</span>(kit)) <span class="sc">%&gt;%</span></span>
<span id="cb15-468"><a href="#cb15-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample_id, kit, pipeline) <span class="sc">%&gt;%</span></span>
<span id="cb15-469"><a href="#cb15-469" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_total =</span> <span class="fu">sum</span>(n_total), <span class="at">n_hv =</span> <span class="fu">sum</span>(n_hv), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-470"><a href="#cb15-470" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_hv =</span> n_hv<span class="sc">/</span>n_total)</span>
<span id="cb15-471"><a href="#cb15-471" aria-hidden="true" tabindex="-1"></a>g_hv_summ_n <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_counts_summ, <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>n_hv, <span class="at">fill=</span>pipeline)) <span class="sc">+</span></span>
<span id="cb15-472"><a href="#cb15-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-473"><a href="#cb15-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"# Human-infecting virus reads"</span>) <span class="sc">+</span></span>
<span id="cb15-474"><a href="#cb15-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Pipeline"</span>) <span class="sc">+</span></span>
<span id="cb15-475"><a href="#cb15-475" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb15-476"><a href="#cb15-476" aria-hidden="true" tabindex="-1"></a>g_hv_summ_p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_counts_summ, <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>p_hv, <span class="at">fill=</span>pipeline)) <span class="sc">+</span></span>
<span id="cb15-477"><a href="#cb15-477" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb15-478"><a href="#cb15-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"RA(human-infecting virus reads)"</span>) <span class="sc">+</span></span>
<span id="cb15-479"><a href="#cb15-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Pipeline"</span>) <span class="sc">+</span></span>
<span id="cb15-480"><a href="#cb15-480" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-481"><a href="#cb15-481" aria-hidden="true" tabindex="-1"></a>g_hv_summ_legend <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_hv_summ_n)</span>
<span id="cb15-482"><a href="#cb15-482" aria-hidden="true" tabindex="-1"></a>g_hv_summ_n_2 <span class="ot">&lt;-</span> g_hv_summ_n <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb15-483"><a href="#cb15-483" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(g_hv_summ_n_2 <span class="sc">+</span> g_hv_summ_p, g_hv_summ_legend, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.15</span>))</span>
<span id="cb15-484"><a href="#cb15-484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb15-485"><a href="#cb15-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-486"><a href="#cb15-486" aria-hidden="true" tabindex="-1"></a>I'm not sure what exactly is giving rise to the discrepancy between the two pipelines. While some degree of disagreement between two different pipelines isn't surprising, it would be good to have a deeper understanding of what's causing it, and which pipeline is giving results that are more "believable". I'll look more deeply into this in a subsequent notebook entry.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>