<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2023-09-21">
<title>Will’s Public NAO Notebook - Extraction experiment 2: high-level results &amp; interpretation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Extraction experiment 2: high-level results &amp; interpretation</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Comparing RNA yields and quality across extraction kits for settled solids</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">September 21, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p><strong>See also:</strong></p>
<ul>
<li><a href="https://docs.google.com/document/d/1Po6eE3LHc9svHC4yfnDak-B-sKa7o1Pc1noFMoSbFSE/edit#heading=h.ukkrtnihc50j">Experiment writeup</a></li>
<li><a href="https://docs.google.com/spreadsheets/d/1f_KSgM2mPdv01Z6QtuDFeQjFP1Ix7YfHqmn-qtAcSoc/edit#gid=1454827698">Templates and results spreadsheet</a></li>
<li><a href="https://drive.google.com/drive/folders/1yEo2b4RYWbxxYoOu0npSPGSGeWw0swiL">Fragment Analyzer results directory</a></li>
</ul>
<section id="total-rna-quantification" class="level1"><h1>1. Total RNA quantification</h1>
<p>Nanodrop and RNA Qubit assays were performed on replicates A and B of each kit, after <a href="https://docs.google.com/document/d/144ApL4fRZxjZSvVXLzc1VOiHNW39iJxFKjCYGHZ3Ei0/edit#heading=h.iw6f49how6k5">DNA digestion with DNase</a>.Fragment Analyzer was run on replicate A of each kit.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">inpath</span> <span class="op">&lt;-</span> <span class="st">"../data/2023-09-11_extraction-expt-2.tsv"</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">inpath</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false"> <div class="table-caption"><p>Table 1: Data from extraction experiment</p></div>
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["kit"],"name":[2],"type":["chr"],"align":["left"]},{"label":["replicate"],"name":[3],"type":["chr"],"align":["left"]},{"label":["qubit_rna_ngul"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["nanodrop_rna_ngul"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["qubit_nanodrop_ratio"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["nanodrop_rna_260_280"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["nanodrop_rna_260_230"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["fa_rna_ngul"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["elution_volume"],"name":[10],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"Zymo quick-RNA","3":"A","4":"15.80","5":"21.0","6":"0.7523810","7":"1.76","8":"3.91","9":"22.9846","10":"30"},{"1":"1","2":"Zymo quick-RNA","3":"B","4":"15.30","5":"20.5","6":"0.7463415","7":"1.69","8":"2.04","9":"NA","10":"30"},{"1":"2","2":"Zymo quick-DNA/RNA","3":"A","4":"13.70","5":"23.9","6":"0.5732218","7":"1.69","8":"1.90","9":"24.7115","10":"100"},{"1":"2","2":"Zymo quick-DNA/RNA","3":"B","4":"13.10","5":"21.9","6":"0.5981735","7":"1.68","8":"1.37","9":"NA","10":"100"},{"1":"3","2":"Invitrogen PureLink RNA","3":"A","4":"12.60","5":"14.8","6":"0.8513514","7":"1.69","8":"1.81","9":"14.3232","10":"100"},{"1":"3","2":"Invitrogen PureLink RNA","3":"B","4":"11.50","5":"14.4","6":"0.7986111","7":"1.64","8":"1.16","9":"NA","10":"100"},{"1":"4","2":"NucleoSpin Virus","3":"A","4":"14.60","5":"24.5","6":"0.5959184","7":"1.75","8":"3.75","9":"19.0442","10":"60"},{"1":"4","2":"NucleoSpin Virus","3":"B","4":"11.80","5":"21.0","6":"0.5619048","7":"1.64","8":"2.53","9":"NA","10":"60"},{"1":"5","2":"QIAamp MinElute Virus","3":"A","4":"17.10","5":"25.2","6":"0.6785714","7":"1.74","8":"1.90","9":"20.7415","10":"100"},{"1":"5","2":"QIAamp MinElute Virus","3":"B","4":"16.60","5":"23.9","6":"0.6945607","7":"1.69","8":"2.01","9":"NA","10":"100"},{"1":"6","2":"QIAamp Viral RNA mini kit","3":"A","4":"16.70","5":"29.4","6":"0.5680272","7":"1.72","8":"0.73","9":"21.1794","10":"80"},{"1":"6","2":"QIAamp Viral RNA mini kit","3":"B","4":"16.90","5":"32.1","6":"0.5264798","7":"1.75","8":"1.09","9":"NA","10":"80"},{"1":"7","2":"Qiagen AllPrep PowerViral DNA/RNA","3":"A","4":"3.78","5":"10.0","6":"0.3780000","7":"0.90","8":"0.18","9":"1.8471","10":"100"},{"1":"7","2":"Qiagen AllPrep PowerViral DNA/RNA","3":"B","4":"4.92","5":"10.6","6":"0.4641509","7":"1.36","8":"0.40","9":"NA","10":"100"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Raw total concentration readings for each of the instruments are shown below:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">data_processed</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>,<span class="va">kit</span>,<span class="va">replicate</span>,<span class="va">elution_volume</span>,<span class="fu">contains</span><span class="op">(</span><span class="st">"ngul"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">gather</span><span class="op">(</span><span class="va">instrument</span>, <span class="va">concentration</span>, <span class="op">-</span><span class="va">sample</span>, <span class="op">-</span><span class="va">kit</span>, <span class="op">-</span><span class="va">replicate</span>, <span class="op">-</span><span class="va">elution_volume</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument <span class="op">=</span> <span class="fu">str_to_title</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.*"</span>, <span class="st">""</span>, <span class="va">instrument</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         instrument <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Fa"</span>, <span class="st">"Fragment Analyzer"</span>, <span class="va">instrument</span><span class="op">)</span>,</span>
<span>         concentration_adj <span class="op">=</span> <span class="va">concentration</span><span class="op">/</span><span class="va">elution_volume</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">g_conc</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data_processed</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">concentration</span>, color<span class="op">=</span><span class="va">instrument</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span>,<span class="fl">10</span><span class="op">)</span>, </span>
<span>                     name <span class="op">=</span> <span class="st">"Concentration (ng/ul)"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Extraction Kit"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Instrument"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span><span class="va">g_conc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-concentrations-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 1: Raw total RNA concentration readings for Qubit, Nanodrop and Fragment Analyzer.</figcaption></figure>
</div>
</div>
</div>
<p>Readings from Nanodrop and Qubit correlate well with both Pearson’s linear correlation coefficient and Spearman’s rho. Fragment Analyzer readings correlate well with other instruments using Pearson but not Spearman (indicating that it generates different rankings):</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cor_qubit_nanodrop</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul</span>, <span class="va">nanodrop_rna_ngul</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul</span>, <span class="va">nanodrop_rna_ngul</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Qubit"</span>, instrument2 <span class="op">=</span> <span class="st">"Nanodrop"</span><span class="op">)</span></span>
<span><span class="va">cor_qubit_fa</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul</span>, <span class="va">fa_rna_ngul</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul</span>, <span class="va">fa_rna_ngul</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Qubit"</span>, instrument2 <span class="op">=</span> <span class="st">"Fragment Analyzer"</span><span class="op">)</span></span>
<span><span class="va">cor_nanodrop_fa</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">nanodrop_rna_ngul</span>, <span class="va">fa_rna_ngul</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">nanodrop_rna_ngul</span>, <span class="va">fa_rna_ngul</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Nanodrop"</span>, instrument2 <span class="op">=</span> <span class="st">"Fragment Analyzer"</span><span class="op">)</span></span>
<span><span class="va">cor_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cor_qubit_nanodrop</span>, <span class="va">cor_qubit_fa</span>, <span class="va">cor_nanodrop_fa</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">instrument1</span>, <span class="va">instrument2</span>, <span class="va">pearson</span>, <span class="va">spearman</span><span class="op">)</span></span>
<span><span class="va">cor_all</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["instrument1"],"name":[1],"type":["chr"],"align":["left"]},{"label":["instrument2"],"name":[2],"type":["chr"],"align":["left"]},{"label":["pearson"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["spearman"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Qubit","2":"Nanodrop","3":"0.8625213","4":"0.8612356"},{"1":"Qubit","2":"Fragment Analyzer","3":"0.9121571","4":"0.5000000"},{"1":"Nanodrop","2":"Fragment Analyzer","3":"0.8408151","4":"0.4642857"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qubit_fa</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">qubit_rna_ngul</span>, y<span class="op">=</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Qubit concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"FA concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_qubit_nanodrop</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">qubit_rna_ngul</span>, y<span class="op">=</span><span class="va">nanodrop_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Qubit concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Nanodrop concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_nanodrop_fa</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">nanodrop_rna_ngul</span>, y<span class="op">=</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Nanodrop concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"FA concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_qubit_nanodrop</span> <span class="op">+</span> <span class="va">g_qubit_fa</span> <span class="op">+</span> <span class="va">g_nanodrop_fa</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-correlations-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 2: Inter-instrument concentration comparisons.</figcaption></figure>
</div>
</div>
</div>
<p>Nanodrop estimates are consistently higher than Qubit, as (with the exception of Qiagen AllPrep) are Fragment Analyzer estimates. This is expected, as the Qubit assay is highly specific to RNA while the others are affected by residual DNA and (in the case of Nanodrop) various contaminants.</p>
<p>In terms of raw yield, the QIAamp MinElute Virus, QIAamp Viral RNA mini kit, and Zymo quick-RNA kit come out the best here. The Qiagen AllPrep PowerViral kit does notably worse than the other kits.</p>
</section><section id="adjusting-for-elution-volume" class="level1"><h1>2. Adjusting for elution volume</h1>
<p>The above analysis looked at the raw outputs of the instrument readings, reflecting the concentrations of each sample after normalizing elution volume. Briefly, each sample was eluted in the volume specified in Table 1 (<code>elution_volume</code>) , then the eluent was diluted to 100ul. This results in readings that are proportional to the total yield achieved by each extraction kit, rather than the raw concentration in the undiluted eluent. However, for sequencing, we might want to use the undiluted eluent, in order to achieve higher concentrations. Hence, it’s useful to look at the concentrations and other readings we would have gotten had we performed the tests on the raw eluent.</p>
<p>Adjusting total concentration for elution volume is easy; we simply divide the reported concentration by the elution volume, then multiply by 100 (the maximum elution volume across all kits). This gives us the following results:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_conc_adj</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data_processed</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">concentration_adj</span>, color<span class="op">=</span><span class="va">instrument</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">80</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">80</span>,<span class="fl">20</span><span class="op">)</span>, </span>
<span>                     name <span class="op">=</span> <span class="st">"Concentration (ng/ul)"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Extraction Kit"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Instrument"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span><span class="va">g_conc_adj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-concentrations-adj-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 3: Total RNA concentration readings for Qubit, Nanodrop and Fragment Analyzer, adjusting for elution volume.</figcaption></figure>
</div>
</div>
</div>
<p>The most obvious result of adjusting for elution volume is that the Zymo quick-RNA kit looks much better, as this kit already had good yield with the smallest elution volume. The Nucleospin Virus and QiaAmp Viral kits receive smaller boosts in performance. Conversely, those kits with large elution volumes (Invitrogen PureLink, QIAamp MinElute, Qiagen AllPrep PowerViral, and Zymo quick-DNA/RNA) look relatively worse.</p>
<p>Surprisingly (to me), adjusting for elution volume dramatically increased the correlation between the different instruments:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">data_adj</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>qubit_rna_ngul_adj <span class="op">=</span> <span class="va">qubit_rna_ngul</span> <span class="op">/</span> <span class="va">elution_volume</span> <span class="op">*</span> <span class="fl">100</span>,</span>
<span>         nanodrop_rna_ngul_adj <span class="op">=</span> <span class="va">nanodrop_rna_ngul</span> <span class="op">/</span> <span class="va">elution_volume</span> <span class="op">*</span> <span class="fl">100</span>,</span>
<span>         fa_rna_ngul_adj <span class="op">=</span> <span class="va">fa_rna_ngul</span> <span class="op">/</span> <span class="va">elution_volume</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">cor_qubit_nanodrop_adj</span> <span class="op">&lt;-</span> <span class="va">data_adj</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul_adj</span>, <span class="va">nanodrop_rna_ngul_adj</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul_adj</span>, <span class="va">nanodrop_rna_ngul_adj</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Qubit"</span>, instrument2 <span class="op">=</span> <span class="st">"Nanodrop"</span><span class="op">)</span></span>
<span><span class="va">cor_qubit_fa_adj</span> <span class="op">&lt;-</span> <span class="va">data_adj</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul_adj</span>, <span class="va">fa_rna_ngul_adj</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">qubit_rna_ngul_adj</span>, <span class="va">fa_rna_ngul_adj</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Qubit"</span>, instrument2 <span class="op">=</span> <span class="st">"Fragment Analyzer"</span><span class="op">)</span></span>
<span><span class="va">cor_nanodrop_fa_adj</span> <span class="op">&lt;-</span> <span class="va">data_adj</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">fa_rna_ngul</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pearson <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">nanodrop_rna_ngul_adj</span>, <span class="va">fa_rna_ngul_adj</span><span class="op">)</span>,</span>
<span>            spearman <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span><span class="op">(</span><span class="va">nanodrop_rna_ngul_adj</span>, <span class="va">fa_rna_ngul_adj</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument1 <span class="op">=</span> <span class="st">"Nanodrop"</span>, instrument2 <span class="op">=</span> <span class="st">"Fragment Analyzer"</span><span class="op">)</span></span>
<span><span class="va">cor_all_adj</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">cor_qubit_nanodrop_adj</span>, <span class="va">cor_qubit_fa_adj</span>, <span class="va">cor_nanodrop_fa_adj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">instrument1</span>, <span class="va">instrument2</span>, <span class="va">pearson</span>, <span class="va">spearman</span><span class="op">)</span></span>
<span><span class="va">cor_all_adj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["instrument1"],"name":[1],"type":["chr"],"align":["left"]},{"label":["instrument2"],"name":[2],"type":["chr"],"align":["left"]},{"label":["pearson"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["spearman"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Qubit","2":"Nanodrop","3":"0.9761951","4":"0.9989005"},{"1":"Qubit","2":"Fragment Analyzer","3":"0.9901551","4":"0.9642857"},{"1":"Nanodrop","2":"Fragment Analyzer","3":"0.9714544","4":"0.9642857"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qubit_fa_adj</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data_adj</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">qubit_rna_ngul_adj</span>, y<span class="op">=</span><span class="va">fa_rna_ngul_adj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. Qubit concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. FA concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_qubit_nanodrop_adj</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data_adj</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">qubit_rna_ngul_adj</span>, y<span class="op">=</span><span class="va">nanodrop_rna_ngul_adj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. Qubit concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. Nanodrop concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_nanodrop_fa_adj</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data_adj</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">nanodrop_rna_ngul_adj</span>, y<span class="op">=</span><span class="va">fa_rna_ngul_adj</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. Nanodrop concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adj. FA concentration (ng/ul)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_qubit_nanodrop_adj</span> <span class="op">+</span> <span class="va">g_qubit_fa_adj</span> <span class="op">+</span> <span class="va">g_nanodrop_fa_adj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-correlations-adj-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 4: Inter-instrument concentration comparisons after adjusting for influent volume.</figcaption></figure>
</div>
</div>
</div>
<p>I’m not sure why this should be; it’s as though some of the instruments are adjusting for the post-elution dilution more than others, which doesn’t make much sense to me. Still, it’s helpful validation that all the instruments are producing useful readouts.</p>
</section><section id="fragment-size-distributions" class="level1"><h1>3. Fragment size distributions</h1>
<p>The fragment size distributions measured by the Fragment Analyzer are as follows:</p>
<p><img src="../img/2023-09-18_fa-sizes.png" class="img-fluid"><img src="../img/2023-09-18_fa-gel.png" class="img-fluid" style="width:30.0%"></p>
<p>With the exception of the AllPrep PowerViral kit (sample 20, in purple) all of the kits show similar size distributions, with significant fragment density between roughly 100 and 500 nt and peaks around 140 and 175 nt. All samples are lacking the large peak at 115nt seen in the FemtoPulse analysis of earlier influent samples, as well as the heavy tail of long fragments.</p>
</section><section id="sample-quality" class="level1"><h1>4. Sample quality</h1>
<section id="nanodrop-260280" class="level2"><h2 class="anchored" data-anchor-id="nanodrop-260280">Nanodrop 260/280</h2>
<p>The standard measures of nucleic acid purity from Nanodrop measurements are the 260/280 and 260/230 absorption ratios. The <a href="https://assets.thermofisher.com/TFS-Assets/CAD/Product-Bulletins/TN52646-E-0215M-NucleicAcid.pdf">standard advice</a> is that, for a pure RNA sample, 260/280 should be around 2.0, while pure DNA should be around 1.8. Lower ratios are typically indicative of contamination with protein, or some specific contaminants such as phenol.</p>
<p>The 260/280 ratios measured for the DNase-treated samples are as follows:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_260_280</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_rect</span><span class="op">(</span>ymin <span class="op">=</span> <span class="fl">1.8</span>, ymax <span class="op">=</span> <span class="fl">2.2</span>, xmin <span class="op">=</span> <span class="fl">0</span>, xmax <span class="op">=</span> <span class="cn">Inf</span>, fill <span class="op">=</span> <span class="st">"black"</span>, alpha <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_rect</span><span class="op">(</span>ymin <span class="op">=</span> <span class="fl">1.9</span>, ymax <span class="op">=</span> <span class="fl">2.1</span>, xmin <span class="op">=</span> <span class="fl">0</span>, xmax <span class="op">=</span> <span class="cn">Inf</span>, fill <span class="op">=</span> <span class="st">"black"</span>, alpha <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">nanodrop_rna_260_280</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">2.3</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     name <span class="op">=</span> <span class="st">"Absorption ratio (260nm/280nm)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span><span class="va">g_260_280</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-260-280-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 5: 260/280 ratios for DNase-treated samples.</figcaption></figure>
</div>
</div>
</div>
<p>None of the kits returned 260/280 ratios at the expected level for pure RNA, though most were not drastically low. Several returned ratios close to (though below) 1.8, which might be indicative of significant residual DNA in the sample; it’s hard to distinguish between this and protein contamination without doing DNA Qubit measurements on the DNase-treated samples. Free dNTP nucleotides produced by the DNase treatment still absorb at DNA-like ratios, so DNA-like readings may not be indicative of poor quality here.</p>
<p>As with yield, the Qiagen AllPrep PowerViral kit does notably worse than the other kits.</p>
</section><section id="nanodrop-260230" class="level2"><h2 class="anchored" data-anchor-id="nanodrop-260230">Nanodrop 260/230</h2>
<p>The 260/230 absorption ratio gives a useful indicator of the presence of various contaminants, including phenol, carbohydrates, guanidine, and various salts. The standard advice is that a pure sample should have a 260/230 of around 2.0-2.2, with lower ratios indicating contamination. Readings for our DNase-treated samples are as follows:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_260_230</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_rect</span><span class="op">(</span>ymin <span class="op">=</span> <span class="fl">1.9</span>, ymax <span class="op">=</span> <span class="fl">2.3</span>, xmin <span class="op">=</span> <span class="fl">0</span>, xmax <span class="op">=</span> <span class="cn">Inf</span>, fill <span class="op">=</span> <span class="st">"black"</span>, alpha <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_rect</span><span class="op">(</span>ymin <span class="op">=</span> <span class="fl">2.0</span>, ymax <span class="op">=</span> <span class="fl">2.2</span>, xmin <span class="op">=</span> <span class="fl">0</span>, xmax <span class="op">=</span> <span class="cn">Inf</span>, fill <span class="op">=</span> <span class="st">"black"</span>, alpha <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>, y<span class="op">=</span><span class="va">nanodrop_rna_260_230</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="co">#limits = c(0.8, 2.3),</span></span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">5</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     name <span class="op">=</span> <span class="st">"Absorption ratio (260nm/230nm)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span><span class="va">g_260_230</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-260-230-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 6: 260/230 ratios for DNase-treated samples.</figcaption></figure>
</div>
</div>
</div>
<p>As we can see, many kits have significantly lower 260/230 ratios than the recommended range, indicating contamination – most likely with guanidinium salts involved in nucleic acid extraction. Combined with the 260/280 results above, this suggests that many kits would benefit from an additional cleanup step. As with other metrics, the Qiagen AllPrep PowerViral kit comes out looking worst.</p>
<p>Two kits (NucleoSpin Virus and Zymo quick-RNA) have at least one replicate with significantly higher 260/230 ratios than the recommended values. It’s not very clear what could cause this, but <a href="https://assets.thermofisher.com/TFS-Assets/CAD/Product-Bulletins/TN52646-E-0215M-NucleicAcid.pdf">one resource</a> I found suggested misblanking as the likely culprit. I’d recommend repeating the nanodrop measurements for these samples to see if the pattern persists.</p>
</section><section id="nanodropqubit-ratio" class="level2"><h2 class="anchored" data-anchor-id="nanodropqubit-ratio">Nanodrop/Qubit ratio</h2>
<p>An informal heuristic measurement of quality is to compare the measured RNA concentration with Qubit (which is highly specific) to Nanodrop (which is not); the higher the latter compared to the former, the more other material is likely contributing to the Nanodrop reading. The ratios for our DNase-treated samples are as follows:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_ratio</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">kit</span>,y<span class="op">=</span><span class="va">qubit_nanodrop_ratio</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, </span>
<span>                     name <span class="op">=</span> <span class="st">"RNA concentration ratio (Qubit/Nanodrop)"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Extraction Kit"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, angle <span class="op">=</span> <span class="fl">45</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span>
<span><span class="va">g_ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="2023-09-12_settled-solids-extraction-test_files/figure-html/plot-nanodrop-qubit-ratio-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure 7: Nanodrop/Qubit ratios for DNase-treated samples.</figcaption></figure>
</div>
</div>
</div>
<p>According to this metric, the Invitrogen PureLink RNA, QIAamp MinElute Virus, and Zymo quick-RNA kits come out looking relatively good, while the Qiagen AllPrep PowerViral kit comes out looking worst.</p>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb11" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Extraction experiment 2: high-level results &amp; interpretation"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Comparing RNA yields and quality across extraction kits for settled solids"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2023-09-21</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>**See also:**</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">Experiment writeup</span><span class="co">](https://docs.google.com/document/d/1Po6eE3LHc9svHC4yfnDak-B-sKa7o1Pc1noFMoSbFSE/edit#heading=h.ukkrtnihc50j)</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">Templates and results spreadsheet</span><span class="co">](https://docs.google.com/spreadsheets/d/1f_KSgM2mPdv01Z6QtuDFeQjFP1Ix7YfHqmn-qtAcSoc/edit#gid=1454827698)</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">Fragment Analyzer results directory</span><span class="co">](https://drive.google.com/drive/folders/1yEo2b4RYWbxxYoOu0npSPGSGeWw0swiL)</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1. Total RNA quantification</span></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>Nanodrop and RNA Qubit assays were performed on replicates A and B of each kit, after <span class="co">[</span><span class="ot">DNA digestion with DNase</span><span class="co">](https://docs.google.com/document/d/144ApL4fRZxjZSvVXLzc1VOiHNW39iJxFKjCYGHZ3Ei0/edit#heading=h.iw6f49how6k5)</span>.Fragment Analyzer was run on replicate A of each kit.</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-data</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Table 1: Data from extraction experiment"</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>inpath <span class="ot">&lt;-</span> <span class="st">"../data/2023-09-11_extraction-expt-2.tsv"</span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(inpath, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>data</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>Raw total concentration readings for each of the instruments are shown below:</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-concentrations</span></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 1: Raw total RNA concentration readings for Qubit, Nanodrop and Fragment Analyzer."</span></span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>data_processed <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,kit,replicate,elution_volume,<span class="fu">contains</span>(<span class="st">"ngul"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(instrument, concentration, <span class="sc">-</span>sample, <span class="sc">-</span>kit, <span class="sc">-</span>replicate, <span class="sc">-</span>elution_volume) <span class="sc">%&gt;%</span></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument =</span> <span class="fu">str_to_title</span>(<span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, instrument)),</span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">instrument =</span> <span class="fu">gsub</span>(<span class="st">"Fa"</span>, <span class="st">"Fragment Analyzer"</span>, instrument),</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">concentration_adj =</span> concentration<span class="sc">/</span>elution_volume <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>g_conc <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data_processed, <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>concentration, <span class="at">color=</span>instrument)) <span class="sc">+</span></span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">40</span>,<span class="dv">10</span>), </span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"Concentration (ng/ul)"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Extraction Kit"</span>) <span class="sc">+</span></span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Instrument"</span>) <span class="sc">+</span></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> </span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>g_conc</span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>Readings from Nanodrop and Qubit correlate well with both Pearson's linear correlation coefficient and Spearman's rho. Fragment Analyzer readings correlate well with other instruments using Pearson but not Spearman (indicating that it generates different rankings):</span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>cor_qubit_nanodrop <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> </span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(qubit_rna_ngul, nanodrop_rna_ngul),</span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(qubit_rna_ngul, nanodrop_rna_ngul, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Qubit"</span>, <span class="at">instrument2 =</span> <span class="st">"Nanodrop"</span>)</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a>cor_qubit_fa <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(fa_rna_ngul)) <span class="sc">%&gt;%</span></span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(qubit_rna_ngul, fa_rna_ngul),</span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(qubit_rna_ngul, fa_rna_ngul, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Qubit"</span>, <span class="at">instrument2 =</span> <span class="st">"Fragment Analyzer"</span>)</span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>cor_nanodrop_fa <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(fa_rna_ngul)) <span class="sc">%&gt;%</span></span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(nanodrop_rna_ngul, fa_rna_ngul),</span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(nanodrop_rna_ngul, fa_rna_ngul, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Nanodrop"</span>, <span class="at">instrument2 =</span> <span class="st">"Fragment Analyzer"</span>)</span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>cor_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cor_qubit_nanodrop, cor_qubit_fa, cor_nanodrop_fa) <span class="sc">%&gt;%</span></span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(instrument1, instrument2, pearson, spearman)</span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>cor_all</span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-correlations</span></span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 2: Inter-instrument concentration comparisons."</span></span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>g_qubit_fa <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>qubit_rna_ngul, <span class="at">y=</span>fa_rna_ngul)) <span class="sc">+</span></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Qubit concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"FA concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>g_qubit_nanodrop <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>qubit_rna_ngul, <span class="at">y=</span>nanodrop_rna_ngul)) <span class="sc">+</span></span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Qubit concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Nanodrop concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a>g_nanodrop_fa <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>nanodrop_rna_ngul, <span class="at">y=</span>fa_rna_ngul)) <span class="sc">+</span></span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Nanodrop concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"FA concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a>g_qubit_nanodrop <span class="sc">+</span> g_qubit_fa <span class="sc">+</span> g_nanodrop_fa</span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a>Nanodrop estimates are consistently higher than Qubit, as (with the exception of Qiagen AllPrep) are Fragment Analyzer estimates. This is expected, as the Qubit assay is highly specific to RNA while the others are affected by residual DNA and (in the case of Nanodrop) various contaminants.</span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a>In terms of raw yield, the QIAamp MinElute Virus, QIAamp Viral RNA mini kit, and Zymo quick-RNA kit come out the best here. The Qiagen AllPrep PowerViral kit does notably worse than the other kits.</span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a><span class="fu"># 2. Adjusting for elution volume</span></span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a>The above analysis looked at the raw outputs of the instrument readings, reflecting the concentrations of each sample after normalizing elution volume. Briefly, each sample was eluted in the volume specified in Table 1 (<span class="in">`elution_volume`</span>) , then the eluent was diluted to 100ul. This results in readings that are proportional to the total yield achieved by each extraction kit, rather than the raw concentration in the undiluted eluent. However, for sequencing, we might want to use the undiluted eluent, in order to achieve higher concentrations. Hence, it's useful to look at the concentrations and other readings we would have gotten had we performed the tests on the raw eluent.</span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a>Adjusting total concentration for elution volume is easy; we simply divide the reported concentration by the elution volume, then multiply by 100 (the maximum elution volume across all kits). This gives us the following results:</span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-concentrations-adj</span></span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 3: Total RNA concentration readings for Qubit, Nanodrop and Fragment Analyzer, adjusting for elution volume."</span></span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a>g_conc_adj <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data_processed, <span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>concentration_adj, <span class="at">color=</span>instrument)) <span class="sc">+</span></span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-140"><a href="#cb11-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">80</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">80</span>,<span class="dv">20</span>), </span>
<span id="cb11-141"><a href="#cb11-141" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"Concentration (ng/ul)"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-142"><a href="#cb11-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Extraction Kit"</span>) <span class="sc">+</span></span>
<span id="cb11-143"><a href="#cb11-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Instrument"</span>) <span class="sc">+</span></span>
<span id="cb11-144"><a href="#cb11-144" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> </span>
<span id="cb11-145"><a href="#cb11-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-146"><a href="#cb11-146" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-147"><a href="#cb11-147" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb11-148"><a href="#cb11-148" aria-hidden="true" tabindex="-1"></a>g_conc_adj</span>
<span id="cb11-149"><a href="#cb11-149" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-150"><a href="#cb11-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-151"><a href="#cb11-151" aria-hidden="true" tabindex="-1"></a>The most obvious result of adjusting for elution volume is that the Zymo quick-RNA kit looks much better, as this kit already had good yield with the smallest elution volume. The Nucleospin Virus and QiaAmp Viral kits receive smaller boosts in performance. Conversely, those kits with large elution volumes (Invitrogen PureLink, QIAamp MinElute, Qiagen AllPrep PowerViral, and Zymo quick-DNA/RNA) look relatively worse.</span>
<span id="cb11-152"><a href="#cb11-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-153"><a href="#cb11-153" aria-hidden="true" tabindex="-1"></a>Surprisingly (to me), adjusting for elution volume dramatically increased the correlation between the different instruments:</span>
<span id="cb11-154"><a href="#cb11-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-157"><a href="#cb11-157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-158"><a href="#cb11-158" aria-hidden="true" tabindex="-1"></a>data_adj <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb11-159"><a href="#cb11-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">qubit_rna_ngul_adj =</span> qubit_rna_ngul <span class="sc">/</span> elution_volume <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb11-160"><a href="#cb11-160" aria-hidden="true" tabindex="-1"></a>         <span class="at">nanodrop_rna_ngul_adj =</span> nanodrop_rna_ngul <span class="sc">/</span> elution_volume <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb11-161"><a href="#cb11-161" aria-hidden="true" tabindex="-1"></a>         <span class="at">fa_rna_ngul_adj =</span> fa_rna_ngul <span class="sc">/</span> elution_volume <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb11-162"><a href="#cb11-162" aria-hidden="true" tabindex="-1"></a>cor_qubit_nanodrop_adj <span class="ot">&lt;-</span> data_adj <span class="sc">%&gt;%</span></span>
<span id="cb11-163"><a href="#cb11-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(qubit_rna_ngul_adj, nanodrop_rna_ngul_adj),</span>
<span id="cb11-164"><a href="#cb11-164" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(qubit_rna_ngul_adj, nanodrop_rna_ngul_adj, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-165"><a href="#cb11-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Qubit"</span>, <span class="at">instrument2 =</span> <span class="st">"Nanodrop"</span>)</span>
<span id="cb11-166"><a href="#cb11-166" aria-hidden="true" tabindex="-1"></a>cor_qubit_fa_adj <span class="ot">&lt;-</span> data_adj <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(fa_rna_ngul)) <span class="sc">%&gt;%</span></span>
<span id="cb11-167"><a href="#cb11-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(qubit_rna_ngul_adj, fa_rna_ngul_adj),</span>
<span id="cb11-168"><a href="#cb11-168" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(qubit_rna_ngul_adj, fa_rna_ngul_adj, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-169"><a href="#cb11-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Qubit"</span>, <span class="at">instrument2 =</span> <span class="st">"Fragment Analyzer"</span>)</span>
<span id="cb11-170"><a href="#cb11-170" aria-hidden="true" tabindex="-1"></a>cor_nanodrop_fa_adj <span class="ot">&lt;-</span> data_adj <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(fa_rna_ngul)) <span class="sc">%&gt;%</span></span>
<span id="cb11-171"><a href="#cb11-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pearson =</span> <span class="fu">cor</span>(nanodrop_rna_ngul_adj, fa_rna_ngul_adj),</span>
<span id="cb11-172"><a href="#cb11-172" aria-hidden="true" tabindex="-1"></a>            <span class="at">spearman =</span> <span class="fu">cor</span>(nanodrop_rna_ngul_adj, fa_rna_ngul_adj, <span class="at">method =</span> <span class="st">"spearman"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-173"><a href="#cb11-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument1 =</span> <span class="st">"Nanodrop"</span>, <span class="at">instrument2 =</span> <span class="st">"Fragment Analyzer"</span>)</span>
<span id="cb11-174"><a href="#cb11-174" aria-hidden="true" tabindex="-1"></a>cor_all_adj <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cor_qubit_nanodrop_adj, cor_qubit_fa_adj, cor_nanodrop_fa_adj) <span class="sc">%&gt;%</span></span>
<span id="cb11-175"><a href="#cb11-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(instrument1, instrument2, pearson, spearman)</span>
<span id="cb11-176"><a href="#cb11-176" aria-hidden="true" tabindex="-1"></a>cor_all_adj</span>
<span id="cb11-177"><a href="#cb11-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-178"><a href="#cb11-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-181"><a href="#cb11-181" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-182"><a href="#cb11-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-correlations-adj</span></span>
<span id="cb11-183"><a href="#cb11-183" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 2.5</span></span>
<span id="cb11-184"><a href="#cb11-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 4: Inter-instrument concentration comparisons after adjusting for influent volume."</span></span>
<span id="cb11-185"><a href="#cb11-185" aria-hidden="true" tabindex="-1"></a>g_qubit_fa_adj <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data_adj, <span class="fu">aes</span>(<span class="at">x=</span>qubit_rna_ngul_adj, <span class="at">y=</span>fa_rna_ngul_adj)) <span class="sc">+</span></span>
<span id="cb11-186"><a href="#cb11-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-187"><a href="#cb11-187" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-188"><a href="#cb11-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. Qubit concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-189"><a href="#cb11-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. FA concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-190"><a href="#cb11-190" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-191"><a href="#cb11-191" aria-hidden="true" tabindex="-1"></a>g_qubit_nanodrop_adj <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data_adj, <span class="fu">aes</span>(<span class="at">x=</span>qubit_rna_ngul_adj, <span class="at">y=</span>nanodrop_rna_ngul_adj)) <span class="sc">+</span></span>
<span id="cb11-192"><a href="#cb11-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-193"><a href="#cb11-193" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-194"><a href="#cb11-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. Qubit concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-195"><a href="#cb11-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. Nanodrop concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-196"><a href="#cb11-196" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-197"><a href="#cb11-197" aria-hidden="true" tabindex="-1"></a>g_nanodrop_fa_adj <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data_adj, <span class="fu">aes</span>(<span class="at">x=</span>nanodrop_rna_ngul_adj, <span class="at">y=</span>fa_rna_ngul_adj)) <span class="sc">+</span></span>
<span id="cb11-198"><a href="#cb11-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb11-199"><a href="#cb11-199" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text_repel(aes(label=kit), box.padding = 0.2, size=3) +</span></span>
<span id="cb11-200"><a href="#cb11-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. Nanodrop concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-201"><a href="#cb11-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adj. FA concentration (ng/ul)"</span>) <span class="sc">+</span></span>
<span id="cb11-202"><a href="#cb11-202" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb11-203"><a href="#cb11-203" aria-hidden="true" tabindex="-1"></a>g_qubit_nanodrop_adj <span class="sc">+</span> g_qubit_fa_adj <span class="sc">+</span> g_nanodrop_fa_adj</span>
<span id="cb11-204"><a href="#cb11-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-205"><a href="#cb11-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-206"><a href="#cb11-206" aria-hidden="true" tabindex="-1"></a>I'm not sure why this should be; it's as though some of the instruments are adjusting for the post-elution dilution more than others, which doesn't make much sense to me. Still, it's helpful validation that all the instruments are producing useful readouts.</span>
<span id="cb11-207"><a href="#cb11-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-208"><a href="#cb11-208" aria-hidden="true" tabindex="-1"></a><span class="fu"># 3. Fragment size distributions</span></span>
<span id="cb11-209"><a href="#cb11-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-210"><a href="#cb11-210" aria-hidden="true" tabindex="-1"></a>The fragment size distributions measured by the Fragment Analyzer are as follows:</span>
<span id="cb11-211"><a href="#cb11-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-212"><a href="#cb11-212" aria-hidden="true" tabindex="-1"></a><span class="al">![](../img/2023-09-18_fa-sizes.png)</span> <span class="al">![](../img/2023-09-18_fa-gel.png)</span>{width="30%"}</span>
<span id="cb11-213"><a href="#cb11-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-214"><a href="#cb11-214" aria-hidden="true" tabindex="-1"></a>With the exception of the AllPrep PowerViral kit (sample 20, in purple) all of the kits show similar size distributions, with significant fragment density between roughly 100 and 500 nt and peaks around 140 and 175 nt. All samples are lacking the large peak at 115nt seen in the FemtoPulse analysis of earlier influent samples, as well as the heavy tail of long fragments.</span>
<span id="cb11-215"><a href="#cb11-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-216"><a href="#cb11-216" aria-hidden="true" tabindex="-1"></a><span class="fu"># 4. Sample quality</span></span>
<span id="cb11-217"><a href="#cb11-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-218"><a href="#cb11-218" aria-hidden="true" tabindex="-1"></a><span class="fu">## Nanodrop 260/280</span></span>
<span id="cb11-219"><a href="#cb11-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-220"><a href="#cb11-220" aria-hidden="true" tabindex="-1"></a>The standard measures of nucleic acid purity from Nanodrop measurements are the 260/280 and 260/230 absorption ratios. The <span class="co">[</span><span class="ot">standard advice</span><span class="co">](https://assets.thermofisher.com/TFS-Assets/CAD/Product-Bulletins/TN52646-E-0215M-NucleicAcid.pdf)</span> is that, for a pure RNA sample, 260/280 should be around 2.0, while pure DNA should be around 1.8. Lower ratios are typically indicative of contamination with protein, or some specific contaminants such as phenol.</span>
<span id="cb11-221"><a href="#cb11-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-222"><a href="#cb11-222" aria-hidden="true" tabindex="-1"></a>The 260/280 ratios measured for the DNase-treated samples are as follows:</span>
<span id="cb11-223"><a href="#cb11-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-226"><a href="#cb11-226" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-227"><a href="#cb11-227" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-260-280</span></span>
<span id="cb11-228"><a href="#cb11-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb11-229"><a href="#cb11-229" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 5: 260/280 ratios for DNase-treated samples."</span></span>
<span id="cb11-230"><a href="#cb11-230" aria-hidden="true" tabindex="-1"></a>g_260_280 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb11-231"><a href="#cb11-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="at">ymin =</span> <span class="fl">1.8</span>, <span class="at">ymax =</span> <span class="fl">2.2</span>, <span class="at">xmin =</span> <span class="dv">0</span>, <span class="at">xmax =</span> <span class="cn">Inf</span>, <span class="at">fill =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb11-232"><a href="#cb11-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="at">ymin =</span> <span class="fl">1.9</span>, <span class="at">ymax =</span> <span class="fl">2.1</span>, <span class="at">xmin =</span> <span class="dv">0</span>, <span class="at">xmax =</span> <span class="cn">Inf</span>, <span class="at">fill =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb11-233"><a href="#cb11-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>nanodrop_rna_260_280), <span class="at">shape =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb11-234"><a href="#cb11-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">2.3</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">5</span>,<span class="fl">0.2</span>),</span>
<span id="cb11-235"><a href="#cb11-235" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"Absorption ratio (260nm/280nm)"</span>) <span class="sc">+</span></span>
<span id="cb11-236"><a href="#cb11-236" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> </span>
<span id="cb11-237"><a href="#cb11-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-238"><a href="#cb11-238" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-239"><a href="#cb11-239" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb11-240"><a href="#cb11-240" aria-hidden="true" tabindex="-1"></a>g_260_280</span>
<span id="cb11-241"><a href="#cb11-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-242"><a href="#cb11-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-243"><a href="#cb11-243" aria-hidden="true" tabindex="-1"></a>None of the kits returned 260/280 ratios at the expected level for pure RNA, though most were not drastically low. Several returned ratios close to (though below) 1.8, which might be indicative of significant residual DNA in the sample; it’s hard to distinguish between this and protein contamination without doing DNA Qubit measurements on the DNase-treated samples. Free dNTP nucleotides produced by the DNase treatment still absorb at DNA-like ratios, so DNA-like readings may not be indicative of poor quality here.</span>
<span id="cb11-244"><a href="#cb11-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-245"><a href="#cb11-245" aria-hidden="true" tabindex="-1"></a>As with yield, the Qiagen AllPrep PowerViral kit does notably worse than the other kits.</span>
<span id="cb11-246"><a href="#cb11-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-247"><a href="#cb11-247" aria-hidden="true" tabindex="-1"></a><span class="fu">## Nanodrop 260/230</span></span>
<span id="cb11-248"><a href="#cb11-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-249"><a href="#cb11-249" aria-hidden="true" tabindex="-1"></a>The 260/230 absorption ratio gives a useful indicator of the presence of various contaminants, including phenol, carbohydrates, guanidine, and various salts. The standard advice is that a pure sample should have a 260/230 of around 2.0-2.2, with lower ratios indicating contamination. Readings for our DNase-treated samples are as follows:</span>
<span id="cb11-250"><a href="#cb11-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-253"><a href="#cb11-253" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-254"><a href="#cb11-254" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-260-230</span></span>
<span id="cb11-255"><a href="#cb11-255" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb11-256"><a href="#cb11-256" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 6: 260/230 ratios for DNase-treated samples."</span></span>
<span id="cb11-257"><a href="#cb11-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-258"><a href="#cb11-258" aria-hidden="true" tabindex="-1"></a>g_260_230 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb11-259"><a href="#cb11-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="at">ymin =</span> <span class="fl">1.9</span>, <span class="at">ymax =</span> <span class="fl">2.3</span>, <span class="at">xmin =</span> <span class="dv">0</span>, <span class="at">xmax =</span> <span class="cn">Inf</span>, <span class="at">fill =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb11-260"><a href="#cb11-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="at">ymin =</span> <span class="fl">2.0</span>, <span class="at">ymax =</span> <span class="fl">2.2</span>, <span class="at">xmin =</span> <span class="dv">0</span>, <span class="at">xmax =</span> <span class="cn">Inf</span>, <span class="at">fill =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.02</span>) <span class="sc">+</span></span>
<span id="cb11-261"><a href="#cb11-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>kit, <span class="at">y=</span>nanodrop_rna_260_230), <span class="at">shape =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb11-262"><a href="#cb11-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="co">#limits = c(0.8, 2.3),</span></span>
<span id="cb11-263"><a href="#cb11-263" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">5</span>,<span class="fl">0.2</span>),</span>
<span id="cb11-264"><a href="#cb11-264" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"Absorption ratio (260nm/230nm)"</span>) <span class="sc">+</span></span>
<span id="cb11-265"><a href="#cb11-265" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> </span>
<span id="cb11-266"><a href="#cb11-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-267"><a href="#cb11-267" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-268"><a href="#cb11-268" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb11-269"><a href="#cb11-269" aria-hidden="true" tabindex="-1"></a>g_260_230</span>
<span id="cb11-270"><a href="#cb11-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-271"><a href="#cb11-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-272"><a href="#cb11-272" aria-hidden="true" tabindex="-1"></a>As we can see, many kits have significantly lower 260/230 ratios than the recommended range, indicating contamination – most likely with guanidinium salts involved in nucleic acid extraction. Combined with the 260/280 results above, this suggests that many kits would benefit from an additional cleanup step. As with other metrics, the Qiagen AllPrep PowerViral kit comes out looking worst.</span>
<span id="cb11-273"><a href="#cb11-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-274"><a href="#cb11-274" aria-hidden="true" tabindex="-1"></a>Two kits (NucleoSpin Virus and Zymo quick-RNA) have at least one replicate with significantly higher 260/230 ratios than the recommended values. It’s not very clear what could cause this, but <span class="co">[</span><span class="ot">one resource</span><span class="co">](https://assets.thermofisher.com/TFS-Assets/CAD/Product-Bulletins/TN52646-E-0215M-NucleicAcid.pdf)</span> I found suggested misblanking as the likely culprit. I’d recommend repeating the nanodrop measurements for these samples to see if the pattern persists.</span>
<span id="cb11-275"><a href="#cb11-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-276"><a href="#cb11-276" aria-hidden="true" tabindex="-1"></a><span class="fu">## Nanodrop/Qubit ratio</span></span>
<span id="cb11-277"><a href="#cb11-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-278"><a href="#cb11-278" aria-hidden="true" tabindex="-1"></a>An informal heuristic measurement of quality is to compare the measured RNA concentration with Qubit (which is highly specific) to Nanodrop (which is not); the higher the latter compared to the former, the more other material is likely contributing to the Nanodrop reading. The ratios for our DNase-treated samples are as follows:</span>
<span id="cb11-279"><a href="#cb11-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-282"><a href="#cb11-282" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-283"><a href="#cb11-283" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-nanodrop-qubit-ratio</span></span>
<span id="cb11-284"><a href="#cb11-284" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb11-285"><a href="#cb11-285" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Figure 7: Nanodrop/Qubit ratios for DNase-treated samples."</span></span>
<span id="cb11-286"><a href="#cb11-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-287"><a href="#cb11-287" aria-hidden="true" tabindex="-1"></a>g_ratio <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>kit,<span class="at">y=</span>qubit_nanodrop_ratio)) <span class="sc">+</span></span>
<span id="cb11-288"><a href="#cb11-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb11-289"><a href="#cb11-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), </span>
<span id="cb11-290"><a href="#cb11-290" aria-hidden="true" tabindex="-1"></a>                     <span class="at">name =</span> <span class="st">"RNA concentration ratio (Qubit/Nanodrop)"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-291"><a href="#cb11-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Extraction Kit"</span>) <span class="sc">+</span></span>
<span id="cb11-292"><a href="#cb11-292" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> </span>
<span id="cb11-293"><a href="#cb11-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-294"><a href="#cb11-294" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-295"><a href="#cb11-295" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb11-296"><a href="#cb11-296" aria-hidden="true" tabindex="-1"></a>g_ratio</span>
<span id="cb11-297"><a href="#cb11-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-298"><a href="#cb11-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-299"><a href="#cb11-299" aria-hidden="true" tabindex="-1"></a>According to this metric, the Invitrogen PureLink RNA, QIAamp MinElute Virus, and Zymo quick-RNA kits come out looking relatively good, while the Qiagen AllPrep PowerViral kit comes out looking worst.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>