<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-04-01">
<title>Will’s Public NAO Notebook - Workflow analysis of Spurbeck et al.&nbsp;(2023)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Spurbeck et al.&nbsp;(2023)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Cave carpa.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 1, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>Continuing my analysis of datasets from the <a href="https://doi.org/10.1101/2023.12.22.23300450">P2RA preprint</a>, I analyzed the data from Spurbeck et al.&nbsp;(2023), a wastewater RNA-sequencing study from Ohio. Samples for this study underwent a variety of processing protocols at different research centers across the state. Along with Rothman and Crits-Christoph, Spurbeck is one of three RNA-sequencing studies that underwent full in-depth analysis in the P2RA study, so is worth looking at closely here.</p>
<p>This one turned out to be a bit of a saga. As we’ll see in the section on human-virus identification, it took multiple tries and several substantial changes to the pipeline to get things to a state I was happy with. Still, I am happy with the outcome and think the changes will improve analysis of future datasets.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The Spurbeck data comprises 55 samples from 8 processing groups, with 4 to 6 samples per group. The number of sequencing read pairs per sample varied widely from 4.5M-106.7M (mean 33.4M). Taken together, the samples totaled roughly 1.8B read pairs (425 gigabases of sequence). Read qualities were generally high but in need of some light preprocessing. Adapter levels were moderate. Inferred duplication levels were fairly high: 14-91% with a mean of 55%.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Data input paths</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-04-01_spurbeck/"</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats_1.tsv"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_adapter_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_base_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_sequence_stats.tsv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">adapter_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_base_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_seq_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize basic stats</span></span>
<span><span class="va">scale_fill_grp</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Set3"</span>, name<span class="op">=</span><span class="st">"Processing group"</span><span class="op">)</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">group</span>, fill<span class="op">=</span><span class="va">group</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span><span class="va">g_nreads_raw</span> <span class="op">&lt;-</span> <span class="va">g_basic</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_nreads_raw_2</span> <span class="op">&lt;-</span> <span class="va">g_nreads_raw</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">legend_group</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_nreads_raw</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">g_nbases_raw</span> <span class="op">&lt;-</span> <span class="va">g_basic</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_bases_approx</span><span class="op">)</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Total base pairs (approx)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_ndup_raw</span> <span class="op">&lt;-</span> <span class="va">g_basic</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Duplicates (FASTQC)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_basic_raw</span> <span class="op">&lt;-</span> <span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_nreads_raw_2</span> <span class="op">+</span> <span class="va">g_nbases_raw</span> <span class="op">+</span> <span class="va">g_ndup_raw</span>, <span class="va">legend_group</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_basic_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">scale_color_grp</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>,palette<span class="op">=</span><span class="st">"Set3"</span>,name<span class="op">=</span><span class="st">"Processing group"</span><span class="op">)</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">group</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Significant numbers of reads were lost at each stage in the preprocessing pipeline. In total, cleaning, deduplication and initial ribodepletion removed 17-96% (mean 72%) of input reads, while secondary ribodepletion removed an additional 0-11% (mean 6%).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span>, <span class="va">stage</span>, <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"1.9-89.8 (mean 13.9)","3":"1.9-89.8 (mean 13.9)"},{"1":"Deduplication","2":"8.9-95.9 (mean 37.3)","3":"2.7-65.9 (mean 23.4)"},{"1":"Initial ribodepletion","2":"16.7-96.4 (mean 72.5)","3":"0.2-62.9 (mean 35.2)"},{"1":"Secondary ribodepletion","2":"17-96.6 (mean 78.7)","3":"0.1-10.9 (mean 6.3)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">n_read_pairs</span>,fill<span class="op">=</span><span class="va">group</span>,group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span>,fill<span class="op">=</span><span class="va">group</span>,group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Data cleaning with FASTP was mostly successful at removing adapters; however, detectable levels of Illumina Universal Adapter sequences persisted through the preprocessing pipeline. FASTP was successful at improving read quality.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">group</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, deduplication was quite effective at reducing measured duplicate levels, with FASTQC-measured levels falling from an average of 55% to one of 22%:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">percent_duplicates</span>, fill<span class="op">=</span><span class="va">group</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">mean_seq_len</span>, fill<span class="op">=</span><span class="va">group</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_grp</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">legend_loc</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_dup_stages</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import Bracken data</span></span>
<span><span class="va">bracken_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"bracken_counts.tsv"</span><span class="op">)</span></span>
<span><span class="va">bracken</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">bracken_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">total_assigned</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  kraken_assigned_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">kraken_assigned_reads</span><span class="op">)</span>,</span>
<span>  added_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">added_reads</span><span class="op">)</span>,</span>
<span>  new_est_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">new_est_reads</span><span class="op">)</span>,</span>
<span>  fraction_total_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">fraction_total_reads</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bracken_spread</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"sample"</span>, names_from <span class="op">=</span> <span class="st">"name"</span>, values_from <span class="op">=</span> <span class="st">"new_est_reads"</span><span class="op">)</span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span>, <span class="va">stage</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"group"</span><span class="op">)</span>, names_from<span class="op">=</span><span class="st">"stage"</span>, values_from<span class="op">=</span><span class="st">"n_read_pairs"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_preproc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">total_assigned</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>assigned <span class="op">=</span> <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">bracken_spread</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span><span class="co"># Assess composition</span></span>
<span><span class="va">read_comp</span> <span class="op">&lt;-</span> <span class="fu">transmute</span><span class="op">(</span><span class="va">read_counts</span>, sample<span class="op">=</span><span class="va">sample</span>, group<span class="op">=</span><span class="va">group</span>,</span>
<span>                       n_filtered <span class="op">=</span> <span class="va">raw_concat</span><span class="op">-</span><span class="va">cleaned</span>,</span>
<span>                       n_duplicate <span class="op">=</span> <span class="va">cleaned</span><span class="op">-</span><span class="va">dedup</span>,</span>
<span>                       n_ribosomal <span class="op">=</span> <span class="op">(</span><span class="va">dedup</span><span class="op">-</span><span class="va">ribo_initial</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="va">ribo_initial</span><span class="op">-</span><span class="va">ribo_secondary</span><span class="op">)</span>,</span>
<span>                       n_unassigned <span class="op">=</span> <span class="va">ribo_secondary</span><span class="op">-</span><span class="va">assigned</span>,</span>
<span>                       n_bacterial <span class="op">=</span> <span class="va">bacteria</span>,</span>
<span>                       n_archaeal <span class="op">=</span> <span class="va">archaea</span>,</span>
<span>                       n_viral <span class="op">=</span> <span class="va">viruses</span>,</span>
<span>                       n_human <span class="op">=</span> <span class="va">eukaryota</span><span class="op">)</span></span>
<span><span class="va">read_comp_long</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">read_comp</span>, <span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">group</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"classification"</span>,</span>
<span>                               names_prefix <span class="op">=</span> <span class="st">"n_"</span>, values_to <span class="op">=</span> <span class="st">"n_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu">str_to_sentence</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_comp_long</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Reads\n"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">group</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">read_comp_minor</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_comp_minor</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Reads\n"</span>,</span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">group</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, ncol<span class="op">=</span><span class="fl">5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_total</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="st">"All groups"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">p_reads_summ_total</span>, <span class="va">p_reads_summ_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">group</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">classification</span>, <span class="va">group</span>, <span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">group</span>, values_from <span class="op">=</span> <span class="va">display</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["classification"],"name":[1],"type":["fct"],"align":["left"]},{"label":["All groups"],"name":[2],"type":["chr"],"align":["left"]},{"label":["A"],"name":[3],"type":["chr"],"align":["left"]},{"label":["B"],"name":[4],"type":["chr"],"align":["left"]},{"label":["C"],"name":[5],"type":["chr"],"align":["left"]},{"label":["D"],"name":[6],"type":["chr"],"align":["left"]},{"label":["E"],"name":[7],"type":["chr"],"align":["left"]},{"label":["F"],"name":[8],"type":["chr"],"align":["left"]},{"label":["G"],"name":[9],"type":["chr"],"align":["left"]},{"label":["H"],"name":[10],"type":["chr"],"align":["left"]},{"label":["I"],"name":[11],"type":["chr"],"align":["left"]},{"label":["J"],"name":[12],"type":["chr"],"align":["left"]}],"data":[{"1":"Excluded","2":"26-100% (mean 55%)","3":"33-51% (mean 38%)","4":"26-43% (mean 34%)","5":"46-76% (mean 54%)","6":"36-68% (mean 50%)","7":"63-100% (mean 80%)","8":"43-99% (mean 71%)","9":"42-96% (mean 65%)","10":"57-99% (mean 82%)","11":"27-46% (mean 36%)","12":"35-63% (mean 44%)"},{"1":"Ribosomal","2":"0.26-72% (mean 41%)","3":"48-66% (mean 61%)","4":"56-72% (mean 64%)","5":"1.2-52% (mean 38%)","6":"9.7-56% (mean 36%)","7":"0.27-34% (mean 17%)","8":"0.9-54% (mean 27%)","9":"0.41-57% (mean 34%)","10":"0.26-40% (mean 16%)","11":"51-71% (mean 62%)","12":"34-62% (mean 53%)"},{"1":"Bacterial","2":"0.052-24% (mean 3.4%)","3":"1.1-1.7% (mean 1.5%)","4":"1.3-1.9% (mean 1.5%)","5":"2.3-23% (mean 8%)","6":"2.9-24% (mean 14%)","7":"0.052-7.6% (mean 2.7%)","8":"0.072-2.9% (mean 1.7%)","9":"0.17-5.5% (mean 1.8%)","10":"0.5-5.7% (mean 2.2%)","11":"2-3.2% (mean 2.6%)","12":"2.8-3.5% (mean 3%)"},{"1":"Archaeal","2":"0.000087-0.17% (mean 0.013%)","3":"0.0025-0.0059% (mean 0.0041%)","4":"0.0034-0.019% (mean 0.0089%)","5":"0.0031-0.063% (mean 0.02%)","6":"0.02-0.036% (mean 0.028%)","7":"0.00021-0.016% (mean 0.0073%)","8":"0.000087-0.025% (mean 0.0078%)","9":"0.00043-0.0084% (mean 0.0045%)","10":"0.0057-0.17% (mean 0.041%)","11":"0.0044-0.021% (mean 0.012%)","12":"0.0044-0.0092% (mean 0.0062%)"},{"1":"Viral","2":"0.00018-0.33% (mean 0.04%)","3":"0.00018-0.00031% (mean 0.00024%)","4":"0.00019-0.0011% (mean 0.00041%)","5":"0.0012-0.021% (mean 0.0067%)","6":"0.013-0.099% (mean 0.045%)","7":"0.0065-0.079% (mean 0.047%)","8":"0.013-0.23% (mean 0.1%)","9":"0.011-0.33% (mean 0.096%)","10":"0.018-0.14% (mean 0.1%)","11":"0.0012-0.0027% (mean 0.002%)","12":"0.0012-0.0022% (mean 0.0015%)"},{"1":"Human","2":"0.001-0.4% (mean 0.034%)","3":"0.0017-0.0032% (mean 0.0024%)","4":"0.001-0.0099% (mean 0.0046%)","5":"0.0023-0.16% (mean 0.044%)","6":"0.023-0.082% (mean 0.045%)","7":"0.0088-0.12% (mean 0.065%)","8":"0.006-0.4% (mean 0.089%)","9":"0.0046-0.15% (mean 0.043%)","10":"0.0062-0.17% (mean 0.046%)","11":"0.0025-0.0086% (mean 0.0058%)","12":"0.0034-0.0057% (mean 0.0042%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Average composition varied substantially between groups. Four groups (A, B, I, J) showed high within-group consistency, with high levels of ribosomal reads and very low levels of assigned reads. Other groups showed much more variability between samples, with higher average levels of assigned reads.</p>
<p>Overall, the average relative abundance of viral reads was 0.04%; however, groups F, G &amp; H showed substantially higher average abundance of around 0.1%. Even these elevated groups, however, still show lower total viral abundance than <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-02-27_rothman-1.html">Rothman</a> or <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-02-04_crits-christoph-1.html">Crits-Christoph</a>, so this doesn’t seem particularly noteworthy.</p>
</section><section id="human-infecting-virus-reads-validation-round-1" class="level1"><h1>Human-infecting virus reads: validation, round 1</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples, using a pipeline identical to that described in my <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-02-27_rothman-1.html">entry on unenriched Rothman samples</a>. This process identified a total of 31,610 read pairs across all samples (0.007% of surviving reads):</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_reads_filtered_1_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered_1.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_1</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_1_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_1</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_1_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered_1</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_1</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_1</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span><span class="va">g_mrg_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_1</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-hist-1-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_hist_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_1</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-hist-1-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>To analyze all these reads in a reasonable amount of time, I set up a dedicated EC2 instance, downloaded nt, and ran BLASTN locally there, otherwise using the same parameters I’ve used for past datasets:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_1_num</span> <span class="op">&lt;-</span> <span class="va">mrg_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">sample</span>, <span class="st">"_"</span>, <span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_1_fasta</span> <span class="op">&lt;-</span>  <span class="va">mrg_1_num</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_1_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_1_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_1_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"spurbeck-putative-viral-1.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># NB: Importing partially-processed BLAST results to save storage space</span></span>
<span><span class="co"># Import BLAST results</span></span>
<span><span class="va">blast_results_1_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"putative-viral-blast-best-1.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_results_1</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_1_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject</span></span>
<span><span class="va">blast_results_1_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each query</span></span>
<span><span class="va">blast_results_1_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_1_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_1_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_1_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, </span>
<span>         seq_num <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span>, </span>
<span>         sample <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">head</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">paste</span>, collapse<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_1_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_1_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            best_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_1_viral</span> <span class="op">&lt;-</span> <span class="va">blast_results_1_paired</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span>,</span>
<span>         viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_1_assign</span> <span class="op">&lt;-</span> <span class="va">mrg_1_num</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span></span>
<span><span class="va">blast_results_1_assign</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">blast_results_1_viral</span>, <span class="va">mrg_1_assign</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_1_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_1_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_1_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_1_num</span>, <span class="va">blast_results_1_out</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_mrg_blast_1</span> <span class="op">&lt;-</span> <span class="va">mrg_1_blast</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">viral_status_out</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_blast_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-blast-scatter-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_hist_blast_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_1_blast</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_blast_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-blast-hist-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There results were…okay. There were a lot of false positives, but nearly all of them were low-scoring and excluded by my usual score filters. Most true positives, meanwhile, had high enough scores to be retained by those filters. However, there were enough high-scoring false positives and low-scoring true positives to drag down my precision and sensitivity, resulting in an F1 score (at a disjunctive score threshold of 20) a little under 90% – quite a few percentage points lower than I usually aim for.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span>, <span class="va">conjunctive</span>, <span class="va">include_special</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">include_special</span><span class="op">)</span> <span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">tab</span>, <span class="va">viral_status_out</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TRUE"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span></span>
<span>    conjunctive <span class="op">=</span> <span class="va">conjunctive</span>,</span>
<span>    retain_score_conjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">&amp;</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>, </span>
<span>    retain_score_disjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>    retain_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="va">retain_score_conjunctive</span>, <span class="va">retain_score_disjunctive</span><span class="op">)</span>,</span>
<span>    retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, include_special <span class="op">=</span> <span class="va">include_special</span>, </span>
<span>                conjunctive <span class="op">=</span> <span class="va">conjunctive</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inc_special</span><span class="op">=</span><span class="cn">FALSE</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span>, include_special<span class="op">=</span><span class="va">inc_special</span><span class="op">)</span></span>
<span>  <span class="va">stats_conj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_disj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_conj</span>, <span class="va">stats_disj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">threshold</span><span class="op">:</span><span class="va">conjunctive</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>conj_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats_all</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">stats_1</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_1_blast</span><span class="op">)</span></span>
<span><span class="va">threshold_opt_1</span> <span class="op">&lt;-</span> <span class="va">stats_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_stats_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_1</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>data <span class="op">=</span> <span class="va">threshold_opt_1</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>xintercept<span class="op">=</span><span class="va">threshold</span><span class="op">)</span>,</span>
<span>             color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-f1-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Digging into taxonomic assignments more deeply, we find that low-scoring false positives are primarily mapped by Bowtie2 to SARS-CoV-2, while higher-scoring false positives are mainly mapped to a variety of parvoviruses and parvo-like viruses, as well as Orf virus (cows again?). High-scoring true positives map to a wide range of viruses, but low-scoring true-positives primarily map to human gammaherpesvirus 4. Given that the latter is a DNA virus, I find this quite suspicious.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fp_1</span> <span class="op">&lt;-</span> <span class="va">mrg_1_blast</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_1_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp_1</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_1_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_1_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_1_major</span> <span class="op">&lt;-</span> <span class="va">fp_1</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_1_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_fp_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_1_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-composition-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Since sensitivity is more of a problem here than precision, I decided to dig into into the “true positive” herpesvirus reads. There are 1,197 of these in total, 93% of which are low scoring. When I look into the top taxids that BLAST maps these reads to, we see the following:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># Get BLAST staxids</span></span>
<span><span class="va">herp_seqs_1</span> <span class="op">&lt;-</span> <span class="va">mrg_1_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">10376</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">adj_score_max</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">herp_seqs_1_total</span> <span class="op">&lt;-</span> <span class="va">herp_seqs_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_seqs_total"</span><span class="op">)</span></span>
<span><span class="va">herp_blast_1_hits</span> <span class="op">&lt;-</span> <span class="va">herp_seqs_1</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_1_paired</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herp_blast_1_hits_top</span> <span class="op">&lt;-</span> <span class="va">herp_blast_1_hits</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="fl">3004166</span>, </span>
<span>                     <span class="st">"Caudopunctatus cichlid hepacivirus"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herp_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">herp_blast_1_hits_top</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>, </span>
<span>                   <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">n</span>, y<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Mapped Read Pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_herp_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-herpes-hits-1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Two things strike me as notable about these results. First, human gammaherpesvirus 4 is only the second-most-common subject taxid, after SARS-CoV-2, an unrelated virus with a different nucleic-acid type. Hmm. Second, close behind these two viruses is a distinctly non-viral taxon, the common carp. This jumped out at me, because the common carp genome is <a href="https://dgg32.medium.com/carp-in-the-soil-1168818d2191">somewhat notorious</a> for being contaminated with Illumina adapter sequences. This makes me suspect that Illumina adapter contamination is playing a role in these results.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">carp_hits_1</span> <span class="op">&lt;-</span> <span class="va">herp_blast_1_hits</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="fl">7962</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_1_blast</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">adj_score_max</span>, <span class="va">query_seq_fwd</span>, <span class="va">query_seq_rev</span><span class="op">)</span></span>
<span><span class="va">carp_hits_1_out</span> <span class="op">&lt;-</span> <span class="va">carp_hits_1</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">sample</span>, <span class="st">"_"</span>, <span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">carp_hits_1_fasta</span> <span class="op">&lt;-</span> <span class="va">carp_hits_1_out</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, </span>
<span>         header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">carp_hits_1_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">carp_hits_1_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">carp_hits_1_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"spurbeck-carp-hits-1.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Inspecting these reads manually, I find that a large fraction do indeed show substantial adapter content. In particular, of the 1658 individual reads queried, at least 1338 (81%) showed a strong match to the Illumina multiplexing primer. As such, better removal of these adapter sequences would likely remove most or all of these “false true positives”, potentially significantly improving my results for this dataset.</p>
</section><section id="human-infecting-virus-reads-validation-round-2" class="level1"><h1>Human-infecting virus reads: validation, round 2</h1>
<p>To address this problem, I added <a href="https://cutadapt.readthedocs.io/en/stable/guide.html">Cutadapt</a> to the pipeline, using settings that allowed it to trim known adaptors internal to the read:</p>
<pre><code>cutadapt -b file:&lt;adapter_file&gt; -B file:&lt;adapter_file&gt; -m 15 -e 0.25 --action=trim -o &lt;fwd_reads_out&gt; -p &lt;rev_reads_out&gt; &lt;fwd_reads_in&gt; &lt;rev_reads_in&gt;</code></pre>
<p>Running this additional preprocessing step reduced the total number of reads surviving cleaning by 10.4M, or an average of 189k reads per sample. The number of putative human-viral reads, meanwhile, was reduced from 31,610 to 19,799, a 37% decrease. This reduction is primarily due to a large decrease in the number of low-scoring Bowtie2-only putative HV hits.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">basic_stats_2_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats_2.tsv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_2</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_2_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">basic_stats_2_sum</span> <span class="op">&lt;-</span> <span class="va">basic_stats_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_read_pairs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, n_bases_approx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"With Cutadapt"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_sum</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_read_pairs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, n_bases_approx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Without Cutadapt"</span><span class="op">)</span></span>
<span><span class="va">sum_cat</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">basic_stats_sum</span>, <span class="va">basic_stats_2_sum</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_sum_cat_reads</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sum_cat</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">n_read_pairs</span>, fill<span class="op">=</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Preprocessing"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read Pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_sum_cat_reads</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/cutadapt-total-reads-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_reads_filtered_2_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered_2.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_2</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_2_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_2</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_2_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered_2</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Make new labelled HV dataset</span></span>
<span><span class="va">mrg_2</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_2</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge and plot histogram</span></span>
<span><span class="va">mrg_2_join</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_1</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label<span class="op">=</span><span class="st">"Without Cutadapt"</span><span class="op">)</span>,</span>
<span>                      <span class="va">mrg_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label<span class="op">=</span><span class="st">"With Cutadapt"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">g_hist_2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_2_join</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">label</span><span class="op">~</span><span class="va">kraken_label</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-hist-2-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>Comparing the lengths of the same sequences in the old versus new results, we see that many reads are reduced in length as a result of adding Cutadapt. As such, it made sense to repeat the BLAST analysis on the reprocessed reads rather than using the BLAST assignments from the previous analysis.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_num_comp</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">mrg_2</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">seq_num</span><span class="op">)</span>, </span>
<span>                           <span class="va">mrg_1_num</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">seq_num</span><span class="op">)</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_num_diff</span> <span class="op">&lt;-</span> <span class="va">mrg_1_num</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, query_len_fwd_old <span class="op">=</span> <span class="va">query_len_fwd</span>,</span>
<span>                                   query_len_rev_old <span class="op">=</span> <span class="va">query_len_rev</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mrg_num_comp</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, query_len_fwd_new <span class="op">=</span> <span class="va">query_len_fwd</span>, </span>
<span>                                  query_len_rev_new <span class="op">=</span> <span class="va">query_len_rev</span><span class="op">)</span>,</span>
<span>             by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>query_len_fwd_diff <span class="op">=</span> <span class="va">query_len_fwd_new</span> <span class="op">-</span> <span class="va">query_len_fwd_old</span>,</span>
<span>         query_len_rev_diff <span class="op">=</span> <span class="va">query_len_rev_new</span> <span class="op">-</span> <span class="va">query_len_rev_old</span><span class="op">)</span></span>
<span><span class="va">mrg_num_diff_long</span> <span class="op">&lt;-</span> <span class="va">mrg_num_diff</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, Forward<span class="op">=</span><span class="va">query_len_fwd_diff</span>, Reverse<span class="op">=</span><span class="va">query_len_rev_diff</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">seq_num</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"read"</span>, values_to<span class="op">=</span><span class="st">"length_difference"</span><span class="op">)</span></span>
<span><span class="va">g_len_diff</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_num_diff_long</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">length_difference</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">4</span>, boundary<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Effect of Cutadapt on read length"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# of Reads"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">read</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_len_diff</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-cutadapt-length-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_2_num</span> <span class="op">&lt;-</span> <span class="va">mrg_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">sample</span>, <span class="st">"_"</span>, <span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_2_fasta</span> <span class="op">&lt;-</span>  <span class="va">mrg_2_num</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_2_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_1_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_2_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"spurbeck-putative-viral-2.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import BLAST results (again, pre-filtered to save space)</span></span>
<span><span class="va">blast_results_2_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"putative-viral-blast-best-2.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_results_2</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_2_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject</span></span>
<span><span class="va">blast_results_2_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_2</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each query</span></span>
<span><span class="va">blast_results_2_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_2_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_2_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_2_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, </span>
<span>         seq_num <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span>, </span>
<span>         sample <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">head</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">paste</span>, collapse<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_2_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_2_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            best_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_2_viral</span> <span class="op">&lt;-</span> <span class="va">blast_results_2_paired</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span>,</span>
<span>         viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_2_assign</span> <span class="op">&lt;-</span> <span class="va">mrg_2_num</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span></span>
<span><span class="va">blast_results_2_assign</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">blast_results_2_viral</span>, <span class="va">mrg_2_assign</span>, </span>
<span>                                    by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_2_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_2_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The addition of Cutadapt results in a substantial decrease in low-scoring putative HV sequences, resulting in a large improvement in measured sensitivity and F1 score:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_2_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_2_num</span>, <span class="va">blast_results_2_out</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine and label</span></span>
<span><span class="va">hist_blast_2_prep</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_1_blast</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label<span class="op">=</span><span class="st">"Without Cutadapt"</span><span class="op">)</span>,</span>
<span>                               <span class="va">mrg_2_blast</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label<span class="op">=</span><span class="st">"With Cutadapt"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hist_blast_2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hist_blast_2_prep</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="va">label</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_blast_2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-blast-hist-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_2</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_2_blast</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"With Cutadapt"</span><span class="op">)</span></span>
<span><span class="va">stats_comb</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_1</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Without Cutadapt"</span><span class="op">)</span>, <span class="va">stats_2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_stats_2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_comb</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">label</span>, linetype<span class="op">=</span><span class="va">conj_label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span>, name <span class="op">=</span> <span class="st">"Configuration"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Threshold type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">metric</span>, nrow<span class="op">=</span><span class="fl">2</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-f1-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>At a disjunctive threshold of 20, excluding “fake true positives” arising from adapter contamination improved measured sensitivity from 86% to 97%, bringing the measured F1 score up from 89% to 95%. I would feel much better about using these results for further downstream analyses.</p>
<p>That said, I think further improvement is possible. While addition of Cutadapt processing substantially improved measured sensitivity, measured precision only improved slightly. Looking at the apparent taxonomic makeup of putative HV reads, we see that, while low-scoring “true positives” mapping to herpesviruses have been mostly eliminated, high-scoring false positives still map primarily to parvoviruses and parvo-like viruses:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Calculate composition for 2nd attempt</span></span>
<span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">fp_2</span> <span class="op">&lt;-</span> <span class="va">mrg_2_blast</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, <span class="va">name</span><span class="op">)</span>,</span>
<span>         major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">fp_2_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp_2</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">major</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_2_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_2_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_2_major</span> <span class="op">&lt;-</span> <span class="va">fp_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_2_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine composition</span></span>
<span><span class="va">fp_major_comb</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">fp_1_major</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Without Cutadapt"</span><span class="op">)</span>,</span>
<span>                           <span class="va">fp_2_major</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="st">"With Cutadapt"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>label <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">name_display</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Palette</span></span>
<span><span class="va">palette_fp_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="st">"#999999"</span><span class="op">)</span></span>
<span><span class="va">g_fp_2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_major_comb</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, label <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span>, values<span class="op">=</span><span class="va">palette_fp_2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">label</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp_2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-composition-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Not only are these parvoviruses and parvo-like viruses DNA viruses (and thus suspicious as abundant components of the RNA virome), they are also the subject of a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3964066/">famous controversy</a> in early viral metagenomics, where viruses apparently widespread in Chinese hepatitis patients were found to arise from spin column contamination. In fact, these viruses seem not to be human-infecting at all; the authors of the study reporting the contamination suggested that they might be viruses of the diatom algae used as the source of silica for these columns. As such, it seems appropriate to remove these from the human-virus database used to identify putative HV reads.</p>
<p>We also see a significant number of false-positive reads mapped to Orf virus. Historically this has been a sign of contamination with bovine sequences, and indeed the top taxa mapped to these reads by BLAST include <em>Bos taurus</em> (cattle), <em>Bos mutus</em> (wild yak), and <em>Cervus elaphus</em> (red deer).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">orf_taxid</span> <span class="op">&lt;-</span> <span class="fl">10258</span></span>
<span><span class="va">orf_seqs</span> <span class="op">&lt;-</span> <span class="va">mrg_1_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">10258</span><span class="op">)</span></span>
<span><span class="va">orf_matches</span> <span class="op">&lt;-</span> <span class="va">orf_seqs</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">blast_results_1_paired</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">orf_staxids</span> <span class="op">&lt;-</span> <span class="va">orf_matches</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span></span>
<span><span class="va">orf_fp_out</span> <span class="op">&lt;-</span> <span class="va">orf_seqs</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">taxid</span>, <span class="st">"_"</span>, <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">orf_fp_fasta</span> <span class="op">&lt;-</span> <span class="va">orf_fp_out</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">orf_fp_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">orf_fp_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">orf_fp_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"spurbeck-orf-fp.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The ideal approach to removing these false positives would be to add the cow genome to the Kraken database we use for validation of Bowtie2 assignments, along with the pig genome and probably some other known contaminants causing issues. This is probably worth doing at some point, but represents a substantial amount of additional work; I’d rather use a simpler alternative right now if one is available. One option that comes to mind is to try replacing the BBMap aligner I’m using to detect and remove contaminants with Bowtie2; in quick experiments, running Bowtie2 (<code>--sensitive</code>) on the putative Orf virus sequences above, using the same dataset of contaminants for the index, successfully removed about two-thirds of them. While this doesn’t guarantee that Bowtie2 would perform as well on the whole population of contaminating cow sequences, it suggests that using Bowtie2 in place of BBMap is worth trying.</p>
<p>As such, I re-ran the HV detection pipeline again, having made two changes: first, removing Parvovirus NIH-CQV and the parvo-like viruses from the HV database used to identify putative HV reads, and secondly, replacing BBMap with Bowtie2 for contaminant removal.</p>
</section><section id="human-infecting-virus-reads-validation-round-3" class="level1"><h1>Human-infecting virus reads: validation, round 3</h1>
<p>I tried re-running the HV detection pipeline with three different sets of Bowtie2 settings: (a) <code>--sensitive</code>, (b) <code>--local --sensitive-local</code>, and (c) <code>--local --very-sensitive-local</code>. In total, these find 26,370, 15,338 and 13,160 putative human-viral reads, respectively, compared to 31,610 for my initial attempt and 19,799 after the addition of Cutadapt:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_reads_filtered_3_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered_3"</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"b"</span>, <span class="st">"c"</span><span class="op">)</span>, <span class="st">".tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_3_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">hv_reads_filtered_3_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="va">hv_reads_filtered_3_raw</span><span class="op">[[</span><span class="va">n</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>                                <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Bowtie2"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"(a)"</span>,<span class="st">"(b)"</span>,<span class="st">"(c)"</span><span class="op">)</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_3</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span>, <span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_3_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu">bind_rows</span><span class="op">(</span><span class="va">n_hv_filtered_2_summ</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="st">"BBMap"</span><span class="op">)</span>, <span class="va">n_hv_filtered_3_summ</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">attempt</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["attempt"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n_putative"],"name":[2],"type":["int"],"align":["right"]},{"label":["n_total"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["p_reads"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pc_reads"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"BBMap","2":"19799","3":"428705958","4":"4.618317e-05","5":"0.004618317"},{"1":"Bowtie2(a)","2":"26370","3":"428705958","4":"6.151069e-05","5":"0.006151069"},{"1":"Bowtie2(b)","2":"15338","3":"428705958","4":"3.577744e-05","5":"0.003577744"},{"1":"Bowtie2(c)","2":"13160","3":"428705958","4":"3.069703e-05","5":"0.003069703"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Make new labelled HV dataset</span></span>
<span><span class="va">mrg_3</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_3</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge and plot histogram</span></span>
<span><span class="va">mrg_join_3</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="st">"BBMap"</span><span class="op">)</span>, <span class="va">mrg_3</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">g_hist_3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_join_3</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">attempt</span><span class="op">~</span><span class="va">kraken_label</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-hist-3-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>In all cases, the great majority of false-positive sequences from previous attempts were successfully removed (85%/89%/90%, respectively), along with a small number of true-positives (0.19%/0.23%/0.23%, respectively). At the same time, a number of new putative HV sequences arose as a result of the change in filtering algorithm: 8,512/2,224/1,166 respectively. The great majority of these (97.6%/99.1%/98.8%) are low-scoring, suggesting that they are mostly or entirely false matches that are being let through by Bowtie2 that were previously being caught by BBMap.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_2_blast_prep</span> <span class="op">&lt;-</span> <span class="va">mrg_2_blast</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">seq_num</span>, <span class="va">viral_status</span>, <span class="va">viral_status_out</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_old</span> <span class="op">&lt;-</span> <span class="va">mrg_join_3</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_2_blast_prep</span>, by <span class="op">=</span> <span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>viral_status_old <span class="op">=</span> <span class="va">viral_status</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_old_out <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">viral_status_out</span><span class="op">)</span>, <span class="st">"UNKNOWN"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">viral_status_out</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_old_cum</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_old</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>vs_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"HV status:\n"</span>, <span class="va">viral_status_old_out</span><span class="op">)</span>,</span>
<span>         attempt_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Attempt"</span>, <span class="va">attempt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_old_summ</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_old_cum</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">viral_status_old_out</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         vs_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"HV status:"</span>, <span class="va">viral_status_old_out</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_status_summ</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_3_blast_old_summ</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">attempt</span>, y<span class="op">=</span><span class="va">n</span>, fill<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">score_label</span><span class="op">~</span><span class="va">vs_label</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use "none" instead as
of ggplot2 3.3.4.</code></pre>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_status_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-blast-3a-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Checking putative Orf virus reads specifically, all three Bowtie2-based attempts successfully remove most putative Orf reads from the previous attempt, while introducing some number of new putative hits (of which I assume the vast majority are fake hits arising from bovine contamination). In attempt (a), these new putative hits outnumber the old hits that were removed, resulting in a net increase in putative (but, I’m fairly confident, false) Orf-virus hits, whether total or high-scoring. Conversely, attempts (b) and (c) manage to remove even more old putative Orf hits while creating many fewer new ones, resulting in a large net decrease in total and high-scoring putative Orf hits:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">orf_seqs_3</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_old_cum</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">10258</span><span class="op">)</span></span>
<span><span class="va">orf_seqs_3_summ</span> <span class="op">&lt;-</span> <span class="va">orf_seqs_3</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">viral_status_old_out</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span></span>
<span><span class="va">g_orf_total</span> <span class="op">&lt;-</span> <span class="va">orf_seqs_3_summ</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">attempt</span>, y<span class="op">=</span><span class="va">n</span>, fill<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Total reads mapped to Orf virus"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">250</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_orf_high</span> <span class="op">&lt;-</span> <span class="va">orf_seqs_3_summ</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">attempt</span>, y<span class="op">=</span><span class="va">n</span>, fill<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"High-scoring reads mapped to Orf virus"</span>,</span>
<span>                     limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">250</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_orf_total</span> <span class="op">+</span> <span class="va">g_orf_high</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-orf-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Next, I validated all new putative HV sequences with BLASTN as before, then evaluated each attempt’s performance against all the putative HV reads identified by any attempt:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_3_fasta_prep</span> <span class="op">&lt;-</span> <span class="va">mrg_join_3</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">mrg_2_blast</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span></span>
<span><span class="va">mrg_3_fasta</span> <span class="op">&lt;-</span> <span class="va">mrg_3_fasta_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">seq_id</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_3_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_2_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_3_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"spurbeck-putative-viral-3.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import BLAST results (again, pre-filtered to save space)</span></span>
<span><span class="co"># blast_cols &lt;- c("qseqid", "sseqid", "sgi", "staxid", "qlen", "evalue", "bitscore", "qcovs", "length", "pident", "mismatch", "gapopen", "sstrand", "qstart", "qend", "sstart", "send")</span></span>
<span><span class="co"># blast_results_3_path_0 &lt;- file.path(data_dir, "spurbeck-putative-viral-3.blast.gz")</span></span>
<span><span class="co"># blast_results_3 &lt;- read_tsv(blast_results_3_path_0, show_col_types = FALSE,</span></span>
<span><span class="co">#                           col_names = blast_cols, col_types = cols(.default="c"))</span></span>
<span></span>
<span><span class="va">blast_results_3_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"putative-viral-blast-best-3.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_results_3</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_3_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject</span></span>
<span><span class="va">blast_results_3_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each query</span></span>
<span><span class="va">blast_results_3_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_3_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_3_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_3_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span>, </span>
<span>         seq_id <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_3_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_3_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            best_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_3_viral</span> <span class="op">&lt;-</span> <span class="va">blast_results_3_paired</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span>,</span>
<span>         viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_3_assign</span> <span class="op">&lt;-</span> <span class="va">mrg_3_fasta_prep</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span></span>
<span><span class="va">blast_results_3_assign</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">blast_results_3_viral</span>, <span class="va">mrg_3_assign</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_3_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_3_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status_new <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_new <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_new</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_3_blast_new</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_old</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_3_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_new</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">viral_status_old</span>, <span class="va">viral_status_new</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="va">viral_status</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill_fwd</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">seq_id</span>, <span class="va">adj_score_fwd</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"attempt"</span>, values_from<span class="op">=</span><span class="st">"adj_score_fwd"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">seq_id</span>, names_to <span class="op">=</span> <span class="st">"attempt"</span>, values_to <span class="op">=</span> <span class="st">"adj_score_fwd"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill_rev</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">seq_id</span>, <span class="va">adj_score_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"attempt"</span>, values_from<span class="op">=</span><span class="st">"adj_score_rev"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">seq_id</span>, names_to <span class="op">=</span> <span class="st">"attempt"</span>, values_to <span class="op">=</span> <span class="st">"adj_score_rev"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill_ass</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">seq_id</span>, <span class="va">assigned_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"attempt"</span>, values_from<span class="op">=</span><span class="st">"assigned_hv"</span>, values_fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">seq_id</span>, names_to <span class="op">=</span> <span class="st">"attempt"</span>, values_to <span class="op">=</span> <span class="st">"assigned_hv"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill_hit</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">seq_id</span>, <span class="va">hit_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"attempt"</span>, values_from<span class="op">=</span><span class="st">"hit_hv"</span>, values_fill<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">seq_id</span>, names_to <span class="op">=</span> <span class="st">"attempt"</span>, values_to <span class="op">=</span> <span class="st">"hit_hv"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill_ref</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">sample</span>, <span class="va">viral_status_out</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>.groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">sample</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">mrg_3_blast_fill</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_3_blast_fill_fwd</span>, <span class="va">mrg_3_blast_fill_rev</span>, </span>
<span>                              by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"attempt"</span>, <span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_3_blast_fill_ass</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"attempt"</span>, <span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_3_blast_fill_hit</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"attempt"</span>, <span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_3_blast_fill_ref</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">attempts</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_fill</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">summarize</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span></span>
<span><span class="va">stats_3_raw</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast_fill</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">group_split</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">range_f1</span><span class="op">)</span></span>
<span><span class="va">stats_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="va">stats_3_raw</span><span class="op">[[</span><span class="va">n</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="va">attempts</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">bind_rows</span></span>
<span><span class="va">g_stats_3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_3</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">attempt</span>, linetype<span class="op">=</span><span class="va">conj_label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span>, name<span class="op">=</span><span class="st">"Configuration"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Threshold type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">metric</span>, nrow<span class="op">=</span><span class="fl">2</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span>, linetype<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-f1-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>At a disjunctive threshold of 20, Bowtie2(c) (<code>--very-sensitive-local</code>) performs the best, with an F1 score of 0.979; Bowtie2(b) (<code>--sensitive-local</code>) is very close behind. In both cases, precision (&gt;=0.988) is higher than sensitivity (~0.969), suggesting that any further improvements would come from investigating low-scoring true-positives:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">fp_3</span> <span class="op">&lt;-</span> <span class="va">mrg_3_blast</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">viral_status_out</span>, </span>
<span>                           highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, <span class="va">name</span><span class="op">)</span>,</span>
<span>         major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">fp_3_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">major</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_3_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_3_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_3_major</span> <span class="op">&lt;-</span> <span class="va">fp_3</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">attempt</span>, <span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_3_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"VS:"</span>, <span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="co">#ifelse(viral_status_out, "True positive", "False positive"))</span></span>
<span><span class="va">palette_fp_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"Dark2"</span><span class="op">)</span>, <span class="st">"#999999"</span><span class="op">)</span></span>
<span><span class="va">g_fp_3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_3_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_fp_3</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">attempt</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">8</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp_3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-composition-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>I suspect that small additional gains are possible here, since I’m suspicious about the low-scoring “true-positives” mapping to human betaherpesvirus 5 and human mastadenovirus F. However, I’ve already spent a long time optimizing the results for this dataset, and the results are now good enough that I feel okay with leaving this here. Going forward, I’ll use Bowtie2(c) as my alignment strategy for the HV identification pipeline.</p>
</section><section id="human-infecting-virus-reads-analysis" class="level1"><h1>Human-infecting virus reads: analysis</h1>
<p>After several rounds of validation and refinement, we finally come to actually analyzing the human-infecting virus content of Spurbeck et al.&nbsp;This section might seem disappointingly short compared to all the effort expended to get here.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">group</span>, <span class="va">date</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts &amp; RA</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">attempt</span> <span class="op">==</span> <span class="st">"Bowtie2(c)"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_group</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span><span class="va">read_counts_total</span> <span class="op">&lt;-</span> <span class="va">read_counts_group</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         group <span class="op">=</span> <span class="st">"All groups"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_total</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">group</span>, <span class="st">"All groups"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">group</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Applying a disjunctive cutoff at S=20 identifies 9,990 reads as human-viral out of 1.84B total reads, for a relative HV abundance of <span class="math inline">\(5.44 \times 10^{-6}\)</span>. This compares to <span class="math inline">\(2.8 \times 10^{-4}\)</span> on the public dashboard, corresponding to the results for Kraken-only identification: a roughly 2x increase, smaller than the 4-5x increases seen for Crits-Christoph and Rothman. Relative HV abundances for individual sample groups ranged from <span class="math inline">\(9.19 \times 10^{-8}\)</span> to <span class="math inline">\(1.58 \times 10^{-5}\)</span>; as with total virus reads, groups F, G &amp; H showed the highest relative abundance:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">group</span>, y<span class="op">=</span><span class="va">p_reads_hv</span>, color<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span>, name <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-ra-plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Digging into particular viruses, we see that <em>Mamastrovirus</em>, <em>Mastadenovirus, Rotavirus</em> and <em>Betacoronavirus</em> are the most abundant genera across samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694009</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"SARS-CoV"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">568715</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Astrovirus MLB1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194965</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_genera</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count relative abundance for genera</span></span>
<span><span class="va">hv_genera_counts_raw</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genera</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">read_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">group</span>, <span class="va">n_reads_raw</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"group"</span><span class="op">)</span></span>
<span><span class="va">hv_genera_counts_all</span> <span class="op">&lt;-</span> <span class="va">hv_genera_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="st">"All groups"</span><span class="op">)</span></span>
<span><span class="va">hv_genera_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_genera_counts_raw</span>, <span class="va">hv_genera_counts_all</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute ranks for species and genera and restrict to high-ranking taxa</span></span>
<span><span class="va">max_rank_genera</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">hv_genera_counts_ranked</span> <span class="op">&lt;-</span> <span class="va">hv_genera_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, ties.method<span class="op">=</span><span class="st">"max"</span><span class="op">)</span>,</span>
<span>         highrank <span class="op">=</span> <span class="va">rank</span> <span class="op">&lt;=</span> <span class="va">max_rank_genera</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>highrank_any <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">highrank</span><span class="op">)</span>,</span>
<span>         name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highrank_any</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>mean_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">mean_rank</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">group</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">group</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot composition</span></span>
<span><span class="va">palette_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span>, <span class="st">"#888888"</span><span class="op">)</span></span>
<span><span class="va">g_vcomp_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_genera_counts_ranked</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">group</span>, y<span class="op">=</span><span class="va">n_reads_hv</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_rank</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Fraction of HV reads"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_vcomp_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-composition-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><em>Mamastrovirus</em> and <em>Rotavirus</em> are predominantly enteric RNA viruses whose prominence here makes sense, while the prevalence of <em>Betacoronavirus</em> is probably a result of the ongoing SARS-CoV-2 pandemic. As a DNA virus, the abundance of <em>Mastadenovirus</em> is a bit more surprising; however, this finding is consistent with the public dashboard, and is also borne out by the other BLASTN matches for these sequences, all of which are other adenovirus taxa:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">masta_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genera</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span><span class="op">==</span><span class="st">"Mastadenovirus"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span><span class="va">masta_hits</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">blast_results_2_paired</span> <span class="op">%&gt;%</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_2_blast</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                        <span class="va">blast_results_3_paired</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, <span class="op">-</span><span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">masta_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">masta_hits_sorted</span> <span class="op">&lt;-</span> <span class="va">masta_hits</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">masta_hits_sorted_head</span> <span class="op">&lt;-</span> <span class="va">masta_hits_sorted</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_masta</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">masta_hits_sorted_head</span>,</span>
<span>                  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">n</span>, y<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"# Mapped read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_masta</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-01_spurbeck_files/figure-html/hv-mastadenovirus-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>Compared to the last few datasets I analyzed, the analysis of Spurbeck took a long time, numerous attempts, and a lot of computational resources. However, as I said at the start of this post, I’m happy with the outcome and am confident it will improve analysis of future datasets. While the overall prevalence of human-infecting viruses is fairly low in Spurbeck compared to other wastewater datasets I’ve looked at, its inclusion as a core dataset for the P2RA analysis make it especially important to process in a reliable and high-quality manner.</p>
<p>Next, I’ll turn my attention to more datasets included in the P2RA analysis, as well as some air-sampling datasets we’re interested in for another project.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb54" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Spurbeck et al. (2023)"</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Cave carpa."</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-04-01</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a>Continuing my analysis of datasets from the <span class="co">[</span><span class="ot">P2RA preprint</span><span class="co">](https://doi.org/10.1101/2023.12.22.23300450)</span>, I analyzed the data from Spurbeck et al. (2023), a wastewater RNA-sequencing study from Ohio. Samples for this study underwent a variety of processing protocols at different research centers across the state. Along with Rothman and Crits-Christoph, Spurbeck is one of three RNA-sequencing studies that underwent full in-depth analysis in the P2RA study, so is worth looking at closely here.</span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a>This one turned out to be a bit of a saga. As we'll see in the section on human-virus identification, it took multiple tries and several substantial changes to the pipeline to get things to a state I was happy with. Still, I am happy with the outcome and think the changes will improve analysis of future datasets.</span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a>The Spurbeck data comprises 55 samples from 8 processing groups, with 4 to 6 samples per group. The number of sequencing read pairs per sample varied widely from 4.5M-106.7M (mean 33.4M). Taken together, the samples totaled roughly 1.8B read pairs (425 gigabases of sequence). Read qualities were generally high but in need of some light preprocessing. Adapter levels were moderate. Inferred duplication levels were fairly high: 14-91% with a mean of 55%.</span>
<span id="cb54-42"><a href="#cb54-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-45"><a href="#cb54-45" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-46"><a href="#cb54-46" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-47"><a href="#cb54-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb54-48"><a href="#cb54-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb54-49"><a href="#cb54-49" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-04-01_spurbeck/"</span></span>
<span id="cb54-50"><a href="#cb54-50" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb54-51"><a href="#cb54-51" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats_1.tsv"</span>)</span>
<span id="cb54-52"><a href="#cb54-52" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_adapter_stats.tsv"</span>)</span>
<span id="cb54-53"><a href="#cb54-53" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_base_stats.tsv"</span>)</span>
<span id="cb54-54"><a href="#cb54-54" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_sequence_stats.tsv"</span>)</span>
<span id="cb54-55"><a href="#cb54-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-56"><a href="#cb54-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb54-57"><a href="#cb54-57" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-58"><a href="#cb54-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-59"><a href="#cb54-59" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb54-60"><a href="#cb54-60" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb54-61"><a href="#cb54-61" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-62"><a href="#cb54-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span id="cb54-63"><a href="#cb54-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-64"><a href="#cb54-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb54-65"><a href="#cb54-65" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb54-66"><a href="#cb54-66" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-67"><a href="#cb54-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb54-68"><a href="#cb54-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-69"><a href="#cb54-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb54-70"><a href="#cb54-70" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb54-71"><a href="#cb54-71" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-72"><a href="#cb54-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span id="cb54-73"><a href="#cb54-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-74"><a href="#cb54-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb54-75"><a href="#cb54-75" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb54-76"><a href="#cb54-76" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-77"><a href="#cb54-77" aria-hidden="true" tabindex="-1"></a>    <span class="co"># mutate(sample = sub("_2$", "", sample)) %&gt;%</span></span>
<span id="cb54-78"><a href="#cb54-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-79"><a href="#cb54-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb54-80"><a href="#cb54-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb54-81"><a href="#cb54-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-82"><a href="#cb54-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb54-83"><a href="#cb54-83" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb54-84"><a href="#cb54-84" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb54-85"><a href="#cb54-85" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb54-86"><a href="#cb54-86" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb54-87"><a href="#cb54-87" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-88"><a href="#cb54-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-91"><a href="#cb54-91" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-92"><a href="#cb54-92" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb54-93"><a href="#cb54-93" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-94"><a href="#cb54-94" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb54-95"><a href="#cb54-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-96"><a href="#cb54-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize basic stats</span></span>
<span id="cb54-97"><a href="#cb54-97" aria-hidden="true" tabindex="-1"></a>scale_fill_grp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Set3"</span>, <span class="at">name=</span><span class="st">"Processing group"</span>)</span>
<span id="cb54-98"><a href="#cb54-98" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">fill=</span>group, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb54-99"><a href="#cb54-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_grp</span>() <span class="sc">+</span> theme_kit</span>
<span id="cb54-100"><a href="#cb54-100" aria-hidden="true" tabindex="-1"></a>g_nreads_raw <span class="ot">&lt;-</span> g_basic <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs), <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb54-101"><a href="#cb54-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb54-102"><a href="#cb54-102" aria-hidden="true" tabindex="-1"></a>g_nreads_raw_2 <span class="ot">&lt;-</span> g_nreads_raw <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb54-103"><a href="#cb54-103" aria-hidden="true" tabindex="-1"></a>legend_group <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_nreads_raw)</span>
<span id="cb54-104"><a href="#cb54-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-105"><a href="#cb54-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-106"><a href="#cb54-106" aria-hidden="true" tabindex="-1"></a>g_nbases_raw <span class="ot">&lt;-</span> g_basic <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_bases_approx), <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb54-107"><a href="#cb54-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Total base pairs (approx)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span> </span>
<span id="cb54-108"><a href="#cb54-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb54-109"><a href="#cb54-109" aria-hidden="true" tabindex="-1"></a>g_ndup_raw <span class="ot">&lt;-</span> g_basic <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates), <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb54-110"><a href="#cb54-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Duplicates (FASTQC)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb54-111"><a href="#cb54-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb54-112"><a href="#cb54-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-113"><a href="#cb54-113" aria-hidden="true" tabindex="-1"></a>g_basic_raw <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(g_nreads_raw_2 <span class="sc">+</span> g_nbases_raw <span class="sc">+</span> g_ndup_raw, legend_group, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.1</span>))</span>
<span id="cb54-114"><a href="#cb54-114" aria-hidden="true" tabindex="-1"></a>g_basic_raw</span>
<span id="cb54-115"><a href="#cb54-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-116"><a href="#cb54-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-119"><a href="#cb54-119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-120"><a href="#cb54-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb54-121"><a href="#cb54-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-122"><a href="#cb54-122" aria-hidden="true" tabindex="-1"></a>scale_color_grp <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer,<span class="at">palette=</span><span class="st">"Set3"</span>,<span class="at">name=</span><span class="st">"Processing group"</span>)</span>
<span id="cb54-123"><a href="#cb54-123" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>group, <span class="at">linetype=</span>read_pair, </span>
<span id="cb54-124"><a href="#cb54-124" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb54-125"><a href="#cb54-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_grp</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb54-126"><a href="#cb54-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb54-127"><a href="#cb54-127" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb54-128"><a href="#cb54-128" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-129"><a href="#cb54-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-130"><a href="#cb54-130" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb54-131"><a href="#cb54-131" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb54-132"><a href="#cb54-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb54-133"><a href="#cb54-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>),</span>
<span id="cb54-134"><a href="#cb54-134" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-135"><a href="#cb54-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb54-136"><a href="#cb54-136" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-137"><a href="#cb54-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>adapter)</span>
<span id="cb54-138"><a href="#cb54-138" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb54-139"><a href="#cb54-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-140"><a href="#cb54-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb54-141"><a href="#cb54-141" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb54-142"><a href="#cb54-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-143"><a href="#cb54-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-144"><a href="#cb54-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb54-145"><a href="#cb54-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb54-146"><a href="#cb54-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb54-147"><a href="#cb54-147" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb54-148"><a href="#cb54-148" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb54-149"><a href="#cb54-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-150"><a href="#cb54-150" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb54-151"><a href="#cb54-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-152"><a href="#cb54-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-153"><a href="#cb54-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb54-154"><a href="#cb54-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-155"><a href="#cb54-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb54-156"><a href="#cb54-156" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb54-157"><a href="#cb54-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-158"><a href="#cb54-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-159"><a href="#cb54-159" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb54-160"><a href="#cb54-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-161"><a href="#cb54-161" aria-hidden="true" tabindex="-1"></a>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Significant numbers of reads were lost at each stage in the preprocessing pipeline. In total, cleaning, deduplication and initial ribodepletion removed 17-96% (mean 72%) of input reads, while secondary ribodepletion removed an additional 0-11% (mean 6%).</span>
<span id="cb54-162"><a href="#cb54-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-165"><a href="#cb54-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-166"><a href="#cb54-166" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb54-167"><a href="#cb54-167" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb54-168"><a href="#cb54-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, group, stage, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb54-169"><a href="#cb54-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, group) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, group, stage) <span class="sc">%&gt;%</span></span>
<span id="cb54-170"><a href="#cb54-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb54-171"><a href="#cb54-171" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb54-172"><a href="#cb54-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb54-173"><a href="#cb54-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb54-174"><a href="#cb54-174" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs))</span>
<span id="cb54-175"><a href="#cb54-175" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb54-176"><a href="#cb54-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb54-177"><a href="#cb54-177" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>)) <span class="sc">%&gt;%</span> <span class="fu">tail</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-178"><a href="#cb54-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb54-179"><a href="#cb54-179" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb54-180"><a href="#cb54-180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-181"><a href="#cb54-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-184"><a href="#cb54-184" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-185"><a href="#cb54-185" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb54-186"><a href="#cb54-186" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-187"><a href="#cb54-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb54-188"><a href="#cb54-188" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>n_read_pairs,<span class="at">fill=</span>group,<span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb54-189"><a href="#cb54-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_grp</span>() <span class="sc">+</span></span>
<span id="cb54-190"><a href="#cb54-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-191"><a href="#cb54-191" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-192"><a href="#cb54-192" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb54-193"><a href="#cb54-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-194"><a href="#cb54-194" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb54-195"><a href="#cb54-195" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>p_reads_lost_abs_marginal,<span class="at">fill=</span>group,<span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb54-196"><a href="#cb54-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_grp</span>() <span class="sc">+</span></span>
<span id="cb54-197"><a href="#cb54-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb54-198"><a href="#cb54-198" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-199"><a href="#cb54-199" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb54-200"><a href="#cb54-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-201"><a href="#cb54-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-202"><a href="#cb54-202" aria-hidden="true" tabindex="-1"></a>Data cleaning with FASTP was mostly successful at removing adapters; however, detectable levels of Illumina Universal Adapter sequences persisted through the preprocessing pipeline. FASTP was successful at improving read quality.</span>
<span id="cb54-203"><a href="#cb54-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-206"><a href="#cb54-206" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-207"><a href="#cb54-207" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-208"><a href="#cb54-208" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb54-209"><a href="#cb54-209" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb54-210"><a href="#cb54-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-211"><a href="#cb54-211" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>group, <span class="at">linetype=</span>read_pair, </span>
<span id="cb54-212"><a href="#cb54-212" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb54-213"><a href="#cb54-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_grp</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb54-214"><a href="#cb54-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb54-215"><a href="#cb54-215" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb54-216"><a href="#cb54-216" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-217"><a href="#cb54-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-218"><a href="#cb54-218" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb54-219"><a href="#cb54-219" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb54-220"><a href="#cb54-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb54-221"><a href="#cb54-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb54-222"><a href="#cb54-222" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-223"><a href="#cb54-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb54-224"><a href="#cb54-224" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-225"><a href="#cb54-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb54-226"><a href="#cb54-226" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb54-227"><a href="#cb54-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-228"><a href="#cb54-228" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb54-229"><a href="#cb54-229" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb54-230"><a href="#cb54-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-231"><a href="#cb54-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-232"><a href="#cb54-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb54-233"><a href="#cb54-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb54-234"><a href="#cb54-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb54-235"><a href="#cb54-235" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-236"><a href="#cb54-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb54-237"><a href="#cb54-237" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb54-238"><a href="#cb54-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-239"><a href="#cb54-239" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb54-240"><a href="#cb54-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-241"><a href="#cb54-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb54-242"><a href="#cb54-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb54-243"><a href="#cb54-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-244"><a href="#cb54-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-245"><a href="#cb54-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb54-246"><a href="#cb54-246" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb54-247"><a href="#cb54-247" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-248"><a href="#cb54-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-249"><a href="#cb54-249" aria-hidden="true" tabindex="-1"></a>According to FASTQC, deduplication was quite effective at reducing measured duplicate levels, with FASTQC-measured levels falling from an average of 55% to one of 22%:</span>
<span id="cb54-250"><a href="#cb54-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-253"><a href="#cb54-253" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-254"><a href="#cb54-254" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb54-255"><a href="#cb54-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-256"><a href="#cb54-256" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>percent_duplicates, <span class="at">fill=</span>group, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb54-257"><a href="#cb54-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_grp</span>() <span class="sc">+</span></span>
<span id="cb54-258"><a href="#cb54-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-259"><a href="#cb54-259" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-260"><a href="#cb54-260" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>mean_seq_len, <span class="at">fill=</span>group, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb54-261"><a href="#cb54-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_grp</span>() <span class="sc">+</span></span>
<span id="cb54-262"><a href="#cb54-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-263"><a href="#cb54-263" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-264"><a href="#cb54-264" aria-hidden="true" tabindex="-1"></a>legend_loc <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_dup_stages)</span>
<span id="cb54-265"><a href="#cb54-265" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb54-266"><a href="#cb54-266" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb54-267"><a href="#cb54-267" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-268"><a href="#cb54-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-269"><a href="#cb54-269" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb54-270"><a href="#cb54-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-271"><a href="#cb54-271" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb54-272"><a href="#cb54-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-275"><a href="#cb54-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-276"><a href="#cb54-276" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb54-277"><a href="#cb54-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-278"><a href="#cb54-278" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bracken data</span></span>
<span id="cb54-279"><a href="#cb54-279" aria-hidden="true" tabindex="-1"></a>bracken_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"bracken_counts.tsv"</span>)</span>
<span id="cb54-280"><a href="#cb54-280" aria-hidden="true" tabindex="-1"></a>bracken <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(bracken_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-281"><a href="#cb54-281" aria-hidden="true" tabindex="-1"></a>total_assigned <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(</span>
<span id="cb54-282"><a href="#cb54-282" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Total"</span>,</span>
<span id="cb54-283"><a href="#cb54-283" aria-hidden="true" tabindex="-1"></a>  <span class="at">kraken_assigned_reads =</span> <span class="fu">sum</span>(kraken_assigned_reads),</span>
<span id="cb54-284"><a href="#cb54-284" aria-hidden="true" tabindex="-1"></a>  <span class="at">added_reads =</span> <span class="fu">sum</span>(added_reads),</span>
<span id="cb54-285"><a href="#cb54-285" aria-hidden="true" tabindex="-1"></a>  <span class="at">new_est_reads =</span> <span class="fu">sum</span>(new_est_reads),</span>
<span id="cb54-286"><a href="#cb54-286" aria-hidden="true" tabindex="-1"></a>  <span class="at">fraction_total_reads =</span> <span class="fu">sum</span>(fraction_total_reads)</span>
<span id="cb54-287"><a href="#cb54-287" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb54-288"><a href="#cb54-288" aria-hidden="true" tabindex="-1"></a>bracken_spread <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">select</span>(name, sample, new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb54-289"><a href="#cb54-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">tolower</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb54-290"><a href="#cb54-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"sample"</span>, <span class="at">names_from =</span> <span class="st">"name"</span>, <span class="at">values_from =</span> <span class="st">"new_est_reads"</span>)</span>
<span id="cb54-291"><a href="#cb54-291" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb54-292"><a href="#cb54-292" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb54-293"><a href="#cb54-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, group, stage, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb54-294"><a href="#cb54-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"group"</span>), <span class="at">names_from=</span><span class="st">"stage"</span>, <span class="at">values_from=</span><span class="st">"n_read_pairs"</span>)</span>
<span id="cb54-295"><a href="#cb54-295" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_preproc <span class="sc">%&gt;%</span></span>
<span id="cb54-296"><a href="#cb54-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total_assigned <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, new_est_reads), <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-297"><a href="#cb54-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">assigned =</span> new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb54-298"><a href="#cb54-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(bracken_spread, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb54-299"><a href="#cb54-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Assess composition</span></span>
<span id="cb54-300"><a href="#cb54-300" aria-hidden="true" tabindex="-1"></a>read_comp <span class="ot">&lt;-</span> <span class="fu">transmute</span>(read_counts, <span class="at">sample=</span>sample, <span class="at">group=</span>group,</span>
<span id="cb54-301"><a href="#cb54-301" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_filtered =</span> raw_concat<span class="sc">-</span>cleaned,</span>
<span id="cb54-302"><a href="#cb54-302" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_duplicate =</span> cleaned<span class="sc">-</span>dedup,</span>
<span id="cb54-303"><a href="#cb54-303" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_ribosomal =</span> (dedup<span class="sc">-</span>ribo_initial) <span class="sc">+</span> (ribo_initial<span class="sc">-</span>ribo_secondary),</span>
<span id="cb54-304"><a href="#cb54-304" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_unassigned =</span> ribo_secondary<span class="sc">-</span>assigned,</span>
<span id="cb54-305"><a href="#cb54-305" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_bacterial =</span> bacteria,</span>
<span id="cb54-306"><a href="#cb54-306" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_archaeal =</span> archaea,</span>
<span id="cb54-307"><a href="#cb54-307" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_viral =</span> viruses,</span>
<span id="cb54-308"><a href="#cb54-308" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_human =</span> eukaryota)</span>
<span id="cb54-309"><a href="#cb54-309" aria-hidden="true" tabindex="-1"></a>read_comp_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(read_comp, <span class="sc">-</span>(sample<span class="sc">:</span>group), <span class="at">names_to =</span> <span class="st">"classification"</span>,</span>
<span id="cb54-310"><a href="#cb54-310" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_prefix =</span> <span class="st">"n_"</span>, <span class="at">values_to =</span> <span class="st">"n_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-311"><a href="#cb54-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(<span class="fu">str_to_sentence</span>(classification))) <span class="sc">%&gt;%</span></span>
<span id="cb54-312"><a href="#cb54-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb54-313"><a href="#cb54-313" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb54-314"><a href="#cb54-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group, classification) <span class="sc">%&gt;%</span></span>
<span id="cb54-315"><a href="#cb54-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-316"><a href="#cb54-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb54-317"><a href="#cb54-317" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb54-318"><a href="#cb54-318" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb54-319"><a href="#cb54-319" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-320"><a href="#cb54-320" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb54-321"><a href="#cb54-321" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_comp_long, <span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb54-322"><a href="#cb54-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb54-323"><a href="#cb54-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Reads</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb54-324"><a href="#cb54-324" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb54-325"><a href="#cb54-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb54-326"><a href="#cb54-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>group, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb54-327"><a href="#cb54-327" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-328"><a href="#cb54-328" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb54-329"><a href="#cb54-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-330"><a href="#cb54-330" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb54-331"><a href="#cb54-331" aria-hidden="true" tabindex="-1"></a>read_comp_minor <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb54-332"><a href="#cb54-332" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb54-333"><a href="#cb54-333" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_comp_minor, <span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb54-334"><a href="#cb54-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb54-335"><a href="#cb54-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Reads</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb54-336"><a href="#cb54-336" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb54-337"><a href="#cb54-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb54-338"><a href="#cb54-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>group, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb54-339"><a href="#cb54-339" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-340"><a href="#cb54-340" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb54-341"><a href="#cb54-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-342"><a href="#cb54-342" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-343"><a href="#cb54-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-346"><a href="#cb54-346" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-347"><a href="#cb54-347" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-table</span></span>
<span id="cb54-348"><a href="#cb54-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-349"><a href="#cb54-349" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span></span>
<span id="cb54-350"><a href="#cb54-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb54-351"><a href="#cb54-351" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb54-352"><a href="#cb54-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, group) <span class="sc">%&gt;%</span></span>
<span id="cb54-353"><a href="#cb54-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-354"><a href="#cb54-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, group) <span class="sc">%&gt;%</span></span>
<span id="cb54-355"><a href="#cb54-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb54-356"><a href="#cb54-356" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb54-357"><a href="#cb54-357" aria-hidden="true" tabindex="-1"></a>p_reads_summ_total <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span></span>
<span id="cb54-358"><a href="#cb54-358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb54-359"><a href="#cb54-359" aria-hidden="true" tabindex="-1"></a>                  <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb54-360"><a href="#cb54-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, group) <span class="sc">%&gt;%</span></span>
<span id="cb54-361"><a href="#cb54-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-362"><a href="#cb54-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb54-363"><a href="#cb54-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb54-364"><a href="#cb54-364" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb54-365"><a href="#cb54-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"All groups"</span>)</span>
<span id="cb54-366"><a href="#cb54-366" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(p_reads_summ_total, p_reads_summ_group) <span class="sc">%&gt;%</span></span>
<span id="cb54-367"><a href="#cb54-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">fct_inorder</span>(group),</span>
<span id="cb54-368"><a href="#cb54-368" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb54-369"><a href="#cb54-369" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb54-370"><a href="#cb54-370" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb54-371"><a href="#cb54-371" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb54-372"><a href="#cb54-372" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb54-373"><a href="#cb54-373" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb54-374"><a href="#cb54-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(classification, group, display) <span class="sc">%&gt;%</span></span>
<span id="cb54-375"><a href="#cb54-375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>group, <span class="at">values_from =</span> display)</span>
<span id="cb54-376"><a href="#cb54-376" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb54-377"><a href="#cb54-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-378"><a href="#cb54-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-379"><a href="#cb54-379" aria-hidden="true" tabindex="-1"></a>Average composition varied substantially between groups. Four groups (A, B, I, J) showed high within-group consistency, with high levels of ribosomal reads and very low levels of assigned reads. Other groups showed much more variability between samples, with higher average levels of assigned reads.</span>
<span id="cb54-380"><a href="#cb54-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-381"><a href="#cb54-381" aria-hidden="true" tabindex="-1"></a>Overall, the average relative abundance of viral reads was 0.04%; however, groups F, G &amp; H showed substantially higher average abundance of around 0.1%. Even these elevated groups, however, still show lower total viral abundance than <span class="co">[</span><span class="ot">Rothman</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-02-27_rothman-1.html)</span> or <span class="co">[</span><span class="ot">Crits-Christoph</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-02-04_crits-christoph-1.html)</span>, so this doesn't seem particularly noteworthy.</span>
<span id="cb54-382"><a href="#cb54-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-383"><a href="#cb54-383" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation, round 1</span></span>
<span id="cb54-384"><a href="#cb54-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-385"><a href="#cb54-385" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples, using a pipeline identical to that described in my <span class="co">[</span><span class="ot">entry on unenriched Rothman samples</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-02-27_rothman-1.html)</span>. This process identified a total of 31,610 read pairs across all samples (0.007% of surviving reads):</span>
<span id="cb54-386"><a href="#cb54-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-389"><a href="#cb54-389" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-390"><a href="#cb54-390" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-nreads-1</span></span>
<span id="cb54-391"><a href="#cb54-391" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_1_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered_1.tsv.gz"</span>)</span>
<span id="cb54-392"><a href="#cb54-392" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_1 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_1_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-393"><a href="#cb54-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-394"><a href="#cb54-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb54-395"><a href="#cb54-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb54-396"><a href="#cb54-396" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_1 <span class="ot">&lt;-</span> hv_reads_filtered_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, group) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb54-397"><a href="#cb54-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-398"><a href="#cb54-398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb54-399"><a href="#cb54-399" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_1_summ <span class="ot">&lt;-</span> n_hv_filtered_1 <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb54-400"><a href="#cb54-400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-401"><a href="#cb54-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-404"><a href="#cb54-404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-405"><a href="#cb54-405" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-406"><a href="#cb54-406" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb54-407"><a href="#cb54-407" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-hist-1</span></span>
<span id="cb54-408"><a href="#cb54-408" aria-hidden="true" tabindex="-1"></a>mrg_1 <span class="ot">&lt;-</span> hv_reads_filtered_1 <span class="sc">%&gt;%</span></span>
<span id="cb54-409"><a href="#cb54-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb54-410"><a href="#cb54-410" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb54-411"><a href="#cb54-411" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb54-412"><a href="#cb54-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb54-413"><a href="#cb54-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(),</span>
<span id="cb54-414"><a href="#cb54-414" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb54-415"><a href="#cb54-415" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span id="cb54-416"><a href="#cb54-416" aria-hidden="true" tabindex="-1"></a>g_mrg_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_1, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev)) <span class="sc">+</span></span>
<span id="cb54-417"><a href="#cb54-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb54-418"><a href="#cb54-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-419"><a href="#cb54-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-420"><a href="#cb54-420" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb54-421"><a href="#cb54-421" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb54-422"><a href="#cb54-422" aria-hidden="true" tabindex="-1"></a>g_mrg_1</span>
<span id="cb54-423"><a href="#cb54-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-424"><a href="#cb54-424" aria-hidden="true" tabindex="-1"></a>g_hist_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_1, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb54-425"><a href="#cb54-425" aria-hidden="true" tabindex="-1"></a>g_hist_1</span>
<span id="cb54-426"><a href="#cb54-426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-427"><a href="#cb54-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-428"><a href="#cb54-428" aria-hidden="true" tabindex="-1"></a>To analyze all these reads in a reasonable amount of time, I set up a dedicated EC2 instance, downloaded nt, and ran BLASTN locally there, otherwise using the same parameters I've used for past datasets:</span>
<span id="cb54-429"><a href="#cb54-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-432"><a href="#cb54-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-433"><a href="#cb54-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-fasta-1</span></span>
<span id="cb54-434"><a href="#cb54-434" aria-hidden="true" tabindex="-1"></a>mrg_1_num <span class="ot">&lt;-</span> mrg_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb54-435"><a href="#cb54-435" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb54-436"><a href="#cb54-436" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, sample, <span class="st">"_"</span>, seq_num))</span>
<span id="cb54-437"><a href="#cb54-437" aria-hidden="true" tabindex="-1"></a>mrg_1_fasta <span class="ot">&lt;-</span>  mrg_1_num <span class="sc">%&gt;%</span> </span>
<span id="cb54-438"><a href="#cb54-438" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-439"><a href="#cb54-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-440"><a href="#cb54-440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb54-441"><a href="#cb54-441" aria-hidden="true" tabindex="-1"></a>mrg_1_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_1_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-442"><a href="#cb54-442" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_1_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"spurbeck-putative-viral-1.fasta"</span>))</span>
<span id="cb54-443"><a href="#cb54-443" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-444"><a href="#cb54-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-447"><a href="#cb54-447" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-448"><a href="#cb54-448" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-449"><a href="#cb54-449" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-1</span></span>
<span id="cb54-450"><a href="#cb54-450" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb54-451"><a href="#cb54-451" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-452"><a href="#cb54-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-453"><a href="#cb54-453" aria-hidden="true" tabindex="-1"></a><span class="co"># NB: Importing partially-processed BLAST results to save storage space</span></span>
<span id="cb54-454"><a href="#cb54-454" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb54-455"><a href="#cb54-455" aria-hidden="true" tabindex="-1"></a>blast_results_1_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"putative-viral-blast-best-1.tsv.gz"</span>)</span>
<span id="cb54-456"><a href="#cb54-456" aria-hidden="true" tabindex="-1"></a>blast_results_1 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_1_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb54-457"><a href="#cb54-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-458"><a href="#cb54-458" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject</span></span>
<span id="cb54-459"><a href="#cb54-459" aria-hidden="true" tabindex="-1"></a>blast_results_1_best <span class="ot">&lt;-</span> blast_results_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> </span>
<span id="cb54-460"><a href="#cb54-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb54-461"><a href="#cb54-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb54-462"><a href="#cb54-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-463"><a href="#cb54-463" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each query</span></span>
<span id="cb54-464"><a href="#cb54-464" aria-hidden="true" tabindex="-1"></a>blast_results_1_ranked <span class="ot">&lt;-</span> blast_results_1_best <span class="sc">%&gt;%</span> </span>
<span id="cb54-465"><a href="#cb54-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb54-466"><a href="#cb54-466" aria-hidden="true" tabindex="-1"></a>blast_results_1_highrank <span class="ot">&lt;-</span> blast_results_1_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-467"><a href="#cb54-467" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>), </span>
<span id="cb54-468"><a href="#cb54-468" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_num =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>), </span>
<span id="cb54-469"><a href="#cb54-469" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(head, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-470"><a href="#cb54-470" aria-hidden="true" tabindex="-1"></a>           <span class="fu">sapply</span>(paste, <span class="at">collapse=</span><span class="st">"_"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-471"><a href="#cb54-471" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num))</span>
<span id="cb54-472"><a href="#cb54-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-473"><a href="#cb54-473" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb54-474"><a href="#cb54-474" aria-hidden="true" tabindex="-1"></a>blast_results_1_paired <span class="ot">&lt;-</span> blast_results_1_highrank <span class="sc">%&gt;%</span></span>
<span id="cb54-475"><a href="#cb54-475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb54-476"><a href="#cb54-476" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb54-477"><a href="#cb54-477" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_rank =</span> <span class="fu">min</span>(rank),</span>
<span id="cb54-478"><a href="#cb54-478" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb54-479"><a href="#cb54-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-480"><a href="#cb54-480" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb54-481"><a href="#cb54-481" aria-hidden="true" tabindex="-1"></a>blast_results_1_viral <span class="ot">&lt;-</span> blast_results_1_paired <span class="sc">%&gt;%</span></span>
<span id="cb54-482"><a href="#cb54-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid,</span>
<span id="cb54-483"><a href="#cb54-483" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb54-484"><a href="#cb54-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-485"><a href="#cb54-485" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb54-486"><a href="#cb54-486" aria-hidden="true" tabindex="-1"></a>mrg_1_assign <span class="ot">&lt;-</span> mrg_1_num <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, taxid, assigned_taxid)</span>
<span id="cb54-487"><a href="#cb54-487" aria-hidden="true" tabindex="-1"></a>blast_results_1_assign <span class="ot">&lt;-</span> <span class="fu">full_join</span>(blast_results_1_viral, mrg_1_assign, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-488"><a href="#cb54-488" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb54-489"><a href="#cb54-489" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb54-490"><a href="#cb54-490" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb54-491"><a href="#cb54-491" aria-hidden="true" tabindex="-1"></a>blast_results_1_out <span class="ot">&lt;-</span> blast_results_1_assign <span class="sc">%&gt;%</span></span>
<span id="cb54-492"><a href="#cb54-492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb54-493"><a href="#cb54-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb54-494"><a href="#cb54-494" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb54-495"><a href="#cb54-495" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb54-496"><a href="#cb54-496" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-497"><a href="#cb54-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>))</span>
<span id="cb54-498"><a href="#cb54-498" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-499"><a href="#cb54-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-502"><a href="#cb54-502" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-503"><a href="#cb54-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb54-504"><a href="#cb54-504" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-505"><a href="#cb54-505" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-scatter-1</span></span>
<span id="cb54-506"><a href="#cb54-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-507"><a href="#cb54-507" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb54-508"><a href="#cb54-508" aria-hidden="true" tabindex="-1"></a>mrg_1_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_1_num, blast_results_1_out, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-509"><a href="#cb54-509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb54-510"><a href="#cb54-510" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb54-511"><a href="#cb54-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-512"><a href="#cb54-512" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb54-513"><a href="#cb54-513" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_1 <span class="ot">&lt;-</span> mrg_1_blast <span class="sc">%&gt;%</span></span>
<span id="cb54-514"><a href="#cb54-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb54-515"><a href="#cb54-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb54-516"><a href="#cb54-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-517"><a href="#cb54-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-518"><a href="#cb54-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb54-519"><a href="#cb54-519" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(viral_status_out<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb54-520"><a href="#cb54-520" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb54-521"><a href="#cb54-521" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_1</span>
<span id="cb54-522"><a href="#cb54-522" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-523"><a href="#cb54-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-526"><a href="#cb54-526" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-527"><a href="#cb54-527" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-hist-1</span></span>
<span id="cb54-528"><a href="#cb54-528" aria-hidden="true" tabindex="-1"></a>g_hist_blast_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_1_blast, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb54-529"><a href="#cb54-529" aria-hidden="true" tabindex="-1"></a>g_hist_blast_1</span>
<span id="cb54-530"><a href="#cb54-530" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-531"><a href="#cb54-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-532"><a href="#cb54-532" aria-hidden="true" tabindex="-1"></a>There results were...okay. There were a lot of false positives, but nearly all of them were low-scoring and excluded by my usual score filters. Most true positives, meanwhile, had high enough scores to be retained by those filters. However, there were enough high-scoring false positives and low-scoring true positives to drag down my precision and sensitivity, resulting in an F1 score (at a disjunctive score threshold of 20) a little under 90% – quite a few percentage points lower than I usually aim for.</span>
<span id="cb54-533"><a href="#cb54-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-536"><a href="#cb54-536" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-537"><a href="#cb54-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-f1-1</span></span>
<span id="cb54-538"><a href="#cb54-538" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold, conjunctive, include_special){</span>
<span id="cb54-539"><a href="#cb54-539" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>include_special) tab <span class="ot">&lt;-</span> <span class="fu">filter</span>(tab, viral_status_out <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>))</span>
<span id="cb54-540"><a href="#cb54-540" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb54-541"><a href="#cb54-541" aria-hidden="true" tabindex="-1"></a>    <span class="at">conjunctive =</span> conjunctive,</span>
<span id="cb54-542"><a href="#cb54-542" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_conjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">&amp;</span> adj_score_rev <span class="sc">&gt;</span> score_threshold), </span>
<span id="cb54-543"><a href="#cb54-543" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_disjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb54-544"><a href="#cb54-544" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score =</span> <span class="fu">ifelse</span>(conjunctive, retain_score_conjunctive, retain_score_disjunctive),</span>
<span id="cb54-545"><a href="#cb54-545" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb54-546"><a href="#cb54-546" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb54-547"><a href="#cb54-547" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb54-548"><a href="#cb54-548" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb54-549"><a href="#cb54-549" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb54-550"><a href="#cb54-550" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb54-551"><a href="#cb54-551" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb54-552"><a href="#cb54-552" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb54-553"><a href="#cb54-553" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb54-554"><a href="#cb54-554" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb54-555"><a href="#cb54-555" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">include_special =</span> include_special, </span>
<span id="cb54-556"><a href="#cb54-556" aria-hidden="true" tabindex="-1"></a>                <span class="at">conjunctive =</span> conjunctive, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb54-557"><a href="#cb54-557" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb54-558"><a href="#cb54-558" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb54-559"><a href="#cb54-559" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb54-560"><a href="#cb54-560" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, <span class="at">inc_special=</span><span class="cn">FALSE</span>, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb54-561"><a href="#cb54-561" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab, <span class="at">include_special=</span>inc_special)</span>
<span id="cb54-562"><a href="#cb54-562" aria-hidden="true" tabindex="-1"></a>  stats_conj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb54-563"><a href="#cb54-563" aria-hidden="true" tabindex="-1"></a>  stats_disj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb54-564"><a href="#cb54-564" aria-hidden="true" tabindex="-1"></a>  stats_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_conj, stats_disj) <span class="sc">%&gt;%</span></span>
<span id="cb54-565"><a href="#cb54-565" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>(threshold<span class="sc">:</span>conjunctive), <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-566"><a href="#cb54-566" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">conj_label =</span> <span class="fu">ifelse</span>(conjunctive, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span>))</span>
<span id="cb54-567"><a href="#cb54-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats_all)</span>
<span id="cb54-568"><a href="#cb54-568" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb54-569"><a href="#cb54-569" aria-hidden="true" tabindex="-1"></a>stats_1 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_1_blast)</span>
<span id="cb54-570"><a href="#cb54-570" aria-hidden="true" tabindex="-1"></a>threshold_opt_1 <span class="ot">&lt;-</span> stats_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(conj_label) <span class="sc">%&gt;%</span> </span>
<span id="cb54-571"><a href="#cb54-571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-572"><a href="#cb54-572" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">==</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="fu">min</span>(threshold))</span>
<span id="cb54-573"><a href="#cb54-573" aria-hidden="true" tabindex="-1"></a>g_stats_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_1, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb54-574"><a href="#cb54-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">data =</span> threshold_opt_1, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">xintercept=</span>threshold),</span>
<span id="cb54-575"><a href="#cb54-575" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb54-576"><a href="#cb54-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb54-577"><a href="#cb54-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-578"><a href="#cb54-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-579"><a href="#cb54-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set3"</span>) <span class="sc">+</span></span>
<span id="cb54-580"><a href="#cb54-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>conj_label) <span class="sc">+</span></span>
<span id="cb54-581"><a href="#cb54-581" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-582"><a href="#cb54-582" aria-hidden="true" tabindex="-1"></a>g_stats_1</span>
<span id="cb54-583"><a href="#cb54-583" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-584"><a href="#cb54-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-585"><a href="#cb54-585" aria-hidden="true" tabindex="-1"></a>Digging into taxonomic assignments more deeply, we find that low-scoring false positives are primarily mapped by Bowtie2 to SARS-CoV-2, while higher-scoring false positives are mainly mapped to a variety of parvoviruses and parvo-like viruses, as well as Orf virus (cows again?). High-scoring true positives map to a wide range of viruses, but low-scoring true-positives primarily map to human gammaherpesvirus 4. Given that the latter is a DNA virus, I find this quite suspicious.</span>
<span id="cb54-586"><a href="#cb54-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-589"><a href="#cb54-589" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-590"><a href="#cb54-590" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-1</span></span>
<span id="cb54-591"><a href="#cb54-591" aria-hidden="true" tabindex="-1"></a>fp_1 <span class="ot">&lt;-</span> mrg_1_blast <span class="sc">%&gt;%</span> </span>
<span id="cb54-592"><a href="#cb54-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, taxid) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb54-593"><a href="#cb54-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-594"><a href="#cb54-594" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-595"><a href="#cb54-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span></span>
<span id="cb54-596"><a href="#cb54-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(taxid <span class="sc">==</span> <span class="dv">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, name))</span>
<span id="cb54-597"><a href="#cb54-597" aria-hidden="true" tabindex="-1"></a>fp_1_major_tab <span class="ot">&lt;-</span> fp_1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb54-598"><a href="#cb54-598" aria-hidden="true" tabindex="-1"></a>fp_1_major_list <span class="ot">&lt;-</span> fp_1_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb54-599"><a href="#cb54-599" aria-hidden="true" tabindex="-1"></a>fp_1_major <span class="ot">&lt;-</span> fp_1 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">major =</span> p <span class="sc">&gt;</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-600"><a href="#cb54-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-601"><a href="#cb54-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb54-602"><a href="#cb54-602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb54-603"><a href="#cb54-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_1_major_list),</span>
<span id="cb54-604"><a href="#cb54-604" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb54-605"><a href="#cb54-605" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb54-606"><a href="#cb54-606" aria-hidden="true" tabindex="-1"></a>g_fp_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_1_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb54-607"><a href="#cb54-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb54-608"><a href="#cb54-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb54-609"><a href="#cb54-609" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb54-610"><a href="#cb54-610" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-611"><a href="#cb54-611" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb54-612"><a href="#cb54-612" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb54-613"><a href="#cb54-613" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb54-614"><a href="#cb54-614" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-615"><a href="#cb54-615" aria-hidden="true" tabindex="-1"></a>g_fp_1</span>
<span id="cb54-616"><a href="#cb54-616" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-617"><a href="#cb54-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-618"><a href="#cb54-618" aria-hidden="true" tabindex="-1"></a>Since sensitivity is more of a problem here than precision, I decided to dig into into the "true positive" herpesvirus reads. There are 1,197 of these in total, 93% of which are low scoring. When I look into the top taxids that BLAST maps these reads to, we see the following:</span>
<span id="cb54-619"><a href="#cb54-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-622"><a href="#cb54-622" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-623"><a href="#cb54-623" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-herpes-hits-1</span></span>
<span id="cb54-624"><a href="#cb54-624" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb54-625"><a href="#cb54-625" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb54-626"><a href="#cb54-626" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-627"><a href="#cb54-627" aria-hidden="true" tabindex="-1"></a><span class="co"># Get BLAST staxids</span></span>
<span id="cb54-628"><a href="#cb54-628" aria-hidden="true" tabindex="-1"></a>herp_seqs_1 <span class="ot">&lt;-</span> mrg_1_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> <span class="dv">10376</span>, viral_status_out) <span class="sc">%&gt;%</span></span>
<span id="cb54-629"><a href="#cb54-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, taxid, adj_score_max) <span class="sc">%&gt;%</span></span>
<span id="cb54-630"><a href="#cb54-630" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb54-631"><a href="#cb54-631" aria-hidden="true" tabindex="-1"></a>herp_seqs_1_total <span class="ot">&lt;-</span> herp_seqs_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(highscore) <span class="sc">%&gt;%</span> </span>
<span id="cb54-632"><a href="#cb54-632" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_seqs_total"</span>)</span>
<span id="cb54-633"><a href="#cb54-633" aria-hidden="true" tabindex="-1"></a>herp_blast_1_hits <span class="ot">&lt;-</span> herp_seqs_1 <span class="sc">%&gt;%</span> </span>
<span id="cb54-634"><a href="#cb54-634" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_1_paired, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-635"><a href="#cb54-635" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid))</span>
<span id="cb54-636"><a href="#cb54-636" aria-hidden="true" tabindex="-1"></a>herp_blast_1_hits_top <span class="ot">&lt;-</span> herp_blast_1_hits <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(staxid) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb54-637"><a href="#cb54-637" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-638"><a href="#cb54-638" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name=</span><span class="fu">ifelse</span>(staxid <span class="sc">==</span> <span class="dv">3004166</span>, </span>
<span id="cb54-639"><a href="#cb54-639" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Caudopunctatus cichlid hepacivirus"</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb54-640"><a href="#cb54-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_inorder</span>(name))</span>
<span id="cb54-641"><a href="#cb54-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-642"><a href="#cb54-642" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb54-643"><a href="#cb54-643" aria-hidden="true" tabindex="-1"></a>g_herp_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(herp_blast_1_hits_top <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>), </span>
<span id="cb54-644"><a href="#cb54-644" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">y=</span><span class="fu">fct_inorder</span>(name))) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb54-645"><a href="#cb54-645" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"# Mapped Read Pairs"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> </span>
<span id="cb54-646"><a href="#cb54-646" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb54-647"><a href="#cb54-647" aria-hidden="true" tabindex="-1"></a>g_herp_1</span>
<span id="cb54-648"><a href="#cb54-648" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-649"><a href="#cb54-649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-650"><a href="#cb54-650" aria-hidden="true" tabindex="-1"></a>Two things strike me as notable about these results. First, human gammaherpesvirus 4 is only the second-most-common subject taxid, after SARS-CoV-2, an unrelated virus with a different nucleic-acid type. Hmm. Second, close behind these two viruses is a distinctly non-viral taxon, the common carp. This jumped out at me, because the common carp genome is <span class="co">[</span><span class="ot">somewhat notorious</span><span class="co">](https://dgg32.medium.com/carp-in-the-soil-1168818d2191)</span> for being contaminated with Illumina adapter sequences. This makes me suspect that Illumina adapter contamination is playing a role in these results.</span>
<span id="cb54-651"><a href="#cb54-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-654"><a href="#cb54-654" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-655"><a href="#cb54-655" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-carp-hits-1</span></span>
<span id="cb54-656"><a href="#cb54-656" aria-hidden="true" tabindex="-1"></a>carp_hits_1 <span class="ot">&lt;-</span> herp_blast_1_hits <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> <span class="dv">7962</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-657"><a href="#cb54-657" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb54-658"><a href="#cb54-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_1_blast, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-659"><a href="#cb54-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, taxid, adj_score_max, query_seq_fwd, query_seq_rev)</span>
<span id="cb54-660"><a href="#cb54-660" aria-hidden="true" tabindex="-1"></a>carp_hits_1_out <span class="ot">&lt;-</span> carp_hits_1 <span class="sc">%&gt;%</span></span>
<span id="cb54-661"><a href="#cb54-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, sample, <span class="st">"_"</span>, seq_num))</span>
<span id="cb54-662"><a href="#cb54-662" aria-hidden="true" tabindex="-1"></a>carp_hits_1_fasta <span class="ot">&lt;-</span> carp_hits_1_out <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-663"><a href="#cb54-663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, </span>
<span id="cb54-664"><a href="#cb54-664" aria-hidden="true" tabindex="-1"></a>         <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-665"><a href="#cb54-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb54-666"><a href="#cb54-666" aria-hidden="true" tabindex="-1"></a>carp_hits_1_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(carp_hits_1_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-667"><a href="#cb54-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-668"><a href="#cb54-668" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(carp_hits_1_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"spurbeck-carp-hits-1.fasta"</span>))</span>
<span id="cb54-669"><a href="#cb54-669" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-670"><a href="#cb54-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-671"><a href="#cb54-671" aria-hidden="true" tabindex="-1"></a>Inspecting these reads manually, I find that a large fraction do indeed show substantial adapter content. In particular, of the 1658 individual reads queried, at least 1338 (81%) showed a strong match to the Illumina multiplexing primer. As such, better removal of these adapter sequences would likely remove most or all of these "false true positives", potentially significantly improving my results for this dataset.</span>
<span id="cb54-672"><a href="#cb54-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-673"><a href="#cb54-673" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation, round 2</span></span>
<span id="cb54-674"><a href="#cb54-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-675"><a href="#cb54-675" aria-hidden="true" tabindex="-1"></a>To address this problem, I added <span class="co">[</span><span class="ot">Cutadapt</span><span class="co">](https://cutadapt.readthedocs.io/en/stable/guide.html)</span> to the pipeline, using settings that allowed it to trim known adaptors internal to the read:</span>
<span id="cb54-676"><a href="#cb54-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-677"><a href="#cb54-677" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb54-678"><a href="#cb54-678" aria-hidden="true" tabindex="-1"></a><span class="in">cutadapt -b file:&lt;adapter_file&gt; -B file:&lt;adapter_file&gt; -m 15 -e 0.25 --action=trim -o &lt;fwd_reads_out&gt; -p &lt;rev_reads_out&gt; &lt;fwd_reads_in&gt; &lt;rev_reads_in&gt;</span></span>
<span id="cb54-679"><a href="#cb54-679" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-680"><a href="#cb54-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-681"><a href="#cb54-681" aria-hidden="true" tabindex="-1"></a>Running this additional preprocessing step reduced the total number of reads surviving cleaning by 10.4M, or an average of 189k reads per sample. The number of putative human-viral reads, meanwhile, was reduced from 31,610 to 19,799, a 37% decrease. This reduction is primarily due to a large decrease in the number of low-scoring Bowtie2-only putative HV hits.</span>
<span id="cb54-682"><a href="#cb54-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-685"><a href="#cb54-685" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-686"><a href="#cb54-686" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: cutadapt-total-reads</span></span>
<span id="cb54-687"><a href="#cb54-687" aria-hidden="true" tabindex="-1"></a>basic_stats_2_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats_2.tsv"</span>)</span>
<span id="cb54-688"><a href="#cb54-688" aria-hidden="true" tabindex="-1"></a>basic_stats_2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_2_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-689"><a href="#cb54-689" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-690"><a href="#cb54-690" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb54-691"><a href="#cb54-691" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb54-692"><a href="#cb54-692" aria-hidden="true" tabindex="-1"></a>basic_stats_2_sum <span class="ot">&lt;-</span> basic_stats_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_read_pairs =</span> <span class="fu">sum</span>(n_read_pairs), <span class="at">n_bases_approx =</span> <span class="fu">sum</span>(n_bases_approx)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"With Cutadapt"</span>)</span>
<span id="cb54-693"><a href="#cb54-693" aria-hidden="true" tabindex="-1"></a>basic_stats_sum <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_read_pairs =</span> <span class="fu">sum</span>(n_read_pairs), <span class="at">n_bases_approx =</span> <span class="fu">sum</span>(n_bases_approx)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"Without Cutadapt"</span>)</span>
<span id="cb54-694"><a href="#cb54-694" aria-hidden="true" tabindex="-1"></a>sum_cat <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(basic_stats_sum, basic_stats_2_sum) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label=</span><span class="fu">fct_inorder</span>(label))</span>
<span id="cb54-695"><a href="#cb54-695" aria-hidden="true" tabindex="-1"></a>g_sum_cat_reads <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sum_cat, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>n_read_pairs, <span class="at">fill=</span>label)) <span class="sc">+</span></span>
<span id="cb54-696"><a href="#cb54-696" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> </span>
<span id="cb54-697"><a href="#cb54-697" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Preprocessing"</span>) <span class="sc">+</span> </span>
<span id="cb54-698"><a href="#cb54-698" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read Pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-699"><a href="#cb54-699" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-700"><a href="#cb54-700" aria-hidden="true" tabindex="-1"></a>g_sum_cat_reads</span>
<span id="cb54-701"><a href="#cb54-701" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-702"><a href="#cb54-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-705"><a href="#cb54-705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-706"><a href="#cb54-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-nreads-2</span></span>
<span id="cb54-707"><a href="#cb54-707" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_2_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered_2.tsv.gz"</span>)</span>
<span id="cb54-708"><a href="#cb54-708" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_2_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-709"><a href="#cb54-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-710"><a href="#cb54-710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb54-711"><a href="#cb54-711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb54-712"><a href="#cb54-712" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_2 <span class="ot">&lt;-</span> hv_reads_filtered_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, group) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb54-713"><a href="#cb54-713" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_2_summ <span class="ot">&lt;-</span> n_hv_filtered_2 <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb54-714"><a href="#cb54-714" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-715"><a href="#cb54-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-718"><a href="#cb54-718" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-719"><a href="#cb54-719" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-720"><a href="#cb54-720" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb54-721"><a href="#cb54-721" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-hist-2</span></span>
<span id="cb54-722"><a href="#cb54-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-723"><a href="#cb54-723" aria-hidden="true" tabindex="-1"></a><span class="co"># Make new labelled HV dataset</span></span>
<span id="cb54-724"><a href="#cb54-724" aria-hidden="true" tabindex="-1"></a>mrg_2 <span class="ot">&lt;-</span> hv_reads_filtered_2 <span class="sc">%&gt;%</span></span>
<span id="cb54-725"><a href="#cb54-725" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb54-726"><a href="#cb54-726" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb54-727"><a href="#cb54-727" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb54-728"><a href="#cb54-728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb54-729"><a href="#cb54-729" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(),</span>
<span id="cb54-730"><a href="#cb54-730" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb54-731"><a href="#cb54-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-732"><a href="#cb54-732" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and plot histogram</span></span>
<span id="cb54-733"><a href="#cb54-733" aria-hidden="true" tabindex="-1"></a>mrg_2_join <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_1 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label=</span><span class="st">"Without Cutadapt"</span>),</span>
<span id="cb54-734"><a href="#cb54-734" aria-hidden="true" tabindex="-1"></a>                      mrg_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label=</span><span class="st">"With Cutadapt"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-735"><a href="#cb54-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">fct_inorder</span>(label))</span>
<span id="cb54-736"><a href="#cb54-736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-737"><a href="#cb54-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-738"><a href="#cb54-738" aria-hidden="true" tabindex="-1"></a>g_hist_2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_2_join, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> </span>
<span id="cb54-739"><a href="#cb54-739" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> </span>
<span id="cb54-740"><a href="#cb54-740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(label<span class="sc">~</span>kraken_label) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb54-741"><a href="#cb54-741" aria-hidden="true" tabindex="-1"></a>g_hist_2</span>
<span id="cb54-742"><a href="#cb54-742" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-743"><a href="#cb54-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-744"><a href="#cb54-744" aria-hidden="true" tabindex="-1"></a>Comparing the lengths of the same sequences in the old versus new results, we see that many reads are reduced in length as a result of adding Cutadapt. As such, it made sense to repeat the BLAST analysis on the reprocessed reads rather than using the BLAST assignments from the previous analysis.</span>
<span id="cb54-745"><a href="#cb54-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-748"><a href="#cb54-748" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-749"><a href="#cb54-749" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-cutadapt-length</span></span>
<span id="cb54-750"><a href="#cb54-750" aria-hidden="true" tabindex="-1"></a>mrg_num_comp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(mrg_2 <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>seq_num), </span>
<span id="cb54-751"><a href="#cb54-751" aria-hidden="true" tabindex="-1"></a>                           mrg_1_num <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(seq_id, seq_num), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"seq_id"</span>))</span>
<span id="cb54-752"><a href="#cb54-752" aria-hidden="true" tabindex="-1"></a>mrg_num_diff <span class="ot">&lt;-</span> mrg_1_num <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, <span class="at">query_len_fwd_old =</span> query_len_fwd,</span>
<span id="cb54-753"><a href="#cb54-753" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">query_len_rev_old =</span> query_len_rev) <span class="sc">%&gt;%</span> </span>
<span id="cb54-754"><a href="#cb54-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(mrg_num_comp <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, <span class="at">query_len_fwd_new =</span> query_len_fwd, </span>
<span id="cb54-755"><a href="#cb54-755" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">query_len_rev_new =</span> query_len_rev),</span>
<span id="cb54-756"><a href="#cb54-756" aria-hidden="true" tabindex="-1"></a>             <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-757"><a href="#cb54-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">query_len_fwd_diff =</span> query_len_fwd_new <span class="sc">-</span> query_len_fwd_old,</span>
<span id="cb54-758"><a href="#cb54-758" aria-hidden="true" tabindex="-1"></a>         <span class="at">query_len_rev_diff =</span> query_len_rev_new <span class="sc">-</span> query_len_rev_old)</span>
<span id="cb54-759"><a href="#cb54-759" aria-hidden="true" tabindex="-1"></a>mrg_num_diff_long <span class="ot">&lt;-</span> mrg_num_diff <span class="sc">%&gt;%</span> </span>
<span id="cb54-760"><a href="#cb54-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, <span class="at">Forward=</span>query_len_fwd_diff, <span class="at">Reverse=</span>query_len_rev_diff) <span class="sc">%&gt;%</span></span>
<span id="cb54-761"><a href="#cb54-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(sample<span class="sc">:</span>seq_num), <span class="at">names_to=</span><span class="st">"read"</span>, <span class="at">values_to=</span><span class="st">"length_difference"</span>)</span>
<span id="cb54-762"><a href="#cb54-762" aria-hidden="true" tabindex="-1"></a>g_len_diff <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_num_diff_long, <span class="fu">aes</span>(<span class="at">x=</span>length_difference)) <span class="sc">+</span></span>
<span id="cb54-763"><a href="#cb54-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">4</span>, <span class="at">boundary=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb54-764"><a href="#cb54-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Effect of Cutadapt on read length"</span>) <span class="sc">+</span></span>
<span id="cb54-765"><a href="#cb54-765" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# of Reads"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-766"><a href="#cb54-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>read) <span class="sc">+</span></span>
<span id="cb54-767"><a href="#cb54-767" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-768"><a href="#cb54-768" aria-hidden="true" tabindex="-1"></a>g_len_diff</span>
<span id="cb54-769"><a href="#cb54-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-770"><a href="#cb54-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-773"><a href="#cb54-773" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-774"><a href="#cb54-774" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-fasta-2</span></span>
<span id="cb54-775"><a href="#cb54-775" aria-hidden="true" tabindex="-1"></a>mrg_2_num <span class="ot">&lt;-</span> mrg_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb54-776"><a href="#cb54-776" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb54-777"><a href="#cb54-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, sample, <span class="st">"_"</span>, seq_num))</span>
<span id="cb54-778"><a href="#cb54-778" aria-hidden="true" tabindex="-1"></a>mrg_2_fasta <span class="ot">&lt;-</span>  mrg_2_num <span class="sc">%&gt;%</span> </span>
<span id="cb54-779"><a href="#cb54-779" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-780"><a href="#cb54-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-781"><a href="#cb54-781" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb54-782"><a href="#cb54-782" aria-hidden="true" tabindex="-1"></a>mrg_2_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_1_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-783"><a href="#cb54-783" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_2_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"spurbeck-putative-viral-2.fasta"</span>))</span>
<span id="cb54-784"><a href="#cb54-784" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-785"><a href="#cb54-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-788"><a href="#cb54-788" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-789"><a href="#cb54-789" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-2</span></span>
<span id="cb54-790"><a href="#cb54-790" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results (again, pre-filtered to save space)</span></span>
<span id="cb54-791"><a href="#cb54-791" aria-hidden="true" tabindex="-1"></a>blast_results_2_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"putative-viral-blast-best-2.tsv.gz"</span>)</span>
<span id="cb54-792"><a href="#cb54-792" aria-hidden="true" tabindex="-1"></a>blast_results_2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_2_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb54-793"><a href="#cb54-793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-794"><a href="#cb54-794" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject</span></span>
<span id="cb54-795"><a href="#cb54-795" aria-hidden="true" tabindex="-1"></a>blast_results_2_best <span class="ot">&lt;-</span> blast_results_2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> </span>
<span id="cb54-796"><a href="#cb54-796" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb54-797"><a href="#cb54-797" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb54-798"><a href="#cb54-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-799"><a href="#cb54-799" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each query</span></span>
<span id="cb54-800"><a href="#cb54-800" aria-hidden="true" tabindex="-1"></a>blast_results_2_ranked <span class="ot">&lt;-</span> blast_results_2_best <span class="sc">%&gt;%</span> </span>
<span id="cb54-801"><a href="#cb54-801" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb54-802"><a href="#cb54-802" aria-hidden="true" tabindex="-1"></a>blast_results_2_highrank <span class="ot">&lt;-</span> blast_results_2_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-803"><a href="#cb54-803" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>), </span>
<span id="cb54-804"><a href="#cb54-804" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_num =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>), </span>
<span id="cb54-805"><a href="#cb54-805" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(head, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-806"><a href="#cb54-806" aria-hidden="true" tabindex="-1"></a>           <span class="fu">sapply</span>(paste, <span class="at">collapse=</span><span class="st">"_"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-807"><a href="#cb54-807" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num))</span>
<span id="cb54-808"><a href="#cb54-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-809"><a href="#cb54-809" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb54-810"><a href="#cb54-810" aria-hidden="true" tabindex="-1"></a>blast_results_2_paired <span class="ot">&lt;-</span> blast_results_2_highrank <span class="sc">%&gt;%</span></span>
<span id="cb54-811"><a href="#cb54-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb54-812"><a href="#cb54-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb54-813"><a href="#cb54-813" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_rank =</span> <span class="fu">min</span>(rank),</span>
<span id="cb54-814"><a href="#cb54-814" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb54-815"><a href="#cb54-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-816"><a href="#cb54-816" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb54-817"><a href="#cb54-817" aria-hidden="true" tabindex="-1"></a>blast_results_2_viral <span class="ot">&lt;-</span> blast_results_2_paired <span class="sc">%&gt;%</span></span>
<span id="cb54-818"><a href="#cb54-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid,</span>
<span id="cb54-819"><a href="#cb54-819" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb54-820"><a href="#cb54-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-821"><a href="#cb54-821" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb54-822"><a href="#cb54-822" aria-hidden="true" tabindex="-1"></a>mrg_2_assign <span class="ot">&lt;-</span> mrg_2_num <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, taxid, assigned_taxid)</span>
<span id="cb54-823"><a href="#cb54-823" aria-hidden="true" tabindex="-1"></a>blast_results_2_assign <span class="ot">&lt;-</span> <span class="fu">full_join</span>(blast_results_2_viral, mrg_2_assign, </span>
<span id="cb54-824"><a href="#cb54-824" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-825"><a href="#cb54-825" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb54-826"><a href="#cb54-826" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb54-827"><a href="#cb54-827" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb54-828"><a href="#cb54-828" aria-hidden="true" tabindex="-1"></a>blast_results_2_out <span class="ot">&lt;-</span> blast_results_2_assign <span class="sc">%&gt;%</span></span>
<span id="cb54-829"><a href="#cb54-829" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb54-830"><a href="#cb54-830" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb54-831"><a href="#cb54-831" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb54-832"><a href="#cb54-832" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb54-833"><a href="#cb54-833" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-834"><a href="#cb54-834" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>))</span>
<span id="cb54-835"><a href="#cb54-835" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-836"><a href="#cb54-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-837"><a href="#cb54-837" aria-hidden="true" tabindex="-1"></a>The addition of Cutadapt results in a substantial decrease in low-scoring putative HV sequences, resulting in a large improvement in measured sensitivity and F1 score:</span>
<span id="cb54-838"><a href="#cb54-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-841"><a href="#cb54-841" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-842"><a href="#cb54-842" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-hist-2</span></span>
<span id="cb54-843"><a href="#cb54-843" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb54-844"><a href="#cb54-844" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb54-845"><a href="#cb54-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-846"><a href="#cb54-846" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb54-847"><a href="#cb54-847" aria-hidden="true" tabindex="-1"></a>mrg_2_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_2_num, blast_results_2_out, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-848"><a href="#cb54-848" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb54-849"><a href="#cb54-849" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb54-850"><a href="#cb54-850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-851"><a href="#cb54-851" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and label</span></span>
<span id="cb54-852"><a href="#cb54-852" aria-hidden="true" tabindex="-1"></a>hist_blast_2_prep <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_1_blast <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label=</span><span class="st">"Without Cutadapt"</span>),</span>
<span id="cb54-853"><a href="#cb54-853" aria-hidden="true" tabindex="-1"></a>                               mrg_2_blast <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label=</span><span class="st">"With Cutadapt"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-854"><a href="#cb54-854" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">fct_inorder</span>(label))</span>
<span id="cb54-855"><a href="#cb54-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-856"><a href="#cb54-856" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb54-857"><a href="#cb54-857" aria-hidden="true" tabindex="-1"></a>g_hist_blast_2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hist_blast_2_prep, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(label<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb54-858"><a href="#cb54-858" aria-hidden="true" tabindex="-1"></a>g_hist_blast_2</span>
<span id="cb54-859"><a href="#cb54-859" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-860"><a href="#cb54-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-863"><a href="#cb54-863" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-864"><a href="#cb54-864" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-f1-2</span></span>
<span id="cb54-865"><a href="#cb54-865" aria-hidden="true" tabindex="-1"></a>stats_2 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_2_blast) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"With Cutadapt"</span>)</span>
<span id="cb54-866"><a href="#cb54-866" aria-hidden="true" tabindex="-1"></a>stats_comb <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_1 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"Without Cutadapt"</span>), stats_2)</span>
<span id="cb54-867"><a href="#cb54-867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-868"><a href="#cb54-868" aria-hidden="true" tabindex="-1"></a>g_stats_2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_comb, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>label, <span class="at">linetype=</span>conj_label)) <span class="sc">+</span></span>
<span id="cb54-869"><a href="#cb54-869" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb54-870"><a href="#cb54-870" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="cn">NA</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-871"><a href="#cb54-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-872"><a href="#cb54-872" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name =</span> <span class="st">"Configuration"</span>) <span class="sc">+</span></span>
<span id="cb54-873"><a href="#cb54-873" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name=</span><span class="st">"Threshold type"</span>) <span class="sc">+</span></span>
<span id="cb54-874"><a href="#cb54-874" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>metric, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb54-875"><a href="#cb54-875" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-876"><a href="#cb54-876" aria-hidden="true" tabindex="-1"></a>g_stats_2</span>
<span id="cb54-877"><a href="#cb54-877" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-878"><a href="#cb54-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-879"><a href="#cb54-879" aria-hidden="true" tabindex="-1"></a>At a disjunctive threshold of 20, excluding "fake true positives" arising from adapter contamination improved measured sensitivity from 86% to 97%, bringing the measured F1 score up from 89% to 95%. I would feel much better about using these results for further downstream analyses.</span>
<span id="cb54-880"><a href="#cb54-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-881"><a href="#cb54-881" aria-hidden="true" tabindex="-1"></a>That said, I think further improvement is possible. While addition of Cutadapt processing substantially improved measured sensitivity, measured precision only improved slightly. Looking at the apparent taxonomic makeup of putative HV reads, we see that, while low-scoring "true positives" mapping to herpesviruses have been mostly eliminated, high-scoring false positives still map primarily to parvoviruses and parvo-like viruses:</span>
<span id="cb54-882"><a href="#cb54-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-885"><a href="#cb54-885" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-886"><a href="#cb54-886" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-2</span></span>
<span id="cb54-887"><a href="#cb54-887" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb54-888"><a href="#cb54-888" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate composition for 2nd attempt</span></span>
<span id="cb54-889"><a href="#cb54-889" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb54-890"><a href="#cb54-890" aria-hidden="true" tabindex="-1"></a>fp_2 <span class="ot">&lt;-</span> mrg_2_blast <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(viral_status_out, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb54-891"><a href="#cb54-891" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span> </span>
<span id="cb54-892"><a href="#cb54-892" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-893"><a href="#cb54-893" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-894"><a href="#cb54-894" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span></span>
<span id="cb54-895"><a href="#cb54-895" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(taxid <span class="sc">==</span> <span class="dv">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, name),</span>
<span id="cb54-896"><a href="#cb54-896" aria-hidden="true" tabindex="-1"></a>         <span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold)</span>
<span id="cb54-897"><a href="#cb54-897" aria-hidden="true" tabindex="-1"></a>fp_2_major_tab <span class="ot">&lt;-</span> fp_2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(major) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb54-898"><a href="#cb54-898" aria-hidden="true" tabindex="-1"></a>fp_2_major_list <span class="ot">&lt;-</span> fp_2_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb54-899"><a href="#cb54-899" aria-hidden="true" tabindex="-1"></a>fp_2_major <span class="ot">&lt;-</span> fp_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-900"><a href="#cb54-900" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb54-901"><a href="#cb54-901" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_2_major_list),</span>
<span id="cb54-902"><a href="#cb54-902" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb54-903"><a href="#cb54-903" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb54-904"><a href="#cb54-904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-905"><a href="#cb54-905" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine composition</span></span>
<span id="cb54-906"><a href="#cb54-906" aria-hidden="true" tabindex="-1"></a>fp_major_comb <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(fp_1_major <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"Without Cutadapt"</span>),</span>
<span id="cb54-907"><a href="#cb54-907" aria-hidden="true" tabindex="-1"></a>                           fp_2_major <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">label =</span> <span class="st">"With Cutadapt"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-908"><a href="#cb54-908" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">fct_inorder</span>(label)) <span class="sc">%&gt;%</span></span>
<span id="cb54-909"><a href="#cb54-909" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.character</span>(name_display)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">str_detect</span>(name_display, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-910"><a href="#cb54-910" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">fct_inorder</span>(name_display))</span>
<span id="cb54-911"><a href="#cb54-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-912"><a href="#cb54-912" aria-hidden="true" tabindex="-1"></a><span class="co"># Palette</span></span>
<span id="cb54-913"><a href="#cb54-913" aria-hidden="true" tabindex="-1"></a>palette_fp_2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="st">"#999999"</span>)</span>
<span id="cb54-914"><a href="#cb54-914" aria-hidden="true" tabindex="-1"></a>g_fp_2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_major_comb, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb54-915"><a href="#cb54-915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb54-916"><a href="#cb54-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb54-917"><a href="#cb54-917" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">label =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb54-918"><a href="#cb54-918" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>, <span class="at">values=</span>palette_fp_2) <span class="sc">+</span></span>
<span id="cb54-919"><a href="#cb54-919" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(label<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb54-920"><a href="#cb54-920" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb54-921"><a href="#cb54-921" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-922"><a href="#cb54-922" aria-hidden="true" tabindex="-1"></a>g_fp_2</span>
<span id="cb54-923"><a href="#cb54-923" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-924"><a href="#cb54-924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-925"><a href="#cb54-925" aria-hidden="true" tabindex="-1"></a>Not only are these parvoviruses and parvo-like viruses DNA viruses (and thus suspicious as abundant components of the RNA virome), they are also the subject of a <span class="co">[</span><span class="ot">famous controversy</span><span class="co">](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3964066/)</span> in early viral metagenomics, where viruses apparently widespread in Chinese hepatitis patients were found to arise from spin column contamination. In fact, these viruses seem not to be human-infecting at all; the authors of the study reporting the contamination suggested that they might be viruses of the diatom algae used as the source of silica for these columns. As such, it seems appropriate to remove these from the human-virus database used to identify putative HV reads.</span>
<span id="cb54-926"><a href="#cb54-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-927"><a href="#cb54-927" aria-hidden="true" tabindex="-1"></a>We also see a significant number of false-positive reads mapped to Orf virus. Historically this has been a sign of contamination with bovine sequences, and indeed the top taxa mapped to these reads by BLAST include *Bos taurus* (cattle), *Bos mutus* (wild yak), and *Cervus elaphus* (red deer).</span>
<span id="cb54-928"><a href="#cb54-928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-931"><a href="#cb54-931" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-932"><a href="#cb54-932" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-orf-1</span></span>
<span id="cb54-933"><a href="#cb54-933" aria-hidden="true" tabindex="-1"></a>orf_taxid <span class="ot">&lt;-</span> <span class="dv">10258</span></span>
<span id="cb54-934"><a href="#cb54-934" aria-hidden="true" tabindex="-1"></a>orf_seqs <span class="ot">&lt;-</span> mrg_1_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> <span class="dv">10258</span>)</span>
<span id="cb54-935"><a href="#cb54-935" aria-hidden="true" tabindex="-1"></a>orf_matches <span class="ot">&lt;-</span> orf_seqs <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(blast_results_1_paired, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>))</span>
<span id="cb54-936"><a href="#cb54-936" aria-hidden="true" tabindex="-1"></a>orf_staxids <span class="ot">&lt;-</span> orf_matches <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(staxid) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>)</span>
<span id="cb54-937"><a href="#cb54-937" aria-hidden="true" tabindex="-1"></a>orf_fp_out <span class="ot">&lt;-</span> orf_seqs <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb54-938"><a href="#cb54-938" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-939"><a href="#cb54-939" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, taxid, <span class="st">"_"</span>, <span class="fu">row_number</span>()))</span>
<span id="cb54-940"><a href="#cb54-940" aria-hidden="true" tabindex="-1"></a>orf_fp_fasta <span class="ot">&lt;-</span> orf_fp_out <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-941"><a href="#cb54-941" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-942"><a href="#cb54-942" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb54-943"><a href="#cb54-943" aria-hidden="true" tabindex="-1"></a>orf_fp_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(orf_fp_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-944"><a href="#cb54-944" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(orf_fp_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"spurbeck-orf-fp.fasta"</span>))</span>
<span id="cb54-945"><a href="#cb54-945" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-946"><a href="#cb54-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-947"><a href="#cb54-947" aria-hidden="true" tabindex="-1"></a>The ideal approach to removing these false positives would be to add the cow genome to the Kraken database we use for validation of Bowtie2 assignments, along with the pig genome and probably some other known contaminants causing issues. This is probably worth doing at some point, but represents a substantial amount of additional work; I'd rather use a simpler alternative right now if one is available. One option that comes to mind is to try replacing the BBMap aligner I'm using to detect and remove contaminants with Bowtie2; in quick experiments, running Bowtie2 (<span class="in">`--sensitive`</span>) on the putative Orf virus sequences above, using the same dataset of contaminants for the index, successfully removed about two-thirds of them. While this doesn't guarantee that Bowtie2 would perform as well on the whole population of contaminating cow sequences, it suggests that using Bowtie2 in place of BBMap is worth trying.</span>
<span id="cb54-948"><a href="#cb54-948" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-949"><a href="#cb54-949" aria-hidden="true" tabindex="-1"></a>As such, I re-ran the HV detection pipeline again, having made two changes: first, removing Parvovirus NIH-CQV and the parvo-like viruses from the HV database used to identify putative HV reads, and secondly, replacing BBMap with Bowtie2 for contaminant removal.</span>
<span id="cb54-950"><a href="#cb54-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-951"><a href="#cb54-951" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation, round 3</span></span>
<span id="cb54-952"><a href="#cb54-952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-953"><a href="#cb54-953" aria-hidden="true" tabindex="-1"></a>I tried re-running the HV detection pipeline with three different sets of Bowtie2 settings: (a) <span class="in">`--sensitive`</span>, (b) <span class="in">`--local --sensitive-local`</span>, and (c) <span class="in">`--local --very-sensitive-local`</span>. In total, these find 26,370, 15,338 and 13,160 putative human-viral reads, respectively, compared to 31,610 for my initial attempt and 19,799 after the addition of Cutadapt:</span>
<span id="cb54-954"><a href="#cb54-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-957"><a href="#cb54-957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-958"><a href="#cb54-958" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-nreads-3</span></span>
<span id="cb54-959"><a href="#cb54-959" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_3_paths <span class="ot">&lt;-</span> <span class="fu">paste0</span>(data_dir, <span class="st">"hv_hits_putative_filtered_3"</span>, </span>
<span id="cb54-960"><a href="#cb54-960" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(<span class="st">"a"</span>, <span class="st">"b"</span>, <span class="st">"c"</span>), <span class="st">".tsv.gz"</span>)</span>
<span id="cb54-961"><a href="#cb54-961" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_3_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hv_reads_filtered_3_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-962"><a href="#cb54-962" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_3 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="cf">function</span>(n) hv_reads_filtered_3_raw[[n]] <span class="sc">%&gt;%</span> </span>
<span id="cb54-963"><a href="#cb54-963" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="fu">paste0</span>(<span class="st">"Bowtie2"</span>, <span class="fu">c</span>(<span class="st">"(a)"</span>,<span class="st">"(b)"</span>,<span class="st">"(c)"</span>)[n]))) <span class="sc">%&gt;%</span> </span>
<span id="cb54-964"><a href="#cb54-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-965"><a href="#cb54-965" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-966"><a href="#cb54-966" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb54-967"><a href="#cb54-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb54-968"><a href="#cb54-968" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_3 <span class="ot">&lt;-</span> hv_reads_filtered_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, group, attempt) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb54-969"><a href="#cb54-969" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_3_summ <span class="ot">&lt;-</span> n_hv_filtered_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(attempt) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb54-970"><a href="#cb54-970" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(n_hv_filtered_2_summ <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="st">"BBMap"</span>), n_hv_filtered_3_summ) <span class="sc">%&gt;%</span> </span>
<span id="cb54-971"><a href="#cb54-971" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(attempt, <span class="fu">everything</span>())</span>
<span id="cb54-972"><a href="#cb54-972" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-973"><a href="#cb54-973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-976"><a href="#cb54-976" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-977"><a href="#cb54-977" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-hist-3</span></span>
<span id="cb54-978"><a href="#cb54-978" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb54-979"><a href="#cb54-979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-980"><a href="#cb54-980" aria-hidden="true" tabindex="-1"></a><span class="co"># Make new labelled HV dataset</span></span>
<span id="cb54-981"><a href="#cb54-981" aria-hidden="true" tabindex="-1"></a>mrg_3 <span class="ot">&lt;-</span> hv_reads_filtered_3 <span class="sc">%&gt;%</span></span>
<span id="cb54-982"><a href="#cb54-982" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb54-983"><a href="#cb54-983" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb54-984"><a href="#cb54-984" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb54-985"><a href="#cb54-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb54-986"><a href="#cb54-986" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(),</span>
<span id="cb54-987"><a href="#cb54-987" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb54-988"><a href="#cb54-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-989"><a href="#cb54-989" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge and plot histogram</span></span>
<span id="cb54-990"><a href="#cb54-990" aria-hidden="true" tabindex="-1"></a>mrg_join_3 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="st">"BBMap"</span>), mrg_3)</span>
<span id="cb54-991"><a href="#cb54-991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-992"><a href="#cb54-992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-993"><a href="#cb54-993" aria-hidden="true" tabindex="-1"></a>g_hist_3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_join_3, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> </span>
<span id="cb54-994"><a href="#cb54-994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> </span>
<span id="cb54-995"><a href="#cb54-995" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(attempt<span class="sc">~</span>kraken_label) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb54-996"><a href="#cb54-996" aria-hidden="true" tabindex="-1"></a>g_hist_3</span>
<span id="cb54-997"><a href="#cb54-997" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-998"><a href="#cb54-998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-999"><a href="#cb54-999" aria-hidden="true" tabindex="-1"></a>In all cases, the great majority of false-positive sequences from previous attempts were successfully removed (85%/89%/90%, respectively), along with a small number of true-positives (0.19%/0.23%/0.23%, respectively). At the same time, a number of new putative HV sequences arose as a result of the change in filtering algorithm: 8,512/2,224/1,166 respectively. The great majority of these (97.6%/99.1%/98.8%) are low-scoring, suggesting that they are mostly or entirely false matches that are being let through by Bowtie2 that were previously being caught by BBMap.</span>
<span id="cb54-1000"><a href="#cb54-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1003"><a href="#cb54-1003" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1004"><a href="#cb54-1004" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-3a</span></span>
<span id="cb54-1005"><a href="#cb54-1005" aria-hidden="true" tabindex="-1"></a>mrg_2_blast_prep <span class="ot">&lt;-</span> mrg_2_blast <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb54-1006"><a href="#cb54-1006" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(seq_id, seq_num, viral_status, viral_status_out)</span>
<span id="cb54-1007"><a href="#cb54-1007" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_old <span class="ot">&lt;-</span> mrg_join_3 <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>seq_num) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb54-1008"><a href="#cb54-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_2_blast_prep, <span class="at">by =</span> <span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1009"><a href="#cb54-1009" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">viral_status_old =</span> viral_status) <span class="sc">%&gt;%</span></span>
<span id="cb54-1010"><a href="#cb54-1010" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_old_out =</span> <span class="fu">replace_na</span>(<span class="fu">as.character</span>(viral_status_out), <span class="st">"UNKNOWN"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1011"><a href="#cb54-1011" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>viral_status_out)</span>
<span id="cb54-1012"><a href="#cb54-1012" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_old_cum <span class="ot">&lt;-</span> mrg_3_blast_old <span class="sc">%&gt;%</span></span>
<span id="cb54-1013"><a href="#cb54-1013" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">vs_label =</span> <span class="fu">paste0</span>(<span class="st">"HV status:</span><span class="sc">\n</span><span class="st">"</span>, viral_status_old_out),</span>
<span id="cb54-1014"><a href="#cb54-1014" aria-hidden="true" tabindex="-1"></a>         <span class="at">attempt_label =</span> <span class="fu">paste</span>(<span class="st">"Attempt"</span>, attempt))</span>
<span id="cb54-1015"><a href="#cb54-1015" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_old_summ <span class="ot">&lt;-</span> mrg_3_blast_old_cum <span class="sc">%&gt;%</span> </span>
<span id="cb54-1016"><a href="#cb54-1016" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(attempt, viral_status_old_out, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb54-1017"><a href="#cb54-1017" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_label =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb54-1018"><a href="#cb54-1018" aria-hidden="true" tabindex="-1"></a>         <span class="at">vs_label =</span> <span class="fu">paste</span>(<span class="st">"HV status:"</span>, viral_status_old_out))</span>
<span id="cb54-1019"><a href="#cb54-1019" aria-hidden="true" tabindex="-1"></a>g_status_summ <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_3_blast_old_summ, <span class="fu">aes</span>(<span class="at">x=</span>attempt, <span class="at">y=</span>n, <span class="at">fill=</span>attempt)) <span class="sc">+</span></span>
<span id="cb54-1020"><a href="#cb54-1020" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb54-1021"><a href="#cb54-1021" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(score_label<span class="sc">~</span>vs_label, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb54-1022"><a href="#cb54-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span></span>
<span id="cb54-1023"><a href="#cb54-1023" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb54-1024"><a href="#cb54-1024" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb54-1025"><a href="#cb54-1025" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1026"><a href="#cb54-1026" aria-hidden="true" tabindex="-1"></a>g_status_summ</span>
<span id="cb54-1027"><a href="#cb54-1027" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1028"><a href="#cb54-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1029"><a href="#cb54-1029" aria-hidden="true" tabindex="-1"></a>Checking putative Orf virus reads specifically, all three Bowtie2-based attempts successfully remove most putative Orf reads from the previous attempt, while introducing some number of new putative hits (of which I assume the vast majority are fake hits arising from bovine contamination). In attempt (a), these new putative hits outnumber the old hits that were removed, resulting in a net increase in putative (but, I'm fairly confident, false) Orf-virus hits, whether total or high-scoring. Conversely, attempts (b) and (c) manage to remove even more old putative Orf hits while creating many fewer new ones, resulting in a large net decrease in total and high-scoring putative Orf hits:</span>
<span id="cb54-1030"><a href="#cb54-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1033"><a href="#cb54-1033" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1034"><a href="#cb54-1034" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-orf-2</span></span>
<span id="cb54-1035"><a href="#cb54-1035" aria-hidden="true" tabindex="-1"></a>orf_seqs_3 <span class="ot">&lt;-</span> mrg_3_blast_old_cum <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> <span class="dv">10258</span>)</span>
<span id="cb54-1036"><a href="#cb54-1036" aria-hidden="true" tabindex="-1"></a>orf_seqs_3_summ <span class="ot">&lt;-</span> orf_seqs_3 <span class="sc">%&gt;%</span> </span>
<span id="cb54-1037"><a href="#cb54-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(attempt, viral_status_old_out, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1038"><a href="#cb54-1038" aria-hidden="true" tabindex="-1"></a>  count</span>
<span id="cb54-1039"><a href="#cb54-1039" aria-hidden="true" tabindex="-1"></a>g_orf_total <span class="ot">&lt;-</span> orf_seqs_3_summ <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(attempt) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1040"><a href="#cb54-1040" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>attempt, <span class="at">y=</span>n, <span class="at">fill=</span>attempt)) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb54-1041"><a href="#cb54-1041" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Total reads mapped to Orf virus"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">250</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1042"><a href="#cb54-1042" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb54-1043"><a href="#cb54-1043" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb54-1044"><a href="#cb54-1044" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1045"><a href="#cb54-1045" aria-hidden="true" tabindex="-1"></a>g_orf_high <span class="ot">&lt;-</span> orf_seqs_3_summ <span class="sc">%&gt;%</span> <span class="fu">filter</span>(highscore) <span class="sc">%&gt;%</span></span>
<span id="cb54-1046"><a href="#cb54-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>attempt, <span class="at">y=</span>n, <span class="at">fill=</span>attempt)) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb54-1047"><a href="#cb54-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb54-1048"><a href="#cb54-1048" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb54-1049"><a href="#cb54-1049" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"High-scoring reads mapped to Orf virus"</span>,</span>
<span id="cb54-1050"><a href="#cb54-1050" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">250</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1051"><a href="#cb54-1051" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1052"><a href="#cb54-1052" aria-hidden="true" tabindex="-1"></a>g_orf_total <span class="sc">+</span> g_orf_high</span>
<span id="cb54-1053"><a href="#cb54-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1054"><a href="#cb54-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1055"><a href="#cb54-1055" aria-hidden="true" tabindex="-1"></a>Next, I validated all new putative HV sequences with BLASTN as before, then evaluated each attempt's performance against all the putative HV reads identified by any attempt:</span>
<span id="cb54-1056"><a href="#cb54-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1059"><a href="#cb54-1059" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1060"><a href="#cb54-1060" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-fasta-3</span></span>
<span id="cb54-1061"><a href="#cb54-1061" aria-hidden="true" tabindex="-1"></a>mrg_3_fasta_prep <span class="ot">&lt;-</span> mrg_join_3 <span class="sc">%&gt;%</span></span>
<span id="cb54-1062"><a href="#cb54-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> seq_id <span class="sc">%in%</span> mrg_2_blast<span class="sc">$</span>seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb54-1063"><a href="#cb54-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb54-1064"><a href="#cb54-1064" aria-hidden="true" tabindex="-1"></a>mrg_3_fasta <span class="ot">&lt;-</span> mrg_3_fasta_prep <span class="sc">%&gt;%</span></span>
<span id="cb54-1065"><a href="#cb54-1065" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, seq_id)) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-1066"><a href="#cb54-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-1067"><a href="#cb54-1067" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb54-1068"><a href="#cb54-1068" aria-hidden="true" tabindex="-1"></a>mrg_3_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_2_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb54-1069"><a href="#cb54-1069" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_3_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"spurbeck-putative-viral-3.fasta"</span>))</span>
<span id="cb54-1070"><a href="#cb54-1070" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1071"><a href="#cb54-1071" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1074"><a href="#cb54-1074" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1075"><a href="#cb54-1075" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-3</span></span>
<span id="cb54-1076"><a href="#cb54-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1077"><a href="#cb54-1077" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results (again, pre-filtered to save space)</span></span>
<span id="cb54-1078"><a href="#cb54-1078" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_cols &lt;- c("qseqid", "sseqid", "sgi", "staxid", "qlen", "evalue", "bitscore", "qcovs", "length", "pident", "mismatch", "gapopen", "sstrand", "qstart", "qend", "sstart", "send")</span></span>
<span id="cb54-1079"><a href="#cb54-1079" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results_3_path_0 &lt;- file.path(data_dir, "spurbeck-putative-viral-3.blast.gz")</span></span>
<span id="cb54-1080"><a href="#cb54-1080" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results_3 &lt;- read_tsv(blast_results_3_path_0, show_col_types = FALSE,</span></span>
<span id="cb54-1081"><a href="#cb54-1081" aria-hidden="true" tabindex="-1"></a><span class="co">#                           col_names = blast_cols, col_types = cols(.default="c"))</span></span>
<span id="cb54-1082"><a href="#cb54-1082" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1083"><a href="#cb54-1083" aria-hidden="true" tabindex="-1"></a>blast_results_3_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"putative-viral-blast-best-3.tsv.gz"</span>)</span>
<span id="cb54-1084"><a href="#cb54-1084" aria-hidden="true" tabindex="-1"></a>blast_results_3 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_3_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb54-1085"><a href="#cb54-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1086"><a href="#cb54-1086" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject</span></span>
<span id="cb54-1087"><a href="#cb54-1087" aria-hidden="true" tabindex="-1"></a>blast_results_3_best <span class="ot">&lt;-</span> blast_results_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb54-1088"><a href="#cb54-1088" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1089"><a href="#cb54-1089" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb54-1090"><a href="#cb54-1090" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1091"><a href="#cb54-1091" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each query</span></span>
<span id="cb54-1092"><a href="#cb54-1092" aria-hidden="true" tabindex="-1"></a>blast_results_3_ranked <span class="ot">&lt;-</span> blast_results_3_best <span class="sc">%&gt;%</span> </span>
<span id="cb54-1093"><a href="#cb54-1093" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb54-1094"><a href="#cb54-1094" aria-hidden="true" tabindex="-1"></a>blast_results_3_highrank <span class="ot">&lt;-</span> blast_results_3_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1095"><a href="#cb54-1095" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="dv">2</span>), </span>
<span id="cb54-1096"><a href="#cb54-1096" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_id =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1097"><a href="#cb54-1097" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore))</span>
<span id="cb54-1098"><a href="#cb54-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1099"><a href="#cb54-1099" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb54-1100"><a href="#cb54-1100" aria-hidden="true" tabindex="-1"></a>blast_results_3_paired <span class="ot">&lt;-</span> blast_results_3_highrank <span class="sc">%&gt;%</span></span>
<span id="cb54-1101"><a href="#cb54-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb54-1102"><a href="#cb54-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb54-1103"><a href="#cb54-1103" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_rank =</span> <span class="fu">min</span>(rank),</span>
<span id="cb54-1104"><a href="#cb54-1104" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb54-1105"><a href="#cb54-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1106"><a href="#cb54-1106" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb54-1107"><a href="#cb54-1107" aria-hidden="true" tabindex="-1"></a>blast_results_3_viral <span class="ot">&lt;-</span> blast_results_3_paired <span class="sc">%&gt;%</span></span>
<span id="cb54-1108"><a href="#cb54-1108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid,</span>
<span id="cb54-1109"><a href="#cb54-1109" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb54-1110"><a href="#cb54-1110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1111"><a href="#cb54-1111" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb54-1112"><a href="#cb54-1112" aria-hidden="true" tabindex="-1"></a>mrg_3_assign <span class="ot">&lt;-</span> mrg_3_fasta_prep <span class="sc">%&gt;%</span> <span class="fu">select</span>(seq_id, taxid, assigned_taxid)</span>
<span id="cb54-1113"><a href="#cb54-1113" aria-hidden="true" tabindex="-1"></a>blast_results_3_assign <span class="ot">&lt;-</span> <span class="fu">full_join</span>(blast_results_3_viral, mrg_3_assign, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1114"><a href="#cb54-1114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb54-1115"><a href="#cb54-1115" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb54-1116"><a href="#cb54-1116" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb54-1117"><a href="#cb54-1117" aria-hidden="true" tabindex="-1"></a>blast_results_3_out <span class="ot">&lt;-</span> blast_results_3_assign <span class="sc">%&gt;%</span></span>
<span id="cb54-1118"><a href="#cb54-1118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb54-1119"><a href="#cb54-1119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status_new =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb54-1120"><a href="#cb54-1120" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb54-1121"><a href="#cb54-1121" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb54-1122"><a href="#cb54-1122" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1123"><a href="#cb54-1123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_new =</span> <span class="fu">replace_na</span>(viral_status_new, <span class="dv">0</span>))</span>
<span id="cb54-1124"><a href="#cb54-1124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1125"><a href="#cb54-1125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1128"><a href="#cb54-1128" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1129"><a href="#cb54-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-3-fill</span></span>
<span id="cb54-1130"><a href="#cb54-1130" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_new <span class="ot">&lt;-</span> mrg_3_blast_old <span class="sc">%&gt;%</span></span>
<span id="cb54-1131"><a href="#cb54-1131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_3_out, <span class="at">by=</span><span class="st">"seq_id"</span>)</span>
<span id="cb54-1132"><a href="#cb54-1132" aria-hidden="true" tabindex="-1"></a>mrg_3_blast <span class="ot">&lt;-</span> mrg_3_blast_new <span class="sc">%&gt;%</span></span>
<span id="cb54-1133"><a href="#cb54-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">pmax</span>(viral_status_old, viral_status_new, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb54-1134"><a href="#cb54-1134" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb54-1135"><a href="#cb54-1135" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> viral_status <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb54-1136"><a href="#cb54-1136" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill_fwd <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">select</span>(attempt, seq_id, adj_score_fwd) <span class="sc">%&gt;%</span></span>
<span id="cb54-1137"><a href="#cb54-1137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"attempt"</span>, <span class="at">values_from=</span><span class="st">"adj_score_fwd"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1138"><a href="#cb54-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>seq_id, <span class="at">names_to =</span> <span class="st">"attempt"</span>, <span class="at">values_to =</span> <span class="st">"adj_score_fwd"</span>)</span>
<span id="cb54-1139"><a href="#cb54-1139" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill_rev <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">select</span>(attempt, seq_id, adj_score_rev) <span class="sc">%&gt;%</span></span>
<span id="cb54-1140"><a href="#cb54-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"attempt"</span>, <span class="at">values_from=</span><span class="st">"adj_score_rev"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1141"><a href="#cb54-1141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>seq_id, <span class="at">names_to =</span> <span class="st">"attempt"</span>, <span class="at">values_to =</span> <span class="st">"adj_score_rev"</span>)</span>
<span id="cb54-1142"><a href="#cb54-1142" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill_ass <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">select</span>(attempt, seq_id, assigned_hv) <span class="sc">%&gt;%</span></span>
<span id="cb54-1143"><a href="#cb54-1143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"attempt"</span>, <span class="at">values_from=</span><span class="st">"assigned_hv"</span>, <span class="at">values_fill=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1144"><a href="#cb54-1144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>seq_id, <span class="at">names_to =</span> <span class="st">"attempt"</span>, <span class="at">values_to =</span> <span class="st">"assigned_hv"</span>)</span>
<span id="cb54-1145"><a href="#cb54-1145" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill_hit <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">select</span>(attempt, seq_id, hit_hv) <span class="sc">%&gt;%</span></span>
<span id="cb54-1146"><a href="#cb54-1146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"attempt"</span>, <span class="at">values_from=</span><span class="st">"hit_hv"</span>, <span class="at">values_fill=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1147"><a href="#cb54-1147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>seq_id, <span class="at">names_to =</span> <span class="st">"attempt"</span>, <span class="at">values_to =</span> <span class="st">"hit_hv"</span>)</span>
<span id="cb54-1148"><a href="#cb54-1148" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill_ref <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seq_id, sample, viral_status_out, taxid) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seq_id, sample, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">taxid =</span> taxid[<span class="dv">1</span>], <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb54-1149"><a href="#cb54-1149" aria-hidden="true" tabindex="-1"></a>mrg_3_blast_fill <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_3_blast_fill_fwd, mrg_3_blast_fill_rev, </span>
<span id="cb54-1150"><a href="#cb54-1150" aria-hidden="true" tabindex="-1"></a>                              <span class="at">by=</span><span class="fu">c</span>(<span class="st">"attempt"</span>, <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1151"><a href="#cb54-1151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_3_blast_fill_ass, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"attempt"</span>, <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1152"><a href="#cb54-1152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_3_blast_fill_hit, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"attempt"</span>, <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1153"><a href="#cb54-1153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_3_blast_fill_ref, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1154"><a href="#cb54-1154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb54-1155"><a href="#cb54-1155" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1156"><a href="#cb54-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1159"><a href="#cb54-1159" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1160"><a href="#cb54-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-f1-3</span></span>
<span id="cb54-1161"><a href="#cb54-1161" aria-hidden="true" tabindex="-1"></a>attempts <span class="ot">&lt;-</span> mrg_3_blast_fill <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(attempt) <span class="sc">%&gt;%</span> summarize <span class="sc">%&gt;%</span> <span class="fu">pull</span>(attempt)</span>
<span id="cb54-1162"><a href="#cb54-1162" aria-hidden="true" tabindex="-1"></a>stats_3_raw <span class="ot">&lt;-</span> mrg_3_blast_fill <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(attempt) <span class="sc">%&gt;%</span> group_split <span class="sc">%&gt;%</span></span>
<span id="cb54-1163"><a href="#cb54-1163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(range_f1)</span>
<span id="cb54-1164"><a href="#cb54-1164" aria-hidden="true" tabindex="-1"></a>stats_3 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="cf">function</span>(n) stats_3_raw[[n]] <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span>attempts[n])) <span class="sc">%&gt;%</span></span>
<span id="cb54-1165"><a href="#cb54-1165" aria-hidden="true" tabindex="-1"></a>  bind_rows</span>
<span id="cb54-1166"><a href="#cb54-1166" aria-hidden="true" tabindex="-1"></a>g_stats_3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_3, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>attempt, <span class="at">linetype=</span>conj_label)) <span class="sc">+</span></span>
<span id="cb54-1167"><a href="#cb54-1167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb54-1168"><a href="#cb54-1168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="cn">NA</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1169"><a href="#cb54-1169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1170"><a href="#cb54-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name=</span><span class="st">"Configuration"</span>) <span class="sc">+</span></span>
<span id="cb54-1171"><a href="#cb54-1171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name=</span><span class="st">"Threshold type"</span>) <span class="sc">+</span></span>
<span id="cb54-1172"><a href="#cb54-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>metric, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb54-1173"><a href="#cb54-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>), <span class="at">linetype=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb54-1174"><a href="#cb54-1174" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb54-1175"><a href="#cb54-1175" aria-hidden="true" tabindex="-1"></a>g_stats_3</span>
<span id="cb54-1176"><a href="#cb54-1176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1177"><a href="#cb54-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1178"><a href="#cb54-1178" aria-hidden="true" tabindex="-1"></a>At a disjunctive threshold of 20, Bowtie2(c) (<span class="in">`--very-sensitive-local`</span>) performs the best, with an F1 score of 0.979; Bowtie2(b) (<span class="in">`--sensitive-local`</span>) is very close behind. In both cases, precision (<span class="sc">\&gt;</span>=0.988) is higher than sensitivity (\~0.969), suggesting that any further improvements would come from investigating low-scoring true-positives:</span>
<span id="cb54-1179"><a href="#cb54-1179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1182"><a href="#cb54-1182" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1183"><a href="#cb54-1183" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-3</span></span>
<span id="cb54-1184"><a href="#cb54-1184" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb54-1185"><a href="#cb54-1185" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb54-1186"><a href="#cb54-1186" aria-hidden="true" tabindex="-1"></a>fp_3 <span class="ot">&lt;-</span> mrg_3_blast <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(attempt, viral_status_out, </span>
<span id="cb54-1187"><a href="#cb54-1187" aria-hidden="true" tabindex="-1"></a>                           <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb54-1188"><a href="#cb54-1188" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span> </span>
<span id="cb54-1189"><a href="#cb54-1189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(attempt, viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-1190"><a href="#cb54-1190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1191"><a href="#cb54-1191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1192"><a href="#cb54-1192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(taxid <span class="sc">==</span> <span class="dv">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, name),</span>
<span id="cb54-1193"><a href="#cb54-1193" aria-hidden="true" tabindex="-1"></a>         <span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold)</span>
<span id="cb54-1194"><a href="#cb54-1194" aria-hidden="true" tabindex="-1"></a>fp_3_major_tab <span class="ot">&lt;-</span> fp_3 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(major) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb54-1195"><a href="#cb54-1195" aria-hidden="true" tabindex="-1"></a>fp_3_major_list <span class="ot">&lt;-</span> fp_3_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb54-1196"><a href="#cb54-1196" aria-hidden="true" tabindex="-1"></a>fp_3_major <span class="ot">&lt;-</span> fp_3 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1197"><a href="#cb54-1197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(attempt, viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb54-1198"><a href="#cb54-1198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_3_major_list),</span>
<span id="cb54-1199"><a href="#cb54-1199" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb54-1200"><a href="#cb54-1200" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">paste</span>(<span class="st">"VS:"</span>, viral_status_out)) <span class="co">#ifelse(viral_status_out, "True positive", "False positive"))</span></span>
<span id="cb54-1201"><a href="#cb54-1201" aria-hidden="true" tabindex="-1"></a>palette_fp_3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Dark2"</span>), <span class="st">"#999999"</span>)</span>
<span id="cb54-1202"><a href="#cb54-1202" aria-hidden="true" tabindex="-1"></a>g_fp_3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_3_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb54-1203"><a href="#cb54-1203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb54-1204"><a href="#cb54-1204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb54-1205"><a href="#cb54-1205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1206"><a href="#cb54-1206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_fp_3, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb54-1207"><a href="#cb54-1207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(attempt<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb54-1208"><a href="#cb54-1208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb54-1209"><a href="#cb54-1209" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1210"><a href="#cb54-1210" aria-hidden="true" tabindex="-1"></a>g_fp_3</span>
<span id="cb54-1211"><a href="#cb54-1211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1212"><a href="#cb54-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1213"><a href="#cb54-1213" aria-hidden="true" tabindex="-1"></a>I suspect that small additional gains are possible here, since I'm suspicious about the low-scoring "true-positives" mapping to human betaherpesvirus 5 and human mastadenovirus F. However, I've already spent a long time optimizing the results for this dataset, and the results are now good enough that I feel okay with leaving this here. Going forward, I'll use Bowtie2(c) as my alignment strategy for the HV identification pipeline.</span>
<span id="cb54-1214"><a href="#cb54-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1215"><a href="#cb54-1215" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: analysis</span></span>
<span id="cb54-1216"><a href="#cb54-1216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1217"><a href="#cb54-1217" aria-hidden="true" tabindex="-1"></a>After several rounds of validation and refinement, we finally come to actually analyzing the human-infecting virus content of Spurbeck et al. This section might seem disappointingly short compared to all the effort expended to get here.</span>
<span id="cb54-1218"><a href="#cb54-1218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1221"><a href="#cb54-1221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1222"><a href="#cb54-1222" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-ra-calc</span></span>
<span id="cb54-1223"><a href="#cb54-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1224"><a href="#cb54-1224" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb54-1225"><a href="#cb54-1225" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb54-1226"><a href="#cb54-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, group, date, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb54-1227"><a href="#cb54-1227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1228"><a href="#cb54-1228" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts &amp; RA</span></span>
<span id="cb54-1229"><a href="#cb54-1229" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg_3 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(attempt <span class="sc">==</span> <span class="st">"Bowtie2(c)"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1230"><a href="#cb54-1230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb54-1231"><a href="#cb54-1231" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>)</span>
<span id="cb54-1232"><a href="#cb54-1232" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb54-1233"><a href="#cb54-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1234"><a href="#cb54-1234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>),</span>
<span id="cb54-1235"><a href="#cb54-1235" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb54-1236"><a href="#cb54-1236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1237"><a href="#cb54-1237" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb54-1238"><a href="#cb54-1238" aria-hidden="true" tabindex="-1"></a>read_counts_group <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb54-1239"><a href="#cb54-1239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb54-1240"><a href="#cb54-1240" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1241"><a href="#cb54-1241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb54-1242"><a href="#cb54-1242" aria-hidden="true" tabindex="-1"></a>read_counts_total <span class="ot">&lt;-</span> read_counts_group <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-1243"><a href="#cb54-1243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb54-1244"><a href="#cb54-1244" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1245"><a href="#cb54-1245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb54-1246"><a href="#cb54-1246" aria-hidden="true" tabindex="-1"></a>         <span class="at">group =</span> <span class="st">"All groups"</span>)</span>
<span id="cb54-1247"><a href="#cb54-1247" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts_group <span class="sc">%&gt;%</span></span>
<span id="cb54-1248"><a href="#cb54-1248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(read_counts_total) <span class="sc">%&gt;%</span></span>
<span id="cb54-1249"><a href="#cb54-1249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(group) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">str_detect</span>(group, <span class="st">"All groups"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1250"><a href="#cb54-1250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">fct_inorder</span>(group))</span>
<span id="cb54-1251"><a href="#cb54-1251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1252"><a href="#cb54-1252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1253"><a href="#cb54-1253" aria-hidden="true" tabindex="-1"></a>Applying a disjunctive cutoff at S=20 identifies 9,990 reads as human-viral out of 1.84B total reads, for a relative HV abundance of $5.44 \times 10^{-6}$. This compares to $2.8 \times 10^{-4}$ on the public dashboard, corresponding to the results for Kraken-only identification: a roughly 2x increase, smaller than the 4-5x increases seen for Crits-Christoph and Rothman. Relative HV abundances for individual sample groups ranged from $9.19 \times 10^{-8}$ to $1.58 \times 10^{-5}$; as with total virus reads, groups F, G &amp; H showed the highest relative abundance:</span>
<span id="cb54-1254"><a href="#cb54-1254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1257"><a href="#cb54-1257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1258"><a href="#cb54-1258" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-ra-plot</span></span>
<span id="cb54-1259"><a href="#cb54-1259" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb54-1260"><a href="#cb54-1260" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>p_reads_hv, <span class="at">color=</span>group)) <span class="sc">+</span></span>
<span id="cb54-1261"><a href="#cb54-1261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb54-1262"><a href="#cb54-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb54-1263"><a href="#cb54-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>, <span class="at">name =</span> <span class="st">"Group"</span>) <span class="sc">+</span></span>
<span id="cb54-1264"><a href="#cb54-1264" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1265"><a href="#cb54-1265" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb54-1266"><a href="#cb54-1266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1267"><a href="#cb54-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1268"><a href="#cb54-1268" aria-hidden="true" tabindex="-1"></a>Digging into particular viruses, we see that *Mamastrovirus*, *Mastadenovirus, Rotavirus* and *Betacoronavirus* are the most abundant genera across samples:</span>
<span id="cb54-1269"><a href="#cb54-1269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1272"><a href="#cb54-1272" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1273"><a href="#cb54-1273" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-composition-4</span></span>
<span id="cb54-1274"><a href="#cb54-1274" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb54-1275"><a href="#cb54-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1276"><a href="#cb54-1276" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb54-1277"><a href="#cb54-1277" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb54-1278"><a href="#cb54-1278" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb54-1279"><a href="#cb54-1279" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb54-1280"><a href="#cb54-1280" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb54-1281"><a href="#cb54-1281" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694009</span>] <span class="ot">&lt;-</span> <span class="st">"SARS-CoV"</span></span>
<span id="cb54-1282"><a href="#cb54-1282" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb54-1283"><a href="#cb54-1283" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">568715</span>] <span class="ot">&lt;-</span> <span class="st">"Astrovirus MLB1"</span></span>
<span id="cb54-1284"><a href="#cb54-1284" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194965</span>] <span class="ot">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span id="cb54-1285"><a href="#cb54-1285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1286"><a href="#cb54-1286" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb54-1287"><a href="#cb54-1287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1288"><a href="#cb54-1288" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb54-1289"><a href="#cb54-1289" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb54-1290"><a href="#cb54-1290" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb54-1291"><a href="#cb54-1291" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb54-1292"><a href="#cb54-1292" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb54-1293"><a href="#cb54-1293" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb54-1294"><a href="#cb54-1294" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb54-1295"><a href="#cb54-1295" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1296"><a href="#cb54-1296" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb54-1297"><a href="#cb54-1297" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb54-1298"><a href="#cb54-1298" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb54-1299"><a href="#cb54-1299" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb54-1300"><a href="#cb54-1300" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb54-1301"><a href="#cb54-1301" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb54-1302"><a href="#cb54-1302" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb54-1303"><a href="#cb54-1303" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb54-1304"><a href="#cb54-1304" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1305"><a href="#cb54-1305" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb54-1306"><a href="#cb54-1306" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb54-1307"><a href="#cb54-1307" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb54-1308"><a href="#cb54-1308" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb54-1309"><a href="#cb54-1309" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb54-1310"><a href="#cb54-1310" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb54-1311"><a href="#cb54-1311" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb54-1312"><a href="#cb54-1312" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb54-1313"><a href="#cb54-1313" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb54-1314"><a href="#cb54-1314" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1315"><a href="#cb54-1315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb54-1316"><a href="#cb54-1316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb54-1317"><a href="#cb54-1317" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb54-1318"><a href="#cb54-1318" aria-hidden="true" tabindex="-1"></a>hv_reads_genera <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb54-1319"><a href="#cb54-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1320"><a href="#cb54-1320" aria-hidden="true" tabindex="-1"></a><span class="co"># Count relative abundance for genera</span></span>
<span id="cb54-1321"><a href="#cb54-1321" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_raw <span class="ot">&lt;-</span> hv_reads_genera <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(group, name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1322"><a href="#cb54-1322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(hv_status), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1323"><a href="#cb54-1323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(read_counts_agg <span class="sc">%&gt;%</span> <span class="fu">select</span>(group, n_reads_raw), <span class="at">by=</span><span class="st">"group"</span>)</span>
<span id="cb54-1324"><a href="#cb54-1324" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_all <span class="ot">&lt;-</span> hv_genera_counts_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1325"><a href="#cb54-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb54-1326"><a href="#cb54-1326" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1327"><a href="#cb54-1327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"All groups"</span>)</span>
<span id="cb54-1328"><a href="#cb54-1328" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_genera_counts_raw, hv_genera_counts_all) <span class="sc">%&gt;%</span></span>
<span id="cb54-1329"><a href="#cb54-1329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb54-1330"><a href="#cb54-1330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1331"><a href="#cb54-1331" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute ranks for species and genera and restrict to high-ranking taxa</span></span>
<span id="cb54-1332"><a href="#cb54-1332" aria-hidden="true" tabindex="-1"></a>max_rank_genera <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb54-1333"><a href="#cb54-1333" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_ranked <span class="ot">&lt;-</span> hv_genera_counts_agg <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb54-1334"><a href="#cb54-1334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">rank</span>(<span class="fu">desc</span>(n_reads_hv), <span class="at">ties.method=</span><span class="st">"max"</span>),</span>
<span id="cb54-1335"><a href="#cb54-1335" aria-hidden="true" tabindex="-1"></a>         <span class="at">highrank =</span> rank <span class="sc">&lt;=</span> max_rank_genera) <span class="sc">%&gt;%</span></span>
<span id="cb54-1336"><a href="#cb54-1336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb54-1337"><a href="#cb54-1337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">highrank_any =</span> <span class="fu">any</span>(highrank),</span>
<span id="cb54-1338"><a href="#cb54-1338" aria-hidden="true" tabindex="-1"></a>         <span class="at">name_display =</span> <span class="fu">ifelse</span>(highrank_any, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1339"><a href="#cb54-1339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name_display) <span class="sc">%&gt;%</span></span>
<span id="cb54-1340"><a href="#cb54-1340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_rank =</span> <span class="fu">mean</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1341"><a href="#cb54-1341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(mean_rank) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb54-1342"><a href="#cb54-1342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">fct_inorder</span>(name_display)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1343"><a href="#cb54-1343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">str_detect</span>(group, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1344"><a href="#cb54-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">fct_inorder</span>(group))</span>
<span id="cb54-1345"><a href="#cb54-1345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1346"><a href="#cb54-1346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1347"><a href="#cb54-1347" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition</span></span>
<span id="cb54-1348"><a href="#cb54-1348" aria-hidden="true" tabindex="-1"></a>palette_rank <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>), <span class="st">"#888888"</span>)</span>
<span id="cb54-1349"><a href="#cb54-1349" aria-hidden="true" tabindex="-1"></a>g_vcomp_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_genera_counts_ranked, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>n_reads_hv, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb54-1350"><a href="#cb54-1350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb54-1351"><a href="#cb54-1351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_rank, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb54-1352"><a href="#cb54-1352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Fraction of HV reads"</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb54-1353"><a href="#cb54-1353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb54-1354"><a href="#cb54-1354" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb54-1355"><a href="#cb54-1355" aria-hidden="true" tabindex="-1"></a>g_vcomp_genera</span>
<span id="cb54-1356"><a href="#cb54-1356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1357"><a href="#cb54-1357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1358"><a href="#cb54-1358" aria-hidden="true" tabindex="-1"></a>*Mamastrovirus* and *Rotavirus* are predominantly enteric RNA viruses whose prominence here makes sense, while the prevalence of *Betacoronavirus* is probably a result of the ongoing SARS-CoV-2 pandemic. As a DNA virus, the abundance of *Mastadenovirus* is a bit more surprising; however, this finding is consistent with the public dashboard, and is also borne out by the other BLASTN matches for these sequences, all of which are other adenovirus taxa:</span>
<span id="cb54-1359"><a href="#cb54-1359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1362"><a href="#cb54-1362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb54-1363"><a href="#cb54-1363" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-mastadenovirus</span></span>
<span id="cb54-1364"><a href="#cb54-1364" aria-hidden="true" tabindex="-1"></a>masta_ids <span class="ot">&lt;-</span> hv_reads_genera <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name<span class="sc">==</span><span class="st">"Mastadenovirus"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(seq_id)</span>
<span id="cb54-1365"><a href="#cb54-1365" aria-hidden="true" tabindex="-1"></a>masta_hits <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(blast_results_2_paired <span class="sc">%&gt;%</span> <span class="fu">full_join</span>(mrg_2_blast <span class="sc">%&gt;%</span> <span class="fu">select</span>(seq_id, sample, seq_num), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)),</span>
<span id="cb54-1366"><a href="#cb54-1366" aria-hidden="true" tabindex="-1"></a>                        blast_results_3_paired) <span class="sc">%&gt;%</span></span>
<span id="cb54-1367"><a href="#cb54-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>sample, <span class="sc">-</span>seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb54-1368"><a href="#cb54-1368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> masta_ids) <span class="sc">%&gt;%</span></span>
<span id="cb54-1369"><a href="#cb54-1369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid))</span>
<span id="cb54-1370"><a href="#cb54-1370" aria-hidden="true" tabindex="-1"></a>masta_hits_sorted <span class="ot">&lt;-</span> masta_hits <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(staxid) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb54-1371"><a href="#cb54-1371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1372"><a href="#cb54-1372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">fct_inorder</span>(name))</span>
<span id="cb54-1373"><a href="#cb54-1373" aria-hidden="true" tabindex="-1"></a>masta_hits_sorted_head <span class="ot">&lt;-</span> masta_hits_sorted <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb54-1374"><a href="#cb54-1374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name=</span><span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(name)))</span>
<span id="cb54-1375"><a href="#cb54-1375" aria-hidden="true" tabindex="-1"></a>g_masta <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(masta_hits_sorted_head,</span>
<span id="cb54-1376"><a href="#cb54-1376" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">y=</span><span class="fu">fct_inorder</span>(name))) <span class="sc">+</span> <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb54-1377"><a href="#cb54-1377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"# Mapped read pairs"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span></span>
<span id="cb54-1378"><a href="#cb54-1378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb54-1379"><a href="#cb54-1379" aria-hidden="true" tabindex="-1"></a>g_masta</span>
<span id="cb54-1380"><a href="#cb54-1380" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb54-1381"><a href="#cb54-1381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1382"><a href="#cb54-1382" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb54-1383"><a href="#cb54-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1384"><a href="#cb54-1384" aria-hidden="true" tabindex="-1"></a>Compared to the last few datasets I analyzed, the analysis of Spurbeck took a long time, numerous attempts, and a lot of computational resources. However, as I said at the start of this post, I'm happy with the outcome and am confident it will improve analysis of future datasets. While the overall prevalence of human-infecting viruses is fairly low in Spurbeck compared to other wastewater datasets I've looked at, its inclusion as a core dataset for the P2RA analysis make it especially important to process in a reliable and high-quality manner.</span>
<span id="cb54-1385"><a href="#cb54-1385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-1386"><a href="#cb54-1386" aria-hidden="true" tabindex="-1"></a>Next, I'll turn my attention to more datasets included in the P2RA analysis, as well as some air-sampling datasets we're interested in for another project.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>