<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-04-30">
<title>Will’s Public NAO Notebook - Workflow analysis of Brinch et al.&nbsp;(2020)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Brinch et al.&nbsp;(2020)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Wastewater from Copenhagen.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 30, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>In this entry, I’m analyzing <a href="https://doi.org/10.1128%2FmSystems.00841-20">Brinch et al.&nbsp;(2020)</a>, a large DNA-sequencing study of raw influent samples from three Copenhagen treatment plants. 12-hour composite samples were collected “at irregular intervals” between 2015 and 2018; the sample processing protocol isn’t described in great detail, but samples were centrifuged and the pellet retained for extraction, a procedure that (as in Bengtsson-Palme) we expect to select against viruses. About half the samples were sequenced on an Illumina MiSeq, the rest on a NextSeq; “the change of sequencing platform was circumstantial and was not part of the experiment design.” The platform used varied in temporal blocks (blue = NextSeq):</p>
<p><img src="../img/2024-04-30_brinch.png" class="img-fluid"></p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The Brinch dataset contained sample information from 322 samples, composed as follows:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Importing the data is a bit more complicated this time as the samples are split across two pipeline runs</span></span>
<span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-04-30_brinch"</span></span>
<span><span class="va">data_dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"/"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data input paths</span></span>
<span><span class="va">libraries_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_basic_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_adapter_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_base_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">instrument_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"instruments.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">libraries_paths</span>, <span class="va">read_csv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span><span class="va">instruments</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">instrument_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>instrument <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"^Illumina "</span>, <span class="st">""</span>, <span class="va">instrument</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>year <span class="op">=</span> <span class="fu">year</span><span class="op">(</span><span class="va">date</span><span class="op">)</span>,</span>
<span>         month <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"-\\d+$"</span>, <span class="st">""</span>, <span class="va">date</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span>, <span class="va">date</span>, <span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>         location_alias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">location</span> <span class="op">==</span> <span class="st">"Amager"</span>, <span class="st">"RL"</span>,</span>
<span>                                 <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">location</span> <span class="op">==</span> <span class="st">"Valby"</span>, <span class="st">"RD"</span>, <span class="st">"RA"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">month</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample_alias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">location_alias</span>, <span class="va">month</span>, <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>,</span>
<span>         sample_alias <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample_alias</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">instruments</span>, by<span class="op">=</span><span class="st">"library"</span><span class="op">)</span></span>
<span>         </span>
<span><span class="co"># Make table</span></span>
<span><span class="va">count_samples</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">year</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"year"</span>, values_from<span class="op">=</span><span class="st">"n"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>Location <span class="op">=</span> <span class="va">location</span><span class="op">)</span></span>
<span><span class="va">count_samples</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Location"],"name":[1],"type":["fct"],"align":["left"]},{"label":["2015"],"name":[2],"type":["int"],"align":["right"]},{"label":["2016"],"name":[3],"type":["int"],"align":["right"]},{"label":["2017"],"name":[4],"type":["int"],"align":["right"]},{"label":["2018"],"name":[5],"type":["int"],"align":["right"]}],"data":[{"1":"Amager","2":"9","3":"75","4":"28","5":"11"},{"1":"Avedoere","2":"3","3":"27","4":"16","5":"9"},{"1":"Valby","2":"8","3":"72","4":"42","5":"22"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">import_basic</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span>, <span class="va">sample_alias</span>, <span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>           sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">import_basic_paired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">import_basic</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic</span><span class="op">(</span><span class="va">basic_stats_paths</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">adapter_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_base_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_seq_stats_paths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get key values for readout</span></span>
<span><span class="va">raw_read_counts</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_lib <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, rmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, rmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            rmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, </span>
<span>            rtot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            btot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span>,</span>
<span>            dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The 169 NextSeq libraries yielded 3.6M-63M (mean 23.8M) read pairs per sample, while the 153 MiSeq libraries yielded 10-7.4M (mean 2.3M) read pairs. The total number of read pairs across all instruments was 4.4B read pairs (1.27 terabases of sequence), of which 93% of read pairs (89% of bases) came from NextSeq runs.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>, <span class="va">location</span>, <span class="va">instrument</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">instrument</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span><span class="va">scale_fill_loc</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span></span>
<span><span class="va">scale_fill_ins</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Dark2"</span>, name<span class="op">=</span><span class="st">"Instrument"</span><span class="op">)</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, </span>
<span>                  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">date</span>, y<span class="op">=</span><span class="va">value</span>, fill<span class="op">=</span><span class="va">instrument</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">location</span>,<span class="va">date</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_ins</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span><span class="op">~</span><span class="va">location</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space<span class="op">=</span><span class="st">"free_x"</span>, switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>In most samples, read qualities were high at the 5’ end of the read but tailed off substantially over the course of the read, especially in long MiSeq reads. A minority of MiSeq samples had poor read quality across the length of the read. Adapter levels were fairly low, but significant enough to require trimming. Duplicate levels were low (0-11%, mean 3%) in MiSeq libraries (unsurprising given their small size) but moderate (3-59%, mean 28%) in the larger NextSeq libraries.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">scale_color_loc</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span></span>
<span><span class="va">scale_color_ins</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Dark2"</span>, name<span class="op">=</span><span class="st">"Instrument"</span><span class="op">)</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">instrument</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_ins</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">rmax</span> <span class="op">&lt;-</span> <span class="fl">240</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">1</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">rmax</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">rmax</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. As expected given the observed difference in duplication levels, few reads were lost during deduplication in MiSeq libraries, while a somewhat larger (but still not huge) number were lost in NextSeq libraries. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">sample_alias</span>, <span class="va">location</span>, <span class="va">date</span>, <span class="va">instrument</span>, </span>
<span>         <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span>Instrument<span class="op">=</span><span class="va">instrument</span>, Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Instrument"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Stage"],"name":[2],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[3],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[4],"type":["chr"],"align":["left"]}],"data":[{"1":"MiSeq","2":"Trimming & filtering","3":"2.7-66.4 (mean 6.3)","4":"2.7-66.4 (mean 6.3)"},{"1":"MiSeq","2":"Deduplication","3":"3.9-66.4 (mean 8.6)","4":"0-8.5 (mean 2.3)"},{"1":"MiSeq","2":"Initial ribodepletion","3":"4.4-68 (mean 9.1)","4":"0-5.9 (mean 0.5)"},{"1":"MiSeq","2":"Secondary ribodepletion","3":"4.5-68.2 (mean 9.3)","4":"0-0.2 (mean 0.1)"},{"1":"NextSeq 550","2":"Trimming & filtering","3":"2.8-10 (mean 4.6)","4":"2.8-10 (mean 4.6)"},{"1":"NextSeq 550","2":"Deduplication","3":"6.9-51.3 (mean 24.4)","4":"1.9-47.4 (mean 19.8)"},{"1":"NextSeq 550","2":"Initial ribodepletion","3":"7.4-51.6 (mean 24.8)","4":"0.2-0.7 (mean 0.4)"},{"1":"NextSeq 550","2":"Secondary ribodepletion","3":"7.5-51.6 (mean 24.9)","4":"0-0.1 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">instrument</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_ins</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">location</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">instrument</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_ins</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">location</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Data cleaning was very successful at removing adapters and mostly successful at improving read qualities, though some libraries still dipped below my preferred quality thresholds later in the read:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">instrument</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_ins</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 28% to 8% in NextSeq libraries and from 3% to 1% in MiSeq samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">instrument</span>,<span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import composition data</span></span>
<span><span class="va">comp_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"taxonomic_composition.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">comp_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">classification</span>, levels <span class="op">=</span> <span class="va">classifications</span><span class="op">)</span>,</span>
<span>         n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">instrument</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">instrument</span><span class="op">~</span><span class="va">location</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, ncol<span class="op">=</span><span class="fl">3</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line<span class="op">=</span><span class="cn">FALSE</span>, width<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_xblank</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">comp_minor</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">comp_minor</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">instrument</span>, <span class="va">classification</span>, read_fraction<span class="op">=</span><span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">instrument</span>, <span class="va">classification</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["instrument"],"name":[1],"type":["chr"],"align":["left"]},{"label":["classification"],"name":[2],"type":["fct"],"align":["left"]},{"label":["read_fraction"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"MiSeq","2":"Excluded","3":"44-90% (mean 63%)"},{"1":"MiSeq","2":"Ribosomal","3":"0-5.9% (mean 0.65%)"},{"1":"MiSeq","2":"Bacterial","3":"9.2-55% (mean 35%)"},{"1":"MiSeq","2":"Archaeal","3":"0-0.32% (mean 0.09%)"},{"1":"MiSeq","2":"Viral","3":"0-7.7% (mean 0.41%)"},{"1":"MiSeq","2":"Human","3":"0-19% (mean 0.42%)"},{"1":"NextSeq 550","2":"Excluded","3":"51-88% (mean 70%)"},{"1":"NextSeq 550","2":"Ribosomal","3":"0.32-0.8% (mean 0.48%)"},{"1":"NextSeq 550","2":"Bacterial","3":"11-48% (mean 29%)"},{"1":"NextSeq 550","2":"Archaeal","3":"0.0099-0.42% (mean 0.075%)"},{"1":"NextSeq 550","2":"Viral","3":"0.017-1.1% (mean 0.34%)"},{"1":"NextSeq 550","2":"Human","3":"0.01-1.5% (mean 0.1%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>In most samples, the majority of reads were either filtered, duplicates, or unassigned. Among assigned reads, the vast majority in most samples were bacterial, which is unsurprising given the sample processing protocols used. A few samples showed elevated levels of human reads, which might be real or due to experimenter contamination. Viral fraction averaged 0.41% in MiSeq reads and 0.34% in NextSeq reads.</p>
<p>As is common for DNA data, viral reads were overwhelmingly dominated by <em>Caudoviricetes</em> phages:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get Kraken reports</span></span>
<span><span class="va">reports_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"kraken_reports.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">reports</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">reports_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to viral taxa</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"minimizers"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span></span>
<span><span class="co"># Identify major viral classes</span></span>
<span><span class="va">viral_classes_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_classes_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">instrument</span>, <span class="va">location</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_minor</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">instrument</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">instrument</span>, <span class="va">location</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_classes_major</span>, <span class="va">viral_classes_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_classes_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_classes</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_classes_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">g_classes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/viral-class-composition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples. A total of 17,448 reads were identified as putatively human-viral:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import HV read data</span></span>
<span><span class="va">hv_reads_filtered_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">hv_reads_filtered_paths</span>, <span class="va">read_tsv</span>,</span>
<span>                            show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collapse multi-entry sequences</span></span>
<span><span class="va">rmax</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">max</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">collapse</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">==</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">x</span>, collapse<span class="op">=</span><span class="st">"/"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>            genome_id <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span>,</span>
<span>            taxid_best <span class="op">=</span> <span class="va">taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            best_alignment_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_fwd</span><span class="op">)</span>,</span>
<span>            best_alignment_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_rev</span><span class="op">)</span>,</span>
<span>            query_len_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_fwd</span><span class="op">)</span>,</span>
<span>            query_len_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_rev</span><span class="op">)</span>,</span>
<span>            query_seq_fwd <span class="op">=</span> <span class="va">query_seq_fwd</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_fwd</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            query_seq_rev <span class="op">=</span> <span class="va">query_seq_rev</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_rev</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            classified <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">classified</span><span class="op">)</span>,</span>
<span>            assigned_name <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">assigned_name</span><span class="op">)</span>,</span>
<span>            assigned_taxid_best <span class="op">=</span> <span class="va">assigned_taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            assigned_taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">assigned_taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            assigned_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">assigned_hv</span><span class="op">)</span>,</span>
<span>            hit_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">hit_hv</span><span class="op">)</span>,</span>
<span>            encoded_hits <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">encoded_hits</span><span class="op">)</span>,</span>
<span>            adj_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>,</span>
<span>            adj_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span></span>
<span>            <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot results</span></span>
<span><span class="va">geom_vhist</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_histogram</span>, binwidth<span class="op">=</span><span class="fl">5</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">g_vhist_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> </span>
<span><span class="va">g_vhist_0</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg</span><span class="op">)</span></span>
<span><span class="va">g_vhist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-hv-scores-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>BLASTing these reads against nt:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import paired BLAST results</span></span>
<span><span class="va">blast_paired_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"hv_hits_blast_paired.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">blast_paired_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_paired</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">match_taxid</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxid_1</span>, <span class="va">taxid_2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"/"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"/"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">p1</span><span class="op">|</span><span class="va">p2</span><span class="op">|</span><span class="va">p3</span>, <span class="cn">NULL</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">adj_score_max</span><span class="op">)</span></span>
<span><span class="va">blast_assign</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">blast_viral</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_out</span> <span class="op">&lt;-</span> <span class="va">blast_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg</span>, <span class="va">blast_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot RNA</span></span>
<span><span class="va">g_vhist_1</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg_blast</span>, mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>fill<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span></span>
<span><span class="va">g_vhist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-blast-results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Applying my standard disjunctive score threshold of 20 achieved precision, sensitivity and F1 scores all &gt;99%:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>retain_score <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>           retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span><span class="op">)</span></span>
<span>  <span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="va">threshold</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adjusted Score Threshold (Disjunctive)"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-f1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span>Threshold<span class="op">=</span><span class="va">threshold</span>, Metric<span class="op">=</span><span class="va">metric</span>, Value<span class="op">=</span><span class="va">value</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Threshold"],"name":[1],"type":["int"],"align":["right"]},{"label":["Metric"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"20","2":"sensitivity","3":"0.9907761"},{"1":"20","2":"specificity","3":"0.7587822"},{"1":"20","2":"precision","3":"0.9939294"},{"1":"20","2":"f1","3":"0.9923502"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Looking into the composition of different read groups, a plurality of high-scoring false positives map to human alphaherpesvirus 1 strain RH2 according to Bowtie2. BLASTN maps these sequences to a variety of bacterial taxa, especially <em>E. coli</em> and various <em>Klebsiella</em> species. I’m already screening against one strain of E. coli, so if I wanted to tackle this I’d need to add more strains or another species, probably <em>Klebsiella pneumoniae</em>. However, in this case the scores are good enough that I don’t think it’s worth it.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.04</span></span>
<span></span>
<span><span class="co"># Add missing viral taxa</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">211787</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">509154</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">493803</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">427343</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">340907</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194959</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span></span>
<span></span>
<span><span class="co"># Prepare data</span></span>
<span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_major</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_fp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/fp-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxid_rh2</span> <span class="op">&lt;-</span> <span class="fl">946522</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span>                         <span class="fl">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span>                         <span class="fl">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span>                         <span class="fl">187126</span>, <span class="st">"Nesoenas mayeri"</span>,</span>
<span>                         <span class="fl">244366</span>, <span class="st">"Klebsiella variicola"</span>,</span>
<span>                         <span class="fl">208224</span>, <span class="st">"Enterobacter kobei"</span></span>
<span>                         <span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="va">tax_names_new</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tax_names</span><span class="op">$</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span></span>
<span><span class="va">ref_name_rh2</span> <span class="op">&lt;-</span> <span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">ref_taxid_rh2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get major matches</span></span>
<span><span class="va">mrg_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid_best</span> <span class="op">==</span> <span class="va">ref_taxid_rh2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">highscore</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid_best<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid_best"</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count</span> <span class="op">&lt;-</span> <span class="va">fp_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, </span>
<span>           <span class="va">taxid_best</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count_major</span> <span class="op">&lt;-</span> <span class="va">fp_staxid_count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, </span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_staxid_count_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">status_display</span><span class="op">~</span><span class="va">score_display</span>, scales<span class="op">=</span><span class="st">"free"</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_rh2</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_rh2</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/ptt-blast-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-overall-relative-abundance" class="level1"><h1>Human-infecting viruses: overall relative abundance</h1>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">date</span>, <span class="va">year</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid_all <span class="op">=</span> <span class="va">taxid</span>, taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_grp</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">year</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span> <span class="st">"All samples"</span>, year <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">year</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_counts_yr</span> <span class="op">&lt;-</span> <span class="va">read_counts_grp</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">year</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">read_counts_loc</span> <span class="op">&lt;-</span> <span class="va">read_counts_grp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>year <span class="op">=</span> <span class="st">"All years"</span><span class="op">)</span></span>
<span><span class="va">read_counts_tot</span> <span class="op">&lt;-</span> <span class="va">read_counts_loc</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">year</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_grp</span>, <span class="va">read_counts_yr</span>,</span>
<span>                             <span class="va">read_counts_loc</span>, <span class="va">read_counts_tot</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">location</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">libraries</span><span class="op">$</span><span class="va">location</span><span class="op">)</span>, <span class="st">"All locations"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         year <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">year</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fl">2015</span><span class="op">:</span><span class="fl">2018</span><span class="op">)</span>, <span class="st">"All years"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Applying a disjunctive cutoff at S=20 identifies 16,967 read pairs as human-viral. This gives an overall relative HV abundance of <span class="math inline">\(3.88 \times 10^{-6}\)</span>. Measured RAs are somewhat lower on average in 2015, and higher than average in 2016, but overall year-on-year variation isn’t huge on a log scale:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">year</span>, color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1e-6</span>, <span class="fl">1e-5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_loc</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/plot-hv-ra-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This overall RA is somewhat lower than most RNA datasets I’ve previously analyzed, and on the low end for DNA datasets as well, probably due to the antiviral selection of the sample processing protocol used:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collate past RA values</span></span>
<span><span class="va">ra_past</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Collate new RA values</span></span>
<span><span class="va">ra_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                  <span class="st">"Brinch"</span>, <span class="fl">3.88e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">scale_color_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                 name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">ra_comp</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">ra_past</span>, <span class="va">ra_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">ra_comp</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">dataset</span>, x<span class="op">=</span><span class="va">ra</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/ra-hv-past-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-taxonomy-and-composition" class="level1"><h1>Human-infecting viruses: taxonomy and composition</h1>
<p>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 15 samples, 4 from influent and 11 from primary sludge, met this criterion.</p>
<p>At the family level, most samples across all locations were overwhelmingly dominated by <em>Adenoviridae</em>, followed distantly by <em>Polyomaviridae</em>, <em>Papillomaviridae</em> and <em>Poxviridae</em>:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">585893</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333922</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334207</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">369960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333924</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687329</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">325455</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333750</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334202</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">197911</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">186938</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333926</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337051</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337043</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334204</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334208</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333928</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337039</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333929</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337042</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334203</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333757</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337050</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333767</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333754</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687363</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687342</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687359</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194441</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334209</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194965</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333930</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337048</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337041</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 9"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337049</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 11"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337044</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 5"</span></span>
<span></span>
<span></span>
<span><span class="co"># Filter samples and add viral taxa information</span></span>
<span><span class="va">samples_keep</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_reads_hv</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">samples_keep</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_family</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"family"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.08</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">year</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">year</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">year</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Adenoviruses were in turn overwhelmingly dominated by human mastadenovirus F:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_adeno</span> <span class="op">&lt;-</span> <span class="fl">10508</span></span>
<span></span>
<span><span class="co"># Get set of adenoviridae reads</span></span>
<span><span class="va">adeno_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">adeno_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each adenoviridae species</span></span>
<span><span class="va">adeno_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_adeno <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">adeno_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_adeno</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_adeno_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">adeno_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adenoviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Adenoviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_adeno_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-species-adeno-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">adeno_species_collate</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxids_hv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">130309</span><span class="op">)</span></span>
<span><span class="va">ref_names_hv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ref_taxids_hv</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">first</span><span class="op">)</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">257877</span>, <span class="st">"Macaca thibetana thibetana"</span>,</span>
<span>                         <span class="fl">256321</span>, <span class="st">"Lentiviral transfer vector pHsCXW"</span>,</span>
<span>                         <span class="fl">419242</span>, <span class="st">"Shuttle vector pLvCmvMYOCDHA"</span>,</span>
<span>                         <span class="fl">419243</span>, <span class="st">"Shuttle vector pLvCmvLacZ"</span>,</span>
<span>                         <span class="fl">421868</span>, <span class="st">"Cloning vector pLvCmvLacZ.Gfp"</span>,</span>
<span>                         <span class="fl">421869</span>, <span class="st">"Cloning vector pLvCmvMyocardin.Gfp"</span>,</span>
<span>                         <span class="fl">426303</span>, <span class="st">"Lentiviral vector pNL-GFP-RRE(SA)"</span>,</span>
<span>                         <span class="fl">436015</span>, <span class="st">"Lentiviral transfer vector pFTMGW"</span>,</span>
<span>                         <span class="fl">454257</span>, <span class="st">"Shuttle vector pLvCmvMYOCD2aHA"</span>,</span>
<span>                         <span class="fl">476184</span>, <span class="st">"Shuttle vector pLV.mMyoD::ERT2.eGFP"</span>,</span>
<span>                         <span class="fl">476185</span>, <span class="st">"Shuttle vector pLV.hMyoD.eGFP"</span>,</span>
<span>                         <span class="fl">591936</span>, <span class="st">"Piliocolobus tephrosceles"</span>,</span>
<span>                         <span class="fl">627481</span>, <span class="st">"Lentiviral transfer vector pFTM3GW"</span>,</span>
<span>                         <span class="fl">680261</span>, <span class="st">"Self-inactivating lentivirus vector pLV.C-EF1a.cyt-bGal.dCpG"</span>,</span>
<span>                         <span class="fl">2952778</span>, <span class="st">"Expression vector pLV[Exp]-EGFP:T2A:Puro-EF1A"</span>,</span>
<span>                         <span class="fl">3022699</span>, <span class="st">"Vector PAS_122122"</span>,</span>
<span>                         <span class="fl">3025913</span>, <span class="st">"Vector pSIN-WP-mPGK-GDNF"</span>,</span>
<span>                         <span class="fl">3105863</span>, <span class="st">"Vector pLKO.1-ZsGreen1"</span>,</span>
<span>                         <span class="fl">3105864</span>, <span class="st">"Vector pLKO.1-ZsGreen1 mouse Wfs1 shRNA"</span>,</span>
<span>                         <span class="fl">3108001</span>, <span class="st">"Cloning vector pLVSIN-CMV_Neo_v4.0"</span>,</span>
<span>                         <span class="fl">3109234</span>, <span class="st">"Vector pTwist+Kan+High"</span>,</span>
<span>                         <span class="fl">3117662</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L"</span>,</span>
<span>                         <span class="fl">3117663</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L:T2A:mRuby3"</span>,</span>
<span>                         <span class="fl">3117664</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;hMAPT[NM_005910.6](ns):T2A:mRuby3"</span>,</span>
<span>                         <span class="fl">3117665</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3"</span>,</span>
<span>                         <span class="fl">3117666</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3/NFAT3 fusion protein"</span>,</span>
<span>                         <span class="fl">3117667</span>, <span class="st">"Cloning vector pLV[Exp]-Neo-mPGK&gt;{EGFP-hSEPT6}"</span>,</span>
<span>                         <span class="fl">438045</span>, <span class="st">"Xenotropic MuLV-related virus"</span>,</span>
<span>                         <span class="fl">447135</span>, <span class="st">"Myodes glareolus"</span>,</span>
<span>                         <span class="fl">590745</span>, <span class="st">"Mus musculus mobilized endogenous polytropic provirus"</span>,</span>
<span>                         <span class="fl">181858</span>, <span class="st">"Murine AIDS virus-related provirus"</span>,</span>
<span>                         <span class="fl">356663</span>, <span class="st">"Xenotropic MuLV-related virus VP35"</span>,</span>
<span>                         <span class="fl">356664</span>, <span class="st">"Xenotropic MuLV-related virus VP42"</span>,</span>
<span>                         <span class="fl">373193</span>, <span class="st">"Xenotropic MuLV-related virus VP62"</span>,</span>
<span>                         <span class="fl">286419</span>, <span class="st">"Canis lupus dingo"</span>,</span>
<span>                         <span class="fl">415978</span>, <span class="st">"Sus scrofa scrofa"</span>,</span>
<span>                         <span class="fl">494514</span>, <span class="st">"Vulpes lagopus"</span>,</span>
<span>                         <span class="fl">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span>                         <span class="fl">3119969</span>, <span class="st">"Bubalus kerabau"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span>, <span class="va">tax_names_new</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get matches</span></span>
<span><span class="va">hv_blast_staxids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">ref_taxids_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count matches</span></span>
<span><span class="va">hv_blast_counts</span> <span class="op">&lt;-</span> <span class="va">hv_blast_staxids</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset to major matches</span></span>
<span><span class="va">hv_blast_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_blast_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">25</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_blast</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_blast_counts_major</span>, mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">name_display</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span>, space<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, </span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_blast</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/adeno-blast-hits-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>In investigating more minor viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_polyoma</span> <span class="op">&lt;-</span> <span class="fl">151341</span></span>
<span></span>
<span><span class="co"># Get set of polyomaviridae reads</span></span>
<span><span class="va">polyoma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">polyoma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each polyomaviridae species</span></span>
<span><span class="va">polyoma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_polyoma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">polyoma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_polyoma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_polyoma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">polyoma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Polyomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Polyomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_polyoma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-species-polyoma-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">polyoma_species_collate</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.35</span></span>
<span><span class="va">taxid_papilloma</span> <span class="op">&lt;-</span> <span class="fl">151340</span></span>
<span></span>
<span><span class="co"># Get set of papillomaviridae reads</span></span>
<span><span class="va">papilloma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">papilloma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each papillomaviridae species</span></span>
<span><span class="va">papilloma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_papilloma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">papilloma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_papilloma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_papilloma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">papilloma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Papillomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Papillomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_papilloma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-species-papilloma-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">papilloma_species_collate</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_pox</span> <span class="op">&lt;-</span> <span class="fl">10240</span></span>
<span></span>
<span><span class="co"># Get set of poxviridae reads</span></span>
<span><span class="va">pox_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">pox_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each poxviridae species</span></span>
<span><span class="va">pox_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_pox <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">pox_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_pox</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_pox_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">pox_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Poxviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Poxviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_pox_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-species-pox-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">pox_species_collate</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_herpes</span> <span class="op">&lt;-</span> <span class="fl">10292</span></span>
<span></span>
<span><span class="co"># Get set of herpesviridae reads</span></span>
<span><span class="va">herpes_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">herpes_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each herpesviridae species</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/hv-species-herpes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the latter case, some subset of human alphaherpesvirus 1 reads will be arising from strain RH2, which we know to be dominated by false positive hits. Digging deeper, we see that these are essentially all of the alphaherpesvirus 1 reads. Hence, these herpesvirus read counts need to be treated with significant skepticism, at least until the RH2 problem is fixed.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Count reads for each herpesviridae taxon</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">mrg_hv_named</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">instrument</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/herpes-species-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry. Note that, as <em>Simplexvirus</em> includes human alphaherpesvirus 1, that count should be treated with significant caution until the RH2 problem is fixed.</p>
<div class="cell" data-warn="false">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define reference genera</span></span>
<span><span class="va">path_genera_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_rna</span>, genome_type<span class="op">=</span><span class="st">"RNA genome"</span><span class="op">)</span>,</span>
<span>                         <span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_dna</span>, genome_type<span class="op">=</span><span class="st">"DNA genome"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count in each sample</span></span>
<span><span class="va">mrg_hv_named_all</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus_all</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named_all</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">n_path_genera</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus_all</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"location"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot out and back to add zero lines</span></span>
<span><span class="va">n_path_genera_out</span> <span class="op">&lt;-</span> <span class="va">n_path_genera</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">n_reads_viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"name"</span>, values_from<span class="op">=</span><span class="st">"n_reads_viral"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, names_to<span class="op">=</span><span class="st">"name"</span>, values_to<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Aggregate across dates</span></span>
<span><span class="va">n_path_genera_stype</span> <span class="op">&lt;-</span> <span class="va">n_path_genera_out</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">genome_type</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="st">"All samples"</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         na_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_path_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_path_genera_stype</span>,</span>
<span>                        <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">name</span>, x<span class="op">=</span><span class="va">p_reads_viral</span>, color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_loc</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">genome_type</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_path_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_brinch_files/figure-html/ra-genera-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>This is the first major DNA dataset from wastewater that I’ve analyzed using this pipeline, and at first I was nervous about how it would perform. Most of the optimization I’ve done on the HV detection pipeline has been on RNA datasets, so I was concerned that we would see major problems with false positives that we haven’t encountered with other datasets. In the event, however, things went well. There were some preventable false positives (especially from human alphaherpesvirus 1 strain RH2) but they were rare enough relative to correct assignments that the pipeline’s overall performance scores were very high. Overall I wouldn’t trust the performance of this pipeline on DNA data much less than on RNA.</p>
<p>In terms of the results themselves, it was interesting to see the remarkable preponderance of adenoviruses, and especially human mastadenovirus F, among human-infecting viruses. This isn’t a pattern we’ve seen in RNA data, and it will be interesting to see if it persists in other DNA datasets.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb50" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Brinch et al. (2020)"</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Wastewater from Copenhagen."</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-04-30</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preamble</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a><span class="co"># GGplot themes and scales</span></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>theme_rotate <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>theme_xblank <span class="ot">&lt;-</span> theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a>In this entry, I'm analyzing <span class="co">[</span><span class="ot">Brinch et al. (2020)</span><span class="co">](https://doi.org/10.1128%2FmSystems.00841-20)</span>, a large DNA-sequencing study of raw influent samples from three Copenhagen treatment plants. 12-hour composite samples were collected "at irregular intervals" between 2015 and 2018; the sample processing protocol isn't described in great detail, but samples were centrifuged and the pellet retained for extraction, a procedure that (as in Bengtsson-Palme) we expect to select against viruses. About half the samples were sequenced on an Illumina MiSeq, the rest on a NextSeq; “the change of sequencing platform was circumstantial and was not part of the experiment design.” The platform used varied in temporal blocks (blue = NextSeq):</span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a><span class="al">![](/img/2024-04-30_brinch.png)</span></span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-50"><a href="#cb50-50" aria-hidden="true" tabindex="-1"></a>The Brinch dataset contained sample information from 322 samples, composed as follows:</span>
<span id="cb50-51"><a href="#cb50-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-55"><a href="#cb50-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-56"><a href="#cb50-56" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: import-qc-data</span></span>
<span id="cb50-57"><a href="#cb50-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-58"><a href="#cb50-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Importing the data is a bit more complicated this time as the samples are split across two pipeline runs</span></span>
<span id="cb50-59"><a href="#cb50-59" aria-hidden="true" tabindex="-1"></a>data_dir_base <span class="ot">&lt;-</span> <span class="st">"../data/2024-04-30_brinch"</span></span>
<span id="cb50-60"><a href="#cb50-60" aria-hidden="true" tabindex="-1"></a>data_dirs <span class="ot">&lt;-</span> <span class="fu">paste</span>(data_dir_base, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">sep=</span><span class="st">"/"</span>)</span>
<span id="cb50-61"><a href="#cb50-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-62"><a href="#cb50-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb50-63"><a href="#cb50-63" aria-hidden="true" tabindex="-1"></a>libraries_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb50-64"><a href="#cb50-64" aria-hidden="true" tabindex="-1"></a>basic_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb50-65"><a href="#cb50-65" aria-hidden="true" tabindex="-1"></a>adapter_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb50-66"><a href="#cb50-66" aria-hidden="true" tabindex="-1"></a>quality_base_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb50-67"><a href="#cb50-67" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb50-68"><a href="#cb50-68" aria-hidden="true" tabindex="-1"></a>instrument_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"instruments.csv"</span>)</span>
<span id="cb50-69"><a href="#cb50-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-70"><a href="#cb50-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb50-71"><a href="#cb50-71" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(libraries_paths, read_csv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb50-72"><a href="#cb50-72" aria-hidden="true" tabindex="-1"></a>instruments <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(instrument_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-73"><a href="#cb50-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">instrument =</span> <span class="fu">sub</span>(<span class="st">"^Illumina "</span>, <span class="st">""</span>, instrument))</span>
<span id="cb50-74"><a href="#cb50-74" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries_raw <span class="sc">%&gt;%</span></span>
<span id="cb50-75"><a href="#cb50-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(location) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">fct_inorder</span>(location)) <span class="sc">%&gt;%</span></span>
<span id="cb50-76"><a href="#cb50-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">year =</span> <span class="fu">year</span>(date),</span>
<span id="cb50-77"><a href="#cb50-77" aria-hidden="true" tabindex="-1"></a>         <span class="at">month =</span> <span class="fu">gsub</span>(<span class="st">"-</span><span class="sc">\\</span><span class="st">d+$"</span>, <span class="st">""</span>, date)) <span class="sc">%&gt;%</span></span>
<span id="cb50-78"><a href="#cb50-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(location, date, sample) <span class="sc">%&gt;%</span></span>
<span id="cb50-79"><a href="#cb50-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">fct_inorder</span>(sample),</span>
<span id="cb50-80"><a href="#cb50-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">location_alias =</span> <span class="fu">ifelse</span>(location <span class="sc">==</span> <span class="st">"Amager"</span>, <span class="st">"RL"</span>,</span>
<span id="cb50-81"><a href="#cb50-81" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(location <span class="sc">==</span> <span class="st">"Valby"</span>, <span class="st">"RD"</span>, <span class="st">"RA"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb50-82"><a href="#cb50-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location, month) <span class="sc">%&gt;%</span></span>
<span id="cb50-83"><a href="#cb50-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_alias =</span> <span class="fu">paste</span>(location_alias, month, <span class="fu">row_number</span>(), <span class="at">sep=</span><span class="st">"_"</span>),</span>
<span id="cb50-84"><a href="#cb50-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample_alias =</span> <span class="fu">fct_inorder</span>(sample_alias)) <span class="sc">%&gt;%</span></span>
<span id="cb50-85"><a href="#cb50-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(instruments, <span class="at">by=</span><span class="st">"library"</span>)</span>
<span id="cb50-86"><a href="#cb50-86" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb50-87"><a href="#cb50-87" aria-hidden="true" tabindex="-1"></a><span class="co"># Make table</span></span>
<span id="cb50-88"><a href="#cb50-88" aria-hidden="true" tabindex="-1"></a>count_samples <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location, year) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb50-89"><a href="#cb50-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"year"</span>, <span class="at">values_from=</span><span class="st">"n"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-90"><a href="#cb50-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Location =</span> location)</span>
<span id="cb50-91"><a href="#cb50-91" aria-hidden="true" tabindex="-1"></a>count_samples</span>
<span id="cb50-92"><a href="#cb50-92" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-93"><a href="#cb50-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-96"><a href="#cb50-96" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-97"><a href="#cb50-97" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-qc-data</span></span>
<span id="cb50-98"><a href="#cb50-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-99"><a href="#cb50-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb50-100"><a href="#cb50-100" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb50-101"><a href="#cb50-101" aria-hidden="true" tabindex="-1"></a>import_basic <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb50-102"><a href="#cb50-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb50-103"><a href="#cb50-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-104"><a href="#cb50-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(location, sample_alias, sample) <span class="sc">%&gt;%</span></span>
<span id="cb50-105"><a href="#cb50-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb50-106"><a href="#cb50-106" aria-hidden="true" tabindex="-1"></a>           <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb50-107"><a href="#cb50-107" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-108"><a href="#cb50-108" aria-hidden="true" tabindex="-1"></a>import_basic_paired <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb50-109"><a href="#cb50-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">import_basic</span>(paths) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(read_pair) <span class="sc">%&gt;%</span> </span>
<span id="cb50-110"><a href="#cb50-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb50-111"><a href="#cb50-111" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-112"><a href="#cb50-112" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">import_basic</span>(basic_stats_paths)</span>
<span id="cb50-113"><a href="#cb50-113" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(adapter_stats_paths)</span>
<span id="cb50-114"><a href="#cb50-114" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_base_stats_paths)</span>
<span id="cb50-115"><a href="#cb50-115" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_seq_stats_paths)</span>
<span id="cb50-116"><a href="#cb50-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-117"><a href="#cb50-117" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb50-118"><a href="#cb50-118" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb50-119"><a href="#cb50-119" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb50-120"><a href="#cb50-120" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb50-121"><a href="#cb50-121" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb50-122"><a href="#cb50-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-123"><a href="#cb50-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Get key values for readout</span></span>
<span id="cb50-124"><a href="#cb50-124" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(instrument) <span class="sc">%&gt;%</span> </span>
<span id="cb50-125"><a href="#cb50-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_lib =</span> <span class="fu">n</span>(), <span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb50-126"><a href="#cb50-126" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb50-127"><a href="#cb50-127" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb50-128"><a href="#cb50-128" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb50-129"><a href="#cb50-129" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb50-130"><a href="#cb50-130" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-131"><a href="#cb50-131" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-132"><a href="#cb50-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-133"><a href="#cb50-133" aria-hidden="true" tabindex="-1"></a>The 169 NextSeq libraries yielded 3.6M-63M (mean 23.8M) read pairs per sample, while the 153 MiSeq libraries yielded 10-7.4M (mean 2.3M) read pairs. The total number of read pairs across all instruments was 4.4B read pairs (1.27 terabases of sequence), of which 93% of read pairs (89% of bases) came from NextSeq runs.</span>
<span id="cb50-134"><a href="#cb50-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-137"><a href="#cb50-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-138"><a href="#cb50-138" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb50-139"><a href="#cb50-139" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-140"><a href="#cb50-140" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb50-141"><a href="#cb50-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-142"><a href="#cb50-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb50-143"><a href="#cb50-143" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb50-144"><a href="#cb50-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, date, location, instrument,</span>
<span id="cb50-145"><a href="#cb50-145" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb50-146"><a href="#cb50-146" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb50-147"><a href="#cb50-147" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb50-148"><a href="#cb50-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(sample<span class="sc">:</span>instrument), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-149"><a href="#cb50-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric))</span>
<span id="cb50-150"><a href="#cb50-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-151"><a href="#cb50-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb50-152"><a href="#cb50-152" aria-hidden="true" tabindex="-1"></a>scale_fill_loc <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>)</span>
<span id="cb50-153"><a href="#cb50-153" aria-hidden="true" tabindex="-1"></a>scale_fill_ins <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name=</span><span class="st">"Instrument"</span>)</span>
<span id="cb50-154"><a href="#cb50-154" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, </span>
<span id="cb50-155"><a href="#cb50-155" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>date, <span class="at">y=</span>value, <span class="at">fill=</span>instrument, <span class="at">group=</span><span class="fu">interaction</span>(location,date))) <span class="sc">+</span></span>
<span id="cb50-156"><a href="#cb50-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb50-157"><a href="#cb50-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-158"><a href="#cb50-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb50-159"><a href="#cb50-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_ins</span>() <span class="sc">+</span> </span>
<span id="cb50-160"><a href="#cb50-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric<span class="sc">~</span>location, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space=</span><span class="st">"free_x"</span>, <span class="at">switch=</span><span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb50-161"><a href="#cb50-161" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb50-162"><a href="#cb50-162" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb50-163"><a href="#cb50-163" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"plain"</span>)</span>
<span id="cb50-164"><a href="#cb50-164" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-165"><a href="#cb50-165" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb50-166"><a href="#cb50-166" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-167"><a href="#cb50-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-168"><a href="#cb50-168" aria-hidden="true" tabindex="-1"></a>In most samples, read qualities were high at the 5' end of the read but tailed off substantially over the course of the read, especially in long MiSeq reads. A minority of MiSeq samples had poor read quality across the length of the read. Adapter levels were fairly low, but significant enough to require trimming. Duplicate levels were low (0-11%, mean 3%) in MiSeq libraries (unsurprising given their small size) but moderate (3-59%, mean 28%) in the larger NextSeq libraries.</span>
<span id="cb50-169"><a href="#cb50-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-172"><a href="#cb50-172" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-173"><a href="#cb50-173" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb50-174"><a href="#cb50-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-175"><a href="#cb50-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb50-176"><a href="#cb50-176" aria-hidden="true" tabindex="-1"></a>scale_color_loc <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>)</span>
<span id="cb50-177"><a href="#cb50-177" aria-hidden="true" tabindex="-1"></a>scale_color_ins <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name=</span><span class="st">"Instrument"</span>)</span>
<span id="cb50-178"><a href="#cb50-178" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>instrument, <span class="at">linetype=</span>read_pair, </span>
<span id="cb50-179"><a href="#cb50-179" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb50-180"><a href="#cb50-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_ins</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb50-181"><a href="#cb50-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb50-182"><a href="#cb50-182" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb50-183"><a href="#cb50-183" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb50-184"><a href="#cb50-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-185"><a href="#cb50-185" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb50-186"><a href="#cb50-186" aria-hidden="true" tabindex="-1"></a>rmax <span class="ot">&lt;-</span> <span class="dv">240</span></span>
<span id="cb50-187"><a href="#cb50-187" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb50-188"><a href="#cb50-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb50-189"><a href="#cb50-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb50-190"><a href="#cb50-190" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">1</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-191"><a href="#cb50-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb50-192"><a href="#cb50-192" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,rmax,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-193"><a href="#cb50-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb50-194"><a href="#cb50-194" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb50-195"><a href="#cb50-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-196"><a href="#cb50-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb50-197"><a href="#cb50-197" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb50-198"><a href="#cb50-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-199"><a href="#cb50-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-200"><a href="#cb50-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb50-201"><a href="#cb50-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb50-202"><a href="#cb50-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb50-203"><a href="#cb50-203" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,rmax,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb50-204"><a href="#cb50-204" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb50-205"><a href="#cb50-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-206"><a href="#cb50-206" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb50-207"><a href="#cb50-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-208"><a href="#cb50-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-209"><a href="#cb50-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb50-210"><a href="#cb50-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-211"><a href="#cb50-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb50-212"><a href="#cb50-212" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb50-213"><a href="#cb50-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-214"><a href="#cb50-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-215"><a href="#cb50-215" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb50-216"><a href="#cb50-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-217"><a href="#cb50-217" aria-hidden="true" tabindex="-1"></a>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. As expected given the observed difference in duplication levels, few reads were lost during deduplication in MiSeq libraries, while a somewhat larger (but still not huge) number were lost in NextSeq libraries. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</span>
<span id="cb50-218"><a href="#cb50-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-221"><a href="#cb50-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-222"><a href="#cb50-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb50-223"><a href="#cb50-223" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb50-224"><a href="#cb50-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, sample_alias, location, date, instrument, </span>
<span id="cb50-225"><a href="#cb50-225" aria-hidden="true" tabindex="-1"></a>         percent_duplicates, n_read_pairs, stage) <span class="sc">%&gt;%</span></span>
<span id="cb50-226"><a href="#cb50-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb50-227"><a href="#cb50-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> <span class="fu">replace_na</span>(n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs), <span class="dv">0</span>),</span>
<span id="cb50-228"><a href="#cb50-228" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb50-229"><a href="#cb50-229" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb50-230"><a href="#cb50-230" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb50-231"><a href="#cb50-231" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> <span class="fu">replace_na</span>(p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs), <span class="dv">0</span>))</span>
<span id="cb50-232"><a href="#cb50-232" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb50-233"><a href="#cb50-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="at">Instrument=</span>instrument, <span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb50-234"><a href="#cb50-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb50-235"><a href="#cb50-235" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-236"><a href="#cb50-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-237"><a href="#cb50-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb50-238"><a href="#cb50-238" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb50-239"><a href="#cb50-239" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-240"><a href="#cb50-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-243"><a href="#cb50-243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-244"><a href="#cb50-244" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb50-245"><a href="#cb50-245" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-246"><a href="#cb50-246" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb50-247"><a href="#cb50-247" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb50-248"><a href="#cb50-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-249"><a href="#cb50-249" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>instrument, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb50-250"><a href="#cb50-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_ins</span>() <span class="sc">+</span></span>
<span id="cb50-251"><a href="#cb50-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>location, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb50-252"><a href="#cb50-252" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb50-253"><a href="#cb50-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-254"><a href="#cb50-254" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb50-255"><a href="#cb50-255" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb50-256"><a href="#cb50-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb50-257"><a href="#cb50-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb50-258"><a href="#cb50-258" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb50-259"><a href="#cb50-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-260"><a href="#cb50-260" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb50-261"><a href="#cb50-261" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>instrument, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb50-262"><a href="#cb50-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb50-263"><a href="#cb50-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb50-264"><a href="#cb50-264" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-265"><a href="#cb50-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_ins</span>() <span class="sc">+</span></span>
<span id="cb50-266"><a href="#cb50-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>location, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb50-267"><a href="#cb50-267" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb50-268"><a href="#cb50-268" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb50-269"><a href="#cb50-269" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-270"><a href="#cb50-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-271"><a href="#cb50-271" aria-hidden="true" tabindex="-1"></a>Data cleaning was very successful at removing adapters and mostly successful at improving read qualities, though some libraries still dipped below my preferred quality thresholds later in the read:</span>
<span id="cb50-272"><a href="#cb50-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-275"><a href="#cb50-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-276"><a href="#cb50-276" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-277"><a href="#cb50-277" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb50-278"><a href="#cb50-278" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb50-279"><a href="#cb50-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-280"><a href="#cb50-280" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>instrument, <span class="at">linetype=</span>read_pair, </span>
<span id="cb50-281"><a href="#cb50-281" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb50-282"><a href="#cb50-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_ins</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb50-283"><a href="#cb50-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb50-284"><a href="#cb50-284" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb50-285"><a href="#cb50-285" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb50-286"><a href="#cb50-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-287"><a href="#cb50-287" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb50-288"><a href="#cb50-288" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb50-289"><a href="#cb50-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb50-290"><a href="#cb50-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb50-291"><a href="#cb50-291" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-292"><a href="#cb50-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb50-293"><a href="#cb50-293" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-294"><a href="#cb50-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb50-295"><a href="#cb50-295" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb50-296"><a href="#cb50-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-297"><a href="#cb50-297" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb50-298"><a href="#cb50-298" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb50-299"><a href="#cb50-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-300"><a href="#cb50-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-301"><a href="#cb50-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb50-302"><a href="#cb50-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb50-303"><a href="#cb50-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb50-304"><a href="#cb50-304" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-305"><a href="#cb50-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb50-306"><a href="#cb50-306" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb50-307"><a href="#cb50-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-308"><a href="#cb50-308" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb50-309"><a href="#cb50-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-310"><a href="#cb50-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-311"><a href="#cb50-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb50-312"><a href="#cb50-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-313"><a href="#cb50-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-314"><a href="#cb50-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb50-315"><a href="#cb50-315" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb50-316"><a href="#cb50-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-317"><a href="#cb50-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-318"><a href="#cb50-318" aria-hidden="true" tabindex="-1"></a>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 28% to 8% in NextSeq libraries and from 3% to 1% in MiSeq samples:</span>
<span id="cb50-319"><a href="#cb50-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-322"><a href="#cb50-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-323"><a href="#cb50-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb50-324"><a href="#cb50-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3.5</span></span>
<span id="cb50-325"><a href="#cb50-325" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb50-326"><a href="#cb50-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-327"><a href="#cb50-327" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(instrument,stage) <span class="sc">%&gt;%</span> </span>
<span id="cb50-328"><a href="#cb50-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb50-329"><a href="#cb50-329" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-330"><a href="#cb50-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-331"><a href="#cb50-331" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb50-332"><a href="#cb50-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb50-333"><a href="#cb50-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb50-334"><a href="#cb50-334" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb50-335"><a href="#cb50-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-336"><a href="#cb50-336" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb50-337"><a href="#cb50-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb50-338"><a href="#cb50-338" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb50-339"><a href="#cb50-339" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-340"><a href="#cb50-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-341"><a href="#cb50-341" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb50-342"><a href="#cb50-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-343"><a href="#cb50-343" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb50-344"><a href="#cb50-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-347"><a href="#cb50-347" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-348"><a href="#cb50-348" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb50-349"><a href="#cb50-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-350"><a href="#cb50-350" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb50-351"><a href="#cb50-351" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb50-352"><a href="#cb50-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-353"><a href="#cb50-353" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb50-354"><a href="#cb50-354" aria-hidden="true" tabindex="-1"></a>comp_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb50-355"><a href="#cb50-355" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(comp_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb50-356"><a href="#cb50-356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-357"><a href="#cb50-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels =</span> classifications),</span>
<span id="cb50-358"><a href="#cb50-358" aria-hidden="true" tabindex="-1"></a>         <span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-359"><a href="#cb50-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb50-360"><a href="#cb50-360" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-361"><a href="#cb50-361" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb50-362"><a href="#cb50-362" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb50-363"><a href="#cb50-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location, instrument, classification) <span class="sc">%&gt;%</span></span>
<span id="cb50-364"><a href="#cb50-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-365"><a href="#cb50-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb50-366"><a href="#cb50-366" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb50-367"><a href="#cb50-367" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb50-368"><a href="#cb50-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-369"><a href="#cb50-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-372"><a href="#cb50-372" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-373"><a href="#cb50-373" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb50-374"><a href="#cb50-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb50-375"><a href="#cb50-375" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb50-376"><a href="#cb50-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-377"><a href="#cb50-377" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb50-378"><a href="#cb50-378" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb50-379"><a href="#cb50-379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(instrument<span class="sc">~</span>location, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol=</span><span class="dv">3</span>,</span>
<span id="cb50-380"><a href="#cb50-380" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line=</span><span class="cn">FALSE</span>, <span class="at">width=</span><span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb50-381"><a href="#cb50-381" aria-hidden="true" tabindex="-1"></a>  theme_xblank</span>
<span id="cb50-382"><a href="#cb50-382" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb50-383"><a href="#cb50-383" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb50-384"><a href="#cb50-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-385"><a href="#cb50-385" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb50-386"><a href="#cb50-386" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data =</span> comp, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-387"><a href="#cb50-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb50-388"><a href="#cb50-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb50-389"><a href="#cb50-389" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb50-390"><a href="#cb50-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-391"><a href="#cb50-391" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb50-392"><a href="#cb50-392" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb50-393"><a href="#cb50-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb50-394"><a href="#cb50-394" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb50-395"><a href="#cb50-395" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data=</span>comp_minor, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-396"><a href="#cb50-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb50-397"><a href="#cb50-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb50-398"><a href="#cb50-398" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb50-399"><a href="#cb50-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-400"><a href="#cb50-400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-401"><a href="#cb50-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-404"><a href="#cb50-404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-405"><a href="#cb50-405" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-summary</span></span>
<span id="cb50-406"><a href="#cb50-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-407"><a href="#cb50-407" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb50-408"><a href="#cb50-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb50-409"><a href="#cb50-409" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb50-410"><a href="#cb50-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-411"><a href="#cb50-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-412"><a href="#cb50-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-413"><a href="#cb50-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb50-414"><a href="#cb50-414" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-415"><a href="#cb50-415" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> p_reads_summ_group <span class="sc">%&gt;%</span></span>
<span id="cb50-416"><a href="#cb50-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb50-417"><a href="#cb50-417" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb50-418"><a href="#cb50-418" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb50-419"><a href="#cb50-419" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb50-420"><a href="#cb50-420" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb50-421"><a href="#cb50-421" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb50-422"><a href="#cb50-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(instrument, classification, <span class="at">read_fraction=</span>display) <span class="sc">%&gt;%</span></span>
<span id="cb50-423"><a href="#cb50-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(instrument, classification)</span>
<span id="cb50-424"><a href="#cb50-424" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb50-425"><a href="#cb50-425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-426"><a href="#cb50-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-427"><a href="#cb50-427" aria-hidden="true" tabindex="-1"></a>In most samples, the majority of reads were either filtered, duplicates, or unassigned. Among assigned reads, the vast majority in most samples were bacterial, which is unsurprising given the sample processing protocols used. A few samples showed elevated levels of human reads, which might be real or due to experimenter contamination. Viral fraction averaged 0.41% in MiSeq reads and 0.34% in NextSeq reads.</span>
<span id="cb50-428"><a href="#cb50-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-429"><a href="#cb50-429" aria-hidden="true" tabindex="-1"></a>As is common for DNA data, viral reads were overwhelmingly dominated by *Caudoviricetes* phages:</span>
<span id="cb50-430"><a href="#cb50-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-433"><a href="#cb50-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-434"><a href="#cb50-434" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb50-435"><a href="#cb50-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-436"><a href="#cb50-436" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb50-437"><a href="#cb50-437" aria-hidden="true" tabindex="-1"></a>reports_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb50-438"><a href="#cb50-438" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">lapply</span>(reports_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb50-439"><a href="#cb50-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-440"><a href="#cb50-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb50-441"><a href="#cb50-441" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb50-442"><a href="#cb50-442" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-443"><a href="#cb50-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-444"><a href="#cb50-444" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb50-445"><a href="#cb50-445" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-446"><a href="#cb50-446" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb50-447"><a href="#cb50-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb50-448"><a href="#cb50-448" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb50-449"><a href="#cb50-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-450"><a href="#cb50-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-451"><a href="#cb50-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>())</span>
<span id="cb50-452"><a href="#cb50-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-453"><a href="#cb50-453" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb50-454"><a href="#cb50-454" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb50-455"><a href="#cb50-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-456"><a href="#cb50-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-457"><a href="#cb50-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-460"><a href="#cb50-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-461"><a href="#cb50-461" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-class-composition</span></span>
<span id="cb50-462"><a href="#cb50-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-463"><a href="#cb50-463" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb50-464"><a href="#cb50-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-465"><a href="#cb50-465" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral classes</span></span>
<span id="cb50-466"><a href="#cb50-466" aria-hidden="true" tabindex="-1"></a>viral_classes_major_tab <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb50-467"><a href="#cb50-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-468"><a href="#cb50-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-469"><a href="#cb50-469" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb50-470"><a href="#cb50-470" aria-hidden="true" tabindex="-1"></a>viral_classes_major_list <span class="ot">&lt;-</span> viral_classes_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb50-471"><a href="#cb50-471" aria-hidden="true" tabindex="-1"></a>viral_classes_major <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb50-472"><a href="#cb50-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_classes_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb50-473"><a href="#cb50-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, instrument, location, p_reads_viral)</span>
<span id="cb50-474"><a href="#cb50-474" aria-hidden="true" tabindex="-1"></a>viral_classes_minor <span class="ot">&lt;-</span> viral_classes_major <span class="sc">%&gt;%</span> </span>
<span id="cb50-475"><a href="#cb50-475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, instrument, location) <span class="sc">%&gt;%</span></span>
<span id="cb50-476"><a href="#cb50-476" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-477"><a href="#cb50-477" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb50-478"><a href="#cb50-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, instrument, location, p_reads_viral)</span>
<span id="cb50-479"><a href="#cb50-479" aria-hidden="true" tabindex="-1"></a>viral_classes_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_classes_major, viral_classes_minor) <span class="sc">%&gt;%</span></span>
<span id="cb50-480"><a href="#cb50-480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-481"><a href="#cb50-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_classes_major_list, <span class="st">"Other"</span>)),</span>
<span id="cb50-482"><a href="#cb50-482" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> <span class="fu">pmax</span>(p_reads_viral, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-483"><a href="#cb50-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name)</span>
<span id="cb50-484"><a href="#cb50-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-485"><a href="#cb50-485" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb50-486"><a href="#cb50-486" aria-hidden="true" tabindex="-1"></a>g_classes <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-487"><a href="#cb50-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>viral_classes_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-488"><a href="#cb50-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-489"><a href="#cb50-489" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-490"><a href="#cb50-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb50-491"><a href="#cb50-491" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-492"><a href="#cb50-492" aria-hidden="true" tabindex="-1"></a>g_classes</span>
<span id="cb50-493"><a href="#cb50-493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-494"><a href="#cb50-494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-495"><a href="#cb50-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-496"><a href="#cb50-496" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb50-497"><a href="#cb50-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-498"><a href="#cb50-498" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples. A total of 17,448 reads were identified as putatively human-viral:</span>
<span id="cb50-499"><a href="#cb50-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-502"><a href="#cb50-502" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-503"><a href="#cb50-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-read-counts</span></span>
<span id="cb50-504"><a href="#cb50-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-505"><a href="#cb50-505" aria-hidden="true" tabindex="-1"></a><span class="co"># Import HV read data</span></span>
<span id="cb50-506"><a href="#cb50-506" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb50-507"><a href="#cb50-507" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hv_reads_filtered_paths, read_tsv,</span>
<span id="cb50-508"><a href="#cb50-508" aria-hidden="true" tabindex="-1"></a>                            <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-509"><a href="#cb50-509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-510"><a href="#cb50-510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb50-511"><a href="#cb50-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-512"><a href="#cb50-512" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb50-513"><a href="#cb50-513" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb50-514"><a href="#cb50-514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, seq_id) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb50-515"><a href="#cb50-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb50-516"><a href="#cb50-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-517"><a href="#cb50-517" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-518"><a href="#cb50-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb50-519"><a href="#cb50-519" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb50-520"><a href="#cb50-520" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_summ <span class="ot">&lt;-</span> n_hv_filtered <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb50-521"><a href="#cb50-521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total), </span>
<span id="cb50-522"><a href="#cb50-522" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-523"><a href="#cb50-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb50-524"><a href="#cb50-524" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-525"><a href="#cb50-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-528"><a href="#cb50-528" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-529"><a href="#cb50-529" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-scores</span></span>
<span id="cb50-530"><a href="#cb50-530" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-531"><a href="#cb50-531" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb50-532"><a href="#cb50-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-533"><a href="#cb50-533" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse multi-entry sequences</span></span>
<span id="cb50-534"><a href="#cb50-534" aria-hidden="true" tabindex="-1"></a>rmax <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(max, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-535"><a href="#cb50-535" aria-hidden="true" tabindex="-1"></a>collapse <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">ifelse</span>(<span class="fu">all</span>(x <span class="sc">==</span> x[<span class="dv">1</span>]), x[<span class="dv">1</span>], <span class="fu">paste</span>(x, <span class="at">collapse=</span><span class="st">"/"</span>))</span>
<span id="cb50-536"><a href="#cb50-536" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> </span>
<span id="cb50-537"><a href="#cb50-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-538"><a href="#cb50-538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb50-539"><a href="#cb50-539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb50-540"><a href="#cb50-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sample =</span> <span class="fu">collapse</span>(sample),</span>
<span id="cb50-541"><a href="#cb50-541" aria-hidden="true" tabindex="-1"></a>            <span class="at">genome_id =</span> <span class="fu">collapse</span>(genome_id),</span>
<span id="cb50-542"><a href="#cb50-542" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid_best =</span> taxid[<span class="dv">1</span>],</span>
<span id="cb50-543"><a href="#cb50-543" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(taxid)),</span>
<span id="cb50-544"><a href="#cb50-544" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_fwd =</span> <span class="fu">rmax</span>(best_alignment_score_fwd),</span>
<span id="cb50-545"><a href="#cb50-545" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_rev =</span> <span class="fu">rmax</span>(best_alignment_score_rev),</span>
<span id="cb50-546"><a href="#cb50-546" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_fwd =</span> <span class="fu">rmax</span>(query_len_fwd),</span>
<span id="cb50-547"><a href="#cb50-547" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_rev =</span> <span class="fu">rmax</span>(query_len_rev),</span>
<span id="cb50-548"><a href="#cb50-548" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_fwd =</span> query_seq_fwd[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_fwd)][<span class="dv">1</span>],</span>
<span id="cb50-549"><a href="#cb50-549" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_rev =</span> query_seq_rev[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_rev)][<span class="dv">1</span>],</span>
<span id="cb50-550"><a href="#cb50-550" aria-hidden="true" tabindex="-1"></a>            <span class="at">classified =</span> <span class="fu">rmax</span>(classified),</span>
<span id="cb50-551"><a href="#cb50-551" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_name =</span> <span class="fu">collapse</span>(assigned_name),</span>
<span id="cb50-552"><a href="#cb50-552" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid_best =</span> assigned_taxid[<span class="dv">1</span>],</span>
<span id="cb50-553"><a href="#cb50-553" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(assigned_taxid)),</span>
<span id="cb50-554"><a href="#cb50-554" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_hv =</span> <span class="fu">rmax</span>(assigned_hv),</span>
<span id="cb50-555"><a href="#cb50-555" aria-hidden="true" tabindex="-1"></a>            <span class="at">hit_hv =</span> <span class="fu">rmax</span>(hit_hv),</span>
<span id="cb50-556"><a href="#cb50-556" aria-hidden="true" tabindex="-1"></a>            <span class="at">encoded_hits =</span> <span class="fu">collapse</span>(encoded_hits),</span>
<span id="cb50-557"><a href="#cb50-557" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_fwd =</span> <span class="fu">rmax</span>(adj_score_fwd),</span>
<span id="cb50-558"><a href="#cb50-558" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_rev =</span> <span class="fu">rmax</span>(adj_score_rev)</span>
<span id="cb50-559"><a href="#cb50-559" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb50-560"><a href="#cb50-560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-561"><a href="#cb50-561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb50-562"><a href="#cb50-562" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb50-563"><a href="#cb50-563" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb50-564"><a href="#cb50-564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb50-565"><a href="#cb50-565" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb50-566"><a href="#cb50-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-567"><a href="#cb50-567" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot results</span></span>
<span id="cb50-568"><a href="#cb50-568" aria-hidden="true" tabindex="-1"></a>geom_vhist <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_histogram, <span class="at">binwidth=</span><span class="dv">5</span>, <span class="at">boundary=</span><span class="dv">0</span>)</span>
<span id="cb50-569"><a href="#cb50-569" aria-hidden="true" tabindex="-1"></a>g_vhist_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span></span>
<span id="cb50-570"><a href="#cb50-570" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb50-571"><a href="#cb50-571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>)), <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb50-572"><a href="#cb50-572" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> </span>
<span id="cb50-573"><a href="#cb50-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> </span>
<span id="cb50-574"><a href="#cb50-574" aria-hidden="true" tabindex="-1"></a>  theme_base </span>
<span id="cb50-575"><a href="#cb50-575" aria-hidden="true" tabindex="-1"></a>g_vhist_0 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg)</span>
<span id="cb50-576"><a href="#cb50-576" aria-hidden="true" tabindex="-1"></a>g_vhist_0</span>
<span id="cb50-577"><a href="#cb50-577" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-578"><a href="#cb50-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-579"><a href="#cb50-579" aria-hidden="true" tabindex="-1"></a>BLASTing these reads against nt:</span>
<span id="cb50-580"><a href="#cb50-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-583"><a href="#cb50-583" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-584"><a href="#cb50-584" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-blast-data</span></span>
<span id="cb50-585"><a href="#cb50-585" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-586"><a href="#cb50-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-587"><a href="#cb50-587" aria-hidden="true" tabindex="-1"></a><span class="co"># Import paired BLAST results</span></span>
<span id="cb50-588"><a href="#cb50-588" aria-hidden="true" tabindex="-1"></a>blast_paired_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"hv_hits_blast_paired.tsv.gz"</span>)</span>
<span id="cb50-589"><a href="#cb50-589" aria-hidden="true" tabindex="-1"></a>blast_paired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(blast_paired_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb50-590"><a href="#cb50-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-591"><a href="#cb50-591" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb50-592"><a href="#cb50-592" aria-hidden="true" tabindex="-1"></a>blast_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_paired, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-593"><a href="#cb50-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb50-594"><a href="#cb50-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-595"><a href="#cb50-595" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb50-596"><a href="#cb50-596" aria-hidden="true" tabindex="-1"></a>match_taxid <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_1, taxid_2){</span>
<span id="cb50-597"><a href="#cb50-597" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"/"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb50-598"><a href="#cb50-598" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"/"</span>), taxid_2)</span>
<span id="cb50-599"><a href="#cb50-599" aria-hidden="true" tabindex="-1"></a>  p3 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb50-600"><a href="#cb50-600" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">setNames</span>(p1<span class="sc">|</span>p2<span class="sc">|</span>p3, <span class="cn">NULL</span>)</span>
<span id="cb50-601"><a href="#cb50-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb50-602"><a href="#cb50-602" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-603"><a href="#cb50-603" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_id, taxid, assigned_taxid, adj_score_max)</span>
<span id="cb50-604"><a href="#cb50-604" aria-hidden="true" tabindex="-1"></a>blast_assign <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(blast_viral, mrg_assign, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-605"><a href="#cb50-605" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> <span class="fu">match_taxid</span>(staxid, taxid),</span>
<span id="cb50-606"><a href="#cb50-606" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> <span class="fu">match_taxid</span>(staxid, assigned_taxid),</span>
<span id="cb50-607"><a href="#cb50-607" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb50-608"><a href="#cb50-608" aria-hidden="true" tabindex="-1"></a>blast_out <span class="ot">&lt;-</span> blast_assign <span class="sc">%&gt;%</span></span>
<span id="cb50-609"><a href="#cb50-609" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb50-610"><a href="#cb50-610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb50-611"><a href="#cb50-611" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb50-612"><a href="#cb50-612" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb50-613"><a href="#cb50-613" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-614"><a href="#cb50-614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-615"><a href="#cb50-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-618"><a href="#cb50-618" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-619"><a href="#cb50-619" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-results</span></span>
<span id="cb50-620"><a href="#cb50-620" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb50-621"><a href="#cb50-621" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-622"><a href="#cb50-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-623"><a href="#cb50-623" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb50-624"><a href="#cb50-624" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg, blast_out, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-625"><a href="#cb50-625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb50-626"><a href="#cb50-626" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb50-627"><a href="#cb50-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-628"><a href="#cb50-628" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot RNA</span></span>
<span id="cb50-629"><a href="#cb50-629" aria-hidden="true" tabindex="-1"></a>g_vhist_1 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg_blast, <span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">fill=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb50-630"><a href="#cb50-630" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>)</span>
<span id="cb50-631"><a href="#cb50-631" aria-hidden="true" tabindex="-1"></a>g_vhist_1</span>
<span id="cb50-632"><a href="#cb50-632" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-633"><a href="#cb50-633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-634"><a href="#cb50-634" aria-hidden="true" tabindex="-1"></a>Applying my standard disjunctive score threshold of 20 achieved precision, sensitivity and F1 scores all <span class="sc">\&gt;</span>99%:</span>
<span id="cb50-635"><a href="#cb50-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-638"><a href="#cb50-638" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-639"><a href="#cb50-639" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-f1</span></span>
<span id="cb50-640"><a href="#cb50-640" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold){</span>
<span id="cb50-641"><a href="#cb50-641" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> </span>
<span id="cb50-642"><a href="#cb50-642" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">retain_score =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb50-643"><a href="#cb50-643" aria-hidden="true" tabindex="-1"></a>           <span class="at">retain =</span> assigned_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb50-644"><a href="#cb50-644" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb50-645"><a href="#cb50-645" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb50-646"><a href="#cb50-646" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb50-647"><a href="#cb50-647" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb50-648"><a href="#cb50-648" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb50-649"><a href="#cb50-649" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb50-650"><a href="#cb50-650" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb50-651"><a href="#cb50-651" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb50-652"><a href="#cb50-652" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb50-653"><a href="#cb50-653" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb50-654"><a href="#cb50-654" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb50-655"><a href="#cb50-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb50-656"><a href="#cb50-656" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-657"><a href="#cb50-657" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb50-658"><a href="#cb50-658" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab)</span>
<span id="cb50-659"><a href="#cb50-659" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb50-660"><a href="#cb50-660" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>threshold, <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>)</span>
<span id="cb50-661"><a href="#cb50-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats)</span>
<span id="cb50-662"><a href="#cb50-662" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-663"><a href="#cb50-663" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast)</span>
<span id="cb50-664"><a href="#cb50-664" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb50-665"><a href="#cb50-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb50-666"><a href="#cb50-666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb50-667"><a href="#cb50-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-668"><a href="#cb50-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adjusted Score Threshold (Disjunctive)"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-669"><a href="#cb50-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb50-670"><a href="#cb50-670" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb50-671"><a href="#cb50-671" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb50-672"><a href="#cb50-672" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-673"><a href="#cb50-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">Threshold=</span>threshold, <span class="at">Metric=</span>metric, <span class="at">Value=</span>value)</span>
<span id="cb50-674"><a href="#cb50-674" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-675"><a href="#cb50-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-676"><a href="#cb50-676" aria-hidden="true" tabindex="-1"></a>Looking into the composition of different read groups, a plurality of high-scoring false positives map to human alphaherpesvirus 1 strain RH2 according to Bowtie2. BLASTN maps these sequences to a variety of bacterial taxa, especially *E. coli* and various *Klebsiella* species. I'm already screening against one strain of E. coli, so if I wanted to tackle this I'd need to add more strains or another species, probably *Klebsiella pneumoniae*. However, in this case the scores are good enough that I don't think it's worth it.</span>
<span id="cb50-677"><a href="#cb50-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-680"><a href="#cb50-680" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-681"><a href="#cb50-681" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fp</span></span>
<span id="cb50-682"><a href="#cb50-682" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-683"><a href="#cb50-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-684"><a href="#cb50-684" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.04</span></span>
<span id="cb50-685"><a href="#cb50-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-686"><a href="#cb50-686" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing viral taxa</span></span>
<span id="cb50-687"><a href="#cb50-687" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">211787</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span id="cb50-688"><a href="#cb50-688" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">509154</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span id="cb50-689"><a href="#cb50-689" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">493803</span>] <span class="ot">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span id="cb50-690"><a href="#cb50-690" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">427343</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span id="cb50-691"><a href="#cb50-691" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194958</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span id="cb50-692"><a href="#cb50-692" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">340907</span>] <span class="ot">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span id="cb50-693"><a href="#cb50-693" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194959</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span id="cb50-694"><a href="#cb50-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-695"><a href="#cb50-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-696"><a href="#cb50-696" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb50-697"><a href="#cb50-697" aria-hidden="true" tabindex="-1"></a>fp <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> </span>
<span id="cb50-698"><a href="#cb50-698" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb50-699"><a href="#cb50-699" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-700"><a href="#cb50-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid =</span> taxid_best) <span class="sc">%&gt;%</span></span>
<span id="cb50-701"><a href="#cb50-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-702"><a href="#cb50-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb50-703"><a href="#cb50-703" aria-hidden="true" tabindex="-1"></a>fp_major_tab <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb50-704"><a href="#cb50-704" aria-hidden="true" tabindex="-1"></a>fp_major_list <span class="ot">&lt;-</span> fp_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb50-705"><a href="#cb50-705" aria-hidden="true" tabindex="-1"></a>fp_major <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb50-706"><a href="#cb50-706" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-707"><a href="#cb50-707" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb50-708"><a href="#cb50-708" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb50-709"><a href="#cb50-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_major_list),</span>
<span id="cb50-710"><a href="#cb50-710" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb50-711"><a href="#cb50-711" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb50-712"><a href="#cb50-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-713"><a href="#cb50-713" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-714"><a href="#cb50-714" aria-hidden="true" tabindex="-1"></a>g_fp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb50-715"><a href="#cb50-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb50-716"><a href="#cb50-716" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb50-717"><a href="#cb50-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-718"><a href="#cb50-718" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-719"><a href="#cb50-719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_viral, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb50-720"><a href="#cb50-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb50-721"><a href="#cb50-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-722"><a href="#cb50-722" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb50-723"><a href="#cb50-723" aria-hidden="true" tabindex="-1"></a>g_fp</span>
<span id="cb50-724"><a href="#cb50-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-725"><a href="#cb50-725" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-726"><a href="#cb50-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-729"><a href="#cb50-729" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-730"><a href="#cb50-730" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ptt-blast-hits</span></span>
<span id="cb50-731"><a href="#cb50-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-732"><a href="#cb50-732" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb50-733"><a href="#cb50-733" aria-hidden="true" tabindex="-1"></a>ref_taxid_rh2 <span class="ot">&lt;-</span> <span class="dv">946522</span></span>
<span id="cb50-734"><a href="#cb50-734" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb50-735"><a href="#cb50-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-736"><a href="#cb50-736" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb50-737"><a href="#cb50-737" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb50-738"><a href="#cb50-738" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-739"><a href="#cb50-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-740"><a href="#cb50-740" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb50-741"><a href="#cb50-741" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb50-742"><a href="#cb50-742" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb50-743"><a href="#cb50-743" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb50-744"><a href="#cb50-744" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span id="cb50-745"><a href="#cb50-745" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span id="cb50-746"><a href="#cb50-746" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span id="cb50-747"><a href="#cb50-747" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">187126</span>, <span class="st">"Nesoenas mayeri"</span>,</span>
<span id="cb50-748"><a href="#cb50-748" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">244366</span>, <span class="st">"Klebsiella variicola"</span>,</span>
<span id="cb50-749"><a href="#cb50-749" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">208224</span>, <span class="st">"Enterobacter kobei"</span></span>
<span id="cb50-750"><a href="#cb50-750" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb50-751"><a href="#cb50-751" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> tax_names_new <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> staxid <span class="sc">%in%</span> tax_names<span class="sc">$</span>staxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-752"><a href="#cb50-752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(tax_names) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(staxid)</span>
<span id="cb50-753"><a href="#cb50-753" aria-hidden="true" tabindex="-1"></a>ref_name_rh2 <span class="ot">&lt;-</span> tax_names <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> ref_taxid_rh2) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb50-754"><a href="#cb50-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-755"><a href="#cb50-755" aria-hidden="true" tabindex="-1"></a><span class="co"># Get major matches</span></span>
<span id="cb50-756"><a href="#cb50-756" aria-hidden="true" tabindex="-1"></a>mrg_staxid <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid_best <span class="sc">==</span> ref_taxid_rh2) <span class="sc">%&gt;%</span></span>
<span id="cb50-757"><a href="#cb50-757" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(highscore, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>())</span>
<span id="cb50-758"><a href="#cb50-758" aria-hidden="true" tabindex="-1"></a>fp_staxid <span class="ot">&lt;-</span> mrg_staxid <span class="sc">%&gt;%</span></span>
<span id="cb50-759"><a href="#cb50-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-760"><a href="#cb50-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb50-761"><a href="#cb50-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb50-762"><a href="#cb50-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid_best=</span>staxid), <span class="at">by=</span><span class="st">"taxid_best"</span>)</span>
<span id="cb50-763"><a href="#cb50-763" aria-hidden="true" tabindex="-1"></a>fp_staxid_count <span class="ot">&lt;-</span> fp_staxid <span class="sc">%&gt;%</span></span>
<span id="cb50-764"><a href="#cb50-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, </span>
<span id="cb50-765"><a href="#cb50-765" aria-hidden="true" tabindex="-1"></a>           taxid_best, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb50-766"><a href="#cb50-766" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span></span>
<span id="cb50-767"><a href="#cb50-767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best, name) <span class="sc">%&gt;%</span></span>
<span id="cb50-768"><a href="#cb50-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb50-769"><a href="#cb50-769" aria-hidden="true" tabindex="-1"></a>fp_staxid_count_major <span class="ot">&lt;-</span> fp_staxid_count <span class="sc">%&gt;%</span></span>
<span id="cb50-770"><a href="#cb50-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb50-771"><a href="#cb50-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb50-772"><a href="#cb50-772" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, </span>
<span id="cb50-773"><a href="#cb50-773" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb50-774"><a href="#cb50-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-775"><a href="#cb50-775" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-776"><a href="#cb50-776" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_staxid_count_major, <span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span> </span>
<span id="cb50-777"><a href="#cb50-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb50-778"><a href="#cb50-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(status_display<span class="sc">~</span>score_display, <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb50-779"><a href="#cb50-779" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb50-780"><a href="#cb50-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-781"><a href="#cb50-781" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-782"><a href="#cb50-782" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_rh2, <span class="st">" (taxid "</span>, ref_taxid_rh2, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb50-783"><a href="#cb50-783" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb50-784"><a href="#cb50-784" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb50-785"><a href="#cb50-785" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-786"><a href="#cb50-786" aria-hidden="true" tabindex="-1"></a>g</span>
<span id="cb50-787"><a href="#cb50-787" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-788"><a href="#cb50-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-789"><a href="#cb50-789" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: overall relative abundance</span></span>
<span id="cb50-790"><a href="#cb50-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-793"><a href="#cb50-793" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-794"><a href="#cb50-794" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb50-795"><a href="#cb50-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-796"><a href="#cb50-796" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb50-797"><a href="#cb50-797" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb50-798"><a href="#cb50-798" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, instrument, date, year, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb50-799"><a href="#cb50-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-800"><a href="#cb50-800" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb50-801"><a href="#cb50-801" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb50-802"><a href="#cb50-802" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb50-803"><a href="#cb50-803" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb50-804"><a href="#cb50-804" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb50-805"><a href="#cb50-805" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-806"><a href="#cb50-806" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>))</span>
<span id="cb50-807"><a href="#cb50-807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-808"><a href="#cb50-808" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb50-809"><a href="#cb50-809" aria-hidden="true" tabindex="-1"></a>read_counts_grp <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location, year) <span class="sc">%&gt;%</span></span>
<span id="cb50-810"><a href="#cb50-810" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb50-811"><a href="#cb50-811" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-812"><a href="#cb50-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span> <span class="st">"All samples"</span>, <span class="at">year =</span> <span class="fu">as.character</span>(year))</span>
<span id="cb50-813"><a href="#cb50-813" aria-hidden="true" tabindex="-1"></a>read_counts_yr <span class="ot">&lt;-</span> read_counts_grp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, year) <span class="sc">%&gt;%</span></span>
<span id="cb50-814"><a href="#cb50-814" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb50-815"><a href="#cb50-815" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-816"><a href="#cb50-816" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb50-817"><a href="#cb50-817" aria-hidden="true" tabindex="-1"></a>read_counts_loc <span class="ot">&lt;-</span> read_counts_grp <span class="sc">%&gt;%</span></span>
<span id="cb50-818"><a href="#cb50-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location) <span class="sc">%&gt;%</span></span>
<span id="cb50-819"><a href="#cb50-819" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb50-820"><a href="#cb50-820" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-821"><a href="#cb50-821" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">year =</span> <span class="st">"All years"</span>)</span>
<span id="cb50-822"><a href="#cb50-822" aria-hidden="true" tabindex="-1"></a>read_counts_tot <span class="ot">&lt;-</span> read_counts_loc <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, year) <span class="sc">%&gt;%</span></span>
<span id="cb50-823"><a href="#cb50-823" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb50-824"><a href="#cb50-824" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-825"><a href="#cb50-825" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb50-826"><a href="#cb50-826" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(read_counts_grp, read_counts_yr,</span>
<span id="cb50-827"><a href="#cb50-827" aria-hidden="true" tabindex="-1"></a>                             read_counts_loc, read_counts_tot) <span class="sc">%&gt;%</span></span>
<span id="cb50-828"><a href="#cb50-828" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb50-829"><a href="#cb50-829" aria-hidden="true" tabindex="-1"></a>         <span class="at">location =</span> <span class="fu">factor</span>(location, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">levels</span>(libraries<span class="sc">$</span>location), <span class="st">"All locations"</span>)),</span>
<span id="cb50-830"><a href="#cb50-830" aria-hidden="true" tabindex="-1"></a>         <span class="at">year =</span> <span class="fu">factor</span>(year, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">as.character</span>(<span class="dv">2015</span><span class="sc">:</span><span class="dv">2018</span>), <span class="st">"All years"</span>)))</span>
<span id="cb50-831"><a href="#cb50-831" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-832"><a href="#cb50-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-833"><a href="#cb50-833" aria-hidden="true" tabindex="-1"></a>Applying a disjunctive cutoff at S=20 identifies 16,967 read pairs as human-viral. This gives an overall relative HV abundance of $3.88 \times 10^{-6}$. Measured RAs are somewhat lower on average in 2015, and higher than average in 2016, but overall year-on-year variation isn't huge on a log scale:</span>
<span id="cb50-834"><a href="#cb50-834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-837"><a href="#cb50-837" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-838"><a href="#cb50-838" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-ra</span></span>
<span id="cb50-839"><a href="#cb50-839" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb50-840"><a href="#cb50-840" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb50-841"><a href="#cb50-841" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>year, <span class="at">color=</span>location)) <span class="sc">+</span></span>
<span id="cb50-842"><a href="#cb50-842" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_hv)) <span class="sc">+</span></span>
<span id="cb50-843"><a href="#cb50-843" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="fl">1e-6</span>, <span class="fl">1e-5</span>)) <span class="sc">+</span></span>
<span id="cb50-844"><a href="#cb50-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_loc</span>() <span class="sc">+</span> theme_kit</span>
<span id="cb50-845"><a href="#cb50-845" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb50-846"><a href="#cb50-846" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-847"><a href="#cb50-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-848"><a href="#cb50-848" aria-hidden="true" tabindex="-1"></a>This overall RA is somewhat lower than most RNA datasets I've previously analyzed, and on the low end for DNA datasets as well, probably due to the antiviral selection of the sample processing protocol used:</span>
<span id="cb50-849"><a href="#cb50-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-852"><a href="#cb50-852" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-853"><a href="#cb50-853" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-hv-past</span></span>
<span id="cb50-854"><a href="#cb50-854" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-855"><a href="#cb50-855" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate past RA values</span></span>
<span id="cb50-856"><a href="#cb50-856" aria-hidden="true" tabindex="-1"></a>ra_past <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb50-857"><a href="#cb50-857" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-858"><a href="#cb50-858" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-859"><a href="#cb50-859" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-860"><a href="#cb50-860" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-861"><a href="#cb50-861" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-862"><a href="#cb50-862" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb50-863"><a href="#cb50-863" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-864"><a href="#cb50-864" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb50-865"><a href="#cb50-865" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-866"><a href="#cb50-866" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-867"><a href="#cb50-867" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-868"><a href="#cb50-868" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb50-869"><a href="#cb50-869" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span id="cb50-870"><a href="#cb50-870" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-871"><a href="#cb50-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-872"><a href="#cb50-872" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate new RA values</span></span>
<span id="cb50-873"><a href="#cb50-873" aria-hidden="true" tabindex="-1"></a>ra_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb50-874"><a href="#cb50-874" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Brinch"</span>, <span class="fl">3.88e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>)</span>
<span id="cb50-875"><a href="#cb50-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-876"><a href="#cb50-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-877"><a href="#cb50-877" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-878"><a href="#cb50-878" aria-hidden="true" tabindex="-1"></a>scale_color_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb50-879"><a href="#cb50-879" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb50-880"><a href="#cb50-880" aria-hidden="true" tabindex="-1"></a>ra_comp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ra_past, ra_new) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">fct_inorder</span>(dataset))</span>
<span id="cb50-881"><a href="#cb50-881" aria-hidden="true" tabindex="-1"></a>g_ra_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ra_comp, <span class="fu">aes</span>(<span class="at">y=</span>dataset, <span class="at">x=</span>ra, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb50-882"><a href="#cb50-882" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb50-883"><a href="#cb50-883" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb50-884"><a href="#cb50-884" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb50-885"><a href="#cb50-885" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb50-886"><a href="#cb50-886" aria-hidden="true" tabindex="-1"></a>g_ra_comp</span>
<span id="cb50-887"><a href="#cb50-887" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-888"><a href="#cb50-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-889"><a href="#cb50-889" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: taxonomy and composition</span></span>
<span id="cb50-890"><a href="#cb50-890" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-891"><a href="#cb50-891" aria-hidden="true" tabindex="-1"></a>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 15 samples, 4 from influent and 11 from primary sludge, met this criterion.</span>
<span id="cb50-892"><a href="#cb50-892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-893"><a href="#cb50-893" aria-hidden="true" tabindex="-1"></a>At the family level, most samples across all locations were overwhelmingly dominated by *Adenoviridae*, followed distantly by *Polyomaviridae*, *Papillomaviridae* and *Poxviridae*:</span>
<span id="cb50-894"><a href="#cb50-894" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-897"><a href="#cb50-897" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-898"><a href="#cb50-898" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb50-899"><a href="#cb50-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-900"><a href="#cb50-900" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb50-901"><a href="#cb50-901" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb50-902"><a href="#cb50-902" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb50-903"><a href="#cb50-903" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb50-904"><a href="#cb50-904" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">585893</span>] <span class="ot">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span id="cb50-905"><a href="#cb50-905" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333922</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span id="cb50-906"><a href="#cb50-906" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334207</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span id="cb50-907"><a href="#cb50-907" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">369960</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span id="cb50-908"><a href="#cb50-908" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333924</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span id="cb50-909"><a href="#cb50-909" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687329</span>] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb50-910"><a href="#cb50-910" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">325455</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span id="cb50-911"><a href="#cb50-911" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333750</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span id="cb50-912"><a href="#cb50-912" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb50-913"><a href="#cb50-913" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334202</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span id="cb50-914"><a href="#cb50-914" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">197911</span>] <span class="ot">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span id="cb50-915"><a href="#cb50-915" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">186938</span>] <span class="ot">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span id="cb50-916"><a href="#cb50-916" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333926</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span id="cb50-917"><a href="#cb50-917" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337051</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span id="cb50-918"><a href="#cb50-918" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337043</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span id="cb50-919"><a href="#cb50-919" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb50-920"><a href="#cb50-920" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334204</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span id="cb50-921"><a href="#cb50-921" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334208</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span id="cb50-922"><a href="#cb50-922" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333928</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span id="cb50-923"><a href="#cb50-923" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337039</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span id="cb50-924"><a href="#cb50-924" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333929</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span id="cb50-925"><a href="#cb50-925" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337042</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span id="cb50-926"><a href="#cb50-926" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334203</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span id="cb50-927"><a href="#cb50-927" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333757</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span id="cb50-928"><a href="#cb50-928" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337050</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span id="cb50-929"><a href="#cb50-929" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333767</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span id="cb50-930"><a href="#cb50-930" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333754</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span id="cb50-931"><a href="#cb50-931" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687363</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span id="cb50-932"><a href="#cb50-932" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687342</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span id="cb50-933"><a href="#cb50-933" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687359</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span id="cb50-934"><a href="#cb50-934" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194441</span>] <span class="ot">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span id="cb50-935"><a href="#cb50-935" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334209</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span id="cb50-936"><a href="#cb50-936" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194965</span>] <span class="ot">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span id="cb50-937"><a href="#cb50-937" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333930</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 4"</span></span>
<span id="cb50-938"><a href="#cb50-938" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337048</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 1"</span></span>
<span id="cb50-939"><a href="#cb50-939" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337041</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 9"</span></span>
<span id="cb50-940"><a href="#cb50-940" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337049</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 11"</span></span>
<span id="cb50-941"><a href="#cb50-941" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337044</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 5"</span></span>
<span id="cb50-942"><a href="#cb50-942" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-943"><a href="#cb50-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-944"><a href="#cb50-944" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb50-945"><a href="#cb50-945" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb50-946"><a href="#cb50-946" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb50-947"><a href="#cb50-947" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-948"><a href="#cb50-948" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb50-949"><a href="#cb50-949" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb50-950"><a href="#cb50-950" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb50-951"><a href="#cb50-951" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb50-952"><a href="#cb50-952" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb50-953"><a href="#cb50-953" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb50-954"><a href="#cb50-954" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb50-955"><a href="#cb50-955" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb50-956"><a href="#cb50-956" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb50-957"><a href="#cb50-957" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb50-958"><a href="#cb50-958" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb50-959"><a href="#cb50-959" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb50-960"><a href="#cb50-960" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb50-961"><a href="#cb50-961" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb50-962"><a href="#cb50-962" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb50-963"><a href="#cb50-963" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-964"><a href="#cb50-964" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb50-965"><a href="#cb50-965" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb50-966"><a href="#cb50-966" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb50-967"><a href="#cb50-967" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb50-968"><a href="#cb50-968" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb50-969"><a href="#cb50-969" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb50-970"><a href="#cb50-970" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb50-971"><a href="#cb50-971" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb50-972"><a href="#cb50-972" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb50-973"><a href="#cb50-973" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb50-974"><a href="#cb50-974" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb50-975"><a href="#cb50-975" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb50-976"><a href="#cb50-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb50-977"><a href="#cb50-977" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-978"><a href="#cb50-978" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb50-979"><a href="#cb50-979" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb50-980"><a href="#cb50-980" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb50-981"><a href="#cb50-981" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-982"><a href="#cb50-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-985"><a href="#cb50-985" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-986"><a href="#cb50-986" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb50-987"><a href="#cb50-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-988"><a href="#cb50-988" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-989"><a href="#cb50-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-990"><a href="#cb50-990" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.08</span></span>
<span id="cb50-991"><a href="#cb50-991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-992"><a href="#cb50-992" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb50-993"><a href="#cb50-993" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-994"><a href="#cb50-994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, year, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-995"><a href="#cb50-995" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-996"><a href="#cb50-996" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, year) <span class="sc">%&gt;%</span></span>
<span id="cb50-997"><a href="#cb50-997" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-998"><a href="#cb50-998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-999"><a href="#cb50-999" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1000"><a href="#cb50-1000" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1001"><a href="#cb50-1001" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1002"><a href="#cb50-1002" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb50-1003"><a href="#cb50-1003" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1004"><a href="#cb50-1004" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1005"><a href="#cb50-1005" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, year, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1006"><a href="#cb50-1006" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb50-1007"><a href="#cb50-1007" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1008"><a href="#cb50-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1009"><a href="#cb50-1009" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1010"><a href="#cb50-1010" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1011"><a href="#cb50-1011" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1012"><a href="#cb50-1012" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1013"><a href="#cb50-1013" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1014"><a href="#cb50-1014" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1015"><a href="#cb50-1015" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1016"><a href="#cb50-1016" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1017"><a href="#cb50-1017" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1018"><a href="#cb50-1018" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1019"><a href="#cb50-1019" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb50-1020"><a href="#cb50-1020" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1021"><a href="#cb50-1021" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb50-1022"><a href="#cb50-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1023"><a href="#cb50-1023" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb50-1024"><a href="#cb50-1024" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1025"><a href="#cb50-1025" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb50-1026"><a href="#cb50-1026" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1027"><a href="#cb50-1027" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1028"><a href="#cb50-1028" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1029"><a href="#cb50-1029" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1030"><a href="#cb50-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1031"><a href="#cb50-1031" aria-hidden="true" tabindex="-1"></a>Adenoviruses were in turn overwhelmingly dominated by human mastadenovirus F:</span>
<span id="cb50-1032"><a href="#cb50-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1035"><a href="#cb50-1035" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1036"><a href="#cb50-1036" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-adeno</span></span>
<span id="cb50-1037"><a href="#cb50-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1038"><a href="#cb50-1038" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-1039"><a href="#cb50-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1040"><a href="#cb50-1040" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb50-1041"><a href="#cb50-1041" aria-hidden="true" tabindex="-1"></a>taxid_adeno <span class="ot">&lt;-</span> <span class="dv">10508</span></span>
<span id="cb50-1042"><a href="#cb50-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1043"><a href="#cb50-1043" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of adenoviridae reads</span></span>
<span id="cb50-1044"><a href="#cb50-1044" aria-hidden="true" tabindex="-1"></a>adeno_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno) <span class="sc">%&gt;%</span></span>
<span id="cb50-1045"><a href="#cb50-1045" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1046"><a href="#cb50-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb50-1047"><a href="#cb50-1047" aria-hidden="true" tabindex="-1"></a>adeno_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-1048"><a href="#cb50-1048" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno, sample <span class="sc">%in%</span> adeno_samples) <span class="sc">%&gt;%</span></span>
<span id="cb50-1049"><a href="#cb50-1049" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb50-1050"><a href="#cb50-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1051"><a href="#cb50-1051" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each adenoviridae species</span></span>
<span id="cb50-1052"><a href="#cb50-1052" aria-hidden="true" tabindex="-1"></a>adeno_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb50-1053"><a href="#cb50-1053" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> adeno_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1054"><a href="#cb50-1054" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1055"><a href="#cb50-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1056"><a href="#cb50-1056" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1057"><a href="#cb50-1057" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_adeno =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1058"><a href="#cb50-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1059"><a href="#cb50-1059" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1060"><a href="#cb50-1060" aria-hidden="true" tabindex="-1"></a>adeno_species_major_tab <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1061"><a href="#cb50-1061" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">==</span> <span class="fu">max</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1062"><a href="#cb50-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1063"><a href="#cb50-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1064"><a href="#cb50-1064" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1065"><a href="#cb50-1065" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_major <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1066"><a href="#cb50-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> adeno_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1067"><a href="#cb50-1067" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1068"><a href="#cb50-1068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1069"><a href="#cb50-1069" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_adeno =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1070"><a href="#cb50-1070" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_adeno =</span> <span class="fu">sum</span>(p_reads_adeno), </span>
<span id="cb50-1071"><a href="#cb50-1071" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1072"><a href="#cb50-1072" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1073"><a href="#cb50-1073" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(adeno_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1074"><a href="#cb50-1074" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_display <span class="ot">&lt;-</span> adeno_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1075"><a href="#cb50-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_adeno, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1076"><a href="#cb50-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1077"><a href="#cb50-1077" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1078"><a href="#cb50-1078" aria-hidden="true" tabindex="-1"></a>g_adeno_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1079"><a href="#cb50-1079" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>adeno_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1080"><a href="#cb50-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adenoviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1081"><a href="#cb50-1081" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1082"><a href="#cb50-1082" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1083"><a href="#cb50-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1084"><a href="#cb50-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Adenoviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1085"><a href="#cb50-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1086"><a href="#cb50-1086" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1087"><a href="#cb50-1087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1088"><a href="#cb50-1088" aria-hidden="true" tabindex="-1"></a>g_adeno_species</span>
<span id="cb50-1089"><a href="#cb50-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1090"><a href="#cb50-1090" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1091"><a href="#cb50-1091" aria-hidden="true" tabindex="-1"></a>adeno_species_collate <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1092"><a href="#cb50-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_adeno), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1093"><a href="#cb50-1093" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1094"><a href="#cb50-1094" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1095"><a href="#cb50-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1098"><a href="#cb50-1098" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1099"><a href="#cb50-1099" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: adeno-blast-hits</span></span>
<span id="cb50-1100"><a href="#cb50-1100" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb50-1101"><a href="#cb50-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1102"><a href="#cb50-1102" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb50-1103"><a href="#cb50-1103" aria-hidden="true" tabindex="-1"></a>ref_taxids_hv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">130309</span>)</span>
<span id="cb50-1104"><a href="#cb50-1104" aria-hidden="true" tabindex="-1"></a>ref_names_hv <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ref_taxids_hv, <span class="cf">function</span>(x) viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> x) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> first)</span>
<span id="cb50-1105"><a href="#cb50-1105" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb50-1106"><a href="#cb50-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1107"><a href="#cb50-1107" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb50-1108"><a href="#cb50-1108" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb50-1109"><a href="#cb50-1109" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb50-1110"><a href="#cb50-1110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1111"><a href="#cb50-1111" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb50-1112"><a href="#cb50-1112" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb50-1113"><a href="#cb50-1113" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb50-1114"><a href="#cb50-1114" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb50-1115"><a href="#cb50-1115" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">257877</span>, <span class="st">"Macaca thibetana thibetana"</span>,</span>
<span id="cb50-1116"><a href="#cb50-1116" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">256321</span>, <span class="st">"Lentiviral transfer vector pHsCXW"</span>,</span>
<span id="cb50-1117"><a href="#cb50-1117" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">419242</span>, <span class="st">"Shuttle vector pLvCmvMYOCDHA"</span>,</span>
<span id="cb50-1118"><a href="#cb50-1118" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">419243</span>, <span class="st">"Shuttle vector pLvCmvLacZ"</span>,</span>
<span id="cb50-1119"><a href="#cb50-1119" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">421868</span>, <span class="st">"Cloning vector pLvCmvLacZ.Gfp"</span>,</span>
<span id="cb50-1120"><a href="#cb50-1120" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">421869</span>, <span class="st">"Cloning vector pLvCmvMyocardin.Gfp"</span>,</span>
<span id="cb50-1121"><a href="#cb50-1121" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">426303</span>, <span class="st">"Lentiviral vector pNL-GFP-RRE(SA)"</span>,</span>
<span id="cb50-1122"><a href="#cb50-1122" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">436015</span>, <span class="st">"Lentiviral transfer vector pFTMGW"</span>,</span>
<span id="cb50-1123"><a href="#cb50-1123" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">454257</span>, <span class="st">"Shuttle vector pLvCmvMYOCD2aHA"</span>,</span>
<span id="cb50-1124"><a href="#cb50-1124" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">476184</span>, <span class="st">"Shuttle vector pLV.mMyoD::ERT2.eGFP"</span>,</span>
<span id="cb50-1125"><a href="#cb50-1125" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">476185</span>, <span class="st">"Shuttle vector pLV.hMyoD.eGFP"</span>,</span>
<span id="cb50-1126"><a href="#cb50-1126" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">591936</span>, <span class="st">"Piliocolobus tephrosceles"</span>,</span>
<span id="cb50-1127"><a href="#cb50-1127" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">627481</span>, <span class="st">"Lentiviral transfer vector pFTM3GW"</span>,</span>
<span id="cb50-1128"><a href="#cb50-1128" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">680261</span>, <span class="st">"Self-inactivating lentivirus vector pLV.C-EF1a.cyt-bGal.dCpG"</span>,</span>
<span id="cb50-1129"><a href="#cb50-1129" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">2952778</span>, <span class="st">"Expression vector pLV[Exp]-EGFP:T2A:Puro-EF1A"</span>,</span>
<span id="cb50-1130"><a href="#cb50-1130" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3022699</span>, <span class="st">"Vector PAS_122122"</span>,</span>
<span id="cb50-1131"><a href="#cb50-1131" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3025913</span>, <span class="st">"Vector pSIN-WP-mPGK-GDNF"</span>,</span>
<span id="cb50-1132"><a href="#cb50-1132" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3105863</span>, <span class="st">"Vector pLKO.1-ZsGreen1"</span>,</span>
<span id="cb50-1133"><a href="#cb50-1133" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3105864</span>, <span class="st">"Vector pLKO.1-ZsGreen1 mouse Wfs1 shRNA"</span>,</span>
<span id="cb50-1134"><a href="#cb50-1134" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3108001</span>, <span class="st">"Cloning vector pLVSIN-CMV_Neo_v4.0"</span>,</span>
<span id="cb50-1135"><a href="#cb50-1135" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3109234</span>, <span class="st">"Vector pTwist+Kan+High"</span>,</span>
<span id="cb50-1136"><a href="#cb50-1136" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117662</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L"</span>,</span>
<span id="cb50-1137"><a href="#cb50-1137" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117663</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L:T2A:mRuby3"</span>,</span>
<span id="cb50-1138"><a href="#cb50-1138" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117664</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;hMAPT[NM_005910.6](ns):T2A:mRuby3"</span>,</span>
<span id="cb50-1139"><a href="#cb50-1139" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117665</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3"</span>,</span>
<span id="cb50-1140"><a href="#cb50-1140" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117666</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3/NFAT3 fusion protein"</span>,</span>
<span id="cb50-1141"><a href="#cb50-1141" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117667</span>, <span class="st">"Cloning vector pLV[Exp]-Neo-mPGK&gt;{EGFP-hSEPT6}"</span>,</span>
<span id="cb50-1142"><a href="#cb50-1142" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">438045</span>, <span class="st">"Xenotropic MuLV-related virus"</span>,</span>
<span id="cb50-1143"><a href="#cb50-1143" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">447135</span>, <span class="st">"Myodes glareolus"</span>,</span>
<span id="cb50-1144"><a href="#cb50-1144" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">590745</span>, <span class="st">"Mus musculus mobilized endogenous polytropic provirus"</span>,</span>
<span id="cb50-1145"><a href="#cb50-1145" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">181858</span>, <span class="st">"Murine AIDS virus-related provirus"</span>,</span>
<span id="cb50-1146"><a href="#cb50-1146" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">356663</span>, <span class="st">"Xenotropic MuLV-related virus VP35"</span>,</span>
<span id="cb50-1147"><a href="#cb50-1147" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">356664</span>, <span class="st">"Xenotropic MuLV-related virus VP42"</span>,</span>
<span id="cb50-1148"><a href="#cb50-1148" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">373193</span>, <span class="st">"Xenotropic MuLV-related virus VP62"</span>,</span>
<span id="cb50-1149"><a href="#cb50-1149" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">286419</span>, <span class="st">"Canis lupus dingo"</span>,</span>
<span id="cb50-1150"><a href="#cb50-1150" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">415978</span>, <span class="st">"Sus scrofa scrofa"</span>,</span>
<span id="cb50-1151"><a href="#cb50-1151" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">494514</span>, <span class="st">"Vulpes lagopus"</span>,</span>
<span id="cb50-1152"><a href="#cb50-1152" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span id="cb50-1153"><a href="#cb50-1153" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3119969</span>, <span class="st">"Bubalus kerabau"</span>)</span>
<span id="cb50-1154"><a href="#cb50-1154" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(tax_names, tax_names_new)</span>
<span id="cb50-1155"><a href="#cb50-1155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1156"><a href="#cb50-1156" aria-hidden="true" tabindex="-1"></a><span class="co"># Get matches</span></span>
<span id="cb50-1157"><a href="#cb50-1157" aria-hidden="true" tabindex="-1"></a>hv_blast_staxids <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">%in%</span> ref_taxids_hv) <span class="sc">%&gt;%</span></span>
<span id="cb50-1158"><a href="#cb50-1158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb50-1159"><a href="#cb50-1159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1160"><a href="#cb50-1160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1161"><a href="#cb50-1161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name), <span class="at">by=</span><span class="st">"staxid"</span>)</span>
<span id="cb50-1162"><a href="#cb50-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1163"><a href="#cb50-1163" aria-hidden="true" tabindex="-1"></a><span class="co"># Count matches</span></span>
<span id="cb50-1164"><a href="#cb50-1164" aria-hidden="true" tabindex="-1"></a>hv_blast_counts <span class="ot">&lt;-</span> hv_blast_staxids <span class="sc">%&gt;%</span></span>
<span id="cb50-1165"><a href="#cb50-1165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(taxid, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb50-1166"><a href="#cb50-1166" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb50-1167"><a href="#cb50-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1168"><a href="#cb50-1168" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to major matches</span></span>
<span id="cb50-1169"><a href="#cb50-1169" aria-hidden="true" tabindex="-1"></a>hv_blast_counts_major <span class="ot">&lt;-</span> hv_blast_counts <span class="sc">%&gt;%</span> </span>
<span id="cb50-1170"><a href="#cb50-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1171"><a href="#cb50-1171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1172"><a href="#cb50-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1173"><a href="#cb50-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> name)</span>
<span id="cb50-1174"><a href="#cb50-1174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1175"><a href="#cb50-1175" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1176"><a href="#cb50-1176" aria-hidden="true" tabindex="-1"></a>g_hv_blast <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_blast_counts_major, <span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span></span>
<span id="cb50-1177"><a href="#cb50-1177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb50-1178"><a href="#cb50-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(name_display<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>, <span class="at">space=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb50-1179"><a href="#cb50-1179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), </span>
<span id="cb50-1180"><a href="#cb50-1180" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb50-1181"><a href="#cb50-1181" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb50-1182"><a href="#cb50-1182" aria-hidden="true" tabindex="-1"></a>g_hv_blast</span>
<span id="cb50-1183"><a href="#cb50-1183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1184"><a href="#cb50-1184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1185"><a href="#cb50-1185" aria-hidden="true" tabindex="-1"></a>In investigating more minor viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</span>
<span id="cb50-1186"><a href="#cb50-1186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1189"><a href="#cb50-1189" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1190"><a href="#cb50-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-polyoma</span></span>
<span id="cb50-1191"><a href="#cb50-1191" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1192"><a href="#cb50-1192" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-1193"><a href="#cb50-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1194"><a href="#cb50-1194" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb50-1195"><a href="#cb50-1195" aria-hidden="true" tabindex="-1"></a>taxid_polyoma <span class="ot">&lt;-</span> <span class="dv">151341</span></span>
<span id="cb50-1196"><a href="#cb50-1196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1197"><a href="#cb50-1197" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of polyomaviridae reads</span></span>
<span id="cb50-1198"><a href="#cb50-1198" aria-hidden="true" tabindex="-1"></a>polyoma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma) <span class="sc">%&gt;%</span></span>
<span id="cb50-1199"><a href="#cb50-1199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1200"><a href="#cb50-1200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb50-1201"><a href="#cb50-1201" aria-hidden="true" tabindex="-1"></a>polyoma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-1202"><a href="#cb50-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma, sample <span class="sc">%in%</span> polyoma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb50-1203"><a href="#cb50-1203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb50-1204"><a href="#cb50-1204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1205"><a href="#cb50-1205" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each polyomaviridae species</span></span>
<span id="cb50-1206"><a href="#cb50-1206" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb50-1207"><a href="#cb50-1207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> polyoma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1208"><a href="#cb50-1208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1209"><a href="#cb50-1209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1210"><a href="#cb50-1210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1211"><a href="#cb50-1211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_polyoma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1212"><a href="#cb50-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1213"><a href="#cb50-1213" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1214"><a href="#cb50-1214" aria-hidden="true" tabindex="-1"></a>polyoma_species_major_tab <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1215"><a href="#cb50-1215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">==</span> <span class="fu">max</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1216"><a href="#cb50-1216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1217"><a href="#cb50-1217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1218"><a href="#cb50-1218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1219"><a href="#cb50-1219" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_major <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1220"><a href="#cb50-1220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> polyoma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1221"><a href="#cb50-1221" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1222"><a href="#cb50-1222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1223"><a href="#cb50-1223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_polyoma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1224"><a href="#cb50-1224" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_polyoma =</span> <span class="fu">sum</span>(p_reads_polyoma), </span>
<span id="cb50-1225"><a href="#cb50-1225" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1226"><a href="#cb50-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1227"><a href="#cb50-1227" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(polyoma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1228"><a href="#cb50-1228" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_display <span class="ot">&lt;-</span> polyoma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1229"><a href="#cb50-1229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_polyoma, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1230"><a href="#cb50-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1231"><a href="#cb50-1231" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1232"><a href="#cb50-1232" aria-hidden="true" tabindex="-1"></a>g_polyoma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1233"><a href="#cb50-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>polyoma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1234"><a href="#cb50-1234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Polyomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1235"><a href="#cb50-1235" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1236"><a href="#cb50-1236" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1237"><a href="#cb50-1237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1238"><a href="#cb50-1238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Polyomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1239"><a href="#cb50-1239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1240"><a href="#cb50-1240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1241"><a href="#cb50-1241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1242"><a href="#cb50-1242" aria-hidden="true" tabindex="-1"></a>g_polyoma_species</span>
<span id="cb50-1243"><a href="#cb50-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1244"><a href="#cb50-1244" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1245"><a href="#cb50-1245" aria-hidden="true" tabindex="-1"></a>polyoma_species_collate <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1246"><a href="#cb50-1246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_polyoma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1247"><a href="#cb50-1247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1248"><a href="#cb50-1248" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1249"><a href="#cb50-1249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1252"><a href="#cb50-1252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1253"><a href="#cb50-1253" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-papilloma</span></span>
<span id="cb50-1254"><a href="#cb50-1254" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1255"><a href="#cb50-1255" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-1256"><a href="#cb50-1256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1257"><a href="#cb50-1257" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.35</span></span>
<span id="cb50-1258"><a href="#cb50-1258" aria-hidden="true" tabindex="-1"></a>taxid_papilloma <span class="ot">&lt;-</span> <span class="dv">151340</span></span>
<span id="cb50-1259"><a href="#cb50-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1260"><a href="#cb50-1260" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of papillomaviridae reads</span></span>
<span id="cb50-1261"><a href="#cb50-1261" aria-hidden="true" tabindex="-1"></a>papilloma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma) <span class="sc">%&gt;%</span></span>
<span id="cb50-1262"><a href="#cb50-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1263"><a href="#cb50-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb50-1264"><a href="#cb50-1264" aria-hidden="true" tabindex="-1"></a>papilloma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-1265"><a href="#cb50-1265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma, sample <span class="sc">%in%</span> papilloma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb50-1266"><a href="#cb50-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb50-1267"><a href="#cb50-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1268"><a href="#cb50-1268" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each papillomaviridae species</span></span>
<span id="cb50-1269"><a href="#cb50-1269" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb50-1270"><a href="#cb50-1270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> papilloma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1271"><a href="#cb50-1271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1272"><a href="#cb50-1272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1273"><a href="#cb50-1273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1274"><a href="#cb50-1274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_papilloma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1275"><a href="#cb50-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1276"><a href="#cb50-1276" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1277"><a href="#cb50-1277" aria-hidden="true" tabindex="-1"></a>papilloma_species_major_tab <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1278"><a href="#cb50-1278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">==</span> <span class="fu">max</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1279"><a href="#cb50-1279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1280"><a href="#cb50-1280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1281"><a href="#cb50-1281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1282"><a href="#cb50-1282" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_major <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1283"><a href="#cb50-1283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> papilloma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1284"><a href="#cb50-1284" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1285"><a href="#cb50-1285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1286"><a href="#cb50-1286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_papilloma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1287"><a href="#cb50-1287" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_papilloma =</span> <span class="fu">sum</span>(p_reads_papilloma), </span>
<span id="cb50-1288"><a href="#cb50-1288" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1289"><a href="#cb50-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1290"><a href="#cb50-1290" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(papilloma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1291"><a href="#cb50-1291" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_display <span class="ot">&lt;-</span> papilloma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1292"><a href="#cb50-1292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_papilloma, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1293"><a href="#cb50-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1294"><a href="#cb50-1294" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1295"><a href="#cb50-1295" aria-hidden="true" tabindex="-1"></a>g_papilloma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1296"><a href="#cb50-1296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>papilloma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1297"><a href="#cb50-1297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Papillomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1298"><a href="#cb50-1298" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1299"><a href="#cb50-1299" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1300"><a href="#cb50-1300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1301"><a href="#cb50-1301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Papillomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1302"><a href="#cb50-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1303"><a href="#cb50-1303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1304"><a href="#cb50-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1305"><a href="#cb50-1305" aria-hidden="true" tabindex="-1"></a>g_papilloma_species</span>
<span id="cb50-1306"><a href="#cb50-1306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1307"><a href="#cb50-1307" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1308"><a href="#cb50-1308" aria-hidden="true" tabindex="-1"></a>papilloma_species_collate <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1309"><a href="#cb50-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_papilloma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1310"><a href="#cb50-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1311"><a href="#cb50-1311" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1312"><a href="#cb50-1312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1315"><a href="#cb50-1315" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1316"><a href="#cb50-1316" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-pox</span></span>
<span id="cb50-1317"><a href="#cb50-1317" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1318"><a href="#cb50-1318" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-1319"><a href="#cb50-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1320"><a href="#cb50-1320" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb50-1321"><a href="#cb50-1321" aria-hidden="true" tabindex="-1"></a>taxid_pox <span class="ot">&lt;-</span> <span class="dv">10240</span></span>
<span id="cb50-1322"><a href="#cb50-1322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1323"><a href="#cb50-1323" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of poxviridae reads</span></span>
<span id="cb50-1324"><a href="#cb50-1324" aria-hidden="true" tabindex="-1"></a>pox_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox) <span class="sc">%&gt;%</span></span>
<span id="cb50-1325"><a href="#cb50-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1326"><a href="#cb50-1326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb50-1327"><a href="#cb50-1327" aria-hidden="true" tabindex="-1"></a>pox_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-1328"><a href="#cb50-1328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox, sample <span class="sc">%in%</span> pox_samples) <span class="sc">%&gt;%</span></span>
<span id="cb50-1329"><a href="#cb50-1329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb50-1330"><a href="#cb50-1330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1331"><a href="#cb50-1331" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each poxviridae species</span></span>
<span id="cb50-1332"><a href="#cb50-1332" aria-hidden="true" tabindex="-1"></a>pox_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb50-1333"><a href="#cb50-1333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> pox_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1334"><a href="#cb50-1334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1335"><a href="#cb50-1335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1336"><a href="#cb50-1336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1337"><a href="#cb50-1337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_pox =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1338"><a href="#cb50-1338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1339"><a href="#cb50-1339" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1340"><a href="#cb50-1340" aria-hidden="true" tabindex="-1"></a>pox_species_major_tab <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1341"><a href="#cb50-1341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">==</span> <span class="fu">max</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1342"><a href="#cb50-1342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1343"><a href="#cb50-1343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1344"><a href="#cb50-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1345"><a href="#cb50-1345" aria-hidden="true" tabindex="-1"></a>pox_species_counts_major <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1346"><a href="#cb50-1346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> pox_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1347"><a href="#cb50-1347" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1348"><a href="#cb50-1348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1349"><a href="#cb50-1349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_pox =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1350"><a href="#cb50-1350" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_pox =</span> <span class="fu">sum</span>(p_reads_pox), </span>
<span id="cb50-1351"><a href="#cb50-1351" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1352"><a href="#cb50-1352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1353"><a href="#cb50-1353" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(pox_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1354"><a href="#cb50-1354" aria-hidden="true" tabindex="-1"></a>pox_species_counts_display <span class="ot">&lt;-</span> pox_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1355"><a href="#cb50-1355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_pox, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1356"><a href="#cb50-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1357"><a href="#cb50-1357" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1358"><a href="#cb50-1358" aria-hidden="true" tabindex="-1"></a>g_pox_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1359"><a href="#cb50-1359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>pox_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1360"><a href="#cb50-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Poxviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1361"><a href="#cb50-1361" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1362"><a href="#cb50-1362" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1363"><a href="#cb50-1363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1364"><a href="#cb50-1364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Poxviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1365"><a href="#cb50-1365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1366"><a href="#cb50-1366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1367"><a href="#cb50-1367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1368"><a href="#cb50-1368" aria-hidden="true" tabindex="-1"></a>g_pox_species</span>
<span id="cb50-1369"><a href="#cb50-1369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1370"><a href="#cb50-1370" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1371"><a href="#cb50-1371" aria-hidden="true" tabindex="-1"></a>pox_species_collate <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1372"><a href="#cb50-1372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_pox), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1373"><a href="#cb50-1373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1374"><a href="#cb50-1374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1375"><a href="#cb50-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1378"><a href="#cb50-1378" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1379"><a href="#cb50-1379" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-herpes</span></span>
<span id="cb50-1380"><a href="#cb50-1380" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1381"><a href="#cb50-1381" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb50-1382"><a href="#cb50-1382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1383"><a href="#cb50-1383" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb50-1384"><a href="#cb50-1384" aria-hidden="true" tabindex="-1"></a>taxid_herpes <span class="ot">&lt;-</span> <span class="dv">10292</span></span>
<span id="cb50-1385"><a href="#cb50-1385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1386"><a href="#cb50-1386" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of herpesviridae reads</span></span>
<span id="cb50-1387"><a href="#cb50-1387" aria-hidden="true" tabindex="-1"></a>herpes_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes) <span class="sc">%&gt;%</span></span>
<span id="cb50-1388"><a href="#cb50-1388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1389"><a href="#cb50-1389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb50-1390"><a href="#cb50-1390" aria-hidden="true" tabindex="-1"></a>herpes_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb50-1391"><a href="#cb50-1391" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes, sample <span class="sc">%in%</span> herpes_samples) <span class="sc">%&gt;%</span></span>
<span id="cb50-1392"><a href="#cb50-1392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb50-1393"><a href="#cb50-1393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1394"><a href="#cb50-1394" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae species</span></span>
<span id="cb50-1395"><a href="#cb50-1395" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb50-1396"><a href="#cb50-1396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1397"><a href="#cb50-1397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1398"><a href="#cb50-1398" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1399"><a href="#cb50-1399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1400"><a href="#cb50-1400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1401"><a href="#cb50-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1402"><a href="#cb50-1402" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1403"><a href="#cb50-1403" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1404"><a href="#cb50-1404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1405"><a href="#cb50-1405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1406"><a href="#cb50-1406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1407"><a href="#cb50-1407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1408"><a href="#cb50-1408" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1409"><a href="#cb50-1409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1410"><a href="#cb50-1410" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1411"><a href="#cb50-1411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1412"><a href="#cb50-1412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1413"><a href="#cb50-1413" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb50-1414"><a href="#cb50-1414" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1415"><a href="#cb50-1415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1416"><a href="#cb50-1416" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1417"><a href="#cb50-1417" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1418"><a href="#cb50-1418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1419"><a href="#cb50-1419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1420"><a href="#cb50-1420" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1421"><a href="#cb50-1421" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1422"><a href="#cb50-1422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1423"><a href="#cb50-1423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1424"><a href="#cb50-1424" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1425"><a href="#cb50-1425" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1426"><a href="#cb50-1426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1427"><a href="#cb50-1427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1428"><a href="#cb50-1428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1429"><a href="#cb50-1429" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1430"><a href="#cb50-1430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1431"><a href="#cb50-1431" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb50-1432"><a href="#cb50-1432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1433"><a href="#cb50-1433" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1434"><a href="#cb50-1434" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1435"><a href="#cb50-1435" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1436"><a href="#cb50-1436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1437"><a href="#cb50-1437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1438"><a href="#cb50-1438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1439"><a href="#cb50-1439" aria-hidden="true" tabindex="-1"></a>In the latter case, some subset of human alphaherpesvirus 1 reads will be arising from strain RH2, which we know to be dominated by false positive hits. Digging deeper, we see that these are essentially all of the alphaherpesvirus 1 reads. Hence, these herpesvirus read counts need to be treated with significant skepticism, at least until the RH2 problem is fixed.</span>
<span id="cb50-1440"><a href="#cb50-1440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1443"><a href="#cb50-1443" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1444"><a href="#cb50-1444" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: herpes-species</span></span>
<span id="cb50-1445"><a href="#cb50-1445" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb50-1446"><a href="#cb50-1446" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae taxon</span></span>
<span id="cb50-1447"><a href="#cb50-1447" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> mrg_hv_named <span class="sc">%&gt;%</span></span>
<span id="cb50-1448"><a href="#cb50-1448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb50-1449"><a href="#cb50-1449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb50-1450"><a href="#cb50-1450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1451"><a href="#cb50-1451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument) <span class="sc">%&gt;%</span></span>
<span id="cb50-1452"><a href="#cb50-1452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb50-1453"><a href="#cb50-1453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1454"><a href="#cb50-1454" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb50-1455"><a href="#cb50-1455" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1456"><a href="#cb50-1456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1457"><a href="#cb50-1457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1458"><a href="#cb50-1458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1459"><a href="#cb50-1459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb50-1460"><a href="#cb50-1460" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-1461"><a href="#cb50-1461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb50-1462"><a href="#cb50-1462" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1463"><a href="#cb50-1463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, instrument, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb50-1464"><a href="#cb50-1464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb50-1465"><a href="#cb50-1465" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb50-1466"><a href="#cb50-1466" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1467"><a href="#cb50-1467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb50-1468"><a href="#cb50-1468" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb50-1469"><a href="#cb50-1469" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb50-1470"><a href="#cb50-1470" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb50-1471"><a href="#cb50-1471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1472"><a href="#cb50-1472" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1473"><a href="#cb50-1473" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb50-1474"><a href="#cb50-1474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb50-1475"><a href="#cb50-1475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb50-1476"><a href="#cb50-1476" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb50-1477"><a href="#cb50-1477" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb50-1478"><a href="#cb50-1478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb50-1479"><a href="#cb50-1479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb50-1480"><a href="#cb50-1480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb50-1481"><a href="#cb50-1481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb50-1482"><a href="#cb50-1482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1483"><a href="#cb50-1483" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb50-1484"><a href="#cb50-1484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1485"><a href="#cb50-1485" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb50-1486"><a href="#cb50-1486" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1487"><a href="#cb50-1487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1488"><a href="#cb50-1488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb50-1489"><a href="#cb50-1489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1490"><a href="#cb50-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1491"><a href="#cb50-1491" aria-hidden="true" tabindex="-1"></a>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry. Note that, as *Simplexvirus* includes human alphaherpesvirus 1, that count should be treated with significant caution until the RH2 problem is fixed.</span>
<span id="cb50-1492"><a href="#cb50-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1495"><a href="#cb50-1495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-1496"><a href="#cb50-1496" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb50-1497"><a href="#cb50-1497" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-genera</span></span>
<span id="cb50-1498"><a href="#cb50-1498" aria-hidden="true" tabindex="-1"></a><span class="co">#| warn: false</span></span>
<span id="cb50-1499"><a href="#cb50-1499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1500"><a href="#cb50-1500" aria-hidden="true" tabindex="-1"></a><span class="co"># Define reference genera</span></span>
<span id="cb50-1501"><a href="#cb50-1501" aria-hidden="true" tabindex="-1"></a>path_genera_rna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span>)</span>
<span id="cb50-1502"><a href="#cb50-1502" aria-hidden="true" tabindex="-1"></a>path_genera_dna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span id="cb50-1503"><a href="#cb50-1503" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span>)</span>
<span id="cb50-1504"><a href="#cb50-1504" aria-hidden="true" tabindex="-1"></a>path_genera <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">tibble</span>(<span class="at">name=</span>path_genera_rna, <span class="at">genome_type=</span><span class="st">"RNA genome"</span>),</span>
<span id="cb50-1505"><a href="#cb50-1505" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">tibble</span>(<span class="at">name=</span>path_genera_dna, <span class="at">genome_type=</span><span class="st">"DNA genome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1506"><a href="#cb50-1506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb50-1507"><a href="#cb50-1507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1508"><a href="#cb50-1508" aria-hidden="true" tabindex="-1"></a><span class="co"># Count in each sample</span></span>
<span id="cb50-1509"><a href="#cb50-1509" aria-hidden="true" tabindex="-1"></a>mrg_hv_named_all <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb50-1510"><a href="#cb50-1510" aria-hidden="true" tabindex="-1"></a>hv_reads_genus_all <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named_all, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb50-1511"><a href="#cb50-1511" aria-hidden="true" tabindex="-1"></a>n_path_genera <span class="ot">&lt;-</span> hv_reads_genus_all <span class="sc">%&gt;%</span> </span>
<span id="cb50-1512"><a href="#cb50-1512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1513"><a href="#cb50-1513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-1514"><a href="#cb50-1514" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(path_genera, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1515"><a href="#cb50-1515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"location"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-1516"><a href="#cb50-1516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb50-1517"><a href="#cb50-1517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1518"><a href="#cb50-1518" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot out and back to add zero lines</span></span>
<span id="cb50-1519"><a href="#cb50-1519" aria-hidden="true" tabindex="-1"></a>n_path_genera_out <span class="ot">&lt;-</span> n_path_genera <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, name, n_reads_viral) <span class="sc">%&gt;%</span></span>
<span id="cb50-1520"><a href="#cb50-1520" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"name"</span>, <span class="at">values_from=</span><span class="st">"n_reads_viral"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1521"><a href="#cb50-1521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to=</span><span class="st">"name"</span>, <span class="at">values_to=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1522"><a href="#cb50-1522" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1523"><a href="#cb50-1523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(path_genera, <span class="at">by=</span><span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1524"><a href="#cb50-1524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb50-1525"><a href="#cb50-1525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1526"><a href="#cb50-1526" aria-hidden="true" tabindex="-1"></a><span class="do">## Aggregate across dates</span></span>
<span id="cb50-1527"><a href="#cb50-1527" aria-hidden="true" tabindex="-1"></a>n_path_genera_stype <span class="ot">&lt;-</span> n_path_genera_out <span class="sc">%&gt;%</span> </span>
<span id="cb50-1528"><a href="#cb50-1528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid, genome_type, location) <span class="sc">%&gt;%</span></span>
<span id="cb50-1529"><a href="#cb50-1529" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb50-1530"><a href="#cb50-1530" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_viral =</span> <span class="fu">sum</span>(n_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-1531"><a href="#cb50-1531" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="st">"All samples"</span>,</span>
<span id="cb50-1532"><a href="#cb50-1532" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw,</span>
<span id="cb50-1533"><a href="#cb50-1533" aria-hidden="true" tabindex="-1"></a>         <span class="at">na_type =</span> <span class="st">"DNA"</span>)</span>
<span id="cb50-1534"><a href="#cb50-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1535"><a href="#cb50-1535" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb50-1536"><a href="#cb50-1536" aria-hidden="true" tabindex="-1"></a>g_path_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_path_genera_stype,</span>
<span id="cb50-1537"><a href="#cb50-1537" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y=</span>name, <span class="at">x=</span>p_reads_viral, <span class="at">color=</span>location)) <span class="sc">+</span></span>
<span id="cb50-1538"><a href="#cb50-1538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb50-1539"><a href="#cb50-1539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb50-1540"><a href="#cb50-1540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_loc</span>() <span class="sc">+</span></span>
<span id="cb50-1541"><a href="#cb50-1541" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genome_type<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb50-1542"><a href="#cb50-1542" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb50-1543"><a href="#cb50-1543" aria-hidden="true" tabindex="-1"></a>g_path_genera</span>
<span id="cb50-1544"><a href="#cb50-1544" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-1545"><a href="#cb50-1545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1546"><a href="#cb50-1546" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb50-1547"><a href="#cb50-1547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1548"><a href="#cb50-1548" aria-hidden="true" tabindex="-1"></a>This is the first major DNA dataset from wastewater that I've analyzed using this pipeline, and at first I was nervous about how it would perform. Most of the optimization I've done on the HV detection pipeline has been on RNA datasets, so I was concerned that we would see major problems with false positives that we haven't encountered with other datasets. In the event, however, things went well. There were some preventable false positives (especially from human alphaherpesvirus 1 strain RH2) but they were rare enough relative to correct assignments that the pipeline's overall performance scores were very high. Overall I wouldn't trust the performance of this pipeline on DNA data much less than on RNA.</span>
<span id="cb50-1549"><a href="#cb50-1549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-1550"><a href="#cb50-1550" aria-hidden="true" tabindex="-1"></a>In terms of the results themselves, it was interesting to see the remarkable preponderance of adenoviruses, and especially human mastadenovirus F, among human-infecting viruses. This isn't a pattern we've seen in RNA data, and it will be interesting to see if it persists in other DNA datasets.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>