<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-04-19">
<title>Will’s Public NAO Notebook - Workflow analysis of Leung et al.&nbsp;(2021)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Leung et al.&nbsp;(2021)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Air sampling from urban public transit systems.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 19, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>The last in our current run of air sampling datasets is <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01044-7">Leung et al.</a> (2021), a study of active air samples collected in public transit systems from six cities (Denver, Hong Kong, London, NYC, Oslo, Stockholm) from June to September 2017.</p>
<p>Samples from Denver originated from their rail and bus system; all other samples originated from metro systems. Collection took place during working days and working hours. Air samples were collected with the SASS 3100 Dry Air Samplers (filtration) for 30 min at a flowrate of 300 L/min using electret microfibrous filters. Filters were stationed at 1.5m above floor level, facing downward (to avoid direct deposition).</p>
<p>This was a DNA-sequencing study, focused on the bacterial microbiome and resistome. Sample processing followed an ideosyncratic protocol, where samples were pelleted and the pellet and supernatant were processed separately before being recombined for NA extraction and sequencing; I don’t have a great understanding of how this is expected to affect the viral fraction. Samples were sequenced with Illumina HiSeqX 2x150bp.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>In total, the Leung dataset comprised 293 samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Importing the data is a bit more complicated this time as the samples are split across three pipeline runs</span></span>
<span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-04-12_leung"</span></span>
<span><span class="va">data_dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"/"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define geo relationships for filling in</span></span>
<span><span class="va">geo</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">region</span>, <span class="op">~</span><span class="va">country</span>, <span class="op">~</span><span class="va">city</span>,</span>
<span>               <span class="st">"Asia"</span>, <span class="st">"Hong Kong"</span>, <span class="st">"Hong Kong"</span>,</span>
<span>               <span class="st">"Europe"</span>, <span class="st">"Norway"</span>, <span class="st">"Oslo"</span>,</span>
<span>               <span class="st">"Europe"</span>, <span class="st">"Sweden"</span>, <span class="st">"Stockholm"</span>,</span>
<span>               <span class="st">"Europe"</span>, <span class="st">"United Kingdom"</span>, <span class="st">"London"</span>,</span>
<span>               <span class="st">"North America"</span>, <span class="st">"USA"</span>, <span class="st">"New York City"</span>,</span>
<span>               <span class="st">"North America"</span>, <span class="st">"USA"</span>, <span class="st">"Denver"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data input paths</span></span>
<span><span class="va">libraries_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_basic_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_adapter_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_base_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">libraries_paths</span>, <span class="va">read_csv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">bind_rows</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Fix missing entries</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>city <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">city</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">", .*"</span>, <span class="st">""</span>, <span class="va">location</span><span class="op">)</span>, <span class="va">city</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">geo</span>, by<span class="op">=</span><span class="st">"city"</span>, suffix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"_new"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>region <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">region</span> <span class="op">==</span> <span class="st">"uncalculated"</span>, <span class="va">region_new</span>, <span class="va">region</span><span class="op">)</span>,</span>
<span>         country <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">country</span> <span class="op">==</span> <span class="st">"uncalculated"</span>, <span class="va">country_new</span>, <span class="va">country</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">country_new</span>, <span class="op">-</span><span class="va">region_new</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Add sample aliases</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">city</span>, <span class="va">date</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">city</span>, <span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample_count <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         date_alias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">date</span><span class="op">)</span>, <span class="va">sample_count</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>,</span>
<span>         sample_alias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">city</span>, <span class="va">date_alias</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">count_city</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">region</span>, <span class="va">country</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_samples"</span><span class="op">)</span></span>
<span><span class="va">count_city</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["region"],"name":[1],"type":["chr"],"align":["left"]},{"label":["country"],"name":[2],"type":["chr"],"align":["left"]},{"label":["city"],"name":[3],"type":["chr"],"align":["left"]},{"label":["n_samples"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"Asia","2":"Hong Kong","3":"Hong Kong","4":"81"},{"1":"Europe","2":"Norway","3":"Oslo","4":"79"},{"1":"Europe","2":"Sweden","3":"Stockholm","4":"32"},{"1":"Europe","2":"United Kingdom","3":"London","4":"41"},{"1":"North America","2":"USA","3":"Denver","4":"31"},{"1":"North America","2":"USA","3":"New York City","4":"29"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">import_basic</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>           sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">import_basic_paired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">import_basic</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic</span><span class="op">(</span><span class="va">basic_stats_paths</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">adapter_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_base_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_seq_stats_paths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get key values for readout</span></span>
<span><span class="va">raw_read_counts</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>rmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, rmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            rmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, </span>
<span>            rtot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            btot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span>,</span>
<span>            dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>These 293 samples yielded 0.39M-7.86M (mean 4.57M) reads per sample, for a total of 1.34B read pairs (402 gigabases of sequence). Read qualities were high at the 5’ end but dropped off significantly in some samples, in definite need of cleaning. Adapter levels were high. With the exception of a couple of early samples, inferred duplication levels were low (mean 9.4%).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">city</span>, <span class="va">date</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">date</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span><span class="va">scale_fill_city</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                  name<span class="op">=</span><span class="st">"City"</span><span class="op">)</span></span>
<span><span class="va">scale_x_cdate</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_x_date</span>, name<span class="op">=</span><span class="st">"Collection Date"</span>,</span>
<span>                                date_breaks <span class="op">=</span> <span class="st">"1 month"</span>, date_labels <span class="op">=</span> <span class="st">"%Y-%m-%d"</span><span class="op">)</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, </span>
<span>                  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">date</span>, y<span class="op">=</span><span class="va">value</span>, fill<span class="op">=</span><span class="va">city</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">city</span>,<span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_cdate</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space<span class="op">=</span><span class="st">"free_x"</span>, switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_rotate</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">scale_color_city</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                   name<span class="op">=</span><span class="st">"City"</span><span class="op">)</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">city</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Read loss during cleaning was highly variable but averaged 11%, with a further ~7% lost during deduplication and ~0.3% during ribodepletion.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>, <span class="va">city</span>, <span class="va">stage</span>, </span>
<span>         <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span>, City<span class="op">=</span><span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"4.8-88.7 (mean 11)","3":"4.8-88.7 (mean 11)"},{"1":"Deduplication","2":"11.5-89.9 (mean 17.8)","3":"1.2-11 (mean 6.8)"},{"1":"Initial ribodepletion","2":"11.7-89.9 (mean 18.1)","3":"0-0.7 (mean 0.2)"},{"1":"Secondary ribodepletion","2":"11.8-89.9 (mean 18.1)","3":"0-0.4 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">city</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">city</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">city</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">city</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Data cleaning was very successful at removing adapters and improving read qualities:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">city</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 9.4% to 1.7% for DNA reads:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import Bracken data</span></span>
<span><span class="va">bracken_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"bracken_counts.tsv"</span><span class="op">)</span></span>
<span><span class="va">bracken</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">bracken_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span><span class="va">total_assigned</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  kraken_assigned_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">kraken_assigned_reads</span><span class="op">)</span>,</span>
<span>  added_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">added_reads</span><span class="op">)</span>,</span>
<span>  new_est_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">new_est_reads</span><span class="op">)</span>,</span>
<span>  fraction_total_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">fraction_total_reads</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bracken_spread</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"sample"</span>, names_from <span class="op">=</span> <span class="st">"name"</span>, </span>
<span>              values_from <span class="op">=</span> <span class="st">"new_est_reads"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">stage</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"date"</span>, <span class="st">"date_alias"</span>, <span class="st">"city"</span><span class="op">)</span>,</span>
<span>              names_from<span class="op">=</span><span class="st">"stage"</span>, values_from<span class="op">=</span><span class="st">"n_read_pairs"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_preproc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">total_assigned</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>assigned <span class="op">=</span> <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">bracken_spread</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assess composition</span></span>
<span><span class="va">read_comp</span> <span class="op">&lt;-</span> <span class="fu">transmute</span><span class="op">(</span><span class="va">read_counts</span>, <span class="va">sample</span>, <span class="va">date</span>, <span class="va">date_alias</span>, <span class="va">city</span>,</span>
<span>                       n_filtered <span class="op">=</span> <span class="va">raw_concat</span><span class="op">-</span><span class="va">cleaned</span>,</span>
<span>                       n_duplicate <span class="op">=</span> <span class="va">cleaned</span><span class="op">-</span><span class="va">dedup</span>,</span>
<span>                       n_ribosomal <span class="op">=</span> <span class="op">(</span><span class="va">dedup</span><span class="op">-</span><span class="va">ribo_initial</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="va">ribo_initial</span><span class="op">-</span><span class="va">ribo_secondary</span><span class="op">)</span>,</span>
<span>                       n_unassigned <span class="op">=</span> <span class="va">ribo_secondary</span><span class="op">-</span><span class="va">assigned</span>,</span>
<span>                       n_bacterial <span class="op">=</span> <span class="va">bacteria</span>,</span>
<span>                       n_archaeal <span class="op">=</span> <span class="va">archaea</span>,</span>
<span>                       n_viral <span class="op">=</span> <span class="va">viruses</span>,</span>
<span>                       n_human <span class="op">=</span> <span class="va">eukaryota</span><span class="op">)</span></span>
<span><span class="va">read_comp_long</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">read_comp</span>, <span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">city</span><span class="op">)</span>, </span>
<span>                               names_to <span class="op">=</span> <span class="st">"classification"</span>,</span>
<span>                               names_prefix <span class="op">=</span> <span class="st">"n_"</span>, values_to <span class="op">=</span> <span class="st">"n_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu">str_to_sentence</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">city</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">date_alias</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Collection Date"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">city</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data <span class="op">=</span> <span class="va">read_comp_long</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">read_comp_minor</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">7</span>,<span class="fl">9</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">read_comp_minor</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">city</span>, <span class="va">classification</span>, read_fraction<span class="op">=</span><span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">city</span>, <span class="va">classification</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["city"],"name":[1],"type":["chr"],"align":["left"]},{"label":["classification"],"name":[2],"type":["fct"],"align":["left"]},{"label":["read_fraction"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Denver","2":"Excluded","3":"29-90% (mean 68%)"},{"1":"Denver","2":"Ribosomal","3":"0.13-0.92% (mean 0.41%)"},{"1":"Denver","2":"Bacterial","3":"6.8-22% (mean 14%)"},{"1":"Denver","2":"Archaeal","3":"0.0038-0.04% (mean 0.019%)"},{"1":"Denver","2":"Viral","3":"0.004-0.027% (mean 0.011%)"},{"1":"Denver","2":"Human","3":"0.83-63% (mean 17%)"},{"1":"Hong Kong","2":"Excluded","3":"20-85% (mean 47%)"},{"1":"Hong Kong","2":"Ribosomal","3":"0.094-0.85% (mean 0.25%)"},{"1":"Hong Kong","2":"Bacterial","3":"2-72% (mean 23%)"},{"1":"Hong Kong","2":"Archaeal","3":"0.0011-0.17% (mean 0.032%)"},{"1":"Hong Kong","2":"Viral","3":"0.0014-0.042% (mean 0.017%)"},{"1":"Hong Kong","2":"Human","3":"1.1-78% (mean 29%)"},{"1":"London","2":"Excluded","3":"55-78% (mean 65%)"},{"1":"London","2":"Ribosomal","3":"0.24-0.49% (mean 0.32%)"},{"1":"London","2":"Bacterial","3":"15-22% (mean 19%)"},{"1":"London","2":"Archaeal","3":"0.016-0.12% (mean 0.036%)"},{"1":"London","2":"Viral","3":"0.0068-0.019% (mean 0.011%)"},{"1":"London","2":"Human","3":"5.4-28% (mean 15%)"},{"1":"New York City","2":"Excluded","3":"43-73% (mean 63%)"},{"1":"New York City","2":"Ribosomal","3":"0.15-0.36% (mean 0.27%)"},{"1":"New York City","2":"Bacterial","3":"12-27% (mean 19%)"},{"1":"New York City","2":"Archaeal","3":"0.0091-0.081% (mean 0.04%)"},{"1":"New York City","2":"Viral","3":"0.0053-0.025% (mean 0.013%)"},{"1":"New York City","2":"Human","3":"1.8-45% (mean 17%)"},{"1":"Oslo","2":"Excluded","3":"44-92% (mean 71%)"},{"1":"Oslo","2":"Ribosomal","3":"0.081-0.59% (mean 0.31%)"},{"1":"Oslo","2":"Bacterial","3":"5.8-37% (mean 15%)"},{"1":"Oslo","2":"Archaeal","3":"0.00042-0.13% (mean 0.036%)"},{"1":"Oslo","2":"Viral","3":"0.0034-0.036% (mean 0.0092%)"},{"1":"Oslo","2":"Human","3":"0.84-49% (mean 14%)"},{"1":"Stockholm","2":"Excluded","3":"28-78% (mean 65%)"},{"1":"Stockholm","2":"Ribosomal","3":"0.23-0.95% (mean 0.3%)"},{"1":"Stockholm","2":"Bacterial","3":"9.5-70% (mean 15%)"},{"1":"Stockholm","2":"Archaeal","3":"0.0014-0.45% (mean 0.042%)"},{"1":"Stockholm","2":"Viral","3":"0.0073-0.044% (mean 0.014%)"},{"1":"Stockholm","2":"Human","3":"0.94-31% (mean 19%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>In many respects, these resemble the Prussin data: high human fraction (mean 19.6%), high bacterial fraction (mean 18.2%), high unclassified fraction (mean 43.9%), low viral fraction (mean 0.01%). One notable difference is that archaeal reads are more abundant (0.034% compared to 0.016% for Prussin).</p>
<p>As in Prussin, viral DNA reads were dominated by <em>Caudoviricetes</em> phages. Other viral classes that are prominent in at least some samples include <em>Herviviricetes</em> (herpesviruses), <em>Papovaviricetes</em> (polyomaviruses and papillomaviruses), <em>Revtraviricetes</em> (retroviruses + Hep B), and <em>Naldaviricetes</em> (mainly arthropod viruses). I’ll investigate the first three of this latter group in more depth, restricting in each case to samples where that family makes up at least 5% of viral reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get paths to Kraken reports</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">basic_stats_raw</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">report_dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"kraken"</span><span class="op">)</span></span>
<span><span class="va">report_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">report_dirs</span>, <span class="va">list.files</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unlist</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">report_paths</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">str_extract</span><span class="op">(</span><span class="va">report_paths</span>, <span class="st">"SRR\\d*"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract viral taxa</span></span>
<span><span class="va">col_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"pc_reads_total"</span>, <span class="st">"n_reads_clade"</span>, <span class="st">"n_reads_direct"</span>,</span>
<span>               <span class="st">"rank"</span>, <span class="st">"taxid"</span>, <span class="st">"name"</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">report_paths</span>, <span class="va">read_tsv</span>, col_names <span class="op">=</span> <span class="va">col_names</span>,</span>
<span>                             show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">kraken_reports</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kraken_reports_raw</span><span class="op">)</span>, </span>
<span>                         <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">kraken_reports_raw</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>                           <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">kraken_reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Identify major viral classes</span></span>
<span><span class="va">viral_classes_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_classes_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_minor</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_classes_major</span>, <span class="va">viral_classes_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_classes_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_classes</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_classes_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">g_classes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/viral-class-composition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><em>Papovaviricetes</em> are quite heterogeneous across samples, and frequently diverse within samples. <em>Alphapolyomavirus</em> and <em>Alphapapillomavirus</em> are the most abundant genera overall, but <em>Betapapillomavirus, Gammapapillomavirus, Mupapillomavirus</em> and others all have strong showings.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get samples</span></span>
<span><span class="va">papova_taxid</span> <span class="op">&lt;-</span> <span class="fl">2732421</span></span>
<span><span class="va">papova_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">papova_samples</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">papova_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unique</span></span>
<span></span>
<span><span class="co"># Get all taxa in class</span></span>
<span><span class="va">papova_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">papova_taxid</span></span>
<span><span class="va">papova_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">papova_desc_taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">papova_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">papova_desc_taxids_new</span></span>
<span>  <span class="va">papova_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Get read counts</span></span>
<span><span class="va">papova_counts</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_new</span>,</span>
<span>         <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_papova <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genus composition</span></span>
<span><span class="va">papova_genera</span> <span class="op">&lt;-</span> <span class="va">papova_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"G"</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major_tab</span> <span class="op">&lt;-</span> <span class="va">papova_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_papova_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papova_max</span> <span class="op">&gt;=</span> <span class="va">papova_threshold</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major_list</span> <span class="op">&lt;-</span> <span class="va">papova_genera_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major</span> <span class="op">&lt;-</span> <span class="va">papova_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_genera_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_papova</span><span class="op">)</span></span>
<span><span class="va">papova_genera_minor</span> <span class="op">&lt;-</span> <span class="va">papova_genera_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_papova_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_papova <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_papova_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_papova</span><span class="op">)</span></span>
<span><span class="va">papova_genera_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">papova_genera_major</span>, <span class="va">papova_genera_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_genera_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_papova</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_papova_genera</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">papova_genera_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Papovaviricetes Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.02</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_papova_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/papovaviricetes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Only a few samples showed at least 5% prevalence of <em>Herviviricetes</em>, but those that did were typically dominated by one or a small number of species that varied between samples. Of these, human alphaherpesvirus 1 appeared in the most samples, but several other species were prominent in at least one sample:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get samples</span></span>
<span><span class="va">hervi_taxid</span> <span class="op">&lt;-</span> <span class="fl">2731363</span></span>
<span><span class="va">hervi_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">hervi_samples</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">hervi_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unique</span></span>
<span></span>
<span><span class="co"># Get all taxa in class</span></span>
<span><span class="va">hervi_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">hervi_taxid</span></span>
<span><span class="va">hervi_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hervi_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hervi_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">hervi_desc_taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">hervi_desc_taxids_old</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hervi_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">hervi_desc_taxids_new</span></span>
<span>  <span class="va">hervi_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hervi_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hervi_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Get read counts</span></span>
<span><span class="va">hervi_counts</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hervi_desc_taxids_new</span>,</span>
<span>         <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hervi_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hervi <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genus composition</span></span>
<span><span class="va">hervi_genera</span> <span class="op">&lt;-</span> <span class="va">hervi_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"S"</span><span class="op">)</span></span>
<span><span class="va">hervi_genera_major_tab</span> <span class="op">&lt;-</span> <span class="va">hervi_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_hervi_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hervi</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hervi_max</span> <span class="op">&gt;=</span> <span class="va">hervi_threshold</span><span class="op">)</span></span>
<span><span class="va">hervi_genera_major_list</span> <span class="op">&lt;-</span> <span class="va">hervi_genera_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">hervi_genera_major</span> <span class="op">&lt;-</span> <span class="va">hervi_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hervi_genera_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_hervi</span><span class="op">)</span></span>
<span><span class="va">hervi_genera_minor</span> <span class="op">&lt;-</span> <span class="va">hervi_genera_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_hervi_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hervi</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_hervi <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_hervi_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_hervi</span><span class="op">)</span></span>
<span><span class="va">hervi_genera_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hervi_genera_major</span>, <span class="va">hervi_genera_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hervi</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hervi_genera_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hervi</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hervi_genera</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hervi_genera_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Herviviricetes Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hervi_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/herviviricetes-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>Finally, for <em>Revtraviricetes</em>, most samples were dominated by porcine type-C oncovirus, while one was dominated by an avian retrovirus. The last showed significant levels of two murine viruses plus HIV. I’m suspicious of many of these.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get samples</span></span>
<span><span class="va">revtra_taxid</span> <span class="op">&lt;-</span> <span class="fl">2732514</span></span>
<span><span class="va">revtra_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">revtra_samples</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">revtra_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unique</span></span>
<span></span>
<span><span class="co"># Get all taxa in class</span></span>
<span><span class="va">revtra_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">revtra_taxid</span></span>
<span><span class="va">revtra_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">revtra_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">revtra_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">revtra_desc_taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">revtra_desc_taxids_old</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">revtra_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">revtra_desc_taxids_new</span></span>
<span>  <span class="va">revtra_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">revtra_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">revtra_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Get read counts</span></span>
<span><span class="va">revtra_counts</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">revtra_desc_taxids_new</span>,</span>
<span>         <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">revtra_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_revtra <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genus composition</span></span>
<span><span class="va">revtra_genera</span> <span class="op">&lt;-</span> <span class="va">revtra_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"S"</span><span class="op">)</span></span>
<span><span class="va">revtra_genera_major_tab</span> <span class="op">&lt;-</span> <span class="va">revtra_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_revtra_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_revtra</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_revtra_max</span> <span class="op">&gt;=</span> <span class="va">revtra_threshold</span><span class="op">)</span></span>
<span><span class="va">revtra_genera_major_list</span> <span class="op">&lt;-</span> <span class="va">revtra_genera_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">revtra_genera_major</span> <span class="op">&lt;-</span> <span class="va">revtra_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">revtra_genera_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_revtra</span><span class="op">)</span></span>
<span><span class="va">revtra_genera_minor</span> <span class="op">&lt;-</span> <span class="va">revtra_genera_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_revtra_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_revtra</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_revtra <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_revtra_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">p_reads_revtra</span><span class="op">)</span></span>
<span><span class="va">revtra_genera_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">revtra_genera_major</span>, <span class="va">revtra_genera_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_revtra</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">revtra_genera_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_revtra</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_revtra_genera</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">revtra_genera_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% revtraviricetes Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_revtra_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/revtraviricetes-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples. Using the same workflow I used for Prussin et al, I identified 24,278 read pairs as putatively human viral: 0.002% of reads surviving to that stage in the pipeline.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import HV read data</span></span>
<span><span class="va">hv_reads_filtered_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">hv_reads_filtered_paths</span>, <span class="va">read_tsv</span>,</span>
<span>                            show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collapse multi-entry sequences</span></span>
<span><span class="va">rmax</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">max</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">collapse</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">==</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">x</span>, collapse<span class="op">=</span><span class="st">"/"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>            genome_id <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span>,</span>
<span>            taxid_best <span class="op">=</span> <span class="va">taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            best_alignment_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_fwd</span><span class="op">)</span>,</span>
<span>            best_alignment_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_rev</span><span class="op">)</span>,</span>
<span>            query_len_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_fwd</span><span class="op">)</span>,</span>
<span>            query_len_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_rev</span><span class="op">)</span>,</span>
<span>            query_seq_fwd <span class="op">=</span> <span class="va">query_seq_fwd</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_fwd</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            query_seq_rev <span class="op">=</span> <span class="va">query_seq_rev</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_rev</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            classified <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">classified</span><span class="op">)</span>,</span>
<span>            assigned_name <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">assigned_name</span><span class="op">)</span>,</span>
<span>            assigned_taxid_best <span class="op">=</span> <span class="va">assigned_taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            assigned_taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">assigned_taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            assigned_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">assigned_hv</span><span class="op">)</span>,</span>
<span>            hit_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">hit_hv</span><span class="op">)</span>,</span>
<span>            encoded_hits <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">encoded_hits</span><span class="op">)</span>,</span>
<span>            adj_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>,</span>
<span>            adj_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span></span>
<span>            <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_hist_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> </span>
<span><span class="va">g_hist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-hv-scores-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I’ve used for previous datasets.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_fasta</span> <span class="op">&lt;-</span>  <span class="va">mrg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">seq_id</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, </span>
<span>         header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_fasta_sep</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">mrg_fasta</span>, header<span class="op">=</span><span class="va">header1</span>, seq<span class="op">=</span><span class="va">seq1</span><span class="op">)</span>,</span>
<span>                           <span class="fu">select</span><span class="op">(</span><span class="va">mrg_fasta</span>, header<span class="op">=</span><span class="va">header2</span>, seq<span class="op">=</span><span class="va">seq2</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">seq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_fasta_sep</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="va">blast_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"blast"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="va">blast_dir</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">blast_dir</span>, <span class="st">"putative-viral.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import BLAST results</span></span>
<span><span class="co"># blast_results_path &lt;- file.path(data_dir_base, "blast/putative-viral.blast.gz")</span></span>
<span><span class="co"># blast_cols &lt;- c("qseqid", "sseqid", "sgi", "staxid", "qlen", "evalue", "bitscore", "qcovs", "length", "pident", "mismatch", "gapopen", "sstrand", "qstart", "qend", "sstart", "send")</span></span>
<span><span class="co"># blast_results &lt;- read_tsv(blast_results_path, show_col_types = FALSE,</span></span>
<span><span class="co">#                           col_names = blast_cols, col_types = cols(.default="c"))</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"blast/putative-viral-best.blast.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject combination</span></span>
<span><span class="va">blast_results_best</span> <span class="op">&lt;-</span> <span class="va">blast_results</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co"># write_tsv(blast_results_best, file.path(data_dir_base, "blast/putative-viral-best.blast.gz"))</span></span>
<span></span>
<span><span class="co"># Rank hits for each query and filter for high-ranking hits</span></span>
<span><span class="va">blast_results_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, </span>
<span>         seq_id <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_results_paired</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">match_taxid</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxid_1</span>, <span class="va">taxid_2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"/"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"/"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">p1</span><span class="op">|</span><span class="va">p2</span><span class="op">|</span><span class="va">p3</span>, <span class="cn">NULL</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">adj_score_max</span><span class="op">)</span></span>
<span><span class="va">blast_results_assign</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">blast_results_viral</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg</span>, <span class="va">blast_results_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot RNA</span></span>
<span><span class="va">g_hist_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_blast</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span>, fill<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_hist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-blast-results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>For a disjunctive score threshold of 20, the workflow achieves a measured F1 score of 98.0%.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>retain_score <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>           retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span><span class="op">)</span></span>
<span>  <span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="va">threshold</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adjusted Score Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-f1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span>Threshold<span class="op">=</span><span class="va">threshold</span>, Metric<span class="op">=</span><span class="va">metric</span>, Value<span class="op">=</span><span class="va">value</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Threshold"],"name":[1],"type":["int"],"align":["right"]},{"label":["Metric"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"20","2":"sensitivity","3":"0.9681598"},{"1":"20","2":"specificity","3":"0.6560636"},{"1":"20","2":"precision","3":"0.9925402"},{"1":"20","2":"f1","3":"0.9801984"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Looking into the composition of different read groups, the notable observation for me is the high prevalence of Pigeon torque teno virus among high-scoring false positives, with 77 such read pairs. BLAST maps these not to viruses but to their most common hosts, i.e.&nbsp;assorted species of pigeon. That said, the number of false positive PTTV reads is substantially exceeded by the number of true-positive PTTV reads (1883), which do map to appropriate viruses according to BLAST, so the presence of a comparatively small number of false positives seems unlikely to cause too much distortion.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.04</span></span>
<span></span>
<span><span class="co"># Add missing viral taxa</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">211787</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">509154</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">493803</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">427343</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">340907</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194959</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span></span>
<span></span>
<span><span class="co"># Prepare data</span></span>
<span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_major</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_fp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/fp-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxid_ptt</span> <span class="op">&lt;-</span> <span class="fl">2233536</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span>                         <span class="fl">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span>                         <span class="fl">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span>                         <span class="fl">187126</span>, <span class="st">"Nesoenas mayeri"</span></span>
<span>                         <span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="va">tax_names_new</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tax_names</span><span class="op">$</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span></span>
<span><span class="va">ref_name_ptt</span> <span class="op">&lt;-</span> <span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">ref_taxid_ptt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get major matches</span></span>
<span><span class="va">mrg_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid_best</span> <span class="op">==</span> <span class="va">ref_taxid_ptt</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">highscore</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid_best<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid_best"</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count</span> <span class="op">&lt;-</span> <span class="va">fp_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, </span>
<span>           <span class="va">taxid_best</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count_major</span> <span class="op">&lt;-</span> <span class="va">fp_staxid_count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, </span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_staxid_count_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">status_display</span><span class="op">~</span><span class="va">score_display</span>, scales<span class="op">=</span><span class="st">"free"</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_ptt</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_ptt</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/ptt-blast-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-overall-relative-abundance" class="level1"><h1>Human-infecting viruses: overall relative abundance</h1>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid_all <span class="op">=</span> <span class="va">taxid</span>, taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_city</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span> <span class="st">"All samples"</span>, date_alias <span class="op">=</span> <span class="st">"All dates"</span><span class="op">)</span></span>
<span><span class="va">read_counts_total</span> <span class="op">&lt;-</span> <span class="va">read_counts_city</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>city <span class="op">=</span> <span class="st">"All cities"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts_city</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_total</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         city <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">city</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Applying a disjunctive cutoff at S=20 identifies 23,191 read pairs as human-viral. This gives an overall relative HV abundance of <span class="math inline">\(1.73 \times 10^{-5}\)</span>.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">city</span>, color<span class="op">=</span><span class="va">city</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Collection Date"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co">#facet_grid(.~sample_type, scales = "free", space = "free_x") +</span></span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_rotate</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/plot-hv-ra-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is lower than for DNA reads from other air-sampling datasets I’ve analyzed, but not drastically so:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collate past RA values</span></span>
<span><span class="va">ra_past</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Collate new RA values</span></span>
<span><span class="va">ra_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                  <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">scale_color_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                 name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">ra_comp</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">ra_past</span>, <span class="va">ra_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">ra_comp</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">dataset</span>, x<span class="op">=</span><span class="va">ra</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/ra-hv-past-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-taxonomy-and-composition" class="level1"><h1>Human-infecting viruses: taxonomy and composition</h1>
<p>At the family level, most samples across all cities are dominated by <em>Papillomaviridae</em>, <em>Herpesviridae</em>, <em>Anelloviridae</em>, <em>Polyomaviridae, and to a lesser extent</em> <em>Poxviridae</em>:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">585893</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333922</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334207</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">369960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333924</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687329</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">325455</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333750</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334202</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">197911</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">186938</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333926</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337051</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337043</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334204</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334208</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333928</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337039</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333929</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337042</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334203</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333757</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337050</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333767</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333754</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687363</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687342</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687359</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194441</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334209</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span></span>
<span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_family</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"family"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads.</p>
<p>As usual, <em>Papillomaviridae</em> reads are divided among many different viral species. In this case, Betapapillomavirus 1 and 2 are the most prevalent across samples, but many other alpha-, beta-, gamma-, and mupapillomaviruses are highly prevalent in at least some samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.4</span></span>
<span><span class="va">taxid_papilloma</span> <span class="op">&lt;-</span> <span class="fl">151340</span></span>
<span></span>
<span><span class="co"># Get set of Papillomaviridae reads</span></span>
<span><span class="va">papilloma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">papilloma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each Papillomaviridae species</span></span>
<span><span class="va">papilloma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_papilloma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">papilloma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_papilloma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_papilloma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">papilloma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Papillomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Papillomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_papilloma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-species-papilloma-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">papilloma_species_collate</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In terms of total reads across samples, herpesviruses are dominated by Epstein-Barr virus (Human gammaherpesvirus 4), HSV-1 (Human alphaherpesvirus 1), and human cytomegalovirus (Human betaherpesvirus 5). However, numerous other herpesviruses are also present.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.4</span></span>
<span><span class="va">taxid_herpes</span> <span class="op">&lt;-</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"Herpesviridae"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get set of herpesviridae reads</span></span>
<span><span class="va">herpes_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">herpes_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each herpesviridae species</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-species-herpes-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In sharp contrast to the above, my pipeline classifies the great majority of anellovirus reads in all samples into a single species, torque teno virus. Looking online, it looks like there are a lot of “torque teno viruses” within <em>Anelloviridae</em> – for example, Wikipedia says that the genus <em>Alphatorquevirus</em> contains &gt;20 numbered torque teno viruses – so I’m not sure exactly which virus this refers to.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_anello</span> <span class="op">&lt;-</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"Anelloviridae"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get set of anelloviridae reads</span></span>
<span><span class="va">anello_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_anello</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">anello_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_anello</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">anello_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each anelloviridae species</span></span>
<span><span class="va">anello_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">anello_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_anello <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">anello_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">anello_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_anello</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_anello</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_anello</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_anello</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">anello_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">anello_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">anello_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_anello <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_anello <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_anello</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">anello_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">anello_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">anello_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_anello</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_anello_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">anello_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Anelloviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Anelloviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_anello_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-species-anello-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">anello_species_collate</span> <span class="op">&lt;-</span> <span class="va">anello_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_anello</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Polyomaviruses are intermediate; most viruses are dominated by a single species, <em>Alphapolyomavirus quintihominis</em>, but several other viruses in the family are also present.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_polyoma</span> <span class="op">&lt;-</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"Polyomaviridae"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get set of polyomaviridae reads</span></span>
<span><span class="co"># Get set of polyomaviridae reads</span></span>
<span><span class="va">polyoma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">polyoma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each polyomaviridae species</span></span>
<span><span class="va">polyoma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_polyoma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">polyoma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_polyoma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_polyoma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">polyoma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Polyomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Polyomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_polyoma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-species-polyoma-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">polyoma_species_collate</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Finally, poxvirus reads in most samples are dominated by molluscum contagiosum virus (which I expect to be real), followed by Orf virus (which I expect to be fake). These expectations are borne out by BLAST alignments (below).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_pox</span> <span class="op">&lt;-</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"Poxviridae"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get set of poxviridae reads</span></span>
<span><span class="co"># Get set of poxviridae reads</span></span>
<span><span class="va">pox_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">pox_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each poxviridae species</span></span>
<span><span class="va">pox_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_pox <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">pox_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_pox</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_pox_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">pox_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Poxviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Poxviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_pox_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-species-pox-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">pox_species_collate</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxids_hv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10279</span>, <span class="fl">10258</span><span class="op">)</span></span>
<span><span class="va">ref_names_hv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ref_taxids_hv</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">first</span><span class="op">)</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">257877</span>, <span class="st">"Macaca thibetana thibetana"</span>,</span>
<span>                         <span class="fl">256321</span>, <span class="st">"Lentiviral transfer vector pHsCXW"</span>,</span>
<span>                         <span class="fl">419242</span>, <span class="st">"Shuttle vector pLvCmvMYOCDHA"</span>,</span>
<span>                         <span class="fl">419243</span>, <span class="st">"Shuttle vector pLvCmvLacZ"</span>,</span>
<span>                         <span class="fl">421868</span>, <span class="st">"Cloning vector pLvCmvLacZ.Gfp"</span>,</span>
<span>                         <span class="fl">421869</span>, <span class="st">"Cloning vector pLvCmvMyocardin.Gfp"</span>,</span>
<span>                         <span class="fl">426303</span>, <span class="st">"Lentiviral vector pNL-GFP-RRE(SA)"</span>,</span>
<span>                         <span class="fl">436015</span>, <span class="st">"Lentiviral transfer vector pFTMGW"</span>,</span>
<span>                         <span class="fl">454257</span>, <span class="st">"Shuttle vector pLvCmvMYOCD2aHA"</span>,</span>
<span>                         <span class="fl">476184</span>, <span class="st">"Shuttle vector pLV.mMyoD::ERT2.eGFP"</span>,</span>
<span>                         <span class="fl">476185</span>, <span class="st">"Shuttle vector pLV.hMyoD.eGFP"</span>,</span>
<span>                         <span class="fl">591936</span>, <span class="st">"Piliocolobus tephrosceles"</span>,</span>
<span>                         <span class="fl">627481</span>, <span class="st">"Lentiviral transfer vector pFTM3GW"</span>,</span>
<span>                         <span class="fl">680261</span>, <span class="st">"Self-inactivating lentivirus vector pLV.C-EF1a.cyt-bGal.dCpG"</span>,</span>
<span>                         <span class="fl">2952778</span>, <span class="st">"Expression vector pLV[Exp]-EGFP:T2A:Puro-EF1A"</span>,</span>
<span>                         <span class="fl">3022699</span>, <span class="st">"Vector PAS_122122"</span>,</span>
<span>                         <span class="fl">3025913</span>, <span class="st">"Vector pSIN-WP-mPGK-GDNF"</span>,</span>
<span>                         <span class="fl">3105863</span>, <span class="st">"Vector pLKO.1-ZsGreen1"</span>,</span>
<span>                         <span class="fl">3105864</span>, <span class="st">"Vector pLKO.1-ZsGreen1 mouse Wfs1 shRNA"</span>,</span>
<span>                         <span class="fl">3108001</span>, <span class="st">"Cloning vector pLVSIN-CMV_Neo_v4.0"</span>,</span>
<span>                         <span class="fl">3109234</span>, <span class="st">"Vector pTwist+Kan+High"</span>,</span>
<span>                         <span class="fl">3117662</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L"</span>,</span>
<span>                         <span class="fl">3117663</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L:T2A:mRuby3"</span>,</span>
<span>                         <span class="fl">3117664</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;hMAPT[NM_005910.6](ns):T2A:mRuby3"</span>,</span>
<span>                         <span class="fl">3117665</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3"</span>,</span>
<span>                         <span class="fl">3117666</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3/NFAT3 fusion protein"</span>,</span>
<span>                         <span class="fl">3117667</span>, <span class="st">"Cloning vector pLV[Exp]-Neo-mPGK&gt;{EGFP-hSEPT6}"</span>,</span>
<span>                         <span class="fl">438045</span>, <span class="st">"Xenotropic MuLV-related virus"</span>,</span>
<span>                         <span class="fl">447135</span>, <span class="st">"Myodes glareolus"</span>,</span>
<span>                         <span class="fl">590745</span>, <span class="st">"Mus musculus mobilized endogenous polytropic provirus"</span>,</span>
<span>                         <span class="fl">181858</span>, <span class="st">"Murine AIDS virus-related provirus"</span>,</span>
<span>                         <span class="fl">356663</span>, <span class="st">"Xenotropic MuLV-related virus VP35"</span>,</span>
<span>                         <span class="fl">356664</span>, <span class="st">"Xenotropic MuLV-related virus VP42"</span>,</span>
<span>                         <span class="fl">373193</span>, <span class="st">"Xenotropic MuLV-related virus VP62"</span>,</span>
<span>                         <span class="fl">286419</span>, <span class="st">"Canis lupus dingo"</span>,</span>
<span>                         <span class="fl">415978</span>, <span class="st">"Sus scrofa scrofa"</span>,</span>
<span>                         <span class="fl">494514</span>, <span class="st">"Vulpes lagopus"</span>,</span>
<span>                         <span class="fl">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span>                         <span class="fl">3119969</span>, <span class="st">"Bubalus kerabau"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span>, <span class="va">tax_names_new</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get matches</span></span>
<span><span class="va">hv_blast_staxids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">ref_taxids_hv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count matches</span></span>
<span><span class="va">hv_blast_counts</span> <span class="op">&lt;-</span> <span class="va">hv_blast_staxids</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset to major matches</span></span>
<span><span class="va">hv_blast_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_blast_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">25</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="va">ref_names_hv</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"MCV"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_blast</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_blast_counts_major</span>, mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">name_display</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span>, space<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, </span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_blast</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/hv-blast-hits-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define reference genera</span></span>
<span><span class="va">path_genera_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_rna</span>, genome_type<span class="op">=</span><span class="st">"RNA genome"</span><span class="op">)</span>,</span>
<span>                         <span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_dna</span>, genome_type<span class="op">=</span><span class="st">"DNA genome"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count in each sample</span></span>
<span><span class="va">n_path_genera</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date_alias</span>, <span class="va">city</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"date_alias"</span>, <span class="st">"city"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot out and back to add zero lines</span></span>
<span><span class="va">n_path_genera_out</span> <span class="op">&lt;-</span> <span class="va">n_path_genera</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">n_reads_viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"name"</span>, values_from<span class="op">=</span><span class="st">"n_reads_viral"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, names_to<span class="op">=</span><span class="st">"name"</span>, values_to<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Aggregate across dates</span></span>
<span><span class="va">n_path_genera_stype</span> <span class="op">&lt;-</span> <span class="va">n_path_genera_out</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">genome_type</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="st">"All samples"</span>, date<span class="op">=</span><span class="st">"All dates"</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         na_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_path_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_path_genera_stype</span>,</span>
<span>                        <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">name</span>, x<span class="op">=</span><span class="va">p_reads_viral</span>, color<span class="op">=</span><span class="va">city</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_city</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">genome_type</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_path_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-19_leung_files/figure-html/ra-genera-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>This is the third, largest, and final of this tranche of air-sampling datasets that I’ve run through this pipeline. Many of the high-level findings were similar to Prussin and Rosario, including high relative abundance of human reads, low total viral reads, an absence of enteric viruses, and high abundance of papillomaviruses among human-infecting viruses.</p>
<p>In the future, I’ll do a more in-depth comparative analysis across different datasets to compare the abundance of different viruses. For now, though, there are some major updates to the pipeline I want to make before I do any more public analyses.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb51" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Leung et al. (2021)"</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Air sampling from urban public transit systems."</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-04-19</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a>theme_rotate <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb51-29"><a href="#cb51-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb51-30"><a href="#cb51-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb51-31"><a href="#cb51-31" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb51-32"><a href="#cb51-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-33"><a href="#cb51-33" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb51-34"><a href="#cb51-34" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb51-35"><a href="#cb51-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-36"><a href="#cb51-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-37"><a href="#cb51-37" aria-hidden="true" tabindex="-1"></a>The last in our current run of air sampling datasets is <span class="co">[</span><span class="ot">Leung et al.</span><span class="co">](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01044-7)</span> (2021), a study of active air samples collected in public transit systems from six cities (Denver, Hong Kong, London, NYC, Oslo, Stockholm) from June to September 2017.</span>
<span id="cb51-38"><a href="#cb51-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-39"><a href="#cb51-39" aria-hidden="true" tabindex="-1"></a>Samples from Denver originated from their rail and bus system; all other samples originated from metro systems. Collection took place during working days and working hours. Air samples were collected with the SASS 3100 Dry Air Samplers (filtration) for 30 min at a flowrate of 300 L/min using electret microfibrous filters. Filters were stationed at 1.5m above floor level, facing downward (to avoid direct deposition).</span>
<span id="cb51-40"><a href="#cb51-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-41"><a href="#cb51-41" aria-hidden="true" tabindex="-1"></a>This was a DNA-sequencing study, focused on the bacterial microbiome and resistome. Sample processing followed an ideosyncratic protocol, where samples were pelleted and the pellet and supernatant were processed separately before being recombined for NA extraction and sequencing; I don't have a great understanding of how this is expected to affect the viral fraction. Samples were sequenced with Illumina HiSeqX 2x150bp.</span>
<span id="cb51-42"><a href="#cb51-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-43"><a href="#cb51-43" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb51-44"><a href="#cb51-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-45"><a href="#cb51-45" aria-hidden="true" tabindex="-1"></a>In total, the Leung dataset comprised 293 samples:</span>
<span id="cb51-46"><a href="#cb51-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-49"><a href="#cb51-49" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-50"><a href="#cb51-50" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-51"><a href="#cb51-51" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: import-qc-data</span></span>
<span id="cb51-52"><a href="#cb51-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-53"><a href="#cb51-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Importing the data is a bit more complicated this time as the samples are split across three pipeline runs</span></span>
<span id="cb51-54"><a href="#cb51-54" aria-hidden="true" tabindex="-1"></a>data_dir_base <span class="ot">&lt;-</span> <span class="st">"../data/2024-04-12_leung"</span></span>
<span id="cb51-55"><a href="#cb51-55" aria-hidden="true" tabindex="-1"></a>data_dirs <span class="ot">&lt;-</span> <span class="fu">paste</span>(data_dir_base, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">sep=</span><span class="st">"/"</span>)</span>
<span id="cb51-56"><a href="#cb51-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-57"><a href="#cb51-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Define geo relationships for filling in</span></span>
<span id="cb51-58"><a href="#cb51-58" aria-hidden="true" tabindex="-1"></a>geo <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>region, <span class="sc">~</span>country, <span class="sc">~</span>city,</span>
<span id="cb51-59"><a href="#cb51-59" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Asia"</span>, <span class="st">"Hong Kong"</span>, <span class="st">"Hong Kong"</span>,</span>
<span id="cb51-60"><a href="#cb51-60" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Europe"</span>, <span class="st">"Norway"</span>, <span class="st">"Oslo"</span>,</span>
<span id="cb51-61"><a href="#cb51-61" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Europe"</span>, <span class="st">"Sweden"</span>, <span class="st">"Stockholm"</span>,</span>
<span id="cb51-62"><a href="#cb51-62" aria-hidden="true" tabindex="-1"></a>               <span class="st">"Europe"</span>, <span class="st">"United Kingdom"</span>, <span class="st">"London"</span>,</span>
<span id="cb51-63"><a href="#cb51-63" aria-hidden="true" tabindex="-1"></a>               <span class="st">"North America"</span>, <span class="st">"USA"</span>, <span class="st">"New York City"</span>,</span>
<span id="cb51-64"><a href="#cb51-64" aria-hidden="true" tabindex="-1"></a>               <span class="st">"North America"</span>, <span class="st">"USA"</span>, <span class="st">"Denver"</span>)</span>
<span id="cb51-65"><a href="#cb51-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-66"><a href="#cb51-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb51-67"><a href="#cb51-67" aria-hidden="true" tabindex="-1"></a>libraries_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb51-68"><a href="#cb51-68" aria-hidden="true" tabindex="-1"></a>basic_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb51-69"><a href="#cb51-69" aria-hidden="true" tabindex="-1"></a>adapter_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb51-70"><a href="#cb51-70" aria-hidden="true" tabindex="-1"></a>quality_base_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb51-71"><a href="#cb51-71" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb51-72"><a href="#cb51-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-73"><a href="#cb51-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb51-74"><a href="#cb51-74" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(libraries_paths, read_csv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-75"><a href="#cb51-75" aria-hidden="true" tabindex="-1"></a>  bind_rows</span>
<span id="cb51-76"><a href="#cb51-76" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries_raw <span class="sc">%&gt;%</span></span>
<span id="cb51-77"><a href="#cb51-77" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Fix missing entries</span></span>
<span id="cb51-78"><a href="#cb51-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">city =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(city), <span class="fu">sub</span>(<span class="st">", .*"</span>, <span class="st">""</span>, location), city)) <span class="sc">%&gt;%</span></span>
<span id="cb51-79"><a href="#cb51-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(geo, <span class="at">by=</span><span class="st">"city"</span>, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="st">"_new"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-80"><a href="#cb51-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">region =</span> <span class="fu">ifelse</span>(region <span class="sc">==</span> <span class="st">"uncalculated"</span>, region_new, region),</span>
<span id="cb51-81"><a href="#cb51-81" aria-hidden="true" tabindex="-1"></a>         <span class="at">country =</span> <span class="fu">ifelse</span>(country <span class="sc">==</span> <span class="st">"uncalculated"</span>, country_new, country)) <span class="sc">%&gt;%</span></span>
<span id="cb51-82"><a href="#cb51-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>country_new, <span class="sc">-</span>region_new) <span class="sc">%&gt;%</span></span>
<span id="cb51-83"><a href="#cb51-83" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add sample aliases</span></span>
<span id="cb51-84"><a href="#cb51-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(city, date, location) <span class="sc">%&gt;%</span></span>
<span id="cb51-85"><a href="#cb51-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(city, date) <span class="sc">%&gt;%</span></span>
<span id="cb51-86"><a href="#cb51-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_count =</span> <span class="fu">row_number</span>(),</span>
<span id="cb51-87"><a href="#cb51-87" aria-hidden="true" tabindex="-1"></a>         <span class="at">date_alias =</span> <span class="fu">paste</span>(<span class="fu">as.character</span>(date), sample_count, <span class="at">sep=</span><span class="st">"_"</span>),</span>
<span id="cb51-88"><a href="#cb51-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample_alias =</span> <span class="fu">paste</span>(city, date_alias, <span class="at">sep=</span><span class="st">"_"</span>))</span>
<span id="cb51-89"><a href="#cb51-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-90"><a href="#cb51-90" aria-hidden="true" tabindex="-1"></a>count_city <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(region, country, city) <span class="sc">%&gt;%</span> </span>
<span id="cb51-91"><a href="#cb51-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_samples"</span>)</span>
<span id="cb51-92"><a href="#cb51-92" aria-hidden="true" tabindex="-1"></a>count_city</span>
<span id="cb51-93"><a href="#cb51-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-94"><a href="#cb51-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-97"><a href="#cb51-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-98"><a href="#cb51-98" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-qc-data</span></span>
<span id="cb51-99"><a href="#cb51-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-100"><a href="#cb51-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb51-101"><a href="#cb51-101" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb51-102"><a href="#cb51-102" aria-hidden="true" tabindex="-1"></a>import_basic <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb51-103"><a href="#cb51-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb51-104"><a href="#cb51-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb51-105"><a href="#cb51-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb51-106"><a href="#cb51-106" aria-hidden="true" tabindex="-1"></a>           <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb51-107"><a href="#cb51-107" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-108"><a href="#cb51-108" aria-hidden="true" tabindex="-1"></a>import_basic_paired <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb51-109"><a href="#cb51-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">import_basic</span>(paths) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(read_pair) <span class="sc">%&gt;%</span> </span>
<span id="cb51-110"><a href="#cb51-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb51-111"><a href="#cb51-111" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-112"><a href="#cb51-112" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">import_basic</span>(basic_stats_paths)</span>
<span id="cb51-113"><a href="#cb51-113" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(adapter_stats_paths)</span>
<span id="cb51-114"><a href="#cb51-114" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_base_stats_paths)</span>
<span id="cb51-115"><a href="#cb51-115" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_seq_stats_paths)</span>
<span id="cb51-116"><a href="#cb51-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-117"><a href="#cb51-117" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb51-118"><a href="#cb51-118" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb51-119"><a href="#cb51-119" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb51-120"><a href="#cb51-120" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb51-121"><a href="#cb51-121" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb51-122"><a href="#cb51-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-123"><a href="#cb51-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Get key values for readout</span></span>
<span id="cb51-124"><a href="#cb51-124" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb51-125"><a href="#cb51-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb51-126"><a href="#cb51-126" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb51-127"><a href="#cb51-127" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb51-128"><a href="#cb51-128" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb51-129"><a href="#cb51-129" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb51-130"><a href="#cb51-130" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb51-131"><a href="#cb51-131" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-132"><a href="#cb51-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-133"><a href="#cb51-133" aria-hidden="true" tabindex="-1"></a>These 293 samples yielded 0.39M-7.86M (mean 4.57M) reads per sample, for a total of 1.34B read pairs (402 gigabases of sequence). Read qualities were high at the 5' end but dropped off significantly in some samples, in definite need of cleaning. Adapter levels were high. With the exception of a couple of early samples, inferred duplication levels were low (mean 9.4%).</span>
<span id="cb51-134"><a href="#cb51-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-137"><a href="#cb51-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-138"><a href="#cb51-138" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-139"><a href="#cb51-139" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-140"><a href="#cb51-140" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb51-141"><a href="#cb51-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-142"><a href="#cb51-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb51-143"><a href="#cb51-143" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb51-144"><a href="#cb51-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, city, date,</span>
<span id="cb51-145"><a href="#cb51-145" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb51-146"><a href="#cb51-146" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb51-147"><a href="#cb51-147" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb51-148"><a href="#cb51-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(sample<span class="sc">:</span>date), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-149"><a href="#cb51-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric))</span>
<span id="cb51-150"><a href="#cb51-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-151"><a href="#cb51-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb51-152"><a href="#cb51-152" aria-hidden="true" tabindex="-1"></a>scale_fill_city <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb51-153"><a href="#cb51-153" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">name=</span><span class="st">"City"</span>)</span>
<span id="cb51-154"><a href="#cb51-154" aria-hidden="true" tabindex="-1"></a>scale_x_cdate <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_x_date, <span class="at">name=</span><span class="st">"Collection Date"</span>,</span>
<span id="cb51-155"><a href="#cb51-155" aria-hidden="true" tabindex="-1"></a>                                <span class="at">date_breaks =</span> <span class="st">"1 month"</span>, <span class="at">date_labels =</span> <span class="st">"%Y-%m-%d"</span>)</span>
<span id="cb51-156"><a href="#cb51-156" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, </span>
<span id="cb51-157"><a href="#cb51-157" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>date, <span class="at">y=</span>value, <span class="at">fill=</span>city, <span class="at">group=</span><span class="fu">interaction</span>(city,sample))) <span class="sc">+</span></span>
<span id="cb51-158"><a href="#cb51-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb51-159"><a href="#cb51-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_cdate</span>() <span class="sc">+</span></span>
<span id="cb51-160"><a href="#cb51-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-161"><a href="#cb51-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb51-162"><a href="#cb51-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_city</span>() <span class="sc">+</span> </span>
<span id="cb51-163"><a href="#cb51-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space=</span><span class="st">"free_x"</span>, <span class="at">switch=</span><span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb51-164"><a href="#cb51-164" aria-hidden="true" tabindex="-1"></a>  theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb51-165"><a href="#cb51-165" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-166"><a href="#cb51-166" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"plain"</span>)</span>
<span id="cb51-167"><a href="#cb51-167" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb51-168"><a href="#cb51-168" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb51-169"><a href="#cb51-169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-170"><a href="#cb51-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-173"><a href="#cb51-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-174"><a href="#cb51-174" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb51-175"><a href="#cb51-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-176"><a href="#cb51-176" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb51-177"><a href="#cb51-177" aria-hidden="true" tabindex="-1"></a>scale_color_city <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb51-178"><a href="#cb51-178" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">name=</span><span class="st">"City"</span>)</span>
<span id="cb51-179"><a href="#cb51-179" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>city, <span class="at">linetype=</span>read_pair, </span>
<span id="cb51-180"><a href="#cb51-180" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb51-181"><a href="#cb51-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb51-182"><a href="#cb51-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb51-183"><a href="#cb51-183" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb51-184"><a href="#cb51-184" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb51-185"><a href="#cb51-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-186"><a href="#cb51-186" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb51-187"><a href="#cb51-187" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb51-188"><a href="#cb51-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb51-189"><a href="#cb51-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb51-190"><a href="#cb51-190" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-191"><a href="#cb51-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb51-192"><a href="#cb51-192" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-193"><a href="#cb51-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb51-194"><a href="#cb51-194" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb51-195"><a href="#cb51-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-196"><a href="#cb51-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb51-197"><a href="#cb51-197" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb51-198"><a href="#cb51-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-199"><a href="#cb51-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-200"><a href="#cb51-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb51-201"><a href="#cb51-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb51-202"><a href="#cb51-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb51-203"><a href="#cb51-203" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb51-204"><a href="#cb51-204" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb51-205"><a href="#cb51-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-206"><a href="#cb51-206" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb51-207"><a href="#cb51-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-208"><a href="#cb51-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-209"><a href="#cb51-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb51-210"><a href="#cb51-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-211"><a href="#cb51-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb51-212"><a href="#cb51-212" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb51-213"><a href="#cb51-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-214"><a href="#cb51-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-215"><a href="#cb51-215" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb51-216"><a href="#cb51-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-217"><a href="#cb51-217" aria-hidden="true" tabindex="-1"></a>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Read loss during cleaning was highly variable but averaged 11%, with a further \~7% lost during deduplication and \~0.3% during ribodepletion.</span>
<span id="cb51-218"><a href="#cb51-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-221"><a href="#cb51-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-222"><a href="#cb51-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb51-223"><a href="#cb51-223" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb51-224"><a href="#cb51-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, date, city, stage, </span>
<span id="cb51-225"><a href="#cb51-225" aria-hidden="true" tabindex="-1"></a>         percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb51-226"><a href="#cb51-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb51-227"><a href="#cb51-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> <span class="fu">replace_na</span>(n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs), <span class="dv">0</span>),</span>
<span id="cb51-228"><a href="#cb51-228" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb51-229"><a href="#cb51-229" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb51-230"><a href="#cb51-230" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb51-231"><a href="#cb51-231" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> <span class="fu">replace_na</span>(p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs), <span class="dv">0</span>))</span>
<span id="cb51-232"><a href="#cb51-232" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb51-233"><a href="#cb51-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage, <span class="at">City=</span>city) <span class="sc">%&gt;%</span> </span>
<span id="cb51-234"><a href="#cb51-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb51-235"><a href="#cb51-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb51-236"><a href="#cb51-236" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-237"><a href="#cb51-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-238"><a href="#cb51-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb51-239"><a href="#cb51-239" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb51-240"><a href="#cb51-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-241"><a href="#cb51-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-244"><a href="#cb51-244" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-245"><a href="#cb51-245" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb51-246"><a href="#cb51-246" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-247"><a href="#cb51-247" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-248"><a href="#cb51-248" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb51-249"><a href="#cb51-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-250"><a href="#cb51-250" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>city, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb51-251"><a href="#cb51-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span></span>
<span id="cb51-252"><a href="#cb51-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>city, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb51-253"><a href="#cb51-253" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb51-254"><a href="#cb51-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-255"><a href="#cb51-255" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb51-256"><a href="#cb51-256" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb51-257"><a href="#cb51-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb51-258"><a href="#cb51-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb51-259"><a href="#cb51-259" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb51-260"><a href="#cb51-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-261"><a href="#cb51-261" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb51-262"><a href="#cb51-262" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>city, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb51-263"><a href="#cb51-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb51-264"><a href="#cb51-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb51-265"><a href="#cb51-265" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-266"><a href="#cb51-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span></span>
<span id="cb51-267"><a href="#cb51-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>city, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb51-268"><a href="#cb51-268" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb51-269"><a href="#cb51-269" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb51-270"><a href="#cb51-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-271"><a href="#cb51-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-272"><a href="#cb51-272" aria-hidden="true" tabindex="-1"></a>Data cleaning was very successful at removing adapters and improving read qualities:</span>
<span id="cb51-273"><a href="#cb51-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-276"><a href="#cb51-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-277"><a href="#cb51-277" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-278"><a href="#cb51-278" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb51-279"><a href="#cb51-279" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb51-280"><a href="#cb51-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-281"><a href="#cb51-281" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>city, <span class="at">linetype=</span>read_pair, </span>
<span id="cb51-282"><a href="#cb51-282" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb51-283"><a href="#cb51-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb51-284"><a href="#cb51-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb51-285"><a href="#cb51-285" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb51-286"><a href="#cb51-286" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb51-287"><a href="#cb51-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-288"><a href="#cb51-288" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb51-289"><a href="#cb51-289" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb51-290"><a href="#cb51-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb51-291"><a href="#cb51-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb51-292"><a href="#cb51-292" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-293"><a href="#cb51-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb51-294"><a href="#cb51-294" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-295"><a href="#cb51-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb51-296"><a href="#cb51-296" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb51-297"><a href="#cb51-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-298"><a href="#cb51-298" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb51-299"><a href="#cb51-299" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb51-300"><a href="#cb51-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-301"><a href="#cb51-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-302"><a href="#cb51-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb51-303"><a href="#cb51-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb51-304"><a href="#cb51-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb51-305"><a href="#cb51-305" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-306"><a href="#cb51-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb51-307"><a href="#cb51-307" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb51-308"><a href="#cb51-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-309"><a href="#cb51-309" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb51-310"><a href="#cb51-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-311"><a href="#cb51-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-312"><a href="#cb51-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb51-313"><a href="#cb51-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-314"><a href="#cb51-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-315"><a href="#cb51-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb51-316"><a href="#cb51-316" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb51-317"><a href="#cb51-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-318"><a href="#cb51-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-319"><a href="#cb51-319" aria-hidden="true" tabindex="-1"></a>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 9.4% to 1.7% for DNA reads:</span>
<span id="cb51-320"><a href="#cb51-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-323"><a href="#cb51-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-324"><a href="#cb51-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb51-325"><a href="#cb51-325" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-326"><a href="#cb51-326" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb51-327"><a href="#cb51-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-328"><a href="#cb51-328" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb51-329"><a href="#cb51-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb51-330"><a href="#cb51-330" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb51-331"><a href="#cb51-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-332"><a href="#cb51-332" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb51-333"><a href="#cb51-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb51-334"><a href="#cb51-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb51-335"><a href="#cb51-335" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb51-336"><a href="#cb51-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-337"><a href="#cb51-337" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb51-338"><a href="#cb51-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb51-339"><a href="#cb51-339" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb51-340"><a href="#cb51-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-341"><a href="#cb51-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-342"><a href="#cb51-342" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb51-343"><a href="#cb51-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-344"><a href="#cb51-344" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb51-345"><a href="#cb51-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-348"><a href="#cb51-348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-349"><a href="#cb51-349" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb51-350"><a href="#cb51-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-351"><a href="#cb51-351" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bracken data</span></span>
<span id="cb51-352"><a href="#cb51-352" aria-hidden="true" tabindex="-1"></a>bracken_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"bracken_counts.tsv"</span>)</span>
<span id="cb51-353"><a href="#cb51-353" aria-hidden="true" tabindex="-1"></a>bracken <span class="ot">&lt;-</span> <span class="fu">lapply</span>(bracken_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb51-354"><a href="#cb51-354" aria-hidden="true" tabindex="-1"></a>total_assigned <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(</span>
<span id="cb51-355"><a href="#cb51-355" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Total"</span>,</span>
<span id="cb51-356"><a href="#cb51-356" aria-hidden="true" tabindex="-1"></a>  <span class="at">kraken_assigned_reads =</span> <span class="fu">sum</span>(kraken_assigned_reads),</span>
<span id="cb51-357"><a href="#cb51-357" aria-hidden="true" tabindex="-1"></a>  <span class="at">added_reads =</span> <span class="fu">sum</span>(added_reads),</span>
<span id="cb51-358"><a href="#cb51-358" aria-hidden="true" tabindex="-1"></a>  <span class="at">new_est_reads =</span> <span class="fu">sum</span>(new_est_reads),</span>
<span id="cb51-359"><a href="#cb51-359" aria-hidden="true" tabindex="-1"></a>  <span class="at">fraction_total_reads =</span> <span class="fu">sum</span>(fraction_total_reads)</span>
<span id="cb51-360"><a href="#cb51-360" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb51-361"><a href="#cb51-361" aria-hidden="true" tabindex="-1"></a>bracken_spread <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">select</span>(name, sample, new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb51-362"><a href="#cb51-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">tolower</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb51-363"><a href="#cb51-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"sample"</span>, <span class="at">names_from =</span> <span class="st">"name"</span>, </span>
<span id="cb51-364"><a href="#cb51-364" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="st">"new_est_reads"</span>)</span>
<span id="cb51-365"><a href="#cb51-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-366"><a href="#cb51-366" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb51-367"><a href="#cb51-367" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb51-368"><a href="#cb51-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, date, date_alias, city, stage, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb51-369"><a href="#cb51-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"date"</span>, <span class="st">"date_alias"</span>, <span class="st">"city"</span>),</span>
<span id="cb51-370"><a href="#cb51-370" aria-hidden="true" tabindex="-1"></a>              <span class="at">names_from=</span><span class="st">"stage"</span>, <span class="at">values_from=</span><span class="st">"n_read_pairs"</span>)</span>
<span id="cb51-371"><a href="#cb51-371" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_preproc <span class="sc">%&gt;%</span></span>
<span id="cb51-372"><a href="#cb51-372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total_assigned <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, new_est_reads), <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-373"><a href="#cb51-373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">assigned =</span> new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb51-374"><a href="#cb51-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(bracken_spread, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb51-375"><a href="#cb51-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-376"><a href="#cb51-376" aria-hidden="true" tabindex="-1"></a><span class="co"># Assess composition</span></span>
<span id="cb51-377"><a href="#cb51-377" aria-hidden="true" tabindex="-1"></a>read_comp <span class="ot">&lt;-</span> <span class="fu">transmute</span>(read_counts, sample, date, date_alias, city,</span>
<span id="cb51-378"><a href="#cb51-378" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_filtered =</span> raw_concat<span class="sc">-</span>cleaned,</span>
<span id="cb51-379"><a href="#cb51-379" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_duplicate =</span> cleaned<span class="sc">-</span>dedup,</span>
<span id="cb51-380"><a href="#cb51-380" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_ribosomal =</span> (dedup<span class="sc">-</span>ribo_initial) <span class="sc">+</span> (ribo_initial<span class="sc">-</span>ribo_secondary),</span>
<span id="cb51-381"><a href="#cb51-381" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_unassigned =</span> ribo_secondary<span class="sc">-</span>assigned,</span>
<span id="cb51-382"><a href="#cb51-382" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_bacterial =</span> bacteria,</span>
<span id="cb51-383"><a href="#cb51-383" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_archaeal =</span> archaea,</span>
<span id="cb51-384"><a href="#cb51-384" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_viral =</span> viruses,</span>
<span id="cb51-385"><a href="#cb51-385" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_human =</span> eukaryota)</span>
<span id="cb51-386"><a href="#cb51-386" aria-hidden="true" tabindex="-1"></a>read_comp_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(read_comp, <span class="sc">-</span>(sample<span class="sc">:</span>city), </span>
<span id="cb51-387"><a href="#cb51-387" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_to =</span> <span class="st">"classification"</span>,</span>
<span id="cb51-388"><a href="#cb51-388" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_prefix =</span> <span class="st">"n_"</span>, <span class="at">values_to =</span> <span class="st">"n_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-389"><a href="#cb51-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(<span class="fu">str_to_sentence</span>(classification))) <span class="sc">%&gt;%</span></span>
<span id="cb51-390"><a href="#cb51-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb51-391"><a href="#cb51-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-392"><a href="#cb51-392" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb51-393"><a href="#cb51-393" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb51-394"><a href="#cb51-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(city, classification) <span class="sc">%&gt;%</span></span>
<span id="cb51-395"><a href="#cb51-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-396"><a href="#cb51-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb51-397"><a href="#cb51-397" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb51-398"><a href="#cb51-398" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb51-399"><a href="#cb51-399" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-400"><a href="#cb51-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-403"><a href="#cb51-403" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-404"><a href="#cb51-404" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb51-405"><a href="#cb51-405" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb51-406"><a href="#cb51-406" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb51-407"><a href="#cb51-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-408"><a href="#cb51-408" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb51-409"><a href="#cb51-409" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>date_alias, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb51-410"><a href="#cb51-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Collection Date"</span>) <span class="sc">+</span></span>
<span id="cb51-411"><a href="#cb51-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>city, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb51-412"><a href="#cb51-412" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb51-413"><a href="#cb51-413" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb51-414"><a href="#cb51-414" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb51-415"><a href="#cb51-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-416"><a href="#cb51-416" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb51-417"><a href="#cb51-417" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data =</span> read_comp_long, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-418"><a href="#cb51-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb51-419"><a href="#cb51-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb51-420"><a href="#cb51-420" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb51-421"><a href="#cb51-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-422"><a href="#cb51-422" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb51-423"><a href="#cb51-423" aria-hidden="true" tabindex="-1"></a>read_comp_minor <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb51-424"><a href="#cb51-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Other"</span>))</span>
<span id="cb51-425"><a href="#cb51-425" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">9</span>)]</span>
<span id="cb51-426"><a href="#cb51-426" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data=</span>read_comp_minor, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-427"><a href="#cb51-427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb51-428"><a href="#cb51-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb51-429"><a href="#cb51-429" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb51-430"><a href="#cb51-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-431"><a href="#cb51-431" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-432"><a href="#cb51-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-435"><a href="#cb51-435" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-436"><a href="#cb51-436" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-summary</span></span>
<span id="cb51-437"><a href="#cb51-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-438"><a href="#cb51-438" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span></span>
<span id="cb51-439"><a href="#cb51-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb51-440"><a href="#cb51-440" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb51-441"><a href="#cb51-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-442"><a href="#cb51-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-443"><a href="#cb51-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-444"><a href="#cb51-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb51-445"><a href="#cb51-445" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb51-446"><a href="#cb51-446" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> p_reads_summ_group <span class="sc">%&gt;%</span></span>
<span id="cb51-447"><a href="#cb51-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb51-448"><a href="#cb51-448" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb51-449"><a href="#cb51-449" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb51-450"><a href="#cb51-450" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb51-451"><a href="#cb51-451" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb51-452"><a href="#cb51-452" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb51-453"><a href="#cb51-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(city, classification, <span class="at">read_fraction=</span>display) <span class="sc">%&gt;%</span></span>
<span id="cb51-454"><a href="#cb51-454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(city, classification)</span>
<span id="cb51-455"><a href="#cb51-455" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb51-456"><a href="#cb51-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-457"><a href="#cb51-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-458"><a href="#cb51-458" aria-hidden="true" tabindex="-1"></a>In many respects, these resemble the Prussin data: high human fraction (mean 19.6%), high bacterial fraction (mean 18.2%), high unclassified fraction (mean 43.9%), low viral fraction (mean 0.01%). One notable difference is that archaeal reads are more abundant (0.034% compared to 0.016% for Prussin).</span>
<span id="cb51-459"><a href="#cb51-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-460"><a href="#cb51-460" aria-hidden="true" tabindex="-1"></a>As in Prussin, viral DNA reads were dominated by *Caudoviricetes* phages. Other viral classes that are prominent in at least some samples include *Herviviricetes* (herpesviruses), *Papovaviricetes* (polyomaviruses and papillomaviruses), *Revtraviricetes* (retroviruses + Hep B), and *Naldaviricetes* (mainly arthropod viruses). I'll investigate the first three of this latter group in more depth, restricting in each case to samples where that family makes up at least 5% of viral reads.</span>
<span id="cb51-461"><a href="#cb51-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-464"><a href="#cb51-464" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-465"><a href="#cb51-465" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb51-466"><a href="#cb51-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-467"><a href="#cb51-467" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb51-468"><a href="#cb51-468" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb51-469"><a href="#cb51-469" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-470"><a href="#cb51-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-471"><a href="#cb51-471" aria-hidden="true" tabindex="-1"></a><span class="co"># Get paths to Kraken reports</span></span>
<span id="cb51-472"><a href="#cb51-472" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">as.character</span>(basic_stats_raw<span class="sc">$</span>sample)</span>
<span id="cb51-473"><a href="#cb51-473" aria-hidden="true" tabindex="-1"></a>report_dirs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"kraken"</span>)</span>
<span id="cb51-474"><a href="#cb51-474" aria-hidden="true" tabindex="-1"></a>report_paths <span class="ot">&lt;-</span> <span class="fu">lapply</span>(report_dirs, list.files, <span class="at">full.names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> unlist</span>
<span id="cb51-475"><a href="#cb51-475" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(report_paths) <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(report_paths, <span class="st">"SRR</span><span class="sc">\\</span><span class="st">d*"</span>)</span>
<span id="cb51-476"><a href="#cb51-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-477"><a href="#cb51-477" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract viral taxa</span></span>
<span id="cb51-478"><a href="#cb51-478" aria-hidden="true" tabindex="-1"></a>col_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pc_reads_total"</span>, <span class="st">"n_reads_clade"</span>, <span class="st">"n_reads_direct"</span>,</span>
<span id="cb51-479"><a href="#cb51-479" aria-hidden="true" tabindex="-1"></a>               <span class="st">"rank"</span>, <span class="st">"taxid"</span>, <span class="st">"name"</span>)</span>
<span id="cb51-480"><a href="#cb51-480" aria-hidden="true" tabindex="-1"></a>kraken_reports_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(report_paths, read_tsv, <span class="at">col_names =</span> col_names,</span>
<span id="cb51-481"><a href="#cb51-481" aria-hidden="true" tabindex="-1"></a>                             <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-482"><a href="#cb51-482" aria-hidden="true" tabindex="-1"></a>kraken_reports <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(kraken_reports_raw), </span>
<span id="cb51-483"><a href="#cb51-483" aria-hidden="true" tabindex="-1"></a>                         <span class="cf">function</span>(x) kraken_reports_raw[[x]] <span class="sc">%&gt;%</span> </span>
<span id="cb51-484"><a href="#cb51-484" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">mutate</span>(<span class="at">sample =</span> x)) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb51-485"><a href="#cb51-485" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(kraken_reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-486"><a href="#cb51-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb51-487"><a href="#cb51-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb51-488"><a href="#cb51-488" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb51-489"><a href="#cb51-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-490"><a href="#cb51-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct) <span class="sc">%&gt;%</span></span>
<span id="cb51-491"><a href="#cb51-491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>())</span>
<span id="cb51-492"><a href="#cb51-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-493"><a href="#cb51-493" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb51-494"><a href="#cb51-494" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb51-495"><a href="#cb51-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-496"><a href="#cb51-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-497"><a href="#cb51-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-500"><a href="#cb51-500" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-501"><a href="#cb51-501" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-class-composition</span></span>
<span id="cb51-502"><a href="#cb51-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-503"><a href="#cb51-503" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb51-504"><a href="#cb51-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-505"><a href="#cb51-505" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral classes</span></span>
<span id="cb51-506"><a href="#cb51-506" aria-hidden="true" tabindex="-1"></a>viral_classes_major_tab <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb51-507"><a href="#cb51-507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-508"><a href="#cb51-508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-509"><a href="#cb51-509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb51-510"><a href="#cb51-510" aria-hidden="true" tabindex="-1"></a>viral_classes_major_list <span class="ot">&lt;-</span> viral_classes_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb51-511"><a href="#cb51-511" aria-hidden="true" tabindex="-1"></a>viral_classes_major <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb51-512"><a href="#cb51-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_classes_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb51-513"><a href="#cb51-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_viral)</span>
<span id="cb51-514"><a href="#cb51-514" aria-hidden="true" tabindex="-1"></a>viral_classes_minor <span class="ot">&lt;-</span> viral_classes_major <span class="sc">%&gt;%</span> </span>
<span id="cb51-515"><a href="#cb51-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-516"><a href="#cb51-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-517"><a href="#cb51-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb51-518"><a href="#cb51-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_viral)</span>
<span id="cb51-519"><a href="#cb51-519" aria-hidden="true" tabindex="-1"></a>viral_classes_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_classes_major, viral_classes_minor) <span class="sc">%&gt;%</span></span>
<span id="cb51-520"><a href="#cb51-520" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-521"><a href="#cb51-521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_classes_major_list, <span class="st">"Other"</span>)),</span>
<span id="cb51-522"><a href="#cb51-522" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> <span class="fu">pmax</span>(p_reads_viral, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-523"><a href="#cb51-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name)</span>
<span id="cb51-524"><a href="#cb51-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-525"><a href="#cb51-525" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb51-526"><a href="#cb51-526" aria-hidden="true" tabindex="-1"></a>g_classes <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-527"><a href="#cb51-527" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>viral_classes_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-528"><a href="#cb51-528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-529"><a href="#cb51-529" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-530"><a href="#cb51-530" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb51-531"><a href="#cb51-531" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-532"><a href="#cb51-532" aria-hidden="true" tabindex="-1"></a>g_classes</span>
<span id="cb51-533"><a href="#cb51-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-534"><a href="#cb51-534" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-535"><a href="#cb51-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-536"><a href="#cb51-536" aria-hidden="true" tabindex="-1"></a>*Papovaviricetes* are quite heterogeneous across samples, and frequently diverse within samples. *Alphapolyomavirus* and *Alphapapillomavirus* are the most abundant genera overall, but *Betapapillomavirus, Gammapapillomavirus, Mupapillomavirus* and others all have strong showings.</span>
<span id="cb51-537"><a href="#cb51-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-540"><a href="#cb51-540" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-541"><a href="#cb51-541" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: papovaviricetes</span></span>
<span id="cb51-542"><a href="#cb51-542" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-543"><a href="#cb51-543" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-544"><a href="#cb51-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-545"><a href="#cb51-545" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples</span></span>
<span id="cb51-546"><a href="#cb51-546" aria-hidden="true" tabindex="-1"></a>papova_taxid <span class="ot">&lt;-</span> <span class="dv">2732421</span></span>
<span id="cb51-547"><a href="#cb51-547" aria-hidden="true" tabindex="-1"></a>papova_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb51-548"><a href="#cb51-548" aria-hidden="true" tabindex="-1"></a>papova_samples <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> papova_taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_viral <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample) <span class="sc">%&gt;%</span> unique</span>
<span id="cb51-549"><a href="#cb51-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-550"><a href="#cb51-550" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all taxa in class</span></span>
<span id="cb51-551"><a href="#cb51-551" aria-hidden="true" tabindex="-1"></a>papova_desc_taxids_old <span class="ot">&lt;-</span> papova_taxid</span>
<span id="cb51-552"><a href="#cb51-552" aria-hidden="true" tabindex="-1"></a>papova_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(papova_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> papova_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-553"><a href="#cb51-553" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (<span class="fu">length</span>(papova_desc_taxids_new) <span class="sc">&gt;</span> <span class="fu">length</span>(papova_desc_taxids_old)){</span>
<span id="cb51-554"><a href="#cb51-554" aria-hidden="true" tabindex="-1"></a>  papova_desc_taxids_old <span class="ot">&lt;-</span> papova_desc_taxids_new</span>
<span id="cb51-555"><a href="#cb51-555" aria-hidden="true" tabindex="-1"></a>  papova_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(papova_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> papova_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-556"><a href="#cb51-556" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-557"><a href="#cb51-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-558"><a href="#cb51-558" aria-hidden="true" tabindex="-1"></a><span class="co"># Get read counts</span></span>
<span id="cb51-559"><a href="#cb51-559" aria-hidden="true" tabindex="-1"></a>papova_counts <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb51-560"><a href="#cb51-560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">%in%</span> papova_desc_taxids_new,</span>
<span id="cb51-561"><a href="#cb51-561" aria-hidden="true" tabindex="-1"></a>         sample <span class="sc">%in%</span> papova_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-562"><a href="#cb51-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_papova =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb51-563"><a href="#cb51-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-564"><a href="#cb51-564" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genus composition</span></span>
<span id="cb51-565"><a href="#cb51-565" aria-hidden="true" tabindex="-1"></a>papova_genera <span class="ot">&lt;-</span> papova_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"G"</span>)</span>
<span id="cb51-566"><a href="#cb51-566" aria-hidden="true" tabindex="-1"></a>papova_genera_major_tab <span class="ot">&lt;-</span> papova_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-567"><a href="#cb51-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-568"><a href="#cb51-568" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_papova_max =</span> <span class="fu">max</span>(p_reads_papova), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-569"><a href="#cb51-569" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papova_max <span class="sc">&gt;=</span> papova_threshold)</span>
<span id="cb51-570"><a href="#cb51-570" aria-hidden="true" tabindex="-1"></a>papova_genera_major_list <span class="ot">&lt;-</span> papova_genera_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb51-571"><a href="#cb51-571" aria-hidden="true" tabindex="-1"></a>papova_genera_major <span class="ot">&lt;-</span> papova_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-572"><a href="#cb51-572" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> papova_genera_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb51-573"><a href="#cb51-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_papova)</span>
<span id="cb51-574"><a href="#cb51-574" aria-hidden="true" tabindex="-1"></a>papova_genera_minor <span class="ot">&lt;-</span> papova_genera_major <span class="sc">%&gt;%</span> </span>
<span id="cb51-575"><a href="#cb51-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-576"><a href="#cb51-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_papova_major =</span> <span class="fu">sum</span>(p_reads_papova), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-577"><a href="#cb51-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_papova =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_papova_major) <span class="sc">%&gt;%</span></span>
<span id="cb51-578"><a href="#cb51-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_papova)</span>
<span id="cb51-579"><a href="#cb51-579" aria-hidden="true" tabindex="-1"></a>papova_genera_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(papova_genera_major, papova_genera_minor) <span class="sc">%&gt;%</span></span>
<span id="cb51-580"><a href="#cb51-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_papova)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-581"><a href="#cb51-581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(papova_genera_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-582"><a href="#cb51-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_papova, <span class="at">classification=</span>name)</span>
<span id="cb51-583"><a href="#cb51-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-584"><a href="#cb51-584" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-585"><a href="#cb51-585" aria-hidden="true" tabindex="-1"></a>g_papova_genera <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-586"><a href="#cb51-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>papova_genera_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-587"><a href="#cb51-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Papovaviricetes Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.02</span>), </span>
<span id="cb51-588"><a href="#cb51-588" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-589"><a href="#cb51-589" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-590"><a href="#cb51-590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb51-591"><a href="#cb51-591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb51-592"><a href="#cb51-592" aria-hidden="true" tabindex="-1"></a>g_papova_genera</span>
<span id="cb51-593"><a href="#cb51-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-594"><a href="#cb51-594" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-595"><a href="#cb51-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-596"><a href="#cb51-596" aria-hidden="true" tabindex="-1"></a>Only a few samples showed at least 5% prevalence of *Herviviricetes*, but those that did were typically dominated by one or a small number of species that varied between samples. Of these, human alphaherpesvirus 1 appeared in the most samples, but several other species were prominent in at least one sample:</span>
<span id="cb51-597"><a href="#cb51-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-600"><a href="#cb51-600" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-601"><a href="#cb51-601" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: herviviricetes</span></span>
<span id="cb51-602"><a href="#cb51-602" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-603"><a href="#cb51-603" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-604"><a href="#cb51-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-605"><a href="#cb51-605" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples</span></span>
<span id="cb51-606"><a href="#cb51-606" aria-hidden="true" tabindex="-1"></a>hervi_taxid <span class="ot">&lt;-</span> <span class="dv">2731363</span></span>
<span id="cb51-607"><a href="#cb51-607" aria-hidden="true" tabindex="-1"></a>hervi_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb51-608"><a href="#cb51-608" aria-hidden="true" tabindex="-1"></a>hervi_samples <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> hervi_taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_viral <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample) <span class="sc">%&gt;%</span> unique</span>
<span id="cb51-609"><a href="#cb51-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-610"><a href="#cb51-610" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all taxa in class</span></span>
<span id="cb51-611"><a href="#cb51-611" aria-hidden="true" tabindex="-1"></a>hervi_desc_taxids_old <span class="ot">&lt;-</span> hervi_taxid</span>
<span id="cb51-612"><a href="#cb51-612" aria-hidden="true" tabindex="-1"></a>hervi_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(hervi_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> hervi_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-613"><a href="#cb51-613" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (<span class="fu">length</span>(hervi_desc_taxids_new) <span class="sc">&gt;</span> <span class="fu">length</span>(hervi_desc_taxids_old)){</span>
<span id="cb51-614"><a href="#cb51-614" aria-hidden="true" tabindex="-1"></a>  hervi_desc_taxids_old <span class="ot">&lt;-</span> hervi_desc_taxids_new</span>
<span id="cb51-615"><a href="#cb51-615" aria-hidden="true" tabindex="-1"></a>  hervi_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(hervi_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> hervi_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-616"><a href="#cb51-616" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-617"><a href="#cb51-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-618"><a href="#cb51-618" aria-hidden="true" tabindex="-1"></a><span class="co"># Get read counts</span></span>
<span id="cb51-619"><a href="#cb51-619" aria-hidden="true" tabindex="-1"></a>hervi_counts <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb51-620"><a href="#cb51-620" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">%in%</span> hervi_desc_taxids_new,</span>
<span id="cb51-621"><a href="#cb51-621" aria-hidden="true" tabindex="-1"></a>         sample <span class="sc">%in%</span> hervi_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-622"><a href="#cb51-622" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hervi =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb51-623"><a href="#cb51-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-624"><a href="#cb51-624" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genus composition</span></span>
<span id="cb51-625"><a href="#cb51-625" aria-hidden="true" tabindex="-1"></a>hervi_genera <span class="ot">&lt;-</span> hervi_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"S"</span>)</span>
<span id="cb51-626"><a href="#cb51-626" aria-hidden="true" tabindex="-1"></a>hervi_genera_major_tab <span class="ot">&lt;-</span> hervi_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-627"><a href="#cb51-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-628"><a href="#cb51-628" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_hervi_max =</span> <span class="fu">max</span>(p_reads_hervi), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-629"><a href="#cb51-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hervi_max <span class="sc">&gt;=</span> hervi_threshold)</span>
<span id="cb51-630"><a href="#cb51-630" aria-hidden="true" tabindex="-1"></a>hervi_genera_major_list <span class="ot">&lt;-</span> hervi_genera_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb51-631"><a href="#cb51-631" aria-hidden="true" tabindex="-1"></a>hervi_genera_major <span class="ot">&lt;-</span> hervi_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-632"><a href="#cb51-632" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> hervi_genera_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb51-633"><a href="#cb51-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_hervi)</span>
<span id="cb51-634"><a href="#cb51-634" aria-hidden="true" tabindex="-1"></a>hervi_genera_minor <span class="ot">&lt;-</span> hervi_genera_major <span class="sc">%&gt;%</span> </span>
<span id="cb51-635"><a href="#cb51-635" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-636"><a href="#cb51-636" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_hervi_major =</span> <span class="fu">sum</span>(p_reads_hervi), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-637"><a href="#cb51-637" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_hervi =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_hervi_major) <span class="sc">%&gt;%</span></span>
<span id="cb51-638"><a href="#cb51-638" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_hervi)</span>
<span id="cb51-639"><a href="#cb51-639" aria-hidden="true" tabindex="-1"></a>hervi_genera_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hervi_genera_major, hervi_genera_minor) <span class="sc">%&gt;%</span></span>
<span id="cb51-640"><a href="#cb51-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hervi)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-641"><a href="#cb51-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(hervi_genera_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-642"><a href="#cb51-642" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hervi, <span class="at">classification=</span>name)</span>
<span id="cb51-643"><a href="#cb51-643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-644"><a href="#cb51-644" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-645"><a href="#cb51-645" aria-hidden="true" tabindex="-1"></a>g_hervi_genera <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-646"><a href="#cb51-646" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hervi_genera_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-647"><a href="#cb51-647" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Herviviricetes Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-648"><a href="#cb51-648" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-649"><a href="#cb51-649" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-650"><a href="#cb51-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb51-651"><a href="#cb51-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb51-652"><a href="#cb51-652" aria-hidden="true" tabindex="-1"></a>g_hervi_genera</span>
<span id="cb51-653"><a href="#cb51-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-654"><a href="#cb51-654" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-655"><a href="#cb51-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-656"><a href="#cb51-656" aria-hidden="true" tabindex="-1"></a>Finally, for *Revtraviricetes*, most samples were dominated by porcine type-C oncovirus, while one was dominated by an avian retrovirus. The last showed significant levels of two murine viruses plus HIV. I'm suspicious of many of these.</span>
<span id="cb51-657"><a href="#cb51-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-660"><a href="#cb51-660" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-661"><a href="#cb51-661" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: revtraviricetes</span></span>
<span id="cb51-662"><a href="#cb51-662" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-663"><a href="#cb51-663" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-664"><a href="#cb51-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-665"><a href="#cb51-665" aria-hidden="true" tabindex="-1"></a><span class="co"># Get samples</span></span>
<span id="cb51-666"><a href="#cb51-666" aria-hidden="true" tabindex="-1"></a>revtra_taxid <span class="ot">&lt;-</span> <span class="dv">2732514</span></span>
<span id="cb51-667"><a href="#cb51-667" aria-hidden="true" tabindex="-1"></a>revtra_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb51-668"><a href="#cb51-668" aria-hidden="true" tabindex="-1"></a>revtra_samples <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> revtra_taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_viral <span class="sc">&gt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample) <span class="sc">%&gt;%</span> unique</span>
<span id="cb51-669"><a href="#cb51-669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-670"><a href="#cb51-670" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all taxa in class</span></span>
<span id="cb51-671"><a href="#cb51-671" aria-hidden="true" tabindex="-1"></a>revtra_desc_taxids_old <span class="ot">&lt;-</span> revtra_taxid</span>
<span id="cb51-672"><a href="#cb51-672" aria-hidden="true" tabindex="-1"></a>revtra_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(revtra_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> revtra_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-673"><a href="#cb51-673" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (<span class="fu">length</span>(revtra_desc_taxids_new) <span class="sc">&gt;</span> <span class="fu">length</span>(revtra_desc_taxids_old)){</span>
<span id="cb51-674"><a href="#cb51-674" aria-hidden="true" tabindex="-1"></a>  revtra_desc_taxids_old <span class="ot">&lt;-</span> revtra_desc_taxids_new</span>
<span id="cb51-675"><a href="#cb51-675" aria-hidden="true" tabindex="-1"></a>  revtra_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(revtra_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> revtra_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb51-676"><a href="#cb51-676" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-677"><a href="#cb51-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-678"><a href="#cb51-678" aria-hidden="true" tabindex="-1"></a><span class="co"># Get read counts</span></span>
<span id="cb51-679"><a href="#cb51-679" aria-hidden="true" tabindex="-1"></a>revtra_counts <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb51-680"><a href="#cb51-680" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">%in%</span> revtra_desc_taxids_new,</span>
<span id="cb51-681"><a href="#cb51-681" aria-hidden="true" tabindex="-1"></a>         sample <span class="sc">%in%</span> revtra_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-682"><a href="#cb51-682" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_revtra =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb51-683"><a href="#cb51-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-684"><a href="#cb51-684" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genus composition</span></span>
<span id="cb51-685"><a href="#cb51-685" aria-hidden="true" tabindex="-1"></a>revtra_genera <span class="ot">&lt;-</span> revtra_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"S"</span>)</span>
<span id="cb51-686"><a href="#cb51-686" aria-hidden="true" tabindex="-1"></a>revtra_genera_major_tab <span class="ot">&lt;-</span> revtra_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-687"><a href="#cb51-687" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-688"><a href="#cb51-688" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_revtra_max =</span> <span class="fu">max</span>(p_reads_revtra), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-689"><a href="#cb51-689" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_revtra_max <span class="sc">&gt;=</span> revtra_threshold)</span>
<span id="cb51-690"><a href="#cb51-690" aria-hidden="true" tabindex="-1"></a>revtra_genera_major_list <span class="ot">&lt;-</span> revtra_genera_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb51-691"><a href="#cb51-691" aria-hidden="true" tabindex="-1"></a>revtra_genera_major <span class="ot">&lt;-</span> revtra_genera <span class="sc">%&gt;%</span> </span>
<span id="cb51-692"><a href="#cb51-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> revtra_genera_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb51-693"><a href="#cb51-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_revtra)</span>
<span id="cb51-694"><a href="#cb51-694" aria-hidden="true" tabindex="-1"></a>revtra_genera_minor <span class="ot">&lt;-</span> revtra_genera_major <span class="sc">%&gt;%</span> </span>
<span id="cb51-695"><a href="#cb51-695" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-696"><a href="#cb51-696" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_revtra_major =</span> <span class="fu">sum</span>(p_reads_revtra), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-697"><a href="#cb51-697" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_revtra =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_revtra_major) <span class="sc">%&gt;%</span></span>
<span id="cb51-698"><a href="#cb51-698" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, date_alias, city, p_reads_revtra)</span>
<span id="cb51-699"><a href="#cb51-699" aria-hidden="true" tabindex="-1"></a>revtra_genera_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(revtra_genera_major, revtra_genera_minor) <span class="sc">%&gt;%</span></span>
<span id="cb51-700"><a href="#cb51-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_revtra)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-701"><a href="#cb51-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(revtra_genera_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-702"><a href="#cb51-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_revtra, <span class="at">classification=</span>name)</span>
<span id="cb51-703"><a href="#cb51-703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-704"><a href="#cb51-704" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-705"><a href="#cb51-705" aria-hidden="true" tabindex="-1"></a>g_revtra_genera <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-706"><a href="#cb51-706" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>revtra_genera_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-707"><a href="#cb51-707" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% revtraviricetes Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-708"><a href="#cb51-708" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-709"><a href="#cb51-709" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-710"><a href="#cb51-710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb51-711"><a href="#cb51-711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb51-712"><a href="#cb51-712" aria-hidden="true" tabindex="-1"></a>g_revtra_genera</span>
<span id="cb51-713"><a href="#cb51-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-714"><a href="#cb51-714" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-715"><a href="#cb51-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-716"><a href="#cb51-716" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb51-717"><a href="#cb51-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-718"><a href="#cb51-718" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples. Using the same workflow I used for Prussin et al, I identified 24,278 read pairs as putatively human viral: 0.002% of reads surviving to that stage in the pipeline.</span>
<span id="cb51-719"><a href="#cb51-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-722"><a href="#cb51-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-723"><a href="#cb51-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-read-counts</span></span>
<span id="cb51-724"><a href="#cb51-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-725"><a href="#cb51-725" aria-hidden="true" tabindex="-1"></a><span class="co"># Import HV read data</span></span>
<span id="cb51-726"><a href="#cb51-726" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb51-727"><a href="#cb51-727" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hv_reads_filtered_paths, read_tsv,</span>
<span id="cb51-728"><a href="#cb51-728" aria-hidden="true" tabindex="-1"></a>                            <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-729"><a href="#cb51-729" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb51-730"><a href="#cb51-730" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb51-731"><a href="#cb51-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-732"><a href="#cb51-732" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb51-733"><a href="#cb51-733" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb51-734"><a href="#cb51-734" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date, date_alias, city, seq_id) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb51-735"><a href="#cb51-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date, date_alias, city) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb51-736"><a href="#cb51-736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-737"><a href="#cb51-737" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-738"><a href="#cb51-738" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb51-739"><a href="#cb51-739" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb51-740"><a href="#cb51-740" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_summ <span class="ot">&lt;-</span> n_hv_filtered <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb51-741"><a href="#cb51-741" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total), </span>
<span id="cb51-742"><a href="#cb51-742" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-743"><a href="#cb51-743" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb51-744"><a href="#cb51-744" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-745"><a href="#cb51-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-748"><a href="#cb51-748" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-749"><a href="#cb51-749" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-scores</span></span>
<span id="cb51-750"><a href="#cb51-750" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-751"><a href="#cb51-751" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb51-752"><a href="#cb51-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-753"><a href="#cb51-753" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse multi-entry sequences</span></span>
<span id="cb51-754"><a href="#cb51-754" aria-hidden="true" tabindex="-1"></a>rmax <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(max, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb51-755"><a href="#cb51-755" aria-hidden="true" tabindex="-1"></a>collapse <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">ifelse</span>(<span class="fu">all</span>(x <span class="sc">==</span> x[<span class="dv">1</span>]), x[<span class="dv">1</span>], <span class="fu">paste</span>(x, <span class="at">collapse=</span><span class="st">"/"</span>))</span>
<span id="cb51-756"><a href="#cb51-756" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> </span>
<span id="cb51-757"><a href="#cb51-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-758"><a href="#cb51-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb51-759"><a href="#cb51-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb51-760"><a href="#cb51-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sample =</span> <span class="fu">collapse</span>(sample),</span>
<span id="cb51-761"><a href="#cb51-761" aria-hidden="true" tabindex="-1"></a>            <span class="at">genome_id =</span> <span class="fu">collapse</span>(genome_id),</span>
<span id="cb51-762"><a href="#cb51-762" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid_best =</span> taxid[<span class="dv">1</span>],</span>
<span id="cb51-763"><a href="#cb51-763" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(taxid)),</span>
<span id="cb51-764"><a href="#cb51-764" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_fwd =</span> <span class="fu">rmax</span>(best_alignment_score_fwd),</span>
<span id="cb51-765"><a href="#cb51-765" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_rev =</span> <span class="fu">rmax</span>(best_alignment_score_rev),</span>
<span id="cb51-766"><a href="#cb51-766" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_fwd =</span> <span class="fu">rmax</span>(query_len_fwd),</span>
<span id="cb51-767"><a href="#cb51-767" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_rev =</span> <span class="fu">rmax</span>(query_len_rev),</span>
<span id="cb51-768"><a href="#cb51-768" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_fwd =</span> query_seq_fwd[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_fwd)][<span class="dv">1</span>],</span>
<span id="cb51-769"><a href="#cb51-769" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_rev =</span> query_seq_rev[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_rev)][<span class="dv">1</span>],</span>
<span id="cb51-770"><a href="#cb51-770" aria-hidden="true" tabindex="-1"></a>            <span class="at">classified =</span> <span class="fu">rmax</span>(classified),</span>
<span id="cb51-771"><a href="#cb51-771" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_name =</span> <span class="fu">collapse</span>(assigned_name),</span>
<span id="cb51-772"><a href="#cb51-772" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid_best =</span> assigned_taxid[<span class="dv">1</span>],</span>
<span id="cb51-773"><a href="#cb51-773" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(assigned_taxid)),</span>
<span id="cb51-774"><a href="#cb51-774" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_hv =</span> <span class="fu">rmax</span>(assigned_hv),</span>
<span id="cb51-775"><a href="#cb51-775" aria-hidden="true" tabindex="-1"></a>            <span class="at">hit_hv =</span> <span class="fu">rmax</span>(hit_hv),</span>
<span id="cb51-776"><a href="#cb51-776" aria-hidden="true" tabindex="-1"></a>            <span class="at">encoded_hits =</span> <span class="fu">collapse</span>(encoded_hits),</span>
<span id="cb51-777"><a href="#cb51-777" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_fwd =</span> <span class="fu">rmax</span>(adj_score_fwd),</span>
<span id="cb51-778"><a href="#cb51-778" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_rev =</span> <span class="fu">rmax</span>(adj_score_rev)</span>
<span id="cb51-779"><a href="#cb51-779" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb51-780"><a href="#cb51-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-781"><a href="#cb51-781" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb51-782"><a href="#cb51-782" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb51-783"><a href="#cb51-783" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-784"><a href="#cb51-784" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb51-785"><a href="#cb51-785" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb51-786"><a href="#cb51-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-787"><a href="#cb51-787" aria-hidden="true" tabindex="-1"></a>g_hist_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> </span>
<span id="cb51-788"><a href="#cb51-788" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb51-789"><a href="#cb51-789" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-790"><a href="#cb51-790" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>)), <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb51-791"><a href="#cb51-791" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> </span>
<span id="cb51-792"><a href="#cb51-792" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> </span>
<span id="cb51-793"><a href="#cb51-793" aria-hidden="true" tabindex="-1"></a>  theme_base </span>
<span id="cb51-794"><a href="#cb51-794" aria-hidden="true" tabindex="-1"></a>g_hist_0</span>
<span id="cb51-795"><a href="#cb51-795" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-796"><a href="#cb51-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-797"><a href="#cb51-797" aria-hidden="true" tabindex="-1"></a>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I've used for previous datasets.</span>
<span id="cb51-798"><a href="#cb51-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-801"><a href="#cb51-801" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-802"><a href="#cb51-802" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: make-blast-fasta</span></span>
<span id="cb51-803"><a href="#cb51-803" aria-hidden="true" tabindex="-1"></a>mrg_fasta <span class="ot">&lt;-</span>  mrg <span class="sc">%&gt;%</span></span>
<span id="cb51-804"><a href="#cb51-804" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, seq_id)) <span class="sc">%&gt;%</span></span>
<span id="cb51-805"><a href="#cb51-805" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb51-806"><a href="#cb51-806" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, </span>
<span id="cb51-807"><a href="#cb51-807" aria-hidden="true" tabindex="-1"></a>         <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb51-808"><a href="#cb51-808" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb51-809"><a href="#cb51-809" aria-hidden="true" tabindex="-1"></a>mrg_fasta_sep <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">select</span>(mrg_fasta, <span class="at">header=</span>header1, <span class="at">seq=</span>seq1),</span>
<span id="cb51-810"><a href="#cb51-810" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">select</span>(mrg_fasta, <span class="at">header=</span>header2, <span class="at">seq=</span>seq2)) <span class="sc">%&gt;%</span></span>
<span id="cb51-811"><a href="#cb51-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(seq))</span>
<span id="cb51-812"><a href="#cb51-812" aria-hidden="true" tabindex="-1"></a>mrg_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_fasta_sep, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-813"><a href="#cb51-813" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb51-814"><a href="#cb51-814" aria-hidden="true" tabindex="-1"></a>blast_dir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"blast"</span>)</span>
<span id="cb51-815"><a href="#cb51-815" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(blast_dir, <span class="at">showWarnings =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-816"><a href="#cb51-816" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_fasta_out, <span class="fu">file.path</span>(blast_dir, <span class="st">"putative-viral.fasta"</span>))</span>
<span id="cb51-817"><a href="#cb51-817" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-818"><a href="#cb51-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-821"><a href="#cb51-821" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-822"><a href="#cb51-822" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-blast-data</span></span>
<span id="cb51-823"><a href="#cb51-823" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-824"><a href="#cb51-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-825"><a href="#cb51-825" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb51-826"><a href="#cb51-826" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results_path &lt;- file.path(data_dir_base, "blast/putative-viral.blast.gz")</span></span>
<span id="cb51-827"><a href="#cb51-827" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_cols &lt;- c("qseqid", "sseqid", "sgi", "staxid", "qlen", "evalue", "bitscore", "qcovs", "length", "pident", "mismatch", "gapopen", "sstrand", "qstart", "qend", "sstart", "send")</span></span>
<span id="cb51-828"><a href="#cb51-828" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results &lt;- read_tsv(blast_results_path, show_col_types = FALSE,</span></span>
<span id="cb51-829"><a href="#cb51-829" aria-hidden="true" tabindex="-1"></a><span class="co">#                           col_names = blast_cols, col_types = cols(.default="c"))</span></span>
<span id="cb51-830"><a href="#cb51-830" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"blast/putative-viral-best.blast.gz"</span>)</span>
<span id="cb51-831"><a href="#cb51-831" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-832"><a href="#cb51-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-833"><a href="#cb51-833" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject combination</span></span>
<span id="cb51-834"><a href="#cb51-834" aria-hidden="true" tabindex="-1"></a>blast_results_best <span class="ot">&lt;-</span> blast_results <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-835"><a href="#cb51-835" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb51-836"><a href="#cb51-836" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb51-837"><a href="#cb51-837" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(blast_results_best, file.path(data_dir_base, "blast/putative-viral-best.blast.gz"))</span></span>
<span id="cb51-838"><a href="#cb51-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-839"><a href="#cb51-839" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each query and filter for high-ranking hits</span></span>
<span id="cb51-840"><a href="#cb51-840" aria-hidden="true" tabindex="-1"></a>blast_results_ranked <span class="ot">&lt;-</span> blast_results_best <span class="sc">%&gt;%</span> </span>
<span id="cb51-841"><a href="#cb51-841" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb51-842"><a href="#cb51-842" aria-hidden="true" tabindex="-1"></a>blast_results_highrank <span class="ot">&lt;-</span> blast_results_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-843"><a href="#cb51-843" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>), </span>
<span id="cb51-844"><a href="#cb51-844" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_id =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-845"><a href="#cb51-845" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore))</span>
<span id="cb51-846"><a href="#cb51-846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-847"><a href="#cb51-847" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb51-848"><a href="#cb51-848" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb51-849"><a href="#cb51-849" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-850"><a href="#cb51-850" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb51-851"><a href="#cb51-851" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb51-852"><a href="#cb51-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-853"><a href="#cb51-853" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb51-854"><a href="#cb51-854" aria-hidden="true" tabindex="-1"></a>blast_results_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_results_paired, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-855"><a href="#cb51-855" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb51-856"><a href="#cb51-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-857"><a href="#cb51-857" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb51-858"><a href="#cb51-858" aria-hidden="true" tabindex="-1"></a>match_taxid <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_1, taxid_2){</span>
<span id="cb51-859"><a href="#cb51-859" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"/"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb51-860"><a href="#cb51-860" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"/"</span>), taxid_2)</span>
<span id="cb51-861"><a href="#cb51-861" aria-hidden="true" tabindex="-1"></a>  p3 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb51-862"><a href="#cb51-862" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">setNames</span>(p1<span class="sc">|</span>p2<span class="sc">|</span>p3, <span class="cn">NULL</span>)</span>
<span id="cb51-863"><a href="#cb51-863" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb51-864"><a href="#cb51-864" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-865"><a href="#cb51-865" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_id, taxid, assigned_taxid, adj_score_max)</span>
<span id="cb51-866"><a href="#cb51-866" aria-hidden="true" tabindex="-1"></a>blast_results_assign <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(blast_results_viral, mrg_assign, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-867"><a href="#cb51-867" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> <span class="fu">match_taxid</span>(staxid, taxid),</span>
<span id="cb51-868"><a href="#cb51-868" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> <span class="fu">match_taxid</span>(staxid, assigned_taxid),</span>
<span id="cb51-869"><a href="#cb51-869" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb51-870"><a href="#cb51-870" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_assign <span class="sc">%&gt;%</span></span>
<span id="cb51-871"><a href="#cb51-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb51-872"><a href="#cb51-872" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb51-873"><a href="#cb51-873" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb51-874"><a href="#cb51-874" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb51-875"><a href="#cb51-875" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb51-876"><a href="#cb51-876" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-877"><a href="#cb51-877" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-880"><a href="#cb51-880" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-881"><a href="#cb51-881" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-results</span></span>
<span id="cb51-882"><a href="#cb51-882" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-883"><a href="#cb51-883" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-884"><a href="#cb51-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-885"><a href="#cb51-885" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb51-886"><a href="#cb51-886" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg, blast_results_out, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-887"><a href="#cb51-887" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb51-888"><a href="#cb51-888" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb51-889"><a href="#cb51-889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-890"><a href="#cb51-890" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot RNA</span></span>
<span id="cb51-891"><a href="#cb51-891" aria-hidden="true" tabindex="-1"></a>g_hist_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_blast, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max, <span class="at">fill=</span>viral_status_out)) <span class="sc">+</span> </span>
<span id="cb51-892"><a href="#cb51-892" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb51-893"><a href="#cb51-893" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-894"><a href="#cb51-894" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>)), <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb51-895"><a href="#cb51-895" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> </span>
<span id="cb51-896"><a href="#cb51-896" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> </span>
<span id="cb51-897"><a href="#cb51-897" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb51-898"><a href="#cb51-898" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb51-899"><a href="#cb51-899" aria-hidden="true" tabindex="-1"></a>g_hist_1</span>
<span id="cb51-900"><a href="#cb51-900" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-901"><a href="#cb51-901" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-902"><a href="#cb51-902" aria-hidden="true" tabindex="-1"></a>For a disjunctive score threshold of 20, the workflow achieves a measured F1 score of 98.0%.</span>
<span id="cb51-903"><a href="#cb51-903" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-906"><a href="#cb51-906" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-907"><a href="#cb51-907" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-f1</span></span>
<span id="cb51-908"><a href="#cb51-908" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold){</span>
<span id="cb51-909"><a href="#cb51-909" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> </span>
<span id="cb51-910"><a href="#cb51-910" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">retain_score =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb51-911"><a href="#cb51-911" aria-hidden="true" tabindex="-1"></a>           <span class="at">retain =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb51-912"><a href="#cb51-912" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb51-913"><a href="#cb51-913" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb51-914"><a href="#cb51-914" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb51-915"><a href="#cb51-915" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb51-916"><a href="#cb51-916" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb51-917"><a href="#cb51-917" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb51-918"><a href="#cb51-918" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb51-919"><a href="#cb51-919" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb51-920"><a href="#cb51-920" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb51-921"><a href="#cb51-921" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb51-922"><a href="#cb51-922" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb51-923"><a href="#cb51-923" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb51-924"><a href="#cb51-924" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-925"><a href="#cb51-925" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb51-926"><a href="#cb51-926" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab)</span>
<span id="cb51-927"><a href="#cb51-927" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb51-928"><a href="#cb51-928" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>threshold, <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>)</span>
<span id="cb51-929"><a href="#cb51-929" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats)</span>
<span id="cb51-930"><a href="#cb51-930" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-931"><a href="#cb51-931" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast)</span>
<span id="cb51-932"><a href="#cb51-932" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb51-933"><a href="#cb51-933" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb51-934"><a href="#cb51-934" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb51-935"><a href="#cb51-935" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-936"><a href="#cb51-936" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adjusted Score Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-937"><a href="#cb51-937" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb51-938"><a href="#cb51-938" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb51-939"><a href="#cb51-939" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb51-940"><a href="#cb51-940" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-941"><a href="#cb51-941" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">Threshold=</span>threshold, <span class="at">Metric=</span>metric, <span class="at">Value=</span>value)</span>
<span id="cb51-942"><a href="#cb51-942" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-943"><a href="#cb51-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-944"><a href="#cb51-944" aria-hidden="true" tabindex="-1"></a>Looking into the composition of different read groups, the notable observation for me is the high prevalence of Pigeon torque teno virus among high-scoring false positives, with 77 such read pairs. BLAST maps these not to viruses but to their most common hosts, i.e. assorted species of pigeon. That said, the number of false positive PTTV reads is substantially exceeded by the number of true-positive PTTV reads (1883), which do map to appropriate viruses according to BLAST, so the presence of a comparatively small number of false positives seems unlikely to cause too much distortion.</span>
<span id="cb51-945"><a href="#cb51-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-948"><a href="#cb51-948" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-949"><a href="#cb51-949" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fp</span></span>
<span id="cb51-950"><a href="#cb51-950" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb51-951"><a href="#cb51-951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-952"><a href="#cb51-952" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.04</span></span>
<span id="cb51-953"><a href="#cb51-953" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-954"><a href="#cb51-954" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing viral taxa</span></span>
<span id="cb51-955"><a href="#cb51-955" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">211787</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span id="cb51-956"><a href="#cb51-956" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">509154</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span id="cb51-957"><a href="#cb51-957" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">493803</span>] <span class="ot">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span id="cb51-958"><a href="#cb51-958" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">427343</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span id="cb51-959"><a href="#cb51-959" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194958</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span id="cb51-960"><a href="#cb51-960" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">340907</span>] <span class="ot">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span id="cb51-961"><a href="#cb51-961" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194959</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span id="cb51-962"><a href="#cb51-962" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-963"><a href="#cb51-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-964"><a href="#cb51-964" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb51-965"><a href="#cb51-965" aria-hidden="true" tabindex="-1"></a>fp <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> </span>
<span id="cb51-966"><a href="#cb51-966" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb51-967"><a href="#cb51-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-968"><a href="#cb51-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid =</span> taxid_best) <span class="sc">%&gt;%</span></span>
<span id="cb51-969"><a href="#cb51-969" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-970"><a href="#cb51-970" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb51-971"><a href="#cb51-971" aria-hidden="true" tabindex="-1"></a>fp_major_tab <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb51-972"><a href="#cb51-972" aria-hidden="true" tabindex="-1"></a>fp_major_list <span class="ot">&lt;-</span> fp_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb51-973"><a href="#cb51-973" aria-hidden="true" tabindex="-1"></a>fp_major <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb51-974"><a href="#cb51-974" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-975"><a href="#cb51-975" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb51-976"><a href="#cb51-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb51-977"><a href="#cb51-977" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_major_list),</span>
<span id="cb51-978"><a href="#cb51-978" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb51-979"><a href="#cb51-979" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb51-980"><a href="#cb51-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-981"><a href="#cb51-981" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-982"><a href="#cb51-982" aria-hidden="true" tabindex="-1"></a>g_fp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb51-983"><a href="#cb51-983" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-984"><a href="#cb51-984" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb51-985"><a href="#cb51-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-986"><a href="#cb51-986" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-987"><a href="#cb51-987" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_viral, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb51-988"><a href="#cb51-988" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb51-989"><a href="#cb51-989" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb51-990"><a href="#cb51-990" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb51-991"><a href="#cb51-991" aria-hidden="true" tabindex="-1"></a>g_fp</span>
<span id="cb51-992"><a href="#cb51-992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-993"><a href="#cb51-993" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-994"><a href="#cb51-994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-997"><a href="#cb51-997" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-998"><a href="#cb51-998" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ptt-blast-hits</span></span>
<span id="cb51-999"><a href="#cb51-999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1000"><a href="#cb51-1000" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb51-1001"><a href="#cb51-1001" aria-hidden="true" tabindex="-1"></a>ref_taxid_ptt <span class="ot">&lt;-</span> <span class="dv">2233536</span></span>
<span id="cb51-1002"><a href="#cb51-1002" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb51-1003"><a href="#cb51-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1004"><a href="#cb51-1004" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb51-1005"><a href="#cb51-1005" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb51-1006"><a href="#cb51-1006" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-1007"><a href="#cb51-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1008"><a href="#cb51-1008" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb51-1009"><a href="#cb51-1009" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb51-1010"><a href="#cb51-1010" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb51-1011"><a href="#cb51-1011" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb51-1012"><a href="#cb51-1012" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span id="cb51-1013"><a href="#cb51-1013" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span id="cb51-1014"><a href="#cb51-1014" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span id="cb51-1015"><a href="#cb51-1015" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">187126</span>, <span class="st">"Nesoenas mayeri"</span></span>
<span id="cb51-1016"><a href="#cb51-1016" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb51-1017"><a href="#cb51-1017" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> tax_names_new <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> staxid <span class="sc">%in%</span> tax_names<span class="sc">$</span>staxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1018"><a href="#cb51-1018" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(tax_names) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(staxid)</span>
<span id="cb51-1019"><a href="#cb51-1019" aria-hidden="true" tabindex="-1"></a>ref_name_ptt <span class="ot">&lt;-</span> tax_names <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> ref_taxid_ptt) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb51-1020"><a href="#cb51-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1021"><a href="#cb51-1021" aria-hidden="true" tabindex="-1"></a><span class="co"># Get major matches</span></span>
<span id="cb51-1022"><a href="#cb51-1022" aria-hidden="true" tabindex="-1"></a>mrg_staxid <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid_best <span class="sc">==</span> ref_taxid_ptt) <span class="sc">%&gt;%</span></span>
<span id="cb51-1023"><a href="#cb51-1023" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(highscore, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>())</span>
<span id="cb51-1024"><a href="#cb51-1024" aria-hidden="true" tabindex="-1"></a>fp_staxid <span class="ot">&lt;-</span> mrg_staxid <span class="sc">%&gt;%</span></span>
<span id="cb51-1025"><a href="#cb51-1025" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1026"><a href="#cb51-1026" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1027"><a href="#cb51-1027" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb51-1028"><a href="#cb51-1028" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid_best=</span>staxid), <span class="at">by=</span><span class="st">"taxid_best"</span>)</span>
<span id="cb51-1029"><a href="#cb51-1029" aria-hidden="true" tabindex="-1"></a>fp_staxid_count <span class="ot">&lt;-</span> fp_staxid <span class="sc">%&gt;%</span></span>
<span id="cb51-1030"><a href="#cb51-1030" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, </span>
<span id="cb51-1031"><a href="#cb51-1031" aria-hidden="true" tabindex="-1"></a>           taxid_best, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb51-1032"><a href="#cb51-1032" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span></span>
<span id="cb51-1033"><a href="#cb51-1033" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best, name) <span class="sc">%&gt;%</span></span>
<span id="cb51-1034"><a href="#cb51-1034" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb51-1035"><a href="#cb51-1035" aria-hidden="true" tabindex="-1"></a>fp_staxid_count_major <span class="ot">&lt;-</span> fp_staxid_count <span class="sc">%&gt;%</span></span>
<span id="cb51-1036"><a href="#cb51-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1037"><a href="#cb51-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb51-1038"><a href="#cb51-1038" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, </span>
<span id="cb51-1039"><a href="#cb51-1039" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb51-1040"><a href="#cb51-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1041"><a href="#cb51-1041" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1042"><a href="#cb51-1042" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_staxid_count_major, <span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span> </span>
<span id="cb51-1043"><a href="#cb51-1043" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb51-1044"><a href="#cb51-1044" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(status_display<span class="sc">~</span>score_display, <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb51-1045"><a href="#cb51-1045" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb51-1046"><a href="#cb51-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1047"><a href="#cb51-1047" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-1048"><a href="#cb51-1048" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_ptt, <span class="st">" (taxid "</span>, ref_taxid_ptt, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb51-1049"><a href="#cb51-1049" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb51-1050"><a href="#cb51-1050" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-1051"><a href="#cb51-1051" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1052"><a href="#cb51-1052" aria-hidden="true" tabindex="-1"></a>g</span>
<span id="cb51-1053"><a href="#cb51-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1054"><a href="#cb51-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1055"><a href="#cb51-1055" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: overall relative abundance</span></span>
<span id="cb51-1056"><a href="#cb51-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1059"><a href="#cb51-1059" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1060"><a href="#cb51-1060" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb51-1061"><a href="#cb51-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1062"><a href="#cb51-1062" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb51-1063"><a href="#cb51-1063" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb51-1064"><a href="#cb51-1064" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, date_alias, city, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb51-1065"><a href="#cb51-1065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1066"><a href="#cb51-1066" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb51-1067"><a href="#cb51-1067" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb51-1068"><a href="#cb51-1068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb51-1069"><a href="#cb51-1069" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1070"><a href="#cb51-1070" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb51-1071"><a href="#cb51-1071" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1072"><a href="#cb51-1072" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>))</span>
<span id="cb51-1073"><a href="#cb51-1073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1074"><a href="#cb51-1074" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb51-1075"><a href="#cb51-1075" aria-hidden="true" tabindex="-1"></a>read_counts_city <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1076"><a href="#cb51-1076" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb51-1077"><a href="#cb51-1077" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1078"><a href="#cb51-1078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span> <span class="st">"All samples"</span>, <span class="at">date_alias =</span> <span class="st">"All dates"</span>)</span>
<span id="cb51-1079"><a href="#cb51-1079" aria-hidden="true" tabindex="-1"></a>read_counts_total <span class="ot">&lt;-</span> read_counts_city <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, date_alias) <span class="sc">%&gt;%</span></span>
<span id="cb51-1080"><a href="#cb51-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb51-1081"><a href="#cb51-1081" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1082"><a href="#cb51-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">city =</span> <span class="st">"All cities"</span>)</span>
<span id="cb51-1083"><a href="#cb51-1083" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts_city <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1084"><a href="#cb51-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(read_counts_total) <span class="sc">%&gt;%</span></span>
<span id="cb51-1085"><a href="#cb51-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb51-1086"><a href="#cb51-1086" aria-hidden="true" tabindex="-1"></a>         <span class="at">city =</span> <span class="fu">fct_inorder</span>(city))</span>
<span id="cb51-1087"><a href="#cb51-1087" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1088"><a href="#cb51-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1089"><a href="#cb51-1089" aria-hidden="true" tabindex="-1"></a>Applying a disjunctive cutoff at S=20 identifies 23,191 read pairs as human-viral. This gives an overall relative HV abundance of $1.73 \times 10^{-5}$.</span>
<span id="cb51-1090"><a href="#cb51-1090" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1093"><a href="#cb51-1093" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1094"><a href="#cb51-1094" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-ra</span></span>
<span id="cb51-1095"><a href="#cb51-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb51-1096"><a href="#cb51-1096" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb51-1097"><a href="#cb51-1097" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>city, <span class="at">color=</span>city)) <span class="sc">+</span></span>
<span id="cb51-1098"><a href="#cb51-1098" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_hv)) <span class="sc">+</span></span>
<span id="cb51-1099"><a href="#cb51-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1100"><a href="#cb51-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Collection Date"</span>) <span class="sc">+</span></span>
<span id="cb51-1101"><a href="#cb51-1101" aria-hidden="true" tabindex="-1"></a>  <span class="co">#facet_grid(.~sample_type, scales = "free", space = "free_x") +</span></span>
<span id="cb51-1102"><a href="#cb51-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span> theme_rotate</span>
<span id="cb51-1103"><a href="#cb51-1103" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb51-1104"><a href="#cb51-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1105"><a href="#cb51-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1106"><a href="#cb51-1106" aria-hidden="true" tabindex="-1"></a>This is lower than for DNA reads from other air-sampling datasets I've analyzed, but not drastically so:</span>
<span id="cb51-1107"><a href="#cb51-1107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1110"><a href="#cb51-1110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1111"><a href="#cb51-1111" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-hv-past</span></span>
<span id="cb51-1112"><a href="#cb51-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1113"><a href="#cb51-1113" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate past RA values</span></span>
<span id="cb51-1114"><a href="#cb51-1114" aria-hidden="true" tabindex="-1"></a>ra_past <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb51-1115"><a href="#cb51-1115" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1116"><a href="#cb51-1116" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1117"><a href="#cb51-1117" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1118"><a href="#cb51-1118" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1119"><a href="#cb51-1119" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1120"><a href="#cb51-1120" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb51-1121"><a href="#cb51-1121" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1122"><a href="#cb51-1122" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb51-1123"><a href="#cb51-1123" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1124"><a href="#cb51-1124" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1125"><a href="#cb51-1125" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb51-1126"><a href="#cb51-1126" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span id="cb51-1127"><a href="#cb51-1127" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb51-1128"><a href="#cb51-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1129"><a href="#cb51-1129" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate new RA values</span></span>
<span id="cb51-1130"><a href="#cb51-1130" aria-hidden="true" tabindex="-1"></a>ra_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb51-1131"><a href="#cb51-1131" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>)</span>
<span id="cb51-1132"><a href="#cb51-1132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1133"><a href="#cb51-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1134"><a href="#cb51-1134" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1135"><a href="#cb51-1135" aria-hidden="true" tabindex="-1"></a>scale_color_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb51-1136"><a href="#cb51-1136" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb51-1137"><a href="#cb51-1137" aria-hidden="true" tabindex="-1"></a>ra_comp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ra_past, ra_new) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">fct_inorder</span>(dataset))</span>
<span id="cb51-1138"><a href="#cb51-1138" aria-hidden="true" tabindex="-1"></a>g_ra_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ra_comp, <span class="fu">aes</span>(<span class="at">y=</span>dataset, <span class="at">x=</span>ra, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb51-1139"><a href="#cb51-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb51-1140"><a href="#cb51-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb51-1141"><a href="#cb51-1141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1142"><a href="#cb51-1142" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb51-1143"><a href="#cb51-1143" aria-hidden="true" tabindex="-1"></a>g_ra_comp</span>
<span id="cb51-1144"><a href="#cb51-1144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1145"><a href="#cb51-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1146"><a href="#cb51-1146" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: taxonomy and composition</span></span>
<span id="cb51-1147"><a href="#cb51-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1148"><a href="#cb51-1148" aria-hidden="true" tabindex="-1"></a>At the family level, most samples across all cities are dominated by *Papillomaviridae*, *Herpesviridae*, *Anelloviridae*, *Polyomaviridae, and to a lesser extent* *Poxviridae*:</span>
<span id="cb51-1149"><a href="#cb51-1149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1152"><a href="#cb51-1152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1153"><a href="#cb51-1153" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb51-1154"><a href="#cb51-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1155"><a href="#cb51-1155" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb51-1156"><a href="#cb51-1156" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb51-1157"><a href="#cb51-1157" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb51-1158"><a href="#cb51-1158" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb51-1159"><a href="#cb51-1159" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">585893</span>] <span class="ot">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span id="cb51-1160"><a href="#cb51-1160" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333922</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span id="cb51-1161"><a href="#cb51-1161" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334207</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span id="cb51-1162"><a href="#cb51-1162" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">369960</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span id="cb51-1163"><a href="#cb51-1163" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333924</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span id="cb51-1164"><a href="#cb51-1164" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687329</span>] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb51-1165"><a href="#cb51-1165" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">325455</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span id="cb51-1166"><a href="#cb51-1166" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333750</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span id="cb51-1167"><a href="#cb51-1167" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb51-1168"><a href="#cb51-1168" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334202</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span id="cb51-1169"><a href="#cb51-1169" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">197911</span>] <span class="ot">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span id="cb51-1170"><a href="#cb51-1170" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">186938</span>] <span class="ot">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span id="cb51-1171"><a href="#cb51-1171" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333926</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span id="cb51-1172"><a href="#cb51-1172" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337051</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span id="cb51-1173"><a href="#cb51-1173" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337043</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span id="cb51-1174"><a href="#cb51-1174" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb51-1175"><a href="#cb51-1175" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334204</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span id="cb51-1176"><a href="#cb51-1176" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334208</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span id="cb51-1177"><a href="#cb51-1177" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333928</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span id="cb51-1178"><a href="#cb51-1178" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337039</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span id="cb51-1179"><a href="#cb51-1179" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333929</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span id="cb51-1180"><a href="#cb51-1180" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337042</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span id="cb51-1181"><a href="#cb51-1181" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334203</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span id="cb51-1182"><a href="#cb51-1182" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333757</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span id="cb51-1183"><a href="#cb51-1183" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337050</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span id="cb51-1184"><a href="#cb51-1184" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333767</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span id="cb51-1185"><a href="#cb51-1185" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333754</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span id="cb51-1186"><a href="#cb51-1186" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687363</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span id="cb51-1187"><a href="#cb51-1187" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687342</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span id="cb51-1188"><a href="#cb51-1188" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687359</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span id="cb51-1189"><a href="#cb51-1189" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194441</span>] <span class="ot">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span id="cb51-1190"><a href="#cb51-1190" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334209</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span id="cb51-1191"><a href="#cb51-1191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1192"><a href="#cb51-1192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1193"><a href="#cb51-1193" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb51-1194"><a href="#cb51-1194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1195"><a href="#cb51-1195" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb51-1196"><a href="#cb51-1196" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb51-1197"><a href="#cb51-1197" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb51-1198"><a href="#cb51-1198" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb51-1199"><a href="#cb51-1199" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb51-1200"><a href="#cb51-1200" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb51-1201"><a href="#cb51-1201" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb51-1202"><a href="#cb51-1202" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb51-1203"><a href="#cb51-1203" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb51-1204"><a href="#cb51-1204" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb51-1205"><a href="#cb51-1205" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb51-1206"><a href="#cb51-1206" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb51-1207"><a href="#cb51-1207" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb51-1208"><a href="#cb51-1208" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb51-1209"><a href="#cb51-1209" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb51-1210"><a href="#cb51-1210" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1211"><a href="#cb51-1211" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb51-1212"><a href="#cb51-1212" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb51-1213"><a href="#cb51-1213" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb51-1214"><a href="#cb51-1214" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb51-1215"><a href="#cb51-1215" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb51-1216"><a href="#cb51-1216" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb51-1217"><a href="#cb51-1217" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb51-1218"><a href="#cb51-1218" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb51-1219"><a href="#cb51-1219" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb51-1220"><a href="#cb51-1220" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb51-1221"><a href="#cb51-1221" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb51-1222"><a href="#cb51-1222" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb51-1223"><a href="#cb51-1223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb51-1224"><a href="#cb51-1224" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-1225"><a href="#cb51-1225" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb51-1226"><a href="#cb51-1226" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb51-1227"><a href="#cb51-1227" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb51-1228"><a href="#cb51-1228" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1229"><a href="#cb51-1229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1232"><a href="#cb51-1232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1233"><a href="#cb51-1233" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb51-1234"><a href="#cb51-1234" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1235"><a href="#cb51-1235" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-1236"><a href="#cb51-1236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1237"><a href="#cb51-1237" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb51-1238"><a href="#cb51-1238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1239"><a href="#cb51-1239" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb51-1240"><a href="#cb51-1240" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1241"><a href="#cb51-1241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1242"><a href="#cb51-1242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1243"><a href="#cb51-1243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1244"><a href="#cb51-1244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1245"><a href="#cb51-1245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1246"><a href="#cb51-1246" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1247"><a href="#cb51-1247" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1248"><a href="#cb51-1248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1249"><a href="#cb51-1249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb51-1250"><a href="#cb51-1250" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1251"><a href="#cb51-1251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1252"><a href="#cb51-1252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1253"><a href="#cb51-1253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb51-1254"><a href="#cb51-1254" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1255"><a href="#cb51-1255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1256"><a href="#cb51-1256" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1257"><a href="#cb51-1257" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1258"><a href="#cb51-1258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1259"><a href="#cb51-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1260"><a href="#cb51-1260" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1261"><a href="#cb51-1261" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1262"><a href="#cb51-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1263"><a href="#cb51-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1264"><a href="#cb51-1264" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1265"><a href="#cb51-1265" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1266"><a href="#cb51-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb51-1267"><a href="#cb51-1267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1268"><a href="#cb51-1268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb51-1269"><a href="#cb51-1269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1270"><a href="#cb51-1270" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb51-1271"><a href="#cb51-1271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1272"><a href="#cb51-1272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1273"><a href="#cb51-1273" aria-hidden="true" tabindex="-1"></a>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads.</span>
<span id="cb51-1274"><a href="#cb51-1274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1275"><a href="#cb51-1275" aria-hidden="true" tabindex="-1"></a>As usual, *Papillomaviridae* reads are divided among many different viral species. In this case, Betapapillomavirus 1 and 2 are the most prevalent across samples, but many other alpha-, beta-, gamma-, and mupapillomaviruses are highly prevalent in at least some samples.</span>
<span id="cb51-1276"><a href="#cb51-1276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1279"><a href="#cb51-1279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1280"><a href="#cb51-1280" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-papilloma</span></span>
<span id="cb51-1281"><a href="#cb51-1281" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1282"><a href="#cb51-1282" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-1283"><a href="#cb51-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1284"><a href="#cb51-1284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1285"><a href="#cb51-1285" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb51-1286"><a href="#cb51-1286" aria-hidden="true" tabindex="-1"></a>taxid_papilloma <span class="ot">&lt;-</span> <span class="dv">151340</span></span>
<span id="cb51-1287"><a href="#cb51-1287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1288"><a href="#cb51-1288" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of Papillomaviridae reads</span></span>
<span id="cb51-1289"><a href="#cb51-1289" aria-hidden="true" tabindex="-1"></a>papilloma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma) <span class="sc">%&gt;%</span></span>
<span id="cb51-1290"><a href="#cb51-1290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1291"><a href="#cb51-1291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb51-1292"><a href="#cb51-1292" aria-hidden="true" tabindex="-1"></a>papilloma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1293"><a href="#cb51-1293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma, sample <span class="sc">%in%</span> papilloma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-1294"><a href="#cb51-1294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb51-1295"><a href="#cb51-1295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1296"><a href="#cb51-1296" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each Papillomaviridae species</span></span>
<span id="cb51-1297"><a href="#cb51-1297" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb51-1298"><a href="#cb51-1298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> papilloma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb51-1299"><a href="#cb51-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1300"><a href="#cb51-1300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1301"><a href="#cb51-1301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1302"><a href="#cb51-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_papilloma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1303"><a href="#cb51-1303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1304"><a href="#cb51-1304" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1305"><a href="#cb51-1305" aria-hidden="true" tabindex="-1"></a>papilloma_species_major_tab <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1306"><a href="#cb51-1306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">==</span> <span class="fu">max</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1307"><a href="#cb51-1307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1308"><a href="#cb51-1308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1309"><a href="#cb51-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb51-1310"><a href="#cb51-1310" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_major <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1311"><a href="#cb51-1311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> papilloma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb51-1312"><a href="#cb51-1312" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1313"><a href="#cb51-1313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1314"><a href="#cb51-1314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_papilloma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb51-1315"><a href="#cb51-1315" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_papilloma =</span> <span class="fu">sum</span>(p_reads_papilloma), </span>
<span id="cb51-1316"><a href="#cb51-1316" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1317"><a href="#cb51-1317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1318"><a href="#cb51-1318" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(papilloma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1319"><a href="#cb51-1319" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_display <span class="ot">&lt;-</span> papilloma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1320"><a href="#cb51-1320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_papilloma, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1321"><a href="#cb51-1321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1322"><a href="#cb51-1322" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1323"><a href="#cb51-1323" aria-hidden="true" tabindex="-1"></a>g_papilloma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1324"><a href="#cb51-1324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>papilloma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1325"><a href="#cb51-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Papillomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1326"><a href="#cb51-1326" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1327"><a href="#cb51-1327" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1328"><a href="#cb51-1328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb51-1329"><a href="#cb51-1329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Papillomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1330"><a href="#cb51-1330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb51-1331"><a href="#cb51-1331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1332"><a href="#cb51-1332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1333"><a href="#cb51-1333" aria-hidden="true" tabindex="-1"></a>g_papilloma_species</span>
<span id="cb51-1334"><a href="#cb51-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1335"><a href="#cb51-1335" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb51-1336"><a href="#cb51-1336" aria-hidden="true" tabindex="-1"></a>papilloma_species_collate <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1337"><a href="#cb51-1337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_papilloma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1338"><a href="#cb51-1338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb51-1339"><a href="#cb51-1339" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1340"><a href="#cb51-1340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1341"><a href="#cb51-1341" aria-hidden="true" tabindex="-1"></a>In terms of total reads across samples, herpesviruses are dominated by Epstein-Barr virus (Human gammaherpesvirus 4), HSV-1 (Human alphaherpesvirus 1), and human cytomegalovirus (Human betaherpesvirus 5). However, numerous other herpesviruses are also present.</span>
<span id="cb51-1342"><a href="#cb51-1342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1345"><a href="#cb51-1345" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1346"><a href="#cb51-1346" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-herpes</span></span>
<span id="cb51-1347"><a href="#cb51-1347" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1348"><a href="#cb51-1348" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-1349"><a href="#cb51-1349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1350"><a href="#cb51-1350" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb51-1351"><a href="#cb51-1351" aria-hidden="true" tabindex="-1"></a>taxid_herpes <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"Herpesviridae"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)</span>
<span id="cb51-1352"><a href="#cb51-1352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1353"><a href="#cb51-1353" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of herpesviridae reads</span></span>
<span id="cb51-1354"><a href="#cb51-1354" aria-hidden="true" tabindex="-1"></a>herpes_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes) <span class="sc">%&gt;%</span></span>
<span id="cb51-1355"><a href="#cb51-1355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1356"><a href="#cb51-1356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb51-1357"><a href="#cb51-1357" aria-hidden="true" tabindex="-1"></a>herpes_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1358"><a href="#cb51-1358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes, sample <span class="sc">%in%</span> herpes_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-1359"><a href="#cb51-1359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb51-1360"><a href="#cb51-1360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1361"><a href="#cb51-1361" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae species</span></span>
<span id="cb51-1362"><a href="#cb51-1362" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb51-1363"><a href="#cb51-1363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb51-1364"><a href="#cb51-1364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1365"><a href="#cb51-1365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1366"><a href="#cb51-1366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1367"><a href="#cb51-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1368"><a href="#cb51-1368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1369"><a href="#cb51-1369" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1370"><a href="#cb51-1370" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1371"><a href="#cb51-1371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1372"><a href="#cb51-1372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1373"><a href="#cb51-1373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1374"><a href="#cb51-1374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb51-1375"><a href="#cb51-1375" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1376"><a href="#cb51-1376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb51-1377"><a href="#cb51-1377" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1378"><a href="#cb51-1378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1379"><a href="#cb51-1379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb51-1380"><a href="#cb51-1380" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb51-1381"><a href="#cb51-1381" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1382"><a href="#cb51-1382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1383"><a href="#cb51-1383" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1384"><a href="#cb51-1384" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1385"><a href="#cb51-1385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1386"><a href="#cb51-1386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1387"><a href="#cb51-1387" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1388"><a href="#cb51-1388" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1389"><a href="#cb51-1389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1390"><a href="#cb51-1390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1391"><a href="#cb51-1391" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1392"><a href="#cb51-1392" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1393"><a href="#cb51-1393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb51-1394"><a href="#cb51-1394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1395"><a href="#cb51-1395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb51-1396"><a href="#cb51-1396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1397"><a href="#cb51-1397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1398"><a href="#cb51-1398" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb51-1399"><a href="#cb51-1399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1400"><a href="#cb51-1400" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb51-1401"><a href="#cb51-1401" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1402"><a href="#cb51-1402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1403"><a href="#cb51-1403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb51-1404"><a href="#cb51-1404" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1405"><a href="#cb51-1405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1406"><a href="#cb51-1406" aria-hidden="true" tabindex="-1"></a>In sharp contrast to the above, my pipeline classifies the great majority of anellovirus reads in all samples into a single species, torque teno virus. Looking online, it looks like there are a lot of "torque teno viruses" within *Anelloviridae* – for example, Wikipedia says that the genus *Alphatorquevirus* contains <span class="sc">\&gt;</span>20 numbered torque teno viruses – so I'm not sure exactly which virus this refers to.</span>
<span id="cb51-1407"><a href="#cb51-1407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1410"><a href="#cb51-1410" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1411"><a href="#cb51-1411" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-anello</span></span>
<span id="cb51-1412"><a href="#cb51-1412" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1413"><a href="#cb51-1413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1414"><a href="#cb51-1414" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb51-1415"><a href="#cb51-1415" aria-hidden="true" tabindex="-1"></a>taxid_anello <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"Anelloviridae"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)</span>
<span id="cb51-1416"><a href="#cb51-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1417"><a href="#cb51-1417" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of anelloviridae reads</span></span>
<span id="cb51-1418"><a href="#cb51-1418" aria-hidden="true" tabindex="-1"></a>anello_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_anello) <span class="sc">%&gt;%</span></span>
<span id="cb51-1419"><a href="#cb51-1419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1420"><a href="#cb51-1420" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb51-1421"><a href="#cb51-1421" aria-hidden="true" tabindex="-1"></a>anello_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1422"><a href="#cb51-1422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_anello, sample <span class="sc">%in%</span> anello_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-1423"><a href="#cb51-1423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb51-1424"><a href="#cb51-1424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1425"><a href="#cb51-1425" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each anelloviridae species</span></span>
<span id="cb51-1426"><a href="#cb51-1426" aria-hidden="true" tabindex="-1"></a>anello_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb51-1427"><a href="#cb51-1427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> anello_ids) <span class="sc">%&gt;%</span></span>
<span id="cb51-1428"><a href="#cb51-1428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1429"><a href="#cb51-1429" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1430"><a href="#cb51-1430" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1431"><a href="#cb51-1431" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_anello =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1432"><a href="#cb51-1432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1433"><a href="#cb51-1433" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1434"><a href="#cb51-1434" aria-hidden="true" tabindex="-1"></a>anello_species_major_tab <span class="ot">&lt;-</span> anello_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1435"><a href="#cb51-1435" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_anello <span class="sc">==</span> <span class="fu">max</span>(p_reads_anello)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1436"><a href="#cb51-1436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1437"><a href="#cb51-1437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_anello)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1438"><a href="#cb51-1438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_anello <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb51-1439"><a href="#cb51-1439" aria-hidden="true" tabindex="-1"></a>anello_species_counts_major <span class="ot">&lt;-</span> anello_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1440"><a href="#cb51-1440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> anello_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb51-1441"><a href="#cb51-1441" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1442"><a href="#cb51-1442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1443"><a href="#cb51-1443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_anello =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb51-1444"><a href="#cb51-1444" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_anello =</span> <span class="fu">sum</span>(p_reads_anello), </span>
<span id="cb51-1445"><a href="#cb51-1445" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1446"><a href="#cb51-1446" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1447"><a href="#cb51-1447" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(anello_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1448"><a href="#cb51-1448" aria-hidden="true" tabindex="-1"></a>anello_species_counts_display <span class="ot">&lt;-</span> anello_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1449"><a href="#cb51-1449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_anello, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1450"><a href="#cb51-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1451"><a href="#cb51-1451" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1452"><a href="#cb51-1452" aria-hidden="true" tabindex="-1"></a>g_anello_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1453"><a href="#cb51-1453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>anello_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1454"><a href="#cb51-1454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Anelloviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1455"><a href="#cb51-1455" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1456"><a href="#cb51-1456" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1457"><a href="#cb51-1457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb51-1458"><a href="#cb51-1458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Anelloviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1459"><a href="#cb51-1459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb51-1460"><a href="#cb51-1460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1461"><a href="#cb51-1461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1462"><a href="#cb51-1462" aria-hidden="true" tabindex="-1"></a>g_anello_species</span>
<span id="cb51-1463"><a href="#cb51-1463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1464"><a href="#cb51-1464" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb51-1465"><a href="#cb51-1465" aria-hidden="true" tabindex="-1"></a>anello_species_collate <span class="ot">&lt;-</span> anello_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1466"><a href="#cb51-1466" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_anello), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1467"><a href="#cb51-1467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb51-1468"><a href="#cb51-1468" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1469"><a href="#cb51-1469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1470"><a href="#cb51-1470" aria-hidden="true" tabindex="-1"></a>Polyomaviruses are intermediate; most viruses are dominated by a single species, *Alphapolyomavirus quintihominis*, but several other viruses in the family are also present.</span>
<span id="cb51-1471"><a href="#cb51-1471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1474"><a href="#cb51-1474" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1475"><a href="#cb51-1475" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-polyoma</span></span>
<span id="cb51-1476"><a href="#cb51-1476" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1477"><a href="#cb51-1477" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb51-1478"><a href="#cb51-1478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1479"><a href="#cb51-1479" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb51-1480"><a href="#cb51-1480" aria-hidden="true" tabindex="-1"></a>taxid_polyoma <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"Polyomaviridae"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)</span>
<span id="cb51-1481"><a href="#cb51-1481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1482"><a href="#cb51-1482" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of polyomaviridae reads</span></span>
<span id="cb51-1483"><a href="#cb51-1483" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of polyomaviridae reads</span></span>
<span id="cb51-1484"><a href="#cb51-1484" aria-hidden="true" tabindex="-1"></a>polyoma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma) <span class="sc">%&gt;%</span></span>
<span id="cb51-1485"><a href="#cb51-1485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1486"><a href="#cb51-1486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb51-1487"><a href="#cb51-1487" aria-hidden="true" tabindex="-1"></a>polyoma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1488"><a href="#cb51-1488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma, sample <span class="sc">%in%</span> polyoma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-1489"><a href="#cb51-1489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb51-1490"><a href="#cb51-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1491"><a href="#cb51-1491" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each polyomaviridae species</span></span>
<span id="cb51-1492"><a href="#cb51-1492" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb51-1493"><a href="#cb51-1493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> polyoma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb51-1494"><a href="#cb51-1494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1495"><a href="#cb51-1495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1496"><a href="#cb51-1496" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1497"><a href="#cb51-1497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_polyoma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1498"><a href="#cb51-1498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1499"><a href="#cb51-1499" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1500"><a href="#cb51-1500" aria-hidden="true" tabindex="-1"></a>polyoma_species_major_tab <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1501"><a href="#cb51-1501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">==</span> <span class="fu">max</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1502"><a href="#cb51-1502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1503"><a href="#cb51-1503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1504"><a href="#cb51-1504" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb51-1505"><a href="#cb51-1505" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_major <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1506"><a href="#cb51-1506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> polyoma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb51-1507"><a href="#cb51-1507" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1508"><a href="#cb51-1508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1509"><a href="#cb51-1509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_polyoma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb51-1510"><a href="#cb51-1510" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_polyoma =</span> <span class="fu">sum</span>(p_reads_polyoma), </span>
<span id="cb51-1511"><a href="#cb51-1511" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1512"><a href="#cb51-1512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1513"><a href="#cb51-1513" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(polyoma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1514"><a href="#cb51-1514" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_display <span class="ot">&lt;-</span> polyoma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1515"><a href="#cb51-1515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_polyoma, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1516"><a href="#cb51-1516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1517"><a href="#cb51-1517" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1518"><a href="#cb51-1518" aria-hidden="true" tabindex="-1"></a>g_polyoma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1519"><a href="#cb51-1519" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>polyoma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1520"><a href="#cb51-1520" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Polyomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1521"><a href="#cb51-1521" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1522"><a href="#cb51-1522" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1523"><a href="#cb51-1523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb51-1524"><a href="#cb51-1524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Polyomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1525"><a href="#cb51-1525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb51-1526"><a href="#cb51-1526" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1527"><a href="#cb51-1527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1528"><a href="#cb51-1528" aria-hidden="true" tabindex="-1"></a>g_polyoma_species</span>
<span id="cb51-1529"><a href="#cb51-1529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1530"><a href="#cb51-1530" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb51-1531"><a href="#cb51-1531" aria-hidden="true" tabindex="-1"></a>polyoma_species_collate <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1532"><a href="#cb51-1532" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_polyoma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1533"><a href="#cb51-1533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb51-1534"><a href="#cb51-1534" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1535"><a href="#cb51-1535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1536"><a href="#cb51-1536" aria-hidden="true" tabindex="-1"></a>Finally, poxvirus reads in most samples are dominated by molluscum contagiosum virus (which I expect to be real), followed by Orf virus (which I expect to be fake). These expectations are borne out by BLAST alignments (below).</span>
<span id="cb51-1537"><a href="#cb51-1537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1540"><a href="#cb51-1540" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1541"><a href="#cb51-1541" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-pox</span></span>
<span id="cb51-1542"><a href="#cb51-1542" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb51-1543"><a href="#cb51-1543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1544"><a href="#cb51-1544" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb51-1545"><a href="#cb51-1545" aria-hidden="true" tabindex="-1"></a>taxid_pox <span class="ot">&lt;-</span> viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"Poxviridae"</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)</span>
<span id="cb51-1546"><a href="#cb51-1546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1547"><a href="#cb51-1547" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of poxviridae reads</span></span>
<span id="cb51-1548"><a href="#cb51-1548" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of poxviridae reads</span></span>
<span id="cb51-1549"><a href="#cb51-1549" aria-hidden="true" tabindex="-1"></a>pox_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox) <span class="sc">%&gt;%</span></span>
<span id="cb51-1550"><a href="#cb51-1550" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1551"><a href="#cb51-1551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb51-1552"><a href="#cb51-1552" aria-hidden="true" tabindex="-1"></a>pox_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb51-1553"><a href="#cb51-1553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox, sample <span class="sc">%in%</span> pox_samples) <span class="sc">%&gt;%</span></span>
<span id="cb51-1554"><a href="#cb51-1554" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb51-1555"><a href="#cb51-1555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1556"><a href="#cb51-1556" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each poxviridae species</span></span>
<span id="cb51-1557"><a href="#cb51-1557" aria-hidden="true" tabindex="-1"></a>pox_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb51-1558"><a href="#cb51-1558" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> pox_ids) <span class="sc">%&gt;%</span></span>
<span id="cb51-1559"><a href="#cb51-1559" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1560"><a href="#cb51-1560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1561"><a href="#cb51-1561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1562"><a href="#cb51-1562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_pox =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb51-1563"><a href="#cb51-1563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1564"><a href="#cb51-1564" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb51-1565"><a href="#cb51-1565" aria-hidden="true" tabindex="-1"></a>pox_species_major_tab <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1566"><a href="#cb51-1566" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">==</span> <span class="fu">max</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1567"><a href="#cb51-1567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1568"><a href="#cb51-1568" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1569"><a href="#cb51-1569" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb51-1570"><a href="#cb51-1570" aria-hidden="true" tabindex="-1"></a>pox_species_counts_major <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb51-1571"><a href="#cb51-1571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> pox_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb51-1572"><a href="#cb51-1572" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1573"><a href="#cb51-1573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb51-1574"><a href="#cb51-1574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_pox =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb51-1575"><a href="#cb51-1575" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_pox =</span> <span class="fu">sum</span>(p_reads_pox), </span>
<span id="cb51-1576"><a href="#cb51-1576" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1577"><a href="#cb51-1577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb51-1578"><a href="#cb51-1578" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(pox_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb51-1579"><a href="#cb51-1579" aria-hidden="true" tabindex="-1"></a>pox_species_counts_display <span class="ot">&lt;-</span> pox_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb51-1580"><a href="#cb51-1580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_pox, <span class="at">classification =</span> name_display)</span>
<span id="cb51-1581"><a href="#cb51-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1582"><a href="#cb51-1582" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1583"><a href="#cb51-1583" aria-hidden="true" tabindex="-1"></a>g_pox_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb51-1584"><a href="#cb51-1584" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>pox_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb51-1585"><a href="#cb51-1585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Poxviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb51-1586"><a href="#cb51-1586" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb51-1587"><a href="#cb51-1587" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb51-1588"><a href="#cb51-1588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb51-1589"><a href="#cb51-1589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Poxviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb51-1590"><a href="#cb51-1590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb51-1591"><a href="#cb51-1591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb51-1592"><a href="#cb51-1592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1593"><a href="#cb51-1593" aria-hidden="true" tabindex="-1"></a>g_pox_species</span>
<span id="cb51-1594"><a href="#cb51-1594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1595"><a href="#cb51-1595" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb51-1596"><a href="#cb51-1596" aria-hidden="true" tabindex="-1"></a>pox_species_collate <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1597"><a href="#cb51-1597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_pox), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1598"><a href="#cb51-1598" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb51-1599"><a href="#cb51-1599" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1600"><a href="#cb51-1600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1603"><a href="#cb51-1603" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1604"><a href="#cb51-1604" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-blast-hits</span></span>
<span id="cb51-1605"><a href="#cb51-1605" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb51-1606"><a href="#cb51-1606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1607"><a href="#cb51-1607" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb51-1608"><a href="#cb51-1608" aria-hidden="true" tabindex="-1"></a>ref_taxids_hv <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10279</span>, <span class="dv">10258</span>)</span>
<span id="cb51-1609"><a href="#cb51-1609" aria-hidden="true" tabindex="-1"></a>ref_names_hv <span class="ot">&lt;-</span> <span class="fu">sapply</span>(ref_taxids_hv, <span class="cf">function</span>(x) viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> x) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> first)</span>
<span id="cb51-1610"><a href="#cb51-1610" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb51-1611"><a href="#cb51-1611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1612"><a href="#cb51-1612" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb51-1613"><a href="#cb51-1613" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb51-1614"><a href="#cb51-1614" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-1615"><a href="#cb51-1615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1616"><a href="#cb51-1616" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb51-1617"><a href="#cb51-1617" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb51-1618"><a href="#cb51-1618" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb51-1619"><a href="#cb51-1619" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb51-1620"><a href="#cb51-1620" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">257877</span>, <span class="st">"Macaca thibetana thibetana"</span>,</span>
<span id="cb51-1621"><a href="#cb51-1621" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">256321</span>, <span class="st">"Lentiviral transfer vector pHsCXW"</span>,</span>
<span id="cb51-1622"><a href="#cb51-1622" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">419242</span>, <span class="st">"Shuttle vector pLvCmvMYOCDHA"</span>,</span>
<span id="cb51-1623"><a href="#cb51-1623" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">419243</span>, <span class="st">"Shuttle vector pLvCmvLacZ"</span>,</span>
<span id="cb51-1624"><a href="#cb51-1624" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">421868</span>, <span class="st">"Cloning vector pLvCmvLacZ.Gfp"</span>,</span>
<span id="cb51-1625"><a href="#cb51-1625" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">421869</span>, <span class="st">"Cloning vector pLvCmvMyocardin.Gfp"</span>,</span>
<span id="cb51-1626"><a href="#cb51-1626" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">426303</span>, <span class="st">"Lentiviral vector pNL-GFP-RRE(SA)"</span>,</span>
<span id="cb51-1627"><a href="#cb51-1627" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">436015</span>, <span class="st">"Lentiviral transfer vector pFTMGW"</span>,</span>
<span id="cb51-1628"><a href="#cb51-1628" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">454257</span>, <span class="st">"Shuttle vector pLvCmvMYOCD2aHA"</span>,</span>
<span id="cb51-1629"><a href="#cb51-1629" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">476184</span>, <span class="st">"Shuttle vector pLV.mMyoD::ERT2.eGFP"</span>,</span>
<span id="cb51-1630"><a href="#cb51-1630" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">476185</span>, <span class="st">"Shuttle vector pLV.hMyoD.eGFP"</span>,</span>
<span id="cb51-1631"><a href="#cb51-1631" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">591936</span>, <span class="st">"Piliocolobus tephrosceles"</span>,</span>
<span id="cb51-1632"><a href="#cb51-1632" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">627481</span>, <span class="st">"Lentiviral transfer vector pFTM3GW"</span>,</span>
<span id="cb51-1633"><a href="#cb51-1633" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">680261</span>, <span class="st">"Self-inactivating lentivirus vector pLV.C-EF1a.cyt-bGal.dCpG"</span>,</span>
<span id="cb51-1634"><a href="#cb51-1634" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">2952778</span>, <span class="st">"Expression vector pLV[Exp]-EGFP:T2A:Puro-EF1A"</span>,</span>
<span id="cb51-1635"><a href="#cb51-1635" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3022699</span>, <span class="st">"Vector PAS_122122"</span>,</span>
<span id="cb51-1636"><a href="#cb51-1636" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3025913</span>, <span class="st">"Vector pSIN-WP-mPGK-GDNF"</span>,</span>
<span id="cb51-1637"><a href="#cb51-1637" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3105863</span>, <span class="st">"Vector pLKO.1-ZsGreen1"</span>,</span>
<span id="cb51-1638"><a href="#cb51-1638" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3105864</span>, <span class="st">"Vector pLKO.1-ZsGreen1 mouse Wfs1 shRNA"</span>,</span>
<span id="cb51-1639"><a href="#cb51-1639" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3108001</span>, <span class="st">"Cloning vector pLVSIN-CMV_Neo_v4.0"</span>,</span>
<span id="cb51-1640"><a href="#cb51-1640" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3109234</span>, <span class="st">"Vector pTwist+Kan+High"</span>,</span>
<span id="cb51-1641"><a href="#cb51-1641" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117662</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L"</span>,</span>
<span id="cb51-1642"><a href="#cb51-1642" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117663</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;P301L:T2A:mRuby3"</span>,</span>
<span id="cb51-1643"><a href="#cb51-1643" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117664</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;hMAPT[NM_005910.6](ns):T2A:mRuby3"</span>,</span>
<span id="cb51-1644"><a href="#cb51-1644" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117665</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3"</span>,</span>
<span id="cb51-1645"><a href="#cb51-1645" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117666</span>, <span class="st">"Cloning vector pLV[Exp]-CBA&gt;mRuby3/NFAT3 fusion protein"</span>,</span>
<span id="cb51-1646"><a href="#cb51-1646" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3117667</span>, <span class="st">"Cloning vector pLV[Exp]-Neo-mPGK&gt;{EGFP-hSEPT6}"</span>,</span>
<span id="cb51-1647"><a href="#cb51-1647" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">438045</span>, <span class="st">"Xenotropic MuLV-related virus"</span>,</span>
<span id="cb51-1648"><a href="#cb51-1648" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">447135</span>, <span class="st">"Myodes glareolus"</span>,</span>
<span id="cb51-1649"><a href="#cb51-1649" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">590745</span>, <span class="st">"Mus musculus mobilized endogenous polytropic provirus"</span>,</span>
<span id="cb51-1650"><a href="#cb51-1650" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">181858</span>, <span class="st">"Murine AIDS virus-related provirus"</span>,</span>
<span id="cb51-1651"><a href="#cb51-1651" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">356663</span>, <span class="st">"Xenotropic MuLV-related virus VP35"</span>,</span>
<span id="cb51-1652"><a href="#cb51-1652" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">356664</span>, <span class="st">"Xenotropic MuLV-related virus VP42"</span>,</span>
<span id="cb51-1653"><a href="#cb51-1653" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">373193</span>, <span class="st">"Xenotropic MuLV-related virus VP62"</span>,</span>
<span id="cb51-1654"><a href="#cb51-1654" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">286419</span>, <span class="st">"Canis lupus dingo"</span>,</span>
<span id="cb51-1655"><a href="#cb51-1655" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">415978</span>, <span class="st">"Sus scrofa scrofa"</span>,</span>
<span id="cb51-1656"><a href="#cb51-1656" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">494514</span>, <span class="st">"Vulpes lagopus"</span>,</span>
<span id="cb51-1657"><a href="#cb51-1657" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span id="cb51-1658"><a href="#cb51-1658" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3119969</span>, <span class="st">"Bubalus kerabau"</span>)</span>
<span id="cb51-1659"><a href="#cb51-1659" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(tax_names, tax_names_new)</span>
<span id="cb51-1660"><a href="#cb51-1660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1661"><a href="#cb51-1661" aria-hidden="true" tabindex="-1"></a><span class="co"># Get matches</span></span>
<span id="cb51-1662"><a href="#cb51-1662" aria-hidden="true" tabindex="-1"></a>hv_blast_staxids <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">%in%</span> ref_taxids_hv) <span class="sc">%&gt;%</span></span>
<span id="cb51-1663"><a href="#cb51-1663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb51-1664"><a href="#cb51-1664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1665"><a href="#cb51-1665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1666"><a href="#cb51-1666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name), <span class="at">by=</span><span class="st">"staxid"</span>)</span>
<span id="cb51-1667"><a href="#cb51-1667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1668"><a href="#cb51-1668" aria-hidden="true" tabindex="-1"></a><span class="co"># Count matches</span></span>
<span id="cb51-1669"><a href="#cb51-1669" aria-hidden="true" tabindex="-1"></a>hv_blast_counts <span class="ot">&lt;-</span> hv_blast_staxids <span class="sc">%&gt;%</span></span>
<span id="cb51-1670"><a href="#cb51-1670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(taxid, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb51-1671"><a href="#cb51-1671" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb51-1672"><a href="#cb51-1672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1673"><a href="#cb51-1673" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to major matches</span></span>
<span id="cb51-1674"><a href="#cb51-1674" aria-hidden="true" tabindex="-1"></a>hv_blast_counts_major <span class="ot">&lt;-</span> hv_blast_counts <span class="sc">%&gt;%</span> </span>
<span id="cb51-1675"><a href="#cb51-1675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1676"><a href="#cb51-1676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span></span>
<span id="cb51-1677"><a href="#cb51-1677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">25</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1678"><a href="#cb51-1678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">==</span> ref_names_hv[<span class="dv">1</span>], <span class="st">"MCV"</span>, name))</span>
<span id="cb51-1679"><a href="#cb51-1679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1680"><a href="#cb51-1680" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1681"><a href="#cb51-1681" aria-hidden="true" tabindex="-1"></a>g_hv_blast <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_blast_counts_major, <span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span></span>
<span id="cb51-1682"><a href="#cb51-1682" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb51-1683"><a href="#cb51-1683" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(name_display<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>, <span class="at">space=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb51-1684"><a href="#cb51-1684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), </span>
<span id="cb51-1685"><a href="#cb51-1685" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb51-1686"><a href="#cb51-1686" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb51-1687"><a href="#cb51-1687" aria-hidden="true" tabindex="-1"></a>g_hv_blast</span>
<span id="cb51-1688"><a href="#cb51-1688" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1689"><a href="#cb51-1689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1690"><a href="#cb51-1690" aria-hidden="true" tabindex="-1"></a>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</span>
<span id="cb51-1691"><a href="#cb51-1691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1694"><a href="#cb51-1694" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb51-1695"><a href="#cb51-1695" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb51-1696"><a href="#cb51-1696" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-genera</span></span>
<span id="cb51-1697"><a href="#cb51-1697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1698"><a href="#cb51-1698" aria-hidden="true" tabindex="-1"></a><span class="co"># Define reference genera</span></span>
<span id="cb51-1699"><a href="#cb51-1699" aria-hidden="true" tabindex="-1"></a>path_genera_rna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span>)</span>
<span id="cb51-1700"><a href="#cb51-1700" aria-hidden="true" tabindex="-1"></a>path_genera_dna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span id="cb51-1701"><a href="#cb51-1701" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span>)</span>
<span id="cb51-1702"><a href="#cb51-1702" aria-hidden="true" tabindex="-1"></a>path_genera <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">tibble</span>(<span class="at">name=</span>path_genera_rna, <span class="at">genome_type=</span><span class="st">"RNA genome"</span>),</span>
<span id="cb51-1703"><a href="#cb51-1703" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">tibble</span>(<span class="at">name=</span>path_genera_dna, <span class="at">genome_type=</span><span class="st">"DNA genome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1704"><a href="#cb51-1704" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb51-1705"><a href="#cb51-1705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1706"><a href="#cb51-1706" aria-hidden="true" tabindex="-1"></a><span class="co"># Count in each sample</span></span>
<span id="cb51-1707"><a href="#cb51-1707" aria-hidden="true" tabindex="-1"></a>n_path_genera <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb51-1708"><a href="#cb51-1708" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, date_alias, city, name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1709"><a href="#cb51-1709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb51-1710"><a href="#cb51-1710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(path_genera, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1711"><a href="#cb51-1711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"date_alias"</span>, <span class="st">"city"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-1712"><a href="#cb51-1712" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb51-1713"><a href="#cb51-1713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1714"><a href="#cb51-1714" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot out and back to add zero lines</span></span>
<span id="cb51-1715"><a href="#cb51-1715" aria-hidden="true" tabindex="-1"></a>n_path_genera_out <span class="ot">&lt;-</span> n_path_genera <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, name, n_reads_viral) <span class="sc">%&gt;%</span></span>
<span id="cb51-1716"><a href="#cb51-1716" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"name"</span>, <span class="at">values_from=</span><span class="st">"n_reads_viral"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1717"><a href="#cb51-1717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to=</span><span class="st">"name"</span>, <span class="at">values_to=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1718"><a href="#cb51-1718" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1719"><a href="#cb51-1719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(path_genera, <span class="at">by=</span><span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1720"><a href="#cb51-1720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb51-1721"><a href="#cb51-1721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1722"><a href="#cb51-1722" aria-hidden="true" tabindex="-1"></a><span class="do">## Aggregate across dates</span></span>
<span id="cb51-1723"><a href="#cb51-1723" aria-hidden="true" tabindex="-1"></a>n_path_genera_stype <span class="ot">&lt;-</span> n_path_genera_out <span class="sc">%&gt;%</span> </span>
<span id="cb51-1724"><a href="#cb51-1724" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid, genome_type, city) <span class="sc">%&gt;%</span></span>
<span id="cb51-1725"><a href="#cb51-1725" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb51-1726"><a href="#cb51-1726" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_viral =</span> <span class="fu">sum</span>(n_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-1727"><a href="#cb51-1727" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="st">"All samples"</span>, <span class="at">date=</span><span class="st">"All dates"</span>,</span>
<span id="cb51-1728"><a href="#cb51-1728" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw,</span>
<span id="cb51-1729"><a href="#cb51-1729" aria-hidden="true" tabindex="-1"></a>         <span class="at">na_type =</span> <span class="st">"DNA"</span>)</span>
<span id="cb51-1730"><a href="#cb51-1730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1731"><a href="#cb51-1731" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb51-1732"><a href="#cb51-1732" aria-hidden="true" tabindex="-1"></a>g_path_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_path_genera_stype,</span>
<span id="cb51-1733"><a href="#cb51-1733" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y=</span>name, <span class="at">x=</span>p_reads_viral, <span class="at">color=</span>city)) <span class="sc">+</span></span>
<span id="cb51-1734"><a href="#cb51-1734" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb51-1735"><a href="#cb51-1735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb51-1736"><a href="#cb51-1736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_city</span>() <span class="sc">+</span></span>
<span id="cb51-1737"><a href="#cb51-1737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genome_type<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb51-1738"><a href="#cb51-1738" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb51-1739"><a href="#cb51-1739" aria-hidden="true" tabindex="-1"></a>g_path_genera</span>
<span id="cb51-1740"><a href="#cb51-1740" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb51-1741"><a href="#cb51-1741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1742"><a href="#cb51-1742" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb51-1743"><a href="#cb51-1743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1744"><a href="#cb51-1744" aria-hidden="true" tabindex="-1"></a>This is the third, largest, and final of this tranche of air-sampling datasets that I've run through this pipeline. Many of the high-level findings were similar to Prussin and Rosario, including high relative abundance of human reads, low total viral reads, an absence of enteric viruses, and high abundance of papillomaviruses among human-infecting viruses.</span>
<span id="cb51-1745"><a href="#cb51-1745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-1746"><a href="#cb51-1746" aria-hidden="true" tabindex="-1"></a>In the future, I'll do a more in-depth comparative analysis across different datasets to compare the abundance of different viruses. For now, though, there are some major updates to the pipeline I want to make before I do any more public analyses.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>