<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-02-04">
<title>Will’s Public NAO Notebook - Workflow analysis of Rothman et al.&nbsp;(2021), part 1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Rothman et al.&nbsp;(2021), part 1</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Unenriched samples.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 4, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>In my last entry, I finished workflow analysis of <a href="https://doi.org/10.1128%2FmBio.02703-20">Crits-Christoph et al.&nbsp;2021</a> enriched and unenriched MGS data. In this entry, I apply the workflow to another pre-existing dataset from the <a href="https://naobservatory.org/reports/predicting-virus-relative-abundance-in-wastewater/">P2RA project</a>, namely <a href="https://doi.org/10.1128/AEM.01448-21">Rothman et al.&nbsp;2021</a>.</p>
<p>This is a much bigger dataset than Crits-Christoph, with 6 billion reads spread out over 363 different samples. Interestingly, Nextflow seemed to struggle more with the number of samples than their size; when running the entire dataset through the pipeline I kept running into issues with caching that seemed to be due to the sheer number of jobs in the pipeline. I’ll keep working on these, but in the meantime I decided to separate the samples from the dataset into groups to make running them through the pipeline easier.</p>
<p>In this entry, I analyze the results of the pipeline on the 97 unenriched samples from the pipeline. These were collected from 9 California treatment plants between July 2020 and January 2021, pasteurized, filtered through a 0.22um filter and concentrated with 10-kDa Amicon filters. After this, they underwent RNA-seq library prep and were sequenced on an Illumina NovaSeq 6000.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The Rothman unenriched samples totaled roughly 660M read pairs (151 gigabases of sequence). The number of reads per sample varied from 1.3M to 23.5M, with an average of 6.8M reads per sample. The number of reads per treatment plant varied from 6.7 to 165.9M, with an average of 73.3M.</p>
<p>While many samples had low levels of sequence duplication according to FASTQC, others had very high levels of duplication in excess of 75%. The median level of duplication according to FASTQ was roughly 56%. Adapter levels were high.</p>
<p>Read qualities for most samples were consistently very high, but some samples showed a large drop in sequence quality around position 20, suggesting a serious need for preprocessing to remove low-quality bases.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Data input paths</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-02-23_rothman-1/"</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"rothman-libraries-unenriched.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_adapter_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_base_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_sequence_stats.tsv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">sample</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"location"</span>, <span class="st">"month"</span>, <span class="st">"day"</span>, <span class="st">"year"</span>, <span class="st">"x1"</span>, <span class="st">"x2"</span><span class="op">)</span>, remove <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>month <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">month</span><span class="op">)</span>, year <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">year</span><span class="op">)</span>, day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">day</span><span class="op">)</span>,</span>
<span>         enrichment <span class="op">=</span> <span class="st">"Unenriched"</span>, </span>
<span>         date <span class="op">=</span> <span class="fu">ymd</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">year</span>, <span class="va">month</span>, <span class="va">day</span>, sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">x1</span>, <span class="op">-</span><span class="va">x2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">location</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">adapter_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_base_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_seq_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize basic stats</span></span>
<span><span class="va">g_nreads_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">enrichment</span>, y<span class="op">=</span><span class="va">n_read_pairs</span>, fill<span class="op">=</span><span class="va">location</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span><span class="va">legend_location</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_nreads_raw</span><span class="op">)</span></span>
<span><span class="va">g_nreads_raw_2</span> <span class="op">&lt;-</span> <span class="va">g_nreads_raw</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_nbases_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">enrichment</span>, y<span class="op">=</span><span class="va">n_bases_approx</span>, fill<span class="op">=</span><span class="va">location</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Total base pairs (approx)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_ndup_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">enrichment</span>, y<span class="op">=</span><span class="va">percent_duplicates</span>, fill<span class="op">=</span><span class="va">location</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Duplicates (FASTQC)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">g_basic_raw</span> <span class="op">&lt;-</span> <span class="fu">plot_grid</span><span class="op">(</span><span class="va">g_nreads_raw_2</span> <span class="op">+</span> <span class="va">g_nbases_raw</span> <span class="op">+</span> <span class="va">g_ndup_raw</span>, <span class="va">legend_location</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_basic_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">adapter_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">quality_base_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">quality_seq_stats_raw</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-2-3.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-2-4.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>On average, cleaning, deduplication and conservative ribodepletion together removed about 60% of total reads from unenriched samples, with the largest reductions seen during deduplication. Secondary ribodepletion removing a further 6%. However, these summary figures conceal significant inter-sample variation. Unsurprisingly, samples that showed a higher initial level of duplication also showed greater read losses during preprocessing.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">n_read_pairs</span>,fill<span class="op">=</span><span class="va">location</span>,group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">stage</span>, <span class="va">enrichment</span>, <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">enrichment</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span><span class="op">)</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">p_reads_lost</span>,fill<span class="op">=</span><span class="va">location</span>,group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot read losses vs initial duplication levels</span></span>
<span><span class="va">g_loss_dup</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>percent_duplicates_raw <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">percent_duplicates_raw</span>, y<span class="op">=</span><span class="va">p_reads_lost_abs</span>, color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Reads lost by initial ribodepletion)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span><span class="st">"% FASTQC-measured duplicates in raw data"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">20</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_loss_dup</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Data cleaning with FASTP was very successful at removing adapters, with no adapter sequences found by FASTQC at any stage after the raw data. Preprocessing successfully removed the terminal decline in quality seen in many samples, but failed to correct the large mid-read drop in quality seen in many samples. Since this group actually accounted for over half of samples, I decided to retain it for now, but will investigate ways to trim or discard such reads in future analyses, including of this data set.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">adapter_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">quality_base_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">quality_seq_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span>, color<span class="op">=</span><span class="va">location</span>, linetype <span class="op">=</span> <span class="va">read_pair</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Deduplication and ribodepletion were collectively quite effective at reducing measured duplicate levels, with the average detected duplication level after both processes reduced to roughly 14%. <strong>Note that the pipeline still doesn’t have a reverse-complement-sensitive deduplication pipeline, so only same-orientation duplicates have been removed.</strong></p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">percent_duplicates</span>, fill<span class="op">=</span><span class="va">location</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, y<span class="op">=</span><span class="va">mean_seq_len</span>, fill<span class="op">=</span><span class="va">location</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">legend_loc</span> <span class="op">&lt;-</span> <span class="fu">get_legend</span><span class="op">(</span><span class="va">g_dup_stages</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import Bracken data</span></span>
<span><span class="va">bracken_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"bracken_counts.tsv"</span><span class="op">)</span></span>
<span><span class="va">bracken</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">bracken_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">total_assigned</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  kraken_assigned_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">kraken_assigned_reads</span><span class="op">)</span>,</span>
<span>  added_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">added_reads</span><span class="op">)</span>,</span>
<span>  new_est_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">new_est_reads</span><span class="op">)</span>,</span>
<span>  fraction_total_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">fraction_total_reads</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bracken_spread</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"sample"</span>, names_from <span class="op">=</span> <span class="st">"name"</span>, values_from <span class="op">=</span> <span class="st">"new_est_reads"</span><span class="op">)</span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">enrichment</span>, <span class="va">stage</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"enrichment"</span><span class="op">)</span>, names_from<span class="op">=</span><span class="st">"stage"</span>, values_from<span class="op">=</span><span class="st">"n_read_pairs"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_preproc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">total_assigned</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>assigned <span class="op">=</span> <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">bracken_spread</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span><span class="co"># Assess composition</span></span>
<span><span class="va">read_comp</span> <span class="op">&lt;-</span> <span class="fu">transmute</span><span class="op">(</span><span class="va">read_counts</span>, sample<span class="op">=</span><span class="va">sample</span>, location<span class="op">=</span><span class="va">location</span>,</span>
<span>                       enrichment<span class="op">=</span><span class="va">enrichment</span>,</span>
<span>                       n_filtered <span class="op">=</span> <span class="va">raw_concat</span><span class="op">-</span><span class="va">cleaned</span>,</span>
<span>                       n_duplicate <span class="op">=</span> <span class="va">cleaned</span><span class="op">-</span><span class="va">dedup</span>,</span>
<span>                       n_ribosomal <span class="op">=</span> <span class="op">(</span><span class="va">dedup</span><span class="op">-</span><span class="va">ribo_initial</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="va">ribo_initial</span><span class="op">-</span><span class="va">ribo_secondary</span><span class="op">)</span>,</span>
<span>                       n_unassigned <span class="op">=</span> <span class="va">ribo_secondary</span><span class="op">-</span><span class="va">assigned</span>,</span>
<span>                       n_bacterial <span class="op">=</span> <span class="va">bacteria</span>,</span>
<span>                       n_archaeal <span class="op">=</span> <span class="va">archaea</span>,</span>
<span>                       n_viral <span class="op">=</span> <span class="va">viruses</span>,</span>
<span>                       n_human <span class="op">=</span> <span class="va">eukaryota</span><span class="op">)</span></span>
<span><span class="va">read_comp_long</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">read_comp</span>, <span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">enrichment</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"classification"</span>,</span>
<span>                               names_prefix <span class="op">=</span> <span class="st">"n_"</span>, values_to <span class="op">=</span> <span class="st">"n_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu">str_to_sentence</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">enrichment</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_comp_summ</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">enrichment</span>, scales<span class="op">=</span><span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">read_comp_minor</span> <span class="op">&lt;-</span> <span class="va">read_comp_summ</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_comp_minor</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Reads"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0.01</span><span class="op">)</span>,</span>
<span>                     expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">enrichment</span>, scales<span class="op">=</span><span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Overall, in these unenriched samples, between 66% and 86% of reads (mean 74%) are low-quality, duplicates, or unassigned. Of the remainder, 8-21% (mean 15%) were identified as ribosomal, and 2-12% (mean 5%) were otherwise classified as bacterial. The fraction of human reads ranged from 0.4-6.5% (mean 1.8%) and that of viral reads from 0.6-8.6% (mean 4.5%).</p>
<p>The latter of these is strikingly high, about an order of magnitude higher than that observed in Crits-Christoph. In the <a href="https://data.securebio.org/mgs-counts/">public dashboard</a>, high levels of total virus reads are also observed, and attributed almost entirely to elevated levels of tobamoviruses, especially tomato brown rugose fruit virus. Digging into the Kraken reports for these samples, we find that for most samples, <em>Tobamovirus</em> indeed makes up the vast majority of viral reads:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">toba_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"tobamovirus.tsv"</span><span class="op">)</span></span>
<span><span class="va">toba</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">toba_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">g_toba</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">toba</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p_viral_reads_Tobamovirus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">0.01</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% Viral reads assigned to Tobamovirus"</span>,</span>
<span>                     limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# samples"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>,</span>
<span>                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_toba</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples, using a modified form of the pipeline used in my last entry:</p>
<ol type="1">
<li>After initial ribodepletion, surviving reads were aligned to a DB of human-infecting virus genomes with Bowtie2, using permissive alignment parameters.</li>
<li>Reads that aligned successfully with Bowtie2 were filtered to remove human, livestock, cloning vector, and other potential contaminant sequences, then run through Kraken2 using the standard 16GB database.</li>
<li>For surviving reads, the length-adjusted alignment score <span class="math inline">\(S=\frac{\text{bowtie2 alignment score}}{\ln(\text{read length})}\)</span> was calculated, and reads were filtered out if they didn’t meet at least one of the following four criteria:
<ul>
<li>The read pair is <em>assigned</em> to a human-infecting virus by both Kraken and Bowtie2</li>
<li>The read pair contains a Kraken2 <em>hit</em> to the same taxid as it is assigned to by Bowtie2.</li>
<li>The read pair is unassigned by Kraken and <span class="math inline">\(S&gt;15\)</span> for the forward read</li>
<li>The read pair is unassigned by Kraken and <span class="math inline">\(S&gt;15\)</span> for the reverse read</li>
</ul>
</li>
</ol>
<p>Applying all of these filtering steps leaves a total of 16034 read pairs across all unenriched samples (0.007% of surviving reads):</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_reads_filtered_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span><span class="va">g_mrg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>To investigate these putative viral reads, I ran them through BLASTN and analyzed the results in a similar manner to Crits-Christoph:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_num</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">sample</span>, <span class="st">"_"</span>, <span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_fasta</span> <span class="op">&lt;-</span>  <span class="va">mrg_num</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"rothman-putative-viral.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import viral taxids and relationships</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"viral-taxa.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import BLAST results</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"rothman-putative-viral.blast.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                          col_names <span class="op">=</span> <span class="va">blast_cols</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_results</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject, then rank hits for each query</span></span>
<span><span class="va">blast_results_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_viral</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">blast_results_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, </span>
<span>         seq_num <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span>, </span>
<span>         sample <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">head</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">paste</span>, collapse<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span>, <span class="va">viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg_num</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span></span>
<span><span class="va">blast_results_assign</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_num</span>, <span class="va">blast_results_out</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_mrg_blast_0</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">viral_status_out</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_blast_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_hist_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_blast</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This initially looks pretty good, with high-scoring true-positives massively outnumbering false-positives and unclear cases. However, there are also a lot of <em>low</em>-scoring true-positives, which are excluded when the alignment score threshold is placed high enough to exclude the majority of false-positives. This pulls down sensitivity at higher score thresholds; as a result, F1 score is lower than I’d like, peaking at 0.947 for a disjunctive cutoff of 17 and dropping to 0.921 for a cutoff of 20.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span>, <span class="va">conjunctive</span>, <span class="va">include_special</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">include_special</span><span class="op">)</span> <span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">tab</span>, <span class="va">viral_status_out</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TRUE"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span></span>
<span>    conjunctive <span class="op">=</span> <span class="va">conjunctive</span>,</span>
<span>    retain_score_conjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">&amp;</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>, </span>
<span>    retain_score_disjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>    retain_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="va">retain_score_conjunctive</span>, <span class="va">retain_score_disjunctive</span><span class="op">)</span>,</span>
<span>    retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, include_special <span class="op">=</span> <span class="va">include_special</span>, </span>
<span>                conjunctive <span class="op">=</span> <span class="va">conjunctive</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inc_special</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span>, include_special<span class="op">=</span><span class="va">inc_special</span><span class="op">)</span></span>
<span>  <span class="va">stats_conj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_disj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_conj</span>, <span class="va">stats_disj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">threshold</span><span class="op">:</span><span class="va">conjunctive</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>conj_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats_all</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">inc_special</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">threshold_opt_0</span> <span class="op">&lt;-</span> <span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">conj_label</span>,<span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>data <span class="op">=</span> <span class="va">threshold_opt_0</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>xintercept<span class="op">=</span><span class="va">threshold</span><span class="op">)</span>,</span>
<span>             color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Improving precision for high-scoring sequences will only have a marginal effect on F1 score here, as the limiting factor is sensitivity. The best way to improve this would be to find ways to better exclude low-scoring false-positives from the dataset, then lower the score threshold closer to 15. I’m wary about doing this, however, as I suspect low-scoring false-positives will be substantially harder to remove than high-scoring ones, and I’m somewhat less worried about false-negatives than false-positives here in any case. As such, I’m going to stick with my current pipeline for now.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unclear</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"UNCLEAR"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span></span>
<span><span class="va">mrg_unclear_fasta</span> <span class="op">&lt;-</span> <span class="va">mrg_unclear</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unclear_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_unclear_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_unclear_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"rothman-unclear-viral.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Moving on to the small number of “unclear” viral matches (155 total, 107 high-scoring), as with Crits-Christoph, the vast majority of these appear to be true positives when run on NCBI BLAST online:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">unclear_viral_status</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">53</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">13</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">15</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">6</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">12</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">2</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">12</span><span class="op">)</span>, <span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fl">29</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mrg_unclear_out</span> <span class="op">&lt;-</span> <span class="va">mrg_unclear</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>viral_status_true <span class="op">=</span> <span class="va">unclear_viral_status</span><span class="op">)</span></span>
<span><span class="va">g_unclear</span> <span class="op">&lt;-</span> <span class="va">mrg_unclear_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_true</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_unclear</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Since this is two large datasets for which I’ve now gotten this result, I’ll henceforth treat reads for which only one of two reads shows a BLAST viral match as true positives for validation purposes.</p>
</section><section id="human-infecting-virus-reads-relative-abundance" class="level1"><h1>Human-infecting virus reads: relative abundance</h1>
<p>In total, my pipeline identified 12305 reads as human-viral out of 660M total reads, for a relative HV abundance of <span class="math inline">\(1.87 \times 10^{-5}\)</span>. This compares to <span class="math inline">\(4.3 \times 10^{-6}\)</span> on the public dashboard, corresponding to the results for Kraken-only identification: a 4.4-fold increase, similar to that observed for Crits-Christoph. Excluding false-positives identified by BLAST above, the new estimate is reduced slightly, to <span class="math inline">\(1.81 \times 10^{-5}\)</span>.</p>
<p>Aggregating by treatment plant location, we see substantial variation in HV relative abundance, from <span class="math inline">\(7.5 \times 10^{-7}\)</span> for location JWPCP to <span class="math inline">\(5.91 \times 10^{-5}\)</span> for location SB:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">date</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts &amp; RA</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>,</span>
<span>         hv_true <span class="op">=</span> <span class="va">viral_status</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv_all</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv_all"</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv_val</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span>, <span class="va">hv_true</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv_validated"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv_all</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv_val</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv_all <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv_all</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         n_reads_hv_validated <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv_validated</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_hv_all <span class="op">=</span> <span class="va">n_reads_hv_all</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_hv_validated <span class="op">=</span> <span class="va">n_reads_hv_validated</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span><span class="va">read_counts_melted</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">date</span>, <span class="fu">starts_with</span><span class="op">(</span><span class="st">"p"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"group"</span>, values_to <span class="op">=</span> <span class="st">"p"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu">str_to_title</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"p_reads_hv_"</span>, <span class="st">""</span>, <span class="va">group</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span>, <span class="va">date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate by location</span></span>
<span><span class="va">read_counts_loc</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_validated</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv_all <span class="op">=</span> <span class="va">n_reads_hv_all</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_hv_validated <span class="op">=</span> <span class="va">n_reads_hv_validated</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span><span class="va">read_counts_total</span> <span class="op">&lt;-</span> <span class="va">read_counts_loc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_validated</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv_all <span class="op">=</span> <span class="va">n_reads_hv_all</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_hv_validated <span class="op">=</span> <span class="va">n_reads_hv_validated</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts_loc</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_total</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">location</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg_melted</span> <span class="op">&lt;-</span> <span class="va">read_counts_agg</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">location</span>, <span class="fu">starts_with</span><span class="op">(</span><span class="st">"p"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="st">"p"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"group"</span>, values_to <span class="op">=</span> <span class="st">"p"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>group <span class="op">=</span> <span class="fu">str_to_title</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"p_reads_hv_"</span>, <span class="st">""</span>, <span class="va">group</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize</span></span>
<span><span class="va">palette_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span>, <span class="st">"black"</span><span class="op">)</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg_melted</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">p</span>, color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_loc</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">group</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="co">#</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Plotting HV relative abundance over time for each plant, we see that while some plants remain roughly stable over time, several others exhibit a gradual increase in HV relative abundance over the course of the study, possibly related to the winter disease season:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_time</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_melted</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">date</span>, y<span class="op">=</span><span class="va">p</span>, color<span class="op">=</span><span class="va">location</span>, linetype<span class="op">=</span><span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Estimator"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">location</span><span class="op">~</span><span class="va">.</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_phv_time</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Digging into individual viruses, we see that fecal-oral viruses predominate, especially noroviruses (e.g.&nbsp;Norwalk virus) and hudisaviruses, though betacoronaviruses including SARS-CoV-2 make a respectable showing in several samples:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genera</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count relative abundance for species</span></span>
<span><span class="va">hv_species_counts_raw</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">summarize</span><span class="op">(</span>n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">hv_status</span> <span class="op">&amp;</span> <span class="va">hv_true</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">read_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">location</span>, <span class="va">n_reads_raw</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"location"</span><span class="op">)</span></span>
<span><span class="va">hv_species_counts_all</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_validated</span><span class="op">)</span>,</span>
<span>            n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">hv_species_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_species_counts_raw</span>, <span class="va">hv_species_counts_all</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv_all <span class="op">=</span> <span class="va">n_reads_hv_all</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_hv_validated <span class="op">=</span> <span class="va">n_reads_hv_validated</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count relative abundance for genera</span></span>
<span><span class="va">hv_genera_counts_raw</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genera</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">hv_status</span> <span class="op">&amp;</span> <span class="va">hv_true</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">read_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">location</span>, <span class="va">n_reads_raw</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"location"</span><span class="op">)</span></span>
<span><span class="va">hv_genera_counts_all</span> <span class="op">&lt;-</span> <span class="va">hv_genera_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>,</span>
<span>            n_reads_hv_validated <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv_validated</span><span class="op">)</span>,</span>
<span>            n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">hv_genera_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_genera_counts_raw</span>, <span class="va">hv_genera_counts_all</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv_all <span class="op">=</span> <span class="va">n_reads_hv_all</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         p_reads_hv_validated <span class="op">=</span> <span class="va">n_reads_hv_validated</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Compute ranks for species and genera and restrict to high-ranking taxa</span></span>
<span><span class="va">max_rank_species</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">max_rank_genera</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">hv_species_counts_ranked</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>, ties.method<span class="op">=</span><span class="st">"max"</span><span class="op">)</span>,</span>
<span>         highrank <span class="op">=</span> <span class="va">rank</span> <span class="op">&lt;=</span> <span class="va">max_rank_species</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>highrank_any <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">highrank</span><span class="op">)</span>,</span>
<span>         name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highrank_any</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>mean_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">mean_rank</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_genera_counts_ranked</span> <span class="op">&lt;-</span> <span class="va">hv_genera_counts_agg</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rank.html">rank</a></span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_hv_all</span><span class="op">)</span>, ties.method<span class="op">=</span><span class="st">"max"</span><span class="op">)</span>,</span>
<span>         highrank <span class="op">=</span> <span class="va">rank</span> <span class="op">&lt;=</span> <span class="va">max_rank_genera</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>highrank_any <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">highrank</span><span class="op">)</span>,</span>
<span>         name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highrank_any</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>mean_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">rank</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">mean_rank</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot composition</span></span>
<span><span class="va">palette_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_vcomp_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_genera_counts_ranked</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">n_reads_hv_all</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_rank</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Fraction of HV reads"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_vcomp_species</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_species_counts_ranked</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">location</span>, y<span class="op">=</span><span class="va">n_reads_hv_all</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_rank</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Fraction of HV reads"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="va">g_vcomp_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_vcomp_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-23_rothman-1_files/figure-html/unnamed-chunk-20-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>These results make sense aren’t especially surprising. In my next entry, I’ll turn to the enriched samples from Rothman, and compare them to these unenriched samples.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb30" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Rothman et al. (2021), part 1"</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Unenriched samples."</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-02-04</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>In my last entry, I finished workflow analysis of <span class="co">[</span><span class="ot">Crits-Christoph et al. 2021</span><span class="co">](https://doi.org/10.1128%2FmBio.02703-20)</span> enriched and unenriched MGS data. In this entry, I apply the workflow to another pre-existing dataset from the <span class="co">[</span><span class="ot">P2RA project</span><span class="co">](https://naobservatory.org/reports/predicting-virus-relative-abundance-in-wastewater/)</span>, namely <span class="co">[</span><span class="ot">Rothman et al. 2021</span><span class="co">](https://doi.org/10.1128/AEM.01448-21)</span>.</span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>This is a much bigger dataset than Crits-Christoph, with 6 billion reads spread out over 363 different samples. Interestingly, Nextflow seemed to struggle more with the number of samples than their size; when running the entire dataset through the pipeline I kept running into issues with caching that seemed to be due to the sheer number of jobs in the pipeline. I'll keep working on these, but in the meantime I decided to separate the samples from the dataset into groups to make running them through the pipeline easier.</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>In this entry, I analyze the results of the pipeline on the 97 unenriched samples from the pipeline. These were collected from 9 California treatment plants between July 2020 and January 2021, pasteurized, filtered through a 0.22um filter and concentrated with 10-kDa Amicon filters. After this, they underwent RNA-seq library prep and were sequenced on an Illumina NovaSeq 6000.</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>The Rothman unenriched samples totaled roughly 660M read pairs (151 gigabases of sequence). The number of reads per sample varied from 1.3M to 23.5M, with an average of 6.8M reads per sample. The number of reads per treatment plant varied from 6.7 to 165.9M, with an average of 73.3M.</span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a>While many samples had low levels of sequence duplication according to FASTQC, others had very high levels of duplication in excess of 75%. The median level of duplication according to FASTQ was roughly 56%. Adapter levels were high.</span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>Read qualities for most samples were consistently very high, but some samples showed a large drop in sequence quality around position 20, suggesting a serious need for preprocessing to remove low-quality bases.</span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-02-23_rothman-1/"</span></span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"rothman-libraries-unenriched.csv"</span>)</span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats.tsv"</span>)</span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_adapter_stats.tsv"</span>)</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_base_stats.tsv"</span>)</span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_sequence_stats.tsv"</span>)</span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb30-64"><a href="#cb30-64" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-65"><a href="#cb30-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(sample, <span class="fu">c</span>(<span class="st">"location"</span>, <span class="st">"month"</span>, <span class="st">"day"</span>, <span class="st">"year"</span>, <span class="st">"x1"</span>, <span class="st">"x2"</span>), <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-66"><a href="#cb30-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">month =</span> <span class="fu">as.numeric</span>(month), <span class="at">year =</span> <span class="fu">as.numeric</span>(year), <span class="at">day =</span> <span class="fu">as.numeric</span>(day),</span>
<span id="cb30-67"><a href="#cb30-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">enrichment =</span> <span class="st">"Unenriched"</span>, </span>
<span id="cb30-68"><a href="#cb30-68" aria-hidden="true" tabindex="-1"></a>         <span class="at">date =</span> <span class="fu">ymd</span>(<span class="fu">paste</span>(year, month, day, <span class="at">sep=</span><span class="st">"-"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb30-69"><a href="#cb30-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>x1, <span class="sc">-</span>x2) <span class="sc">%&gt;%</span></span>
<span id="cb30-70"><a href="#cb30-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, date) <span class="sc">%&gt;%</span></span>
<span id="cb30-71"><a href="#cb30-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">fct_inorder</span>(location))</span>
<span id="cb30-72"><a href="#cb30-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-73"><a href="#cb30-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb30-74"><a href="#cb30-74" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb30-75"><a href="#cb30-75" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-76"><a href="#cb30-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb30-77"><a href="#cb30-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-78"><a href="#cb30-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, date) <span class="sc">%&gt;%</span></span>
<span id="cb30-79"><a href="#cb30-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb30-80"><a href="#cb30-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb30-81"><a href="#cb30-81" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-82"><a href="#cb30-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb30-83"><a href="#cb30-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-84"><a href="#cb30-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb30-85"><a href="#cb30-85" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb30-86"><a href="#cb30-86" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-87"><a href="#cb30-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb30-88"><a href="#cb30-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-89"><a href="#cb30-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb30-90"><a href="#cb30-90" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb30-91"><a href="#cb30-91" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-92"><a href="#cb30-92" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb30-93"><a href="#cb30-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-94"><a href="#cb30-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb30-95"><a href="#cb30-95" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb30-96"><a href="#cb30-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-97"><a href="#cb30-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb30-98"><a href="#cb30-98" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb30-99"><a href="#cb30-99" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb30-100"><a href="#cb30-100" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb30-101"><a href="#cb30-101" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb30-102"><a href="#cb30-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-103"><a href="#cb30-103" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize basic stats</span></span>
<span id="cb30-104"><a href="#cb30-104" aria-hidden="true" tabindex="-1"></a>g_nreads_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>enrichment, <span class="at">y=</span>n_read_pairs, <span class="at">fill=</span>location, <span class="at">group=</span>sample)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span> theme_kit</span>
<span id="cb30-105"><a href="#cb30-105" aria-hidden="true" tabindex="-1"></a>legend_location <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_nreads_raw)</span>
<span id="cb30-106"><a href="#cb30-106" aria-hidden="true" tabindex="-1"></a>g_nreads_raw_2 <span class="ot">&lt;-</span> g_nreads_raw <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb30-107"><a href="#cb30-107" aria-hidden="true" tabindex="-1"></a>g_nbases_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>enrichment, <span class="at">y=</span>n_bases_approx, <span class="at">fill=</span>location, <span class="at">group=</span>sample)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Total base pairs (approx)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span> theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb30-108"><a href="#cb30-108" aria-hidden="true" tabindex="-1"></a>g_ndup_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>enrichment, <span class="at">y=</span>percent_duplicates, <span class="at">fill=</span>location, <span class="at">group=</span>sample)) <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Duplicates (FASTQC)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">20</span>)) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span> theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb30-109"><a href="#cb30-109" aria-hidden="true" tabindex="-1"></a>g_basic_raw <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(g_nreads_raw_2 <span class="sc">+</span> g_nbases_raw <span class="sc">+</span> g_ndup_raw, legend_location, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.1</span>))</span>
<span id="cb30-110"><a href="#cb30-110" aria-hidden="true" tabindex="-1"></a>g_basic_raw</span>
<span id="cb30-111"><a href="#cb30-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-112"><a href="#cb30-112" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb30-113"><a href="#cb30-113" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adapter_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample, read_pair))) <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-114"><a href="#cb30-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-115"><a href="#cb30-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-116"><a href="#cb30-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>),</span>
<span id="cb30-117"><a href="#cb30-117" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-118"><a href="#cb30-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">140</span>),</span>
<span id="cb30-119"><a href="#cb30-119" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-120"><a href="#cb30-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-121"><a href="#cb30-121" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-122"><a href="#cb30-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>adapter) <span class="sc">+</span> theme_base</span>
<span id="cb30-123"><a href="#cb30-123" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb30-124"><a href="#cb30-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-125"><a href="#cb30-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb30-126"><a href="#cb30-126" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(quality_base_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span></span>
<span id="cb30-127"><a href="#cb30-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-128"><a href="#cb30-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-129"><a href="#cb30-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-130"><a href="#cb30-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-131"><a href="#cb30-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-132"><a href="#cb30-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb30-133"><a href="#cb30-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">140</span>),</span>
<span id="cb30-134"><a href="#cb30-134" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-135"><a href="#cb30-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-136"><a href="#cb30-136" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-137"><a href="#cb30-137" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb30-138"><a href="#cb30-138" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(quality_seq_stats_raw, <span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span></span>
<span id="cb30-139"><a href="#cb30-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-140"><a href="#cb30-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-141"><a href="#cb30-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-142"><a href="#cb30-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-143"><a href="#cb30-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-144"><a href="#cb30-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-145"><a href="#cb30-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-146"><a href="#cb30-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-147"><a href="#cb30-147" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-148"><a href="#cb30-148" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb30-149"><a href="#cb30-149" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb30-150"><a href="#cb30-150" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb30-151"><a href="#cb30-151" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-152"><a href="#cb30-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-153"><a href="#cb30-153" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb30-154"><a href="#cb30-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-155"><a href="#cb30-155" aria-hidden="true" tabindex="-1"></a>On average, cleaning, deduplication and conservative ribodepletion together removed about 60% of total reads from unenriched samples, with the largest reductions seen during deduplication. Secondary ribodepletion removing a further 6%. However, these summary figures conceal significant inter-sample variation. Unsurprisingly, samples that showed a higher initial level of duplication also showed greater read losses during preprocessing.</span>
<span id="cb30-156"><a href="#cb30-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-159"><a href="#cb30-159" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-160"><a href="#cb30-160" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-161"><a href="#cb30-161" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb30-162"><a href="#cb30-162" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>n_read_pairs,<span class="at">fill=</span>location,<span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb30-163"><a href="#cb30-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-164"><a href="#cb30-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-165"><a href="#cb30-165" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-166"><a href="#cb30-166" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb30-167"><a href="#cb30-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-168"><a href="#cb30-168" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb30-169"><a href="#cb30-169" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, location, stage, enrichment, percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb30-170"><a href="#cb30-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, enrichment) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, location, enrichment, stage) <span class="sc">%&gt;%</span></span>
<span id="cb30-171"><a href="#cb30-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs),</span>
<span id="cb30-172"><a href="#cb30-172" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb30-173"><a href="#cb30-173" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb30-174"><a href="#cb30-174" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs)</span>
<span id="cb30-175"><a href="#cb30-175" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>p_reads_lost,<span class="at">fill=</span>location,<span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb30-176"><a href="#cb30-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-177"><a href="#cb30-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb30-178"><a href="#cb30-178" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-179"><a href="#cb30-179" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb30-180"><a href="#cb30-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-181"><a href="#cb30-181" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot read losses vs initial duplication levels</span></span>
<span id="cb30-182"><a href="#cb30-182" aria-hidden="true" tabindex="-1"></a>g_loss_dup <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb30-183"><a href="#cb30-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent_duplicates_raw =</span> percent_duplicates[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb30-184"><a href="#cb30-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-185"><a href="#cb30-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>percent_duplicates_raw, <span class="at">y=</span>p_reads_lost_abs, <span class="at">color=</span>location)) <span class="sc">+</span> </span>
<span id="cb30-186"><a href="#cb30-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb30-187"><a href="#cb30-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-188"><a href="#cb30-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Reads lost by initial ribodepletion)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb30-189"><a href="#cb30-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="st">"% FASTQC-measured duplicates in raw data"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb30-190"><a href="#cb30-190" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">20</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb30-191"><a href="#cb30-191" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb30-192"><a href="#cb30-192" aria-hidden="true" tabindex="-1"></a>g_loss_dup</span>
<span id="cb30-193"><a href="#cb30-193" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-194"><a href="#cb30-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-195"><a href="#cb30-195" aria-hidden="true" tabindex="-1"></a>Data cleaning with FASTP was very successful at removing adapters, with no adapter sequences found by FASTQC at any stage after the raw data. Preprocessing successfully removed the terminal decline in quality seen in many samples, but failed to correct the large mid-read drop in quality seen in many samples. Since this group actually accounted for over half of samples, I decided to retain it for now, but will investigate ways to trim or discard such reads in future analyses, including of this data set.</span>
<span id="cb30-196"><a href="#cb30-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-199"><a href="#cb30-199" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-200"><a href="#cb30-200" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-201"><a href="#cb30-201" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb30-202"><a href="#cb30-202" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(adapter_stats, <span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample, read_pair))) <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-203"><a href="#cb30-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-204"><a href="#cb30-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-205"><a href="#cb30-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>),</span>
<span id="cb30-206"><a href="#cb30-206" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-207"><a href="#cb30-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">140</span>),</span>
<span id="cb30-208"><a href="#cb30-208" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-209"><a href="#cb30-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-210"><a href="#cb30-210" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-211"><a href="#cb30-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter) <span class="sc">+</span> theme_base</span>
<span id="cb30-212"><a href="#cb30-212" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb30-213"><a href="#cb30-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-214"><a href="#cb30-214" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb30-215"><a href="#cb30-215" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(quality_base_stats, <span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span></span>
<span id="cb30-216"><a href="#cb30-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-217"><a href="#cb30-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-218"><a href="#cb30-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-219"><a href="#cb30-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-220"><a href="#cb30-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-221"><a href="#cb30-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb30-222"><a href="#cb30-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">140</span>),</span>
<span id="cb30-223"><a href="#cb30-223" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-224"><a href="#cb30-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-225"><a href="#cb30-225" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-226"><a href="#cb30-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.) <span class="sc">+</span> theme_base</span>
<span id="cb30-227"><a href="#cb30-227" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(quality_seq_stats, <span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences, <span class="at">color=</span>location, <span class="at">linetype =</span> read_pair, <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span></span>
<span id="cb30-228"><a href="#cb30-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-229"><a href="#cb30-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb30-230"><a href="#cb30-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-231"><a href="#cb30-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-232"><a href="#cb30-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb30-233"><a href="#cb30-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-234"><a href="#cb30-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-235"><a href="#cb30-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb30-236"><a href="#cb30-236" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb30-237"><a href="#cb30-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.) <span class="sc">+</span> theme_base</span>
<span id="cb30-238"><a href="#cb30-238" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb30-239"><a href="#cb30-239" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb30-240"><a href="#cb30-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-241"><a href="#cb30-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-242"><a href="#cb30-242" aria-hidden="true" tabindex="-1"></a>Deduplication and ribodepletion were collectively quite effective at reducing measured duplicate levels, with the average detected duplication level after both processes reduced to roughly 14%. **Note that the pipeline still doesn't have a reverse-complement-sensitive deduplication pipeline, so only same-orientation duplicates have been removed.**</span>
<span id="cb30-243"><a href="#cb30-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-246"><a href="#cb30-246" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-247"><a href="#cb30-247" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>percent_duplicates, <span class="at">fill=</span>location, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb30-248"><a href="#cb30-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-249"><a href="#cb30-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-250"><a href="#cb30-250" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-251"><a href="#cb30-251" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span>mean_seq_len, <span class="at">fill=</span>location, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb30-252"><a href="#cb30-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-253"><a href="#cb30-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-254"><a href="#cb30-254" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-255"><a href="#cb30-255" aria-hidden="true" tabindex="-1"></a>legend_loc <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(g_dup_stages)</span>
<span id="cb30-256"><a href="#cb30-256" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb30-257"><a href="#cb30-257" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb30-258"><a href="#cb30-258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-259"><a href="#cb30-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-260"><a href="#cb30-260" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb30-261"><a href="#cb30-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-262"><a href="#cb30-262" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb30-263"><a href="#cb30-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-266"><a href="#cb30-266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-267"><a href="#cb30-267" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bracken data</span></span>
<span id="cb30-268"><a href="#cb30-268" aria-hidden="true" tabindex="-1"></a>bracken_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"bracken_counts.tsv"</span>)</span>
<span id="cb30-269"><a href="#cb30-269" aria-hidden="true" tabindex="-1"></a>bracken <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(bracken_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-270"><a href="#cb30-270" aria-hidden="true" tabindex="-1"></a>total_assigned <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(</span>
<span id="cb30-271"><a href="#cb30-271" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Total"</span>,</span>
<span id="cb30-272"><a href="#cb30-272" aria-hidden="true" tabindex="-1"></a>  <span class="at">kraken_assigned_reads =</span> <span class="fu">sum</span>(kraken_assigned_reads),</span>
<span id="cb30-273"><a href="#cb30-273" aria-hidden="true" tabindex="-1"></a>  <span class="at">added_reads =</span> <span class="fu">sum</span>(added_reads),</span>
<span id="cb30-274"><a href="#cb30-274" aria-hidden="true" tabindex="-1"></a>  <span class="at">new_est_reads =</span> <span class="fu">sum</span>(new_est_reads),</span>
<span id="cb30-275"><a href="#cb30-275" aria-hidden="true" tabindex="-1"></a>  <span class="at">fraction_total_reads =</span> <span class="fu">sum</span>(fraction_total_reads)</span>
<span id="cb30-276"><a href="#cb30-276" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-277"><a href="#cb30-277" aria-hidden="true" tabindex="-1"></a>bracken_spread <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">select</span>(name, sample, new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb30-278"><a href="#cb30-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">tolower</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb30-279"><a href="#cb30-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"sample"</span>, <span class="at">names_from =</span> <span class="st">"name"</span>, <span class="at">values_from =</span> <span class="st">"new_est_reads"</span>)</span>
<span id="cb30-280"><a href="#cb30-280" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb30-281"><a href="#cb30-281" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb30-282"><a href="#cb30-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, enrichment, stage, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb30-283"><a href="#cb30-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"enrichment"</span>), <span class="at">names_from=</span><span class="st">"stage"</span>, <span class="at">values_from=</span><span class="st">"n_read_pairs"</span>)</span>
<span id="cb30-284"><a href="#cb30-284" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_preproc <span class="sc">%&gt;%</span></span>
<span id="cb30-285"><a href="#cb30-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total_assigned <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, new_est_reads), <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-286"><a href="#cb30-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">assigned =</span> new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb30-287"><a href="#cb30-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(bracken_spread, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb30-288"><a href="#cb30-288" aria-hidden="true" tabindex="-1"></a><span class="co"># Assess composition</span></span>
<span id="cb30-289"><a href="#cb30-289" aria-hidden="true" tabindex="-1"></a>read_comp <span class="ot">&lt;-</span> <span class="fu">transmute</span>(read_counts, <span class="at">sample=</span>sample, <span class="at">location=</span>location,</span>
<span id="cb30-290"><a href="#cb30-290" aria-hidden="true" tabindex="-1"></a>                       <span class="at">enrichment=</span>enrichment,</span>
<span id="cb30-291"><a href="#cb30-291" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_filtered =</span> raw_concat<span class="sc">-</span>cleaned,</span>
<span id="cb30-292"><a href="#cb30-292" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_duplicate =</span> cleaned<span class="sc">-</span>dedup,</span>
<span id="cb30-293"><a href="#cb30-293" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_ribosomal =</span> (dedup<span class="sc">-</span>ribo_initial) <span class="sc">+</span> (ribo_initial<span class="sc">-</span>ribo_secondary),</span>
<span id="cb30-294"><a href="#cb30-294" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_unassigned =</span> ribo_secondary<span class="sc">-</span>assigned,</span>
<span id="cb30-295"><a href="#cb30-295" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_bacterial =</span> bacteria,</span>
<span id="cb30-296"><a href="#cb30-296" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_archaeal =</span> archaea,</span>
<span id="cb30-297"><a href="#cb30-297" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_viral =</span> viruses,</span>
<span id="cb30-298"><a href="#cb30-298" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_human =</span> eukaryota)</span>
<span id="cb30-299"><a href="#cb30-299" aria-hidden="true" tabindex="-1"></a>read_comp_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(read_comp, <span class="sc">-</span>(sample<span class="sc">:</span>enrichment), <span class="at">names_to =</span> <span class="st">"classification"</span>,</span>
<span id="cb30-300"><a href="#cb30-300" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_prefix =</span> <span class="st">"n_"</span>, <span class="at">values_to =</span> <span class="st">"n_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-301"><a href="#cb30-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(<span class="fu">str_to_sentence</span>(classification))) <span class="sc">%&gt;%</span></span>
<span id="cb30-302"><a href="#cb30-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb30-303"><a href="#cb30-303" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb30-304"><a href="#cb30-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location, enrichment, classification) <span class="sc">%&gt;%</span></span>
<span id="cb30-305"><a href="#cb30-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-306"><a href="#cb30-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb30-307"><a href="#cb30-307" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb30-308"><a href="#cb30-308" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb30-309"><a href="#cb30-309" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-310"><a href="#cb30-310" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb30-311"><a href="#cb30-311" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_comp_summ, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb30-312"><a href="#cb30-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb30-313"><a href="#cb30-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb30-314"><a href="#cb30-314" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb30-315"><a href="#cb30-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb30-316"><a href="#cb30-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>enrichment, <span class="at">scales=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb30-317"><a href="#cb30-317" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-318"><a href="#cb30-318" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb30-319"><a href="#cb30-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-320"><a href="#cb30-320" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb30-321"><a href="#cb30-321" aria-hidden="true" tabindex="-1"></a>read_comp_minor <span class="ot">&lt;-</span> read_comp_summ <span class="sc">%&gt;%</span> <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb30-322"><a href="#cb30-322" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb30-323"><a href="#cb30-323" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_comp_minor, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb30-324"><a href="#cb30-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb30-325"><a href="#cb30-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% Reads"</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">0.1</span>,<span class="fl">0.01</span>),</span>
<span id="cb30-326"><a href="#cb30-326" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb30-327"><a href="#cb30-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>) <span class="sc">+</span></span>
<span id="cb30-328"><a href="#cb30-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>enrichment, <span class="at">scales=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb30-329"><a href="#cb30-329" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-330"><a href="#cb30-330" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb30-331"><a href="#cb30-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-332"><a href="#cb30-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-333"><a href="#cb30-333" aria-hidden="true" tabindex="-1"></a>Overall, in these unenriched samples, between 66% and 86% of reads (mean 74%) are low-quality, duplicates, or unassigned. Of the remainder, 8-21% (mean 15%) were identified as ribosomal, and 2-12% (mean 5%) were otherwise classified as bacterial. The fraction of human reads ranged from 0.4-6.5% (mean 1.8%) and that of viral reads from 0.6-8.6% (mean 4.5%).</span>
<span id="cb30-334"><a href="#cb30-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-335"><a href="#cb30-335" aria-hidden="true" tabindex="-1"></a>The latter of these is strikingly high, about an order of magnitude higher than that observed in Crits-Christoph. In the <span class="co">[</span><span class="ot">public dashboard</span><span class="co">](https://data.securebio.org/mgs-counts/)</span>, high levels of total virus reads are also observed, and attributed almost entirely to elevated levels of tobamoviruses, especially tomato brown rugose fruit virus. Digging into the Kraken reports for these samples, we find that for most samples, *Tobamovirus* indeed makes up the vast majority of viral reads:</span>
<span id="cb30-336"><a href="#cb30-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-339"><a href="#cb30-339" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-340"><a href="#cb30-340" aria-hidden="true" tabindex="-1"></a>toba_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"tobamovirus.tsv"</span>)</span>
<span id="cb30-341"><a href="#cb30-341" aria-hidden="true" tabindex="-1"></a>toba <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(toba_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-342"><a href="#cb30-342" aria-hidden="true" tabindex="-1"></a>g_toba <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(toba, <span class="fu">aes</span>(<span class="at">x=</span>p_viral_reads_Tobamovirus)) <span class="sc">+</span></span>
<span id="cb30-343"><a href="#cb30-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.01</span>, <span class="at">boundary=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb30-344"><a href="#cb30-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"% Viral reads assigned to Tobamovirus"</span>,</span>
<span id="cb30-345"><a href="#cb30-345" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb30-346"><a href="#cb30-346" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb30-347"><a href="#cb30-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# samples"</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>),</span>
<span id="cb30-348"><a href="#cb30-348" aria-hidden="true" tabindex="-1"></a>                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-349"><a href="#cb30-349" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb30-350"><a href="#cb30-350" aria-hidden="true" tabindex="-1"></a>g_toba</span>
<span id="cb30-351"><a href="#cb30-351" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-352"><a href="#cb30-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-353"><a href="#cb30-353" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb30-354"><a href="#cb30-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-355"><a href="#cb30-355" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples, using a modified form of the pipeline used in my last entry:</span>
<span id="cb30-356"><a href="#cb30-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-357"><a href="#cb30-357" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>After initial ribodepletion, surviving reads were aligned to a DB of human-infecting virus genomes with Bowtie2, using permissive alignment parameters.</span>
<span id="cb30-358"><a href="#cb30-358" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Reads that aligned successfully with Bowtie2 were filtered to remove human, livestock, cloning vector, and other potential contaminant sequences, then run through Kraken2 using the standard 16GB database.</span>
<span id="cb30-359"><a href="#cb30-359" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>For surviving reads, the length-adjusted alignment score $S=\frac{\text{bowtie2 alignment score}}{\ln(\text{read length})}$ was calculated, and reads were filtered out if they didn't meet at least one of the following four criteria:</span>
<span id="cb30-360"><a href="#cb30-360" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is *assigned* to a human-infecting virus by both Kraken and Bowtie2</span>
<span id="cb30-361"><a href="#cb30-361" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair contains a Kraken2 *hit* to the same taxid as it is assigned to by Bowtie2.</span>
<span id="cb30-362"><a href="#cb30-362" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is unassigned by Kraken and $S&gt;15$ for the forward read</span>
<span id="cb30-363"><a href="#cb30-363" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is unassigned by Kraken and $S&gt;15$ for the reverse read</span>
<span id="cb30-364"><a href="#cb30-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-365"><a href="#cb30-365" aria-hidden="true" tabindex="-1"></a>Applying all of these filtering steps leaves a total of 16034 read pairs across all unenriched samples (0.007% of surviving reads):</span>
<span id="cb30-366"><a href="#cb30-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-369"><a href="#cb30-369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-370"><a href="#cb30-370" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb30-371"><a href="#cb30-371" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-372"><a href="#cb30-372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-373"><a href="#cb30-373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, date) <span class="sc">%&gt;%</span></span>
<span id="cb30-374"><a href="#cb30-374" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb30-375"><a href="#cb30-375" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, location, enrichment) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb30-376"><a href="#cb30-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-377"><a href="#cb30-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-380"><a href="#cb30-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-381"><a href="#cb30-381" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-382"><a href="#cb30-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb30-383"><a href="#cb30-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-384"><a href="#cb30-384" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb30-385"><a href="#cb30-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb30-386"><a href="#cb30-386" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb30-387"><a href="#cb30-387" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb30-388"><a href="#cb30-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb30-389"><a href="#cb30-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(),</span>
<span id="cb30-390"><a href="#cb30-390" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb30-391"><a href="#cb30-391" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span id="cb30-392"><a href="#cb30-392" aria-hidden="true" tabindex="-1"></a>g_mrg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev)) <span class="sc">+</span></span>
<span id="cb30-393"><a href="#cb30-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb30-394"><a href="#cb30-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-395"><a href="#cb30-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-396"><a href="#cb30-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb30-397"><a href="#cb30-397" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb30-398"><a href="#cb30-398" aria-hidden="true" tabindex="-1"></a>g_mrg</span>
<span id="cb30-399"><a href="#cb30-399" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-400"><a href="#cb30-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-401"><a href="#cb30-401" aria-hidden="true" tabindex="-1"></a>To investigate these putative viral reads, I ran them through BLASTN and analyzed the results in a similar manner to Crits-Christoph:</span>
<span id="cb30-402"><a href="#cb30-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-405"><a href="#cb30-405" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-406"><a href="#cb30-406" aria-hidden="true" tabindex="-1"></a>mrg_num <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb30-407"><a href="#cb30-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb30-408"><a href="#cb30-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, sample, <span class="st">"_"</span>, seq_num))</span>
<span id="cb30-409"><a href="#cb30-409" aria-hidden="true" tabindex="-1"></a>mrg_fasta <span class="ot">&lt;-</span>  mrg_num <span class="sc">%&gt;%</span> </span>
<span id="cb30-410"><a href="#cb30-410" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb30-411"><a href="#cb30-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb30-412"><a href="#cb30-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb30-413"><a href="#cb30-413" aria-hidden="true" tabindex="-1"></a>mrg_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-414"><a href="#cb30-414" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"rothman-putative-viral.fasta"</span>))</span>
<span id="cb30-415"><a href="#cb30-415" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-416"><a href="#cb30-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-419"><a href="#cb30-419" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-420"><a href="#cb30-420" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-421"><a href="#cb30-421" aria-hidden="true" tabindex="-1"></a><span class="co"># Import viral taxids and relationships</span></span>
<span id="cb30-422"><a href="#cb30-422" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"viral-taxa.tsv.gz"</span>)</span>
<span id="cb30-423"><a href="#cb30-423" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb30-424"><a href="#cb30-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-425"><a href="#cb30-425" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb30-426"><a href="#cb30-426" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"rothman-putative-viral.blast.gz"</span>)</span>
<span id="cb30-427"><a href="#cb30-427" aria-hidden="true" tabindex="-1"></a>blast_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span>)</span>
<span id="cb30-428"><a href="#cb30-428" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb30-429"><a href="#cb30-429" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col_names =</span> blast_cols, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb30-430"><a href="#cb30-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-431"><a href="#cb30-431" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb30-432"><a href="#cb30-432" aria-hidden="true" tabindex="-1"></a>blast_results_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_results, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid)</span>
<span id="cb30-433"><a href="#cb30-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-434"><a href="#cb30-434" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject, then rank hits for each query</span></span>
<span id="cb30-435"><a href="#cb30-435" aria-hidden="true" tabindex="-1"></a>blast_results_best <span class="ot">&lt;-</span> blast_results_viral <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> </span>
<span id="cb30-436"><a href="#cb30-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb30-437"><a href="#cb30-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb30-438"><a href="#cb30-438" aria-hidden="true" tabindex="-1"></a>blast_results_ranked <span class="ot">&lt;-</span> blast_results_best <span class="sc">%&gt;%</span> </span>
<span id="cb30-439"><a href="#cb30-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb30-440"><a href="#cb30-440" aria-hidden="true" tabindex="-1"></a>blast_results_highrank <span class="ot">&lt;-</span> blast_results_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-441"><a href="#cb30-441" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>), </span>
<span id="cb30-442"><a href="#cb30-442" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_num =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>), </span>
<span id="cb30-443"><a href="#cb30-443" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(head, <span class="at">n=</span><span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-444"><a href="#cb30-444" aria-hidden="true" tabindex="-1"></a>           <span class="fu">sapply</span>(paste, <span class="at">collapse=</span><span class="st">"_"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-445"><a href="#cb30-445" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num))</span>
<span id="cb30-446"><a href="#cb30-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-447"><a href="#cb30-447" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb30-448"><a href="#cb30-448" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb30-449"><a href="#cb30-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid, viral) <span class="sc">%&gt;%</span></span>
<span id="cb30-450"><a href="#cb30-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb30-451"><a href="#cb30-451" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-452"><a href="#cb30-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb30-453"><a href="#cb30-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-454"><a href="#cb30-454" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb30-455"><a href="#cb30-455" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg_num <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, taxid, assigned_taxid)</span>
<span id="cb30-456"><a href="#cb30-456" aria-hidden="true" tabindex="-1"></a>blast_results_assign <span class="ot">&lt;-</span> <span class="fu">left_join</span>(blast_results_paired, mrg_assign, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-457"><a href="#cb30-457" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb30-458"><a href="#cb30-458" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb30-459"><a href="#cb30-459" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb30-460"><a href="#cb30-460" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_assign <span class="sc">%&gt;%</span></span>
<span id="cb30-461"><a href="#cb30-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb30-462"><a href="#cb30-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb30-463"><a href="#cb30-463" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb30-464"><a href="#cb30-464" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb30-465"><a href="#cb30-465" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb30-466"><a href="#cb30-466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-467"><a href="#cb30-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-470"><a href="#cb30-470" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-471"><a href="#cb30-471" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb30-472"><a href="#cb30-472" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-473"><a href="#cb30-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-474"><a href="#cb30-474" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb30-475"><a href="#cb30-475" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_num, blast_results_out, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-476"><a href="#cb30-476" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb30-477"><a href="#cb30-477" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb30-478"><a href="#cb30-478" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb30-479"><a href="#cb30-479" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb30-480"><a href="#cb30-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-481"><a href="#cb30-481" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb30-482"><a href="#cb30-482" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_0 <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span></span>
<span id="cb30-483"><a href="#cb30-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb30-484"><a href="#cb30-484" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb30-485"><a href="#cb30-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-486"><a href="#cb30-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-487"><a href="#cb30-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb30-488"><a href="#cb30-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(viral_status_out<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb30-489"><a href="#cb30-489" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb30-490"><a href="#cb30-490" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_0</span>
<span id="cb30-491"><a href="#cb30-491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-492"><a href="#cb30-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-495"><a href="#cb30-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-496"><a href="#cb30-496" aria-hidden="true" tabindex="-1"></a>g_hist_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_blast, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb30-497"><a href="#cb30-497" aria-hidden="true" tabindex="-1"></a>g_hist_0</span>
<span id="cb30-498"><a href="#cb30-498" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-499"><a href="#cb30-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-500"><a href="#cb30-500" aria-hidden="true" tabindex="-1"></a>This initially looks pretty good, with high-scoring true-positives massively outnumbering false-positives and unclear cases. However, there are also a lot of *low*-scoring true-positives, which are excluded when the alignment score threshold is placed high enough to exclude the majority of false-positives. This pulls down sensitivity at higher score thresholds; as a result, F1 score is lower than I'd like, peaking at 0.947 for a disjunctive cutoff of 17 and dropping to 0.921 for a cutoff of 20.</span>
<span id="cb30-501"><a href="#cb30-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-504"><a href="#cb30-504" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-505"><a href="#cb30-505" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold, conjunctive, include_special){</span>
<span id="cb30-506"><a href="#cb30-506" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>include_special) tab <span class="ot">&lt;-</span> <span class="fu">filter</span>(tab, viral_status_out <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>))</span>
<span id="cb30-507"><a href="#cb30-507" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb30-508"><a href="#cb30-508" aria-hidden="true" tabindex="-1"></a>    <span class="at">conjunctive =</span> conjunctive,</span>
<span id="cb30-509"><a href="#cb30-509" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_conjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">&amp;</span> adj_score_rev <span class="sc">&gt;</span> score_threshold), </span>
<span id="cb30-510"><a href="#cb30-510" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_disjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb30-511"><a href="#cb30-511" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score =</span> <span class="fu">ifelse</span>(conjunctive, retain_score_conjunctive, retain_score_disjunctive),</span>
<span id="cb30-512"><a href="#cb30-512" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb30-513"><a href="#cb30-513" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb30-514"><a href="#cb30-514" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb30-515"><a href="#cb30-515" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb30-516"><a href="#cb30-516" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb30-517"><a href="#cb30-517" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb30-518"><a href="#cb30-518" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb30-519"><a href="#cb30-519" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb30-520"><a href="#cb30-520" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb30-521"><a href="#cb30-521" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb30-522"><a href="#cb30-522" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">include_special =</span> include_special, </span>
<span id="cb30-523"><a href="#cb30-523" aria-hidden="true" tabindex="-1"></a>                <span class="at">conjunctive =</span> conjunctive, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb30-524"><a href="#cb30-524" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb30-525"><a href="#cb30-525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb30-526"><a href="#cb30-526" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-527"><a href="#cb30-527" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, inc_special, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb30-528"><a href="#cb30-528" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab, <span class="at">include_special=</span>inc_special)</span>
<span id="cb30-529"><a href="#cb30-529" aria-hidden="true" tabindex="-1"></a>  stats_conj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb30-530"><a href="#cb30-530" aria-hidden="true" tabindex="-1"></a>  stats_disj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb30-531"><a href="#cb30-531" aria-hidden="true" tabindex="-1"></a>  stats_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_conj, stats_disj) <span class="sc">%&gt;%</span></span>
<span id="cb30-532"><a href="#cb30-532" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>(threshold<span class="sc">:</span>conjunctive), <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-533"><a href="#cb30-533" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">conj_label =</span> <span class="fu">ifelse</span>(conjunctive, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span>))</span>
<span id="cb30-534"><a href="#cb30-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats_all)</span>
<span id="cb30-535"><a href="#cb30-535" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-536"><a href="#cb30-536" aria-hidden="true" tabindex="-1"></a>inc_special <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb30-537"><a href="#cb30-537" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">0</span>)</span>
<span id="cb30-538"><a href="#cb30-538" aria-hidden="true" tabindex="-1"></a>threshold_opt_0 <span class="ot">&lt;-</span> stats_0 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(conj_label,attempt) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-539"><a href="#cb30-539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">==</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="fu">min</span>(threshold))</span>
<span id="cb30-540"><a href="#cb30-540" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb30-541"><a href="#cb30-541" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">data =</span> threshold_opt_0, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">xintercept=</span>threshold),</span>
<span id="cb30-542"><a href="#cb30-542" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb30-543"><a href="#cb30-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-544"><a href="#cb30-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-545"><a href="#cb30-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-546"><a href="#cb30-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set3"</span>) <span class="sc">+</span></span>
<span id="cb30-547"><a href="#cb30-547" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>conj_label) <span class="sc">+</span></span>
<span id="cb30-548"><a href="#cb30-548" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb30-549"><a href="#cb30-549" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb30-550"><a href="#cb30-550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-551"><a href="#cb30-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-552"><a href="#cb30-552" aria-hidden="true" tabindex="-1"></a>Improving precision for high-scoring sequences will only have a marginal effect on F1 score here, as the limiting factor is sensitivity. The best way to improve this would be to find ways to better exclude low-scoring false-positives from the dataset, then lower the score threshold closer to 15. I'm wary about doing this, however, as I suspect low-scoring false-positives will be substantially harder to remove than high-scoring ones, and I'm somewhat less worried about false-negatives than false-positives here in any case. As such, I'm going to stick with my current pipeline for now.</span>
<span id="cb30-553"><a href="#cb30-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-556"><a href="#cb30-556" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-557"><a href="#cb30-557" aria-hidden="true" tabindex="-1"></a>mrg_unclear <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"UNCLEAR"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-558"><a href="#cb30-558" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample, seq_num)</span>
<span id="cb30-559"><a href="#cb30-559" aria-hidden="true" tabindex="-1"></a>mrg_unclear_fasta <span class="ot">&lt;-</span> mrg_unclear  <span class="sc">%&gt;%</span></span>
<span id="cb30-560"><a href="#cb30-560" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb30-561"><a href="#cb30-561" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb30-562"><a href="#cb30-562" aria-hidden="true" tabindex="-1"></a>mrg_unclear_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_unclear_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-563"><a href="#cb30-563" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_unclear_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"rothman-unclear-viral.fasta"</span>))</span>
<span id="cb30-564"><a href="#cb30-564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-565"><a href="#cb30-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-566"><a href="#cb30-566" aria-hidden="true" tabindex="-1"></a>Moving on to the small number of "unclear" viral matches (155 total, 107 high-scoring), as with Crits-Christoph, the vast majority of these appear to be true positives when run on NCBI BLAST online:</span>
<span id="cb30-567"><a href="#cb30-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-570"><a href="#cb30-570" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-571"><a href="#cb30-571" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-572"><a href="#cb30-572" aria-hidden="true" tabindex="-1"></a>unclear_viral_status <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb30-573"><a href="#cb30-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">53</span>), <span class="cn">FALSE</span>, <span class="cn">TRUE</span>,</span>
<span id="cb30-574"><a href="#cb30-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">13</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">15</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">6</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">12</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">5</span>),</span>
<span id="cb30-575"><a href="#cb30-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">2</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">12</span>), <span class="cn">FALSE</span>, <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">29</span>)</span>
<span id="cb30-576"><a href="#cb30-576" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb30-577"><a href="#cb30-577" aria-hidden="true" tabindex="-1"></a>mrg_unclear_out <span class="ot">&lt;-</span> mrg_unclear <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">viral_status_true =</span> unclear_viral_status)</span>
<span id="cb30-578"><a href="#cb30-578" aria-hidden="true" tabindex="-1"></a>g_unclear <span class="ot">&lt;-</span> mrg_unclear_out <span class="sc">%&gt;%</span></span>
<span id="cb30-579"><a href="#cb30-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_true)) <span class="sc">+</span></span>
<span id="cb30-580"><a href="#cb30-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb30-581"><a href="#cb30-581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-582"><a href="#cb30-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-583"><a href="#cb30-583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb30-584"><a href="#cb30-584" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb30-585"><a href="#cb30-585" aria-hidden="true" tabindex="-1"></a>g_unclear</span>
<span id="cb30-586"><a href="#cb30-586" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-587"><a href="#cb30-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-588"><a href="#cb30-588" aria-hidden="true" tabindex="-1"></a>Since this is two large datasets for which I've now gotten this result, I'll henceforth treat reads for which only one of two reads shows a BLAST viral match as true positives for validation purposes.</span>
<span id="cb30-589"><a href="#cb30-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-590"><a href="#cb30-590" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: relative abundance</span></span>
<span id="cb30-591"><a href="#cb30-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-592"><a href="#cb30-592" aria-hidden="true" tabindex="-1"></a>In total, my pipeline identified 12305 reads as human-viral out of 660M total reads, for a relative HV abundance of $1.87 \times 10^{-5}$. This compares to $4.3 \times 10^{-6}$ on the public dashboard, corresponding to the results for Kraken-only identification: a 4.4-fold increase, similar to that observed for Crits-Christoph. Excluding false-positives identified by BLAST above, the new estimate is reduced slightly, to $1.81 \times 10^{-5}$.</span>
<span id="cb30-593"><a href="#cb30-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-594"><a href="#cb30-594" aria-hidden="true" tabindex="-1"></a>Aggregating by treatment plant location, we see substantial variation in HV relative abundance, from $7.5 \times 10^{-7}$ for location JWPCP to $5.91 \times 10^{-5}$ for location SB:</span>
<span id="cb30-595"><a href="#cb30-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-598"><a href="#cb30-598" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-599"><a href="#cb30-599" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb30-600"><a href="#cb30-600" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-601"><a href="#cb30-601" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, date, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb30-602"><a href="#cb30-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-603"><a href="#cb30-603" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts &amp; RA</span></span>
<span id="cb30-604"><a href="#cb30-604" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span></span>
<span id="cb30-605"><a href="#cb30-605" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>,</span>
<span id="cb30-606"><a href="#cb30-606" aria-hidden="true" tabindex="-1"></a>         <span class="at">hv_true =</span> viral_status <span class="sc">&gt;=</span> <span class="dv">1</span>)</span>
<span id="cb30-607"><a href="#cb30-607" aria-hidden="true" tabindex="-1"></a>read_counts_hv_all <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb30-608"><a href="#cb30-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv_all"</span>)</span>
<span id="cb30-609"><a href="#cb30-609" aria-hidden="true" tabindex="-1"></a>read_counts_hv_val <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status, hv_true) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb30-610"><a href="#cb30-610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv_validated"</span>)</span>
<span id="cb30-611"><a href="#cb30-611" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span></span>
<span id="cb30-612"><a href="#cb30-612" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv_all, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-613"><a href="#cb30-613" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_hv_val, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-614"><a href="#cb30-614" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv_all =</span> <span class="fu">replace_na</span>(n_reads_hv_all, <span class="dv">0</span>),</span>
<span id="cb30-615"><a href="#cb30-615" aria-hidden="true" tabindex="-1"></a>         <span class="at">n_reads_hv_validated =</span> <span class="fu">replace_na</span>(n_reads_hv_validated, <span class="dv">0</span>),</span>
<span id="cb30-616"><a href="#cb30-616" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_all =</span> n_reads_hv_all<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-617"><a href="#cb30-617" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_validated =</span> n_reads_hv_validated<span class="sc">/</span>n_reads_raw)</span>
<span id="cb30-618"><a href="#cb30-618" aria-hidden="true" tabindex="-1"></a>read_counts_melted <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> </span>
<span id="cb30-619"><a href="#cb30-619" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, date, <span class="fu">starts_with</span>(<span class="st">"p"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-620"><a href="#cb30-620" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p"</span>), <span class="at">names_to =</span> <span class="st">"group"</span>, <span class="at">values_to =</span> <span class="st">"p"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-621"><a href="#cb30-621" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">str_to_title</span>(<span class="fu">sub</span>(<span class="st">"p_reads_hv_"</span>, <span class="st">""</span>, group))) <span class="sc">%&gt;%</span></span>
<span id="cb30-622"><a href="#cb30-622" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(location, date) <span class="sc">%&gt;%</span></span>
<span id="cb30-623"><a href="#cb30-623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb30-624"><a href="#cb30-624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-625"><a href="#cb30-625" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate by location</span></span>
<span id="cb30-626"><a href="#cb30-626" aria-hidden="true" tabindex="-1"></a>read_counts_loc <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb30-627"><a href="#cb30-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb30-628"><a href="#cb30-628" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(n_reads_hv_all),</span>
<span id="cb30-629"><a href="#cb30-629" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(n_reads_hv_validated)) <span class="sc">%&gt;%</span></span>
<span id="cb30-630"><a href="#cb30-630" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv_all =</span> n_reads_hv_all<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-631"><a href="#cb30-631" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_validated =</span> n_reads_hv_validated<span class="sc">/</span>n_reads_raw)</span>
<span id="cb30-632"><a href="#cb30-632" aria-hidden="true" tabindex="-1"></a>read_counts_total <span class="ot">&lt;-</span> read_counts_loc <span class="sc">%&gt;%</span></span>
<span id="cb30-633"><a href="#cb30-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb30-634"><a href="#cb30-634" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(n_reads_hv_all),</span>
<span id="cb30-635"><a href="#cb30-635" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(n_reads_hv_validated)) <span class="sc">%&gt;%</span></span>
<span id="cb30-636"><a href="#cb30-636" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv_all =</span> n_reads_hv_all<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-637"><a href="#cb30-637" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_validated =</span> n_reads_hv_validated<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-638"><a href="#cb30-638" aria-hidden="true" tabindex="-1"></a>         <span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb30-639"><a href="#cb30-639" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts_loc <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">as.character</span>(location)) <span class="sc">%&gt;%</span></span>
<span id="cb30-640"><a href="#cb30-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(read_counts_total) <span class="sc">%&gt;%</span></span>
<span id="cb30-641"><a href="#cb30-641" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">fct_inorder</span>(location))</span>
<span id="cb30-642"><a href="#cb30-642" aria-hidden="true" tabindex="-1"></a>read_counts_agg_melted <span class="ot">&lt;-</span> read_counts_agg <span class="sc">%&gt;%</span> </span>
<span id="cb30-643"><a href="#cb30-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(location, <span class="fu">starts_with</span>(<span class="st">"p"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-644"><a href="#cb30-644" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">starts_with</span>(<span class="st">"p"</span>), <span class="at">names_to =</span> <span class="st">"group"</span>, <span class="at">values_to =</span> <span class="st">"p"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-645"><a href="#cb30-645" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">str_to_title</span>(<span class="fu">sub</span>(<span class="st">"p_reads_hv_"</span>, <span class="st">""</span>, group)))</span>
<span id="cb30-646"><a href="#cb30-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-647"><a href="#cb30-647" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb30-648"><a href="#cb30-648" aria-hidden="true" tabindex="-1"></a>palette_loc <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>), <span class="st">"black"</span>)</span>
<span id="cb30-649"><a href="#cb30-649" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg_melted, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>p, <span class="at">color=</span>location)) <span class="sc">+</span></span>
<span id="cb30-650"><a href="#cb30-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb30-651"><a href="#cb30-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb30-652"><a href="#cb30-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>palette_loc, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-653"><a href="#cb30-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>group, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span> </span>
<span id="cb30-654"><a href="#cb30-654" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))<span class="co">#</span></span>
<span id="cb30-655"><a href="#cb30-655" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb30-656"><a href="#cb30-656" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-657"><a href="#cb30-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-658"><a href="#cb30-658" aria-hidden="true" tabindex="-1"></a>Plotting HV relative abundance over time for each plant, we see that while some plants remain roughly stable over time, several others exhibit a gradual increase in HV relative abundance over the course of the study, possibly related to the winter disease season:</span>
<span id="cb30-659"><a href="#cb30-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-662"><a href="#cb30-662" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-663"><a href="#cb30-663" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb30-664"><a href="#cb30-664" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb30-665"><a href="#cb30-665" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb30-666"><a href="#cb30-666" aria-hidden="true" tabindex="-1"></a>g_phv_time <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_melted, <span class="fu">aes</span>(<span class="at">x=</span>date, <span class="at">y=</span>p, <span class="at">color=</span>location, <span class="at">linetype=</span>group)) <span class="sc">+</span></span>
<span id="cb30-667"><a href="#cb30-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb30-668"><a href="#cb30-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Location"</span>) <span class="sc">+</span></span>
<span id="cb30-669"><a href="#cb30-669" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb30-670"><a href="#cb30-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Estimator"</span>) <span class="sc">+</span></span>
<span id="cb30-671"><a href="#cb30-671" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(location<span class="sc">~</span>.) <span class="sc">+</span></span>
<span id="cb30-672"><a href="#cb30-672" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb30-673"><a href="#cb30-673" aria-hidden="true" tabindex="-1"></a>g_phv_time</span>
<span id="cb30-674"><a href="#cb30-674" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-675"><a href="#cb30-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-676"><a href="#cb30-676" aria-hidden="true" tabindex="-1"></a>Digging into individual viruses, we see that fecal-oral viruses predominate, especially noroviruses (e.g. Norwalk virus) and hudisaviruses, though betacoronaviruses including SARS-CoV-2 make a respectable showing in several samples:</span>
<span id="cb30-677"><a href="#cb30-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-680"><a href="#cb30-680" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-681"><a href="#cb30-681" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb30-682"><a href="#cb30-682" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb30-683"><a href="#cb30-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-684"><a href="#cb30-684" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb30-685"><a href="#cb30-685" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb30-686"><a href="#cb30-686" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb30-687"><a href="#cb30-687" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb30-688"><a href="#cb30-688" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb30-689"><a href="#cb30-689" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb30-690"><a href="#cb30-690" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb30-691"><a href="#cb30-691" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb30-692"><a href="#cb30-692" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb30-693"><a href="#cb30-693" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb30-694"><a href="#cb30-694" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb30-695"><a href="#cb30-695" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb30-696"><a href="#cb30-696" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb30-697"><a href="#cb30-697" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb30-698"><a href="#cb30-698" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb30-699"><a href="#cb30-699" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb30-700"><a href="#cb30-700" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb30-701"><a href="#cb30-701" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb30-702"><a href="#cb30-702" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb30-703"><a href="#cb30-703" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb30-704"><a href="#cb30-704" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb30-705"><a href="#cb30-705" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb30-706"><a href="#cb30-706" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-707"><a href="#cb30-707" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb30-708"><a href="#cb30-708" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb30-709"><a href="#cb30-709" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb30-710"><a href="#cb30-710" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb30-711"><a href="#cb30-711" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb30-712"><a href="#cb30-712" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb30-713"><a href="#cb30-713" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-714"><a href="#cb30-714" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb30-715"><a href="#cb30-715" aria-hidden="true" tabindex="-1"></a>hv_reads_genera <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb30-716"><a href="#cb30-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-717"><a href="#cb30-717" aria-hidden="true" tabindex="-1"></a><span class="co"># Count relative abundance for species</span></span>
<span id="cb30-718"><a href="#cb30-718" aria-hidden="true" tabindex="-1"></a>hv_species_counts_raw <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location, name) <span class="sc">%&gt;%</span></span>
<span id="cb30-719"><a href="#cb30-719" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(hv_status),</span>
<span id="cb30-720"><a href="#cb30-720" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(hv_status <span class="sc">&amp;</span> hv_true), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-721"><a href="#cb30-721" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(read_counts_agg <span class="sc">%&gt;%</span> <span class="fu">select</span>(location, n_reads_raw), <span class="at">by=</span><span class="st">"location"</span>)</span>
<span id="cb30-722"><a href="#cb30-722" aria-hidden="true" tabindex="-1"></a>hv_species_counts_all <span class="ot">&lt;-</span> hv_species_counts_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb30-723"><a href="#cb30-723" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(n_reads_hv_all),</span>
<span id="cb30-724"><a href="#cb30-724" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(n_reads_hv_validated),</span>
<span id="cb30-725"><a href="#cb30-725" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw)) <span class="sc">%&gt;%</span></span>
<span id="cb30-726"><a href="#cb30-726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb30-727"><a href="#cb30-727" aria-hidden="true" tabindex="-1"></a>hv_species_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_species_counts_raw, hv_species_counts_all) <span class="sc">%&gt;%</span></span>
<span id="cb30-728"><a href="#cb30-728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv_all =</span> n_reads_hv_all<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-729"><a href="#cb30-729" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_validated =</span> n_reads_hv_validated<span class="sc">/</span>n_reads_raw)</span>
<span id="cb30-730"><a href="#cb30-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-731"><a href="#cb30-731" aria-hidden="true" tabindex="-1"></a><span class="co"># Count relative abundance for genera</span></span>
<span id="cb30-732"><a href="#cb30-732" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_raw <span class="ot">&lt;-</span> hv_reads_genera <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location, name) <span class="sc">%&gt;%</span></span>
<span id="cb30-733"><a href="#cb30-733" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(hv_status),</span>
<span id="cb30-734"><a href="#cb30-734" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(hv_status <span class="sc">&amp;</span> hv_true), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-735"><a href="#cb30-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(read_counts_agg <span class="sc">%&gt;%</span> <span class="fu">select</span>(location, n_reads_raw), <span class="at">by=</span><span class="st">"location"</span>)</span>
<span id="cb30-736"><a href="#cb30-736" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_all <span class="ot">&lt;-</span> hv_genera_counts_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb30-737"><a href="#cb30-737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv_all =</span> <span class="fu">sum</span>(n_reads_hv_all),</span>
<span id="cb30-738"><a href="#cb30-738" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv_validated =</span> <span class="fu">sum</span>(n_reads_hv_validated),</span>
<span id="cb30-739"><a href="#cb30-739" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw)) <span class="sc">%&gt;%</span></span>
<span id="cb30-740"><a href="#cb30-740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb30-741"><a href="#cb30-741" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_genera_counts_raw, hv_genera_counts_all) <span class="sc">%&gt;%</span></span>
<span id="cb30-742"><a href="#cb30-742" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv_all =</span> n_reads_hv_all<span class="sc">/</span>n_reads_raw,</span>
<span id="cb30-743"><a href="#cb30-743" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv_validated =</span> n_reads_hv_validated<span class="sc">/</span>n_reads_raw)</span>
<span id="cb30-744"><a href="#cb30-744" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-745"><a href="#cb30-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-748"><a href="#cb30-748" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb30-749"><a href="#cb30-749" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb30-750"><a href="#cb30-750" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute ranks for species and genera and restrict to high-ranking taxa</span></span>
<span id="cb30-751"><a href="#cb30-751" aria-hidden="true" tabindex="-1"></a>max_rank_species <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb30-752"><a href="#cb30-752" aria-hidden="true" tabindex="-1"></a>max_rank_genera <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb30-753"><a href="#cb30-753" aria-hidden="true" tabindex="-1"></a>hv_species_counts_ranked <span class="ot">&lt;-</span> hv_species_counts_agg <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb30-754"><a href="#cb30-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">rank</span>(<span class="fu">desc</span>(n_reads_hv_all), <span class="at">ties.method=</span><span class="st">"max"</span>),</span>
<span id="cb30-755"><a href="#cb30-755" aria-hidden="true" tabindex="-1"></a>         <span class="at">highrank =</span> rank <span class="sc">&lt;=</span> max_rank_species) <span class="sc">%&gt;%</span></span>
<span id="cb30-756"><a href="#cb30-756" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb30-757"><a href="#cb30-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">highrank_any =</span> <span class="fu">any</span>(highrank),</span>
<span id="cb30-758"><a href="#cb30-758" aria-hidden="true" tabindex="-1"></a>         <span class="at">name_display =</span> <span class="fu">ifelse</span>(highrank_any, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-759"><a href="#cb30-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name_display) <span class="sc">%&gt;%</span></span>
<span id="cb30-760"><a href="#cb30-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_rank =</span> <span class="fu">mean</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb30-761"><a href="#cb30-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(mean_rank) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb30-762"><a href="#cb30-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">fct_inorder</span>(name_display))</span>
<span id="cb30-763"><a href="#cb30-763" aria-hidden="true" tabindex="-1"></a>hv_genera_counts_ranked <span class="ot">&lt;-</span> hv_genera_counts_agg <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location) <span class="sc">%&gt;%</span></span>
<span id="cb30-764"><a href="#cb30-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">rank</span>(<span class="fu">desc</span>(n_reads_hv_all), <span class="at">ties.method=</span><span class="st">"max"</span>),</span>
<span id="cb30-765"><a href="#cb30-765" aria-hidden="true" tabindex="-1"></a>         <span class="at">highrank =</span> rank <span class="sc">&lt;=</span> max_rank_genera) <span class="sc">%&gt;%</span></span>
<span id="cb30-766"><a href="#cb30-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb30-767"><a href="#cb30-767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">highrank_any =</span> <span class="fu">any</span>(highrank),</span>
<span id="cb30-768"><a href="#cb30-768" aria-hidden="true" tabindex="-1"></a>         <span class="at">name_display =</span> <span class="fu">ifelse</span>(highrank_any, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb30-769"><a href="#cb30-769" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name_display) <span class="sc">%&gt;%</span></span>
<span id="cb30-770"><a href="#cb30-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_rank =</span> <span class="fu">mean</span>(rank)) <span class="sc">%&gt;%</span></span>
<span id="cb30-771"><a href="#cb30-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(mean_rank) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb30-772"><a href="#cb30-772" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">fct_inorder</span>(name_display))</span>
<span id="cb30-773"><a href="#cb30-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-774"><a href="#cb30-774" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition</span></span>
<span id="cb30-775"><a href="#cb30-775" aria-hidden="true" tabindex="-1"></a>palette_rank <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))</span>
<span id="cb30-776"><a href="#cb30-776" aria-hidden="true" tabindex="-1"></a>g_vcomp_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_genera_counts_ranked, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>n_reads_hv_all, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb30-777"><a href="#cb30-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb30-778"><a href="#cb30-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_rank, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb30-779"><a href="#cb30-779" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Fraction of HV reads"</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-780"><a href="#cb30-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb30-781"><a href="#cb30-781" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-782"><a href="#cb30-782" aria-hidden="true" tabindex="-1"></a>g_vcomp_species <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_species_counts_ranked, <span class="fu">aes</span>(<span class="at">x=</span>location, <span class="at">y=</span>n_reads_hv_all, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb30-783"><a href="#cb30-783" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb30-784"><a href="#cb30-784" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_rank, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb30-785"><a href="#cb30-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Fraction of HV reads"</span>, <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb30-786"><a href="#cb30-786" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">7</span>)) <span class="sc">+</span></span>
<span id="cb30-787"><a href="#cb30-787" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb30-788"><a href="#cb30-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-789"><a href="#cb30-789" aria-hidden="true" tabindex="-1"></a>g_vcomp_genera</span>
<span id="cb30-790"><a href="#cb30-790" aria-hidden="true" tabindex="-1"></a>g_vcomp_species</span>
<span id="cb30-791"><a href="#cb30-791" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb30-792"><a href="#cb30-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-793"><a href="#cb30-793" aria-hidden="true" tabindex="-1"></a>These results make sense aren't especially surprising. In my next entry, I'll turn to the enriched samples from Rothman, and compare them to these unenriched samples.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>