<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-04-30">
<title>Will’s Public NAO Notebook - Workflow analysis of Bengtsson-Palme et al.&nbsp;(2021)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Bengtsson-Palme et al.&nbsp;(2021)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Grab samples from Swedish treatment plants.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 30, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>Returning to wastewater datasets, today I’m analyzing <a href="https://doi.org/10.1016/j.scitotenv.2016.06.228">Bengtsson-Palme et al.</a> (2016), a study of grab samples taken from three treatment plants in Sweden in September 2012. This was a DNA-sequencing study focused on investigating AMR in sewage communities. Influent, effluent and sludge samples were taken from each location, with a total of 70 samples across all sites: 20 influent, 10 effluent and 40 sludge. Liquid samples were filtered through a 1mm seive, then centrifuged, retaining the pellet (note that we expect this to select against viruses), which was then resuspended and underwent DNA extraction. Sludge samples weren’t concentrated, but went directly to DNA extraction. Samples were sequenced on an Illumina HiSeq 2500, producing 2x101bp reads.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The sample composition of the Bengtsson-Palme dataset was as follows:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Importing the data is a bit more complicated this time as the samples are split across three pipeline runs</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-04-30_bengtsson-palme"</span></span>
<span></span>
<span><span class="co"># Data input paths</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_adapter_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_base_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">libraries_path</span>, <span class="va">read_csv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Process sample types</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Inlet"</span>, <span class="va">sample_type</span><span class="op">)</span>, <span class="st">"Influent"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Primary"</span>, <span class="va">sample_type</span><span class="op">)</span>, <span class="st">"Sludge (Primary/Surplus)"</span>,</span>
<span>                                      <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Surplus"</span>, <span class="va">sample_type</span><span class="op">)</span>, <span class="st">"Sludge (Primary/Surplus)"</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Digested"</span>, <span class="va">sample_type</span><span class="op">)</span>, <span class="st">"Sludge (Other)"</span>,</span>
<span>                                                    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Kemikond"</span>, <span class="va">sample_type</span><span class="op">)</span>, <span class="st">"Sludge (Other)"</span>,</span>
<span>                                                           <span class="st">"Effluent"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         sample_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sample_group</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Influent"</span>, <span class="st">"Effluent"</span>, <span class="st">"Sludge (Primary/Surplus)"</span>, </span>
<span>                                                        <span class="st">"Sludge (Other)"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         sludge <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Sludge"</span>, <span class="va">sample_group</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">location</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make table</span></span>
<span><span class="va">count_samples</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample_group</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"location"</span>, values_from<span class="op">=</span><span class="st">"n"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>`Sample Type`<span class="op">=</span><span class="va">sample_group</span><span class="op">)</span></span>
<span><span class="va">count_samples</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Sample Type"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Henriksdal"],"name":[2],"type":["int"],"align":["right"]},{"label":["Käppala"],"name":[3],"type":["int"],"align":["right"]},{"label":["Uppsala"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"Influent","2":"8","3":"4","4":"8"},{"1":"Effluent","2":"4","3":"4","4":"2"},{"1":"Sludge (Primary/Surplus)","2":"8","3":"8","4":"8"},{"1":"Sludge (Other)","2":"4","3":"8","4":"4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">import_basic</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">arrange</span><span class="op">(</span><span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>           sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">import_basic_paired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">import_basic</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic</span><span class="op">(</span><span class="va">basic_stats_path</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">adapter_stats_path</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_base_stats_path</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_seq_stats_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get key values for readout</span></span>
<span><span class="va">raw_read_counts</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>rmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, rmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            rmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, </span>
<span>            rtot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            btot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span>,</span>
<span>            dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>These 70 samples yielded 23.7M-61.3M (mean 38.7M) reads per sample, for a total of 2.7B read pairs (539 gigabases of sequence). Read qualities were mostly high but tailed off towards the 3’ end, requiring some trimming. Adapter levels were high. Inferred duplication levels were low in sludge samples (1-12%, mean 4%) but much higher in liquid samples samples (22-90%, mean 46%), implying lower available sequence diversity in the latter sample groups.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">sample_group</span>, <span class="va">location</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">location</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span><span class="va">scale_fill_st</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>, name<span class="op">=</span><span class="st">"Sample Type"</span><span class="op">)</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, </span>
<span>                  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">value</span>, fill<span class="op">=</span><span class="va">sample_group</span>, group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample_group</span>,<span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span><span class="op">~</span><span class="va">location</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space<span class="op">=</span><span class="st">"free_x"</span>, switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">scale_color_st</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                   name<span class="op">=</span><span class="st">"Sample Type"</span><span class="op">)</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">sample_group</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. As expected given the observed difference in duplication levels, many more reads were lost during deduplication in liquid samples than sludge samples. Conversely, trimming and filtering consistently removed more reads in sludge than in liquid samples, though the effect was less dramatic than for deduplication. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">sludge</span>, <span class="va">stage</span>, </span>
<span>         <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span>`Sludge?`<span class="op">=</span><span class="va">sludge</span>, Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Sludge?"],"name":[1],"type":["lgl"],"align":["right"]},{"label":["Stage"],"name":[2],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[3],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[4],"type":["chr"],"align":["left"]}],"data":[{"1":"FALSE","2":"Trimming & filtering","3":"2.2-4.2 (mean 3.1)","4":"2.2-4.2 (mean 3.1)"},{"1":"FALSE","2":"Deduplication","3":"23.4-86.1 (mean 44.1)","4":"20.5-82.4 (mean 41)"},{"1":"FALSE","2":"Initial ribodepletion","3":"23.7-86.1 (mean 44.3)","4":"0-0.3 (mean 0.2)"},{"1":"FALSE","2":"Secondary ribodepletion","3":"23.8-86.1 (mean 44.3)","4":"0-0.1 (mean 0.1)"},{"1":"TRUE","2":"Trimming & filtering","3":"4.5-11.5 (mean 7.1)","4":"4.5-11.5 (mean 7.1)"},{"1":"TRUE","2":"Deduplication","3":"5.6-12.5 (mean 8.3)","4":"0.3-4.3 (mean 1.2)"},{"1":"TRUE","2":"Initial ribodepletion","3":"6.2-12.6 (mean 8.6)","4":"0.1-0.6 (mean 0.2)"},{"1":"TRUE","2":"Secondary ribodepletion","3":"6.3-12.6 (mean 8.6)","4":"0.1-0.1 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">sample_group</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">location</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">sample_group</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">location</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Data cleaning was very successful at removing adapters and improving read qualities:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">sample_group</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 45% to 5% in liquid samples and from 6% to 3% in sludge samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sludge</span>,<span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import composition data</span></span>
<span><span class="va">comp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"taxonomic_composition.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">comp_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">classification</span>, levels <span class="op">=</span> <span class="va">classifications</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">location</span><span class="op">~</span><span class="va">sample_group</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, ncol<span class="op">=</span><span class="fl">4</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line<span class="op">=</span><span class="cn">FALSE</span>, width<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_xblank</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">comp_minor</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">comp_minor</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_group</span>, <span class="va">classification</span>, read_fraction<span class="op">=</span><span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample_group</span>, <span class="va">classification</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample_group"],"name":[1],"type":["fct"],"align":["left"]},{"label":["classification"],"name":[2],"type":["fct"],"align":["left"]},{"label":["read_fraction"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Influent","2":"Excluded","3":"81-92% (mean 86%)"},{"1":"Influent","2":"Ribosomal","3":"0.18-0.4% (mean 0.31%)"},{"1":"Influent","2":"Bacterial","3":"7.6-19% (mean 13%)"},{"1":"Influent","2":"Archaeal","3":"0.0081-0.021% (mean 0.013%)"},{"1":"Influent","2":"Viral","3":"0.0046-0.017% (mean 0.011%)"},{"1":"Influent","2":"Human","3":"0.012-0.036% (mean 0.026%)"},{"1":"Effluent","2":"Excluded","3":"94-99% (mean 96%)"},{"1":"Effluent","2":"Ribosomal","3":"0.044-0.18% (mean 0.12%)"},{"1":"Effluent","2":"Bacterial","3":"0.84-5.8% (mean 3.6%)"},{"1":"Effluent","2":"Archaeal","3":"0.00074-0.012% (mean 0.0066%)"},{"1":"Effluent","2":"Viral","3":"0.00022-0.01% (mean 0.0027%)"},{"1":"Effluent","2":"Human","3":"0.012-0.062% (mean 0.038%)"},{"1":"Sludge (Primary/Surplus)","2":"Excluded","3":"68-95% (mean 84%)"},{"1":"Sludge (Primary/Surplus)","2":"Ribosomal","3":"0.2-0.64% (mean 0.38%)"},{"1":"Sludge (Primary/Surplus)","2":"Bacterial","3":"4.6-31% (mean 16%)"},{"1":"Sludge (Primary/Surplus)","2":"Archaeal","3":"0.013-0.038% (mean 0.026%)"},{"1":"Sludge (Primary/Surplus)","2":"Viral","3":"0.0035-0.023% (mean 0.011%)"},{"1":"Sludge (Primary/Surplus)","2":"Human","3":"0.026-0.086% (mean 0.052%)"},{"1":"Sludge (Other)","2":"Excluded","3":"91-97% (mean 96%)"},{"1":"Sludge (Other)","2":"Ribosomal","3":"0.18-0.27% (mean 0.21%)"},{"1":"Sludge (Other)","2":"Bacterial","3":"2.5-8.2% (mean 3.6%)"},{"1":"Sludge (Other)","2":"Archaeal","3":"0.079-0.34% (mean 0.22%)"},{"1":"Sludge (Other)","2":"Viral","3":"0.00095-0.013% (mean 0.0044%)"},{"1":"Sludge (Other)","2":"Human","3":"0.0054-0.023% (mean 0.01%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>In all sample types, the majority of reads were either filtered, duplicates, or unassigned. Among assigned reads, the vast majority were bacterial, which is unsurprising given the sample processing protocols used. Total viral fraction averaged 0.011% in influent and primary sludge, and considerably lower in effluent and treated sludge. The human fraction was also low, averaging 0.033% across all sample types. Interestingly, treated sludge showed far higher fractions of archaeal reads than other sample types, possibly due to the anaerobic conditions experienced during sludge treatment.</p>
<p>As is common for DNA data, viral reads were overwhelmingly dominated by <em>Caudoviricetes</em> phages:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get Kraken reports</span></span>
<span><span class="va">reports_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"kraken_reports.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">reports</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">reports_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to viral taxa</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"minimizers"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span></span>
<span><span class="co"># Identify major viral classes</span></span>
<span><span class="va">viral_classes_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_classes_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">sample_group</span>, <span class="va">location</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_minor</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">sample_group</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">sample_group</span>, <span class="va">location</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_classes_major</span>, <span class="va">viral_classes_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_classes_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_classes</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_classes_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">g_classes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/viral-class-composition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples. A grand total of 565 reads were identified as putatively human-viral, with many samples showing fewer than 5 total HV read pairs. Even this total likely overstates total human-viral presence, however, as more than 300 of these reads had only low alignment scores to their putative viral sources:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import HV read data</span></span>
<span><span class="va">hv_reads_filtered_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">hv_reads_filtered_path</span>, <span class="va">read_tsv</span>,</span>
<span>                            show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collapse multi-entry sequences</span></span>
<span><span class="va">rmax</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">max</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">collapse</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">==</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">x</span>, collapse<span class="op">=</span><span class="st">"/"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>            genome_id <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span>,</span>
<span>            taxid_best <span class="op">=</span> <span class="va">taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            best_alignment_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_fwd</span><span class="op">)</span>,</span>
<span>            best_alignment_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_rev</span><span class="op">)</span>,</span>
<span>            query_len_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_fwd</span><span class="op">)</span>,</span>
<span>            query_len_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_rev</span><span class="op">)</span>,</span>
<span>            query_seq_fwd <span class="op">=</span> <span class="va">query_seq_fwd</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_fwd</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            query_seq_rev <span class="op">=</span> <span class="va">query_seq_rev</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_rev</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            classified <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">classified</span><span class="op">)</span>,</span>
<span>            assigned_name <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">assigned_name</span><span class="op">)</span>,</span>
<span>            assigned_taxid_best <span class="op">=</span> <span class="va">assigned_taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            assigned_taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">assigned_taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            assigned_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">assigned_hv</span><span class="op">)</span>,</span>
<span>            hit_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">hit_hv</span><span class="op">)</span>,</span>
<span>            encoded_hits <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">encoded_hits</span><span class="op">)</span>,</span>
<span>            adj_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>,</span>
<span>            adj_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span></span>
<span>            <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot results</span></span>
<span><span class="va">geom_vhist</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_histogram</span>, binwidth<span class="op">=</span><span class="fl">5</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">g_vhist_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> </span>
<span><span class="va">g_vhist_0</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg</span><span class="op">)</span></span>
<span><span class="va">g_vhist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-hv-scores-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>BLASTing these reads against nt</p>
<p>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I’ve used for previous datasets. As expected, most low-scoring sequences were not found to be true viral hits:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import paired BLAST results</span></span>
<span><span class="va">blast_paired_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_blast_paired.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_paired</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_paired_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_paired</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">match_taxid</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxid_1</span>, <span class="va">taxid_2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"/"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"/"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">p1</span><span class="op">|</span><span class="va">p2</span><span class="op">|</span><span class="va">p3</span>, <span class="cn">NULL</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">adj_score_max</span><span class="op">)</span></span>
<span><span class="va">blast_assign</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">blast_viral</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_out</span> <span class="op">&lt;-</span> <span class="va">blast_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg</span>, <span class="va">blast_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot RNA</span></span>
<span><span class="va">g_vhist_1</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg_blast</span>, mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>fill<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span></span>
<span><span class="va">g_vhist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-blast-results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In order to achieve decent performance metrics under these conditions, I needed to exclude low-scoring reads with Kraken hits as well as those without. Doing this at my normal disjunctive score threshold of 20 gave a passable precision (93%) but poor sensitivity (79%), leading to a poor overall F1 score (85%):</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>retain_score <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>           retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span><span class="op">)</span></span>
<span>  <span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="va">threshold</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adjusted Score Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-f1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span>Threshold<span class="op">=</span><span class="va">threshold</span>, Metric<span class="op">=</span><span class="va">metric</span>, Value<span class="op">=</span><span class="va">value</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Threshold"],"name":[1],"type":["int"],"align":["right"]},{"label":["Metric"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"20","2":"sensitivity","3":"0.7873303"},{"1":"20","2":"specificity","3":"0.9622093"},{"1":"20","2":"precision","3":"0.9304813"},{"1":"20","2":"f1","3":"0.8529412"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Looking into the composition of different read groups, the bulk of high-scoring false positives map to human alphaherpesvirus 1 strain RH2 according to Bowtie2. BLASTN maps these sequences to a variety of bacterial taxa, especially <em>E. coli</em> and various <em>Klebsiella</em> species.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Add missing viral taxa</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">211787</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">509154</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">493803</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">427343</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">340907</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194959</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span></span>
<span></span>
<span><span class="co"># Prepare data</span></span>
<span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_major</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_fp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/fp-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxid_rh2</span> <span class="op">&lt;-</span> <span class="fl">946522</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span>                         <span class="fl">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span>                         <span class="fl">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span>                         <span class="fl">187126</span>, <span class="st">"Nesoenas mayeri"</span>,</span>
<span>                         <span class="fl">244366</span>, <span class="st">"Klebsiella variicola"</span>,</span>
<span>                         <span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="va">tax_names_new</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tax_names</span><span class="op">$</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span></span>
<span><span class="va">ref_name_rh2</span> <span class="op">&lt;-</span> <span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">ref_taxid_rh2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get major matches</span></span>
<span><span class="va">mrg_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid_best</span> <span class="op">==</span> <span class="va">ref_taxid_rh2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">highscore</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid_best<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid_best"</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count</span> <span class="op">&lt;-</span> <span class="va">fp_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, </span>
<span>           <span class="va">taxid_best</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">taxid_best</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count_major</span> <span class="op">&lt;-</span> <span class="va">fp_staxid_count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, </span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_staxid_count_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">status_display</span><span class="op">~</span><span class="va">score_display</span>, scales<span class="op">=</span><span class="st">"free"</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_rh2</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_rh2</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/ptt-blast-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-overall-relative-abundance" class="level1"><h1>Human-infecting viruses: overall relative abundance</h1>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid_all <span class="op">=</span> <span class="va">taxid</span>, taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_grp</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span> <span class="st">"All samples"</span><span class="op">)</span></span>
<span><span class="va">read_counts_st</span> <span class="op">&lt;-</span> <span class="va">read_counts_grp</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">read_counts_loc</span> <span class="op">&lt;-</span> <span class="va">read_counts_grp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample_group <span class="op">=</span> <span class="st">"All sample types"</span><span class="op">)</span></span>
<span><span class="va">read_counts_tot</span> <span class="op">&lt;-</span> <span class="va">read_counts_loc</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>location <span class="op">=</span> <span class="st">"All locations"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_grp</span>, <span class="va">read_counts_st</span>,</span>
<span>                             <span class="va">read_counts_loc</span>, <span class="va">read_counts_tot</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">location</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">libraries</span><span class="op">$</span><span class="va">location</span><span class="op">)</span>, <span class="st">"All locations"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         sample_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sample_group</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">libraries</span><span class="op">$</span><span class="va">sample_group</span><span class="op">)</span>, <span class="st">"All sample types"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Applying a disjunctive cutoff at S=20 identifies 240 read pairs as human-viral. This gives an overall relative HV abundance of <span class="math inline">\(8.86 \times 10^{-8}\)</span>. While very low across all sample types, HV RA was noticeably higher in influent and primary sludge than in sample types that had undergone more extensive processing (effluent and processed sludge):</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample_group</span>, color<span class="op">=</span><span class="va">location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Location"</span>, palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_kit</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/plot-hv-ra-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is by far the lowest HV relative abundance I’ve seen across any of the datasets I’ve analyzed:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collate past RA values</span></span>
<span><span class="va">ra_past</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Collate new RA values</span></span>
<span><span class="va">ra_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                  <span class="st">"Bengtsson-Palme"</span>, <span class="fl">8.86e-8</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">scale_color_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                 name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">ra_comp</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">ra_past</span>, <span class="va">ra_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">ra_comp</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">dataset</span>, x<span class="op">=</span><span class="va">ra</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/ra-hv-past-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-taxonomy-and-composition" class="level1"><h1>Human-infecting viruses: taxonomy and composition</h1>
<p>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 15 samples, 4 from influent and 11 from primary sludge, met this criterion.</p>
<p>At the family level, most samples across all locations were dominated by <em>Poxviridae</em>, <em>Adenoviridae</em>, <em>Picornaviridae,</em> and <em>Herpesviridae</em>, with several other families making an appearance</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">585893</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333922</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334207</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">369960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333924</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687329</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">325455</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333750</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334202</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">197911</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">186938</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333926</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337051</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337043</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334204</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334208</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333928</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337039</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333929</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337042</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334203</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333757</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337050</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333767</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333754</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687363</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687342</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687359</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194441</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334209</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194965</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span></span>
<span></span>
<span><span class="co"># Filter samples and add viral taxa information</span></span>
<span><span class="va">samples_keep</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_reads_hv</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">samples_keep</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_family</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"family"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.08</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_pox</span> <span class="op">&lt;-</span> <span class="fl">10240</span></span>
<span></span>
<span><span class="co"># Get set of poxviridae reads</span></span>
<span><span class="va">pox_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">pox_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_pox</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each poxviridae species</span></span>
<span><span class="va">pox_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_pox <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">pox_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pox</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_pox <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pox_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pox_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_pox</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_pox_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">pox_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Poxviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Poxviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_pox_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/hv-species-pox-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">pox_species_collate</span> <span class="op">&lt;-</span> <span class="va">pox_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_pox</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_adeno</span> <span class="op">&lt;-</span> <span class="fl">10508</span></span>
<span></span>
<span><span class="co"># Get set of adenoviridae reads</span></span>
<span><span class="va">adeno_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">adeno_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each adenoviridae species</span></span>
<span><span class="va">adeno_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_adeno <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">adeno_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_adeno</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_adeno_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">adeno_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% adenoviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of adenoviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_adeno_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/hv-species-adeno-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">adeno_species_collate</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_picorna</span> <span class="op">&lt;-</span> <span class="fl">12058</span></span>
<span></span>
<span><span class="co"># Get set of picornaviridae reads</span></span>
<span><span class="va">picorna_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_picorna</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">picorna_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_picorna</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">picorna_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each picornaviridae species</span></span>
<span><span class="va">picorna_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">picorna_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_picorna <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">picorna_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">picorna_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_picorna</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_picorna</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_picorna</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_picorna</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">picorna_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">picorna_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">picorna_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_picorna <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_picorna <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_picorna</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">picorna_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">picorna_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">picorna_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_picorna</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_picorna_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">picorna_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Picornaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Picornaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_picorna_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/hv-species-picorna-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">picorna_species_collate</span> <span class="op">&lt;-</span> <span class="va">picorna_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_picorna</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_herpes</span> <span class="op">&lt;-</span> <span class="fl">10292</span></span>
<span></span>
<span><span class="co"># Get set of herpesviridae reads</span></span>
<span><span class="va">herpes_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">herpes_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each herpesviridae species</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/hv-species-herpes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In the latter case, some subset of human alphaherpesvirus 1 reads will be arising from strain RH2, which we know to be dominated by false positive hits. Digging deeper, we see that these are essentially all of the alphaherpesvirus 1 reads. Hence, these herpesvirus read counts need to be treated with significant skepticism.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_hv_named</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["seq_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["sample"],"name":[2],"type":["chr"],"align":["left"]},{"label":["genome_id"],"name":[3],"type":["chr"],"align":["left"]},{"label":["taxid"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["taxid_all"],"name":[5],"type":["chr"],"align":["left"]},{"label":["best_alignment_score_fwd"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["best_alignment_score_rev"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["query_len_fwd"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["query_len_rev"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["query_seq_fwd"],"name":[10],"type":["chr"],"align":["left"]},{"label":["query_seq_rev"],"name":[11],"type":["chr"],"align":["left"]},{"label":["classified"],"name":[12],"type":["int"],"align":["right"]},{"label":["assigned_name"],"name":[13],"type":["chr"],"align":["left"]},{"label":["assigned_taxid_best"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["assigned_taxid"],"name":[15],"type":["chr"],"align":["left"]},{"label":["assigned_hv"],"name":[16],"type":["int"],"align":["right"]},{"label":["hit_hv"],"name":[17],"type":["int"],"align":["right"]},{"label":["encoded_hits"],"name":[18],"type":["chr"],"align":["left"]},{"label":["adj_score_fwd"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["adj_score_rev"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["library"],"name":[21],"type":["chr"],"align":["left"]},{"label":["location"],"name":[22],"type":["fct"],"align":["left"]},{"label":["sample_type"],"name":[23],"type":["chr"],"align":["left"]},{"label":["sample_alias"],"name":[24],"type":["dbl"],"align":["right"]},{"label":["bioproject"],"name":[25],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[26],"type":["chr"],"align":["left"]},{"label":["sample_group"],"name":[27],"type":["fct"],"align":["left"]},{"label":["sludge"],"name":[28],"type":["lgl"],"align":["right"]},{"label":["kraken_label"],"name":[29],"type":["chr"],"align":["left"]},{"label":["adj_score_max"],"name":[30],"type":["dbl"],"align":["right"]},{"label":["highscore"],"name":[31],"type":["lgl"],"align":["right"]},{"label":["hv_status"],"name":[32],"type":["lgl"],"align":["right"]},{"label":["name"],"name":[33],"type":["chr"],"align":["left"]},{"label":["rank"],"name":[34],"type":["chr"],"align":["left"]},{"label":["parent_taxid"],"name":[35],"type":["dbl"],"align":["right"]}],"data":[{"1":"ERR1414209.12335053","2":"ERR1414209","3":"MZ983573.1","4":"10529","5":"10529","6":"184","7":"183","8":"92","9":"94","10":"CTCTAAGCTTAGACGGAGGGGGAAACCTTGGTTTGGATCTTCTCGCTCCCTTAGTTGTAAATTCTAGCAACGCGCTAACCCTATCTGCTTCA","11":"TAACAAGAATAGGAGATGGTCTACCATTTGATAATGGGGTTATAAAAGCATATGTTGCCGGGGGAATGAGACTCTCAGGGGGTAAAATAATTTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:34 0:61","19":"40.69186","20":"40.27914","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"40.69186","31":"TRUE","32":"TRUE","33":"Human adenovirus 31","34":"serotype","35":"129875"},{"1":"ERR1414209.12655300","2":"ERR1414209","3":"KY628978.1","4":"10325","5":"10325","6":"-Inf","7":"48","8":"-Inf","9":"50","10":"NA","11":"CTCGCGCCCACTACGCCCAACCTCGAGGCGCTGTGCGCCAAGTTCATCGA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:50 A:34 10325:2 0:15","19":"0.00000","20":"12.26987","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.26987","31":"FALSE","32":"TRUE","33":"Macacine alphaherpesvirus 1","34":"species","35":"10294"},{"1":"ERR1414209.14369686","2":"ERR1414209","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"97","10":"NA","11":"TCTTTAGGTAGTTTATTTACAATANGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:56 A:59 0:2 38804:3 0:34","19":"0.00000","20":"14.86433","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.86433","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414209.17488341","2":"ERR1414209","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"69","8":"-Inf","9":"92","10":"NA","11":"TTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:59 A:34 0:41 38804:3 0:15","19":"0.00000","20":"15.25945","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.25945","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414209.20000857","2":"ERR1414209","3":"KY569018.1","4":"10243","5":"10243","6":"56","7":"-Inf","8":"34","9":"-Inf","10":"GAGAGAATGAGAATGAGAATGAGAATGAGAATGA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:34 0:54","19":"15.88040","20":"0.00000","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.88040","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414209.2062573","2":"ERR1414209","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"-Inf","8":"97","9":"-Inf","10":"TTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:28 38804:3 0:32 A:34 0:51","19":"15.08292","20":"0.00000","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.08292","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414209.27029605","2":"ERR1414209","3":"OQ468651.1","4":"10244","5":"10244","6":"-Inf","7":"72","8":"-Inf","9":"89","10":"NA","11":"TTCAACAATCTCAGTACAAAGATTAGAACTCTTGATGGTTCCTAAATTTTTCTGATTGCTCTTTTCGTTGCAAGCATCTTTATATAAGA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:59 A:34 0:56","19":"0.00000","20":"16.04051","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.04051","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414209.29502351","2":"ERR1414209","3":"EU918736.1","4":"638313","5":"638313","6":"144","7":"61","8":"92","9":"41","10":"TGGGGGAAGTGATCCAACGCTGCAAAACTCTTCTGGCTACGCGTCCTTTTGAACCGGAAGAGGCAGATATCTATCATGCTTTAAAGCGAGCT","11":"AACCTACACTGCCACATTATAGTCGGGGGAGAAGGCTTGAG","12":"1","13":"Human bocavirus 3","14":"638313","15":"638313","16":"1","17":"1","18":"0:34 638313:2 0:9 638313:2 0:11 A:34 0:8","19":"31.84581","20":"16.42623","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"31.84581","31":"TRUE","32":"TRUE","33":"unknown virus","34":"no rank","35":"1511871"},{"1":"ERR1414209.3495318","2":"ERR1414209","3":"MF001305.1","4":"2025360","5":"2025360","6":"86","7":"-Inf","8":"98","9":"-Inf","10":"TCAGCAAGATTTTCATTGACGATCCCATCACTTGCAGCAAAAAATGCCAATACATGTTTTATAAAGAACTTTTCATTATCGTTGAGCTTATGCTCCCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:65","19":"18.75695","20":"0.00000","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"18.75695","31":"FALSE","32":"FALSE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414209.4219337","2":"ERR1414209","3":"JX878426.1","4":"10244","5":"10244","6":"83","7":"83","8":"75","9":"75","10":"AGATTTTCATTGACTATACCATCACTGGCTGCAAAAAAAGCTAAAACATGTTTTATGAAATGTTTTTCATCATCA","11":"AGATTTTCATTGACTATACCATCACTGGCTGCAAAAAAAGCTAAAACATGTTTTATGAAATGTTTTTCATCATCA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:41","19":"19.22414","20":"19.22414","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"19.22414","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414209.8084574","2":"ERR1414209","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"72","8":"-Inf","9":"97","10":"NA","11":"TATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCAT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:64 A:34 0:48 38804:3 0:13","19":"0.00000","20":"15.73870","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.73870","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414209.813353","2":"ERR1414209","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"83","10":"NA","11":"TAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTAAGCC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:34 0:50","19":"0.00000","20":"15.38865","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.38865","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414209.943636","2":"ERR1414209","3":"MH341446.1","4":"10245","5":"10245","6":"-Inf","7":"74","8":"-Inf","9":"74","10":"NA","11":"TGCTTTCCTGCAAAGATCAATCGTTGTTGATCCGGTGGAATGCCTTCCTTATCTTGAATCTTGGCTTTCACTTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:40 9606:4 0:2 A:34 0:41","19":"0.00000","20":"17.19305","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.19305","31":"FALSE","32":"FALSE","33":"Vaccinia virus","34":"species","35":"10242"},{"1":"ERR1414209.9599058","2":"ERR1414209","3":"KP010356.1","4":"10258","5":"10258","6":"-Inf","7":"72","8":"-Inf","9":"97","10":"NA","11":"AGAATAAAATTCTCCTAGATAACTAGATAACTAGATAACTAGATAACTAGATAACTAGATAACTAGATGACTAGTTTATCTCGATTTCGCCCACATT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:64","19":"0.00000","20":"15.73870","21":"ERR1414209","22":"Uppsala","23":"Primary/Surplus","24":"102","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.73870","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414211.11168880","2":"ERR1414211","3":"LT960370.1","4":"1891762","5":"1891762","6":"181","7":"181","8":"94","9":"94","10":"GTTTTAGGGACAGGATACTCATTCATTGTAACCAGGCCTGGTGGAAATATTTGGGTTCTTTTGTTTAAATGTTTCTTTTCTAAATTAACCTTAA","11":"AGCCTGCAGGGTTTTAGGGACAGGATACTCATTCATTGTAACCAGGCCTGGTGGAAATATTTGGGTTCTTTTGTTTAAATGTTTCTTTTCTAAA","12":"1","13":"Betapolyomavirus hominis","14":"1891762","15":"1891762","16":"1","17":"1","18":"0:5 151341:1 0:15 1891762:5 0:20 1891762:3 0:16 1891762:5","19":"39.83893","20":"39.83893","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"39.83893","31":"TRUE","32":"TRUE","33":"Betapolyomavirus hominis","34":"species","35":"1891714"},{"1":"ERR1414211.13113705","2":"ERR1414211","3":"KT600066.1","4":"1820160","5":"1820160","6":"-Inf","7":"165","8":"-Inf","9":"97","10":"NA","11":"TCAACTGTGGTAACCTATGGGAACTATGCGAGGAGTATCGTATCGCTTGGATTGCTCTATCATTCACGGTCGCTGAGAGTCAGACTCAGAACAACCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:50 A:34 0:64","19":"0.00000","20":"36.06785","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"36.06785","31":"TRUE","32":"TRUE","33":"Human feces pecovirus","34":"species","35":"12429"},{"1":"ERR1414211.21159929","2":"ERR1414211","3":"GQ499302.1","4":"688436","5":"688436","6":"-Inf","7":"58","8":"-Inf","9":"96","10":"NA","11":"AATGAAACTTGTGATTTCTTTTCATATGAGATTCCAAATGGTGCAAATACTTTTCTTTGTCTAATATATAAAGTATCTTGTCCACCGTTTTTTGAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:61 A:34 0:28 688436:2 0:33","19":"0.00000","20":"12.70718","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.70718","31":"FALSE","32":"TRUE","33":"unknown virus","34":"species","35":"1299308"},{"1":"ERR1414211.25105168","2":"ERR1414211","3":"MZ983594.1","4":"10529","5":"10529","6":"178","7":"176","8":"94","9":"88","10":"ATAGGAACACAGAGCTGTCCTACCAACTTATGTTGGATGCTTTAGGCGATAGAAGCCGTTATTTTTCCATGTGGAATTCTGCCGTCGACCACCG","11":"TAGGACAACAAGCAGCACCTAACAGACCAAATTACATTGCCTTCAGAGACAATTTTATTGGGCTTATGTATTATAACAGCACTGGAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:55","19":"39.17862","20":"39.30908","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"39.30908","31":"TRUE","32":"TRUE","33":"Human adenovirus 31","34":"serotype","35":"129875"},{"1":"ERR1414211.25825478","2":"ERR1414211","3":"OP047712.1","4":"10524","5":"10524","6":"194","7":"185","8":"97","9":"96","10":"CGACGGAGCCAGCGTCAGGTTCGATAGCATTAATCTGTACGCCAACTTTTTTCCCATGGCTCACAACACCGCCTCCACCTTGGAAGCTATGCTGCGC","11":"AATCTGTACGCCAACTTTTTTCCCATGGCTCACAACACCGCCTCCACCTTGGAAGCTATGCTGCGCAATGACACCAACGACCAATCATTTAACGAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:51 10509:1 0:41","19":"42.40705","20":"40.53153","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414211.5891857","2":"ERR1414211","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"96","10":"NA","11":"GTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:46 A:34 0:41 38804:3 0:19","19":"0.00000","20":"14.89807","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.89807","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414211.6434150","2":"ERR1414211","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"92","10":"NA","11":"AATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCATTA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:64 A:34 0:50 38804:3 0:6","19":"0.00000","20":"15.03830","21":"ERR1414211","22":"Uppsala","23":"Primary/Surplus","24":"104","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.03830","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.10214658","2":"ERR1414212","3":"MH607142.1","4":"2200830","5":"2200830","6":"74","7":"79","8":"98","9":"90","10":"AGCAACACGCTTTTCGAAAAAATTAGTTTTACCCTGCAACGAGATAATTTCCATGAAATCGAACGGGTTGGTTGCATTGTATATTTTAGCATAGCCAA","11":"CCCGATTTTTTGTACTCAGCAACACGCTTTTCGAAAAAATTAGTTTTACCCTGCAACGAGATAATTTCCATGAAATCGAACGGGTTGGTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53 2810512:2 0:26","19":"16.13970","20":"17.55630","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.55630","31":"FALSE","32":"FALSE","33":"Akhmeta virus","34":"species","35":"10242"},{"1":"ERR1414212.15656941","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"56","7":"-Inf","8":"91","9":"-Inf","10":"CAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:5 38804:3 0:49 A:34 0:59","19":"12.41449","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.41449","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.16118683","2":"ERR1414212","3":"AJ293568.1","4":"132475","5":"132475","6":"-Inf","7":"75","8":"-Inf","9":"96","10":"NA","11":"ATATTCTGAAACACGTTTTTCAAAGAAATTTGTTTTTCCTTCCAAAGAAATTGTCTCCATAAAATCGAAAGGATTTTCGCGGTTATACACTTTTGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:145","19":"0.00000","20":"16.43170","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.43170","31":"FALSE","32":"FALSE","33":"Yaba-like disease virus","34":"no rank","35":"99000"},{"1":"ERR1414212.16543063","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"57","7":"-Inf","8":"97","9":"-Inf","10":"CAGCTTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:5 38804:3 0:55 A:34 0:65","19":"12.45980","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.45980","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.190573","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"93","10":"NA","11":"TGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:56 A:34 0:44 38804:3 0:13","19":"0.00000","20":"15.00243","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.20328636","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"94","9":"-Inf","10":"GGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:37 38804:3 0:20 A:34 0:65","19":"14.96711","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.96711","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.22565410","2":"ERR1414212","3":"JQ410350.1","4":"1452750","5":"1452750","6":"-Inf","7":"78","8":"-Inf","9":"92","10":"NA","11":"GACATTCATTCGGACACATCAATGTCCACTCACCATCTTCCTTCACACGTTTCATGAATAGATCCGGGATCCAAAGCGCATAAAATAAATCA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:59","19":"0.00000","20":"17.24981","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.24981","31":"FALSE","32":"FALSE","33":"Ectromelia virus ERPV","34":"no rank","35":"12643"},{"1":"ERR1414212.24410221","2":"ERR1414212","3":"OQ348891.1","4":"10244","5":"10244","6":"73","7":"73","8":"80","9":"80","10":"CTCGTTCACAATTCCATCACTTGCTGCAAAAAATGCCAATACATGTTTTATAAAATGCCTCTCATCATCATTGAGCTTGT","11":"CTCGTTCACAATTCCATCACTTGCTGCAAAAAATGCCAATACATGTTTTATAAAATGCCTCTCATCATCATTGAGCTTGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:46","19":"16.65896","20":"16.65896","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.65896","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414212.25952717","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"93","10":"NA","11":"GTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTAAGCCTC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:64 A:34 0:56 38804:3 0:1","19":"0.00000","20":"15.00243","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.27466979","2":"ERR1414212","3":"EF420157.1","4":"99000","5":"99000","6":"76","7":"-Inf","8":"96","9":"-Inf","10":"TTGTCCTTTGCATCTTTAATATATGTATCTATTAGTAAGGAATAGGTTTCTGAATGAATATTCTCCATCATGATCTGAAAACCATAAAAGAACTTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:56 2627091:3 0:3 A:34 0:63","19":"16.65079","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.65079","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414212.29643441","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"57","7":"-Inf","8":"93","9":"-Inf","10":"TTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:51 38804:3 0:5 A:34 0:59","19":"12.57557","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.57557","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.31538802","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"101","9":"-Inf","10":"TTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCATTAAGCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:53 38804:3 0:11 A:34 0:62","19":"14.73418","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.73418","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414212.32535983","2":"ERR1414212","3":"KY549149.1","4":"10243","5":"10243","6":"70","7":"70","8":"97","9":"90","10":"ACACAAACGACACAAGGGACACAAGGGACACAAGAGATACAAGAGATACAAGAGATACAAGAGATACAAGAGATACAAGAGATACAAAAGATACAAG","11":"AGATACAAGAGATACAAGAGATACAAGAGATACAAGAGATACAAGAGATACAAAAGATACAAGAGATACAAGAGATACAAGAGATACAAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:57","19":"15.30151","20":"15.55621","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.55621","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414212.4891783","2":"ERR1414212","3":"LT797682.1","4":"10310","5":"10310","6":"71","7":"-Inf","8":"90","9":"-Inf","10":"CACTTGCGTATGCGGTTGGCCAGGCCCTCGGCCTCGATCTGGGCCCGGAGCTGGGGTGAGGGGCCAAGTCCCTCGATATTGACCAGCTTG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:56 A:34 0:9 2482763:1 0:18","19":"15.77845","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.77845","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 2","34":"species","35":"10294"},{"1":"ERR1414212.5734924","2":"ERR1414212","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"93","9":"-Inf","10":"CAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:7 38804:3 0:49 A:34 0:56","19":"15.00243","20":"0.00000","21":"ERR1414212","22":"Uppsala","23":"Surplus","24":"105","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.11880754","2":"ERR1414213","3":"JQ410350.1","4":"1452750","5":"1452750","6":"75","7":"-Inf","8":"98","9":"-Inf","10":"GACATTCATTCGGACACATCAATGTCCACTCACCATCTTCCTTCACACGTTTCATGAATAGATCCGGGATCCAAAGCGCATAAAATAAATCACGTGCT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:50","19":"16.35780","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.35780","31":"FALSE","32":"FALSE","33":"Ectromelia virus ERPV","34":"no rank","35":"12643"},{"1":"ERR1414213.11945856","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"69","8":"-Inf","9":"87","10":"NA","11":"ATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTAAGCC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:54","19":"0.00000","20":"15.45039","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.45039","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.15050710","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"65","7":"54","8":"87","9":"86","10":"CATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCATTAAGCCTCTTTTTTT","11":"TATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:62 38804:3 0:45","19":"14.55471","20":"12.12299","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.55471","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.15318118","2":"ERR1414213","3":"EF420157.1","4":"99000","5":"99000","6":"-Inf","7":"72","8":"-Inf","9":"95","10":"NA","11":"TTGTCCTTTGCATCTTTAATATATGTATCTATTAGTAAGGAATAGGTTTCTGAATGAATATTCTCCATCATGATCTGAAAACCATAAAAGAACTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:62 A:34 0:57 2627091:3 0:2","19":"0.00000","20":"15.81070","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.81070","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414213.15724727","2":"ERR1414213","3":"MN702453.1","4":"10244","5":"10244","6":"-Inf","7":"78","8":"-Inf","9":"88","10":"NA","11":"TCTCAGCAAGATTTTCATTGACTATACCATCACTGGCTGCAAAAAAAGCTAAAACATGTTTTATGAAATGTTTTTCATCATCATTCAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:49 A:34 0:55","19":"0.00000","20":"17.42107","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.42107","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414213.16316667","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"-Inf","8":"92","9":"-Inf","10":"TAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTAAGCCTCTTTTTTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:34 0:60","19":"15.25945","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.25945","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.1979946","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"93","9":"-Inf","10":"TAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:34 38804:3 0:22 A:34 0:61","19":"15.00243","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.2103980","2":"ERR1414213","3":"OP328312.1","4":"10244","5":"10244","6":"-Inf","7":"75","8":"-Inf","9":"91","10":"NA","11":"AGTTCTCGTTCACAATTCCATCACTTGCTGCAAAAAATGCCAATACATGTTTTATAAAATGCCTCTCATCATCATTGAGTTTGTACTGCCA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:55 1917830:3 A:34 0:58","19":"0.00000","20":"16.62654","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.62654","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414213.21462110","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"-Inf","8":"98","9":"-Inf","10":"CTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:32 38804:3 0:29 A:34 0:5","19":"15.04918","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.04918","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.23263598","2":"ERR1414213","3":"JX878426.1","4":"10244","5":"10244","6":"71","7":"-Inf","8":"97","9":"-Inf","10":"TTCCCTAACCATATTTTCGGCAAGATTTTCGTTTACAATACCATCACTGGCAGCAAAAAAAGCTAAGACGTGTTTTATAAAATATTTTTCATCCTCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:24 2591634:2 0:37 A:34 0:9","19":"15.52011","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.52011","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414213.24454607","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"69","8":"-Inf","9":"93","10":"NA","11":"ATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCATTAAGCCTCTTTTTTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:60","19":"0.00000","20":"15.22305","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.22305","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.247495","2":"ERR1414213","3":"EF420156.1","4":"99000","5":"99000","6":"-Inf","7":"72","8":"-Inf","9":"94","10":"NA","11":"CTTTTTTGGTATTCAGCTACCCGCTTTTCGAAGAAATTAGTTTTTCCTTGCAATGAAATTAATTCCATAAAATCAAAAGGATTTGTGGCATTGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:34 0:33 2906456:5 0:23","19":"0.00000","20":"15.84753","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.84753","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414213.24875634","2":"ERR1414213","3":"JX878426.1","4":"10244","5":"10244","6":"70","7":"-Inf","8":"97","9":"-Inf","10":"TCGGCAAGATTTTCGTTTACAATACCATCACTGGCAGCAAAAAAAGCTAAGACGTGTTTTATAAAATATTTTTCATCCTCCGTCAGTTTATGCTCCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:8 2591634:2 0:53 A:34 0:57","19":"15.30151","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.30151","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414213.26410523","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"96","10":"NA","11":"TAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:60 A:34 0:34 38804:3 0:26","19":"0.00000","20":"14.89807","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.89807","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.26777335","2":"ERR1414213","3":"MH709314.1","4":"10335","5":"10335","6":"-Inf","7":"64","8":"-Inf","9":"65","10":"NA","11":"GCTCCCTGCGCCACCGCAGCCCGCGCAGCCCGCGCAGCCCGCGCAGCCTGAAGCGCAGCCTGAGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:44 A:34 0:32","19":"0.00000","20":"15.33159","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.33159","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 3","34":"species","35":"10319"},{"1":"ERR1414213.27384498","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"57","7":"-Inf","8":"94","9":"-Inf","10":"CTTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:52 38804:3 0:5 A:34 0:61","19":"12.54596","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.54596","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.29342201","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"59","7":"-Inf","8":"86","9":"-Inf","10":"TGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:6 38804:3 0:43 A:34 0:68","19":"13.24549","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"13.24549","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.30008802","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"98","9":"-Inf","10":"GCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:23 38804:3 0:91","19":"14.83108","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.83108","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.32406191","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"54","7":"-Inf","8":"89","9":"-Inf","10":"AGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:4 38804:3 0:48 A:34 0:61","19":"12.03038","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.03038","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.3279446","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"89","9":"-Inf","10":"CTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:25 38804:3 0:27 A:34 0:65","19":"15.14937","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.14937","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.424632","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"56","7":"-Inf","8":"94","9":"-Inf","10":"CTTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:5 38804:3 0:52 A:34 0:50","19":"12.32586","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.32586","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.447877","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"93","10":"NA","11":"TAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:51 A:34 0:23 38804:3 0:34","19":"0.00000","20":"15.00243","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.4756107","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"57","7":"-Inf","8":"94","9":"-Inf","10":"CTTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:52 38804:3 0:5 A:34 0:52","19":"12.54596","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.54596","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.5720413","2":"ERR1414213","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"98","9":"-Inf","10":"CTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGAC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:25 38804:3 0:36 A:34 0:64","19":"14.83108","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.83108","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414213.7760456","2":"ERR1414213","3":"AF037218.1","4":"10372","5":"10372","6":"71","7":"-Inf","8":"98","9":"-Inf","10":"CTAGCCCTAACCCTAGCCCTAACCCTAGCCCTAACCCTAGCCCTAACCCTAAAGCCCGTAGCCTGAGCAACAAGACCGCCCATCACTGTAGCCCTGGC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:63","19":"15.48539","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.48539","31":"FALSE","32":"FALSE","33":"Human betaherpesvirus 7","34":"species","35":"40272"},{"1":"ERR1414213.8536362","2":"ERR1414213","3":"EF420157.1","4":"99000","5":"99000","6":"74","7":"-Inf","8":"98","9":"-Inf","10":"TTTTTCAAAGAAATTTGTTTTTCCTTCCAAAGAAATTGTCTCCATAAAATCGAAAGGATTTTCGCGGTTATACACTTTTGGACAATTCAATGAATTCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:62","19":"16.13970","20":"0.00000","21":"ERR1414213","22":"Uppsala","23":"Surplus","24":"106","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.13970","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414215.10895059","2":"ERR1414215","3":"MN244296.1","4":"2200830","5":"2200830","6":"72","7":"-Inf","8":"98","9":"-Inf","10":"AGAATACATTTCCGAATGAATATTTTCCATGGCGATTTGAAAACCATAGAAACAACGCGCTTCAGGAATCTGTACTTCACGCATAAAATTCGTAGCCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"2732007:5 0:59 A:34 0:64","19":"15.70349","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.70349","31":"FALSE","32":"FALSE","33":"Akhmeta virus","34":"species","35":"10242"},{"1":"ERR1414215.11881351","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"57","8":"-Inf","9":"97","10":"NA","11":"CAGCTTCAGTTATGATTGCTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:63 A:34 0:56 38804:3 0:5","19":"0.00000","20":"12.45980","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.45980","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.12333477","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"69","8":"-Inf","9":"98","10":"NA","11":"CTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:65 A:34 0:30 38804:3 0:32","19":"0.00000","20":"15.04918","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.04918","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.14502244","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"92","9":"-Inf","10":"CTTCTAGACTTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:18 38804:3 0:37 A:34 0:64","19":"15.03830","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.03830","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.14652477","2":"ERR1414215","3":"DQ184476.1","4":"10258","5":"10258","6":"-Inf","7":"52","8":"-Inf","9":"30","10":"NA","11":"TTGTGGTCGTGGTCGTGGTCGTGGTCGTGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:31","19":"0.00000","20":"15.28873","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.28873","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414215.16336912","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"69","8":"-Inf","9":"94","10":"NA","11":"ACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:62 A:34 0:50 38804:3 0:8","19":"0.00000","20":"15.18722","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.18722","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.17366977","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"66","7":"-Inf","8":"79","9":"-Inf","10":"TCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTAAGCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:45 A:34 0:64","19":"15.10488","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.10488","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.18369909","2":"ERR1414215","3":"DQ184476.1","4":"10258","5":"10258","6":"58","7":"-Inf","8":"30","9":"-Inf","10":"TGCTCGTGCTCGCGCTCGCGCTCGCGCTCG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:30 0:58","19":"17.05282","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.05282","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414215.19344687","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"94","10":"NA","11":"ACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATCCTGGCATTA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:64 A:34 0:50 38804:3 0:8","19":"0.00000","20":"14.96711","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.96711","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.1996785","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"68","8":"-Inf","9":"93","10":"NA","11":"TCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:62 A:34 0:31 38804:3 0:26","19":"0.00000","20":"15.00243","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.00243","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.2295939","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"68","8":"94","9":"98","10":"AATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAATCCTGGCATTAAGCCTCTTTTTTTCAAC","11":"CTGTGATTTTGCATGTATGATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:66 38804:3 0:31","19":"15.18722","20":"14.83108","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.18722","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.23221791","2":"ERR1414215","3":"OP531887.1","4":"10244","5":"10244","6":"75","7":"-Inf","8":"92","9":"-Inf","10":"CCTAAGCATATTTTCTGCCAAGTTCTCGTTCACAATTCCATCACTTGCTGCAAAAAATGCCAATACATGTTTTATAAAATGCCTCTCATCAT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:58 A:34 0:52","19":"16.58636","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.58636","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414215.23291914","2":"ERR1414215","3":"EF420156.1","4":"99000","5":"99000","6":"79","7":"-Inf","8":"96","9":"-Inf","10":"CCTTTGCATCTTTAATATATGTATCTATTAGTAAGGAATAGGTTTCTGAATGAATATTCTCCATCATGATCTGAAAACCATAAAAGAACTTTGCTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:52 2627091:3 0:7 A:34 0:61","19":"17.30806","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.30806","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414215.29450676","2":"ERR1414215","3":"KY549144.1","4":"10243","5":"10243","6":"-Inf","7":"70","8":"-Inf","9":"87","10":"NA","11":"CAGTCCGGGCATAAGGGAACGCTTTTTGAGCCAAAAGATTGCACAGAACGACCCAGAAAAGAAGATTCCTTCGACAGCTGCAAATGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:62 A:34 0:54","19":"0.00000","20":"15.67430","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.67430","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414215.30031571","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"68","7":"-Inf","8":"96","9":"-Inf","10":"TTTCTTTAGGTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:31 38804:3 0:28 A:34 0:61","19":"14.89807","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.89807","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.30883218","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"-Inf","8":"101","9":"-Inf","10":"TTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAATGACAGTCCTGGCATTAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:14 38804:3 0:50 A:34 0:63","19":"14.95086","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"14.95086","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.31641709","2":"ERR1414215","3":"AY386371.1","4":"38804","5":"38804","6":"69","7":"-Inf","8":"96","9":"-Inf","10":"GTAGTTTATTTACAATATGTTTATTATAAATCATACATGCAAAATCACAGTGCATTCCTTCATCTCTTGATATCAATTCATTGGAAAAAGACAATC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:40 38804:3 0:19 A:34 0:59","19":"15.11716","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"15.11716","31":"FALSE","32":"TRUE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414215.33235627","2":"ERR1414215","3":"LT896721.1","4":"10243","5":"10243","6":"-Inf","7":"80","8":"-Inf","9":"94","10":"NA","11":"CGCGAACCATGTTTTCAGCAAGATTTTCATTGACTATTCCATCACTTGCAGCAAAGAAAGCTAAAACATGTTTTATAAAATGTTTTTCGTCAGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:61","19":"0.00000","20":"17.60837","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.60837","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414215.33976789","2":"ERR1414215","3":"MF001305.1","4":"2025360","5":"2025360","6":"115","7":"-Inf","8":"94","9":"-Inf","10":"CATGTTTTCAGCAAGATTTTCATTGACTATCCCATCACTGGCCGCAAAAAATGCCAACACATGTTTGATGAAGTGTTTTTCTCCATCTGTAAGT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:42","19":"25.31203","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"25.31203","31":"TRUE","32":"TRUE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414215.5781819","2":"ERR1414215","3":"OR264389.1","4":"10244","5":"10244","6":"73","7":"-Inf","8":"81","9":"-Inf","10":"CGACCTAAGCATATTTTCTGCCAAGTTCTCGTTCACAATTCCATCACTTGCTGCAAAAAATGCCAATACATGTTTTATAAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:47 A:34 1917830:4 0:57","19":"16.61187","20":"0.00000","21":"ERR1414215","22":"Uppsala","23":"Surplus","24":"108","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.61187","31":"FALSE","32":"FALSE","33":"Monkeypox virus","34":"species","35":"10242"},{"1":"ERR1414241.13459159","2":"ERR1414241","3":"ON075053.1","4":"1476554","5":"1476554","6":"57","7":"57","8":"36","9":"36","10":"CTCAGAACCCCTAGACATCGTTGCCTTGGTGAGCCG","11":"CTCAGAACCCCTAGACATCGTTGCCTTGGTGAGCCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"976:2","19":"15.90615","20":"15.90615","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.90615","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414241.13849324","2":"ERR1414241","3":"MF118167.1","4":"2017081","5":"2017081","6":"103","7":"-Inf","8":"92","9":"-Inf","10":"GGCCCATAGATTAAAGAAAAAGAAAGAAAAGAAAAATTTTATAAATGCTCATTGAGGGAACAACTGTGTTGTTTACACAATGTGCGGAGTCG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:44 9606:3 0:38","19":"22.77860","20":"0.00000","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"22.77860","31":"TRUE","32":"TRUE","33":"Human fecal virus Jorvi2","34":"species","35":"642248"},{"1":"ERR1414241.16290313","2":"ERR1414241","3":"OP581396.1","4":"28284","5":"28284","6":"146","7":"146","8":"73","9":"73","10":"GGGAAGGCGTGGAAAAACTTGGAAACCCCTTTGTGCCCTCCCAAATTTTCCATGCACTCATCCATAATGATGG","11":"GGGAAGGCGTGGAAAAACTTGGAAACCCCTTTGTGCCCTCCCAAATTTTCCATGCACTCATCCATAATGATGG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"130309:1 0:1 130309:8 0:7 130309:5 0:3 130309:2 0:5 10509:1 0:6","19":"34.02899","20":"34.02899","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"34.02899","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414241.22254649","2":"ERR1414241","3":"LC708006.1","4":"2021738","5":"2021738","6":"84","7":"106","8":"90","9":"94","10":"CTGCTCACACACAATAACATCACCATCCACCCACTCACACTCACACACGAATGTCCTCTATCTCTTTTGTCCCTCCTTTACAAGTAATGG","11":"CATCCACCCACTCACACTCACACACGAATGTCCTCTATCTCTTTTGTCCCTCCTTTACAAGTAATGGACATTCTCATCTTTTTTTTGCATTCAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:83","19":"18.66746","20":"23.33109","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"23.33109","31":"TRUE","32":"TRUE","33":"Hudisavirus sp.","34":"species","35":"12429"},{"1":"ERR1414241.23478325","2":"ERR1414241","3":"OP581404.1","4":"28284","5":"28284","6":"54","7":"54","8":"27","9":"27","10":"CTTGCAAAAGAGCGTGTTCCCAATCTC","11":"CTTGCAAAAGAGCGTGTTCCCAATCTC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:21","19":"16.38431","20":"16.38431","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"16.38431","31":"FALSE","32":"FALSE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414241.26678678","2":"ERR1414241","3":"AB618031.1","4":"946522","5":"946522","6":"76","7":"77","8":"85","9":"85","10":"TTCACACCATCAAAGATAATTCGCGTTTGTCCTTCCTGAAGCCAGGCCTCGTCAACACAAAATGTGAGCGAGTAACAACCTGTGG","11":"CACACCATCAAAGATAATTCGCGTTTGTCCTTCCTGAAGCCAGGCCTCGTCAACACAAAATGTGAGCGAGTAACAACCTGTGGGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53","19":"17.10690","20":"17.33199","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"17.33199","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414241.29037182","2":"ERR1414241","3":"MN244296.1","4":"2200830","5":"2200830","6":"69","7":"-Inf","8":"97","9":"-Inf","10":"TCTCTACTACCACCATCCGCATCGGTCACACCCATTTTTTGATATTCGCCTACACGTTTCTCAAAGAAATTCGTTTTCCCTTCAAGAGATATATTTG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:126","19":"15.08292","20":"0.00000","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.08292","31":"FALSE","32":"FALSE","33":"Akhmeta virus","34":"species","35":"10242"},{"1":"ERR1414241.30969331","2":"ERR1414241","3":"KY040275.1","4":"10280","5":"10280","6":"178","7":"158","8":"89","9":"79","10":"GAGGCCACTCGCAAGTACCCCCGCGCCGCTTTCGACACCGGCATCGTTCTTCGCGACCACGCTCGAGATCTCCTCGGGTACACGCGCTG","11":"CACTCCTGCCTCAATGTCCTTTGAAGAGCAGCACGTTGTTTGGACGAACCTGACCGTGGAAGACAACTGCACGATCATG","12":"1","13":"Molluscum contagiosum virus subtype 1","14":"10280","15":"10280","16":"1","17":"1","18":"0:3 10280:1 0:12 10280:7 0:12 10280:5 0:15 A:34 10280:5 0:1 10280:3 0:7 10280:3 0:5 10280:1 0:21","19":"39.65570","20":"36.16018","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"39.65570","31":"TRUE","32":"TRUE","33":"Molluscum contagiosum virus subtype 1","34":"no rank","35":"10279"},{"1":"ERR1414241.33801257","2":"ERR1414241","3":"AB618031.1","4":"946522","5":"946522","6":"99","7":"99","8":"65","9":"65","10":"CGTTAGCCAGAGTTGTCCGGCGCTCTTCGGCTGCGGTAGTCCAGGCAGTTCAATCAACTGTTTAC","11":"CGTTAGCCAGAGTTGTCCGGCGCTCTTCGGCTGCGGTAGTCCAGGCAGTTCAATCAACTGTTTAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:31","19":"23.71606","20":"23.71606","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"23.71606","31":"TRUE","32":"TRUE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414241.5141162","2":"ERR1414241","3":"OP581398.1","4":"28284","5":"28284","6":"196","7":"186","8":"98","9":"93","10":"CGGCGTGGATTTCACGCAAAGTCGTCTGAACAACTTGCTAGGAATACGCAAGCGAATGCCCTTTCAAAAAGGTTTCCAAATCATGTATGAGGATTTGG","11":"CACCAACGAGGCCTTTCACCCAGACATTGTTTTGCTACCTGGTTGCGGCGTGGATTTCACGCAAAGTCGTCTGAACAACTTGCTAGGAATACG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:1 130309:5 0:15 130309:4 0:9 130309:5 0:4 130309:5 0:6 130309:7 0:7 130309:5 0:2 130309:5 0:6 130309:7 0:8 130309:5 0:2 130309:1","19":"42.74839","20":"41.03605","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414241.9073197","2":"ERR1414241","3":"KF268320.1","4":"130310","5":"130310","6":"98","7":"98","8":"49","9":"49","10":"GGAATAAGCTCTCTTTGTGTCTCCGGTGATTCCTTCCAATAGGTGAGTG","11":"GGAATAAGCTCTCTTTGTGTCTCCGGTGATTCCTTCCAATAGGTGAGTG","12":"1","13":"Human mastadenovirus D","14":"130310","15":"130310","16":"1","17":"1","18":"0:7 130310:5 0:1 130310:2","19":"25.18102","20":"25.18102","21":"ERR1414241","22":"Uppsala","23":"Inlet-C","24":"145","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"25.18102","31":"TRUE","32":"TRUE","33":"Human mastadenovirus D","34":"species","35":"10509"},{"1":"ERR1414243.16108947","2":"ERR1414243","3":"MF001305.1","4":"2025360","5":"2025360","6":"63","7":"63","8":"65","9":"65","10":"CTGATTTTTGATATTCGGCCACTTTCTTTTCGAAGAAATTGGTTTTTCCCTGAAGTGAAATCATG","11":"CTGATTTTTGATATTCGGCCACTTTCTTTTCGAAGAAATTGGTTTTTCCCTGAAGTGAAATCATG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:27 761193:4","19":"15.09204","20":"15.09204","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.09204","31":"FALSE","32":"FALSE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414243.16516783","2":"ERR1414243","3":"KC610695.1","4":"1307169","5":"1307169","6":"54","7":"54","8":"32","9":"32","10":"TCCATTTTCGTTTGTGAATTTAATTGAATTAG","11":"TCCATTTTCGTTTGTGAATTTAATTGAATTAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:31","19":"15.58111","20":"15.58111","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.58111","31":"FALSE","32":"FALSE","33":"Rotavirus A RVA/Pig-wt/ITA/3BS/2009/G9P[23]","34":"no rank","35":"42567"},{"1":"ERR1414243.16796328","2":"ERR1414243","3":"AB618031.1","4":"946522","5":"946522","6":"94","7":"-Inf","8":"96","9":"-Inf","10":"GGGCGCACCATCACATTCGCCAGATGTGTTGCTGGTTTATGCAGCAAAGAGACATCACGGAAGATGCCACTCATCCGCCACATATCCTGATCTTCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:10 91347:1 0:103","19":"20.59440","20":"0.00000","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"20.59440","31":"TRUE","32":"TRUE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414243.17723061","2":"ERR1414243","3":"OP581349.1","4":"10524","5":"10524","6":"180","7":"180","8":"90","9":"90","10":"CGCGGGGAGGCTGAAGCCTGAGAGCGAGAGAGCGGATTTGGGAGCTGTCCAGGGAGTCCAGGAAGGCTTCGCTGAGGCCGGCGGGCAAAG","11":"CGCGGGGAGGCTGAAGCCTGAGAGCGAGAGAGCGGATTTGGGAGCTGTCCAGGGAGTCCAGGAAGGCTTCGCTGAGGCCGGCGGGCAAAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:56","19":"40.00169","20":"40.00169","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"40.00169","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.19352799","2":"ERR1414243","3":"OP555454.1","4":"10524","5":"10524","6":"176","7":"192","8":"88","9":"96","10":"CCTCGGGAGGCTACAATGGTTCCTCGGAGGCGGAAGGTTGAGGGAGGGTTTACATCATTTGACGCGGACGGGCGCCTGGGGGCAGTGG","11":"GGAAGGTTGAGGGAGGGTTTACATCATTTGACGCGGACGGGCGCCTGGGGGCAGTGGCGGTTCGGGACCCGCCTGAAGAGGAGGTAGAAGCACGTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:93","19":"39.30908","20":"42.06515","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"42.06515","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.20282851","2":"ERR1414243","3":"OP047710.1","4":"10524","5":"10524","6":"194","7":"172","8":"97","9":"86","10":"TGCATTCGTGCTGATATTTGTAAAAGCGGGCGCAATACTCGCAGCGTTGCACGTTCTGCACCTCTTGAATGAGGTGTACCTGGCGCCCTCGCACGAG","11":"AATACTCGCAGCGTTGCACGTTCTGCACCTCTTGAATGAGGTGTACCTGGCGCCCTCGCACGAGAAATCGGAGAGGGAAATTAAGA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:3 130309:2 0:3 130309:5 0:72","19":"42.40705","20":"38.61396","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.2136784","2":"ERR1414243","3":"OR777224.1","4":"880565","5":"880565","6":"163","7":"143","8":"85","9":"75","10":"CCTCGCCACCTATCACATCTTTTTCAAGAACCAAAAGATCCCCGTCTCCTGCCGCGCCAACCGCACCCGCGCCGACACGCTCCTC","11":"CAAGAACCAAAAGATCCCCGTCTCCTGCCGCGCCAACCGCACCCGCGCCGACACGCTCCTCGCTCTGGGGCCCGG","12":"1","13":"Human mastadenovirus D","14":"130310","15":"130310","16":"1","17":"0","18":"0:6 130310:2 0:5 10509:9 0:1 10509:5 0:12 10509:5 0:20","19":"36.68980","20":"33.12111","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"36.68980","31":"TRUE","32":"TRUE","33":"Human adenovirus 56","34":"no rank","35":"130310"},{"1":"ERR1414243.23865782","2":"ERR1414243","3":"HE963025.1","4":"1209820","5":"1209820","6":"121","7":"103","8":"93","9":"84","10":"AGATGTCAGTTTAGACATAGTGAATGAAACTTGCAAATATCCAGATTTTTTAACTATGGCCAATGATGTATATGGCGATGCATGTTTCTTTTT","11":"AGATGTCAGTTTAGACATAGTGAATGAAACTTGCAAATATCCAGATTTTTTAACTATGGCCAATGATGTATATGGCGATGCATG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:59","19":"26.69550","20":"23.24628","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"26.69550","31":"TRUE","32":"TRUE","33":"Human papillomavirus 159","34":"serotype","35":"333924"},{"1":"ERR1414243.28798577","2":"ERR1414243","3":"MG925782.1","4":"10524","5":"10524","6":"194","7":"202","8":"97","9":"101","10":"GGAATTGTAACTGAACCTGACCCCGAAGAGGGTACTAGCAGTGGGCAAAGGGGGGAGAAAAGGAAGTTAGAAAATGACGGGGCGGATTTTCTAAAGG","11":"CCTCAAACGGAATTGTAACTGAACCTGACCCCGAAGAGGGTACTAGCAGTGGGCAAAGGGGGGAGAAAAGGAAGTTAGAAAATGACGGGGCGGATTTTCTA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:71","19":"42.40705","20":"43.76917","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"43.76917","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.31361367","2":"ERR1414243","3":"ON075053.1","4":"1476554","5":"1476554","6":"55","7":"45","8":"34","9":"26","10":"CCCCTAGTCATCGTCGCCTTGGTGAGCCGTTACC","11":"CATCGTCGCCTTGGTGAGCCGTTACC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:27","19":"15.59682","20":"13.81175","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.59682","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414243.33452104","2":"ERR1414243","3":"MW567963.1","4":"10524","5":"10524","6":"194","7":"162","8":"97","9":"81","10":"CACGTAGGCAGCCAGGAATCGACCCATGAGCTCTGTAAAGGTTTTCTGGGTGGTGAATGTCAAAGACAATCCGCGGCGTTCCTCATTCAACCAGGTC","11":"CGTCGTCATTCTTGATTTGGACCACGCTGCGACCCCAGCGGTTCTGAACAACTTTGGTTTTGCTAGGTTGCTCCTTGAGGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:48","19":"42.40705","20":"36.86469","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.37807457","2":"ERR1414243","3":"ON815894.1","4":"130309","5":"130309","6":"186","7":"186","8":"93","9":"93","10":"ACTTTCTTACACTAGCCATTGAACGCCCGCTAGCTCTATCCAGTAACAGAGCAGTGACCCTTAAGTATTCACCACCTTTAAAAATAGAAAACG","11":"CTTACACTAGCCATTGAACGCCCGCTAGCTCTATCCAGTAACAGAGCAGTGACCCTTAAGTATTCACCACCTTTAAAAATAGAAAACGAAAAC","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"1","18":"0:8 130309:2 0:10 130309:4 0:38 130309:1 0:1","19":"41.03605","20":"41.03605","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"41.03605","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414243.6115827","2":"ERR1414243","3":"KF303071.1","4":"10524","5":"10524","6":"192","7":"183","8":"96","9":"94","10":"CCTGGCCTGTTTTTGAGCATAACATGCTTATGCGTTGTACCATGCATTTGGGGGCTCGTCGTGGCATGTTTTCTCCATATCAGAGTAATTTTTGCC","11":"CCGAAAATGTGTGCTTTTTAGTGCTGAAGGGAACTGCCAGTGTTAAGCATAATATGATTTGTGGCACTGGTCACTCTCAGCTGCTAACTTGCGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:62 A:34 0:61","19":"42.06515","20":"40.27914","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"42.06515","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414243.7522396","2":"ERR1414243","3":"ON075053.1","4":"1476554","5":"1476554","6":"56","7":"57","8":"39","9":"39","10":"TTCCTCTCAGAACCCCTAGACATCGTTGCCTTGGTAAGC","11":"TTCCTCTCAGAACCCCTAGACATCGTTGCCTTGGTAAGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:5","19":"15.28567","20":"15.55863","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.55863","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414243.7593606","2":"ERR1414243","3":"X74471.1","4":"10609","5":"10609","6":"59","7":"-Inf","8":"44","9":"-Inf","10":"CTCGTCCCGGTCCCGGTCCCGGTCCCGGTCCCGGTCCCGGTCCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:10 A:34 0:59","19":"15.59118","20":"0.00000","21":"ERR1414243","22":"Käppala","23":"Inlet","24":"147","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.59118","31":"FALSE","32":"FALSE","33":"human papillomavirus 25","34":"serotype","35":"337051"},{"1":"ERR1414249.13014143","2":"ERR1414249","3":"ON075053.1","4":"1476554","5":"1476554","6":"57","7":"57","8":"41","9":"41","10":"GGGACCTTCCTCTCAGAACCCCTAGACATCGTTGCCTTGGT","11":"GGGACCTTCCTCTCAGAACCCCTAGACATCGTTGCCTTGGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:7","19":"15.34910","20":"15.34910","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.34910","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414249.14312654","2":"ERR1414249","3":"ON075053.1","4":"1476554","5":"1476554","6":"69","7":"69","8":"41","9":"41","10":"ATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGTGAGCCG","11":"ATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGTGAGCCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"976:2 0:5","19":"18.58049","20":"18.58049","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"18.58049","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414249.17392585","2":"ERR1414249","3":"OP581403.1","4":"28284","5":"28284","6":"176","7":"176","8":"88","9":"88","10":"CCCACCGCTGCGTTGGTGAAAAATTACAGCCAAGTCAAAAAAGATGCGATTTTCCAAATTACCAATCACGGCTTCCACTAAGGCTGGC","11":"CCCACCGCTGCGTTGGTGAAAAATTACAGCCAAGTCAAAAAAGATGCGATTTTCCAAATTACCAATCACGGCTTCCACTAAGGCTGGC","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:24 130309:1 0:4 130309:10 0:4 130309:7 0:4","19":"39.30908","20":"39.30908","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"39.30908","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414249.1833196","2":"ERR1414249","3":"OP581393.1","4":"28284","5":"28284","6":"122","7":"106","8":"61","9":"53","10":"AGCATGGCTCAGCCCCGCCGGGGGAATATCTACTGGGTGCGAGATGCGTCGGGCGTGCGGG","11":"AGCATGGCTCAGCCCCGCCGGGGGAATATCTACTGGGTGCGAGATGCGTCGGG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:2 130309:1 0:4 130309:5 0:15","19":"29.67739","20":"26.69829","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"29.67739","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414249.20092041","2":"ERR1414249","3":"KX544841.1","4":"10359","5":"10359","6":"67","7":"67","8":"69","9":"79","10":"TGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGC","11":"CTTCTGCTGGTGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGCTGGTGCTGGCTCTTCTGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:35 A:34 0:46","19":"15.82388","20":"15.33375","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.82388","31":"FALSE","32":"FALSE","33":"Human betaherpesvirus 5","34":"species","35":"10358"},{"1":"ERR1414249.21111489","2":"ERR1414249","3":"AY243312.1","4":"10254","5":"10254","6":"79","7":"79","8":"83","9":"83","10":"GAGATCTTTTTTTGAGCCAGAAAATTGCACAGAATGATCCAGAAAAGAAGATTCCTTCAACAGCAGCAAAGGCAACAAGTCTT","11":"GAGATCTTTTTTTGAGCCAGAAAATTGCACAGAATGATCCAGAAAAGAAGATTCCTTCAACAGCAGCAAAGGCAACAAGTCTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:49","19":"17.87799","20":"17.87799","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"17.87799","31":"FALSE","32":"FALSE","33":"Vaccinia virus WR","34":"no rank","35":"10245"},{"1":"ERR1414249.25445618","2":"ERR1414249","3":"AY386371.1","4":"38804","5":"38804","6":"-Inf","7":"74","8":"-Inf","9":"98","10":"NA","11":"ATTACTCCTGCTTTTTGATATTCACCTACTCTCTTTTCAAAGAAGTTGGTTTTTCCTTCAAGCGATATTAATTCCATAAAATCGAAAGGATTGGTAAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:136","19":"0.00000","20":"16.13970","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"16.13970","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414249.26082567","2":"ERR1414249","3":"ON075053.1","4":"1476554","5":"1476554","6":"58","7":"58","8":"32","9":"32","10":"TCAGAACCCCTAGACATCGTCGCCTTGGTAGG","11":"TCAGAACCCCTAGACATCGTCGCCTTGGTAGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:31","19":"16.73526","20":"16.73526","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"16.73526","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414249.30866721","2":"ERR1414249","3":"AB618031.1","4":"946522","5":"946522","6":"99","7":"99","8":"61","9":"61","10":"GCCAGAGTTGTCCGGCGCTCTTCGGCTGCGGTAGTCCAGGCAGTTCAATCAACTGTTTACC","11":"GCCAGAGTTGTCCGGCGCTCTTCGGCTGCGGTAGTCCAGGCAGTTCAATCAACTGTTTACC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:27","19":"24.08247","20":"24.08247","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"24.08247","31":"TRUE","32":"TRUE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414249.35613626","2":"ERR1414249","3":"OP581380.1","4":"28284","5":"28284","6":"152","7":"192","8":"76","9":"96","10":"GCATCGTTGACCGCGGCTTGTCGCAGAATCTCCTGCACGTCGCCAGAGTTGTCTTGGTAGGCAATCTCCGACATAA","11":"GTTCAACAGAATCAATATCGGCATCGTTGACCGCGGCTTGTCGCAGAATCTCCTGCACGTCGCCAGAGTTGTCTTGGTAGGCAATCTCCGACATAA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:29 130309:5 0:16 130309:5 0:5 130309:2","19":"35.09798","20":"42.06515","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.06515","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414249.39715615","2":"ERR1414249","3":"OP581405.1","4":"28284","5":"28284","6":"162","7":"202","8":"81","9":"101","10":"GGTGCGGTTTCAAACATGTGACCGTGTAGCCTTTAACTGCGCAATGCAGAGCTTGGGCCCTGGGCTTATTGGCATGAGTGG","11":"GCCTATGTGCTGGGGAATGGAGCTGTGGTGCGGTTTCAAACATGTGACCGTGTAGCCTTTAACTGCGCAATGCAGAGCTTGGGCCCTGGGCTTATTGGCAT","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"130309:3 0:14 130309:6 0:12 130309:2 0:9 130309:1 0:5 130309:2 0:7 130309:3 0:9","19":"36.86469","20":"43.76917","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"43.76917","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414249.40625778","2":"ERR1414249","3":"ON075053.1","4":"1476554","5":"1476554","6":"64","7":"64","8":"42","9":"42","10":"GGGGATCATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGT","11":"GGGGATCATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:8","19":"17.12297","20":"17.12297","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"17.12297","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414249.41687551","2":"ERR1414249","3":"OP581407.1","4":"28284","5":"28284","6":"194","7":"80","8":"97","9":"40","10":"ACTCAGCGAACCCAACTATTTGGGAGGGGAGGACGACGTGCTCCTGAAGCACATAGCGCGACAGAGCACCATTGTACGAGAAGCCCTCAAGGAATGC","11":"TGCCTGAAGCACTGGCTACTTATCACATCTTTTTCAAGAA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"130309:1 0:5 130309:3 0:3 130309:8 0:9 130309:1 0:16 130309:3 0:14 A:34 0:1 130309:5 0:1","19":"42.40705","20":"21.68680","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414249.42245620","2":"ERR1414249","3":"KP264964.1","4":"1595998","5":"1595998","6":"57","7":"59","8":"38","9":"38","10":"CTGATGTCTGATGTCTGATGTCTGATGTCTGAGGTTTT","11":"CTGATGTCTGATGTCTGATGTCTGATGTCTGAGGTTTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:4 A:34 0:5","19":"15.66973","20":"16.21955","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"16.21955","31":"FALSE","32":"FALSE","33":"Human smacovirus 1","34":"species","35":"2498817"},{"1":"ERR1414249.42382968","2":"ERR1414249","3":"HM773823.1","4":"28875","5":"28875","6":"57","7":"57","8":"44","9":"44","10":"TAGATTTTCAAAATTTTCTATTGAGAAAAATTGATCCATAAATG","11":"TAGATTTTCAAAATTTTCTATTGAGAAAAATTGATCCATAAATG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:10","19":"15.06267","20":"15.06267","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.06267","31":"FALSE","32":"FALSE","33":"Rotavirus A","34":"species","35":"10912"},{"1":"ERR1414249.8420395","2":"ERR1414249","3":"KF908243.1","4":"340907","5":"340907","6":"-Inf","7":"62","8":"-Inf","9":"61","10":"NA","11":"GCGGCAGCCTGCAGGGCTGGGCCAGCCAGGGCGTGCTGCTGCTCAACACCAGCCTGACGGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:42 80864:1 0:1","19":"0.00000","20":"15.08195","21":"ERR1414249","22":"Uppsala","23":"Inlet-Adb","24":"153","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.08195","31":"FALSE","32":"FALSE","33":"Papiine alphaherpesvirus 2","34":"species","35":"10294"},{"1":"ERR1414250.11153663","2":"ERR1414250","3":"JF909601.1","4":"944645","5":"944645","6":"71","7":"48","8":"92","9":"27","10":"AAAGACTTATCTTCGCCGGAAAGCAACTCGAAGACGGAAGAAGCTTGTCTGATTACAACATCCAAAAATAATCAACTCTTCACTTAGTCTTG","11":"CAGCAAAGACTTATCTTCGCCGGAAAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:62","19":"15.70175","20":"14.56383","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.70175","31":"FALSE","32":"FALSE","33":"Senegalvirus marseillevirus","34":"species","35":"1513458"},{"1":"ERR1414250.15157512","2":"ERR1414250","3":"OP581409.1","4":"28284","5":"28284","6":"184","7":"196","8":"92","9":"98","10":"GTAACATCTGCATCGGTGCCGTAACCTTCGCTGCTTTACTTAAAAAGCCTGTGATTTCATTTACCACCCCAGCACTTGGATTACATGAAGAT","11":"TGCATCGGTGCCGTAACCTTCGCTGCTTTACTTAAAAAGCCTGTGATTTCATTTACCACCCCAGCACTTGGATTACATGAAGATCTGTGTTCTTTTTT","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:9 130309:1 0:31 130309:4 0:3 130309:7 0:12 130309:5","19":"40.69186","20":"42.74839","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414250.17431204","2":"ERR1414250","3":"ON075053.1","4":"1476554","5":"1476554","6":"64","7":"64","8":"42","9":"42","10":"ATCATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGTAGGC","11":"ATCATCCTCTCAGAACCCCTAGACATCGTCGCCTTGGTAGGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:5 2:3","19":"17.12297","20":"17.12297","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"17.12297","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414250.28268098","2":"ERR1414250","3":"ON075053.1","4":"1476554","5":"1476554","6":"50","7":"50","8":"27","9":"27","10":"AACCCCTAGACATCGTCGCCTTGGTAG","11":"AACCCCTAGACATCGTCGCCTTGGTAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:21","19":"15.17065","20":"15.17065","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.17065","31":"FALSE","32":"FALSE","33":"Bovine kobuvirus","34":"no rank","35":"194965"},{"1":"ERR1414250.29382526","2":"ERR1414250","3":"OP581398.1","4":"28284","5":"28284","6":"194","7":"192","8":"97","9":"96","10":"CTGCCAGAGGGTGTACGTTTGTTGCATGTTGGAAAGGGGTGGTGGGGCGAAACAAGAGTCAAATGTCTGTAAAGAAGTGTGTGTTTGAACGTTGCAT","11":"GGCGCGGATTCGCCATAATGCGGGCTCTGATAATGTGTGTTTTTTACTGCTAAAGGGAACTGCCAGTGTAAAGCATAACATGATTTGTGGCGGTGG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:10 130309:3 0:5 130309:1 0:39 130309:1 0:4 A:34 0:4 130309:3 0:13 130309:5 0:32 130309:5 0:1","19":"42.40705","20":"42.06515","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414250.36164693","2":"ERR1414250","3":"OP562382.1","4":"10258","5":"10258","6":"-Inf","7":"58","8":"-Inf","9":"39","10":"NA","11":"GTGGCGGTCTTGCCGGAGCCGGAGCCGCCGATCACGACG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:13","19":"0.00000","20":"15.83159","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"15.83159","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414250.3774279","2":"ERR1414250","3":"J02226.1","4":"10632","5":"10632","6":"179","7":"189","8":"93","9":"98","10":"GTGCAATCTCAAGTTATGAACACAGAGCACAAGGCGTACCTAGATAAGAACAAAGCATATCCTGTTGAATGTTGGGTTCCTGATCCCACCAGA","11":"GCAATCTCAAGTTATGAACACAGAGCACAAGGCGTACCTAGATAAGAACAAAGCATATCCTGTTGAATGTTGGGTTCCTGATCCCACCAGAAATGAAA","12":"1","13":"JC polyomavirus","14":"10632","15":"10632","16":"1","17":"1","18":"0:19 10632:1 0:7 10632:4 0:15 10632:2 0:2 10632:5 0:11","19":"39.49169","20":"41.22167","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"41.22167","31":"TRUE","32":"TRUE","33":"JC polyomavirus","34":"no rank","35":"1891763"},{"1":"ERR1414250.5072821","2":"ERR1414250","3":"OP555454.1","4":"10524","5":"10524","6":"188","7":"194","8":"94","9":"97","10":"CTCCACTAGGGTGTCATCCCCCGAGAGATGCATAACTAGCATAAACGGGACCAGTTGCTTACCAAAGGCTCCCATCCACGTGTAGGTCTCAACG","11":"GATGCATAACTAGCATAAACGGGACCAGTTGCTTACCAAAGGCTCCCATCCACGTGTAGGTCTCAACGTCGTAGGTGACGAAGAGGCGTTCAACGCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:89","19":"41.37966","20":"42.40705","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"No hit or\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414250.7747757","2":"ERR1414250","3":"OP581401.1","4":"28284","5":"28284","6":"164","7":"164","8":"82","9":"82","10":"TATTCAAATGTTCACACTCTAAAAGTCCAGGAGTCCTTTCCATTGTGGCAATAGGCACAGAATCCTCAGAGGGAGGTGGAAG","11":"TATTCAAATGTTCACACTCTAAAAGTCCAGGAGTCCTTTCCATTGTGGCAATAGGCACAGAATCCTCAGAGGGAGGTGGAAG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"130309:3 0:1 130309:5 0:13 130309:5 0:21","19":"37.21589","20":"37.21589","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"37.21589","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414250.8359767","2":"ERR1414250","3":"OP581388.1","4":"28284","5":"28284","6":"192","7":"180","8":"96","9":"90","10":"GTGAAGGTATTAACAAATCGATTAAGTTGGGAGGGATGCATGCGAGGGGAAATCAGATGCATTTTGGCCTGAATTTTTAGGTTGGCAATGTTGCCA","11":"AATCGATTAAGTTGGGAGGGATGCATGCGAGGGGAAATCAGATGCATTTTGGCCTGAATTTTTAGGTTGGCAATGTTGCCACCGAGATCA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:10 130309:1 0:9 130309:6 0:7 130309:4 0:12 130309:2 10509:7 0:13","19":"42.06515","20":"40.00169","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"42.06515","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414250.8530132","2":"ERR1414250","3":"OP581370.1","4":"10524","5":"10524","6":"164","7":"164","8":"82","9":"82","10":"CTACAGCGACGGAACATGGAGAAAAAACTATCCCGTGTTTGACAACGAAGGGATACTAGCAAACAGTGCCACGTGGGGTTAT","11":"CTACAGCGACGGAACATGGAGAAAAAACTATCCCGTGTTTGACAACGAAGGGATACTAGCAAACAGTGCCACGTGGGGTTAT","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:14 130309:4 0:10 130309:1 0:19","19":"37.21589","20":"37.21589","21":"ERR1414250","22":"Uppsala","23":"Inlet-Adb","24":"154","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Influent","28":"FALSE","29":"Kraken2 HV\\nassignment","30":"37.21589","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414270.11948271","2":"ERR1414270","3":"AM501482.1","4":"126794","5":"126794","6":"70","7":"-Inf","8":"98","9":"-Inf","10":"TGCATTAATTTCCTTTTGATTATTTTCTATTGCAATACCTTTAGCTAATTTTAAAAAATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTAG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:57","19":"15.26728","20":"0.00000","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.26728","31":"FALSE","32":"FALSE","33":"Vaccinia virus Ankara","34":"no rank","35":"10245"},{"1":"ERR1414270.120434","2":"ERR1414270","3":"MK955317.1","4":"130309","5":"130309","6":"48","7":"108","8":"24","9":"54","10":"CAATGAGTTTATGTCTACCAACAA","11":"TGAACCGGAAAACCAACGCTCCTCGTTACGAGTGGTTCGAGTTCACTTTGCCAG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:24 0:21","19":"15.10358","20":"27.07456","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"27.07456","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414270.15019889","2":"ERR1414270","3":"OP581351.1","4":"10524","5":"10524","6":"140","7":"140","8":"70","9":"70","10":"AAGGAAGTTAGAAAATGACGGGGCGGATTTTCTAAAGGAGTTAACCTTGAGTTTAATGTCTCGTTGTTAT","11":"AAGGAAGTTAGAAAATGACGGGGCGGATTTTCTAAAGGAGTTAACCTTGAGTTTAATGTCTCGTTGTTAT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:36","19":"32.95284","20":"32.95284","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"32.95284","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414270.16878447","2":"ERR1414270","3":"U37537.1","4":"186160","5":"186160","6":"176","7":"176","8":"88","9":"88","10":"TGTGCCTGTGATCCACCTGCAAGGTGATGCAAACTGTTTAAAGTGCTTCAGATACAGGGTGCAAAAACATAAAGACACACTGTTTGTG","11":"TGTGCCTGTGATCCACCTGCAAGGTGATGCAAACTGTTTAAAGTGCTTCAGATACAGGGTGCAAAAACATAAAGACACACTGTTTGTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:16 337043:5 0:33","19":"39.30908","20":"39.30908","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"39.30908","31":"TRUE","32":"TRUE","33":"unknown virus","34":"no rank","35":"333753"},{"1":"ERR1414270.22816464","2":"ERR1414270","3":"KF268335.1","4":"130310","5":"130310","6":"126","7":"126","8":"63","9":"63","10":"TGATCTCTTTGGGCATGTCTCTGTTCACAATCTGACGGTACCATGGGAAGCGCTGGTTGAACA","11":"TGATCTCTTTGGGCATGTCTCTGTTCACAATCTGACGGTACCATGGGAAGCGCTGGTTGAACA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:29","19":"30.41176","20":"30.41176","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"30.41176","31":"TRUE","32":"TRUE","33":"Human mastadenovirus D","34":"species","35":"10509"},{"1":"ERR1414270.27325383","2":"ERR1414270","3":"LN827593.1","4":"10376","5":"10376","6":"188","7":"194","8":"94","9":"97","10":"TGCAAAAACAGACAAAGAGACCGTGAGCGGTTACCGGGGCGCAGGGCCTCTGCCGGGAAGCCCACCCGGGCCAGGGCCCGGTAAAGCAGGTACC","11":"GAGCGGTTACCGGGGCGCAGGGCCTCTGCCGGGAAGCCCACCCGGGCCAGGGCCCGGTAAAGCAGGTACCAGTATTCATCCGGCACCTTGCGTGCCA","12":"1","13":"Human gammaherpesvirus 4","14":"10376","15":"10376","16":"1","17":"1","18":"0:12 10376:5 0:1 10376:5 0:8 10376:5 0:14 10376:7 0:27 10376:3","19":"41.37966","20":"42.40705","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.28004169","2":"ERR1414270","3":"AY961628.3","4":"10376","5":"10376","6":"173","7":"164","8":"97","9":"82","10":"GGGATGAAGAGGGGAGGGCTGGAGGCCGGGGCCGCGGAGTCCGGAGCCGCAGAGGCCGGAGCCGCAGAGGCCGGAGCCGCAGAGGCCGGAGCCGCAG","11":"AATCCCTTTTTCCTCTTGGGAGTCTGAGGCGCTGCGCTGCCCGACGCCCTTGGTGGCGTGGAGGGAGCGGGGACCCCGGGGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:49","19":"37.81660","20":"37.21589","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"37.81660","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.28731166","2":"ERR1414270","3":"AB618031.1","4":"946522","5":"946522","6":"-Inf","7":"69","8":"-Inf","9":"97","10":"NA","11":"TCTGGAAGATCAGGATATGTGGCGCATGAGCGGTATTTTTCGCGATGTGACCCTGCTGCATAAACCGGAAACTCGTTTAGTGAATGTACAAGCGCGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:56 91347:1 0:16","19":"0.00000","20":"15.08292","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.08292","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414270.29211422","2":"ERR1414270","3":"OP555455.1","4":"10524","5":"10524","6":"196","7":"166","8":"98","9":"83","10":"TTATGCCGACAGAGGGGCAGAAATTGAGTCTGGGAATATTTTTGCCATGGAAATCAATTTGGCGGCCAATCTCTGGCGCAGCTTCTTATATTCCAATG","11":"CTGCTTTCCGCTGGGAGGGTCTGCAGCTACAGACACGTATTCTGGCATAAAAGCAAACAGCGGAAACTGGACTGCGGACAACG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:50","19":"42.74839","20":"37.56641","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414270.33522773","2":"ERR1414270","3":"MK540317.1","4":"10376","5":"10376","6":"188","7":"188","8":"94","9":"94","10":"GCCCCCTCCCCTGCGCCTTCGTGCACACCAAGGACAAGACGTCCAGCCTGCAGTGGCCCATGTACCTCCTGACTAATGAGAAGTCCAAGGCCTT","11":"GCCCCCTCCCCTGCGCCTTCGTGCACACCAAGGACAAGACGTCCAGCCTGCAGTGGCCCATGTACCTCCTGACTAATGAGAAGTCCAAGGCCTT","12":"1","13":"Human gammaherpesvirus 4","14":"10376","15":"10376","16":"1","17":"1","18":"0:22 10376:2 0:17 10376:5 0:3 10376:7 0:4","19":"41.37966","20":"41.37966","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"41.37966","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.34501615","2":"ERR1414270","3":"ON815884.1","4":"130309","5":"130309","6":"-Inf","7":"56","8":"-Inf","9":"28","10":"NA","11":"ATGCTTTTACTTTTAAATGGTCAGCCGA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:55 A:29","19":"0.00000","20":"16.80569","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.80569","31":"FALSE","32":"FALSE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414270.35485278","2":"ERR1414270","3":"AP019115.1","4":"10376","5":"10376","6":"188","7":"190","8":"94","9":"95","10":"TCCTGTTTAGTTGAAGAAAACAAGCTATTGGATTGTAACACACATTTTACGCTTTGTTCCTTAGATTTGTTCTGCATGTTATTACTGATTGTCG","11":"TGCTAGCCTCCGCGCTAATCGCTGGTGGCAGTATTTTGCAAACAAACTTCAAGAGTTTAAGCAGCACTGAATTTATACCCAGTGAGTATCTATTT","12":"1","13":"Human gammaherpesvirus 4","14":"10376","15":"10376","16":"1","17":"1","18":"0:15 10376:5 0:16 10376:2 0:6 10376:3 0:8 10376:3 0:2 A:34 0:12 10376:1 0:5 10376:1 0:5 10376:3 0:3 10376:5 0:27","19":"41.37966","20":"41.72269","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"41.72269","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.38153519","2":"ERR1414270","3":"AB850648.1","4":"10376","5":"10376","6":"108","7":"-Inf","8":"93","9":"-Inf","10":"GTAGCGTCAATGCAATTTTGAAAACCTTCCAAATCGACATCGGGTTGTGAAAATATTTCTGGTGCATTTGCGAGCAATCGGACATTTGACATT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:10 10376:5 0:44 A:34 0:51","19":"23.82739","20":"0.00000","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"23.82739","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.42702793","2":"ERR1414270","3":"ON442321.1","4":"10524","5":"10524","6":"188","7":"176","8":"94","9":"88","10":"GAAATTGAGTCTGGGAATATTTTTGCCATGGAAATCAATTTGGCGGCCAATCTCTGGCGCAGCTTCTTATATTCCAATGTAGCCCTGTATTTGC","11":"CGCAGCTTCTTATATTCCAATGTAGCCCTGTATTTGCCTGACTCATACAAAATTACGCCAGACAACATTACACTACCCGAAAACAAGA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:111","19":"41.37966","20":"39.30908","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.37966","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414270.4272769","2":"ERR1414270","3":"LN827571.1","4":"10376","5":"10376","6":"172","7":"180","8":"86","9":"90","10":"GGGGAGAACGGTGGCCAGACGGCGCTTGCGTCTGCCCCCGGAGCCCTGCCCTCCTCCACCCAGCAGCAGCCCGGCCGAGGCCTGCG","11":"GTCCAGACTCCGGGGGTCCAGCCGGGCGCACACGCTCAGCTCAACCGCCAGGGCGTACACCTGGCTGTACGCCGCCGCCAGCAGCCCCGA","12":"1","13":"Human herpesvirus 4 type 2","14":"12509","15":"12509","16":"1","17":"1","18":"0:43 10376:2 0:1 10376:4 0:2 A:34 0:2 10376:3 0:9 10376:9 0:33 12509:1","19":"38.61396","20":"40.00169","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"40.00169","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.42967948","2":"ERR1414270","3":"MG021312.1","4":"10376","5":"10376","6":"-Inf","7":"156","8":"-Inf","9":"88","10":"NA","11":"CCTCCGGGCCCGGACGGCCGCCCTCGGCCCCCCAGACAGACCCCAGGGTCTCCAGGCAGGGTCCGGCATCTCCAGGGGTTTAAGGCGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:29 0:41 12509:1 0:13","19":"0.00000","20":"34.84214","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"34.84214","31":"TRUE","32":"TRUE","33":"Human gammaherpesvirus 4","34":"species","35":"10375"},{"1":"ERR1414270.6145281","2":"ERR1414270","3":"OP174922.1","4":"10524","5":"10524","6":"190","7":"190","8":"95","9":"95","10":"GAGGCCTTCAAGGATTACTCTGGGTTTCCGCTGACGGTGGAGGAGTTGAGTCGGACCTACGAAGCTAATCTCTTTTCACCGCGTGTGCCCCCAAA","11":"GAGGCCTTCAAGGATTACTCTGGGTTTCCGCTGACGGTGGAGGAGTTGAGTCGGACCTACGAAGCTAATCTCTTTTCACCGCGTGTGCCCCCAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:61","19":"41.72269","20":"41.72269","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.72269","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414270.9368372","2":"ERR1414270","3":"MN369532.1","4":"10245","5":"10245","6":"71","7":"-Inf","8":"97","9":"-Inf","10":"TCCTGGACAACATGGTGAAACCCCATCTCTACTAAAAATACAAAAATTAGCTGGGTAGACCTTTATGATAGATAGTTACACTTTGATTTGAGTATGC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:64","19":"15.52011","20":"0.00000","21":"ERR1414270","22":"Käppala","23":"Primary","24":"225","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.52011","31":"FALSE","32":"FALSE","33":"Vaccinia virus","34":"species","35":"10242"},{"1":"ERR1414271.10274778","2":"ERR1414271","3":"OP047711.1","4":"10524","5":"10524","6":"146","7":"146","8":"73","9":"73","10":"GATGGGGAAGGGTACAACGTGGCTCAATGTAACATGACCAAAGACTGGTTTCTGATTCAGATGCTTAGTCACT","11":"GATGGGGAAGGGTACAACGTGGCTCAATGTAACATGACCAAAGACTGGTTTCTGATTCAGATGCTTAGTCACT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:39","19":"34.02899","20":"34.02899","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"34.02899","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.1379483","2":"ERR1414271","3":"OQ518335.1","4":"10515","5":"10515","6":"194","7":"194","8":"97","9":"97","10":"CTAGCCTAAAGCCCGTGACACTGCAGCAGGTGCTGCCCGCGCTTGCACCGTCCGAAGAAAAGCGCGGCCTAAAGCGCGAGTCTGGTGACTTGGCACC","11":"GTGCTGCCCGCGCTTGCACCGTCCGAAGAAAAGCGCGGCCTAAAGCGCGAGTCTGGTGACTTGGCACCCACCGTGCAGCTGATGGTACCCAAGCGCC","12":"1","13":"Human mastadenovirus C","14":"129951","15":"129951","16":"1","17":"0","18":"0:1 129951:1 0:69 129951:2 0:9 129951:5 0:5","19":"42.40705","20":"42.40705","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 2","34":"serotype","35":"129951"},{"1":"ERR1414271.15674318","2":"ERR1414271","3":"OR735187.1","4":"10515","5":"10515","6":"82","7":"186","8":"41","9":"93","10":"TCTTGATAGGCGATCTCGGCCATGAACTGCTCGATCTCTTC","11":"GACGACGGGCCCGGTGAGCTTGAACCTGAAAGAGAGTTCGACAGAATCAATTTCGGTGTCGTTGACGGCGGCCTGGCGCAAAATCTCCTGCAC","12":"1","13":"Human mastadenovirus C","14":"129951","15":"129951","16":"1","17":"0","18":"10509:2 0:5 A:34 0:12 129951:1 0:16 129951:1 0:6 129951:1 0:14 129951:5 0:4","19":"22.08117","20":"41.03605","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"41.03605","31":"TRUE","32":"TRUE","33":"Human adenovirus 2","34":"serotype","35":"129951"},{"1":"ERR1414271.168803","2":"ERR1414271","3":"MW567963.1","4":"10524","5":"10524","6":"174","7":"170","8":"87","9":"88","10":"TACCAATGCTAAGCATGAGCTTAGTTTAAAACTTGGCAGCAACATAACTTTAGAAAATGGGTTACTTTCGGCCACAGTTCCCACTGT","11":"ATTGGCCACTGCCGCACCACTGACAGTAAGCAACAACCAGCTTAGTATTAACACGGGCAGAGGTTTAGTTATAACTAATAATGCCGTA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:53 A:34 0:12 130309:3 0:2 130309:6 0:32","19":"38.96184","20":"37.96900","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"38.96184","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.21212185","2":"ERR1414271","3":"KY053526.1","4":"10258","5":"10258","6":"-Inf","7":"52","8":"-Inf","9":"30","10":"NA","11":"GTCGTGGTCGTGGTCGTGGTCGTGGTCGTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:31","19":"0.00000","20":"15.28873","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.28873","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414271.24798637","2":"ERR1414271","3":"KP745658.1","4":"10359","5":"10359","6":"70","7":"-Inf","8":"98","9":"-Inf","10":"CACCAAAAAAACTACGACCACAACAAAGAAAACAACCACTACCACCAAAAAATAAAGATTGGATTCACATCGTGTTTTTAAACAAAAAGCAGAACTTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:133","19":"15.26728","20":"0.00000","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.26728","31":"FALSE","32":"FALSE","33":"Human betaherpesvirus 5","34":"species","35":"10358"},{"1":"ERR1414271.24876064","2":"ERR1414271","3":"OP581374.1","4":"10524","5":"10524","6":"-Inf","7":"98","8":"-Inf","9":"84","10":"NA","11":"TTCCTGCATCACAAAATGCTCTGGGAAATTGTGTGGGCCTGGAACCCCAAGAGCCATACCTGTTACCTCTTTGACCCGTTTGGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:56 A:34 0:51","19":"0.00000","20":"22.11782","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"22.11782","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.25668766","2":"ERR1414271","3":"MW567962.1","4":"10524","5":"10524","6":"187","7":"102","8":"97","9":"51","10":"TCAACAGAATCAATCTCCGCATCGTTGACCGCGACCTGACGCAGAATTTCCTGCACGTAGCCCGAGTTGTCTTGGTATGCGATCTCGGACATAAACT","11":"CAACGCGGCGGTTGAGTTCCTGGATACGTCTTTGTTGAGTGAAGACCAAGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:18","19":"40.87690","20":"25.94215","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"40.87690","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.26784737","2":"ERR1414271","3":"MZ983604.1","4":"10529","5":"10529","6":"180","7":"170","8":"90","9":"85","10":"CGGCACCAGCCAGCTGAACGGGGGCGCCTTCAATTGGAACAGTATCTGGAGCGGTCTTAAAAATTTTGGTTCCACAATTAAGACATATGG","11":"TTAAGACATATGGCTCCAAGGCGTGGAACAGCCAAACCGGCCAGATGCTAAGGGACAAGTTAAAAGACCAGAACTTCCAACAAAA","12":"1","13":"Human mastadenovirus A","14":"129875","15":"129875","16":"1","17":"0","18":"0:9 129875:2 10509:5 0:58 129875:3 0:51","19":"40.00169","20":"38.26544","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"40.00169","31":"TRUE","32":"TRUE","33":"Human adenovirus 31","34":"serotype","35":"129875"},{"1":"ERR1414271.27479065","2":"ERR1414271","3":"OP047707.1","4":"10524","5":"10524","6":"174","7":"186","8":"87","9":"93","10":"TGAAGGGGATCCGAATGAGGAAGCGGTAGATGGTATGTTTCCTAACTGGATGCTGTCCGAGGACCACAGTGCTGACAGTGGTGCGGC","11":"TTTCTCCTCCGTTGCCGCCGAGATGAGAATGCTGCCGGACTTCTTCACCGGGAACTGGGATGACATGTTCCAGGGGCTGCTGGAGGCTGAGCA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53 A:34 0:60","19":"38.96184","20":"41.03605","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.03605","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.35243601","2":"ERR1414271","3":"MZ983607.1","4":"10529","5":"10529","6":"182","7":"182","8":"91","9":"91","10":"AAAGGTATAGAGGAAATTGTGTACTTTTCAAACTTTACAAACCACTTACTTTTAAACTGTTCCTGCTCTAACTCCTTAATCTCATGGTTTG","11":"AAAGGTATAGAGGAAATTGTGTACTTTTCAAACTTTACAAACCACTTACTTTTAAACTGTTCCTGCTCTAACTCCTTAATCTCATGGTTTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:43 129875:1 0:13","19":"40.34708","20":"40.34708","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"40.34708","31":"TRUE","32":"TRUE","33":"Human adenovirus 31","34":"serotype","35":"129875"},{"1":"ERR1414271.3796086","2":"ERR1414271","3":"ON442315.1","4":"10524","5":"10524","6":"187","7":"188","8":"97","9":"94","10":"TTCACCGCCTTGGCGAGCAACCGCCGCGTCCGACGGGCACAGCATCGCACAGAAAGCTCATCCAAGTTTTTACAATAGGTACAGTCCAACACCACCA","11":"CGTTCCATTAAGCCCCGCAGTAATCCCTGTCTCCGGGGTTCCACCAAGCTTTTCAGAAGTGACCTAACGGTGAAGCTTCCCAAAACTTTCACCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:49 130309:5 0:7","19":"40.87690","20":"41.37966","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.37966","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414271.39111776","2":"ERR1414271","3":"OK422496.1","4":"32605","5":"32605","6":"58","7":"77","8":"81","9":"97","10":"TAAAGCTCAAAAATTATGGTATGCTATTTTAGATGCTCAAACTGAAACTGGTACTCCATTCATGCTTTATAAAGATGCTTG","11":"CTATTTTAGATGCTCAAACTGAAACTGGTACTCCATTCATGCTTTATAAAGATGCTTGTAACAAAAAAACTAATCAAAAGAACTTGGGTATCATCAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:86","19":"13.19847","20":"16.83166","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.83166","31":"FALSE","32":"FALSE","33":"Buffalopox virus","34":"no rank","35":"10245"},{"1":"ERR1414271.42382628","2":"ERR1414271","3":"OL468961.1","4":"10255","5":"10255","6":"62","7":"62","8":"53","9":"52","10":"CAATCCCTTCTTCCATCCGTAAAAATGCATACTTGTTAGTTTACCCATAGTTG","11":"CAATCCCTTCTTCCATCCGTAAAAATGCATACTTGTTAGTTTACCCATAGTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:18","19":"15.61598","20":"15.69126","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.69126","31":"FALSE","32":"FALSE","33":"Variola virus","34":"species","35":"10242"},{"1":"ERR1414271.5160421","2":"ERR1414271","3":"ON815886.1","4":"130309","5":"130309","6":"184","7":"194","8":"92","9":"97","10":"AGTTTTGGAGTGAGCTACAAAGTTATCAGAGCCTCCGACGCTTGCTGGAGTTGGCTTCTGCCAGAACTTCCAGCTGTTGGAGGTTTATTTTT","11":"GTCCGCGGGTATATAAGCAGGTGGAGTGGTTTCAGTGTTAGTTTATTCTGATGGAGTTTTGGAGTGAGCTACAAAGTTATCAGAGCCTCCGACGCTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:26 130309:1 0:85","19":"40.69186","20":"42.40705","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414271.5974125","2":"ERR1414271","3":"AB618031.1","4":"946522","5":"946522","6":"98","7":"-Inf","8":"92","9":"-Inf","10":"CACATTCACTAAACGGGTTTCCGGTTTATGCAGCAGTGACACATCACGGAAAATACCGCTCATGCGCCACATATCCTGATCTTCCAGATAAC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"1236:4 0:134","19":"21.67284","20":"0.00000","21":"ERR1414271","22":"Käppala","23":"Primary","24":"226","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"21.67284","31":"TRUE","32":"TRUE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414273.13309218","2":"ERR1414273","3":"ON532825.1","4":"10524","5":"10524","6":"196","7":"190","8":"98","9":"95","10":"GATCTCCACCAGCCCCAGTTGGTAAGCAGCCGCAAACTCGGCGGGGGTATTGTTCCTGGCGGCCCTACAAAAGCAATATCGTCGCCTGCCTCCAGCAC","11":"GTCCCCGCCGCGGCCAGGCTTGTTGAAAGCCAGCCGTAGATGGGACACCGTTTCCATCGCAGGATCTCCACCAGCCCCAGTTGGTAAGCAGCCGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:126","19":"42.74839","20":"41.72269","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414273.1449015","2":"ERR1414273","3":"X55965.1","4":"333753","5":"333753","6":"182","7":"194","8":"91","9":"97","10":"TTGCGCTACAACTGTGCAAAGAACAGGGACACCGTCTCCAAGAACATGAGCATGCTGTTAAACATTCCCGAGAAGCATATGCTCATAGAAC","11":"CTGTTAAACATTCCCGAGAAGCATATGCTCATAGAACCACCAAAACTAAGAAGTACACCTGCTGCCTTGTACTGGTACAAGACATCCATGGGTAATG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:117","19":"40.34708","20":"42.40705","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"unknown virus","34":"serotype","35":"337043"},{"1":"ERR1414273.16872365","2":"ERR1414273","3":"MF118167.1","4":"2017081","5":"2017081","6":"146","7":"160","8":"87","9":"94","10":"TATTCGGAACCTTTTGACATGGATCCAAAGCAATTAGCTTATCCATGTGAGCCTTGGACCAGTCCAAAGTGAAAAAAAGATATTTAC","11":"ACCTTTTGACATGGATCCAAAGCAATTAGCTTATCCATGTGAGCCTTGGACCAGTCCAAAGTGAAAAAAAGATATTTACGAATGCAGATTGCAT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:34 2017081:3 0:31","19":"32.69212","20":"35.21673","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"35.21673","31":"TRUE","32":"TRUE","33":"Human fecal virus Jorvi2","34":"species","35":"642248"},{"1":"ERR1414273.17515661","2":"ERR1414273","3":"ON532824.1","4":"10524","5":"10524","6":"122","7":"-Inf","8":"95","9":"-Inf","10":"TATGCACTTTCACTTAGCCAATAGTTGAAGTGCTAAAGACTGGTTTCTGATTCAGATGCTTAGTCACTACAACATTGGATATCAGGGTTTTTATG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:61 A:34 0:63","19":"26.79036","20":"0.00000","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"26.79036","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414273.17746671","2":"ERR1414273","3":"MF001305.1","4":"2025360","5":"2025360","6":"-Inf","7":"106","8":"-Inf","9":"93","10":"NA","11":"TGTATATTGCACTTCTCTCACCATGTTTTCAGCAAGATTTTCATTGACTATCCCATCACTGGCCGCAAAAAATGCCAACACATGTTTGATGAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:60","19":"0.00000","20":"23.38614","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"23.38614","31":"TRUE","32":"TRUE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414273.21905497","2":"ERR1414273","3":"AB618031.1","4":"946522","5":"946522","6":"-Inf","7":"78","8":"-Inf","9":"98","10":"NA","11":"GGCTTCGATACGCCCATTTATACCAACGTGACCTATCCCATTGCCGTGAATCCACCGTTTGTCCCTGAGGAAAACCCCACAGGTTGTTACTCGCTCAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:15 976:1 0:132","19":"0.00000","20":"17.01212","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.01212","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414273.22092353","2":"ERR1414273","3":"OP581373.1","4":"10524","5":"10524","6":"176","7":"188","8":"88","9":"94","10":"CCAAAGTGTGGCTAGTTTTAGTAAAATGCAACGGCATGGTTAACGGGACCATATCCATTAAAGCTCAGAAAGGCATTTTACTTAGACC","11":"CTCCTAACGCCACTTTTTATGAATCACTAGACGCCAAAGTGTGGCTAGTTTTAGTAAAATGCAACGGCATGGTTAACGGGACCATATCCATTAA","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:20 130309:5 0:15 130309:1 0:29 130309:12 0:5","19":"39.30908","20":"41.37966","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"41.37966","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414273.26453800","2":"ERR1414273","3":"MF001305.1","4":"2025360","5":"2025360","6":"83","7":"77","8":"97","9":"86","10":"AACCATAAAAGAACTTTGCTTCTGTATATTGCACTTCTCTCACCATGTTTTCAGCAAGATTTTCATTGACTATCCCATCACTGGCCGCAAAAAATGC","11":"AAGAACTTTGCTTCTGTATATTGCACTTCTCTCACCATGTTTTCAGCAAGATTTTCATTGACTATCCCATCACTGGCCGCAAAAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:55","19":"18.14322","20":"17.28648","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"18.14322","31":"FALSE","32":"FALSE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414273.27927739","2":"ERR1414273","3":"MF001305.1","4":"2025360","5":"2025360","6":"-Inf","7":"97","8":"-Inf","9":"89","10":"NA","11":"ATCCTGCATGAAGTTCACTACGAGATTTTCATTAACGATACCATCGCTGGCCGCGAAAAATGCCAATACATGTTTGATAAAATGGCGTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53 A:34 0:56","19":"0.00000","20":"21.61013","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"21.61013","31":"TRUE","32":"TRUE","33":"NY_014 poxvirus","34":"species","35":"2025358"},{"1":"ERR1414273.34748759","2":"ERR1414273","3":"MK962810.1","4":"130309","5":"130309","6":"134","7":"191","8":"67","9":"98","10":"TATCACAGAAGGGAAGACCAACTGCAGCGCACGCTGGAAGACGCCGAGGCACTGTTCAATAAATACT","11":"CCTTACATGTGGAGCTATCATCCACAGATGGGCTTGGCAGCTGGAGCATCTCAGGACTACTCTAGTCGCATGAACTGGCTAAGTGCCGGCCCCCATAT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:15 130309:5 0:13 A:34 0:65","19":"31.86915","20":"41.65787","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"41.65787","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414273.40457603","2":"ERR1414273","3":"EF420156.1","4":"99000","5":"99000","6":"-Inf","7":"76","8":"-Inf","9":"89","10":"NA","11":"TCCTGCTTTTTGGTATTCACCAACACGCTTTTCAAAGAAATTTGTTTTTCCTTGTAAAGATATCATATCCATAAAATCGAATGGATTTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:13 132475:4 0:39","19":"0.00000","20":"16.93165","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.93165","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414273.44412117","2":"ERR1414273","3":"KY628973.1","4":"10325","5":"10325","6":"50","7":"-Inf","8":"27","9":"-Inf","10":"GTCGTCCTCGTCGTCTTCGGCGTCGTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:15","19":"15.17065","20":"0.00000","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.17065","31":"FALSE","32":"FALSE","33":"Macacine alphaherpesvirus 1","34":"species","35":"10294"},{"1":"ERR1414273.44719823","2":"ERR1414273","3":"KP264964.1","4":"1595998","5":"1595998","6":"-Inf","7":"68","8":"-Inf","9":"84","10":"NA","11":"GTCTGATGTCTGATGTCTGATGTCTGATGTCTGATGTCTGATGTCTGATAACATTATGTAGTAACAATCTAAATTTTTAAAAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:85","19":"0.00000","20":"15.34706","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.34706","31":"FALSE","32":"FALSE","33":"Human smacovirus 1","34":"species","35":"2498817"},{"1":"ERR1414273.45604607","2":"ERR1414273","3":"X74476.1","4":"333764","5":"333764","6":"-Inf","7":"134","8":"-Inf","9":"101","10":"NA","11":"GGGCTGTAAGCCTCCCTTAGGTGAACATTGGGGAACAGGCACACCATGTGCTACTCAGAATCCTGGGGAATGTCCTCCACTAGAATTAAAAAACACCACTA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:67 A:34 0:68","19":"0.00000","20":"29.03499","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"29.03499","31":"TRUE","32":"TRUE","33":"unknown virus","34":"serotype","35":"337049"},{"1":"ERR1414273.45682484","2":"ERR1414273","3":"LT896723.1","4":"10243","5":"10243","6":"62","7":"77","8":"97","9":"97","10":"ATCAAGATTATCAAGATTATCAAGATTAGCAAGATTAGCAAGATTAGCTGAAATTTTAATTTCTAAAAACTTTTGATTTTGTTTTAAAAATTCTTTA","11":"AGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTAT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:63 A:34 0:64","19":"13.55277","20":"16.83166","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.83166","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414273.48417017","2":"ERR1414273","3":"AM501482.1","4":"126794","5":"126794","6":"-Inf","7":"72","8":"-Inf","9":"96","10":"NA","11":"TACCTTTAGCTAATTTTAAAAAATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTAGCTGAAATTTTAATTTCTAAAAACTTTTTATTTC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:63","19":"0.00000","20":"15.77443","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.77443","31":"FALSE","32":"FALSE","33":"Vaccinia virus Ankara","34":"no rank","35":"10245"},{"1":"ERR1414273.52888291","2":"ERR1414273","3":"KY766069.1","4":"64320","5":"64320","6":"-Inf","7":"75","8":"-Inf","9":"98","10":"NA","11":"TATGGCTCACTCCTGTAATCCCAGCACTTTGGGAGGCCAAGGCTGGCGGATCACCTGAGGTCAGGAGCCACTTATCATAGATTTTTAGCGTGGCTTGC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:62 A:34 0:63 9606:2","19":"0.00000","20":"16.35780","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.35780","31":"FALSE","32":"FALSE","33":"Zika virus","34":"species","35":"11051"},{"1":"ERR1414273.54796324","2":"ERR1414273","3":"ON815894.1","4":"130309","5":"130309","6":"188","7":"196","8":"94","9":"98","10":"AGTCTGTCTGGTGGGCTGATTTGGAAGATGAGTTTAAAAATGGAAACATGAATCTTTTATACAAGTATGGATTTGAACAGCTGAAAACACATTG","11":"TGGGCAAAGGGGGGAGAAAAGGAAGTTAGAAAATGACGGGGCGGATTTTCTAAAGGAGTTAACCTTGAGTTTAATGTCTCGTTGTTATCCTGAGTCTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:65","19":"41.37966","20":"42.74839","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414273.56131764","2":"ERR1414273","3":"KJ206566.1","4":"1525173","5":"1525173","6":"68","7":"186","8":"34","9":"96","10":"AGCCAGGACCCGTGGATCCCGTACATGAGCGGGA","11":"TCCCGAAAACTTCGATGAACTGGTCGCCGCACAACTGGAGCGAGAAGCAGAGACACGCGGACCAAGCCTGGATGGAGAATCTGTACGGGGACAGCC","12":"1","13":"Human circovirus VS6600022","14":"1525173","15":"1525173","16":"1","17":"1","18":"A:34 0:8 1525173:5 0:12 1525173:5 0:12 1525173:2 0:1 1525173:2 0:10 1525173:1 0:5","19":"19.28334","20":"40.75062","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"40.75062","31":"TRUE","32":"TRUE","33":"Human circovirus VS6600022","34":"species","35":"345249"},{"1":"ERR1414273.59881094","2":"ERR1414273","3":"OP581370.1","4":"10524","5":"10524","6":"174","7":"184","8":"87","9":"92","10":"AAATTCCTGTGCGATCGCACCATGTGGCGCATTCCGTTTTCCAGCAACTTCATGTCTATGGGGGCACTGACCGACCTGGGGCAGAAC","11":"TACAACATTGGATATCAGGGTTTTTATGTACCCGAAAGCTACAAAGATAGAATGTATTCCTTTTTCCGAAACTTCCAGCCAATGAGTCGTCA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:3 10509:2 0:48 A:34 0:59","19":"38.96184","20":"40.69186","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"40.69186","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414273.60648685","2":"ERR1414273","3":"MZ882390.1","4":"108098","5":"108098","6":"55","7":"-Inf","8":"34","9":"-Inf","10":"CTGAAACTGAAACTGAAACTGAAACTGAAACTGA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:34 0:51","19":"15.59682","20":"0.00000","21":"ERR1414273","22":"Käppala","23":"Primary","24":"228","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.59682","31":"FALSE","32":"FALSE","33":"Human mastadenovirus B","34":"species","35":"10509"},{"1":"ERR1414275.11838034","2":"ERR1414275","3":"HM011556.1","4":"493803","5":"493803","6":"-Inf","7":"174","8":"-Inf","9":"87","10":"NA","11":"CAGGGAATCTCTTAGATTTGCCTTTGGGGAAAAGTGTAAAGTATAACTAAATCTTGCTATTAATGTTTTGGGAATAAAATAATCATT","12":"1","13":"Merkel cell polyomavirus","14":"493803","15":"493803","16":"1","17":"1","18":"0:48 663365:5 0:11 A:34 0:4 493803:5 0:2 493803:8 0:5 493803:1 0:14 493803:1 0:5 493803:4 0:1 493803:4","19":"0.00000","20":"38.96184","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"38.96184","31":"TRUE","32":"TRUE","33":"Merkel cell polyomavirus","34":"no rank","35":"1891726"},{"1":"ERR1414275.1723973","2":"ERR1414275","3":"MF118167.1","4":"2017081","5":"2017081","6":"94","7":"87","8":"60","9":"96","10":"AACTGGCAATAATATATAGTTATTTCTTACATTATTATAATATAACTTAACATATTCCTT","11":"AAGGAATATGTTAAGTTATATTATAATAATGTAAGAAATCACTATATATTTATGGATGCGGGAGAAGAATGCAACTTGGTGGAATTAAGTAATTGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:26 A:34 0:63","19":"22.95850","20":"19.06077","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"22.95850","31":"TRUE","32":"TRUE","33":"Human fecal virus Jorvi2","34":"species","35":"642248"},{"1":"ERR1414275.23187176","2":"ERR1414275","3":"GQ845446.1","4":"765056","5":"765056","6":"124","7":"124","8":"62","9":"62","10":"GGAAAGCGCAGCAGAGACGACGAACCAGATCCAGGTCCGTCGGGTCCCAAGAAAGGCGGATT","11":"GGAAAGCGCAGCAGAGACGACGAACCAGATCCAGGTCCGTCGGGTCCCAAGAAAGGCGGATT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:28","19":"30.04506","20":"30.04506","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"30.04506","31":"TRUE","32":"TRUE","33":"human papillomavirus 124","34":"serotype","35":"337051"},{"1":"ERR1414275.27100682","2":"ERR1414275","3":"OP581347.1","4":"10524","5":"10524","6":"182","7":"188","8":"91","9":"94","10":"GGAGTTTTCCAGCTTGGAGGTGCCGGCCGCTCCTCCTTCACTCCTCGCCAGGCGTACCTGACACTCCAGAGCTCTTCTTCCCAGCCTCGCT","11":"CGCGGTGAGCGAATCTGTGGACGGCTACGACTGAATCCCAATGGTGCGTCCGTGACTGTGTGGCTGCAACATCTACATCGGCGCCTTAATCCTT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:57 A:34 0:61","19":"40.34708","20":"41.37966","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.37966","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414275.31205118","2":"ERR1414275","3":"AB618031.1","4":"946522","5":"946522","6":"-Inf","7":"48","8":"-Inf","9":"24","10":"NA","11":"CCGCCAAGACTGTTACCCATCGCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:25","19":"0.00000","20":"15.10358","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.10358","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414275.39183707","2":"ERR1414275","3":"OP562382.1","4":"10258","5":"10258","6":"76","7":"-Inf","8":"88","9":"-Inf","10":"CGGAGAAATGTTTTGATCGTGCCACCAGGCTCCCCCGTCCCTGGGATTCTCCAGGCAAGAACACTGGAGTGGGTTGCCATTTCCTTCT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:54 A:34 1241979:3 0:61","19":"16.97438","20":"0.00000","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.97438","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414275.45544085","2":"ERR1414275","3":"LT993228.1","4":"10243","5":"10243","6":"79","7":"-Inf","8":"97","9":"-Inf","10":"TATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTATCAAGATTAGCTGAAATTTTAATTTCTAAAAACTTTTGATTTTTTTCTAA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:113","19":"17.26885","20":"0.00000","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.26885","31":"FALSE","32":"FALSE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414275.6793712","2":"ERR1414275","3":"AY386371.1","4":"38804","5":"38804","6":"74","7":"-Inf","8":"91","9":"-Inf","10":"TTTTTTCAGGGTCTTTGATATAAGTATCAATTAAAAGAGAATACATTTCTGAATGAATATTTTCGATGGCAATTTGAAAACCATAAAAACA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:57 A:34 0:63","19":"16.40486","20":"0.00000","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.40486","31":"FALSE","32":"FALSE","33":"Yaba monkey tumor virus","34":"species","35":"10282"},{"1":"ERR1414275.7265485","2":"ERR1414275","3":"U31783.1","4":"37111","5":"37111","6":"174","7":"174","8":"87","9":"87","10":"GGTAGAAGTCGCAGCGCCACGTCATCTCGGGGATACACTACGGTGTCCCCACTGTCTTCCACACTAACTACCAAGTATGGCAATGTC","11":"GGTAGAAGTCGCAGCGCCACGTCATCTCGGGGATACACTACGGTGTCCCCACTGTCTTCCACACTAACTACCAAGTATGGCAATGTC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53","19":"38.96184","20":"38.96184","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"38.96184","31":"TRUE","32":"TRUE","33":"Human papillomavirus 28","34":"serotype","35":"337039"},{"1":"ERR1414275.8380016","2":"ERR1414275","3":"OP581406.1","4":"28284","5":"28284","6":"138","7":"138","8":"69","9":"69","10":"CCTCCACCCGTGCCCACCGTTCCGCCCATGCCGGCTCCGACGCTTGGCACCGCCGTTTCTCGCCCCACC","11":"CCTCCACCCGTGCCCACCGTTCCGCCCATGCCGGCTCCGACGCTTGGCACCGCCGTTTCTCGCCCCACC","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"130309:1 0:29 130309:4 0:1","19":"32.59247","20":"32.59247","21":"ERR1414275","22":"Henriksdal","23":"Primary","24":"233","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"32.59247","31":"TRUE","32":"TRUE","33":"Human adenovirus 40","34":"serotype","35":"130309"},{"1":"ERR1414276.10595779","2":"ERR1414276","3":"ON532817.1","4":"10524","5":"10524","6":"156","7":"156","8":"78","9":"78","10":"CAGCCGCGATAATGAAACTTTTCGAGAAGCTATGCTTAACATTGCTGAGCCCGAGGGGCGCTGGCTTTTAGACCTAGT","11":"CAGCCGCGATAATGAAACTTTTCGAGAAGCTATGCTTAACATTGCTGAGCCCGAGGGGCGCTGGCTTTTAGACCTAGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:44","19":"35.80685","20":"35.80685","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"35.80685","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.13201476","2":"ERR1414276","3":"KC813493.1","4":"10243","5":"10243","6":"104","7":"-Inf","8":"98","9":"-Inf","10":"CTGATTTTTGGTATTCCGCCACACGTTTTTCGAAGAAGTTAGTTTTTCCTTGTAACGAAATCATTTCCATGAAATCGAATGGATTGGTTGCATTGTAT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:41 398049:5 0:18 A:34 0:2","19":"22.68282","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"22.68282","31":"TRUE","32":"TRUE","33":"Cowpox virus","34":"species","35":"10242"},{"1":"ERR1414276.15807888","2":"ERR1414276","3":"OP562382.1","4":"10258","5":"10258","6":"-Inf","7":"77","8":"-Inf","9":"83","10":"NA","11":"CTGGGATTCTCCAGGCAAGAACACTGGAGTGGGTTGCCGTTTCCTTCTCCAATGCATGAAAGTGAAAAGTGAAAGTGAAGTCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:61 A:34 0:50","19":"0.00000","20":"17.42539","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"17.42539","31":"FALSE","32":"FALSE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414276.16295411","2":"ERR1414276","3":"OP581346.1","4":"10524","5":"10524","6":"-Inf","7":"196","8":"-Inf","9":"98","10":"NA","11":"ATTGAACCCGCCCAATTGGCCCGCTGCCCAGGTGTATCAGGAAAACCCCGCTCCGACCACAGTTCTCCTGCCACGCGACGCTGAGGCCGAAGTCCAAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:54 A:34 0:65","19":"0.00000","20":"42.74839","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.17729352","2":"ERR1414276","3":"LT797626.1","4":"10310","5":"10310","6":"52","7":"-Inf","8":"31","9":"-Inf","10":"GGGCCTGATCATCGGCGCACTGGCCGCCGTG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:71","19":"15.14275","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.14275","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 2","34":"species","35":"10294"},{"1":"ERR1414276.19194541","2":"ERR1414276","3":"OR628212.1","4":"10524","5":"10524","6":"184","7":"188","8":"96","9":"98","10":"GCATTTTACTTAGACCTACAGCTAGTTTTATTTCCTTTGTCATGTATTTCTACAGCTACGGAACATGGAGAAAAAACTATCCCGTGTTTGACAACG","11":"CTACAGCTAGTTTTATTTCCTTTGTCATGTATTTCTACAGCTACGGAACATGGAGAAAAAACTATCCCGTGTTTGACAACGAAGGGATACTAGCAAAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:68 130309:1 0:10","19":"40.31244","20":"41.00356","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.00356","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.20851204","2":"ERR1414276","3":"OR628210.1","4":"10524","5":"10524","6":"196","7":"194","8":"98","9":"97","10":"TAAAATCAACATTAAGAGTGGCACAACAAACACAAACACGCATGTAAATTCTACACACATGCTTTTCCCATTTAGACAGTATCATGTCCCACCACATT","11":"GTGGCAATGGGCACAAAATCCTCAGAGGGAGGTGGAAGGCGGTGGGTTTGGTACGAACTCAGTTTGTAGCGAAACCATCTGTCGCGTTGCATCATAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:64 A:34 0:64","19":"42.74839","20":"42.40705","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.25404221","2":"ERR1414276","3":"KY766069.1","4":"64320","5":"64320","6":"74","7":"-Inf","8":"98","9":"-Inf","10":"TCCTCGAGCAGGTTCAGGCAGCGCAGAAAGGTGGACTTGCCCGAGTACTTGGGAGGCTGAGGCAGGAGAATTGCTTGAACCGGGAGGCAGAGGTTGCA","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:19 9606:1 0:94","19":"16.13970","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.13970","31":"FALSE","32":"FALSE","33":"Zika virus","34":"species","35":"11051"},{"1":"ERR1414276.25910329","2":"ERR1414276","3":"EF420156.1","4":"99000","5":"99000","6":"-Inf","7":"75","8":"-Inf","9":"98","10":"NA","11":"GTATTCGGCAACTCTTTTTTCAAAGAAATTTGTTTTTCCTTGCAAAGAAATCATTTCCATAAAAGGAAAAGGATTGCTTACGCCATAGTGCTTGTTAC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:80 2949996:2 0:44","19":"0.00000","20":"16.35780","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.35780","31":"FALSE","32":"FALSE","33":"Tanapox virus","34":"species","35":"10282"},{"1":"ERR1414276.37108503","2":"ERR1414276","3":"OP047700.1","4":"10524","5":"10524","6":"124","7":"-Inf","8":"72","9":"-Inf","10":"GTAACGCCGACAATTTGACAGTAACGGACGCATTGCTATTAGTAATAGCAACCGAACTCGAAGTGTACCATC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:38 A:34 0:64","19":"28.99455","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"28.99455","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.39249709","2":"ERR1414276","3":"AJ293568.1","4":"132475","5":"132475","6":"54","7":"-Inf","8":"34","9":"-Inf","10":"TTCCATAAAATCAAAAGGATTTTTAGAACCATAG","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:34 0:59","19":"15.31324","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.31324","31":"FALSE","32":"FALSE","33":"Yaba-like disease virus","34":"no rank","35":"99000"},{"1":"ERR1414276.40376066","2":"ERR1414276","3":"AB618031.1","4":"946522","5":"946522","6":"63","7":"-Inf","8":"65","9":"-Inf","10":"GCCGCCAGCGCCAGCAGCACGCTCAGACGGATTGCCATTTTCATCATATTTAATCAGCGACTGAT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:31 A:34 0:61","19":"15.09204","20":"0.00000","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.09204","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414276.40910781","2":"ERR1414276","3":"OP047700.1","4":"10524","5":"10524","6":"172","7":"194","8":"86","9":"97","10":"ATATAAGCAGGTGGAGTGGTTTCAGTGTTAGTTTATTCTGATGGAGTTTTGGAGTGAGCTACAAAGTTATCAGAGCCTCCGACGCT","11":"GGTTTCTGGGCTTGCTTTCATTTGTTTTATATTGGATCAATGGAGCGCCCAAACCCATCTGTCGGAGGGGTATACTCTGGATTACATGACAATGGCC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:52 A:34 0:64","19":"38.61396","20":"42.40705","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.40705","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.4359948","2":"ERR1414276","3":"MT123292.2","4":"2697049","5":"2697049","6":"-Inf","7":"68","8":"-Inf","9":"86","10":"NA","11":"GTCCCATTAAACGGCGGCTGTAACCCTAACGGTCCTAAGGTAGCGAAATTCCTTGTCAAATAAGTATTGACCTGCATGAGTGATGT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:84 661478:1 0:21","19":"0.00000","20":"15.26599","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"15.26599","31":"FALSE","32":"FALSE","33":"Severe acute respiratory syndrome coronavirus 2","34":"no rank","35":"694009"},{"1":"ERR1414276.4657994","2":"ERR1414276","3":"MK962807.1","4":"130309","5":"130309","6":"172","7":"190","8":"86","9":"95","10":"TTGCTTCTACCACTCAATTAACTTTGCATGGTGTGTTTTTTCAAAATTGCAGCGGTGTTTGCGTGGATTCCTGGGGTAGGGTGTCT","11":"ACCATTAAGAAAACTGTTAATATACGTAAATGTGCCTATGTAATTGGAAATGGAGCCGTGGTTAGATTCCAAACCTTTGACCGGGTGGTTTTTAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:28 130309:1 0:23 A:34 0:62","19":"38.61396","20":"41.72269","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"41.72269","31":"TRUE","32":"TRUE","33":"Human mastadenovirus F","34":"species","35":"10509"},{"1":"ERR1414276.5174493","2":"ERR1414276","3":"OP581346.1","4":"10524","5":"10524","6":"178","7":"190","8":"89","9":"95","10":"GGTGCGCTCGTCAGGAAGCAACCGTACCTGCCACCCGCGGTTGTGCAGGGTGATCAGATCTATGCTAGTGGCCACCTCGCCCCTTAGGG","11":"TGCAGGGTGATCAGATCTATGCTAGTGGCCACCTCGCCCCTTAGGGGTTCGTTGGTCCAGCACAAGCGTCCTCCCTTGCGCGAGCAGTATGGGGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:104","19":"39.65570","20":"41.72269","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.72269","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.7202440","2":"ERR1414276","3":"OP174917.1","4":"10524","5":"10524","6":"188","7":"196","8":"94","9":"98","10":"AGGAACAAACATAAATAAAGACAATGGGATTCAGATAGGAACAGATACAACAAATCAGCCTATTTATGCCGACAAAACCTATCAACCAGAGCCT","11":"GGCGGTCAGGCTAGTCTCATAACGAACGGAACCGATCAAACTCTAACAAGCGACGTTAACTTACAGTTCTTTGCCTTGCCAAGCACTCCTAATGAGCC","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:60 A:34 0:65","19":"41.37966","20":"42.74839","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"42.74839","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414276.8997076","2":"ERR1414276","3":"AB618031.1","4":"946522","5":"946522","6":"-Inf","7":"77","8":"-Inf","9":"94","10":"NA","11":"GTGCGCATCCACATCAAATGTGAGCGAGTAACAACCTGTGGGGTTTTCCTCAGGGACAAACGGTGGATTCACGGCAATGGGATAGGTCACGTTG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"A:25 0:61","19":"0.00000","20":"16.94805","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"16.94805","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414276.9585471","2":"ERR1414276","3":"ON442328.1","4":"10524","5":"10524","6":"108","7":"108","8":"54","9":"54","10":"CGAAGCTTGCCTTTGGAAATGTGCTGACAGCTGGGACAGTAGAGGCATTTCAGG","11":"CGAAGCTTGCCTTTGGAAATGTGCTGACAGCTGGGACAGTAGAGGCATTTCAGG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:20","19":"27.07456","20":"27.07456","21":"ERR1414276","22":"Henriksdal","23":"Primary","24":"234","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"27.07456","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414277.15139561","2":"ERR1414277","3":"AY386264.1","4":"10258","5":"10258","6":"54","7":"-Inf","8":"89","9":"-Inf","10":"TTAACTTGTTAACTCGTTAGTACTCTAACTTGTTAACTTGTTAACTTGTTAACTCGTTAACTCATTAGCACTTTAATCAAAGAGTTGTT","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"1","18":"0:30 10258:3 0:22 A:34 0:64","19":"12.03038","20":"0.00000","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nhit","30":"12.03038","31":"FALSE","32":"TRUE","33":"Orf virus","34":"species","35":"10257"},{"1":"ERR1414277.2774156","2":"ERR1414277","3":"OP581346.1","4":"10524","5":"10524","6":"186","7":"155","8":"93","9":"91","10":"GTTAAGGTAAAGCCACAATAAGCAGCTTGAAAAACTGGAGTAACACAATGTAAAATGTGCAGCAAAAACTCCGACATGTTTGTCTTTAAAAAA","11":"TAAATAAAACATCAGGGGCTGAGGAACCACCACCGAAATAAAAATAGGTCGTAACAAAAACATTGTGTCCTGCAACAAAAAAAAATATTAA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:53 130309:1 0:5 A:34 0:58","19":"41.03605","20":"34.36152","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"41.03605","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414277.33730654","2":"ERR1414277","3":"AB618031.1","4":"946522","5":"946522","6":"-Inf","7":"103","8":"-Inf","9":"90","10":"NA","11":"TCGGGGAAGACCAGGCCGTTCATACAGAACTGGCGATCGTTTGGTGTATCGCCAAAATCACCGCCGTAGGCGTAGCCAACGCTACCGTCG","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:24 570:2 0:47","19":"0.00000","20":"22.88986","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"22.88986","31":"TRUE","32":"TRUE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414277.41272846","2":"ERR1414277","3":"OP581365.1","4":"10524","5":"10524","6":"182","7":"182","8":"91","9":"91","10":"AAGTGCTGCACCTCTTGGACTAGTGGACAAAACACTAAAAGTTTTGTTTTCTAGCCCCCTCTATCTAGATAACAACTTTCTTACACTAGCC","11":"TTGAACGCCCGCTAGCTCTATCCAGTAACAGAGCAGTGACCCTTAAGTATTCACCACCTTTAAAAATAGAAAACGAAAACTTAACCCTAAG","12":"1","13":"Human mastadenovirus F","14":"130309","15":"130309","16":"1","17":"0","18":"0:5 130309:3 0:49 A:34 0:3 130309:4 0:38 130309:1 0:8 130309:4","19":"40.34708","20":"40.34708","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"Kraken2 HV\\nassignment","30":"40.34708","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414277.41336538","2":"ERR1414277","3":"OP555452.1","4":"10524","5":"10524","6":"122","7":"-Inf","8":"61","9":"-Inf","10":"GCACTTCGCGTTCAAAGGCGCCCGGCGCTTCCTCGATTTCTTCTTCCATGAATTCTTCCTC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:27 A:34 0:53","19":"29.67739","20":"0.00000","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"29.67739","31":"TRUE","32":"TRUE","33":"Human adenovirus 41","34":"serotype","35":"130309"},{"1":"ERR1414277.42145577","2":"ERR1414277","3":"AB618031.1","4":"946522","5":"946522","6":"81","7":"-Inf","8":"88","9":"-Inf","10":"CTGCACGTTCACTAAACGGGTTTCCGGTTTATGCAACAAGGTCACATCACGGAAAATACCGCTCATGCGCCACATATCCTGATCTTCC","11":"NA","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:85","19":"18.09111","20":"0.00000","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"18.09111","31":"FALSE","32":"FALSE","33":"Human alphaherpesvirus 1 strain RH2","34":"no rank","35":"10298"},{"1":"ERR1414277.5885285","2":"ERR1414277","3":"OR777199.1","4":"10529","5":"10529","6":"-Inf","7":"183","8":"-Inf","9":"94","10":"NA","11":"ACATATTTACAGGGTGATACAACTAAACCTATTACAATGAAAGTTGTATTTAATGGCAATGCAACAGTGGATGGATACTCTCTAACATTTATCT","12":"0","13":"unclassified","14":"0","15":"0","16":"0","17":"0","18":"0:61 A:34 0:61","19":"0.00000","20":"40.27914","21":"ERR1414277","22":"Henriksdal","23":"Primary","24":"235","25":"PRJEB14051","26":"Bengtsson-Palme 2016","27":"Sludge (Primary/Surplus)","28":"TRUE","29":"No hit or\\nassignment","30":"40.27914","31":"TRUE","32":"TRUE","33":"Human adenovirus 31","34":"serotype","35":"129875"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Count reads for each herpesviridae taxon</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">mrg_hv_named</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define reference genera</span></span>
<span><span class="va">path_genera_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_rna</span>, genome_type<span class="op">=</span><span class="st">"RNA genome"</span><span class="op">)</span>,</span>
<span>                         <span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_dna</span>, genome_type<span class="op">=</span><span class="st">"DNA genome"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count in each sample</span></span>
<span><span class="va">mrg_hv_named_all</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus_all</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named_all</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">n_path_genera</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus_all</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">sample_group</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"sample_group"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot out and back to add zero lines</span></span>
<span><span class="va">n_path_genera_out</span> <span class="op">&lt;-</span> <span class="va">n_path_genera</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">n_reads_viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"name"</span>, values_from<span class="op">=</span><span class="st">"n_reads_viral"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, names_to<span class="op">=</span><span class="st">"name"</span>, values_to<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Aggregate across dates</span></span>
<span><span class="va">n_path_genera_stype</span> <span class="op">&lt;-</span> <span class="va">n_path_genera_out</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">genome_type</span>, <span class="va">sample_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="st">"All samples"</span>, location<span class="op">=</span><span class="st">"All locations"</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         na_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_path_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_path_genera_stype</span>,</span>
<span>                        <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">name</span>, x<span class="op">=</span><span class="va">p_reads_viral</span>, color<span class="op">=</span><span class="va">sample_group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_st</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">genome_type</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_path_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-30_bengtsson-palme_files/figure-html/ra-genera-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>This dataset is a cautionary tale about the dangers of looking for viruses in samples that were processed with something else in mind. The sample preparation methods used here would be expected to select against viruses, and it shows, with by far the lowest overall human-viral RA of any dataset I’ve investigated so far. This very low abundance had knock-on effects, including worse pipeline performance as the low rate of false positives exceeded the even lower rate of true ones. Even so, there were some interesting results, such as the higher level of human viruses in influent and primary sludge compared to more processed sample types.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb47" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Bengtsson-Palme et al. (2021)"</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Grab samples from Swedish treatment plants."</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-04-30</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> true</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preamble</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a><span class="co"># GGplot themes and scales</span></span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>theme_rotate <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>theme_xblank <span class="ot">&lt;-</span> theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>Returning to wastewater datasets, today I'm analyzing <span class="co">[</span><span class="ot">Bengtsson-Palme et al.</span><span class="co">](https://doi.org/10.1016/j.scitotenv.2016.06.228)</span> (2016), a study of grab samples taken from three treatment plants in Sweden in September 2012. This was a DNA-sequencing study focused on investigating AMR in sewage communities. Influent, effluent and sludge samples were taken from each location, with a total of 70 samples across all sites: 20 influent, 10 effluent and 40 sludge. Liquid samples were filtered through a 1mm seive, then centrifuged, retaining the pellet (note that we expect this to select against viruses), which was then resuspended and underwent DNA extraction. Sludge samples weren't concentrated, but went directly to DNA extraction. Samples were sequenced on an Illumina HiSeq 2500, producing 2x101bp reads.</span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>The sample composition of the Bengtsson-Palme dataset was as follows:</span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: import-qc-data</span></span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Importing the data is a bit more complicated this time as the samples are split across three pipeline runs</span></span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-04-30_bengtsson-palme"</span></span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb47-68"><a href="#cb47-68" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(libraries_path, read_csv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb47-69"><a href="#cb47-69" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries_raw <span class="sc">%&gt;%</span></span>
<span id="cb47-70"><a href="#cb47-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Process sample types</span></span>
<span id="cb47-71"><a href="#cb47-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_group =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Inlet"</span>, sample_type), <span class="st">"Influent"</span>,</span>
<span id="cb47-72"><a href="#cb47-72" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Primary"</span>, sample_type), <span class="st">"Sludge (Primary/Surplus)"</span>,</span>
<span id="cb47-73"><a href="#cb47-73" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Surplus"</span>, sample_type), <span class="st">"Sludge (Primary/Surplus)"</span>,</span>
<span id="cb47-74"><a href="#cb47-74" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Digested"</span>, sample_type), <span class="st">"Sludge (Other)"</span>,</span>
<span id="cb47-75"><a href="#cb47-75" aria-hidden="true" tabindex="-1"></a>                                                    <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Kemikond"</span>, sample_type), <span class="st">"Sludge (Other)"</span>,</span>
<span id="cb47-76"><a href="#cb47-76" aria-hidden="true" tabindex="-1"></a>                                                           <span class="st">"Effluent"</span>))))),</span>
<span id="cb47-77"><a href="#cb47-77" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample_group =</span> <span class="fu">factor</span>(sample_group, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Influent"</span>, <span class="st">"Effluent"</span>, <span class="st">"Sludge (Primary/Surplus)"</span>, </span>
<span id="cb47-78"><a href="#cb47-78" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"Sludge (Other)"</span>)),</span>
<span id="cb47-79"><a href="#cb47-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">sludge =</span> <span class="fu">grepl</span>(<span class="st">"Sludge"</span>, sample_group)) <span class="sc">%&gt;%</span></span>
<span id="cb47-80"><a href="#cb47-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(location, sample_group, sample) <span class="sc">%&gt;%</span></span>
<span id="cb47-81"><a href="#cb47-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="fu">fct_inorder</span>(location),</span>
<span id="cb47-82"><a href="#cb47-82" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb47-83"><a href="#cb47-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-84"><a href="#cb47-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Make table</span></span>
<span id="cb47-85"><a href="#cb47-85" aria-hidden="true" tabindex="-1"></a>count_samples <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_group, location) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb47-86"><a href="#cb47-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"location"</span>, <span class="at">values_from=</span><span class="st">"n"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-87"><a href="#cb47-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span><span class="ot">=</span>sample_group)</span>
<span id="cb47-88"><a href="#cb47-88" aria-hidden="true" tabindex="-1"></a>count_samples</span>
<span id="cb47-89"><a href="#cb47-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-90"><a href="#cb47-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-93"><a href="#cb47-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-94"><a href="#cb47-94" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-qc-data</span></span>
<span id="cb47-95"><a href="#cb47-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-96"><a href="#cb47-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb47-97"><a href="#cb47-97" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb47-98"><a href="#cb47-98" aria-hidden="true" tabindex="-1"></a>import_basic <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb47-99"><a href="#cb47-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb47-100"><a href="#cb47-100" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-101"><a href="#cb47-101" aria-hidden="true" tabindex="-1"></a>      <span class="fu">arrange</span>(location, sample_group, sample) <span class="sc">%&gt;%</span></span>
<span id="cb47-102"><a href="#cb47-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb47-103"><a href="#cb47-103" aria-hidden="true" tabindex="-1"></a>           <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb47-104"><a href="#cb47-104" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-105"><a href="#cb47-105" aria-hidden="true" tabindex="-1"></a>import_basic_paired <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb47-106"><a href="#cb47-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">import_basic</span>(paths) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(read_pair) <span class="sc">%&gt;%</span> </span>
<span id="cb47-107"><a href="#cb47-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb47-108"><a href="#cb47-108" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-109"><a href="#cb47-109" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">import_basic</span>(basic_stats_path)</span>
<span id="cb47-110"><a href="#cb47-110" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(adapter_stats_path)</span>
<span id="cb47-111"><a href="#cb47-111" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_base_stats_path)</span>
<span id="cb47-112"><a href="#cb47-112" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_seq_stats_path)</span>
<span id="cb47-113"><a href="#cb47-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-114"><a href="#cb47-114" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb47-115"><a href="#cb47-115" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb47-116"><a href="#cb47-116" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb47-117"><a href="#cb47-117" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb47-118"><a href="#cb47-118" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb47-119"><a href="#cb47-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-120"><a href="#cb47-120" aria-hidden="true" tabindex="-1"></a><span class="co"># Get key values for readout</span></span>
<span id="cb47-121"><a href="#cb47-121" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb47-122"><a href="#cb47-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb47-123"><a href="#cb47-123" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb47-124"><a href="#cb47-124" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb47-125"><a href="#cb47-125" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb47-126"><a href="#cb47-126" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb47-127"><a href="#cb47-127" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb47-128"><a href="#cb47-128" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-129"><a href="#cb47-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-130"><a href="#cb47-130" aria-hidden="true" tabindex="-1"></a>These 70 samples yielded 23.7M-61.3M (mean 38.7M) reads per sample, for a total of 2.7B read pairs (539 gigabases of sequence). Read qualities were mostly high but tailed off towards the 3' end, requiring some trimming. Adapter levels were high. Inferred duplication levels were low in sludge samples (1-12%, mean 4%) but much higher in liquid samples samples (22-90%, mean 46%), implying lower available sequence diversity in the latter sample groups.</span>
<span id="cb47-131"><a href="#cb47-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-134"><a href="#cb47-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-135"><a href="#cb47-135" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb47-136"><a href="#cb47-136" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-137"><a href="#cb47-137" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb47-138"><a href="#cb47-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-139"><a href="#cb47-139" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb47-140"><a href="#cb47-140" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb47-141"><a href="#cb47-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, sample_group, location,</span>
<span id="cb47-142"><a href="#cb47-142" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb47-143"><a href="#cb47-143" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb47-144"><a href="#cb47-144" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb47-145"><a href="#cb47-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(sample<span class="sc">:</span>location), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-146"><a href="#cb47-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric))</span>
<span id="cb47-147"><a href="#cb47-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-148"><a href="#cb47-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb47-149"><a href="#cb47-149" aria-hidden="true" tabindex="-1"></a>scale_fill_st <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>, <span class="at">name=</span><span class="st">"Sample Type"</span>)</span>
<span id="cb47-150"><a href="#cb47-150" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, </span>
<span id="cb47-151"><a href="#cb47-151" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>value, <span class="at">fill=</span>sample_group, <span class="at">group=</span><span class="fu">interaction</span>(sample_group,sample))) <span class="sc">+</span></span>
<span id="cb47-152"><a href="#cb47-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb47-153"><a href="#cb47-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-154"><a href="#cb47-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb47-155"><a href="#cb47-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_st</span>() <span class="sc">+</span> </span>
<span id="cb47-156"><a href="#cb47-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric<span class="sc">~</span>location, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space=</span><span class="st">"free_x"</span>, <span class="at">switch=</span><span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb47-157"><a href="#cb47-157" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb47-158"><a href="#cb47-158" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-159"><a href="#cb47-159" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"plain"</span>)</span>
<span id="cb47-160"><a href="#cb47-160" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb47-161"><a href="#cb47-161" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb47-162"><a href="#cb47-162" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-163"><a href="#cb47-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-166"><a href="#cb47-166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-167"><a href="#cb47-167" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb47-168"><a href="#cb47-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-169"><a href="#cb47-169" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb47-170"><a href="#cb47-170" aria-hidden="true" tabindex="-1"></a>scale_color_st <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb47-171"><a href="#cb47-171" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">name=</span><span class="st">"Sample Type"</span>)</span>
<span id="cb47-172"><a href="#cb47-172" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>sample_group, <span class="at">linetype=</span>read_pair, </span>
<span id="cb47-173"><a href="#cb47-173" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb47-174"><a href="#cb47-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_st</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb47-175"><a href="#cb47-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb47-176"><a href="#cb47-176" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb47-177"><a href="#cb47-177" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb47-178"><a href="#cb47-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-179"><a href="#cb47-179" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb47-180"><a href="#cb47-180" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb47-181"><a href="#cb47-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb47-182"><a href="#cb47-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb47-183"><a href="#cb47-183" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-184"><a href="#cb47-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb47-185"><a href="#cb47-185" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-186"><a href="#cb47-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb47-187"><a href="#cb47-187" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb47-188"><a href="#cb47-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-189"><a href="#cb47-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb47-190"><a href="#cb47-190" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb47-191"><a href="#cb47-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-192"><a href="#cb47-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-193"><a href="#cb47-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb47-194"><a href="#cb47-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb47-195"><a href="#cb47-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb47-196"><a href="#cb47-196" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb47-197"><a href="#cb47-197" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb47-198"><a href="#cb47-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-199"><a href="#cb47-199" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb47-200"><a href="#cb47-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-201"><a href="#cb47-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-202"><a href="#cb47-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb47-203"><a href="#cb47-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-204"><a href="#cb47-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb47-205"><a href="#cb47-205" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb47-206"><a href="#cb47-206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-207"><a href="#cb47-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-208"><a href="#cb47-208" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb47-209"><a href="#cb47-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-210"><a href="#cb47-210" aria-hidden="true" tabindex="-1"></a>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. As expected given the observed difference in duplication levels, many more reads were lost during deduplication in liquid samples than sludge samples. Conversely, trimming and filtering consistently removed more reads in sludge than in liquid samples, though the effect was less dramatic than for deduplication. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</span>
<span id="cb47-211"><a href="#cb47-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-214"><a href="#cb47-214" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-215"><a href="#cb47-215" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb47-216"><a href="#cb47-216" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb47-217"><a href="#cb47-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, sample_group, sludge, stage, </span>
<span id="cb47-218"><a href="#cb47-218" aria-hidden="true" tabindex="-1"></a>         percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb47-219"><a href="#cb47-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb47-220"><a href="#cb47-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> <span class="fu">replace_na</span>(n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs), <span class="dv">0</span>),</span>
<span id="cb47-221"><a href="#cb47-221" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb47-222"><a href="#cb47-222" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb47-223"><a href="#cb47-223" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb47-224"><a href="#cb47-224" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> <span class="fu">replace_na</span>(p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs), <span class="dv">0</span>))</span>
<span id="cb47-225"><a href="#cb47-225" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb47-226"><a href="#cb47-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Sludge?</span><span class="st">`</span><span class="ot">=</span>sludge, <span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb47-227"><a href="#cb47-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb47-228"><a href="#cb47-228" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-229"><a href="#cb47-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-230"><a href="#cb47-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb47-231"><a href="#cb47-231" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb47-232"><a href="#cb47-232" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-233"><a href="#cb47-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-236"><a href="#cb47-236" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-237"><a href="#cb47-237" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb47-238"><a href="#cb47-238" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-239"><a href="#cb47-239" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb47-240"><a href="#cb47-240" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb47-241"><a href="#cb47-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-242"><a href="#cb47-242" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>sample_group, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb47-243"><a href="#cb47-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_st</span>() <span class="sc">+</span></span>
<span id="cb47-244"><a href="#cb47-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>location, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb47-245"><a href="#cb47-245" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb47-246"><a href="#cb47-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-247"><a href="#cb47-247" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb47-248"><a href="#cb47-248" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb47-249"><a href="#cb47-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb47-250"><a href="#cb47-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb47-251"><a href="#cb47-251" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb47-252"><a href="#cb47-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-253"><a href="#cb47-253" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb47-254"><a href="#cb47-254" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>sample_group, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb47-255"><a href="#cb47-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb47-256"><a href="#cb47-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb47-257"><a href="#cb47-257" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-258"><a href="#cb47-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_st</span>() <span class="sc">+</span></span>
<span id="cb47-259"><a href="#cb47-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>location, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb47-260"><a href="#cb47-260" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb47-261"><a href="#cb47-261" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb47-262"><a href="#cb47-262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-263"><a href="#cb47-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-264"><a href="#cb47-264" aria-hidden="true" tabindex="-1"></a>Data cleaning was very successful at removing adapters and improving read qualities:</span>
<span id="cb47-265"><a href="#cb47-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-268"><a href="#cb47-268" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-269"><a href="#cb47-269" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-270"><a href="#cb47-270" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb47-271"><a href="#cb47-271" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb47-272"><a href="#cb47-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-273"><a href="#cb47-273" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>sample_group, <span class="at">linetype=</span>read_pair, </span>
<span id="cb47-274"><a href="#cb47-274" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb47-275"><a href="#cb47-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_st</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb47-276"><a href="#cb47-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb47-277"><a href="#cb47-277" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb47-278"><a href="#cb47-278" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb47-279"><a href="#cb47-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-280"><a href="#cb47-280" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb47-281"><a href="#cb47-281" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb47-282"><a href="#cb47-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb47-283"><a href="#cb47-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb47-284"><a href="#cb47-284" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-285"><a href="#cb47-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb47-286"><a href="#cb47-286" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-287"><a href="#cb47-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb47-288"><a href="#cb47-288" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb47-289"><a href="#cb47-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-290"><a href="#cb47-290" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb47-291"><a href="#cb47-291" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb47-292"><a href="#cb47-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-293"><a href="#cb47-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-294"><a href="#cb47-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb47-295"><a href="#cb47-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb47-296"><a href="#cb47-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb47-297"><a href="#cb47-297" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-298"><a href="#cb47-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb47-299"><a href="#cb47-299" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb47-300"><a href="#cb47-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-301"><a href="#cb47-301" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb47-302"><a href="#cb47-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-303"><a href="#cb47-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-304"><a href="#cb47-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb47-305"><a href="#cb47-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-306"><a href="#cb47-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-307"><a href="#cb47-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb47-308"><a href="#cb47-308" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb47-309"><a href="#cb47-309" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-310"><a href="#cb47-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-311"><a href="#cb47-311" aria-hidden="true" tabindex="-1"></a>According to FASTQC, cleaning + deduplication was very effective at reducing measured duplicate levels, which fell from an average of 45% to 5% in liquid samples and from 6% to 3% in sludge samples:</span>
<span id="cb47-312"><a href="#cb47-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-315"><a href="#cb47-315" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-316"><a href="#cb47-316" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb47-317"><a href="#cb47-317" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3.5</span></span>
<span id="cb47-318"><a href="#cb47-318" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb47-319"><a href="#cb47-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-320"><a href="#cb47-320" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sludge,stage) <span class="sc">%&gt;%</span> </span>
<span id="cb47-321"><a href="#cb47-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb47-322"><a href="#cb47-322" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb47-323"><a href="#cb47-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-324"><a href="#cb47-324" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb47-325"><a href="#cb47-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb47-326"><a href="#cb47-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb47-327"><a href="#cb47-327" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb47-328"><a href="#cb47-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-329"><a href="#cb47-329" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb47-330"><a href="#cb47-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb47-331"><a href="#cb47-331" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb47-332"><a href="#cb47-332" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-333"><a href="#cb47-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-334"><a href="#cb47-334" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb47-335"><a href="#cb47-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-336"><a href="#cb47-336" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb47-337"><a href="#cb47-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-340"><a href="#cb47-340" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-341"><a href="#cb47-341" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb47-342"><a href="#cb47-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-343"><a href="#cb47-343" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb47-344"><a href="#cb47-344" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb47-345"><a href="#cb47-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-346"><a href="#cb47-346" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb47-347"><a href="#cb47-347" aria-hidden="true" tabindex="-1"></a>comp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb47-348"><a href="#cb47-348" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(comp_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-349"><a href="#cb47-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-350"><a href="#cb47-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels =</span> classifications))</span>
<span id="cb47-351"><a href="#cb47-351" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-352"><a href="#cb47-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-353"><a href="#cb47-353" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb47-354"><a href="#cb47-354" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb47-355"><a href="#cb47-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(location, sample_group, classification) <span class="sc">%&gt;%</span></span>
<span id="cb47-356"><a href="#cb47-356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-357"><a href="#cb47-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb47-358"><a href="#cb47-358" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb47-359"><a href="#cb47-359" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb47-360"><a href="#cb47-360" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-361"><a href="#cb47-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-364"><a href="#cb47-364" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-365"><a href="#cb47-365" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb47-366"><a href="#cb47-366" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb47-367"><a href="#cb47-367" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb47-368"><a href="#cb47-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-369"><a href="#cb47-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb47-370"><a href="#cb47-370" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb47-371"><a href="#cb47-371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(location<span class="sc">~</span>sample_group, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol=</span><span class="dv">4</span>,</span>
<span id="cb47-372"><a href="#cb47-372" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line=</span><span class="cn">FALSE</span>, <span class="at">width=</span><span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb47-373"><a href="#cb47-373" aria-hidden="true" tabindex="-1"></a>  theme_xblank</span>
<span id="cb47-374"><a href="#cb47-374" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb47-375"><a href="#cb47-375" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb47-376"><a href="#cb47-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-377"><a href="#cb47-377" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb47-378"><a href="#cb47-378" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data =</span> comp, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-379"><a href="#cb47-379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb47-380"><a href="#cb47-380" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb47-381"><a href="#cb47-381" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb47-382"><a href="#cb47-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-383"><a href="#cb47-383" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb47-384"><a href="#cb47-384" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb47-385"><a href="#cb47-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb47-386"><a href="#cb47-386" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb47-387"><a href="#cb47-387" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data=</span>comp_minor, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-388"><a href="#cb47-388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb47-389"><a href="#cb47-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb47-390"><a href="#cb47-390" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb47-391"><a href="#cb47-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-392"><a href="#cb47-392" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-393"><a href="#cb47-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-396"><a href="#cb47-396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-397"><a href="#cb47-397" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-summary</span></span>
<span id="cb47-398"><a href="#cb47-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-399"><a href="#cb47-399" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb47-400"><a href="#cb47-400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb47-401"><a href="#cb47-401" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb47-402"><a href="#cb47-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-403"><a href="#cb47-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-404"><a href="#cb47-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-405"><a href="#cb47-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb47-406"><a href="#cb47-406" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb47-407"><a href="#cb47-407" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> p_reads_summ_group <span class="sc">%&gt;%</span></span>
<span id="cb47-408"><a href="#cb47-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb47-409"><a href="#cb47-409" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb47-410"><a href="#cb47-410" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb47-411"><a href="#cb47-411" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb47-412"><a href="#cb47-412" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb47-413"><a href="#cb47-413" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb47-414"><a href="#cb47-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_group, classification, <span class="at">read_fraction=</span>display) <span class="sc">%&gt;%</span></span>
<span id="cb47-415"><a href="#cb47-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample_group, classification)</span>
<span id="cb47-416"><a href="#cb47-416" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb47-417"><a href="#cb47-417" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-418"><a href="#cb47-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-419"><a href="#cb47-419" aria-hidden="true" tabindex="-1"></a>In all sample types, the majority of reads were either filtered, duplicates, or unassigned. Among assigned reads, the vast majority were bacterial, which is unsurprising given the sample processing protocols used. Total viral fraction averaged 0.011% in influent and primary sludge, and considerably lower in effluent and treated sludge. The human fraction was also low, averaging 0.033% across all sample types. Interestingly, treated sludge showed far higher fractions of archaeal reads than other sample types, possibly due to the anaerobic conditions experienced during sludge treatment.</span>
<span id="cb47-420"><a href="#cb47-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-421"><a href="#cb47-421" aria-hidden="true" tabindex="-1"></a>As is common for DNA data, viral reads were overwhelmingly dominated by *Caudoviricetes* phages:</span>
<span id="cb47-422"><a href="#cb47-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-425"><a href="#cb47-425" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-426"><a href="#cb47-426" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb47-427"><a href="#cb47-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-428"><a href="#cb47-428" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Kraken reports</span></span>
<span id="cb47-429"><a href="#cb47-429" aria-hidden="true" tabindex="-1"></a>reports_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"kraken_reports.tsv.gz"</span>)</span>
<span id="cb47-430"><a href="#cb47-430" aria-hidden="true" tabindex="-1"></a>reports <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reports_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-431"><a href="#cb47-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-432"><a href="#cb47-432" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb47-433"><a href="#cb47-433" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb47-434"><a href="#cb47-434" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-435"><a href="#cb47-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-436"><a href="#cb47-436" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to viral taxa</span></span>
<span id="cb47-437"><a href="#cb47-437" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-438"><a href="#cb47-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb47-439"><a href="#cb47-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb47-440"><a href="#cb47-440" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb47-441"><a href="#cb47-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-442"><a href="#cb47-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"minimizers"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-443"><a href="#cb47-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>())</span>
<span id="cb47-444"><a href="#cb47-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-445"><a href="#cb47-445" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb47-446"><a href="#cb47-446" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb47-447"><a href="#cb47-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-448"><a href="#cb47-448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-449"><a href="#cb47-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-452"><a href="#cb47-452" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-453"><a href="#cb47-453" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-class-composition</span></span>
<span id="cb47-454"><a href="#cb47-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-455"><a href="#cb47-455" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb47-456"><a href="#cb47-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-457"><a href="#cb47-457" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral classes</span></span>
<span id="cb47-458"><a href="#cb47-458" aria-hidden="true" tabindex="-1"></a>viral_classes_major_tab <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb47-459"><a href="#cb47-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-460"><a href="#cb47-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-461"><a href="#cb47-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb47-462"><a href="#cb47-462" aria-hidden="true" tabindex="-1"></a>viral_classes_major_list <span class="ot">&lt;-</span> viral_classes_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb47-463"><a href="#cb47-463" aria-hidden="true" tabindex="-1"></a>viral_classes_major <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb47-464"><a href="#cb47-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_classes_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb47-465"><a href="#cb47-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, sample_group, location, p_reads_viral)</span>
<span id="cb47-466"><a href="#cb47-466" aria-hidden="true" tabindex="-1"></a>viral_classes_minor <span class="ot">&lt;-</span> viral_classes_major <span class="sc">%&gt;%</span> </span>
<span id="cb47-467"><a href="#cb47-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, sample_group, location) <span class="sc">%&gt;%</span></span>
<span id="cb47-468"><a href="#cb47-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-469"><a href="#cb47-469" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb47-470"><a href="#cb47-470" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, sample_group, location, p_reads_viral)</span>
<span id="cb47-471"><a href="#cb47-471" aria-hidden="true" tabindex="-1"></a>viral_classes_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_classes_major, viral_classes_minor) <span class="sc">%&gt;%</span></span>
<span id="cb47-472"><a href="#cb47-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-473"><a href="#cb47-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_classes_major_list, <span class="st">"Other"</span>)),</span>
<span id="cb47-474"><a href="#cb47-474" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> <span class="fu">pmax</span>(p_reads_viral, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-475"><a href="#cb47-475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name)</span>
<span id="cb47-476"><a href="#cb47-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-477"><a href="#cb47-477" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb47-478"><a href="#cb47-478" aria-hidden="true" tabindex="-1"></a>g_classes <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-479"><a href="#cb47-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>viral_classes_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-480"><a href="#cb47-480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-481"><a href="#cb47-481" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-482"><a href="#cb47-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb47-483"><a href="#cb47-483" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb47-484"><a href="#cb47-484" aria-hidden="true" tabindex="-1"></a>g_classes</span>
<span id="cb47-485"><a href="#cb47-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-486"><a href="#cb47-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-487"><a href="#cb47-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-488"><a href="#cb47-488" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb47-489"><a href="#cb47-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-490"><a href="#cb47-490" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples. A grand total of 565 reads were identified as putatively human-viral, with many samples showing fewer than 5 total HV read pairs. Even this total likely overstates total human-viral presence, however, as more than 300 of these reads had only low alignment scores to their putative viral sources:</span>
<span id="cb47-491"><a href="#cb47-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-494"><a href="#cb47-494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-495"><a href="#cb47-495" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-read-counts</span></span>
<span id="cb47-496"><a href="#cb47-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-497"><a href="#cb47-497" aria-hidden="true" tabindex="-1"></a><span class="co"># Import HV read data</span></span>
<span id="cb47-498"><a href="#cb47-498" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb47-499"><a href="#cb47-499" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hv_reads_filtered_path, read_tsv,</span>
<span id="cb47-500"><a href="#cb47-500" aria-hidden="true" tabindex="-1"></a>                            <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-501"><a href="#cb47-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb47-502"><a href="#cb47-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb47-503"><a href="#cb47-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-504"><a href="#cb47-504" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb47-505"><a href="#cb47-505" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb47-506"><a href="#cb47-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, seq_id) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb47-507"><a href="#cb47-507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb47-508"><a href="#cb47-508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-509"><a href="#cb47-509" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-510"><a href="#cb47-510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb47-511"><a href="#cb47-511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb47-512"><a href="#cb47-512" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_summ <span class="ot">&lt;-</span> n_hv_filtered <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb47-513"><a href="#cb47-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total), </span>
<span id="cb47-514"><a href="#cb47-514" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-515"><a href="#cb47-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb47-516"><a href="#cb47-516" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-517"><a href="#cb47-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-520"><a href="#cb47-520" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-521"><a href="#cb47-521" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-scores</span></span>
<span id="cb47-522"><a href="#cb47-522" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-523"><a href="#cb47-523" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb47-524"><a href="#cb47-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-525"><a href="#cb47-525" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse multi-entry sequences</span></span>
<span id="cb47-526"><a href="#cb47-526" aria-hidden="true" tabindex="-1"></a>rmax <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(max, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb47-527"><a href="#cb47-527" aria-hidden="true" tabindex="-1"></a>collapse <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">ifelse</span>(<span class="fu">all</span>(x <span class="sc">==</span> x[<span class="dv">1</span>]), x[<span class="dv">1</span>], <span class="fu">paste</span>(x, <span class="at">collapse=</span><span class="st">"/"</span>))</span>
<span id="cb47-528"><a href="#cb47-528" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> </span>
<span id="cb47-529"><a href="#cb47-529" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-530"><a href="#cb47-530" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb47-531"><a href="#cb47-531" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb47-532"><a href="#cb47-532" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sample =</span> <span class="fu">collapse</span>(sample),</span>
<span id="cb47-533"><a href="#cb47-533" aria-hidden="true" tabindex="-1"></a>            <span class="at">genome_id =</span> <span class="fu">collapse</span>(genome_id),</span>
<span id="cb47-534"><a href="#cb47-534" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid_best =</span> taxid[<span class="dv">1</span>],</span>
<span id="cb47-535"><a href="#cb47-535" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(taxid)),</span>
<span id="cb47-536"><a href="#cb47-536" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_fwd =</span> <span class="fu">rmax</span>(best_alignment_score_fwd),</span>
<span id="cb47-537"><a href="#cb47-537" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_rev =</span> <span class="fu">rmax</span>(best_alignment_score_rev),</span>
<span id="cb47-538"><a href="#cb47-538" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_fwd =</span> <span class="fu">rmax</span>(query_len_fwd),</span>
<span id="cb47-539"><a href="#cb47-539" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_rev =</span> <span class="fu">rmax</span>(query_len_rev),</span>
<span id="cb47-540"><a href="#cb47-540" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_fwd =</span> query_seq_fwd[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_fwd)][<span class="dv">1</span>],</span>
<span id="cb47-541"><a href="#cb47-541" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_rev =</span> query_seq_rev[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_rev)][<span class="dv">1</span>],</span>
<span id="cb47-542"><a href="#cb47-542" aria-hidden="true" tabindex="-1"></a>            <span class="at">classified =</span> <span class="fu">rmax</span>(classified),</span>
<span id="cb47-543"><a href="#cb47-543" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_name =</span> <span class="fu">collapse</span>(assigned_name),</span>
<span id="cb47-544"><a href="#cb47-544" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid_best =</span> assigned_taxid[<span class="dv">1</span>],</span>
<span id="cb47-545"><a href="#cb47-545" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(assigned_taxid)),</span>
<span id="cb47-546"><a href="#cb47-546" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_hv =</span> <span class="fu">rmax</span>(assigned_hv),</span>
<span id="cb47-547"><a href="#cb47-547" aria-hidden="true" tabindex="-1"></a>            <span class="at">hit_hv =</span> <span class="fu">rmax</span>(hit_hv),</span>
<span id="cb47-548"><a href="#cb47-548" aria-hidden="true" tabindex="-1"></a>            <span class="at">encoded_hits =</span> <span class="fu">collapse</span>(encoded_hits),</span>
<span id="cb47-549"><a href="#cb47-549" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_fwd =</span> <span class="fu">rmax</span>(adj_score_fwd),</span>
<span id="cb47-550"><a href="#cb47-550" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_rev =</span> <span class="fu">rmax</span>(adj_score_rev)</span>
<span id="cb47-551"><a href="#cb47-551" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb47-552"><a href="#cb47-552" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-553"><a href="#cb47-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb47-554"><a href="#cb47-554" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb47-555"><a href="#cb47-555" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb47-556"><a href="#cb47-556" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb47-557"><a href="#cb47-557" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb47-558"><a href="#cb47-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-559"><a href="#cb47-559" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot results</span></span>
<span id="cb47-560"><a href="#cb47-560" aria-hidden="true" tabindex="-1"></a>geom_vhist <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_histogram, <span class="at">binwidth=</span><span class="dv">5</span>, <span class="at">boundary=</span><span class="dv">0</span>)</span>
<span id="cb47-561"><a href="#cb47-561" aria-hidden="true" tabindex="-1"></a>g_vhist_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span></span>
<span id="cb47-562"><a href="#cb47-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb47-563"><a href="#cb47-563" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>)), <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb47-564"><a href="#cb47-564" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> </span>
<span id="cb47-565"><a href="#cb47-565" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> </span>
<span id="cb47-566"><a href="#cb47-566" aria-hidden="true" tabindex="-1"></a>  theme_base </span>
<span id="cb47-567"><a href="#cb47-567" aria-hidden="true" tabindex="-1"></a>g_vhist_0 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg)</span>
<span id="cb47-568"><a href="#cb47-568" aria-hidden="true" tabindex="-1"></a>g_vhist_0</span>
<span id="cb47-569"><a href="#cb47-569" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-570"><a href="#cb47-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-571"><a href="#cb47-571" aria-hidden="true" tabindex="-1"></a>BLASTing these reads against nt</span>
<span id="cb47-572"><a href="#cb47-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-573"><a href="#cb47-573" aria-hidden="true" tabindex="-1"></a>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I've used for previous datasets. As expected, most low-scoring sequences were not found to be true viral hits:</span>
<span id="cb47-574"><a href="#cb47-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-577"><a href="#cb47-577" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-578"><a href="#cb47-578" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-blast-data</span></span>
<span id="cb47-579"><a href="#cb47-579" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-580"><a href="#cb47-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-581"><a href="#cb47-581" aria-hidden="true" tabindex="-1"></a><span class="co"># Import paired BLAST results</span></span>
<span id="cb47-582"><a href="#cb47-582" aria-hidden="true" tabindex="-1"></a>blast_paired_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_blast_paired.tsv.gz"</span>)</span>
<span id="cb47-583"><a href="#cb47-583" aria-hidden="true" tabindex="-1"></a>blast_paired <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_paired_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-584"><a href="#cb47-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-585"><a href="#cb47-585" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb47-586"><a href="#cb47-586" aria-hidden="true" tabindex="-1"></a>blast_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_paired, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-587"><a href="#cb47-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb47-588"><a href="#cb47-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-589"><a href="#cb47-589" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb47-590"><a href="#cb47-590" aria-hidden="true" tabindex="-1"></a>match_taxid <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_1, taxid_2){</span>
<span id="cb47-591"><a href="#cb47-591" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"/"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb47-592"><a href="#cb47-592" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"/"</span>), taxid_2)</span>
<span id="cb47-593"><a href="#cb47-593" aria-hidden="true" tabindex="-1"></a>  p3 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb47-594"><a href="#cb47-594" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">setNames</span>(p1<span class="sc">|</span>p2<span class="sc">|</span>p3, <span class="cn">NULL</span>)</span>
<span id="cb47-595"><a href="#cb47-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb47-596"><a href="#cb47-596" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-597"><a href="#cb47-597" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_id, taxid, assigned_taxid, adj_score_max)</span>
<span id="cb47-598"><a href="#cb47-598" aria-hidden="true" tabindex="-1"></a>blast_assign <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(blast_viral, mrg_assign, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-599"><a href="#cb47-599" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> <span class="fu">match_taxid</span>(staxid, taxid),</span>
<span id="cb47-600"><a href="#cb47-600" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> <span class="fu">match_taxid</span>(staxid, assigned_taxid),</span>
<span id="cb47-601"><a href="#cb47-601" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb47-602"><a href="#cb47-602" aria-hidden="true" tabindex="-1"></a>blast_out <span class="ot">&lt;-</span> blast_assign <span class="sc">%&gt;%</span></span>
<span id="cb47-603"><a href="#cb47-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb47-604"><a href="#cb47-604" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb47-605"><a href="#cb47-605" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb47-606"><a href="#cb47-606" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb47-607"><a href="#cb47-607" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb47-608"><a href="#cb47-608" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-609"><a href="#cb47-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-612"><a href="#cb47-612" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-613"><a href="#cb47-613" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-results</span></span>
<span id="cb47-614"><a href="#cb47-614" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb47-615"><a href="#cb47-615" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-616"><a href="#cb47-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-617"><a href="#cb47-617" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb47-618"><a href="#cb47-618" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg, blast_out, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-619"><a href="#cb47-619" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb47-620"><a href="#cb47-620" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb47-621"><a href="#cb47-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-622"><a href="#cb47-622" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot RNA</span></span>
<span id="cb47-623"><a href="#cb47-623" aria-hidden="true" tabindex="-1"></a>g_vhist_1 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg_blast, <span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">fill=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb47-624"><a href="#cb47-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>)</span>
<span id="cb47-625"><a href="#cb47-625" aria-hidden="true" tabindex="-1"></a>g_vhist_1</span>
<span id="cb47-626"><a href="#cb47-626" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-627"><a href="#cb47-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-628"><a href="#cb47-628" aria-hidden="true" tabindex="-1"></a>In order to achieve decent performance metrics under these conditions, I needed to exclude low-scoring reads with Kraken hits as well as those without. Doing this at my normal disjunctive score threshold of 20 gave a passable precision (93%) but poor sensitivity (79%), leading to a poor overall F1 score (85%):</span>
<span id="cb47-629"><a href="#cb47-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-632"><a href="#cb47-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-633"><a href="#cb47-633" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-f1</span></span>
<span id="cb47-634"><a href="#cb47-634" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold){</span>
<span id="cb47-635"><a href="#cb47-635" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> </span>
<span id="cb47-636"><a href="#cb47-636" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">retain_score =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb47-637"><a href="#cb47-637" aria-hidden="true" tabindex="-1"></a>           <span class="at">retain =</span> assigned_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb47-638"><a href="#cb47-638" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb47-639"><a href="#cb47-639" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb47-640"><a href="#cb47-640" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb47-641"><a href="#cb47-641" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb47-642"><a href="#cb47-642" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb47-643"><a href="#cb47-643" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb47-644"><a href="#cb47-644" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb47-645"><a href="#cb47-645" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb47-646"><a href="#cb47-646" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb47-647"><a href="#cb47-647" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb47-648"><a href="#cb47-648" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb47-649"><a href="#cb47-649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb47-650"><a href="#cb47-650" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-651"><a href="#cb47-651" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb47-652"><a href="#cb47-652" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab)</span>
<span id="cb47-653"><a href="#cb47-653" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb47-654"><a href="#cb47-654" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>threshold, <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>)</span>
<span id="cb47-655"><a href="#cb47-655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats)</span>
<span id="cb47-656"><a href="#cb47-656" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-657"><a href="#cb47-657" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast)</span>
<span id="cb47-658"><a href="#cb47-658" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb47-659"><a href="#cb47-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb47-660"><a href="#cb47-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb47-661"><a href="#cb47-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-662"><a href="#cb47-662" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adjusted Score Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-663"><a href="#cb47-663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb47-664"><a href="#cb47-664" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb47-665"><a href="#cb47-665" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb47-666"><a href="#cb47-666" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-667"><a href="#cb47-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">Threshold=</span>threshold, <span class="at">Metric=</span>metric, <span class="at">Value=</span>value)</span>
<span id="cb47-668"><a href="#cb47-668" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-669"><a href="#cb47-669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-670"><a href="#cb47-670" aria-hidden="true" tabindex="-1"></a>Looking into the composition of different read groups, the bulk of high-scoring false positives map to human alphaherpesvirus 1 strain RH2 according to Bowtie2. BLASTN maps these sequences to a variety of bacterial taxa, especially *E. coli* and various *Klebsiella* species.</span>
<span id="cb47-671"><a href="#cb47-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-674"><a href="#cb47-674" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-675"><a href="#cb47-675" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fp</span></span>
<span id="cb47-676"><a href="#cb47-676" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-677"><a href="#cb47-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-678"><a href="#cb47-678" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb47-679"><a href="#cb47-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-680"><a href="#cb47-680" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing viral taxa</span></span>
<span id="cb47-681"><a href="#cb47-681" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">211787</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span id="cb47-682"><a href="#cb47-682" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">509154</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span id="cb47-683"><a href="#cb47-683" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">493803</span>] <span class="ot">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span id="cb47-684"><a href="#cb47-684" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">427343</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span id="cb47-685"><a href="#cb47-685" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194958</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus A"</span></span>
<span id="cb47-686"><a href="#cb47-686" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">340907</span>] <span class="ot">&lt;-</span> <span class="st">"Papiine alphaherpesvirus 2"</span></span>
<span id="cb47-687"><a href="#cb47-687" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194959</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus B"</span></span>
<span id="cb47-688"><a href="#cb47-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-689"><a href="#cb47-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-690"><a href="#cb47-690" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb47-691"><a href="#cb47-691" aria-hidden="true" tabindex="-1"></a>fp <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> </span>
<span id="cb47-692"><a href="#cb47-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb47-693"><a href="#cb47-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-694"><a href="#cb47-694" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid =</span> taxid_best) <span class="sc">%&gt;%</span></span>
<span id="cb47-695"><a href="#cb47-695" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-696"><a href="#cb47-696" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb47-697"><a href="#cb47-697" aria-hidden="true" tabindex="-1"></a>fp_major_tab <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb47-698"><a href="#cb47-698" aria-hidden="true" tabindex="-1"></a>fp_major_list <span class="ot">&lt;-</span> fp_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb47-699"><a href="#cb47-699" aria-hidden="true" tabindex="-1"></a>fp_major <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb47-700"><a href="#cb47-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-701"><a href="#cb47-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb47-702"><a href="#cb47-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb47-703"><a href="#cb47-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_major_list),</span>
<span id="cb47-704"><a href="#cb47-704" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb47-705"><a href="#cb47-705" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb47-706"><a href="#cb47-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-707"><a href="#cb47-707" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-708"><a href="#cb47-708" aria-hidden="true" tabindex="-1"></a>g_fp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb47-709"><a href="#cb47-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-710"><a href="#cb47-710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb47-711"><a href="#cb47-711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-712"><a href="#cb47-712" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-713"><a href="#cb47-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_viral, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb47-714"><a href="#cb47-714" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb47-715"><a href="#cb47-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-716"><a href="#cb47-716" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb47-717"><a href="#cb47-717" aria-hidden="true" tabindex="-1"></a>g_fp</span>
<span id="cb47-718"><a href="#cb47-718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-719"><a href="#cb47-719" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-720"><a href="#cb47-720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-723"><a href="#cb47-723" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-724"><a href="#cb47-724" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ptt-blast-hits</span></span>
<span id="cb47-725"><a href="#cb47-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-726"><a href="#cb47-726" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb47-727"><a href="#cb47-727" aria-hidden="true" tabindex="-1"></a>ref_taxid_rh2 <span class="ot">&lt;-</span> <span class="dv">946522</span></span>
<span id="cb47-728"><a href="#cb47-728" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb47-729"><a href="#cb47-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-730"><a href="#cb47-730" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb47-731"><a href="#cb47-731" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb47-732"><a href="#cb47-732" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb47-733"><a href="#cb47-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-734"><a href="#cb47-734" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb47-735"><a href="#cb47-735" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb47-736"><a href="#cb47-736" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb47-737"><a href="#cb47-737" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb47-738"><a href="#cb47-738" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3082113</span>, <span class="st">"Rangifer tarandus platyrhynchus"</span>,</span>
<span id="cb47-739"><a href="#cb47-739" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3119969</span>, <span class="st">"Bubalus kerabau"</span>,</span>
<span id="cb47-740"><a href="#cb47-740" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">177155</span>, <span class="st">"Streptopelia turtur"</span>,</span>
<span id="cb47-741"><a href="#cb47-741" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">187126</span>, <span class="st">"Nesoenas mayeri"</span>,</span>
<span id="cb47-742"><a href="#cb47-742" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">244366</span>, <span class="st">"Klebsiella variicola"</span>,</span>
<span id="cb47-743"><a href="#cb47-743" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb47-744"><a href="#cb47-744" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> tax_names_new <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> staxid <span class="sc">%in%</span> tax_names<span class="sc">$</span>staxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-745"><a href="#cb47-745" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(tax_names) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(staxid)</span>
<span id="cb47-746"><a href="#cb47-746" aria-hidden="true" tabindex="-1"></a>ref_name_rh2 <span class="ot">&lt;-</span> tax_names <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> ref_taxid_rh2) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb47-747"><a href="#cb47-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-748"><a href="#cb47-748" aria-hidden="true" tabindex="-1"></a><span class="co"># Get major matches</span></span>
<span id="cb47-749"><a href="#cb47-749" aria-hidden="true" tabindex="-1"></a>mrg_staxid <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid_best <span class="sc">==</span> ref_taxid_rh2) <span class="sc">%&gt;%</span></span>
<span id="cb47-750"><a href="#cb47-750" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(highscore, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>())</span>
<span id="cb47-751"><a href="#cb47-751" aria-hidden="true" tabindex="-1"></a>fp_staxid <span class="ot">&lt;-</span> mrg_staxid <span class="sc">%&gt;%</span></span>
<span id="cb47-752"><a href="#cb47-752" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-753"><a href="#cb47-753" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb47-754"><a href="#cb47-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb47-755"><a href="#cb47-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid_best=</span>staxid), <span class="at">by=</span><span class="st">"taxid_best"</span>)</span>
<span id="cb47-756"><a href="#cb47-756" aria-hidden="true" tabindex="-1"></a>fp_staxid_count <span class="ot">&lt;-</span> fp_staxid <span class="sc">%&gt;%</span></span>
<span id="cb47-757"><a href="#cb47-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, </span>
<span id="cb47-758"><a href="#cb47-758" aria-hidden="true" tabindex="-1"></a>           taxid_best, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb47-759"><a href="#cb47-759" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span></span>
<span id="cb47-760"><a href="#cb47-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, taxid_best, name) <span class="sc">%&gt;%</span></span>
<span id="cb47-761"><a href="#cb47-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb47-762"><a href="#cb47-762" aria-hidden="true" tabindex="-1"></a>fp_staxid_count_major <span class="ot">&lt;-</span> fp_staxid_count <span class="sc">%&gt;%</span></span>
<span id="cb47-763"><a href="#cb47-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb47-764"><a href="#cb47-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb47-765"><a href="#cb47-765" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, </span>
<span id="cb47-766"><a href="#cb47-766" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb47-767"><a href="#cb47-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-768"><a href="#cb47-768" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-769"><a href="#cb47-769" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_staxid_count_major, <span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span> </span>
<span id="cb47-770"><a href="#cb47-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb47-771"><a href="#cb47-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(status_display<span class="sc">~</span>score_display, <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb47-772"><a href="#cb47-772" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb47-773"><a href="#cb47-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-774"><a href="#cb47-774" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb47-775"><a href="#cb47-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_rh2, <span class="st">" (taxid "</span>, ref_taxid_rh2, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb47-776"><a href="#cb47-776" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb47-777"><a href="#cb47-777" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb47-778"><a href="#cb47-778" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-779"><a href="#cb47-779" aria-hidden="true" tabindex="-1"></a>g</span>
<span id="cb47-780"><a href="#cb47-780" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-781"><a href="#cb47-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-782"><a href="#cb47-782" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: overall relative abundance</span></span>
<span id="cb47-783"><a href="#cb47-783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-786"><a href="#cb47-786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-787"><a href="#cb47-787" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb47-788"><a href="#cb47-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-789"><a href="#cb47-789" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb47-790"><a href="#cb47-790" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb47-791"><a href="#cb47-791" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, location, sample_group, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb47-792"><a href="#cb47-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-793"><a href="#cb47-793" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb47-794"><a href="#cb47-794" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb47-795"><a href="#cb47-795" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb47-796"><a href="#cb47-796" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb47-797"><a href="#cb47-797" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb47-798"><a href="#cb47-798" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-799"><a href="#cb47-799" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>))</span>
<span id="cb47-800"><a href="#cb47-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-801"><a href="#cb47-801" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb47-802"><a href="#cb47-802" aria-hidden="true" tabindex="-1"></a>read_counts_grp <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-803"><a href="#cb47-803" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb47-804"><a href="#cb47-804" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-805"><a href="#cb47-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span> <span class="st">"All samples"</span>)</span>
<span id="cb47-806"><a href="#cb47-806" aria-hidden="true" tabindex="-1"></a>read_counts_st <span class="ot">&lt;-</span> read_counts_grp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-807"><a href="#cb47-807" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb47-808"><a href="#cb47-808" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-809"><a href="#cb47-809" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb47-810"><a href="#cb47-810" aria-hidden="true" tabindex="-1"></a>read_counts_loc <span class="ot">&lt;-</span> read_counts_grp <span class="sc">%&gt;%</span></span>
<span id="cb47-811"><a href="#cb47-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location) <span class="sc">%&gt;%</span></span>
<span id="cb47-812"><a href="#cb47-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb47-813"><a href="#cb47-813" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-814"><a href="#cb47-814" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_group =</span> <span class="st">"All sample types"</span>)</span>
<span id="cb47-815"><a href="#cb47-815" aria-hidden="true" tabindex="-1"></a>read_counts_tot <span class="ot">&lt;-</span> read_counts_loc <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-816"><a href="#cb47-816" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb47-817"><a href="#cb47-817" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-818"><a href="#cb47-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">location =</span> <span class="st">"All locations"</span>)</span>
<span id="cb47-819"><a href="#cb47-819" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(read_counts_grp, read_counts_st,</span>
<span id="cb47-820"><a href="#cb47-820" aria-hidden="true" tabindex="-1"></a>                             read_counts_loc, read_counts_tot) <span class="sc">%&gt;%</span></span>
<span id="cb47-821"><a href="#cb47-821" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb47-822"><a href="#cb47-822" aria-hidden="true" tabindex="-1"></a>         <span class="at">location =</span> <span class="fu">factor</span>(location, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">levels</span>(libraries<span class="sc">$</span>location), <span class="st">"All locations"</span>)),</span>
<span id="cb47-823"><a href="#cb47-823" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample_group =</span> <span class="fu">factor</span>(sample_group, <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">levels</span>(libraries<span class="sc">$</span>sample_group), <span class="st">"All sample types"</span>)))</span>
<span id="cb47-824"><a href="#cb47-824" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-825"><a href="#cb47-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-826"><a href="#cb47-826" aria-hidden="true" tabindex="-1"></a>Applying a disjunctive cutoff at S=20 identifies 240 read pairs as human-viral. This gives an overall relative HV abundance of $8.86 \times 10^{-8}$. While very low across all sample types, HV RA was noticeably higher in influent and primary sludge than in sample types that had undergone more extensive processing (effluent and processed sludge):</span>
<span id="cb47-827"><a href="#cb47-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-830"><a href="#cb47-830" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-831"><a href="#cb47-831" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-ra</span></span>
<span id="cb47-832"><a href="#cb47-832" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb47-833"><a href="#cb47-833" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb47-834"><a href="#cb47-834" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>sample_group, <span class="at">color=</span>location)) <span class="sc">+</span></span>
<span id="cb47-835"><a href="#cb47-835" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_hv)) <span class="sc">+</span></span>
<span id="cb47-836"><a href="#cb47-836" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb47-837"><a href="#cb47-837" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">name=</span><span class="st">"Location"</span>, <span class="at">palette=</span><span class="st">"Dark2"</span>) <span class="sc">+</span> theme_kit</span>
<span id="cb47-838"><a href="#cb47-838" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb47-839"><a href="#cb47-839" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-840"><a href="#cb47-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-841"><a href="#cb47-841" aria-hidden="true" tabindex="-1"></a>This is by far the lowest HV relative abundance I've seen across any of the datasets I've analyzed:</span>
<span id="cb47-842"><a href="#cb47-842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-845"><a href="#cb47-845" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-846"><a href="#cb47-846" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-hv-past</span></span>
<span id="cb47-847"><a href="#cb47-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-848"><a href="#cb47-848" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate past RA values</span></span>
<span id="cb47-849"><a href="#cb47-849" aria-hidden="true" tabindex="-1"></a>ra_past <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb47-850"><a href="#cb47-850" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-851"><a href="#cb47-851" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-852"><a href="#cb47-852" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-853"><a href="#cb47-853" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-854"><a href="#cb47-854" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-855"><a href="#cb47-855" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb47-856"><a href="#cb47-856" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-857"><a href="#cb47-857" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb47-858"><a href="#cb47-858" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-859"><a href="#cb47-859" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-860"><a href="#cb47-860" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-861"><a href="#cb47-861" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb47-862"><a href="#cb47-862" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span id="cb47-863"><a href="#cb47-863" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-864"><a href="#cb47-864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-865"><a href="#cb47-865" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate new RA values</span></span>
<span id="cb47-866"><a href="#cb47-866" aria-hidden="true" tabindex="-1"></a>ra_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb47-867"><a href="#cb47-867" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Bengtsson-Palme"</span>, <span class="fl">8.86e-8</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>)</span>
<span id="cb47-868"><a href="#cb47-868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-869"><a href="#cb47-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-870"><a href="#cb47-870" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-871"><a href="#cb47-871" aria-hidden="true" tabindex="-1"></a>scale_color_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb47-872"><a href="#cb47-872" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb47-873"><a href="#cb47-873" aria-hidden="true" tabindex="-1"></a>ra_comp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ra_past, ra_new) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">fct_inorder</span>(dataset))</span>
<span id="cb47-874"><a href="#cb47-874" aria-hidden="true" tabindex="-1"></a>g_ra_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ra_comp, <span class="fu">aes</span>(<span class="at">y=</span>dataset, <span class="at">x=</span>ra, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb47-875"><a href="#cb47-875" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb47-876"><a href="#cb47-876" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb47-877"><a href="#cb47-877" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb47-878"><a href="#cb47-878" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb47-879"><a href="#cb47-879" aria-hidden="true" tabindex="-1"></a>g_ra_comp</span>
<span id="cb47-880"><a href="#cb47-880" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-881"><a href="#cb47-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-882"><a href="#cb47-882" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: taxonomy and composition</span></span>
<span id="cb47-883"><a href="#cb47-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-884"><a href="#cb47-884" aria-hidden="true" tabindex="-1"></a>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 15 samples, 4 from influent and 11 from primary sludge, met this criterion.</span>
<span id="cb47-885"><a href="#cb47-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-886"><a href="#cb47-886" aria-hidden="true" tabindex="-1"></a>At the family level, most samples across all locations were dominated by *Poxviridae*, *Adenoviridae*, *Picornaviridae,* and *Herpesviridae*, with several other families making an appearance</span>
<span id="cb47-887"><a href="#cb47-887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-890"><a href="#cb47-890" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-891"><a href="#cb47-891" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb47-892"><a href="#cb47-892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-893"><a href="#cb47-893" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb47-894"><a href="#cb47-894" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb47-895"><a href="#cb47-895" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb47-896"><a href="#cb47-896" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb47-897"><a href="#cb47-897" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">585893</span>] <span class="ot">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span id="cb47-898"><a href="#cb47-898" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333922</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span id="cb47-899"><a href="#cb47-899" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334207</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span id="cb47-900"><a href="#cb47-900" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">369960</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span id="cb47-901"><a href="#cb47-901" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333924</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span id="cb47-902"><a href="#cb47-902" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687329</span>] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb47-903"><a href="#cb47-903" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">325455</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span id="cb47-904"><a href="#cb47-904" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333750</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span id="cb47-905"><a href="#cb47-905" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb47-906"><a href="#cb47-906" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334202</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span id="cb47-907"><a href="#cb47-907" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">197911</span>] <span class="ot">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span id="cb47-908"><a href="#cb47-908" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">186938</span>] <span class="ot">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span id="cb47-909"><a href="#cb47-909" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333926</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span id="cb47-910"><a href="#cb47-910" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337051</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span id="cb47-911"><a href="#cb47-911" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337043</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span id="cb47-912"><a href="#cb47-912" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb47-913"><a href="#cb47-913" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334204</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span id="cb47-914"><a href="#cb47-914" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334208</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span id="cb47-915"><a href="#cb47-915" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333928</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span id="cb47-916"><a href="#cb47-916" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337039</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span id="cb47-917"><a href="#cb47-917" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333929</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span id="cb47-918"><a href="#cb47-918" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337042</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span id="cb47-919"><a href="#cb47-919" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334203</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span id="cb47-920"><a href="#cb47-920" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333757</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span id="cb47-921"><a href="#cb47-921" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337050</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span id="cb47-922"><a href="#cb47-922" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333767</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span id="cb47-923"><a href="#cb47-923" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333754</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span id="cb47-924"><a href="#cb47-924" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687363</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span id="cb47-925"><a href="#cb47-925" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687342</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span id="cb47-926"><a href="#cb47-926" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687359</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span id="cb47-927"><a href="#cb47-927" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194441</span>] <span class="ot">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span id="cb47-928"><a href="#cb47-928" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334209</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span id="cb47-929"><a href="#cb47-929" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194965</span>] <span class="ot">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span id="cb47-930"><a href="#cb47-930" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-931"><a href="#cb47-931" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-932"><a href="#cb47-932" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb47-933"><a href="#cb47-933" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb47-934"><a href="#cb47-934" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb47-935"><a href="#cb47-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-936"><a href="#cb47-936" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb47-937"><a href="#cb47-937" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb47-938"><a href="#cb47-938" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb47-939"><a href="#cb47-939" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb47-940"><a href="#cb47-940" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb47-941"><a href="#cb47-941" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb47-942"><a href="#cb47-942" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb47-943"><a href="#cb47-943" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb47-944"><a href="#cb47-944" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb47-945"><a href="#cb47-945" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb47-946"><a href="#cb47-946" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb47-947"><a href="#cb47-947" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb47-948"><a href="#cb47-948" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb47-949"><a href="#cb47-949" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb47-950"><a href="#cb47-950" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb47-951"><a href="#cb47-951" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-952"><a href="#cb47-952" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb47-953"><a href="#cb47-953" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb47-954"><a href="#cb47-954" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb47-955"><a href="#cb47-955" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb47-956"><a href="#cb47-956" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb47-957"><a href="#cb47-957" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb47-958"><a href="#cb47-958" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb47-959"><a href="#cb47-959" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb47-960"><a href="#cb47-960" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb47-961"><a href="#cb47-961" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb47-962"><a href="#cb47-962" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb47-963"><a href="#cb47-963" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb47-964"><a href="#cb47-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb47-965"><a href="#cb47-965" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb47-966"><a href="#cb47-966" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb47-967"><a href="#cb47-967" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb47-968"><a href="#cb47-968" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb47-969"><a href="#cb47-969" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-970"><a href="#cb47-970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-973"><a href="#cb47-973" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-974"><a href="#cb47-974" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb47-975"><a href="#cb47-975" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-976"><a href="#cb47-976" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb47-977"><a href="#cb47-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-978"><a href="#cb47-978" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.08</span></span>
<span id="cb47-979"><a href="#cb47-979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-980"><a href="#cb47-980" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb47-981"><a href="#cb47-981" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb47-982"><a href="#cb47-982" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-983"><a href="#cb47-983" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-984"><a href="#cb47-984" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-985"><a href="#cb47-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-986"><a href="#cb47-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-987"><a href="#cb47-987" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-988"><a href="#cb47-988" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-989"><a href="#cb47-989" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-990"><a href="#cb47-990" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb47-991"><a href="#cb47-991" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-992"><a href="#cb47-992" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-993"><a href="#cb47-993" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-994"><a href="#cb47-994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb47-995"><a href="#cb47-995" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-996"><a href="#cb47-996" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-997"><a href="#cb47-997" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-998"><a href="#cb47-998" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-999"><a href="#cb47-999" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1000"><a href="#cb47-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1001"><a href="#cb47-1001" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1002"><a href="#cb47-1002" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1003"><a href="#cb47-1003" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1004"><a href="#cb47-1004" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1005"><a href="#cb47-1005" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1006"><a href="#cb47-1006" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1007"><a href="#cb47-1007" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb47-1008"><a href="#cb47-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1009"><a href="#cb47-1009" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb47-1010"><a href="#cb47-1010" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1011"><a href="#cb47-1011" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb47-1012"><a href="#cb47-1012" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1013"><a href="#cb47-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1014"><a href="#cb47-1014" aria-hidden="true" tabindex="-1"></a>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</span>
<span id="cb47-1015"><a href="#cb47-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1018"><a href="#cb47-1018" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1019"><a href="#cb47-1019" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-pox</span></span>
<span id="cb47-1020"><a href="#cb47-1020" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-1021"><a href="#cb47-1021" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb47-1022"><a href="#cb47-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1023"><a href="#cb47-1023" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb47-1024"><a href="#cb47-1024" aria-hidden="true" tabindex="-1"></a>taxid_pox <span class="ot">&lt;-</span> <span class="dv">10240</span></span>
<span id="cb47-1025"><a href="#cb47-1025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1026"><a href="#cb47-1026" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of poxviridae reads</span></span>
<span id="cb47-1027"><a href="#cb47-1027" aria-hidden="true" tabindex="-1"></a>pox_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox) <span class="sc">%&gt;%</span></span>
<span id="cb47-1028"><a href="#cb47-1028" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1029"><a href="#cb47-1029" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb47-1030"><a href="#cb47-1030" aria-hidden="true" tabindex="-1"></a>pox_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb47-1031"><a href="#cb47-1031" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_pox, sample <span class="sc">%in%</span> pox_samples) <span class="sc">%&gt;%</span></span>
<span id="cb47-1032"><a href="#cb47-1032" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb47-1033"><a href="#cb47-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1034"><a href="#cb47-1034" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each poxviridae species</span></span>
<span id="cb47-1035"><a href="#cb47-1035" aria-hidden="true" tabindex="-1"></a>pox_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb47-1036"><a href="#cb47-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> pox_ids) <span class="sc">%&gt;%</span></span>
<span id="cb47-1037"><a href="#cb47-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-1038"><a href="#cb47-1038" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1039"><a href="#cb47-1039" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1040"><a href="#cb47-1040" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_pox =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-1041"><a href="#cb47-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1042"><a href="#cb47-1042" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-1043"><a href="#cb47-1043" aria-hidden="true" tabindex="-1"></a>pox_species_major_tab <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1044"><a href="#cb47-1044" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">==</span> <span class="fu">max</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1045"><a href="#cb47-1045" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1046"><a href="#cb47-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_pox)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1047"><a href="#cb47-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pox <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb47-1048"><a href="#cb47-1048" aria-hidden="true" tabindex="-1"></a>pox_species_counts_major <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-1049"><a href="#cb47-1049" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> pox_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb47-1050"><a href="#cb47-1050" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1051"><a href="#cb47-1051" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-1052"><a href="#cb47-1052" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_pox =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb47-1053"><a href="#cb47-1053" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_pox =</span> <span class="fu">sum</span>(p_reads_pox), </span>
<span id="cb47-1054"><a href="#cb47-1054" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1055"><a href="#cb47-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-1056"><a href="#cb47-1056" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(pox_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-1057"><a href="#cb47-1057" aria-hidden="true" tabindex="-1"></a>pox_species_counts_display <span class="ot">&lt;-</span> pox_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-1058"><a href="#cb47-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_pox, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1059"><a href="#cb47-1059" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1060"><a href="#cb47-1060" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1061"><a href="#cb47-1061" aria-hidden="true" tabindex="-1"></a>g_pox_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1062"><a href="#cb47-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>pox_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1063"><a href="#cb47-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Poxviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1064"><a href="#cb47-1064" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1065"><a href="#cb47-1065" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1066"><a href="#cb47-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb47-1067"><a href="#cb47-1067" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Poxviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1068"><a href="#cb47-1068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-1069"><a href="#cb47-1069" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1070"><a href="#cb47-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1071"><a href="#cb47-1071" aria-hidden="true" tabindex="-1"></a>g_pox_species</span>
<span id="cb47-1072"><a href="#cb47-1072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1073"><a href="#cb47-1073" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb47-1074"><a href="#cb47-1074" aria-hidden="true" tabindex="-1"></a>pox_species_collate <span class="ot">&lt;-</span> pox_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1075"><a href="#cb47-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_pox), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1076"><a href="#cb47-1076" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb47-1077"><a href="#cb47-1077" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1078"><a href="#cb47-1078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1081"><a href="#cb47-1081" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1082"><a href="#cb47-1082" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-adeno</span></span>
<span id="cb47-1083"><a href="#cb47-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-1084"><a href="#cb47-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb47-1085"><a href="#cb47-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1086"><a href="#cb47-1086" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb47-1087"><a href="#cb47-1087" aria-hidden="true" tabindex="-1"></a>taxid_adeno <span class="ot">&lt;-</span> <span class="dv">10508</span></span>
<span id="cb47-1088"><a href="#cb47-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1089"><a href="#cb47-1089" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of adenoviridae reads</span></span>
<span id="cb47-1090"><a href="#cb47-1090" aria-hidden="true" tabindex="-1"></a>adeno_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno) <span class="sc">%&gt;%</span></span>
<span id="cb47-1091"><a href="#cb47-1091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1092"><a href="#cb47-1092" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb47-1093"><a href="#cb47-1093" aria-hidden="true" tabindex="-1"></a>adeno_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb47-1094"><a href="#cb47-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno, sample <span class="sc">%in%</span> adeno_samples) <span class="sc">%&gt;%</span></span>
<span id="cb47-1095"><a href="#cb47-1095" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb47-1096"><a href="#cb47-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1097"><a href="#cb47-1097" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each adenoviridae species</span></span>
<span id="cb47-1098"><a href="#cb47-1098" aria-hidden="true" tabindex="-1"></a>adeno_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb47-1099"><a href="#cb47-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> adeno_ids) <span class="sc">%&gt;%</span></span>
<span id="cb47-1100"><a href="#cb47-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-1101"><a href="#cb47-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1102"><a href="#cb47-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1103"><a href="#cb47-1103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_adeno =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-1104"><a href="#cb47-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1105"><a href="#cb47-1105" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-1106"><a href="#cb47-1106" aria-hidden="true" tabindex="-1"></a>adeno_species_major_tab <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1107"><a href="#cb47-1107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">==</span> <span class="fu">max</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1108"><a href="#cb47-1108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1109"><a href="#cb47-1109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1110"><a href="#cb47-1110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb47-1111"><a href="#cb47-1111" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_major <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-1112"><a href="#cb47-1112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> adeno_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb47-1113"><a href="#cb47-1113" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1114"><a href="#cb47-1114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-1115"><a href="#cb47-1115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_adeno =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb47-1116"><a href="#cb47-1116" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_adeno =</span> <span class="fu">sum</span>(p_reads_adeno), </span>
<span id="cb47-1117"><a href="#cb47-1117" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1118"><a href="#cb47-1118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-1119"><a href="#cb47-1119" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(adeno_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-1120"><a href="#cb47-1120" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_display <span class="ot">&lt;-</span> adeno_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-1121"><a href="#cb47-1121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_adeno, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1122"><a href="#cb47-1122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1123"><a href="#cb47-1123" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1124"><a href="#cb47-1124" aria-hidden="true" tabindex="-1"></a>g_adeno_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1125"><a href="#cb47-1125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>adeno_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1126"><a href="#cb47-1126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% adenoviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1127"><a href="#cb47-1127" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1128"><a href="#cb47-1128" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1129"><a href="#cb47-1129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb47-1130"><a href="#cb47-1130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of adenoviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1131"><a href="#cb47-1131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-1132"><a href="#cb47-1132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1133"><a href="#cb47-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1134"><a href="#cb47-1134" aria-hidden="true" tabindex="-1"></a>g_adeno_species</span>
<span id="cb47-1135"><a href="#cb47-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1136"><a href="#cb47-1136" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb47-1137"><a href="#cb47-1137" aria-hidden="true" tabindex="-1"></a>adeno_species_collate <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1138"><a href="#cb47-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_adeno), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1139"><a href="#cb47-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb47-1140"><a href="#cb47-1140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1141"><a href="#cb47-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1144"><a href="#cb47-1144" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1145"><a href="#cb47-1145" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-picorna</span></span>
<span id="cb47-1146"><a href="#cb47-1146" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-1147"><a href="#cb47-1147" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb47-1148"><a href="#cb47-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1149"><a href="#cb47-1149" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb47-1150"><a href="#cb47-1150" aria-hidden="true" tabindex="-1"></a>taxid_picorna <span class="ot">&lt;-</span> <span class="dv">12058</span></span>
<span id="cb47-1151"><a href="#cb47-1151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1152"><a href="#cb47-1152" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of picornaviridae reads</span></span>
<span id="cb47-1153"><a href="#cb47-1153" aria-hidden="true" tabindex="-1"></a>picorna_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_picorna) <span class="sc">%&gt;%</span></span>
<span id="cb47-1154"><a href="#cb47-1154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1155"><a href="#cb47-1155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb47-1156"><a href="#cb47-1156" aria-hidden="true" tabindex="-1"></a>picorna_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb47-1157"><a href="#cb47-1157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_picorna, sample <span class="sc">%in%</span> picorna_samples) <span class="sc">%&gt;%</span></span>
<span id="cb47-1158"><a href="#cb47-1158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb47-1159"><a href="#cb47-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1160"><a href="#cb47-1160" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each picornaviridae species</span></span>
<span id="cb47-1161"><a href="#cb47-1161" aria-hidden="true" tabindex="-1"></a>picorna_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb47-1162"><a href="#cb47-1162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> picorna_ids) <span class="sc">%&gt;%</span></span>
<span id="cb47-1163"><a href="#cb47-1163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-1164"><a href="#cb47-1164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1165"><a href="#cb47-1165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1166"><a href="#cb47-1166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_picorna =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-1167"><a href="#cb47-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1168"><a href="#cb47-1168" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-1169"><a href="#cb47-1169" aria-hidden="true" tabindex="-1"></a>picorna_species_major_tab <span class="ot">&lt;-</span> picorna_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1170"><a href="#cb47-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_picorna <span class="sc">==</span> <span class="fu">max</span>(p_reads_picorna)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1171"><a href="#cb47-1171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1172"><a href="#cb47-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_picorna)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1173"><a href="#cb47-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_picorna <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb47-1174"><a href="#cb47-1174" aria-hidden="true" tabindex="-1"></a>picorna_species_counts_major <span class="ot">&lt;-</span> picorna_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-1175"><a href="#cb47-1175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> picorna_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb47-1176"><a href="#cb47-1176" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1177"><a href="#cb47-1177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-1178"><a href="#cb47-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_picorna =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb47-1179"><a href="#cb47-1179" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_picorna =</span> <span class="fu">sum</span>(p_reads_picorna), </span>
<span id="cb47-1180"><a href="#cb47-1180" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1181"><a href="#cb47-1181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-1182"><a href="#cb47-1182" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(picorna_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-1183"><a href="#cb47-1183" aria-hidden="true" tabindex="-1"></a>picorna_species_counts_display <span class="ot">&lt;-</span> picorna_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-1184"><a href="#cb47-1184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_picorna, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1185"><a href="#cb47-1185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1186"><a href="#cb47-1186" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1187"><a href="#cb47-1187" aria-hidden="true" tabindex="-1"></a>g_picorna_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1188"><a href="#cb47-1188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>picorna_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1189"><a href="#cb47-1189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Picornaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1190"><a href="#cb47-1190" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1191"><a href="#cb47-1191" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1192"><a href="#cb47-1192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb47-1193"><a href="#cb47-1193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Picornaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1194"><a href="#cb47-1194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-1195"><a href="#cb47-1195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1196"><a href="#cb47-1196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1197"><a href="#cb47-1197" aria-hidden="true" tabindex="-1"></a>g_picorna_species</span>
<span id="cb47-1198"><a href="#cb47-1198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1199"><a href="#cb47-1199" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb47-1200"><a href="#cb47-1200" aria-hidden="true" tabindex="-1"></a>picorna_species_collate <span class="ot">&lt;-</span> picorna_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1201"><a href="#cb47-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_picorna), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1202"><a href="#cb47-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb47-1203"><a href="#cb47-1203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1204"><a href="#cb47-1204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1207"><a href="#cb47-1207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1208"><a href="#cb47-1208" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-herpes</span></span>
<span id="cb47-1209"><a href="#cb47-1209" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-1210"><a href="#cb47-1210" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb47-1211"><a href="#cb47-1211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1212"><a href="#cb47-1212" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb47-1213"><a href="#cb47-1213" aria-hidden="true" tabindex="-1"></a>taxid_herpes <span class="ot">&lt;-</span> <span class="dv">10292</span></span>
<span id="cb47-1214"><a href="#cb47-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1215"><a href="#cb47-1215" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of herpesviridae reads</span></span>
<span id="cb47-1216"><a href="#cb47-1216" aria-hidden="true" tabindex="-1"></a>herpes_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes) <span class="sc">%&gt;%</span></span>
<span id="cb47-1217"><a href="#cb47-1217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1218"><a href="#cb47-1218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb47-1219"><a href="#cb47-1219" aria-hidden="true" tabindex="-1"></a>herpes_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb47-1220"><a href="#cb47-1220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes, sample <span class="sc">%in%</span> herpes_samples) <span class="sc">%&gt;%</span></span>
<span id="cb47-1221"><a href="#cb47-1221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb47-1222"><a href="#cb47-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1223"><a href="#cb47-1223" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae species</span></span>
<span id="cb47-1224"><a href="#cb47-1224" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb47-1225"><a href="#cb47-1225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb47-1226"><a href="#cb47-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-1227"><a href="#cb47-1227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1228"><a href="#cb47-1228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1229"><a href="#cb47-1229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-1230"><a href="#cb47-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1231"><a href="#cb47-1231" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-1232"><a href="#cb47-1232" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1233"><a href="#cb47-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1234"><a href="#cb47-1234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1235"><a href="#cb47-1235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1236"><a href="#cb47-1236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb47-1237"><a href="#cb47-1237" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-1238"><a href="#cb47-1238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb47-1239"><a href="#cb47-1239" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1240"><a href="#cb47-1240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-1241"><a href="#cb47-1241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb47-1242"><a href="#cb47-1242" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb47-1243"><a href="#cb47-1243" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1244"><a href="#cb47-1244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-1245"><a href="#cb47-1245" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-1246"><a href="#cb47-1246" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-1247"><a href="#cb47-1247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1248"><a href="#cb47-1248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1249"><a href="#cb47-1249" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1250"><a href="#cb47-1250" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1251"><a href="#cb47-1251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1252"><a href="#cb47-1252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1253"><a href="#cb47-1253" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1254"><a href="#cb47-1254" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1255"><a href="#cb47-1255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb47-1256"><a href="#cb47-1256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1257"><a href="#cb47-1257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-1258"><a href="#cb47-1258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1259"><a href="#cb47-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1260"><a href="#cb47-1260" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb47-1261"><a href="#cb47-1261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1262"><a href="#cb47-1262" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb47-1263"><a href="#cb47-1263" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1264"><a href="#cb47-1264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1265"><a href="#cb47-1265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb47-1266"><a href="#cb47-1266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1267"><a href="#cb47-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1268"><a href="#cb47-1268" aria-hidden="true" tabindex="-1"></a>In the latter case, some subset of human alphaherpesvirus 1 reads will be arising from strain RH2, which we know to be dominated by false positive hits. Digging deeper, we see that these are essentially all of the alphaherpesvirus 1 reads. Hence, these herpesvirus read counts need to be treated with significant skepticism.</span>
<span id="cb47-1269"><a href="#cb47-1269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1272"><a href="#cb47-1272" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1273"><a href="#cb47-1273" aria-hidden="true" tabindex="-1"></a>mrg_hv_named</span>
<span id="cb47-1274"><a href="#cb47-1274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1275"><a href="#cb47-1275" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae taxon</span></span>
<span id="cb47-1276"><a href="#cb47-1276" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> mrg_hv_named <span class="sc">%&gt;%</span></span>
<span id="cb47-1277"><a href="#cb47-1277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb47-1278"><a href="#cb47-1278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb47-1279"><a href="#cb47-1279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1280"><a href="#cb47-1280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1281"><a href="#cb47-1281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb47-1282"><a href="#cb47-1282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1283"><a href="#cb47-1283" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb47-1284"><a href="#cb47-1284" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1285"><a href="#cb47-1285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1286"><a href="#cb47-1286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1287"><a href="#cb47-1287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1288"><a href="#cb47-1288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb47-1289"><a href="#cb47-1289" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb47-1290"><a href="#cb47-1290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb47-1291"><a href="#cb47-1291" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1292"><a href="#cb47-1292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb47-1293"><a href="#cb47-1293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb47-1294"><a href="#cb47-1294" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb47-1295"><a href="#cb47-1295" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1296"><a href="#cb47-1296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb47-1297"><a href="#cb47-1297" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb47-1298"><a href="#cb47-1298" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb47-1299"><a href="#cb47-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb47-1300"><a href="#cb47-1300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1301"><a href="#cb47-1301" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1302"><a href="#cb47-1302" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb47-1303"><a href="#cb47-1303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb47-1304"><a href="#cb47-1304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb47-1305"><a href="#cb47-1305" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb47-1306"><a href="#cb47-1306" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb47-1307"><a href="#cb47-1307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb47-1308"><a href="#cb47-1308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb47-1309"><a href="#cb47-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb47-1310"><a href="#cb47-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb47-1311"><a href="#cb47-1311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1312"><a href="#cb47-1312" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb47-1313"><a href="#cb47-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1314"><a href="#cb47-1314" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb47-1315"><a href="#cb47-1315" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1316"><a href="#cb47-1316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1317"><a href="#cb47-1317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb47-1318"><a href="#cb47-1318" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1319"><a href="#cb47-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1320"><a href="#cb47-1320" aria-hidden="true" tabindex="-1"></a>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</span>
<span id="cb47-1321"><a href="#cb47-1321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1324"><a href="#cb47-1324" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb47-1325"><a href="#cb47-1325" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb47-1326"><a href="#cb47-1326" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-genera</span></span>
<span id="cb47-1327"><a href="#cb47-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1328"><a href="#cb47-1328" aria-hidden="true" tabindex="-1"></a><span class="co"># Define reference genera</span></span>
<span id="cb47-1329"><a href="#cb47-1329" aria-hidden="true" tabindex="-1"></a>path_genera_rna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span>)</span>
<span id="cb47-1330"><a href="#cb47-1330" aria-hidden="true" tabindex="-1"></a>path_genera_dna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span id="cb47-1331"><a href="#cb47-1331" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span>)</span>
<span id="cb47-1332"><a href="#cb47-1332" aria-hidden="true" tabindex="-1"></a>path_genera <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">tibble</span>(<span class="at">name=</span>path_genera_rna, <span class="at">genome_type=</span><span class="st">"RNA genome"</span>),</span>
<span id="cb47-1333"><a href="#cb47-1333" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">tibble</span>(<span class="at">name=</span>path_genera_dna, <span class="at">genome_type=</span><span class="st">"DNA genome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1334"><a href="#cb47-1334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb47-1335"><a href="#cb47-1335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1336"><a href="#cb47-1336" aria-hidden="true" tabindex="-1"></a><span class="co"># Count in each sample</span></span>
<span id="cb47-1337"><a href="#cb47-1337" aria-hidden="true" tabindex="-1"></a>mrg_hv_named_all <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb47-1338"><a href="#cb47-1338" aria-hidden="true" tabindex="-1"></a>hv_reads_genus_all <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named_all, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb47-1339"><a href="#cb47-1339" aria-hidden="true" tabindex="-1"></a>n_path_genera <span class="ot">&lt;-</span> hv_reads_genus_all <span class="sc">%&gt;%</span> </span>
<span id="cb47-1340"><a href="#cb47-1340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, location, sample_group, name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1341"><a href="#cb47-1341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-1342"><a href="#cb47-1342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(path_genera, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1343"><a href="#cb47-1343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"sample_group"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-1344"><a href="#cb47-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb47-1345"><a href="#cb47-1345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1346"><a href="#cb47-1346" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot out and back to add zero lines</span></span>
<span id="cb47-1347"><a href="#cb47-1347" aria-hidden="true" tabindex="-1"></a>n_path_genera_out <span class="ot">&lt;-</span> n_path_genera <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, name, n_reads_viral) <span class="sc">%&gt;%</span></span>
<span id="cb47-1348"><a href="#cb47-1348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"name"</span>, <span class="at">values_from=</span><span class="st">"n_reads_viral"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1349"><a href="#cb47-1349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to=</span><span class="st">"name"</span>, <span class="at">values_to=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1350"><a href="#cb47-1350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1351"><a href="#cb47-1351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(path_genera, <span class="at">by=</span><span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1352"><a href="#cb47-1352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb47-1353"><a href="#cb47-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1354"><a href="#cb47-1354" aria-hidden="true" tabindex="-1"></a><span class="do">## Aggregate across dates</span></span>
<span id="cb47-1355"><a href="#cb47-1355" aria-hidden="true" tabindex="-1"></a>n_path_genera_stype <span class="ot">&lt;-</span> n_path_genera_out <span class="sc">%&gt;%</span> </span>
<span id="cb47-1356"><a href="#cb47-1356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid, genome_type, sample_group) <span class="sc">%&gt;%</span></span>
<span id="cb47-1357"><a href="#cb47-1357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb47-1358"><a href="#cb47-1358" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_viral =</span> <span class="fu">sum</span>(n_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-1359"><a href="#cb47-1359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="st">"All samples"</span>, <span class="at">location=</span><span class="st">"All locations"</span>,</span>
<span id="cb47-1360"><a href="#cb47-1360" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw,</span>
<span id="cb47-1361"><a href="#cb47-1361" aria-hidden="true" tabindex="-1"></a>         <span class="at">na_type =</span> <span class="st">"DNA"</span>)</span>
<span id="cb47-1362"><a href="#cb47-1362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1363"><a href="#cb47-1363" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb47-1364"><a href="#cb47-1364" aria-hidden="true" tabindex="-1"></a>g_path_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_path_genera_stype,</span>
<span id="cb47-1365"><a href="#cb47-1365" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y=</span>name, <span class="at">x=</span>p_reads_viral, <span class="at">color=</span>sample_group)) <span class="sc">+</span></span>
<span id="cb47-1366"><a href="#cb47-1366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb47-1367"><a href="#cb47-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb47-1368"><a href="#cb47-1368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_st</span>() <span class="sc">+</span></span>
<span id="cb47-1369"><a href="#cb47-1369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genome_type<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb47-1370"><a href="#cb47-1370" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb47-1371"><a href="#cb47-1371" aria-hidden="true" tabindex="-1"></a>g_path_genera</span>
<span id="cb47-1372"><a href="#cb47-1372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb47-1373"><a href="#cb47-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1374"><a href="#cb47-1374" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb47-1375"><a href="#cb47-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-1376"><a href="#cb47-1376" aria-hidden="true" tabindex="-1"></a>This dataset is a cautionary tale about the dangers of looking for viruses in samples that were processed with something else in mind. The sample preparation methods used here would be expected to select against viruses, and it shows, with by far the lowest overall human-viral RA of any dataset I've investigated so far. This very low abundance had knock-on effects, including worse pipeline performance as the low rate of false positives exceeded the even lower rate of true ones. Even so, there were some interesting results, such as the higher level of human viruses in influent and primary sludge compared to more processed sample types.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>