<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-02-13">
<title>Will’s Public NAO Notebook - Workflow analysis of Crits-Christoph et al.&nbsp;(2021), part 3</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Crits-Christoph et al.&nbsp;(2021), part 3</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Fixing the virus pipeline.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 13, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>In my <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-02-08_crits-christoph-2.html">last entry</a>, I documented my first attempt at analyzing the human-infecting-virus content of sequence data from <a href="https://doi.org/10.1128%2FmBio.02703-20">Crits-Christoph et al.&nbsp;(2021)</a>. In this entry, I’ll attempt to address the issues that arose in that first analysis and analyze the resulting updated data.</p>
<p>As a reminder, I previously found that the analysis pipeline I developed for our BMC data led to numerous high-scoring false-positives when run on Crits-Christoph data:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Data input paths</span></span>
<span><span class="va">data_dir_old</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-02-04_crits-christoph-2/"</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"cc-libraries.txt"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_taxids_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"human-virus-taxids-all.txt"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"nodes.dmp.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"cat-cc-unenriched.blast.gz"</span><span class="op">)</span></span>
<span><span class="va">bg_causes</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"cc-bad-notr.tsv"</span><span class="op">)</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">header_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"human-viral-headers.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import data</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>enrichment <span class="op">=</span> <span class="fu">str_to_title</span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_taxids</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_taxids_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes</span> <span class="op">&lt;-</span> <span class="fu">read_delim</span><span class="op">(</span><span class="va">tax_nodes_path</span>, delim <span class="op">=</span> <span class="st">"\t|\t"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">X1</span><span class="op">:</span><span class="va">X3</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>child_taxid <span class="op">=</span> <span class="va">X1</span>, parent_taxid <span class="op">=</span> <span class="va">X2</span>, rank <span class="op">=</span> <span class="va">X3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define taxid search function</span></span>
<span><span class="va">expand_taxids</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxids_in</span>, <span class="va">nodes</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="va">taxids_in</span></span>
<span>  <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">taxids_out</span>, <span class="va">taxids_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>    <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, </span>
<span>                         <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">taxids_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">v_taxids</span> <span class="op">&lt;-</span> <span class="fu">expand_taxids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10239</span>, <span class="va">hv_taxids</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span>, <span class="va">tax_nodes</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import BLAST results</span></span>
<span><span class="va">blast_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="va">blast_cols</span>,</span>
<span>                          col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_results</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for the best hit for each sequence and taxid</span></span>
<span><span class="va">blast_results_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_viral</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each sequence</span></span>
<span><span class="va">blast_results_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_best</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter by rank</span></span>
<span><span class="va">blast_results_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">qseqid</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span>, <span class="va">viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, </span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Process Bowtie/Kraken data</span></span>
<span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare BLAST results to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">seq_id</span><span class="op">)</span></span>
<span><span class="va">blast_results_assign</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge with unenriched read data</span></span>
<span><span class="va">mrg_unenriched</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_out</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"seq_id"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unenriched_plot</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unenriched_debug</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unenriched_plot_2</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_debug</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">bg_causes</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"Appears real"</span>, <span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"No match"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">viral_status</span><span class="op">)</span>, <span class="va">viral_status</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_mrg_v2</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_v2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>In total, 992 non-viral read pairs and 83 read pairs of unclear status achieved high length-adjusted alignment scores (<span class="math inline">\(\geq 20\)</span>) when mapped to viral Genbank with Bowtie2. For now, I will focus on the non-viral read pairs, and return to the unclear read pairs after these are in better shape.</p>
<section id="pass-1-excluding-transgenic-and-bacterial-sequences" class="level1"><h1>Pass 1: excluding transgenic and bacterial sequences</h1>
<p>The first major change I made to the pipeline was to curate the reference database for the Bowtie2 alignment to remove transgenic and other inappropriate viral sequences. To do this, I subset the collated FASTA file of viral genomes with seqtk to remove any sequence with “transgenic”, “mutant”, “recombinant”, “unverified” or “draft” in its sequence ID. This removed a total of 493 genomes from the reference DB, leaving a total of 41942 remaining for alignment with Bowtie2.</p>
<p>At the same time, I also added an <em>E. coli</em> genome to the set of contaminant genomes to screen against, joining cow, pig and human. I also moved the screening step to be downstream rather than upstream of the Bowtie2 filtering step, to save computation time and make it quicker to make further modifications downstream if needed. For now, I didn’t make any changes to the BBMap parameters for screening for contaminant genomes.</p>
<p>After passing putative viral read pairs through the same filtering steps as last time, I got the following result:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process new HV data</span></span>
<span><span class="va">data_dir_new</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-02-13_crits-christoph-3/"</span></span>
<span><span class="va">hv_reads_new_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"hv_hits_putative_filtered_1.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_new</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_new_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         contained <span class="op">=</span> <span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">mrg</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span><span class="va">hv_reads_new_unenriched</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_reads_new</span>, <span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span></span>
<span><span class="va">mrg_old_join</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>              <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">cause</span>, <span class="va">viral_status</span>, <span class="va">viral_status_out</span>, genome_id_old<span class="op">=</span><span class="va">genome_id</span>, taxid_old<span class="op">=</span><span class="va">taxid</span><span class="op">)</span></span>
<span><span class="va">mrg_new</span> <span class="op">&lt;-</span> <span class="va">hv_reads_new_unenriched</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_old_join</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"UNCLEAR"</span><span class="op">)</span>,</span>
<span>         kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make initial plot</span></span>
<span><span class="va">g_mrg_new</span> <span class="op">&lt;-</span> <span class="va">mrg_new</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_new</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_hist_new</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_new</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">attempt</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">g_hist_new</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_hist_new</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span>, fill<span class="op">=</span><span class="va">attempt</span>, group<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_hist_new</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This is a clear improvement on the last round: the number of high-scoring non-viral sequences has fallen from 992 to 549, and the number of high-scoring unclear sequences from 83 to 56. The degree of improvement, however, is less than I’d hoped. Looking into the new alignments, this appears to be because many of the sequences mapping to the old transgenic sequences now map to non-transgenic (or at least, not labeled to be transgenic) strains of the same viruses. When BLASTed against nt, however, these sequences still map primarily to synthetic cloning vectors, suggesting these viral genomes may still be genetically modified despite not being labeled as such.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">header_db</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">header_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                      col_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"genome_id"</span>, <span class="st">"genome_name"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unenriched_genomes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_new</span>, <span class="va">header_db</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span><span class="va">bad_genomes</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_genomes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">viral_status_out</span>, values_from<span class="op">=</span><span class="va">n</span>, names_prefix <span class="op">=</span> <span class="st">"n_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_FALSE</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">n_FALSE</span>, <span class="va">n_TRUE</span>, <span class="va">n_UNCLEAR</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_TRUE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_TRUE</span>, <span class="fl">0</span><span class="op">)</span>, n_UNCLEAR <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_UNCLEAR</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_FALSE <span class="op">=</span> <span class="va">n_FALSE</span><span class="op">/</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">+</span><span class="va">n_TRUE</span><span class="op">+</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["genome_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["genome_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_FALSE"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_TRUE"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_UNCLEAR"],"name":[5],"type":["int"],"align":["right"]},{"label":["p_FALSE"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"KM192300.1","2":"Human herpesvirus 5 strain Merlin isolate RCMV1528, complete genome","3":"382","4":"7","5":"8","6":"0.9622166"},{"1":"KY766069.1","2":"Zika virus isolate Pf13/251013-18, complete genome","3":"102","4":"0","5":"3","6":"0.9714286"},{"1":"MN369532.1","2":"Vaccinia virus isolate VK01, complete genome","3":"26","4":"0","5":"0","6":"1.0000000"},{"1":"HM133903.1","2":"Orf virus strain D1701, complete genome","3":"12","4":"0","5":"0","6":"1.0000000"},{"1":"HQ540592.1","2":"Porcine endogenous retrovirus A clone 1347C1, complete genome","3":"11","4":"3","5":"0","6":"0.7857143"},{"1":"AB618031.1","2":"Herpes simplex virus (type 1 /strain RH2) DNA, nearly complete genome","3":"7","4":"0","5":"2","6":"0.7777778"},{"1":"D00835.1","2":"Human immunodeficiency virus 2 proviral DNA, complete genome","3":"5","4":"0","5":"0","6":"1.0000000"},{"1":"JF909601.1","2":"Senegalvirus SSV-A contig6 genomic sequence","3":"1","4":"2","5":"0","6":"0.3333333"},{"1":"JQ410350.1","2":"Ectromelia virus ERPV culture-collection ATCC:VR-1431, complete genome","3":"1","4":"0","5":"0","6":"1.0000000"},{"1":"M81861.1","2":"Cardiovirus A1 strain Ruckert polyprotein gene, complete cds","3":"1","4":"0","5":"2","6":"0.3333333"},{"1":"MH341447.1","2":"Vaccinia virus strain L-IVP_oncoQ, complete genome","3":"1","4":"0","5":"0","6":"1.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unenriched_fasta</span> <span class="op">&lt;-</span> <span class="va">mrg_new</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"FALSE"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>nseq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">nseq</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num_gid <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">genome_id</span>, <span class="st">"_"</span>, <span class="va">seq_num_gid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mu_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_unenriched_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mu_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"cc-bad-gid.fasta"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_out</span> <span class="op">&lt;-</span> <span class="va">bad_genomes</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_new</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"FALSE"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">cause</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>.groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span>, </span>
<span>            by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="pass-2-additional-contaminant-screening" class="level1"><h1>Pass 2: Additional “contaminant” screening</h1>
<p>In my second attempt, I excluded further viral genomes (those with “recombinant” in the genome name) from the Bowtie database, and more importantly added several new sequences to the set of “contaminant” genomes to screen for during viral read identification. Specifically, I added one eukaryotic synthetic construct chromosome, three synthetic cloning vectors, and the <em>Klebsiella pneumoniae</em> genome, all of which came up during pass 1 as matches to false-positive viral sequences. Re-running the analysis with these changes, we see a large improvement:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process new HV data</span></span>
<span><span class="va">hv_reads_newer_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"hv_hits_putative_filtered_2.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_newer</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_newer_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         contained <span class="op">=</span> <span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">mrg</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span><span class="va">hv_reads_newer_unenriched</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_reads_newer</span>, <span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span></span>
<span><span class="va">mrg_newer</span> <span class="op">&lt;-</span> <span class="va">hv_reads_newer_unenriched</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_old_join</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"UNCLEAR"</span><span class="op">)</span>,</span>
<span>         kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make initial plot</span></span>
<span><span class="va">g_mrg_newer</span> <span class="op">&lt;-</span> <span class="va">mrg_newer</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_newer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_hist_newer</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_new</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, <span class="va">mrg_newer</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">attempt</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">g_hist_newer</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_hist_newer</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span>, fill<span class="op">=</span><span class="va">attempt</span>, group<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span></span>
<span><span class="va">g_hist_newer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">counts_newer</span> <span class="op">&lt;-</span> <span class="va">mrg_hist_newer</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">counts_newer</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Viral Status"</span>, <span class="st">"Attempt"</span>, <span class="st">"High-Scoring Read Pairs"</span><span class="op">)</span></span>
<span><span class="va">counts_newer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Viral Status"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Attempt"],"name":[2],"type":["fct"],"align":["left"]},{"label":["High-Scoring Read Pairs"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"FALSE","2":"0","3":"992"},{"1":"FALSE","2":"1","3":"549"},{"1":"FALSE","2":"2","3":"138"},{"1":"UNCLEAR","2":"0","3":"83"},{"1":"UNCLEAR","2":"1","3":"56"},{"1":"UNCLEAR","2":"2","3":"44"},{"1":"TRUE","2":"0","3":"3752"},{"1":"TRUE","2":"1","3":"3660"},{"1":"TRUE","2":"2","3":"3653"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span>, <span class="va">conjunctive</span>, <span class="va">include_special</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">include_special</span><span class="op">)</span> <span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">tab</span>, <span class="va">viral_status_out</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TRUE"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span></span>
<span>    conjunctive <span class="op">=</span> <span class="va">conjunctive</span>,</span>
<span>    retain_score_conjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">&amp;</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>, </span>
<span>    retain_score_disjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>    retain_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="va">retain_score_conjunctive</span>, <span class="va">retain_score_disjunctive</span><span class="op">)</span>,</span>
<span>    retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, include_special <span class="op">=</span> <span class="va">include_special</span>, </span>
<span>                conjunctive <span class="op">=</span> <span class="va">conjunctive</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inc_special</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span>, include_special<span class="op">=</span><span class="va">inc_special</span><span class="op">)</span></span>
<span>  <span class="va">stats_conj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_disj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_conj</span>, <span class="va">stats_disj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">threshold</span><span class="op">:</span><span class="va">conjunctive</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>conj_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats_all</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">inc_special</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">stats_old</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_unenriched_plot_2</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">stats_new</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_new</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">stats_newer</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_newer</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">stats_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_old</span>, <span class="va">stats_new</span>, <span class="va">stats_newer</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">threshold_opt</span> <span class="op">&lt;-</span> <span class="va">stats_all</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">conj_label</span>,<span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_stats_2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_all</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span>,</span>
<span>                    <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The number of high-scoring false positives has been cut by over 400 (almost 75% compared to pass 1) and the optimal F1 score (excluding unclear read pairs for now) increased from 0.938 to 0.967. However, over 100 apparent false positives still remain, still primarily arising from cloning vectors and cow and pig sequences:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_ug_newer</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_newer</span>, <span class="va">header_db</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newer</span> <span class="op">&lt;-</span> <span class="va">mrg_ug_newer</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">viral_status_out</span>, values_from<span class="op">=</span><span class="va">n</span>, names_prefix <span class="op">=</span> <span class="st">"n_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_FALSE</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">n_FALSE</span>, <span class="va">n_TRUE</span>, <span class="va">n_UNCLEAR</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_TRUE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_TRUE</span>, <span class="fl">0</span><span class="op">)</span>, n_UNCLEAR <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_UNCLEAR</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_FALSE <span class="op">=</span> <span class="va">n_FALSE</span><span class="op">/</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">+</span><span class="va">n_TRUE</span><span class="op">+</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newer_caused</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">bad_genomes_newer</span>, <span class="va">bg_causes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">cause</span><span class="op">)</span>,</span>
<span>                                      by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newer_caused</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["genome_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["genome_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_FALSE"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_TRUE"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_UNCLEAR"],"name":[5],"type":["int"],"align":["right"]},{"label":["p_FALSE"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["cause"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"KM192300.1","2":"Human herpesvirus 5 strain Merlin isolate RCMV1528, complete genome","3":"104","4":"0","5":"2","6":"0.9811321","7":"Cloning/expression vector"},{"1":"HM133903.1","2":"Orf virus strain D1701, complete genome","3":"12","4":"0","5":"0","6":"1.0000000","7":"Cow genome"},{"1":"HQ540592.1","2":"Porcine endogenous retrovirus A clone 1347C1, complete genome","3":"11","4":"3","5":"0","6":"0.7857143","7":"Pig genome"},{"1":"AB618031.1","2":"Herpes simplex virus (type 1 /strain RH2) DNA, nearly complete genome","3":"7","4":"0","5":"1","6":"0.8750000","7":"E. coli chromosomal sequence"},{"1":"JF909601.1","2":"Senegalvirus SSV-A contig6 genomic sequence","3":"1","4":"2","5":"0","6":"0.3333333","7":"Arabidopsis genome"},{"1":"JQ410350.1","2":"Ectromelia virus ERPV culture-collection ATCC:VR-1431, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Other bacterial genome"},{"1":"KY766069.1","2":"Zika virus isolate Pf13/251013-18, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Human genome / eukaryotic synthetic construct"},{"1":"MH341447.1","2":"Vaccinia virus strain L-IVP_oncoQ, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Other eukaryotic genome"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unenriched_fasta_2</span> <span class="op">&lt;-</span> <span class="va">mrg_newer</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"FALSE"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>nseq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">nseq</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num_gid <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">genome_id</span>, <span class="st">"_"</span>, <span class="va">seq_num_gid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mu_fasta_out_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_unenriched_fasta_2</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mu_fasta_out_2</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"cc-bad-gid-2.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="pass-3-unmasking-contaminant-genomes" class="level1"><h1>Pass 3: Unmasking “contaminant” genomes</h1>
<p>In my previous attempts at this problem, I mapped reads against “contaminant” genomes having first masked the latter to remove repetitive and low-entropy sequences. This makes sense in many contexts, but may not make sense here: in particular, if the repeat sequences being masked include virus-like transposable elements, this may be responsible for the failure of my current contaminant screening approach to detect and remove some of the non-viral (especially mammalian) sequences being mistaken for viruses by Bowtie2.</p>
<p>In addition to adding a further cloning vector sequence to the contaminant sequence database for this third pass, therefore, I also tried tried re-running my analysis pipeline against an unmasked version of this database. The results looked like this:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import and process new HV data</span></span>
<span><span class="va">hv_reads_newest_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"hv_hits_putative_filtered_3.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_newest</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_newest_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         contained <span class="op">=</span> <span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">mrg</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span><span class="va">hv_reads_newest_unenriched</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_reads_newest</span>, <span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span></span>
<span><span class="va">mrg_newest</span> <span class="op">&lt;-</span> <span class="va">hv_reads_newest_unenriched</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_old_join</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"UNCLEAR"</span><span class="op">)</span>,</span>
<span>         kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make initial plot</span></span>
<span><span class="va">g_mrg_newest</span> <span class="op">&lt;-</span> <span class="va">mrg_newest</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_newest</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_hist_newest</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">mrg_new</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, </span>
<span>                             <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, </span>
<span>                             <span class="va">mrg_newer</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>                             <span class="va">mrg_newest</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">attempt</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">g_hist_newest</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_hist_newest</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span>, fill<span class="op">=</span><span class="va">attempt</span>, group<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_newest</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">counts_newest</span> <span class="op">&lt;-</span> <span class="va">mrg_hist_newest</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">counts_newest</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Viral Status"</span>, <span class="st">"Attempt"</span>, <span class="st">"High-Scoring Read Pairs"</span><span class="op">)</span></span>
<span><span class="va">counts_newest</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Viral Status"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Attempt"],"name":[2],"type":["fct"],"align":["left"]},{"label":["High-Scoring Read Pairs"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"FALSE","2":"0","3":"992"},{"1":"FALSE","2":"1","3":"549"},{"1":"FALSE","2":"2","3":"138"},{"1":"FALSE","2":"3","3":"40"},{"1":"UNCLEAR","2":"0","3":"83"},{"1":"UNCLEAR","2":"1","3":"56"},{"1":"UNCLEAR","2":"2","3":"44"},{"1":"UNCLEAR","2":"3","3":"44"},{"1":"TRUE","2":"0","3":"3752"},{"1":"TRUE","2":"1","3":"3660"},{"1":"TRUE","2":"2","3":"3653"},{"1":"TRUE","2":"3","3":"3653"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_newest</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_newest</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">stats_all_3</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_old</span>, <span class="va">stats_new</span>, <span class="va">stats_newer</span>, <span class="va">stats_newest</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>attempt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">threshold_opt_3</span> <span class="op">&lt;-</span> <span class="va">stats_all_3</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">conj_label</span>,<span class="va">attempt</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_stats_3</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_all_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span>,</span>
<span>                    <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">attempt</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set1"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The number of high-scoring false positives has been cut by another 98, down to just 40, and the optimal F1 score (again excluding unclear read pairs for now) increased from to 0.98. Looking at the remaining false positives, it looks like this is entirely due to more successful removal of remaining cloning vector sequences; unmasking the cow and pig genomes seems to have had little effect:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_ug_newest</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg_newest</span>, <span class="va">header_db</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newest</span> <span class="op">&lt;-</span> <span class="va">mrg_ug_newest</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">viral_status_out</span>, values_from<span class="op">=</span><span class="va">n</span>, names_prefix <span class="op">=</span> <span class="st">"n_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_FALSE</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">n_FALSE</span>, <span class="va">n_TRUE</span>, <span class="va">n_UNCLEAR</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_TRUE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_TRUE</span>, <span class="fl">0</span><span class="op">)</span>, n_UNCLEAR <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_UNCLEAR</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_FALSE <span class="op">=</span> <span class="va">n_FALSE</span><span class="op">/</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">+</span><span class="va">n_TRUE</span><span class="op">+</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newest_caused</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">bad_genomes_newest</span>,</span>
<span>                                       <span class="va">bg_causes</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">cause</span><span class="op">)</span>,</span>
<span>                                       by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_newest_caused</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["genome_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["genome_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_FALSE"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_TRUE"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_UNCLEAR"],"name":[5],"type":["int"],"align":["right"]},{"label":["p_FALSE"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["cause"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"HM133903.1","2":"Orf virus strain D1701, complete genome","3":"12","4":"0","5":"0","6":"1.0000000","7":"Cow genome"},{"1":"HQ540592.1","2":"Porcine endogenous retrovirus A clone 1347C1, complete genome","3":"11","4":"3","5":"0","6":"0.7857143","7":"Pig genome"},{"1":"AB618031.1","2":"Herpes simplex virus (type 1 /strain RH2) DNA, nearly complete genome","3":"7","4":"0","5":"1","6":"0.8750000","7":"E. coli chromosomal sequence"},{"1":"KM192300.1","2":"Human herpesvirus 5 strain Merlin isolate RCMV1528, complete genome","3":"6","4":"0","5":"2","6":"0.7500000","7":"Cloning/expression vector"},{"1":"JF909601.1","2":"Senegalvirus SSV-A contig6 genomic sequence","3":"1","4":"2","5":"0","6":"0.3333333","7":"Arabidopsis genome"},{"1":"JQ410350.1","2":"Ectromelia virus ERPV culture-collection ATCC:VR-1431, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Other bacterial genome"},{"1":"KY766069.1","2":"Zika virus isolate Pf13/251013-18, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Human genome / eukaryotic synthetic construct"},{"1":"MH341447.1","2":"Vaccinia virus strain L-IVP_oncoQ, complete genome","3":"1","4":"0","5":"0","6":"1.0000000","7":"Other eukaryotic genome"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>While I would like to do better at removing these residual sequences, I think the F1 scores I’m getting are now high enough to consider this acceptable performance.</p>
<p>Turning now to the “unclear” sequences, we see that, unlike the false-positive sequences, these are not concentrated in a few specific culprits, but rather spread fairly evenly over numerous viruses (44 sequences across 34 genome IDs). Inspecting these manually with NCBI BLAST, we find that most, but not all, of them appear to be real matches:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">unclear_genomes_newest</span> <span class="op">&lt;-</span> <span class="va">mrg_ug_newest</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">viral_status_out</span>, values_from<span class="op">=</span><span class="va">n</span>, names_prefix <span class="op">=</span> <span class="st">"n_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_UNCLEAR</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">n_UNCLEAR</span>, <span class="va">n_FALSE</span>, <span class="va">n_TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_TRUE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_TRUE</span>, <span class="fl">0</span><span class="op">)</span>, n_FALSE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_FALSE</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_UNCLEAR <span class="op">=</span> <span class="va">n_UNCLEAR</span><span class="op">/</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">+</span><span class="va">n_TRUE</span><span class="op">+</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">unclear_genomes_newest</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["genome_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["genome_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_UNCLEAR"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_FALSE"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_TRUE"],"name":[5],"type":["int"],"align":["right"]},{"label":["p_UNCLEAR"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"MG522853.1","2":"Hudisavirus sp. isolate ETH_P4_2016, complete genome","3":"4","4":"0","5":"28","6":"0.125000000"},{"1":"FM955841.1","2":"Human papillomavirus type 105, complete genome","3":"3","4":"0","5":"18","6":"0.142857143"},{"1":"U31780.1","2":"Human papillomavirus type 22, complete genome","3":"3","4":"0","5":"28","6":"0.096774194"},{"1":"AF246939.1","2":"Human picobirnavirus strain 1-CHN-97 segment 2 RNA-dependent RNA polymerase gene, complete cds","3":"2","4":"0","5":"4","6":"0.333333333"},{"1":"GQ179640.3","2":"Salivirus NG-J1, complete genome","3":"2","4":"0","5":"8","6":"0.200000000"},{"1":"KM192300.1","2":"Human herpesvirus 5 strain Merlin isolate RCMV1528, complete genome","3":"2","4":"6","5":"0","6":"0.250000000"},{"1":"AB186897.1","2":"Human picobirnavirus RNA segment 1 genomic RNA, complete sequence","3":"1","4":"0","5":"3","6":"0.250000000"},{"1":"AB186898.1","2":"Human picobirnavirus RNA segment 2 genomic RNA, complete sequence","3":"1","4":"0","5":"2","6":"0.333333333"},{"1":"AB618031.1","2":"Herpes simplex virus (type 1 /strain RH2) DNA, nearly complete genome","3":"1","4":"7","5":"0","6":"0.125000000"},{"1":"AB747255.1","2":"Saffold virus gene for polyprotein, partial cds, isolate: Pak-3486","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"DQ358078.1","2":"Human coxsackievirus A20 serotype CAV20a, complete genome","3":"1","4":"0","5":"1","6":"0.500000000"},{"1":"EF165067.2","2":"Cardiovirus D1, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"EU410348.1","2":"Human papillomavirus type 110, complete genome","3":"1","4":"0","5":"349","6":"0.002857143"},{"1":"EU708909.1","2":"Human rotavirus HCR3A NSP3 protein (NSP3) gene, complete cds","3":"1","4":"0","5":"3","6":"0.250000000"},{"1":"FJ947080.1","2":"Human papillomavirus type 115 isolate GC02, complete genome","3":"1","4":"0","5":"17","6":"0.055555556"},{"1":"KJ412577.1","2":"Rotavirus A strain RVA/Human-wt/PRY/1471SR/2006/G12P[9] segment 3 RNA capping protein VP3 (VP3) gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KM023140.1","2":"Salivirus FHB, complete genome","3":"1","4":"0","5":"9","6":"0.100000000"},{"1":"KT600065.1","2":"Human feces pecovirus strain PeCV-PE, complete genome","3":"1","4":"0","5":"27","6":"0.035714286"},{"1":"KT600069.1","2":"Human feces smacovirus 3 strain SmaCV3, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KU059788.1","2":"Rotavirus A strain RVA/Human-wt/AUS/WAPC2016/2014/G3P[8] segment 1 RNA-dependent RNA polymerase VP1 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KU356575.1","2":"Human rotavirus A strain RVA/Human-wt/BGN/J253/2010/G2P[4] VP2 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KX814961.1","2":"Rotavirus A isolate RVA/Bat-wt/CHN/YSSK5/2015/G3P[3] NSP3 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC066150.1","2":"Human rotavirus A VP1 gene for VP1 protein, partial cds, strain: RVA/Human-wt/VNM/SP026/2012/G1P[8]","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC169887.1","2":"Rotavirus A RVA/Human-wt/THA/PCB-103/2013/G8P[8] VP3 gene for structural protein VP3, complete cds, strain: RVA/Human-wt/THA/PCB-103/2013/G8P[8]","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC708010.1","2":"Hudisavirus sp. CRESSV19-84-AMS-06 DNA, complete genome","3":"1","4":"0","5":"4","6":"0.200000000"},{"1":"LC708012.1","2":"Hudisavirus sp. CRESSV19-84-AMS-08 DNA, complete genome","3":"1","4":"0","5":"5","6":"0.166666667"},{"1":"LC708014.1","2":"Hudisavirus sp. CRESSV19-84-AMS-10 DNA, complete genome","3":"1","4":"0","5":"1","6":"0.500000000"},{"1":"LT960370.1","2":"Human polyomavirus 1 isolate CAMB-1055 genome assembly, chromosome: I","3":"1","4":"0","5":"16","6":"0.058823529"},{"1":"MF351520.1","2":"Hudisavirus sp. isolate P13_2, complete genome","3":"1","4":"0","5":"4","6":"0.200000000"},{"1":"MH341445.1","2":"Vaccinia virus strain L-IVP_oncoB, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"MH341446.1","2":"Vaccinia virus strain L-IVP_oncoM, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"S45208.1","2":"polyprotein [echovirus 22 EV22, Harris, Genomic RNA Complete, 7339 nt]","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"U31786.1","2":"Human papillomavirus type 37, complete genome","3":"1","4":"0","5":"3","6":"0.250000000"},{"1":"X70829.1","2":"Human papillomavirus type 65 complete genome","3":"1","4":"0","5":"5","6":"0.166666667"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">write_tsv</span><span class="op">(</span><span class="va">unclear_genomes_newest</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"gid-unclear-3-raw.tsv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mrg_unenriched_fasta_3</span> <span class="op">&lt;-</span> <span class="va">mrg_newer</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"UNCLEAR"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>nseq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">nseq</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num_gid <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">genome_id</span>, <span class="st">"_"</span>, <span class="va">seq_num_gid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mu_fasta_out_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_unenriched_fasta_3</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mu_fasta_out_3</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"cc-unclear-gid-3.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">unclear_genomes_caused</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"gid-unclear-3.tsv"</span><span class="op">)</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">mrg_newest_unclear</span> <span class="op">&lt;-</span> <span class="va">mrg_ug_newest</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">cause</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"UNCLEAR"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">unclear_genomes_caused</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">cause</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"Appears real"</span>, <span class="fl">2</span>, </span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"No match"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">viral_status</span><span class="op">)</span>, <span class="va">viral_status</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_newest_unclear</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">cause</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["cause"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"Appears real","2":"38"},{"1":"Cloning/expression vector","2":"2"},{"1":"Other bacterial genome","2":"1"},{"1":"Other eukaryotic genome","2":"3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_unclear</span> <span class="op">&lt;-</span> <span class="va">mrg_newest_unclear</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_unclear</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The false matches don’t appear particularly differentiated from the true matches by alignment score, either using Bowtie2 (above) or BLAST (below):</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">blast_results_seqid</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">qseqid</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span>, read_pair <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span>, sample<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">seq_id</span>, taxid_bowtie <span class="op">=</span> <span class="va">taxid</span><span class="op">)</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">blast_results_unclear</span> <span class="op">&lt;-</span> <span class="va">blast_results_seqid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">mrg_newest_unclear</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">viral_status</span>, <span class="va">viral_status_out</span>, <span class="va">cause</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span></span>
<span><span class="co"># First check if any BLAST taxid is a descendent of the corresponding Bowtie taxid, or vice versa</span></span>
<span><span class="va">taxid_dec_bowtie</span> <span class="op">&lt;-</span> <span class="va">blast_results_unclear</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid_bowtie</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">expand_taxids</span>, nodes<span class="op">=</span><span class="va">tax_nodes</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">taxid_bowtie</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">taxid_dec_blast</span> <span class="op">&lt;-</span> <span class="va">blast_results_unclear</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">expand_taxids</span>, nodes<span class="op">=</span><span class="va">tax_nodes</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">match_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">staxid</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxid_dec_bowtie</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">taxid_bowtie</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">match_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">taxid_bowtie</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxid_dec_bowtie</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">blast_results_unclear</span><span class="op">$</span><span class="va">staxid</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">match_any</span> <span class="op">&lt;-</span> <span class="va">match_1</span> <span class="op">|</span> <span class="va">match_2</span></span>
<span><span class="va">blast_results_matched</span> <span class="op">&lt;-</span> <span class="va">blast_results_unclear</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid_match <span class="op">=</span> <span class="va">match_any</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Otherwise, take the highest-scoring, longest viral match</span></span>
<span><span class="va">blast_results_single</span> <span class="op">&lt;-</span> <span class="va">blast_results_matched</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid_match</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">taxid_match</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">brs_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_single</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">bitscore</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"viral_status_out"</span><span class="op">)</span>, names_from <span class="op">=</span> <span class="st">"read_pair"</span>, </span>
<span>              values_from <span class="op">=</span> <span class="st">"bitscore"</span>, names_prefix <span class="op">=</span> <span class="st">"read_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>read_1 <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">read_1</span>, <span class="fl">0</span><span class="op">)</span>, read_2 <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">read_2</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_brs</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">brs_out</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">read_1</span>, y<span class="op">=</span><span class="va">read_2</span>, color <span class="op">=</span> <span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Best viral bitscore (forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">160</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">160</span>,<span class="fl">40</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Best viral bitscore (reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">160</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">160</span>,<span class="fl">40</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_brs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Overall, I think pulling these out for manual inspection is currently the right call. But if we’re forced to classify them automatically, counting them as true viral matches is probably the better bet.</p>
</section><section id="human-viral-reads-in-crits-christoph-2021-final-assessment" class="level1"><h1>Human viral reads in Crits-Christoph (2021): Final assessment</h1>
<p>Now that I’ve settled on a pipeline and downstream analysis process I’m happy with, we can return to the question of overall human-viral abundance and composition in Crits-Christoph (2021). I’ll use a Bowtie2 alignment score cutoff of 20 here, as this is consistent with previous studies and gives good F1 scores in the tests above.</p>
<p>Using this cutoff, we find a total of 109868/8716917 human-viral reads in panel-enriched samples (<span class="math inline">\(1.26 \times 10^{-2}\)</span> , about 1 in 80), and 4064/297690777 in unenriched samples (<span class="math inline">\(1.37 \times 10^{-5}\)</span> , about 1 in 73,000). Unsurprisingly, enriched samples show much higher overall human-viral abundance than unenriched samples; however, even unenriched samples show much higher relative abundance than <a href="https://data.securebio.org/wills-public-notebook/notebooks/2023-12-19_project-runway-bmc-rna.html">our previous BMC sludge sequences</a> ( <span class="math inline">\(\sim 3 \times 10^{-7}\)</span>):</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_old</span>, <span class="st">"qc_basic_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">hv_reads_newest_cut</span> <span class="op">&lt;-</span> <span class="va">hv_reads_newest</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">hv_reads_newest_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_newest_cut</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge</span></span>
<span><span class="va">hv_reads_ra</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">hv_reads_newest_counts</span>, <span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span><span class="va">hv_reads_total</span> <span class="op">&lt;-</span> <span class="va">hv_reads_ra</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"All"</span>, <span class="fu">str_to_lower</span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span><span class="va">hv_reads_bound</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_reads_ra</span>, <span class="va">hv_reads_total</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span></span>
<span><span class="va">hv_reads_bound</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">hv_reads_bound</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">g_phv</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_reads_bound</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads_hv</span>, color<span class="op">=</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span>, name<span class="op">=</span><span class="st">"Panel enrichment"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">enrichment</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_phv</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In comparison, the old pipeline returns an estimated overall relative abundance of human-infecting viruses in unenriched samples of roughly <span class="math inline">\(3.1 \times 10^{-6}\)</span>, nearly 5 times lower.</p>
<p>Digging into individual viruses, we see large but inconsistent differences in relative abundance between enriched and unenriched samples:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import viral genera</span></span>
<span><span class="va">viral_taxids_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"viral-taxids.tsv"</span><span class="op">)</span></span>
<span><span class="va">viral_taxids</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxids_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">viral_genera</span> <span class="op">&lt;-</span> <span class="va">viral_taxids</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"genus"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get unique name for each viral genus</span></span>
<span><span class="va">viral_genera_unique</span> <span class="op">&lt;-</span> <span class="va">viral_genera</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">viral_genera_duplicate</span> <span class="op">&lt;-</span> <span class="va">viral_genera</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">viral_genera_valid_1</span> <span class="op">&lt;-</span> <span class="va">viral_genera_duplicate</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"virus$"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">viral_genera_unique</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_genera_unique</span>, <span class="va">viral_genera_valid_1</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">viral_genera_valid_2</span> <span class="op">&lt;-</span> <span class="va">viral_genera_valid_1</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_genera_unique</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">" "</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">viral_genera_unique</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_genera_unique</span>, <span class="va">viral_genera_valid_2</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">viral_genera_valid_3</span> <span class="op">&lt;-</span> <span class="va">viral_genera_valid_2</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_genera_unique</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">viral_genera_unique</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_genera_unique</span>, <span class="va">viral_genera_valid_3</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">viral_genera_valid_4</span> <span class="op">&lt;-</span> <span class="va">viral_genera_duplicate</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span> <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_genera_unique</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">viral_genera_unique</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_genera_unique</span>, <span class="va">viral_genera_valid_4</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">write_tsv</span><span class="op">(</span><span class="va">viral_genera_unique</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_new</span>, <span class="st">"viral-genera-unique.tsv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Discover viral genera for HV reads</span></span>
<span><span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"family"</span>, <span class="st">"kingdom"</span>, <span class="st">"order"</span>, <span class="st">"phylum"</span>, <span class="st">"subfamily"</span>, <span class="st">"suborder"</span>, <span class="st">"subphylum"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span><span class="va">hv_read_db</span> <span class="op">&lt;-</span> <span class="va">hv_reads_newest_cut</span></span>
<span><span class="va">tax_nodes_cut</span> <span class="op">&lt;-</span> <span class="fu">rename</span><span class="op">(</span><span class="va">tax_nodes</span>, taxid <span class="op">=</span> <span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids</span><span class="op">)</span></span>
<span><span class="va">hv_read_genus</span> <span class="op">&lt;-</span> <span class="va">hv_read_db</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">viral_genera_unique</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">hv_read_nogenus</span> <span class="op">&lt;-</span> <span class="va">hv_read_db</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_genera_unique</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">tax_nodes_cut</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span><span class="op">)</span></span>
<span><span class="co">#cat(nrow(hv_read_db), nrow(hv_read_genus), nrow(hv_read_nogenus), nrow(hv_read_genus)+nrow(hv_read_nogenus), "\n")</span></span>
<span><span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">hv_read_nogenus</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">hv_read_genus</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_read_genus</span>, <span class="va">hv_read_nogenus</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">viral_genera_unique</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">hv_read_nogenus</span> <span class="op">&lt;-</span> <span class="va">hv_read_nogenus</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_genera_unique</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">tax_nodes_cut</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span><span class="op">)</span></span>
<span>  <span class="co">#cat(nrow(hv_read_db), nrow(hv_read_genus), nrow(hv_read_nogenus), nrow(hv_read_genus)+nrow(hv_read_nogenus), "\n")</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Get taxon names for higher-ranked assignments</span></span>
<span><span class="va">smatch</span> <span class="op">&lt;-</span> <span class="va">hv_read_db</span><span class="op">$</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_read_genus</span><span class="op">$</span><span class="va">seq_id</span></span>
<span><span class="va">hv_read_highrank</span> <span class="op">&lt;-</span> <span class="va">hv_read_db</span><span class="op">[</span><span class="op">!</span><span class="va">smatch</span>,<span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">viral_taxids</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"taxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count viral genera (&amp; unassigned viruses)</span></span>
<span><span class="va">hv_counts_wide</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">hv_read_genus</span>, <span class="va">hv_read_highrank</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"name"</span>, names_from <span class="op">=</span> <span class="st">"enrichment"</span>, values_from <span class="op">=</span> <span class="st">"n"</span>, values_fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> </span>
<span><span class="va">hv_counts</span> <span class="op">&lt;-</span> <span class="va">hv_counts_wide</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">name</span>, names_to <span class="op">=</span> <span class="st">"enrichment"</span>, values_to <span class="op">=</span> <span class="st">"n_reads_virus"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_counts_fraction</span> <span class="op">&lt;-</span> <span class="va">hv_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">hv_reads_total</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">n_reads_hv</span>, <span class="va">n_reads_raw</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"enrichment"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_virus_all <span class="op">=</span> <span class="va">n_reads_virus</span><span class="op">/</span><span class="va">n_reads_raw</span>, p_reads_virus_hv <span class="op">=</span> <span class="va">n_reads_virus</span><span class="op">/</span><span class="va">n_reads_hv</span><span class="op">)</span></span>
<span><span class="va">g_hv_counts</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">hv_counts_fraction</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">name</span>, y<span class="op">=</span><span class="va">p_reads_virus_all</span>, color<span class="op">=</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span>, name<span class="op">=</span><span class="st">"Panel enrichment"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>, hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span>, aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">g_hv_counts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-02-13_crits-christoph-3_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>As expected, viruses included in the respiratory virus panel see large increases in relative abundance in the enriched vs the unenriched samples, with the largest relative increases seen for <em>Bocaparvovirus</em> (Human bocavirus 1, 2c, 3), <em>Betacoronavirus</em> (SARS-CoV-2, OC43, HKU1), and <em>Mastadenovirus</em> (Human adenovirus B1, C2, E4). Confusingly, Orthopoxvirus, Cytomegalovirus and Gemygorvirus all also show substantially increased relative abundance, even though as far as I can tell there are no viruses from those genera in the <a href="https://www.illumina.com/ko-kr/products/by-type/sequencing-kits/library-prep-kits/respiratory-virus-oligo-panel.html">Illumina panel</a>. Numerous other viruses show weaker enrichment; even norovirus shows moderate (~4x) enrichment in the enriched vs the unenriched samples.</p>
<p>Conversely, a number of viruses are found in the unenriched samples that are absent in the enriched samples, including <em>Rotavirus</em>, <em>Flavivirus</em>, <em>Parvovirus</em>, and various papillomaviruses and polyomaviruses. One natural hypothesis for this is that these viruses were excluded by the enrichment panel and so had their relative abundance reduced; however, the enrichment observed for various other not-in-panel viruses calls this into question. An alternative hypothesis is that this difference is simply a consequence of the much deeper sequencing conducted on the unenriched samples (geometric mean of ~340k read pairs per enriched sample vs 42M read pairs per unenriched sample).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_ra_wide</span> <span class="op">&lt;-</span> <span class="va">hv_counts_fraction</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">enrichment</span>, <span class="va">p_reads_virus_all</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"name"</span>, names_from <span class="op">=</span> <span class="st">"enrichment"</span>, values_from <span class="op">=</span> <span class="st">"p_reads_virus_all"</span>, values_fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>ra_enriched <span class="op">=</span> <span class="va">Enriched</span>, ra_unenriched <span class="op">=</span> <span class="va">Unenriched</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>relative_enrichment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">ra_enriched</span><span class="op">/</span><span class="va">ra_unenriched</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">relative_enrichment</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">ra_enriched</span><span class="op">)</span>, <span class="va">ra_unenriched</span><span class="op">)</span></span>
<span><span class="va">hv_ra_wide</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["ra_unenriched"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["ra_enriched"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["relative_enrichment"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"Cyclovirus","2":"0.000000e+00","3":"1.147195e-07","4":"Inf"},{"1":"Orthohepadnavirus","2":"0.000000e+00","3":"1.147195e-07","4":"Inf"},{"1":"Rocahepevirus","2":"0.000000e+00","3":"1.147195e-07","4":"Inf"},{"1":"Bocaparvovirus","2":"6.718381e-08","3":"3.998547e-03","4":"4.77463756"},{"1":"Betacoronavirus","2":"2.687352e-07","3":"8.540634e-03","4":"4.50216552"},{"1":"Mastadenovirus","2":"1.646003e-07","3":"2.397637e-05","4":"2.16335267"},{"1":"Orthopoxvirus","2":"1.041349e-07","3":"1.066891e-05","4":"2.01052372"},{"1":"Cytomegalovirus","2":"6.382462e-08","3":"5.735973e-06","4":"1.95361887"},{"1":"Gemygorvirus","2":"3.359190e-09","3":"1.147195e-07","4":"1.53340247"},{"1":"Human fecal virus Jorvi2","2":"1.881147e-07","3":"2.867986e-06","4":"1.18315445"},{"1":"Gemykibivirus","2":"7.054300e-08","3":"9.177557e-07","4":"1.11427316"},{"1":"Huchismacovirus","2":"6.382462e-08","3":"8.030362e-07","4":"1.09974691"},{"1":"Sapovirus","2":"1.007757e-08","3":"1.147195e-07","4":"1.05628121"},{"1":"Hudisavirus sp.","2":"8.532344e-07","3":"9.177557e-06","4":"1.03165874"},{"1":"Human feces pecovirus","2":"4.434131e-07","3":"4.244620e-06","4":"0.98103026"},{"1":"Enterovirus","2":"7.726138e-08","3":"6.883168e-07","4":"0.94982588"},{"1":"Circular ssDNA virus sp.","2":"8.062057e-08","3":"5.735973e-07","4":"0.85216123"},{"1":"Betapolyomavirus","2":"3.594334e-07","3":"1.491353e-06","4":"0.61796204"},{"1":"Norovirus","2":"1.108533e-07","3":"4.588778e-07","4":"0.61694852"},{"1":"Parapoxvirus","2":"6.046543e-08","3":"1.147195e-07","4":"0.27812996"},{"1":"Simplexvirus","2":"7.726138e-08","3":"1.147195e-07","4":"0.17167463"},{"1":"Husavirus","2":"5.441888e-07","3":"6.883168e-07","4":"0.10203870"},{"1":"Orthopicobirnavirus","2":"8.263608e-07","3":"1.032475e-06","4":"0.09670987"},{"1":"Kobuvirus","2":"4.870826e-07","3":"5.735973e-07","4":"0.07100447"},{"1":"Mamastrovirus","2":"2.720944e-07","3":"1.147195e-07","4":"-0.37508255"},{"1":"Centapoxvirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Circovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Deltapolyomavirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Deltaretrovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Dependoparvovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Ledantevirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Lentivirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Molluscipoxvirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Mupapillomavirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Parechovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Parvo-like hybrid virus UC9","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Pigeon Torque tenovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Porprismacovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Respirovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Roseolovirus","2":"3.359190e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Cardiovirus","2":"6.718381e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Cosavirus","2":"6.718381e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Parvovirus","2":"6.718381e-09","3":"0.000000e+00","4":"-Inf"},{"1":"Yatapoxvirus","2":"1.007757e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Astrovirus VA3","2":"3.023271e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Gemycircularvirus","2":"3.023271e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Marseillevirus","2":"3.695109e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Nupapillomavirus","2":"3.695109e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Flavivirus","2":"4.366948e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Gammaretrovirus","2":"7.726138e-08","3":"0.000000e+00","4":"-Inf"},{"1":"Klassevirus","2":"1.578819e-07","3":"0.000000e+00","4":"-Inf"},{"1":"Gammapapillomavirus","2":"1.813963e-07","3":"0.000000e+00","4":"-Inf"},{"1":"Alphapapillomavirus","2":"2.821720e-07","3":"0.000000e+00","4":"-Inf"},{"1":"Alphapolyomavirus","2":"1.081659e-06","3":"0.000000e+00","4":"-Inf"},{"1":"Betapapillomavirus","2":"2.818361e-06","3":"0.000000e+00","4":"-Inf"},{"1":"Rotavirus","2":"3.560742e-06","3":"0.000000e+00","4":"-Inf"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>At this point, I’m satisfied with my workflow’s ability to produce usable results on the Crits-Christoph data. Next, I’ll apply this updated workflow to another previously-published WMGS dataset, likely <a href="https://doi.org/10.1128/AEM.01448-21">Rothman et al.&nbsp;(2021)</a>.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb27" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Crits-Christoph et al. (2021), part 3"</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Fixing the virus pipeline."</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-02-13</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>In my <span class="co">[</span><span class="ot">last entry</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-02-08_crits-christoph-2.html)</span>, I documented my first attempt at analyzing the human-infecting-virus content of sequence data from <span class="co">[</span><span class="ot">Crits-Christoph et al. (2021)</span><span class="co">](https://doi.org/10.1128%2FmBio.02703-20)</span>. In this entry, I'll attempt to address the issues that arose in that first analysis and analyze the resulting updated data.</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>As a reminder, I previously found that the analysis pipeline I developed for our BMC data led to numerous high-scoring false-positives when run on Crits-Christoph data:</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>data_dir_old <span class="ot">&lt;-</span> <span class="st">"../data/2024-02-04_crits-christoph-2/"</span></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"cc-libraries.txt"</span>)</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>hv_taxids_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"human-virus-taxids-all.txt"</span>)</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>tax_nodes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"nodes.dmp.gz"</span>)</span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"cat-cc-unenriched.blast.gz"</span>)</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>bg_causes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir_old, <span class="st">"cc-bad-notr.tsv"</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>header_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"human-viral-headers.txt"</span>)</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">str_to_title</span>(enrichment))</span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>hv_taxids <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_taxids_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> <span class="st">"taxid"</span>)</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>tax_nodes <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(tax_nodes_path, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">|</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X1<span class="sc">:</span>X3) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">child_taxid =</span> X1, <span class="at">parent_taxid =</span> X2, <span class="at">rank =</span> X3)</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Define taxid search function</span></span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>expand_taxids <span class="ot">&lt;-</span> <span class="cf">function</span>(taxids_in, nodes){</span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>  taxids_out <span class="ot">&lt;-</span> taxids_in</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a>  taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="fu">length</span>(taxids_new) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>    taxids_out <span class="ot">&lt;-</span> <span class="fu">c</span>(taxids_out, taxids_new) <span class="sc">%&gt;%</span> sort</span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a>    taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, </span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a>                         <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(taxids_out)</span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>v_taxids <span class="ot">&lt;-</span> <span class="fu">expand_taxids</span>(<span class="fu">c</span>(<span class="dv">10239</span>, hv_taxids<span class="sc">$</span>taxid), tax_nodes)</span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a>blast_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span>)</span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> blast_cols,</span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a>blast_results_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_results, <span class="at">viral =</span> staxid <span class="sc">%in%</span> v_taxids)</span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for the best hit for each sequence and taxid</span></span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a>blast_results_best <span class="ot">&lt;-</span> blast_results_viral <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each sequence</span></span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a>blast_results_ranked <span class="ot">&lt;-</span> blast_results_best <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter by rank</span></span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a>blast_results_highrank <span class="ot">&lt;-</span> blast_results_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>)</span>
<span id="cb27-99"><a href="#cb27-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-100"><a href="#cb27-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb27-101"><a href="#cb27-101" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb27-102"><a href="#cb27-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(qseqid, <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span>), <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-103"><a href="#cb27-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid, viral) <span class="sc">%&gt;%</span></span>
<span id="cb27-104"><a href="#cb27-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num)) <span class="sc">%&gt;%</span></span>
<span id="cb27-105"><a href="#cb27-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore), </span>
<span id="cb27-106"><a href="#cb27-106" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-107"><a href="#cb27-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb27-108"><a href="#cb27-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-109"><a href="#cb27-109" aria-hidden="true" tabindex="-1"></a><span class="co"># Process Bowtie/Kraken data</span></span>
<span id="cb27-110"><a href="#cb27-110" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb27-111"><a href="#cb27-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb27-112"><a href="#cb27-112" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb27-113"><a href="#cb27-113" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb27-114"><a href="#cb27-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb27-115"><a href="#cb27-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>())</span>
<span id="cb27-116"><a href="#cb27-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-117"><a href="#cb27-117" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare BLAST results to Kraken &amp; Bowtie assignments</span></span>
<span id="cb27-118"><a href="#cb27-118" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-119"><a href="#cb27-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, taxid, assigned_taxid, seq_id)</span>
<span id="cb27-120"><a href="#cb27-120" aria-hidden="true" tabindex="-1"></a>blast_results_assign <span class="ot">&lt;-</span> <span class="fu">left_join</span>(blast_results_paired, mrg_assign, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-121"><a href="#cb27-121" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb27-122"><a href="#cb27-122" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb27-123"><a href="#cb27-123" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb27-124"><a href="#cb27-124" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_assign <span class="sc">%&gt;%</span></span>
<span id="cb27-125"><a href="#cb27-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb27-126"><a href="#cb27-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb27-127"><a href="#cb27-127" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb27-128"><a href="#cb27-128" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb27-129"><a href="#cb27-129" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb27-130"><a href="#cb27-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-131"><a href="#cb27-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with unenriched read data</span></span>
<span id="cb27-132"><a href="#cb27-132" aria-hidden="true" tabindex="-1"></a>mrg_unenriched <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-133"><a href="#cb27-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_out, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"seq_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-134"><a href="#cb27-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>))</span>
<span id="cb27-135"><a href="#cb27-135" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_plot <span class="ot">&lt;-</span> mrg_unenriched <span class="sc">%&gt;%</span></span>
<span id="cb27-136"><a href="#cb27-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb27-137"><a href="#cb27-137" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb27-138"><a href="#cb27-138" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb27-139"><a href="#cb27-139" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_debug <span class="ot">&lt;-</span> mrg_unenriched_plot <span class="sc">%&gt;%</span></span>
<span id="cb27-140"><a href="#cb27-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb27-141"><a href="#cb27-141" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_plot_2 <span class="ot">&lt;-</span> mrg_unenriched_debug <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(bg_causes, <span class="at">by=</span><span class="st">"genome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-142"><a href="#cb27-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb27-143"><a href="#cb27-143" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status =</span> <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"Appears real"</span>, <span class="dv">2</span>, <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"No match"</span>, <span class="fu">pmax</span>(<span class="dv">1</span>, viral_status), viral_status))) <span class="sc">%&gt;%</span></span>
<span id="cb27-144"><a href="#cb27-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb27-145"><a href="#cb27-145" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb27-146"><a href="#cb27-146" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb27-147"><a href="#cb27-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-148"><a href="#cb27-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb27-149"><a href="#cb27-149" aria-hidden="true" tabindex="-1"></a>g_mrg_v2 <span class="ot">&lt;-</span> mrg_unenriched_plot_2 <span class="sc">%&gt;%</span></span>
<span id="cb27-150"><a href="#cb27-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb27-151"><a href="#cb27-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb27-152"><a href="#cb27-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-153"><a href="#cb27-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-154"><a href="#cb27-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-155"><a href="#cb27-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb27-156"><a href="#cb27-156" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-157"><a href="#cb27-157" aria-hidden="true" tabindex="-1"></a>g_mrg_v2</span>
<span id="cb27-158"><a href="#cb27-158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-159"><a href="#cb27-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-160"><a href="#cb27-160" aria-hidden="true" tabindex="-1"></a>In total, 992 non-viral read pairs and 83 read pairs of unclear status achieved high length-adjusted alignment scores ($\geq 20$) when mapped to viral Genbank with Bowtie2. For now, I will focus on the non-viral read pairs, and return to the unclear read pairs after these are in better shape.</span>
<span id="cb27-161"><a href="#cb27-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-162"><a href="#cb27-162" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pass 1: excluding transgenic and bacterial sequences</span></span>
<span id="cb27-163"><a href="#cb27-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-164"><a href="#cb27-164" aria-hidden="true" tabindex="-1"></a>The first major change I made to the pipeline was to curate the reference database for the Bowtie2 alignment to remove transgenic and other inappropriate viral sequences. To do this, I subset the collated FASTA file of viral genomes with seqtk to remove any sequence with "transgenic", "mutant", "recombinant", "unverified" or "draft" in its sequence ID. This removed a total of 493 genomes from the reference DB, leaving a total of 41942 remaining for alignment with Bowtie2.</span>
<span id="cb27-165"><a href="#cb27-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-166"><a href="#cb27-166" aria-hidden="true" tabindex="-1"></a>At the same time, I also added an *E. coli* genome to the set of contaminant genomes to screen against, joining cow, pig and human. I also moved the screening step to be downstream rather than upstream of the Bowtie2 filtering step, to save computation time and make it quicker to make further modifications downstream if needed. For now, I didn't make any changes to the BBMap parameters for screening for contaminant genomes.</span>
<span id="cb27-167"><a href="#cb27-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-168"><a href="#cb27-168" aria-hidden="true" tabindex="-1"></a>After passing putative viral read pairs through the same filtering steps as last time, I got the following result:</span>
<span id="cb27-169"><a href="#cb27-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-172"><a href="#cb27-172" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-173"><a href="#cb27-173" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb27-174"><a href="#cb27-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-175"><a href="#cb27-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process new HV data</span></span>
<span id="cb27-176"><a href="#cb27-176" aria-hidden="true" tabindex="-1"></a>data_dir_new <span class="ot">&lt;-</span> <span class="st">"../data/2024-02-13_crits-christoph-3/"</span></span>
<span id="cb27-177"><a href="#cb27-177" aria-hidden="true" tabindex="-1"></a>hv_reads_new_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_new, <span class="st">"hv_hits_putative_filtered_1.tsv.gz"</span>)</span>
<span id="cb27-178"><a href="#cb27-178" aria-hidden="true" tabindex="-1"></a>hv_reads_new <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_new_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-179"><a href="#cb27-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-180"><a href="#cb27-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb27-181"><a href="#cb27-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample),</span>
<span id="cb27-182"><a href="#cb27-182" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb27-183"><a href="#cb27-183" aria-hidden="true" tabindex="-1"></a>         <span class="at">contained =</span> seq_id <span class="sc">%in%</span> mrg<span class="sc">$</span>seq_id)</span>
<span id="cb27-184"><a href="#cb27-184" aria-hidden="true" tabindex="-1"></a>hv_reads_new_unenriched <span class="ot">&lt;-</span> <span class="fu">filter</span>(hv_reads_new, enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>)</span>
<span id="cb27-185"><a href="#cb27-185" aria-hidden="true" tabindex="-1"></a>mrg_old_join <span class="ot">&lt;-</span> mrg_unenriched_plot_2 <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb27-186"><a href="#cb27-186" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(seq_id, cause, viral_status, viral_status_out, <span class="at">genome_id_old=</span>genome_id, <span class="at">taxid_old=</span>taxid)</span>
<span id="cb27-187"><a href="#cb27-187" aria-hidden="true" tabindex="-1"></a>mrg_new <span class="ot">&lt;-</span> hv_reads_new_unenriched <span class="sc">%&gt;%</span></span>
<span id="cb27-188"><a href="#cb27-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_old_join, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-189"><a href="#cb27-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb27-190"><a href="#cb27-190" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">replace_na</span>(viral_status_out, <span class="st">"UNCLEAR"</span>),</span>
<span id="cb27-191"><a href="#cb27-191" aria-hidden="true" tabindex="-1"></a>         <span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb27-192"><a href="#cb27-192" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb27-193"><a href="#cb27-193" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>)))</span>
<span id="cb27-194"><a href="#cb27-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-195"><a href="#cb27-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Make initial plot</span></span>
<span id="cb27-196"><a href="#cb27-196" aria-hidden="true" tabindex="-1"></a>g_mrg_new <span class="ot">&lt;-</span> mrg_new <span class="sc">%&gt;%</span></span>
<span id="cb27-197"><a href="#cb27-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb27-198"><a href="#cb27-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb27-199"><a href="#cb27-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-200"><a href="#cb27-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-201"><a href="#cb27-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-202"><a href="#cb27-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb27-203"><a href="#cb27-203" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-204"><a href="#cb27-204" aria-hidden="true" tabindex="-1"></a>g_mrg_new</span>
<span id="cb27-205"><a href="#cb27-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-206"><a href="#cb27-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-209"><a href="#cb27-209" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-210"><a href="#cb27-210" aria-hidden="true" tabindex="-1"></a>mrg_hist_new <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_new <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">1</span>), mrg_unenriched_plot_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="fu">factor</span>(attempt, <span class="at">levels=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))) </span>
<span id="cb27-211"><a href="#cb27-211" aria-hidden="true" tabindex="-1"></a>g_hist_new <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_hist_new, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max, <span class="at">fill=</span>attempt, <span class="at">group=</span>attempt)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base</span>
<span id="cb27-212"><a href="#cb27-212" aria-hidden="true" tabindex="-1"></a>g_hist_new</span>
<span id="cb27-213"><a href="#cb27-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-214"><a href="#cb27-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-215"><a href="#cb27-215" aria-hidden="true" tabindex="-1"></a>This is a clear improvement on the last round: the number of high-scoring non-viral sequences has fallen from 992 to 549, and the number of high-scoring unclear sequences from 83 to 56. The degree of improvement, however, is less than I'd hoped. Looking into the new alignments, this appears to be because many of the sequences mapping to the old transgenic sequences now map to non-transgenic (or at least, not labeled to be transgenic) strains of the same viruses. When BLASTed against nt, however, these sequences still map primarily to synthetic cloning vectors, suggesting these viral genomes may still be genetically modified despite not being labeled as such.</span>
<span id="cb27-216"><a href="#cb27-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-219"><a href="#cb27-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-220"><a href="#cb27-220" aria-hidden="true" tabindex="-1"></a>header_db <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(header_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-221"><a href="#cb27-221" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"genome_id"</span>, <span class="st">"genome_name"</span>))</span>
<span id="cb27-222"><a href="#cb27-222" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_genomes <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_new, header_db, <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-223"><a href="#cb27-223" aria-hidden="true" tabindex="-1"></a>bad_genomes <span class="ot">&lt;-</span> mrg_unenriched_genomes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-224"><a href="#cb27-224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id, genome_name, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-225"><a href="#cb27-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>viral_status_out, <span class="at">values_from=</span>n, <span class="at">names_prefix =</span> <span class="st">"n_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-226"><a href="#cb27-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_FALSE <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n_FALSE)) <span class="sc">%&gt;%</span></span>
<span id="cb27-227"><a href="#cb27-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genome_id, genome_name, n_FALSE, n_TRUE, n_UNCLEAR) <span class="sc">%&gt;%</span></span>
<span id="cb27-228"><a href="#cb27-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_TRUE =</span> <span class="fu">replace_na</span>(n_TRUE, <span class="dv">0</span>), <span class="at">n_UNCLEAR =</span> <span class="fu">replace_na</span>(n_UNCLEAR, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-229"><a href="#cb27-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_FALSE =</span> n_FALSE<span class="sc">/</span>(n_FALSE<span class="sc">+</span>n_TRUE<span class="sc">+</span>n_UNCLEAR))</span>
<span id="cb27-230"><a href="#cb27-230" aria-hidden="true" tabindex="-1"></a>bad_genomes</span>
<span id="cb27-231"><a href="#cb27-231" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-232"><a href="#cb27-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-235"><a href="#cb27-235" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-236"><a href="#cb27-236" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_fasta <span class="ot">&lt;-</span> mrg_new <span class="sc">%&gt;%</span> </span>
<span id="cb27-237"><a href="#cb27-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status_out <span class="sc">==</span> <span class="st">"FALSE"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-238"><a href="#cb27-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-239"><a href="#cb27-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">nseq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(nseq), <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb27-240"><a href="#cb27-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-241"><a href="#cb27-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num_gid =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, genome_id, <span class="st">"_"</span>, seq_num_gid)) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb27-242"><a href="#cb27-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb27-243"><a href="#cb27-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb27-244"><a href="#cb27-244" aria-hidden="true" tabindex="-1"></a>mu_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_unenriched_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb27-245"><a href="#cb27-245" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mu_fasta_out, <span class="fu">file.path</span>(data_dir_new, <span class="st">"cc-bad-gid.fasta"</span>))</span>
<span id="cb27-246"><a href="#cb27-246" aria-hidden="true" tabindex="-1"></a>bad_genomes_out <span class="ot">&lt;-</span> bad_genomes <span class="sc">%&gt;%</span></span>
<span id="cb27-247"><a href="#cb27-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_new <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status_out <span class="sc">==</span> <span class="st">"FALSE"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-248"><a href="#cb27-248" aria-hidden="true" tabindex="-1"></a>              <span class="fu">group_by</span>(genome_id, cause) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">.groups=</span><span class="st">"drop"</span>), </span>
<span id="cb27-249"><a href="#cb27-249" aria-hidden="true" tabindex="-1"></a>            <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-250"><a href="#cb27-250" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-251"><a href="#cb27-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-252"><a href="#cb27-252" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pass 2: Additional "contaminant" screening</span></span>
<span id="cb27-253"><a href="#cb27-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-254"><a href="#cb27-254" aria-hidden="true" tabindex="-1"></a>In my second attempt, I excluded further viral genomes (those with "recombinant" in the genome name) from the Bowtie database, and more importantly added several new sequences to the set of "contaminant" genomes to screen for during viral read identification. Specifically, I added one eukaryotic synthetic construct chromosome, three synthetic cloning vectors, and the *Klebsiella pneumoniae* genome, all of which came up during pass 1 as matches to false-positive viral sequences. Re-running the analysis with these changes, we see a large improvement:</span>
<span id="cb27-255"><a href="#cb27-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-258"><a href="#cb27-258" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-259"><a href="#cb27-259" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb27-260"><a href="#cb27-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-261"><a href="#cb27-261" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process new HV data</span></span>
<span id="cb27-262"><a href="#cb27-262" aria-hidden="true" tabindex="-1"></a>hv_reads_newer_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_new, <span class="st">"hv_hits_putative_filtered_2.tsv.gz"</span>)</span>
<span id="cb27-263"><a href="#cb27-263" aria-hidden="true" tabindex="-1"></a>hv_reads_newer <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_newer_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-264"><a href="#cb27-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-265"><a href="#cb27-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb27-266"><a href="#cb27-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample),</span>
<span id="cb27-267"><a href="#cb27-267" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb27-268"><a href="#cb27-268" aria-hidden="true" tabindex="-1"></a>         <span class="at">contained =</span> seq_id <span class="sc">%in%</span> mrg<span class="sc">$</span>seq_id)</span>
<span id="cb27-269"><a href="#cb27-269" aria-hidden="true" tabindex="-1"></a>hv_reads_newer_unenriched <span class="ot">&lt;-</span> <span class="fu">filter</span>(hv_reads_newer, enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>)</span>
<span id="cb27-270"><a href="#cb27-270" aria-hidden="true" tabindex="-1"></a>mrg_newer <span class="ot">&lt;-</span> hv_reads_newer_unenriched <span class="sc">%&gt;%</span></span>
<span id="cb27-271"><a href="#cb27-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_old_join, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-272"><a href="#cb27-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb27-273"><a href="#cb27-273" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">replace_na</span>(viral_status_out, <span class="st">"UNCLEAR"</span>),</span>
<span id="cb27-274"><a href="#cb27-274" aria-hidden="true" tabindex="-1"></a>         <span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb27-275"><a href="#cb27-275" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb27-276"><a href="#cb27-276" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>)))</span>
<span id="cb27-277"><a href="#cb27-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-278"><a href="#cb27-278" aria-hidden="true" tabindex="-1"></a><span class="co"># Make initial plot</span></span>
<span id="cb27-279"><a href="#cb27-279" aria-hidden="true" tabindex="-1"></a>g_mrg_newer <span class="ot">&lt;-</span> mrg_newer <span class="sc">%&gt;%</span></span>
<span id="cb27-280"><a href="#cb27-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb27-281"><a href="#cb27-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb27-282"><a href="#cb27-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-283"><a href="#cb27-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-284"><a href="#cb27-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-285"><a href="#cb27-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb27-286"><a href="#cb27-286" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-287"><a href="#cb27-287" aria-hidden="true" tabindex="-1"></a>g_mrg_newer</span>
<span id="cb27-288"><a href="#cb27-288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-289"><a href="#cb27-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-292"><a href="#cb27-292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-293"><a href="#cb27-293" aria-hidden="true" tabindex="-1"></a>mrg_hist_newer <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_new <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">1</span>), mrg_unenriched_plot_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">0</span>), mrg_newer <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">2</span>)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="fu">factor</span>(attempt, <span class="at">levels=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>))) </span>
<span id="cb27-294"><a href="#cb27-294" aria-hidden="true" tabindex="-1"></a>g_hist_newer <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_hist_newer, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max, <span class="at">fill=</span>attempt, <span class="at">group=</span>attempt)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base</span>
<span id="cb27-295"><a href="#cb27-295" aria-hidden="true" tabindex="-1"></a>g_hist_newer</span>
<span id="cb27-296"><a href="#cb27-296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-297"><a href="#cb27-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-300"><a href="#cb27-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-301"><a href="#cb27-301" aria-hidden="true" tabindex="-1"></a>counts_newer <span class="ot">&lt;-</span> mrg_hist_newer <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(viral_status_out, attempt) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb27-302"><a href="#cb27-302" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts_newer) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Viral Status"</span>, <span class="st">"Attempt"</span>, <span class="st">"High-Scoring Read Pairs"</span>)</span>
<span id="cb27-303"><a href="#cb27-303" aria-hidden="true" tabindex="-1"></a>counts_newer</span>
<span id="cb27-304"><a href="#cb27-304" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-305"><a href="#cb27-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-308"><a href="#cb27-308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-309"><a href="#cb27-309" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold, conjunctive, include_special){</span>
<span id="cb27-310"><a href="#cb27-310" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>include_special) tab <span class="ot">&lt;-</span> <span class="fu">filter</span>(tab, viral_status_out <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>))</span>
<span id="cb27-311"><a href="#cb27-311" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb27-312"><a href="#cb27-312" aria-hidden="true" tabindex="-1"></a>    <span class="at">conjunctive =</span> conjunctive,</span>
<span id="cb27-313"><a href="#cb27-313" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_conjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">&amp;</span> adj_score_rev <span class="sc">&gt;</span> score_threshold), </span>
<span id="cb27-314"><a href="#cb27-314" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_disjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb27-315"><a href="#cb27-315" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score =</span> <span class="fu">ifelse</span>(conjunctive, retain_score_conjunctive, retain_score_disjunctive),</span>
<span id="cb27-316"><a href="#cb27-316" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb27-317"><a href="#cb27-317" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb27-318"><a href="#cb27-318" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb27-319"><a href="#cb27-319" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb27-320"><a href="#cb27-320" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb27-321"><a href="#cb27-321" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb27-322"><a href="#cb27-322" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb27-323"><a href="#cb27-323" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb27-324"><a href="#cb27-324" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb27-325"><a href="#cb27-325" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb27-326"><a href="#cb27-326" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">include_special =</span> include_special, </span>
<span id="cb27-327"><a href="#cb27-327" aria-hidden="true" tabindex="-1"></a>                <span class="at">conjunctive =</span> conjunctive, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb27-328"><a href="#cb27-328" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb27-329"><a href="#cb27-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb27-330"><a href="#cb27-330" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-331"><a href="#cb27-331" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, inc_special, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb27-332"><a href="#cb27-332" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab, <span class="at">include_special=</span>inc_special)</span>
<span id="cb27-333"><a href="#cb27-333" aria-hidden="true" tabindex="-1"></a>  stats_conj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb27-334"><a href="#cb27-334" aria-hidden="true" tabindex="-1"></a>  stats_disj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb27-335"><a href="#cb27-335" aria-hidden="true" tabindex="-1"></a>  stats_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_conj, stats_disj) <span class="sc">%&gt;%</span></span>
<span id="cb27-336"><a href="#cb27-336" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>(threshold<span class="sc">:</span>conjunctive), <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-337"><a href="#cb27-337" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">conj_label =</span> <span class="fu">ifelse</span>(conjunctive, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span>))</span>
<span id="cb27-338"><a href="#cb27-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats_all)</span>
<span id="cb27-339"><a href="#cb27-339" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-340"><a href="#cb27-340" aria-hidden="true" tabindex="-1"></a>inc_special <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb27-341"><a href="#cb27-341" aria-hidden="true" tabindex="-1"></a>stats_old <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_unenriched_plot_2, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">0</span>)</span>
<span id="cb27-342"><a href="#cb27-342" aria-hidden="true" tabindex="-1"></a>stats_new <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_new, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">1</span>)</span>
<span id="cb27-343"><a href="#cb27-343" aria-hidden="true" tabindex="-1"></a>stats_newer <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_newer, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">2</span>)</span>
<span id="cb27-344"><a href="#cb27-344" aria-hidden="true" tabindex="-1"></a>stats_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_old, stats_new, stats_newer) <span class="sc">%&gt;%</span> </span>
<span id="cb27-345"><a href="#cb27-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="fu">as.factor</span>(attempt))</span>
<span id="cb27-346"><a href="#cb27-346" aria-hidden="true" tabindex="-1"></a>threshold_opt <span class="ot">&lt;-</span> stats_all <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(conj_label,attempt) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">==</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="fu">min</span>(threshold))</span>
<span id="cb27-347"><a href="#cb27-347" aria-hidden="true" tabindex="-1"></a>g_stats_2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>),</span>
<span id="cb27-348"><a href="#cb27-348" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>attempt)) <span class="sc">+</span></span>
<span id="cb27-349"><a href="#cb27-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb27-350"><a href="#cb27-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-351"><a href="#cb27-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-352"><a href="#cb27-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>)<span class="sc">+</span></span>
<span id="cb27-353"><a href="#cb27-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>conj_label) <span class="sc">+</span></span>
<span id="cb27-354"><a href="#cb27-354" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb27-355"><a href="#cb27-355" aria-hidden="true" tabindex="-1"></a>g_stats_2</span>
<span id="cb27-356"><a href="#cb27-356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-357"><a href="#cb27-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-358"><a href="#cb27-358" aria-hidden="true" tabindex="-1"></a>The number of high-scoring false positives has been cut by over 400 (almost 75% compared to pass 1) and the optimal F1 score (excluding unclear read pairs for now) increased from 0.938 to 0.967. However, over 100 apparent false positives still remain, still primarily arising from cloning vectors and cow and pig sequences:</span>
<span id="cb27-359"><a href="#cb27-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-362"><a href="#cb27-362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-363"><a href="#cb27-363" aria-hidden="true" tabindex="-1"></a>mrg_ug_newer <span class="ot">&lt;-</span> <span class="fu">left_join</span>(mrg_newer, header_db, <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-364"><a href="#cb27-364" aria-hidden="true" tabindex="-1"></a>bad_genomes_newer <span class="ot">&lt;-</span> mrg_ug_newer <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-365"><a href="#cb27-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id, genome_name, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-366"><a href="#cb27-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>viral_status_out, <span class="at">values_from=</span>n, <span class="at">names_prefix =</span> <span class="st">"n_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-367"><a href="#cb27-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_FALSE <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n_FALSE)) <span class="sc">%&gt;%</span></span>
<span id="cb27-368"><a href="#cb27-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genome_id, genome_name, n_FALSE, n_TRUE, n_UNCLEAR) <span class="sc">%&gt;%</span></span>
<span id="cb27-369"><a href="#cb27-369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_TRUE =</span> <span class="fu">replace_na</span>(n_TRUE, <span class="dv">0</span>), <span class="at">n_UNCLEAR =</span> <span class="fu">replace_na</span>(n_UNCLEAR, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-370"><a href="#cb27-370" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_FALSE =</span> n_FALSE<span class="sc">/</span>(n_FALSE<span class="sc">+</span>n_TRUE<span class="sc">+</span>n_UNCLEAR))</span>
<span id="cb27-371"><a href="#cb27-371" aria-hidden="true" tabindex="-1"></a>bad_genomes_newer_caused <span class="ot">&lt;-</span> <span class="fu">left_join</span>(bad_genomes_newer, bg_causes <span class="sc">%&gt;%</span> <span class="fu">select</span>(genome_id, cause),</span>
<span id="cb27-372"><a href="#cb27-372" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-373"><a href="#cb27-373" aria-hidden="true" tabindex="-1"></a>bad_genomes_newer_caused</span>
<span id="cb27-374"><a href="#cb27-374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-375"><a href="#cb27-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-378"><a href="#cb27-378" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-379"><a href="#cb27-379" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_fasta_2 <span class="ot">&lt;-</span> mrg_newer <span class="sc">%&gt;%</span> </span>
<span id="cb27-380"><a href="#cb27-380" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status_out <span class="sc">==</span> <span class="st">"FALSE"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-381"><a href="#cb27-381" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-382"><a href="#cb27-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">nseq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(nseq), <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb27-383"><a href="#cb27-383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-384"><a href="#cb27-384" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num_gid =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, genome_id, <span class="st">"_"</span>, seq_num_gid)) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb27-385"><a href="#cb27-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb27-386"><a href="#cb27-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb27-387"><a href="#cb27-387" aria-hidden="true" tabindex="-1"></a>mu_fasta_out_2 <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_unenriched_fasta_2, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb27-388"><a href="#cb27-388" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mu_fasta_out_2, <span class="fu">file.path</span>(data_dir_new, <span class="st">"cc-bad-gid-2.fasta"</span>))</span>
<span id="cb27-389"><a href="#cb27-389" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-390"><a href="#cb27-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-391"><a href="#cb27-391" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pass 3: Unmasking "contaminant" genomes</span></span>
<span id="cb27-392"><a href="#cb27-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-393"><a href="#cb27-393" aria-hidden="true" tabindex="-1"></a>In my previous attempts at this problem, I mapped reads against "contaminant" genomes having first masked the latter to remove repetitive and low-entropy sequences. This makes sense in many contexts, but may not make sense here: in particular, if the repeat sequences being masked include virus-like transposable elements, this may be responsible for the failure of my current contaminant screening approach to detect and remove some of the non-viral (especially mammalian) sequences being mistaken for viruses by Bowtie2.</span>
<span id="cb27-394"><a href="#cb27-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-395"><a href="#cb27-395" aria-hidden="true" tabindex="-1"></a>In addition to adding a further cloning vector sequence to the contaminant sequence database for this third pass, therefore, I also tried tried re-running my analysis pipeline against an unmasked version of this database. The results looked like this:</span>
<span id="cb27-396"><a href="#cb27-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-399"><a href="#cb27-399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-400"><a href="#cb27-400" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb27-401"><a href="#cb27-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-402"><a href="#cb27-402" aria-hidden="true" tabindex="-1"></a><span class="co"># Import and process new HV data</span></span>
<span id="cb27-403"><a href="#cb27-403" aria-hidden="true" tabindex="-1"></a>hv_reads_newest_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_new, <span class="st">"hv_hits_putative_filtered_3.tsv.gz"</span>)</span>
<span id="cb27-404"><a href="#cb27-404" aria-hidden="true" tabindex="-1"></a>hv_reads_newest <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_newest_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-405"><a href="#cb27-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-406"><a href="#cb27-406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb27-407"><a href="#cb27-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample),</span>
<span id="cb27-408"><a href="#cb27-408" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb27-409"><a href="#cb27-409" aria-hidden="true" tabindex="-1"></a>         <span class="at">contained =</span> seq_id <span class="sc">%in%</span> mrg<span class="sc">$</span>seq_id)</span>
<span id="cb27-410"><a href="#cb27-410" aria-hidden="true" tabindex="-1"></a>hv_reads_newest_unenriched <span class="ot">&lt;-</span> <span class="fu">filter</span>(hv_reads_newest, enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>)</span>
<span id="cb27-411"><a href="#cb27-411" aria-hidden="true" tabindex="-1"></a>mrg_newest <span class="ot">&lt;-</span> hv_reads_newest_unenriched <span class="sc">%&gt;%</span></span>
<span id="cb27-412"><a href="#cb27-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg_old_join, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-413"><a href="#cb27-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb27-414"><a href="#cb27-414" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">replace_na</span>(viral_status_out, <span class="st">"UNCLEAR"</span>),</span>
<span id="cb27-415"><a href="#cb27-415" aria-hidden="true" tabindex="-1"></a>         <span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb27-416"><a href="#cb27-416" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb27-417"><a href="#cb27-417" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>)))</span>
<span id="cb27-418"><a href="#cb27-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-419"><a href="#cb27-419" aria-hidden="true" tabindex="-1"></a><span class="co"># Make initial plot</span></span>
<span id="cb27-420"><a href="#cb27-420" aria-hidden="true" tabindex="-1"></a>g_mrg_newest <span class="ot">&lt;-</span> mrg_newest <span class="sc">%&gt;%</span></span>
<span id="cb27-421"><a href="#cb27-421" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb27-422"><a href="#cb27-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb27-423"><a href="#cb27-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-424"><a href="#cb27-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-425"><a href="#cb27-425" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-426"><a href="#cb27-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb27-427"><a href="#cb27-427" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-428"><a href="#cb27-428" aria-hidden="true" tabindex="-1"></a>g_mrg_newest</span>
<span id="cb27-429"><a href="#cb27-429" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-430"><a href="#cb27-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-433"><a href="#cb27-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-434"><a href="#cb27-434" aria-hidden="true" tabindex="-1"></a>mrg_hist_newest <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mrg_new <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">1</span>), </span>
<span id="cb27-435"><a href="#cb27-435" aria-hidden="true" tabindex="-1"></a>                             mrg_unenriched_plot_2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="dv">0</span>), </span>
<span id="cb27-436"><a href="#cb27-436" aria-hidden="true" tabindex="-1"></a>                             mrg_newer <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">2</span>),</span>
<span id="cb27-437"><a href="#cb27-437" aria-hidden="true" tabindex="-1"></a>                             mrg_newest <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">3</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-438"><a href="#cb27-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="fu">factor</span>(attempt, <span class="at">levels=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>))) </span>
<span id="cb27-439"><a href="#cb27-439" aria-hidden="true" tabindex="-1"></a>g_hist_newest <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_hist_newest, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max, <span class="at">fill=</span>attempt, <span class="at">group=</span>attempt)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb27-440"><a href="#cb27-440" aria-hidden="true" tabindex="-1"></a>g_hist_newest</span>
<span id="cb27-441"><a href="#cb27-441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-442"><a href="#cb27-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-445"><a href="#cb27-445" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-446"><a href="#cb27-446" aria-hidden="true" tabindex="-1"></a>counts_newest <span class="ot">&lt;-</span> mrg_hist_newest <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(viral_status_out, attempt) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> ungroup</span>
<span id="cb27-447"><a href="#cb27-447" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts_newest) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Viral Status"</span>, <span class="st">"Attempt"</span>, <span class="st">"High-Scoring Read Pairs"</span>)</span>
<span id="cb27-448"><a href="#cb27-448" aria-hidden="true" tabindex="-1"></a>counts_newest</span>
<span id="cb27-449"><a href="#cb27-449" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-450"><a href="#cb27-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-453"><a href="#cb27-453" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-454"><a href="#cb27-454" aria-hidden="true" tabindex="-1"></a>stats_newest <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_newest, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">3</span>)</span>
<span id="cb27-455"><a href="#cb27-455" aria-hidden="true" tabindex="-1"></a>stats_all_3 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_old, stats_new, stats_newer, stats_newest) <span class="sc">%&gt;%</span> </span>
<span id="cb27-456"><a href="#cb27-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">attempt =</span> <span class="fu">as.factor</span>(attempt))</span>
<span id="cb27-457"><a href="#cb27-457" aria-hidden="true" tabindex="-1"></a>threshold_opt_3 <span class="ot">&lt;-</span> stats_all_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(conj_label,attempt) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(value <span class="sc">==</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="fu">min</span>(threshold))</span>
<span id="cb27-458"><a href="#cb27-458" aria-hidden="true" tabindex="-1"></a>g_stats_3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_all_3 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>),</span>
<span id="cb27-459"><a href="#cb27-459" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>attempt)) <span class="sc">+</span></span>
<span id="cb27-460"><a href="#cb27-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb27-461"><a href="#cb27-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-462"><a href="#cb27-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-463"><a href="#cb27-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>)<span class="sc">+</span></span>
<span id="cb27-464"><a href="#cb27-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>conj_label) <span class="sc">+</span></span>
<span id="cb27-465"><a href="#cb27-465" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb27-466"><a href="#cb27-466" aria-hidden="true" tabindex="-1"></a>g_stats_3</span>
<span id="cb27-467"><a href="#cb27-467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-468"><a href="#cb27-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-469"><a href="#cb27-469" aria-hidden="true" tabindex="-1"></a>The number of high-scoring false positives has been cut by another 98, down to just 40, and the optimal F1 score (again excluding unclear read pairs for now) increased from to 0.98. Looking at the remaining false positives, it looks like this is entirely due to more successful removal of remaining cloning vector sequences; unmasking the cow and pig genomes seems to have had little effect:</span>
<span id="cb27-470"><a href="#cb27-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-473"><a href="#cb27-473" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-474"><a href="#cb27-474" aria-hidden="true" tabindex="-1"></a>mrg_ug_newest <span class="ot">&lt;-</span> <span class="fu">left_join</span>(mrg_newest, header_db, <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-475"><a href="#cb27-475" aria-hidden="true" tabindex="-1"></a>bad_genomes_newest <span class="ot">&lt;-</span> mrg_ug_newest <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-476"><a href="#cb27-476" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id, genome_name, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-477"><a href="#cb27-477" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>viral_status_out, <span class="at">values_from=</span>n, <span class="at">names_prefix =</span> <span class="st">"n_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-478"><a href="#cb27-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_FALSE <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n_FALSE)) <span class="sc">%&gt;%</span></span>
<span id="cb27-479"><a href="#cb27-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genome_id, genome_name, n_FALSE, n_TRUE, n_UNCLEAR) <span class="sc">%&gt;%</span></span>
<span id="cb27-480"><a href="#cb27-480" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_TRUE =</span> <span class="fu">replace_na</span>(n_TRUE, <span class="dv">0</span>), <span class="at">n_UNCLEAR =</span> <span class="fu">replace_na</span>(n_UNCLEAR, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-481"><a href="#cb27-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_FALSE =</span> n_FALSE<span class="sc">/</span>(n_FALSE<span class="sc">+</span>n_TRUE<span class="sc">+</span>n_UNCLEAR))</span>
<span id="cb27-482"><a href="#cb27-482" aria-hidden="true" tabindex="-1"></a>bad_genomes_newest_caused <span class="ot">&lt;-</span> <span class="fu">left_join</span>(bad_genomes_newest,</span>
<span id="cb27-483"><a href="#cb27-483" aria-hidden="true" tabindex="-1"></a>                                       bg_causes <span class="sc">%&gt;%</span> <span class="fu">select</span>(genome_id, cause),</span>
<span id="cb27-484"><a href="#cb27-484" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb27-485"><a href="#cb27-485" aria-hidden="true" tabindex="-1"></a>bad_genomes_newest_caused</span>
<span id="cb27-486"><a href="#cb27-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-487"><a href="#cb27-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-488"><a href="#cb27-488" aria-hidden="true" tabindex="-1"></a>While I would like to do better at removing these residual sequences, I think the F1 scores I'm getting are now high enough to consider this acceptable performance.</span>
<span id="cb27-489"><a href="#cb27-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-490"><a href="#cb27-490" aria-hidden="true" tabindex="-1"></a>Turning now to the "unclear" sequences, we see that, unlike the false-positive sequences, these are not concentrated in a few specific culprits, but rather spread fairly evenly over numerous viruses (44 sequences across 34 genome IDs). Inspecting these manually with NCBI BLAST, we find that most, but not all, of them appear to be real matches:</span>
<span id="cb27-491"><a href="#cb27-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-494"><a href="#cb27-494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-495"><a href="#cb27-495" aria-hidden="true" tabindex="-1"></a>unclear_genomes_newest <span class="ot">&lt;-</span> mrg_ug_newest <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-496"><a href="#cb27-496" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id, genome_name, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-497"><a href="#cb27-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>viral_status_out, <span class="at">values_from=</span>n, <span class="at">names_prefix =</span> <span class="st">"n_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-498"><a href="#cb27-498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_UNCLEAR <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n_UNCLEAR)) <span class="sc">%&gt;%</span></span>
<span id="cb27-499"><a href="#cb27-499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genome_id, genome_name, n_UNCLEAR, n_FALSE, n_TRUE) <span class="sc">%&gt;%</span></span>
<span id="cb27-500"><a href="#cb27-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_TRUE =</span> <span class="fu">replace_na</span>(n_TRUE, <span class="dv">0</span>), <span class="at">n_FALSE =</span> <span class="fu">replace_na</span>(n_FALSE, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-501"><a href="#cb27-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_UNCLEAR =</span> n_UNCLEAR<span class="sc">/</span>(n_FALSE<span class="sc">+</span>n_TRUE<span class="sc">+</span>n_UNCLEAR))</span>
<span id="cb27-502"><a href="#cb27-502" aria-hidden="true" tabindex="-1"></a>unclear_genomes_newest</span>
<span id="cb27-503"><a href="#cb27-503" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(unclear_genomes_newest, <span class="fu">file.path</span>(data_dir_new, <span class="st">"gid-unclear-3-raw.tsv"</span>))</span>
<span id="cb27-504"><a href="#cb27-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-505"><a href="#cb27-505" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_fasta_3 <span class="ot">&lt;-</span> mrg_newer <span class="sc">%&gt;%</span> </span>
<span id="cb27-506"><a href="#cb27-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status_out <span class="sc">==</span> <span class="st">"UNCLEAR"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-507"><a href="#cb27-507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-508"><a href="#cb27-508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">nseq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(nseq), <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb27-509"><a href="#cb27-509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-510"><a href="#cb27-510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num_gid =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, genome_id, <span class="st">"_"</span>, seq_num_gid)) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb27-511"><a href="#cb27-511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb27-512"><a href="#cb27-512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb27-513"><a href="#cb27-513" aria-hidden="true" tabindex="-1"></a>mu_fasta_out_3 <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_unenriched_fasta_3, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb27-514"><a href="#cb27-514" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mu_fasta_out_3, <span class="fu">file.path</span>(data_dir_new, <span class="st">"cc-unclear-gid-3.fasta"</span>))</span>
<span id="cb27-515"><a href="#cb27-515" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-516"><a href="#cb27-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-519"><a href="#cb27-519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-520"><a href="#cb27-520" aria-hidden="true" tabindex="-1"></a>unclear_genomes_caused <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir_new, <span class="st">"gid-unclear-3.tsv"</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-521"><a href="#cb27-521" aria-hidden="true" tabindex="-1"></a>mrg_newest_unclear <span class="ot">&lt;-</span> mrg_ug_newest <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>cause) <span class="sc">%&gt;%</span></span>
<span id="cb27-522"><a href="#cb27-522" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status_out <span class="sc">==</span> <span class="st">"UNCLEAR"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-523"><a href="#cb27-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(unclear_genomes_caused <span class="sc">%&gt;%</span> <span class="fu">select</span>(genome_id, cause), <span class="at">by=</span><span class="st">"genome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-524"><a href="#cb27-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb27-525"><a href="#cb27-525" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status =</span> <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"Appears real"</span>, <span class="dv">2</span>, </span>
<span id="cb27-526"><a href="#cb27-526" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"No match"</span>, <span class="fu">pmax</span>(<span class="dv">1</span>, viral_status), viral_status))) <span class="sc">%&gt;%</span></span>
<span id="cb27-527"><a href="#cb27-527" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb27-528"><a href="#cb27-528" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb27-529"><a href="#cb27-529" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb27-530"><a href="#cb27-530" aria-hidden="true" tabindex="-1"></a>mrg_newest_unclear <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cause) <span class="sc">%&gt;%</span> count</span>
<span id="cb27-531"><a href="#cb27-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-532"><a href="#cb27-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-535"><a href="#cb27-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-536"><a href="#cb27-536" aria-hidden="true" tabindex="-1"></a>g_unclear <span class="ot">&lt;-</span> mrg_newest_unclear <span class="sc">%&gt;%</span></span>
<span id="cb27-537"><a href="#cb27-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb27-538"><a href="#cb27-538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb27-539"><a href="#cb27-539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-540"><a href="#cb27-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-541"><a href="#cb27-541" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-542"><a href="#cb27-542" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-543"><a href="#cb27-543" aria-hidden="true" tabindex="-1"></a>g_unclear</span>
<span id="cb27-544"><a href="#cb27-544" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-545"><a href="#cb27-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-546"><a href="#cb27-546" aria-hidden="true" tabindex="-1"></a>The false matches don't appear particularly differentiated from the true matches by alignment score, either using Bowtie2 (above) or BLAST (below):</span>
<span id="cb27-547"><a href="#cb27-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-550"><a href="#cb27-550" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-551"><a href="#cb27-551" aria-hidden="true" tabindex="-1"></a>blast_results_seqid <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb27-552"><a href="#cb27-552" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(qseqid, <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span>), <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-553"><a href="#cb27-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">as.integer</span>(seq_num), <span class="at">read_pair =</span> <span class="fu">as.integer</span>(read_pair), <span class="at">sample=</span><span class="fu">fct_inorder</span>(sample)) <span class="sc">%&gt;%</span></span>
<span id="cb27-554"><a href="#cb27-554" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, seq_id, <span class="at">taxid_bowtie =</span> taxid), <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) </span>
<span id="cb27-555"><a href="#cb27-555" aria-hidden="true" tabindex="-1"></a>blast_results_unclear <span class="ot">&lt;-</span> blast_results_seqid <span class="sc">%&gt;%</span></span>
<span id="cb27-556"><a href="#cb27-556" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(mrg_newest_unclear <span class="sc">%&gt;%</span> <span class="fu">select</span>(seq_id, viral_status, viral_status_out, cause), <span class="at">by =</span> <span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-557"><a href="#cb27-557" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(viral)</span>
<span id="cb27-558"><a href="#cb27-558" aria-hidden="true" tabindex="-1"></a><span class="co"># First check if any BLAST taxid is a descendent of the corresponding Bowtie taxid, or vice versa</span></span>
<span id="cb27-559"><a href="#cb27-559" aria-hidden="true" tabindex="-1"></a>taxid_dec_bowtie <span class="ot">&lt;-</span> blast_results_unclear <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid_bowtie) <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(expand_taxids, <span class="at">nodes=</span>tax_nodes) <span class="sc">%&gt;%</span> <span class="fu">setNames</span>(<span class="fu">unique</span>(blast_results_unclear<span class="sc">$</span>taxid_bowtie))</span>
<span id="cb27-560"><a href="#cb27-560" aria-hidden="true" tabindex="-1"></a>taxid_dec_blast <span class="ot">&lt;-</span> blast_results_unclear <span class="sc">%&gt;%</span> <span class="fu">pull</span>(staxid) <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(expand_taxids, <span class="at">nodes=</span>tax_nodes) <span class="sc">%&gt;%</span> <span class="fu">setNames</span>(<span class="fu">unique</span>(blast_results_unclear<span class="sc">$</span>staxid))</span>
<span id="cb27-561"><a href="#cb27-561" aria-hidden="true" tabindex="-1"></a>match_1 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(blast_results_unclear), <span class="cf">function</span>(n) <span class="fu">as.numeric</span>(blast_results_unclear<span class="sc">$</span>staxid[n]) <span class="sc">%in%</span> taxid_dec_bowtie[[<span class="fu">as.character</span>(blast_results_unclear<span class="sc">$</span>taxid_bowtie[n])]])</span>
<span id="cb27-562"><a href="#cb27-562" aria-hidden="true" tabindex="-1"></a>match_2 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(blast_results_unclear), <span class="cf">function</span>(n) <span class="fu">as.numeric</span>(blast_results_unclear<span class="sc">$</span>taxid_bowtie[n]) <span class="sc">%in%</span> taxid_dec_bowtie[[<span class="fu">as.character</span>(blast_results_unclear<span class="sc">$</span>staxid[n])]])</span>
<span id="cb27-563"><a href="#cb27-563" aria-hidden="true" tabindex="-1"></a>match_any <span class="ot">&lt;-</span> match_1 <span class="sc">|</span> match_2</span>
<span id="cb27-564"><a href="#cb27-564" aria-hidden="true" tabindex="-1"></a>blast_results_matched <span class="ot">&lt;-</span> blast_results_unclear <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid_match =</span> match_any)</span>
<span id="cb27-565"><a href="#cb27-565" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-566"><a href="#cb27-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-569"><a href="#cb27-569" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-570"><a href="#cb27-570" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb27-571"><a href="#cb27-571" aria-hidden="true" tabindex="-1"></a><span class="co"># Otherwise, take the highest-scoring, longest viral match</span></span>
<span id="cb27-572"><a href="#cb27-572" aria-hidden="true" tabindex="-1"></a>blast_results_single <span class="ot">&lt;-</span> blast_results_matched <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, seq_num, read_pair) <span class="sc">%&gt;%</span></span>
<span id="cb27-573"><a href="#cb27-573" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">length =</span> <span class="fu">as.numeric</span>(length)) <span class="sc">%&gt;%</span></span>
<span id="cb27-574"><a href="#cb27-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid_match <span class="sc">==</span> <span class="fu">max</span>(taxid_match)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb27-575"><a href="#cb27-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb27-576"><a href="#cb27-576" aria-hidden="true" tabindex="-1"></a>brs_out <span class="ot">&lt;-</span> blast_results_single <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, bitscore, viral_status_out) <span class="sc">%&gt;%</span></span>
<span id="cb27-577"><a href="#cb27-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"viral_status_out"</span>), <span class="at">names_from =</span> <span class="st">"read_pair"</span>, </span>
<span id="cb27-578"><a href="#cb27-578" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="st">"bitscore"</span>, <span class="at">names_prefix =</span> <span class="st">"read_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-579"><a href="#cb27-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">read_1 =</span> <span class="fu">replace_na</span>(read_1, <span class="dv">0</span>), <span class="at">read_2 =</span> <span class="fu">replace_na</span>(read_2, <span class="dv">0</span>))</span>
<span id="cb27-580"><a href="#cb27-580" aria-hidden="true" tabindex="-1"></a>g_brs <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(brs_out, <span class="fu">aes</span>(<span class="at">x=</span>read_1, <span class="at">y=</span>read_2, <span class="at">color =</span> viral_status_out)) <span class="sc">+</span> </span>
<span id="cb27-581"><a href="#cb27-581" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>, <span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb27-582"><a href="#cb27-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Best viral bitscore (forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">160</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">160</span>,<span class="dv">40</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-583"><a href="#cb27-583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Best viral bitscore (reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">160</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">160</span>,<span class="dv">40</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb27-584"><a href="#cb27-584" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb27-585"><a href="#cb27-585" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb27-586"><a href="#cb27-586" aria-hidden="true" tabindex="-1"></a>g_brs</span>
<span id="cb27-587"><a href="#cb27-587" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-588"><a href="#cb27-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-589"><a href="#cb27-589" aria-hidden="true" tabindex="-1"></a>Overall, I think pulling these out for manual inspection is currently the right call. But if we're forced to classify them automatically, counting them as true viral matches is probably the better bet.</span>
<span id="cb27-590"><a href="#cb27-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-591"><a href="#cb27-591" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human viral reads in Crits-Christoph (2021): Final assessment</span></span>
<span id="cb27-592"><a href="#cb27-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-593"><a href="#cb27-593" aria-hidden="true" tabindex="-1"></a>Now that I've settled on a pipeline and downstream analysis process I'm happy with, we can return to the question of overall human-viral abundance and composition in Crits-Christoph (2021). I'll use a Bowtie2 alignment score cutoff of 20 here, as this is consistent with previous studies and gives good F1 scores in the tests above.</span>
<span id="cb27-594"><a href="#cb27-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-595"><a href="#cb27-595" aria-hidden="true" tabindex="-1"></a>Using this cutoff, we find a total of 109868/8716917 human-viral reads in panel-enriched samples ($1.26 \times 10^{-2}$ , about 1 in 80), and 4064/297690777 in unenriched samples ($1.37 \times 10^{-5}$ , about 1 in 73,000). Unsurprisingly, enriched samples show much higher overall human-viral abundance than unenriched samples; however, even unenriched samples show much higher relative abundance than <span class="co">[</span><span class="ot">our previous BMC sludge sequences</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2023-12-19_project-runway-bmc-rna.html)</span> ( $\sim 3 \times 10^{-7}$):</span>
<span id="cb27-596"><a href="#cb27-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-599"><a href="#cb27-599" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-600"><a href="#cb27-600" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb27-601"><a href="#cb27-601" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_old, <span class="st">"qc_basic_stats.tsv"</span>)</span>
<span id="cb27-602"><a href="#cb27-602" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-603"><a href="#cb27-603" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb27-604"><a href="#cb27-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-605"><a href="#cb27-605" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb27-606"><a href="#cb27-606" aria-hidden="true" tabindex="-1"></a>hv_reads_newest_cut <span class="ot">&lt;-</span> hv_reads_newest <span class="sc">%&gt;%</span></span>
<span id="cb27-607"><a href="#cb27-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb27-608"><a href="#cb27-608" aria-hidden="true" tabindex="-1"></a>hv_reads_newest_counts <span class="ot">&lt;-</span> hv_reads_newest_cut <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, enrichment) <span class="sc">%&gt;%</span> <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb27-609"><a href="#cb27-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-610"><a href="#cb27-610" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge</span></span>
<span id="cb27-611"><a href="#cb27-611" aria-hidden="true" tabindex="-1"></a>hv_reads_ra <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(hv_reads_newest_counts, read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-612"><a href="#cb27-612" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb27-613"><a href="#cb27-613" aria-hidden="true" tabindex="-1"></a>hv_reads_total <span class="ot">&lt;-</span> hv_reads_ra <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(enrichment) <span class="sc">%&gt;%</span></span>
<span id="cb27-614"><a href="#cb27-614" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw)) <span class="sc">%&gt;%</span></span>
<span id="cb27-615"><a href="#cb27-615" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">paste</span>(<span class="st">"All"</span>, <span class="fu">str_to_lower</span>(enrichment)), <span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb27-616"><a href="#cb27-616" aria-hidden="true" tabindex="-1"></a>hv_reads_bound <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_reads_ra, hv_reads_total) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(enrichment)</span>
<span id="cb27-617"><a href="#cb27-617" aria-hidden="true" tabindex="-1"></a>hv_reads_bound<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">fct_inorder</span>(hv_reads_bound<span class="sc">$</span>sample)</span>
<span id="cb27-618"><a href="#cb27-618" aria-hidden="true" tabindex="-1"></a>g_phv <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_reads_bound, <span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads_hv, <span class="at">color=</span>enrichment)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb27-619"><a href="#cb27-619" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb27-620"><a href="#cb27-620" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name=</span><span class="st">"Panel enrichment"</span>) <span class="sc">+</span></span>
<span id="cb27-621"><a href="#cb27-621" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>enrichment, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span> </span>
<span id="cb27-622"><a href="#cb27-622" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb27-623"><a href="#cb27-623" aria-hidden="true" tabindex="-1"></a>g_phv</span>
<span id="cb27-624"><a href="#cb27-624" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-625"><a href="#cb27-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-626"><a href="#cb27-626" aria-hidden="true" tabindex="-1"></a>In comparison, the old pipeline returns an estimated overall relative abundance of human-infecting viruses in unenriched samples of roughly $3.1 \times 10^{-6}$, nearly 5 times lower.</span>
<span id="cb27-627"><a href="#cb27-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-628"><a href="#cb27-628" aria-hidden="true" tabindex="-1"></a>Digging into individual viruses, we see large but inconsistent differences in relative abundance between enriched and unenriched samples:</span>
<span id="cb27-629"><a href="#cb27-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-632"><a href="#cb27-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-633"><a href="#cb27-633" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb27-634"><a href="#cb27-634" aria-hidden="true" tabindex="-1"></a><span class="co"># Import viral genera</span></span>
<span id="cb27-635"><a href="#cb27-635" aria-hidden="true" tabindex="-1"></a>viral_taxids_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_new, <span class="st">"viral-taxids.tsv"</span>)</span>
<span id="cb27-636"><a href="#cb27-636" aria-hidden="true" tabindex="-1"></a>viral_taxids <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxids_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-637"><a href="#cb27-637" aria-hidden="true" tabindex="-1"></a>viral_genera <span class="ot">&lt;-</span> viral_taxids <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"genus"</span>)</span>
<span id="cb27-638"><a href="#cb27-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-639"><a href="#cb27-639" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique name for each viral genus</span></span>
<span id="cb27-640"><a href="#cb27-640" aria-hidden="true" tabindex="-1"></a>viral_genera_unique <span class="ot">&lt;-</span> viral_genera <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb27-641"><a href="#cb27-641" aria-hidden="true" tabindex="-1"></a>viral_genera_duplicate <span class="ot">&lt;-</span> viral_genera <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb27-642"><a href="#cb27-642" aria-hidden="true" tabindex="-1"></a>viral_genera_valid_1 <span class="ot">&lt;-</span> viral_genera_duplicate <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"virus$"</span>, name))</span>
<span id="cb27-643"><a href="#cb27-643" aria-hidden="true" tabindex="-1"></a>viral_genera_unique <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_genera_unique, viral_genera_valid_1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb27-644"><a href="#cb27-644" aria-hidden="true" tabindex="-1"></a>viral_genera_valid_2 <span class="ot">&lt;-</span> viral_genera_valid_1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> taxid <span class="sc">%in%</span> viral_genera_unique<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-645"><a href="#cb27-645" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">" "</span>, name))</span>
<span id="cb27-646"><a href="#cb27-646" aria-hidden="true" tabindex="-1"></a>viral_genera_unique <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_genera_unique, viral_genera_valid_2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb27-647"><a href="#cb27-647" aria-hidden="true" tabindex="-1"></a>viral_genera_valid_3 <span class="ot">&lt;-</span> viral_genera_valid_2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> taxid <span class="sc">%in%</span> viral_genera_unique<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-648"><a href="#cb27-648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb27-649"><a href="#cb27-649" aria-hidden="true" tabindex="-1"></a>viral_genera_unique <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_genera_unique, viral_genera_valid_3 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb27-650"><a href="#cb27-650" aria-hidden="true" tabindex="-1"></a>viral_genera_valid_4 <span class="ot">&lt;-</span> viral_genera_duplicate <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span> taxid <span class="sc">%in%</span> viral_genera_unique<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-651"><a href="#cb27-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb27-652"><a href="#cb27-652" aria-hidden="true" tabindex="-1"></a>viral_genera_unique <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_genera_unique, viral_genera_valid_4 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span>))</span>
<span id="cb27-653"><a href="#cb27-653" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(viral_genera_unique, <span class="fu">file.path</span>(data_dir_new, <span class="st">"viral-genera-unique.tsv"</span>))</span>
<span id="cb27-654"><a href="#cb27-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-655"><a href="#cb27-655" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral genera for HV reads</span></span>
<span id="cb27-656"><a href="#cb27-656" aria-hidden="true" tabindex="-1"></a>high_ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"class"</span>, <span class="st">"family"</span>, <span class="st">"kingdom"</span>, <span class="st">"order"</span>, <span class="st">"phylum"</span>, <span class="st">"subfamily"</span>, <span class="st">"suborder"</span>, <span class="st">"subphylum"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb27-657"><a href="#cb27-657" aria-hidden="true" tabindex="-1"></a>hv_read_db <span class="ot">&lt;-</span> hv_reads_newest_cut</span>
<span id="cb27-658"><a href="#cb27-658" aria-hidden="true" tabindex="-1"></a>tax_nodes_cut <span class="ot">&lt;-</span> <span class="fu">rename</span>(tax_nodes, <span class="at">taxid =</span> child_taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">%in%</span> v_taxids)</span>
<span id="cb27-659"><a href="#cb27-659" aria-hidden="true" tabindex="-1"></a>hv_read_genus <span class="ot">&lt;-</span> hv_read_db <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(viral_genera_unique, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb27-660"><a href="#cb27-660" aria-hidden="true" tabindex="-1"></a>hv_read_nogenus <span class="ot">&lt;-</span> hv_read_db <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>taxid <span class="sc">%in%</span> viral_genera_unique<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-661"><a href="#cb27-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(tax_nodes_cut, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-662"><a href="#cb27-662" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(taxid), <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks)</span>
<span id="cb27-663"><a href="#cb27-663" aria-hidden="true" tabindex="-1"></a><span class="co">#cat(nrow(hv_read_db), nrow(hv_read_genus), nrow(hv_read_nogenus), nrow(hv_read_genus)+nrow(hv_read_nogenus), "\n")</span></span>
<span id="cb27-664"><a href="#cb27-664" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span>(<span class="fu">nrow</span>(hv_read_nogenus) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb27-665"><a href="#cb27-665" aria-hidden="true" tabindex="-1"></a>  hv_read_genus <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_read_genus, hv_read_nogenus <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(viral_genera_unique, <span class="at">by=</span><span class="st">"taxid"</span>))</span>
<span id="cb27-666"><a href="#cb27-666" aria-hidden="true" tabindex="-1"></a>  hv_read_nogenus <span class="ot">&lt;-</span> hv_read_nogenus <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>taxid <span class="sc">%in%</span> viral_genera_unique<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>rank, <span class="sc">-</span>parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-667"><a href="#cb27-667" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(tax_nodes_cut, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb27-668"><a href="#cb27-668" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(taxid), <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks)</span>
<span id="cb27-669"><a href="#cb27-669" aria-hidden="true" tabindex="-1"></a>  <span class="co">#cat(nrow(hv_read_db), nrow(hv_read_genus), nrow(hv_read_nogenus), nrow(hv_read_genus)+nrow(hv_read_nogenus), "\n")</span></span>
<span id="cb27-670"><a href="#cb27-670" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-671"><a href="#cb27-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-672"><a href="#cb27-672" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names for higher-ranked assignments</span></span>
<span id="cb27-673"><a href="#cb27-673" aria-hidden="true" tabindex="-1"></a>smatch <span class="ot">&lt;-</span> hv_read_db<span class="sc">$</span>seq_id <span class="sc">%in%</span> hv_read_genus<span class="sc">$</span>seq_id</span>
<span id="cb27-674"><a href="#cb27-674" aria-hidden="true" tabindex="-1"></a>hv_read_highrank <span class="ot">&lt;-</span> hv_read_db[<span class="sc">!</span>smatch,] <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(viral_taxids <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(taxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>), <span class="at">by =</span> <span class="st">"taxid"</span>)</span>
<span id="cb27-675"><a href="#cb27-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-676"><a href="#cb27-676" aria-hidden="true" tabindex="-1"></a><span class="co"># Count viral genera (&amp; unassigned viruses)</span></span>
<span id="cb27-677"><a href="#cb27-677" aria-hidden="true" tabindex="-1"></a>hv_counts_wide <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(hv_read_genus, hv_read_highrank) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, enrichment) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb27-678"><a href="#cb27-678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"name"</span>, <span class="at">names_from =</span> <span class="st">"enrichment"</span>, <span class="at">values_from =</span> <span class="st">"n"</span>, <span class="at">values_fill =</span> <span class="dv">0</span>) </span>
<span id="cb27-679"><a href="#cb27-679" aria-hidden="true" tabindex="-1"></a>hv_counts <span class="ot">&lt;-</span> hv_counts_wide <span class="sc">%&gt;%</span></span>
<span id="cb27-680"><a href="#cb27-680" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>name, <span class="at">names_to =</span> <span class="st">"enrichment"</span>, <span class="at">values_to =</span> <span class="st">"n_reads_virus"</span>)</span>
<span id="cb27-681"><a href="#cb27-681" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-682"><a href="#cb27-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-685"><a href="#cb27-685" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-686"><a href="#cb27-686" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb27-687"><a href="#cb27-687" aria-hidden="true" tabindex="-1"></a>hv_counts_fraction <span class="ot">&lt;-</span> hv_counts <span class="sc">%&gt;%</span></span>
<span id="cb27-688"><a href="#cb27-688" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(hv_reads_total <span class="sc">%&gt;%</span> <span class="fu">select</span>(enrichment, n_reads_hv, n_reads_raw), <span class="at">by=</span><span class="st">"enrichment"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-689"><a href="#cb27-689" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_virus_all =</span> n_reads_virus<span class="sc">/</span>n_reads_raw, <span class="at">p_reads_virus_hv =</span> n_reads_virus<span class="sc">/</span>n_reads_hv)</span>
<span id="cb27-690"><a href="#cb27-690" aria-hidden="true" tabindex="-1"></a>g_hv_counts <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(hv_counts_fraction, <span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>p_reads_virus_all, <span class="at">color=</span>enrichment)) <span class="sc">+</span></span>
<span id="cb27-691"><a href="#cb27-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb27-692"><a href="#cb27-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">name =</span> <span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb27-693"><a href="#cb27-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>, <span class="at">name=</span><span class="st">"Panel enrichment"</span>) <span class="sc">+</span></span>
<span id="cb27-694"><a href="#cb27-694" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>), <span class="at">aspect.ratio =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">5</span>)</span>
<span id="cb27-695"><a href="#cb27-695" aria-hidden="true" tabindex="-1"></a>g_hv_counts</span>
<span id="cb27-696"><a href="#cb27-696" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-697"><a href="#cb27-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-698"><a href="#cb27-698" aria-hidden="true" tabindex="-1"></a>As expected, viruses included in the respiratory virus panel see large increases in relative abundance in the enriched vs the unenriched samples, with the largest relative increases seen for *Bocaparvovirus* (Human bocavirus 1, 2c, 3), *Betacoronavirus* (SARS-CoV-2, OC43, HKU1), and *Mastadenovirus* (Human adenovirus B1, C2, E4). Confusingly, Orthopoxvirus, Cytomegalovirus and Gemygorvirus all also show substantially increased relative abundance, even though as far as I can tell there are no viruses from those genera in the <span class="co">[</span><span class="ot">Illumina panel</span><span class="co">](https://www.illumina.com/ko-kr/products/by-type/sequencing-kits/library-prep-kits/respiratory-virus-oligo-panel.html)</span>. Numerous other viruses show weaker enrichment; even norovirus shows moderate (\~4x) enrichment in the enriched vs the unenriched samples.</span>
<span id="cb27-699"><a href="#cb27-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-700"><a href="#cb27-700" aria-hidden="true" tabindex="-1"></a>Conversely, a number of viruses are found in the unenriched samples that are absent in the enriched samples, including *Rotavirus*, *Flavivirus*, *Parvovirus*, and various papillomaviruses and polyomaviruses. One natural hypothesis for this is that these viruses were excluded by the enrichment panel and so had their relative abundance reduced; however, the enrichment observed for various other not-in-panel viruses calls this into question. An alternative hypothesis is that this difference is simply a consequence of the much deeper sequencing conducted on the unenriched samples (geometric mean of \~340k read pairs per enriched sample vs 42M read pairs per unenriched sample).</span>
<span id="cb27-701"><a href="#cb27-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-704"><a href="#cb27-704" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-705"><a href="#cb27-705" aria-hidden="true" tabindex="-1"></a>hv_ra_wide <span class="ot">&lt;-</span> hv_counts_fraction <span class="sc">%&gt;%</span> <span class="fu">select</span>(name, enrichment, p_reads_virus_all) <span class="sc">%&gt;%</span></span>
<span id="cb27-706"><a href="#cb27-706" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"name"</span>, <span class="at">names_from =</span> <span class="st">"enrichment"</span>, <span class="at">values_from =</span> <span class="st">"p_reads_virus_all"</span>, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-707"><a href="#cb27-707" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ra_enriched =</span> Enriched, <span class="at">ra_unenriched =</span> Unenriched) <span class="sc">%&gt;%</span></span>
<span id="cb27-708"><a href="#cb27-708" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">relative_enrichment =</span> <span class="fu">log10</span>(ra_enriched<span class="sc">/</span>ra_unenriched)) <span class="sc">%&gt;%</span></span>
<span id="cb27-709"><a href="#cb27-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(relative_enrichment), <span class="fu">desc</span>(ra_enriched), ra_unenriched)</span>
<span id="cb27-710"><a href="#cb27-710" aria-hidden="true" tabindex="-1"></a>hv_ra_wide</span>
<span id="cb27-711"><a href="#cb27-711" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-712"><a href="#cb27-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-713"><a href="#cb27-713" aria-hidden="true" tabindex="-1"></a>At this point, I'm satisfied with my workflow's ability to produce usable results on the Crits-Christoph data. Next, I'll apply this updated workflow to another previously-published WMGS dataset, likely <span class="co">[</span><span class="ot">Rothman et al. (2021)</span><span class="co">](https://doi.org/10.1128/AEM.01448-21)</span>.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>