<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-05-07">
<title>Will’s Public NAO Notebook - Workflow analysis of Maritz et al.&nbsp;(2019)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Maritz et al.&nbsp;(2019)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Wastewater from NYC.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">May 7, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>The final dataset from the P2RA dataset I want to analyze here is <a href="https://www.nature.com/articles/s41467-022-34312-7">Munk et al.&nbsp;(2022)</a>, an enormous dataset of &gt;1,000 raw influent samples from 101 countries collected between 2016 and 2019. As in previous DNA studies like Bengtsson-Palme, samples were centrifuged and only the pellet was retained for sequencing, so we expect viral abundance to be low; nevertheless, this is the largest and most comprehensive DNA wastewater dataset we’ve been able to find to date, so it’s worth having a look at what’s in it. The pellet from each sample was resuspended, was homogenized with bead-beating, underwent DNA extraction and library prep, and was sequenced using Illumina technology; earlier samples were sequenced on an Illumina HiSeq3000, while later samples were sequenced on a NovaSeq6000, both with 2x150bp reads.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The Munk data comprised 1,189 total samples, of which 1,185 had complete metadata. These samples came from 101 countries, with the largest number of samples coming from the USA, Canada, and Denmark:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Importing the data is a bit more complicated this time as the samples are split across seven (!) pipeline runs</span></span>
<span><span class="va">data_dir_base</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-05-06_munk"</span></span>
<span><span class="va">data_dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.dirs</a></span><span class="op">(</span><span class="va">data_dir_base</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data input paths</span></span>
<span><span class="va">libraries_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_basic_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_adapter_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_base_stats.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">ctypes</span> <span class="op">&lt;-</span> <span class="fu">cols</span><span class="op">(</span>date<span class="op">=</span><span class="st">"D"</span>, .default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span></span>
<span><span class="va">libraries_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">libraries_paths</span>, <span class="va">read_csv</span>, col_types <span class="op">=</span> <span class="va">ctypes</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">bind_rows</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Add missing dates</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>date <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"ERR4682809"</span>, <span class="fu">as_date</span><span class="op">(</span><span class="st">"2018-06-01"</span><span class="op">)</span>, <span class="va">date</span><span class="op">)</span>,</span>
<span>         date <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"ERR4682803"</span>, <span class="fu">as_date</span><span class="op">(</span><span class="st">"2018-06-01"</span><span class="op">)</span>, <span class="va">date</span><span class="op">)</span>,</span>
<span>         date <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">"ERR2683170"</span>, <span class="fu">as_date</span><span class="op">(</span><span class="st">"2017-06-01"</span><span class="op">)</span>, <span class="va">date</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Filter samples with unknown dates</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">date</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">date</span>, <span class="va">country</span>, <span class="va">city</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>, date<span class="op">=</span><span class="fu">as_date</span><span class="op">(</span><span class="va">date</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_countries</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>country<span class="op">=</span><span class="fu">fct_inorder</span><span class="op">(</span><span class="va">country</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_countries</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">sample_countries</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">country</span>, y<span class="op">=</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Samples"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">120</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">200</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_countries</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-countries-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>The 1,185 libraries included in this analysis varied dramatically in size, from 33,554 read pairs to over 123 million. The mean number of read pairs per library was 33.5M, and the dataset as a whole comprised 39.7B read pairs and almost 12 terabases of sequence:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">import_basic</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>           sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">import_basic_paired</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">import_basic</span><span class="op">(</span><span class="va">paths</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic</span><span class="op">(</span><span class="va">basic_stats_paths</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">adapter_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_base_stats_paths</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">import_basic_paired</span><span class="op">(</span><span class="va">quality_seq_stats_paths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify small and large datasets</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">libraries_small</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">&lt;=</span> <span class="fl">1e7</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">library</span><span class="op">)</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>small <span class="op">=</span> <span class="va">library</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">libraries_small</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>small <span class="op">=</span> <span class="va">library</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">libraries_small</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>small <span class="op">=</span> <span class="va">library</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">libraries_small</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>small <span class="op">=</span> <span class="va">library</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">libraries_small</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>small <span class="op">=</span> <span class="va">library</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">libraries_small</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get key values for readout</span></span>
<span><span class="va">raw_read_counts</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>rmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, rmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            rmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, </span>
<span>            rtot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            btot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span>,</span>
<span>            dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">date</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">date</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">date</span>, y<span class="op">=</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_date</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space<span class="op">=</span><span class="st">"free_x"</span>, switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>Adapter levels were high, read qualities were variable (in definite need of trimming) and duplicate levels were moderate:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>,</span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">500</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>About 6% of reads on average were lost during cleaning, and a further 10% during deduplication; however, in both cases a minority of samples lost much larger read fractions. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span>, </span>
<span>         <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Stage"],"name":[1],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[2],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"Trimming & filtering","2":"2.1-66.3 (mean 6)","3":"2.1-66.3 (mean 6)"},{"1":"Deduplication","2":"4.9-67.5 (mean 16.4)","3":"0.1-21.2 (mean 10.4)"},{"1":"Initial ribodepletion","2":"5.2-67.6 (mean 16.7)","3":"0.1-0.9 (mean 0.4)"},{"1":"Secondary ribodepletion","2":"5.3-67.6 (mean 16.8)","3":"0-0.3 (mean 0.1)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_base</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span>, data<span class="op">=</span><span class="va">basic_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="va">g_stage_base</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span>, data<span class="op">=</span><span class="va">n_reads_rel</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>As usual, data cleaning was very successful at removing adapters and improving read qualities:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, cleaning + deduplication was mostly effective at reducing measured duplicate levels, though a few samples retained high measured duplicate levels throughout the pipeline:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_base</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span>, data<span class="op">=</span><span class="va">basic_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span>, data<span class="op">=</span><span class="va">basic_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">classifications</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import composition data</span></span>
<span><span class="va">comp_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"taxonomic_composition.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">comp_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">classification</span>, levels <span class="op">=</span> <span class="va">classifications</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">sample</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_xblank</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.ticks.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data <span class="op">=</span> <span class="va">comp</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">comp_minor</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">9</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">comp_minor</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">comp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>Classification<span class="op">=</span><span class="va">classification</span>, </span>
<span>         `Read Fraction`<span class="op">=</span><span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">Classification</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Classification"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Read Fraction"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"Excluded","2":"26-94% (mean 68%)"},{"1":"Ribosomal","2":"0.1-1% (mean 0.44%)"},{"1":"Bacterial","2":"5.2-73% (mean 31%)"},{"1":"Archaeal","2":"0.0048-4.6% (mean 0.15%)"},{"1":"Viral","2":"0.0044-6.9% (mean 0.33%)"},{"1":"Human","2":"0.0058-5.3% (mean 0.17%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>As in previous DNA datasets, the vast majority of classified reads were bacterial in origin. Viral fraction averaged 0.33%, higher than in other DNA wastewater datasets I’ve looked at, and reached &gt;1% in 35 samples. As is common for DNA wastewater data, viral reads were overwhelmingly dominated by <em>Caudoviricetes</em> phages, though <em>Quintoviricetes</em> (parvoviruses) also showed significant prevalence in some samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># # Get Kraken reports</span></span>
<span><span class="co"># reports_paths &lt;- file.path(data_dirs, "kraken_reports.tsv.gz")</span></span>
<span><span class="co"># reports &lt;- lapply(reports_paths, read_tsv, show_col_types = FALSE) %&gt;% bind_rows %&gt;%</span></span>
<span><span class="co">#   inner_join(libraries, by="sample")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # Filter to viral taxa</span></span>
<span><span class="co"># kraken_reports_viral &lt;- filter(reports, taxid %in% viral_taxa$taxid) %&gt;%</span></span>
<span><span class="co">#   group_by(sample) %&gt;%</span></span>
<span><span class="co">#   mutate(p_reads_viral = n_reads_clade/n_reads_clade[1])</span></span>
<span><span class="co"># kraken_reports_viral_cleaned &lt;- kraken_reports_viral %&gt;%</span></span>
<span><span class="co">#   inner_join(libraries, by="sample") %&gt;%</span></span>
<span><span class="co">#   select(-pc_reads_total, -n_reads_direct, -contains("minimizers")) %&gt;%</span></span>
<span><span class="co">#   select(name, taxid, p_reads_viral, n_reads_clade, everything())</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># viral_classes &lt;- kraken_reports_viral_cleaned %&gt;% filter(rank == "C")</span></span>
<span></span>
<span><span class="va">viral_classes_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"viral_classes.tsv.gz"</span><span class="op">)</span></span>
<span><span class="co"># write_tsv(viral_classes, viral_classes_path)</span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_classes_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span></span>
<span><span class="co"># Identify major viral classes</span></span>
<span><span class="va">viral_classes_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="va">major_threshold</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_classes_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_minor</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_classes_major</span>, <span class="va">viral_classes_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_classes_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_classes</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_classes_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">g_classes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/viral-class-composition-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples. A grand total of 331,452 reads were identified as putatively human-viral:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import HV read data</span></span>
<span><span class="va">hv_reads_filtered_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">hv_reads_filtered_paths</span>, <span class="va">read_tsv</span>,</span>
<span>                            show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collapse multi-entry sequences</span></span>
<span><span class="va">rmax</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">max</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">collapse</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">x</span> <span class="op">==</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">x</span>, collapse<span class="op">=</span><span class="st">"/"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>            genome_id <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span>,</span>
<span>            taxid_best <span class="op">=</span> <span class="va">taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            best_alignment_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_fwd</span><span class="op">)</span>,</span>
<span>            best_alignment_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">best_alignment_score_rev</span><span class="op">)</span>,</span>
<span>            query_len_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_fwd</span><span class="op">)</span>,</span>
<span>            query_len_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">query_len_rev</span><span class="op">)</span>,</span>
<span>            query_seq_fwd <span class="op">=</span> <span class="va">query_seq_fwd</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_fwd</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            query_seq_rev <span class="op">=</span> <span class="va">query_seq_rev</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">query_seq_rev</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            classified <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">classified</span><span class="op">)</span>,</span>
<span>            assigned_name <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">assigned_name</span><span class="op">)</span>,</span>
<span>            assigned_taxid_best <span class="op">=</span> <span class="va">assigned_taxid</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>            assigned_taxid <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">assigned_taxid</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            assigned_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">assigned_hv</span><span class="op">)</span>,</span>
<span>            hit_hv <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">hit_hv</span><span class="op">)</span>,</span>
<span>            encoded_hits <span class="op">=</span> <span class="fu">collapse</span><span class="op">(</span><span class="va">encoded_hits</span><span class="op">)</span>,</span>
<span>            adj_score_fwd <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>,</span>
<span>            adj_score_rev <span class="op">=</span> <span class="fu">rmax</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span></span>
<span>            <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span>,</span>
<span>         highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot results</span></span>
<span><span class="va">geom_vhist</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">geom_histogram</span>, binwidth<span class="op">=</span><span class="fl">5</span>, boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">g_vhist_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="va">theme_base</span> </span>
<span><span class="va">g_vhist_0</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg</span><span class="op">)</span></span>
<span><span class="va">g_vhist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-hv-scores-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>BLASTing these reads against nt, we find that the pipeline performs well, with only a single high-scoring false-positive read:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import paired BLAST results</span></span>
<span><span class="va">blast_paired_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dirs</span>, <span class="st">"hv_hits_blast_paired.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">blast_paired_paths</span>, <span class="va">read_tsv</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_paired</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">match_taxid</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxid_1</span>, <span class="va">taxid_2</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"/"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"/"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mapply.html">mapply</a></span><span class="op">(</span><span class="va">grepl</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="va">taxid_1</span>, <span class="st">"$"</span><span class="op">)</span>, <span class="va">taxid_2</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">p1</span><span class="op">|</span><span class="va">p2</span><span class="op">|</span><span class="va">p3</span>, <span class="cn">NULL</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">adj_score_max</span><span class="op">)</span></span>
<span><span class="va">blast_assign</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">blast_viral</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="fu">match_taxid</span><span class="op">(</span><span class="va">staxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_out</span> <span class="op">&lt;-</span> <span class="va">blast_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg</span>, <span class="va">blast_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_vhist_1</span> <span class="op">&lt;-</span> <span class="va">g_vhist_base</span> <span class="op">+</span> <span class="fu">geom_vhist</span><span class="op">(</span>data<span class="op">=</span><span class="va">mrg_blast</span>, mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>fill<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span></span>
<span><span class="va">g_vhist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-blast-results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>My usual disjunctive score threshold of 20 gave precision, sensitivity, and F1 scores all &gt;99%:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> </span>
<span>    <span class="fu">mutate</span><span class="op">(</span>retain_score <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>           retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span><span class="op">)</span></span>
<span>  <span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="va">threshold</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Adjusted Score Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-f1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span>Threshold<span class="op">=</span><span class="va">threshold</span>, Metric<span class="op">=</span><span class="va">metric</span>, Value<span class="op">=</span><span class="va">value</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Threshold"],"name":[1],"type":["int"],"align":["right"]},{"label":["Metric"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Value"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"20","2":"sensitivity","3":"0.9941336"},{"1":"20","2":"specificity","3":"0.8860624"},{"1":"20","2":"precision","3":"0.9983620"},{"1":"20","2":"f1","3":"0.9962433"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-overall-relative-abundance" class="level1"><h1>Human-infecting viruses: overall relative abundance</h1>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>taxid_all <span class="op">=</span> <span class="va">taxid</span>, taxid <span class="op">=</span> <span class="va">taxid_best</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_grp</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">country</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, </span>
<span>            n_samples <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span> <span class="st">"All samples"</span><span class="op">)</span></span>
<span><span class="va">read_counts_tot</span> <span class="op">&lt;-</span> <span class="va">read_counts_grp</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>country<span class="op">=</span> <span class="st">"All countries"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_grp</span>, <span class="va">read_counts_tot</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sample</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">libraries</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>, <span class="st">"All samples"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Applying a disjunctive cutoff at S=20 identifies 325,390 read pairs as human-viral. This gives an overall relative HV abundance of <span class="math inline">\(8.19 \times 10^{-6}\)</span>; higher than any other DNA WW dataset I’ve analyzed and competitive with many RNA datasets:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">country</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/plot-hv-ra-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collate past RA values</span></span>
<span><span class="va">ra_past</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brinch"</span>, <span class="fl">3.88e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Bengtsson-Palme"</span>, <span class="fl">8.86e-8</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Ng"</span>, <span class="fl">2.90e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Maritz"</span>, <span class="fl">9.42e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Collate new RA values</span></span>
<span><span class="va">ra_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                  <span class="st">"Munk"</span>, <span class="fl">8.19e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">scale_color_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                 name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">ra_comp</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">ra_past</span>, <span class="va">ra_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">ra_comp</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">dataset</span>, x<span class="op">=</span><span class="va">ra</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/ra-hv-past-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>One potential explanation for the higher HV fraction in the Munk data compared to other DNA WW datasets is the sample location: whereas Brinch, Maritz, Bengtsson-Palme and Ng are all from highly developed economies with good sanitation, Munk includes samples from numerous countries including many with much lower incomes and development scores. To quickly test this, I took the most recent Human Development Index dataset from the UN (2022<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>) and GDP per capita dataset from the World Bank (PPP, 2019). In both cases, there was a weak negative correlation between the development metric and measured human-viral load:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># HDI</span></span>
<span><span class="va">hdi_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"hdi.csv"</span><span class="op">)</span></span>
<span><span class="va">hdi</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">hdi_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">read_counts_hdi</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">read_counts_grp</span>, <span class="va">hdi</span>, by<span class="op">=</span><span class="st">"country"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         log_p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hdi</span> <span class="op">&lt;-</span> <span class="fu">ggscatter</span><span class="op">(</span><span class="va">read_counts_hdi</span>, x<span class="op">=</span><span class="st">"HDI"</span>, y<span class="op">=</span><span class="st">"p_reads_hv"</span>,</span>
<span>                   add <span class="op">=</span> <span class="st">"reg.line"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">stat_cor</span><span class="op">(</span>method<span class="op">=</span><span class="st">"pearson"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span><span class="st">"HDI (2022)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"HV RA"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_hdi</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/dev-metrics-linear-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># GDP</span></span>
<span><span class="va">gdp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir_base</span>, <span class="st">"gdp.csv"</span><span class="op">)</span></span>
<span><span class="va">gdp</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">gdp_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">read_counts_gdp</span> <span class="op">&lt;-</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">read_counts_grp</span>, <span class="va">gdp</span>, by<span class="op">=</span><span class="st">"country"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         log_p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>,</span>
<span>         log_gdp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">gdp_per_capita_ppp</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_gdp</span> <span class="op">&lt;-</span> <span class="fu">ggscatter</span><span class="op">(</span><span class="va">read_counts_gdp</span>, x<span class="op">=</span><span class="st">"log_gdp"</span>, y<span class="op">=</span><span class="st">"p_reads_hv"</span>,</span>
<span>                   add <span class="op">=</span> <span class="st">"reg.line"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">stat_cor</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span><span class="st">"Log GDP per Capita (PPP, Int$, 2019)"</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"1e+"</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_gdp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/dev-metrics-linear-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-taxonomy-and-composition" class="level1"><h1>Human-infecting viruses: taxonomy and composition</h1>
<p>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 1,129 samples met this criterion.</p>
<p>As usual, at the family level, most samples were dominated by <em>Adenoviridae</em>, <em>Polyomaviridae</em> and <em>Papillomaviridae.</em> Three other families, <em>Parvoviridae</em>, <em>Circoviridae</em> and <em>Herpesviridae</em>, also showed substantial prevalence.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">585893</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333922</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334207</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">369960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333924</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687329</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">325455</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333750</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334202</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">197911</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">186938</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333926</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337051</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337043</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334204</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334208</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333928</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337039</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333929</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337042</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334203</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333757</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337050</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333767</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333754</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687363</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687342</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687359</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194441</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334209</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194965</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333930</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337048</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337041</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 9"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337049</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 11"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337044</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 5"</span></span>
<span></span>
<span><span class="co"># Filter samples and add viral taxa information</span></span>
<span><span class="va">samples_keep</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_reads_hv</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">samples_keep</span>, <span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_family</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"family"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Family composition of human-viral reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent families for text</span></span>
<span><span class="va">hv_family_collate</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">taxid_adeno</span> <span class="op">&lt;-</span> <span class="fl">10508</span></span>
<span></span>
<span><span class="co"># Get set of adenoviridae reads</span></span>
<span><span class="va">adeno_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">adeno_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_adeno</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each adenoviridae species</span></span>
<span><span class="va">adeno_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_adeno <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">adeno_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_adeno</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_adeno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">adeno_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adeno_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_adeno</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_adeno_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">adeno_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adenoviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Adenoviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_adeno_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-adeno-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">adeno_species_collate</span> <span class="op">&lt;-</span> <span class="va">adeno_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_adeno</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_polyoma</span> <span class="op">&lt;-</span> <span class="fl">151341</span></span>
<span></span>
<span><span class="co"># Get set of polyomaviridae reads</span></span>
<span><span class="va">polyoma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">polyoma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_polyoma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each polyomaviridae species</span></span>
<span><span class="va">polyoma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_polyoma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">polyoma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_polyoma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">polyoma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">polyoma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_polyoma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_polyoma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">polyoma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Polyomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Polyomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_polyoma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-polyoma-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">polyoma_species_collate</span> <span class="op">&lt;-</span> <span class="va">polyoma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_polyoma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">taxid_papilloma</span> <span class="op">&lt;-</span> <span class="fl">151340</span></span>
<span></span>
<span><span class="co"># Get set of papillomaviridae reads</span></span>
<span><span class="va">papilloma_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">papilloma_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_papilloma</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each papillomaviridae species</span></span>
<span><span class="va">papilloma_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_papilloma <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">papilloma_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_papilloma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papilloma_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">papilloma_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_papilloma</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_papilloma_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">papilloma_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Papillomaviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Papillomaviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_papilloma_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-papilloma-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">papilloma_species_collate</span> <span class="op">&lt;-</span> <span class="va">papilloma_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_papilloma</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_parvo</span> <span class="op">&lt;-</span> <span class="fl">10780</span></span>
<span></span>
<span><span class="co"># Get set of parvoviridae reads</span></span>
<span><span class="va">parvo_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_parvo</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">parvo_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_parvo</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">parvo_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each parvoviridae species</span></span>
<span><span class="va">parvo_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">parvo_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_parvo <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">parvo_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">parvo_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_parvo</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_parvo</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_parvo</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_parvo</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">parvo_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">parvo_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">parvo_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_parvo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_parvo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_parvo</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">parvo_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">parvo_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">parvo_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_parvo</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_parvo_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">parvo_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Parvoviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Parvoviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_parvo_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-parvo-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">parvo_species_collate</span> <span class="op">&lt;-</span> <span class="va">parvo_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_parvo</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_circo</span> <span class="op">&lt;-</span> <span class="fl">39724</span></span>
<span></span>
<span><span class="co"># Get set of circoviridae reads</span></span>
<span><span class="va">circo_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_circo</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">circo_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_circo</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">circo_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each circoviridae species</span></span>
<span><span class="va">circo_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">circo_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_circo <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">circo_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">circo_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_circo</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_circo</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_circo</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_circo</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">circo_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">circo_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">circo_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_circo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_circo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_circo</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">circo_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">circo_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">circo_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_circo</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_circo_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">circo_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Circoviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Circoviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_circo_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-circo-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">circo_species_collate</span> <span class="op">&lt;-</span> <span class="va">circo_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_circo</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">taxid_herpes</span> <span class="op">&lt;-</span> <span class="fl">10292</span></span>
<span></span>
<span><span class="co"># Get set of herpesviridae reads</span></span>
<span><span class="va">herpes_samples</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">herpes_ids</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">taxid_herpes</span>, <span class="va">sample</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_samples</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads for each herpesviridae species</span></span>
<span><span class="va">herpes_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_ids</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_herpes <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">herpes_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_herpes</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, </span>
<span>                               <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>,</span>
<span>            p_reads_herpes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">herpes_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">herpes_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_herpes</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_herpes_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">herpes_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span>, width<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Herpesviridae Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"Species composition of Herpesviridae reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_herpes_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/hv-species-herpes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get most prominent species for text</span></span>
<span><span class="va">herpes_species_collate</span> <span class="op">&lt;-</span> <span class="va">herpes_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_tot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_herpes</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_reads_tot</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define reference genera</span></span>
<span><span class="va">path_genera_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_rna</span>, genome_type<span class="op">=</span><span class="st">"RNA genome"</span><span class="op">)</span>,</span>
<span>                         <span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_dna</span>, genome_type<span class="op">=</span><span class="st">"DNA genome"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count in each sample</span></span>
<span><span class="va">mrg_hv_named_all</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus_all</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named_all</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">n_path_genera</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus_all</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot out and back to add zero lines</span></span>
<span><span class="va">n_path_genera_out</span> <span class="op">&lt;-</span> <span class="va">n_path_genera</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">n_reads_viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"name"</span>, values_from<span class="op">=</span><span class="st">"n_reads_viral"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, names_to<span class="op">=</span><span class="st">"name"</span>, values_to<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Aggregate across dates</span></span>
<span><span class="va">n_path_genera_stype</span> <span class="op">&lt;-</span> <span class="va">n_path_genera_out</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">genome_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="st">"All samples"</span>, location<span class="op">=</span><span class="st">"All locations"</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span>,</span>
<span>         na_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_path_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_path_genera_stype</span>,</span>
<span>                        <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">name</span>, x<span class="op">=</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">genome_type</span><span class="op">~</span><span class="va">.</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_path_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-05-07_munk_files/figure-html/ra-genera-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>This is the final P2RA dataset I needed to analyze before we finish re-doing that analysis for publication, so I’m pretty happy to have it done. In terms of the results, things mostly look similar to other DNA WW datasets I’ve looked at, with the notable difference that the total fraction of human-infecting viruses is significantly higher. I’m still not sure what’s causing this elevation; the methods used in this study don’t seem any different from other studies that got much lower fractions, and the fact that this study sampled from developing countries seems like only a partial explanation.</p>


<!-- -->

</section><div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>
<ol>
<li id="fn1"><p>I wasn’t able to quickly find any HDI datasets other than the most recent one, and it didn’t seem worth doing serious digging for this quick analysis.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb49" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Maritz et al. (2019)"</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Wastewater from NYC."</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-05-07</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preamble</span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a><span class="co"># GGplot themes and scales</span></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>theme_rotate <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>theme_xblank <span class="ot">&lt;-</span> theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>The final dataset from the P2RA dataset I want to analyze here is <span class="co">[</span><span class="ot">Munk et al. (2022)</span><span class="co">](https://www.nature.com/articles/s41467-022-34312-7)</span>, an enormous dataset of <span class="sc">\&gt;</span>1,000 raw influent samples from 101 countries collected between 2016 and 2019. As in previous DNA studies like Bengtsson-Palme, samples were centrifuged and only the pellet was retained for sequencing, so we expect viral abundance to be low; nevertheless, this is the largest and most comprehensive DNA wastewater dataset we've been able to find to date, so it's worth having a look at what's in it. The pellet from each sample was resuspended, was homogenized with bead-beating, underwent DNA extraction and library prep, and was sequenced using Illumina technology; earlier samples were sequenced on an Illumina HiSeq3000, while later samples were sequenced on a NovaSeq6000, both with 2x150bp reads.</span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-47"><a href="#cb49-47" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb49-48"><a href="#cb49-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-49"><a href="#cb49-49" aria-hidden="true" tabindex="-1"></a>The Munk data comprised 1,189 total samples, of which 1,185 had complete metadata. These samples came from 101 countries, with the largest number of samples coming from the USA, Canada, and Denmark:</span>
<span id="cb49-50"><a href="#cb49-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-53"><a href="#cb49-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-54"><a href="#cb49-54" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-55"><a href="#cb49-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: import-qc-data</span></span>
<span id="cb49-56"><a href="#cb49-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-57"><a href="#cb49-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Importing the data is a bit more complicated this time as the samples are split across seven (!) pipeline runs</span></span>
<span id="cb49-58"><a href="#cb49-58" aria-hidden="true" tabindex="-1"></a>data_dir_base <span class="ot">&lt;-</span> <span class="st">"../data/2024-05-06_munk"</span></span>
<span id="cb49-59"><a href="#cb49-59" aria-hidden="true" tabindex="-1"></a>data_dirs <span class="ot">&lt;-</span> <span class="fu">list.dirs</span>(data_dir_base, <span class="at">recursive =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-60"><a href="#cb49-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-61"><a href="#cb49-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb49-62"><a href="#cb49-62" aria-hidden="true" tabindex="-1"></a>libraries_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb49-63"><a href="#cb49-63" aria-hidden="true" tabindex="-1"></a>basic_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_basic_stats.tsv.gz"</span>)</span>
<span id="cb49-64"><a href="#cb49-64" aria-hidden="true" tabindex="-1"></a>adapter_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_adapter_stats.tsv.gz"</span>)</span>
<span id="cb49-65"><a href="#cb49-65" aria-hidden="true" tabindex="-1"></a>quality_base_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_base_stats.tsv.gz"</span>)</span>
<span id="cb49-66"><a href="#cb49-66" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"qc_quality_sequence_stats.tsv.gz"</span>)</span>
<span id="cb49-67"><a href="#cb49-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-68"><a href="#cb49-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb49-69"><a href="#cb49-69" aria-hidden="true" tabindex="-1"></a>ctypes <span class="ot">&lt;-</span> <span class="fu">cols</span>(<span class="at">date=</span><span class="st">"D"</span>, <span class="at">.default=</span><span class="st">"c"</span>)</span>
<span id="cb49-70"><a href="#cb49-70" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">lapply</span>(libraries_paths, read_csv, <span class="at">col_types =</span> ctypes) <span class="sc">%&gt;%</span></span>
<span id="cb49-71"><a href="#cb49-71" aria-hidden="true" tabindex="-1"></a>  bind_rows</span>
<span id="cb49-72"><a href="#cb49-72" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries_raw <span class="sc">%&gt;%</span></span>
<span id="cb49-73"><a href="#cb49-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add missing dates</span></span>
<span id="cb49-74"><a href="#cb49-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">date =</span> <span class="fu">ifelse</span>(sample <span class="sc">==</span> <span class="st">"ERR4682809"</span>, <span class="fu">as_date</span>(<span class="st">"2018-06-01"</span>), date),</span>
<span id="cb49-75"><a href="#cb49-75" aria-hidden="true" tabindex="-1"></a>         <span class="at">date =</span> <span class="fu">ifelse</span>(sample <span class="sc">==</span> <span class="st">"ERR4682803"</span>, <span class="fu">as_date</span>(<span class="st">"2018-06-01"</span>), date),</span>
<span id="cb49-76"><a href="#cb49-76" aria-hidden="true" tabindex="-1"></a>         <span class="at">date =</span> <span class="fu">ifelse</span>(sample <span class="sc">==</span> <span class="st">"ERR2683170"</span>, <span class="fu">as_date</span>(<span class="st">"2017-06-01"</span>), date)) <span class="sc">%&gt;%</span></span>
<span id="cb49-77"><a href="#cb49-77" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter samples with unknown dates</span></span>
<span id="cb49-78"><a href="#cb49-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(date)) <span class="sc">%&gt;%</span></span>
<span id="cb49-79"><a href="#cb49-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(date, country, city) <span class="sc">%&gt;%</span></span>
<span id="cb49-80"><a href="#cb49-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample), <span class="at">date=</span><span class="fu">as_date</span>(date))</span>
<span id="cb49-81"><a href="#cb49-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-82"><a href="#cb49-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-85"><a href="#cb49-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-86"><a href="#cb49-86" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-countries</span></span>
<span id="cb49-87"><a href="#cb49-87" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb49-88"><a href="#cb49-88" aria-hidden="true" tabindex="-1"></a>sample_countries <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(country) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb49-89"><a href="#cb49-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">country=</span><span class="fu">fct_inorder</span>(country))</span>
<span id="cb49-90"><a href="#cb49-90" aria-hidden="true" tabindex="-1"></a>g_countries <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sample_countries, <span class="fu">aes</span>(<span class="at">x=</span>country, <span class="at">y=</span>n)) <span class="sc">+</span> </span>
<span id="cb49-91"><a href="#cb49-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb49-92"><a href="#cb49-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Samples"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">120</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">200</span>,<span class="dv">20</span>)) <span class="sc">+</span></span>
<span id="cb49-93"><a href="#cb49-93" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)))</span>
<span id="cb49-94"><a href="#cb49-94" aria-hidden="true" tabindex="-1"></a>g_countries</span>
<span id="cb49-95"><a href="#cb49-95" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-96"><a href="#cb49-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-97"><a href="#cb49-97" aria-hidden="true" tabindex="-1"></a>The 1,185 libraries included in this analysis varied dramatically in size, from 33,554 read pairs to over 123 million. The mean number of read pairs per library was 33.5M, and the dataset as a whole comprised 39.7B read pairs and almost 12 terabases of sequence:</span>
<span id="cb49-98"><a href="#cb49-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-101"><a href="#cb49-101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-102"><a href="#cb49-102" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-qc-data</span></span>
<span id="cb49-103"><a href="#cb49-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-104"><a href="#cb49-104" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb49-105"><a href="#cb49-105" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb49-106"><a href="#cb49-106" aria-hidden="true" tabindex="-1"></a>import_basic <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb49-107"><a href="#cb49-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb49-108"><a href="#cb49-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-109"><a href="#cb49-109" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-110"><a href="#cb49-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb49-111"><a href="#cb49-111" aria-hidden="true" tabindex="-1"></a>           <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb49-112"><a href="#cb49-112" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-113"><a href="#cb49-113" aria-hidden="true" tabindex="-1"></a>import_basic_paired <span class="ot">&lt;-</span> <span class="cf">function</span>(paths){</span>
<span id="cb49-114"><a href="#cb49-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">import_basic</span>(paths) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(read_pair) <span class="sc">%&gt;%</span> </span>
<span id="cb49-115"><a href="#cb49-115" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb49-116"><a href="#cb49-116" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-117"><a href="#cb49-117" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">import_basic</span>(basic_stats_paths)</span>
<span id="cb49-118"><a href="#cb49-118" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(adapter_stats_paths)</span>
<span id="cb49-119"><a href="#cb49-119" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_base_stats_paths)</span>
<span id="cb49-120"><a href="#cb49-120" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">import_basic_paired</span>(quality_seq_stats_paths)</span>
<span id="cb49-121"><a href="#cb49-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-122"><a href="#cb49-122" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify small and large datasets</span></span>
<span id="cb49-123"><a href="#cb49-123" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb49-124"><a href="#cb49-124" aria-hidden="true" tabindex="-1"></a>libraries_small <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_read_pairs <span class="sc">&lt;=</span> <span class="fl">1e7</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(library)</span>
<span id="cb49-125"><a href="#cb49-125" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">small =</span> library <span class="sc">%in%</span> libraries_small)</span>
<span id="cb49-126"><a href="#cb49-126" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">small =</span> library <span class="sc">%in%</span> libraries_small)</span>
<span id="cb49-127"><a href="#cb49-127" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">small =</span> library <span class="sc">%in%</span> libraries_small)</span>
<span id="cb49-128"><a href="#cb49-128" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">small =</span> library <span class="sc">%in%</span> libraries_small)</span>
<span id="cb49-129"><a href="#cb49-129" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">small =</span> library <span class="sc">%in%</span> libraries_small)</span>
<span id="cb49-130"><a href="#cb49-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-131"><a href="#cb49-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb49-132"><a href="#cb49-132" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb49-133"><a href="#cb49-133" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb49-134"><a href="#cb49-134" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb49-135"><a href="#cb49-135" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb49-136"><a href="#cb49-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-137"><a href="#cb49-137" aria-hidden="true" tabindex="-1"></a><span class="co"># Get key values for readout</span></span>
<span id="cb49-138"><a href="#cb49-138" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> </span>
<span id="cb49-139"><a href="#cb49-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb49-140"><a href="#cb49-140" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), </span>
<span id="cb49-141"><a href="#cb49-141" aria-hidden="true" tabindex="-1"></a>            <span class="at">rtot =</span> <span class="fu">sum</span>(n_read_pairs),</span>
<span id="cb49-142"><a href="#cb49-142" aria-hidden="true" tabindex="-1"></a>            <span class="at">btot =</span> <span class="fu">sum</span>(n_bases_approx),</span>
<span id="cb49-143"><a href="#cb49-143" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb49-144"><a href="#cb49-144" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-145"><a href="#cb49-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-146"><a href="#cb49-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-149"><a href="#cb49-149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-150"><a href="#cb49-150" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb49-151"><a href="#cb49-151" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-152"><a href="#cb49-152" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb49-153"><a href="#cb49-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-154"><a href="#cb49-154" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb49-155"><a href="#cb49-155" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb49-156"><a href="#cb49-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, date,</span>
<span id="cb49-157"><a href="#cb49-157" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb49-158"><a href="#cb49-158" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb49-159"><a href="#cb49-159" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb49-160"><a href="#cb49-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(sample<span class="sc">:</span>date), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-161"><a href="#cb49-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric))</span>
<span id="cb49-162"><a href="#cb49-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-163"><a href="#cb49-163" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb49-164"><a href="#cb49-164" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, <span class="fu">aes</span>(<span class="at">x=</span>date, <span class="at">y=</span>value)) <span class="sc">+</span></span>
<span id="cb49-165"><a href="#cb49-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb49-166"><a href="#cb49-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_date</span>() <span class="sc">+</span></span>
<span id="cb49-167"><a href="#cb49-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-168"><a href="#cb49-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb49-169"><a href="#cb49-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space=</span><span class="st">"free_x"</span>, <span class="at">switch=</span><span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb49-170"><a href="#cb49-170" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb49-171"><a href="#cb49-171" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-172"><a href="#cb49-172" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"plain"</span>)</span>
<span id="cb49-173"><a href="#cb49-173" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb49-174"><a href="#cb49-174" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb49-175"><a href="#cb49-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-176"><a href="#cb49-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-177"><a href="#cb49-177" aria-hidden="true" tabindex="-1"></a>Adapter levels were high, read qualities were variable (in definite need of trimming) and duplicate levels were moderate:</span>
<span id="cb49-178"><a href="#cb49-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-181"><a href="#cb49-181" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-182"><a href="#cb49-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb49-183"><a href="#cb49-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-184"><a href="#cb49-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb49-185"><a href="#cb49-185" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair,</span>
<span id="cb49-186"><a href="#cb49-186" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb49-187"><a href="#cb49-187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb49-188"><a href="#cb49-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb49-189"><a href="#cb49-189" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb49-190"><a href="#cb49-190" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb49-191"><a href="#cb49-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-192"><a href="#cb49-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb49-193"><a href="#cb49-193" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb49-194"><a href="#cb49-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb49-195"><a href="#cb49-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb49-196"><a href="#cb49-196" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-197"><a href="#cb49-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb49-198"><a href="#cb49-198" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-199"><a href="#cb49-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>adapter)</span>
<span id="cb49-200"><a href="#cb49-200" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb49-201"><a href="#cb49-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-202"><a href="#cb49-202" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb49-203"><a href="#cb49-203" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb49-204"><a href="#cb49-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-205"><a href="#cb49-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-206"><a href="#cb49-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb49-207"><a href="#cb49-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb49-208"><a href="#cb49-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb49-209"><a href="#cb49-209" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb49-210"><a href="#cb49-210" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb49-211"><a href="#cb49-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-212"><a href="#cb49-212" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb49-213"><a href="#cb49-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-214"><a href="#cb49-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-215"><a href="#cb49-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb49-216"><a href="#cb49-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-217"><a href="#cb49-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb49-218"><a href="#cb49-218" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb49-219"><a href="#cb49-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-220"><a href="#cb49-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-221"><a href="#cb49-221" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb49-222"><a href="#cb49-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-223"><a href="#cb49-223" aria-hidden="true" tabindex="-1"></a>About 6% of reads on average were lost during cleaning, and a further 10% during deduplication; however, in both cases a minority of samples lost much larger read fractions. Very few reads were lost during ribodepletion, as expected for DNA sequencing libraries.</span>
<span id="cb49-224"><a href="#cb49-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-227"><a href="#cb49-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-228"><a href="#cb49-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb49-229"><a href="#cb49-229" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb49-230"><a href="#cb49-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, stage, </span>
<span id="cb49-231"><a href="#cb49-231" aria-hidden="true" tabindex="-1"></a>         percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb49-232"><a href="#cb49-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, stage) <span class="sc">%&gt;%</span></span>
<span id="cb49-233"><a href="#cb49-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> <span class="fu">replace_na</span>(n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs), <span class="dv">0</span>),</span>
<span id="cb49-234"><a href="#cb49-234" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb49-235"><a href="#cb49-235" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb49-236"><a href="#cb49-236" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb49-237"><a href="#cb49-237" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> <span class="fu">replace_na</span>(p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs), <span class="dv">0</span>))</span>
<span id="cb49-238"><a href="#cb49-238" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb49-239"><a href="#cb49-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="at">Stage=</span>stage) <span class="sc">%&gt;%</span> </span>
<span id="cb49-240"><a href="#cb49-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb49-241"><a href="#cb49-241" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-242"><a href="#cb49-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-243"><a href="#cb49-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb49-244"><a href="#cb49-244" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb49-245"><a href="#cb49-245" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-246"><a href="#cb49-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-249"><a href="#cb49-249" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-250"><a href="#cb49-250" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb49-251"><a href="#cb49-251" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-252"><a href="#cb49-252" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb49-253"><a href="#cb49-253" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb49-254"><a href="#cb49-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-255"><a href="#cb49-255" aria-hidden="true" tabindex="-1"></a>g_stage_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb49-256"><a href="#cb49-256" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb49-257"><a href="#cb49-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-258"><a href="#cb49-258" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb49-259"><a href="#cb49-259" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_base <span class="sc">+</span></span>
<span id="cb49-260"><a href="#cb49-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs), <span class="at">data=</span>basic_stats) <span class="sc">+</span></span>
<span id="cb49-261"><a href="#cb49-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb49-262"><a href="#cb49-262" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb49-263"><a href="#cb49-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-264"><a href="#cb49-264" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb49-265"><a href="#cb49-265" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> g_stage_base <span class="sc">+</span></span>
<span id="cb49-266"><a href="#cb49-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal), <span class="at">data=</span>n_reads_rel) <span class="sc">+</span></span>
<span id="cb49-267"><a href="#cb49-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb49-268"><a href="#cb49-268" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb49-269"><a href="#cb49-269" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb49-270"><a href="#cb49-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-271"><a href="#cb49-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-272"><a href="#cb49-272" aria-hidden="true" tabindex="-1"></a>As usual, data cleaning was very successful at removing adapters and improving read qualities:</span>
<span id="cb49-273"><a href="#cb49-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-276"><a href="#cb49-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-277"><a href="#cb49-277" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-278"><a href="#cb49-278" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb49-279"><a href="#cb49-279" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb49-280"><a href="#cb49-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-281"><a href="#cb49-281" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">linetype=</span>read_pair, </span>
<span id="cb49-282"><a href="#cb49-282" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb49-283"><a href="#cb49-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb49-284"><a href="#cb49-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb49-285"><a href="#cb49-285" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb49-286"><a href="#cb49-286" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb49-287"><a href="#cb49-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-288"><a href="#cb49-288" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb49-289"><a href="#cb49-289" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb49-290"><a href="#cb49-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb49-291"><a href="#cb49-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb49-292"><a href="#cb49-292" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-293"><a href="#cb49-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb49-294"><a href="#cb49-294" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-295"><a href="#cb49-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb49-296"><a href="#cb49-296" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb49-297"><a href="#cb49-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-298"><a href="#cb49-298" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb49-299"><a href="#cb49-299" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb49-300"><a href="#cb49-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-301"><a href="#cb49-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-302"><a href="#cb49-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb49-303"><a href="#cb49-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb49-304"><a href="#cb49-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb49-305"><a href="#cb49-305" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-306"><a href="#cb49-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb49-307"><a href="#cb49-307" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb49-308"><a href="#cb49-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-309"><a href="#cb49-309" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb49-310"><a href="#cb49-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-311"><a href="#cb49-311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-312"><a href="#cb49-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb49-313"><a href="#cb49-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-314"><a href="#cb49-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-315"><a href="#cb49-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb49-316"><a href="#cb49-316" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb49-317"><a href="#cb49-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-318"><a href="#cb49-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-319"><a href="#cb49-319" aria-hidden="true" tabindex="-1"></a>According to FASTQC, cleaning + deduplication was mostly effective at reducing measured duplicate levels, though a few samples retained high measured duplicate levels throughout the pipeline:</span>
<span id="cb49-320"><a href="#cb49-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-323"><a href="#cb49-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-324"><a href="#cb49-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb49-325"><a href="#cb49-325" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3.5</span></span>
<span id="cb49-326"><a href="#cb49-326" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb49-327"><a href="#cb49-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-328"><a href="#cb49-328" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(stage) <span class="sc">%&gt;%</span> </span>
<span id="cb49-329"><a href="#cb49-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb49-330"><a href="#cb49-330" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-331"><a href="#cb49-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-332"><a href="#cb49-332" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_base <span class="sc">+</span></span>
<span id="cb49-333"><a href="#cb49-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates), <span class="at">data=</span>basic_stats) <span class="sc">+</span></span>
<span id="cb49-334"><a href="#cb49-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb49-335"><a href="#cb49-335" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb49-336"><a href="#cb49-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-337"><a href="#cb49-337" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_base <span class="sc">+</span> </span>
<span id="cb49-338"><a href="#cb49-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len), <span class="at">data=</span>basic_stats) <span class="sc">+</span></span>
<span id="cb49-339"><a href="#cb49-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb49-340"><a href="#cb49-340" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb49-341"><a href="#cb49-341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-342"><a href="#cb49-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-343"><a href="#cb49-343" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb49-344"><a href="#cb49-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-345"><a href="#cb49-345" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb49-346"><a href="#cb49-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-349"><a href="#cb49-349" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-350"><a href="#cb49-350" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb49-351"><a href="#cb49-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-352"><a href="#cb49-352" aria-hidden="true" tabindex="-1"></a>classifications <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Ribosomal"</span>, <span class="st">"Unassigned"</span>,</span>
<span id="cb49-353"><a href="#cb49-353" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Bacterial"</span>, <span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>)</span>
<span id="cb49-354"><a href="#cb49-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-355"><a href="#cb49-355" aria-hidden="true" tabindex="-1"></a><span class="co"># Import composition data</span></span>
<span id="cb49-356"><a href="#cb49-356" aria-hidden="true" tabindex="-1"></a>comp_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"taxonomic_composition.tsv.gz"</span>)</span>
<span id="cb49-357"><a href="#cb49-357" aria-hidden="true" tabindex="-1"></a>comp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(comp_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb49-358"><a href="#cb49-358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-359"><a href="#cb49-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">factor</span>(classification, <span class="at">levels =</span> classifications))</span>
<span id="cb49-360"><a href="#cb49-360" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-361"><a href="#cb49-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-362"><a href="#cb49-362" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb49-363"><a href="#cb49-363" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb49-364"><a href="#cb49-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb49-365"><a href="#cb49-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-366"><a href="#cb49-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb49-367"><a href="#cb49-367" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb49-368"><a href="#cb49-368" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb49-369"><a href="#cb49-369" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-370"><a href="#cb49-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-373"><a href="#cb49-373" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-374"><a href="#cb49-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb49-375"><a href="#cb49-375" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb49-376"><a href="#cb49-376" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb49-377"><a href="#cb49-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-378"><a href="#cb49-378" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb49-379"><a href="#cb49-379" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>sample, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb49-380"><a href="#cb49-380" aria-hidden="true" tabindex="-1"></a>  theme_xblank <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb49-381"><a href="#cb49-381" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb49-382"><a href="#cb49-382" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb49-383"><a href="#cb49-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-384"><a href="#cb49-384" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb49-385"><a href="#cb49-385" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data =</span> comp, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-386"><a href="#cb49-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb49-387"><a href="#cb49-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb49-388"><a href="#cb49-388" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb49-389"><a href="#cb49-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-390"><a href="#cb49-390" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb49-391"><a href="#cb49-391" aria-hidden="true" tabindex="-1"></a>comp_minor <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span> </span>
<span id="cb49-392"><a href="#cb49-392" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Human"</span>, <span class="st">"Other"</span>))</span>
<span id="cb49-393"><a href="#cb49-393" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">6</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb49-394"><a href="#cb49-394" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-395"><a href="#cb49-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>comp_minor, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-396"><a href="#cb49-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb49-397"><a href="#cb49-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb49-398"><a href="#cb49-398" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb49-399"><a href="#cb49-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-400"><a href="#cb49-400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-401"><a href="#cb49-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-404"><a href="#cb49-404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-405"><a href="#cb49-405" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-summary</span></span>
<span id="cb49-406"><a href="#cb49-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-407"><a href="#cb49-407" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> comp <span class="sc">%&gt;%</span></span>
<span id="cb49-408"><a href="#cb49-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb49-409"><a href="#cb49-409" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb49-410"><a href="#cb49-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-411"><a href="#cb49-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-412"><a href="#cb49-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification) <span class="sc">%&gt;%</span></span>
<span id="cb49-413"><a href="#cb49-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb49-414"><a href="#cb49-414" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-415"><a href="#cb49-415" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> p_reads_summ_group <span class="sc">%&gt;%</span></span>
<span id="cb49-416"><a href="#cb49-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb49-417"><a href="#cb49-417" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb49-418"><a href="#cb49-418" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb49-419"><a href="#cb49-419" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb49-420"><a href="#cb49-420" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb49-421"><a href="#cb49-421" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb49-422"><a href="#cb49-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">Classification=</span>classification, </span>
<span id="cb49-423"><a href="#cb49-423" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Read Fraction</span><span class="st">`</span><span class="ot">=</span>display) <span class="sc">%&gt;%</span></span>
<span id="cb49-424"><a href="#cb49-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Classification)</span>
<span id="cb49-425"><a href="#cb49-425" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb49-426"><a href="#cb49-426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-427"><a href="#cb49-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-428"><a href="#cb49-428" aria-hidden="true" tabindex="-1"></a>As in previous DNA datasets, the vast majority of classified reads were bacterial in origin. Viral fraction averaged 0.33%, higher than in other DNA wastewater datasets I've looked at, and reached <span class="sc">\&gt;</span>1% in 35 samples. As is common for DNA wastewater data, viral reads were overwhelmingly dominated by *Caudoviricetes* phages, though *Quintoviricetes* (parvoviruses) also showed significant prevalence in some samples:</span>
<span id="cb49-429"><a href="#cb49-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-432"><a href="#cb49-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-433"><a href="#cb49-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb49-434"><a href="#cb49-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-435"><a href="#cb49-435" aria-hidden="true" tabindex="-1"></a><span class="co"># # Get Kraken reports</span></span>
<span id="cb49-436"><a href="#cb49-436" aria-hidden="true" tabindex="-1"></a><span class="co"># reports_paths &lt;- file.path(data_dirs, "kraken_reports.tsv.gz")</span></span>
<span id="cb49-437"><a href="#cb49-437" aria-hidden="true" tabindex="-1"></a><span class="co"># reports &lt;- lapply(reports_paths, read_tsv, show_col_types = FALSE) %&gt;% bind_rows %&gt;%</span></span>
<span id="cb49-438"><a href="#cb49-438" aria-hidden="true" tabindex="-1"></a><span class="co">#   inner_join(libraries, by="sample")</span></span>
<span id="cb49-439"><a href="#cb49-439" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb49-440"><a href="#cb49-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb49-441"><a href="#cb49-441" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb49-442"><a href="#cb49-442" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-443"><a href="#cb49-443" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb49-444"><a href="#cb49-444" aria-hidden="true" tabindex="-1"></a><span class="co"># # Filter to viral taxa</span></span>
<span id="cb49-445"><a href="#cb49-445" aria-hidden="true" tabindex="-1"></a><span class="co"># kraken_reports_viral &lt;- filter(reports, taxid %in% viral_taxa$taxid) %&gt;%</span></span>
<span id="cb49-446"><a href="#cb49-446" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(sample) %&gt;%</span></span>
<span id="cb49-447"><a href="#cb49-447" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(p_reads_viral = n_reads_clade/n_reads_clade[1])</span></span>
<span id="cb49-448"><a href="#cb49-448" aria-hidden="true" tabindex="-1"></a><span class="co"># kraken_reports_viral_cleaned &lt;- kraken_reports_viral %&gt;%</span></span>
<span id="cb49-449"><a href="#cb49-449" aria-hidden="true" tabindex="-1"></a><span class="co">#   inner_join(libraries, by="sample") %&gt;%</span></span>
<span id="cb49-450"><a href="#cb49-450" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(-pc_reads_total, -n_reads_direct, -contains("minimizers")) %&gt;%</span></span>
<span id="cb49-451"><a href="#cb49-451" aria-hidden="true" tabindex="-1"></a><span class="co">#   select(name, taxid, p_reads_viral, n_reads_clade, everything())</span></span>
<span id="cb49-452"><a href="#cb49-452" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb49-453"><a href="#cb49-453" aria-hidden="true" tabindex="-1"></a><span class="co"># viral_classes &lt;- kraken_reports_viral_cleaned %&gt;% filter(rank == "C")</span></span>
<span id="cb49-454"><a href="#cb49-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-455"><a href="#cb49-455" aria-hidden="true" tabindex="-1"></a>viral_classes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"viral_classes.tsv.gz"</span>)</span>
<span id="cb49-456"><a href="#cb49-456" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(viral_classes, viral_classes_path)</span></span>
<span id="cb49-457"><a href="#cb49-457" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_classes_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-458"><a href="#cb49-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-459"><a href="#cb49-459" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-460"><a href="#cb49-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-463"><a href="#cb49-463" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-464"><a href="#cb49-464" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-class-composition</span></span>
<span id="cb49-465"><a href="#cb49-465" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb49-466"><a href="#cb49-466" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb49-467"><a href="#cb49-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-468"><a href="#cb49-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-469"><a href="#cb49-469" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb49-470"><a href="#cb49-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-471"><a href="#cb49-471" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral classes</span></span>
<span id="cb49-472"><a href="#cb49-472" aria-hidden="true" tabindex="-1"></a>viral_classes_major_tab <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb49-473"><a href="#cb49-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-474"><a href="#cb49-474" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-475"><a href="#cb49-475" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> major_threshold)</span>
<span id="cb49-476"><a href="#cb49-476" aria-hidden="true" tabindex="-1"></a>viral_classes_major_list <span class="ot">&lt;-</span> viral_classes_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb49-477"><a href="#cb49-477" aria-hidden="true" tabindex="-1"></a>viral_classes_major <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb49-478"><a href="#cb49-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_classes_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb49-479"><a href="#cb49-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb49-480"><a href="#cb49-480" aria-hidden="true" tabindex="-1"></a>viral_classes_minor <span class="ot">&lt;-</span> viral_classes_major <span class="sc">%&gt;%</span> </span>
<span id="cb49-481"><a href="#cb49-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-482"><a href="#cb49-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-483"><a href="#cb49-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb49-484"><a href="#cb49-484" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, p_reads_viral)</span>
<span id="cb49-485"><a href="#cb49-485" aria-hidden="true" tabindex="-1"></a>viral_classes_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_classes_major, viral_classes_minor) <span class="sc">%&gt;%</span></span>
<span id="cb49-486"><a href="#cb49-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-487"><a href="#cb49-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_classes_major_list, <span class="st">"Other"</span>)),</span>
<span id="cb49-488"><a href="#cb49-488" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> <span class="fu">pmax</span>(p_reads_viral, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-489"><a href="#cb49-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name)</span>
<span id="cb49-490"><a href="#cb49-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-491"><a href="#cb49-491" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb49-492"><a href="#cb49-492" aria-hidden="true" tabindex="-1"></a>g_classes <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-493"><a href="#cb49-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>viral_classes_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-494"><a href="#cb49-494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-495"><a href="#cb49-495" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-496"><a href="#cb49-496" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb49-497"><a href="#cb49-497" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-498"><a href="#cb49-498" aria-hidden="true" tabindex="-1"></a>g_classes</span>
<span id="cb49-499"><a href="#cb49-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-500"><a href="#cb49-500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-501"><a href="#cb49-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-502"><a href="#cb49-502" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb49-503"><a href="#cb49-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-504"><a href="#cb49-504" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples. A grand total of 331,452 reads were identified as putatively human-viral:</span>
<span id="cb49-505"><a href="#cb49-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-508"><a href="#cb49-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-509"><a href="#cb49-509" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-read-counts</span></span>
<span id="cb49-510"><a href="#cb49-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-511"><a href="#cb49-511" aria-hidden="true" tabindex="-1"></a><span class="co"># Import HV read data</span></span>
<span id="cb49-512"><a href="#cb49-512" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb49-513"><a href="#cb49-513" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">lapply</span>(hv_reads_filtered_paths, read_tsv,</span>
<span id="cb49-514"><a href="#cb49-514" aria-hidden="true" tabindex="-1"></a>                            <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-515"><a href="#cb49-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb49-516"><a href="#cb49-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb49-517"><a href="#cb49-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-518"><a href="#cb49-518" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb49-519"><a href="#cb49-519" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb49-520"><a href="#cb49-520" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_id) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb49-521"><a href="#cb49-521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb49-522"><a href="#cb49-522" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-523"><a href="#cb49-523" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-524"><a href="#cb49-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb49-525"><a href="#cb49-525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb49-526"><a href="#cb49-526" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_summ <span class="ot">&lt;-</span> n_hv_filtered <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb49-527"><a href="#cb49-527" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total), </span>
<span id="cb49-528"><a href="#cb49-528" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-529"><a href="#cb49-529" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb49-530"><a href="#cb49-530" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-531"><a href="#cb49-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-534"><a href="#cb49-534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-535"><a href="#cb49-535" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-scores</span></span>
<span id="cb49-536"><a href="#cb49-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-537"><a href="#cb49-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb49-538"><a href="#cb49-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-539"><a href="#cb49-539" aria-hidden="true" tabindex="-1"></a><span class="co"># Collapse multi-entry sequences</span></span>
<span id="cb49-540"><a href="#cb49-540" aria-hidden="true" tabindex="-1"></a>rmax <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(max, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb49-541"><a href="#cb49-541" aria-hidden="true" tabindex="-1"></a>collapse <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">ifelse</span>(<span class="fu">all</span>(x <span class="sc">==</span> x[<span class="dv">1</span>]), x[<span class="dv">1</span>], <span class="fu">paste</span>(x, <span class="at">collapse=</span><span class="st">"/"</span>))</span>
<span id="cb49-542"><a href="#cb49-542" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> </span>
<span id="cb49-543"><a href="#cb49-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-544"><a href="#cb49-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb49-545"><a href="#cb49-545" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb49-546"><a href="#cb49-546" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">sample =</span> <span class="fu">collapse</span>(sample),</span>
<span id="cb49-547"><a href="#cb49-547" aria-hidden="true" tabindex="-1"></a>            <span class="at">genome_id =</span> <span class="fu">collapse</span>(genome_id),</span>
<span id="cb49-548"><a href="#cb49-548" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid_best =</span> taxid[<span class="dv">1</span>],</span>
<span id="cb49-549"><a href="#cb49-549" aria-hidden="true" tabindex="-1"></a>            <span class="at">taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(taxid)),</span>
<span id="cb49-550"><a href="#cb49-550" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_fwd =</span> <span class="fu">rmax</span>(best_alignment_score_fwd),</span>
<span id="cb49-551"><a href="#cb49-551" aria-hidden="true" tabindex="-1"></a>            <span class="at">best_alignment_score_rev =</span> <span class="fu">rmax</span>(best_alignment_score_rev),</span>
<span id="cb49-552"><a href="#cb49-552" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_fwd =</span> <span class="fu">rmax</span>(query_len_fwd),</span>
<span id="cb49-553"><a href="#cb49-553" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_len_rev =</span> <span class="fu">rmax</span>(query_len_rev),</span>
<span id="cb49-554"><a href="#cb49-554" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_fwd =</span> query_seq_fwd[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_fwd)][<span class="dv">1</span>],</span>
<span id="cb49-555"><a href="#cb49-555" aria-hidden="true" tabindex="-1"></a>            <span class="at">query_seq_rev =</span> query_seq_rev[<span class="sc">!</span><span class="fu">is.na</span>(query_seq_rev)][<span class="dv">1</span>],</span>
<span id="cb49-556"><a href="#cb49-556" aria-hidden="true" tabindex="-1"></a>            <span class="at">classified =</span> <span class="fu">rmax</span>(classified),</span>
<span id="cb49-557"><a href="#cb49-557" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_name =</span> <span class="fu">collapse</span>(assigned_name),</span>
<span id="cb49-558"><a href="#cb49-558" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid_best =</span> assigned_taxid[<span class="dv">1</span>],</span>
<span id="cb49-559"><a href="#cb49-559" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_taxid =</span> <span class="fu">collapse</span>(<span class="fu">as.character</span>(assigned_taxid)),</span>
<span id="cb49-560"><a href="#cb49-560" aria-hidden="true" tabindex="-1"></a>            <span class="at">assigned_hv =</span> <span class="fu">rmax</span>(assigned_hv),</span>
<span id="cb49-561"><a href="#cb49-561" aria-hidden="true" tabindex="-1"></a>            <span class="at">hit_hv =</span> <span class="fu">rmax</span>(hit_hv),</span>
<span id="cb49-562"><a href="#cb49-562" aria-hidden="true" tabindex="-1"></a>            <span class="at">encoded_hits =</span> <span class="fu">collapse</span>(encoded_hits),</span>
<span id="cb49-563"><a href="#cb49-563" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_fwd =</span> <span class="fu">rmax</span>(adj_score_fwd),</span>
<span id="cb49-564"><a href="#cb49-564" aria-hidden="true" tabindex="-1"></a>            <span class="at">adj_score_rev =</span> <span class="fu">rmax</span>(adj_score_rev)</span>
<span id="cb49-565"><a href="#cb49-565" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb49-566"><a href="#cb49-566" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-567"><a href="#cb49-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb49-568"><a href="#cb49-568" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb49-569"><a href="#cb49-569" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb49-570"><a href="#cb49-570" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev),</span>
<span id="cb49-571"><a href="#cb49-571" aria-hidden="true" tabindex="-1"></a>         <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb49-572"><a href="#cb49-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-573"><a href="#cb49-573" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot results</span></span>
<span id="cb49-574"><a href="#cb49-574" aria-hidden="true" tabindex="-1"></a>geom_vhist <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(geom_histogram, <span class="at">binwidth=</span><span class="dv">5</span>, <span class="at">boundary=</span><span class="dv">0</span>)</span>
<span id="cb49-575"><a href="#cb49-575" aria-hidden="true" tabindex="-1"></a>g_vhist_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span></span>
<span id="cb49-576"><a href="#cb49-576" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb49-577"><a href="#cb49-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>)), <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb49-578"><a href="#cb49-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> </span>
<span id="cb49-579"><a href="#cb49-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> </span>
<span id="cb49-580"><a href="#cb49-580" aria-hidden="true" tabindex="-1"></a>  theme_base </span>
<span id="cb49-581"><a href="#cb49-581" aria-hidden="true" tabindex="-1"></a>g_vhist_0 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg)</span>
<span id="cb49-582"><a href="#cb49-582" aria-hidden="true" tabindex="-1"></a>g_vhist_0</span>
<span id="cb49-583"><a href="#cb49-583" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-584"><a href="#cb49-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-585"><a href="#cb49-585" aria-hidden="true" tabindex="-1"></a>BLASTing these reads against nt, we find that the pipeline performs well, with only a single high-scoring false-positive read:</span>
<span id="cb49-586"><a href="#cb49-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-589"><a href="#cb49-589" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-590"><a href="#cb49-590" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-blast-data</span></span>
<span id="cb49-591"><a href="#cb49-591" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-592"><a href="#cb49-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-593"><a href="#cb49-593" aria-hidden="true" tabindex="-1"></a><span class="co"># Import paired BLAST results</span></span>
<span id="cb49-594"><a href="#cb49-594" aria-hidden="true" tabindex="-1"></a>blast_paired_paths <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dirs, <span class="st">"hv_hits_blast_paired.tsv.gz"</span>)</span>
<span id="cb49-595"><a href="#cb49-595" aria-hidden="true" tabindex="-1"></a>blast_paired <span class="ot">&lt;-</span> <span class="fu">lapply</span>(blast_paired_paths, read_tsv, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb49-596"><a href="#cb49-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-597"><a href="#cb49-597" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb49-598"><a href="#cb49-598" aria-hidden="true" tabindex="-1"></a>blast_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_paired, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-599"><a href="#cb49-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb49-600"><a href="#cb49-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-601"><a href="#cb49-601" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb49-602"><a href="#cb49-602" aria-hidden="true" tabindex="-1"></a>match_taxid <span class="ot">&lt;-</span> <span class="cf">function</span>(taxid_1, taxid_2){</span>
<span id="cb49-603"><a href="#cb49-603" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"/"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb49-604"><a href="#cb49-604" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"/"</span>), taxid_2)</span>
<span id="cb49-605"><a href="#cb49-605" aria-hidden="true" tabindex="-1"></a>  p3 <span class="ot">&lt;-</span> <span class="fu">mapply</span>(grepl, <span class="fu">paste0</span>(<span class="st">"^"</span>, taxid_1, <span class="st">"$"</span>), taxid_2)</span>
<span id="cb49-606"><a href="#cb49-606" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">setNames</span>(p1<span class="sc">|</span>p2<span class="sc">|</span>p3, <span class="cn">NULL</span>)</span>
<span id="cb49-607"><a href="#cb49-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb49-608"><a href="#cb49-608" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-609"><a href="#cb49-609" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_id, taxid, assigned_taxid, adj_score_max)</span>
<span id="cb49-610"><a href="#cb49-610" aria-hidden="true" tabindex="-1"></a>blast_assign <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(blast_viral, mrg_assign, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-611"><a href="#cb49-611" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> <span class="fu">match_taxid</span>(staxid, taxid),</span>
<span id="cb49-612"><a href="#cb49-612" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> <span class="fu">match_taxid</span>(staxid, assigned_taxid),</span>
<span id="cb49-613"><a href="#cb49-613" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb49-614"><a href="#cb49-614" aria-hidden="true" tabindex="-1"></a>blast_out <span class="ot">&lt;-</span> blast_assign <span class="sc">%&gt;%</span></span>
<span id="cb49-615"><a href="#cb49-615" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb49-616"><a href="#cb49-616" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb49-617"><a href="#cb49-617" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb49-618"><a href="#cb49-618" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb49-619"><a href="#cb49-619" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb49-620"><a href="#cb49-620" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-621"><a href="#cb49-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-624"><a href="#cb49-624" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-625"><a href="#cb49-625" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-results</span></span>
<span id="cb49-626"><a href="#cb49-626" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb49-627"><a href="#cb49-627" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-628"><a href="#cb49-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-629"><a href="#cb49-629" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb49-630"><a href="#cb49-630" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg, blast_out, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-631"><a href="#cb49-631" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb49-632"><a href="#cb49-632" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb49-633"><a href="#cb49-633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-634"><a href="#cb49-634" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-635"><a href="#cb49-635" aria-hidden="true" tabindex="-1"></a>g_vhist_1 <span class="ot">&lt;-</span> g_vhist_base <span class="sc">+</span> <span class="fu">geom_vhist</span>(<span class="at">data=</span>mrg_blast, <span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">fill=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb49-636"><a href="#cb49-636" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>)</span>
<span id="cb49-637"><a href="#cb49-637" aria-hidden="true" tabindex="-1"></a>g_vhist_1</span>
<span id="cb49-638"><a href="#cb49-638" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-639"><a href="#cb49-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-640"><a href="#cb49-640" aria-hidden="true" tabindex="-1"></a>My usual disjunctive score threshold of 20 gave precision, sensitivity, and F1 scores all <span class="sc">\&gt;</span>99%:</span>
<span id="cb49-641"><a href="#cb49-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-644"><a href="#cb49-644" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-645"><a href="#cb49-645" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-f1</span></span>
<span id="cb49-646"><a href="#cb49-646" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold){</span>
<span id="cb49-647"><a href="#cb49-647" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> </span>
<span id="cb49-648"><a href="#cb49-648" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">retain_score =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb49-649"><a href="#cb49-649" aria-hidden="true" tabindex="-1"></a>           <span class="at">retain =</span> assigned_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb49-650"><a href="#cb49-650" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb49-651"><a href="#cb49-651" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb49-652"><a href="#cb49-652" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb49-653"><a href="#cb49-653" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb49-654"><a href="#cb49-654" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb49-655"><a href="#cb49-655" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb49-656"><a href="#cb49-656" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb49-657"><a href="#cb49-657" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb49-658"><a href="#cb49-658" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb49-659"><a href="#cb49-659" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb49-660"><a href="#cb49-660" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb49-661"><a href="#cb49-661" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb49-662"><a href="#cb49-662" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-663"><a href="#cb49-663" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb49-664"><a href="#cb49-664" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab)</span>
<span id="cb49-665"><a href="#cb49-665" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss) <span class="sc">%&gt;%</span> bind_rows <span class="sc">%&gt;%</span></span>
<span id="cb49-666"><a href="#cb49-666" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>threshold, <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>)</span>
<span id="cb49-667"><a href="#cb49-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats)</span>
<span id="cb49-668"><a href="#cb49-668" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-669"><a href="#cb49-669" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast)</span>
<span id="cb49-670"><a href="#cb49-670" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb49-671"><a href="#cb49-671" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb49-672"><a href="#cb49-672" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb49-673"><a href="#cb49-673" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-674"><a href="#cb49-674" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Adjusted Score Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb49-675"><a href="#cb49-675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb49-676"><a href="#cb49-676" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb49-677"><a href="#cb49-677" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb49-678"><a href="#cb49-678" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-679"><a href="#cb49-679" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">Threshold=</span>threshold, <span class="at">Metric=</span>metric, <span class="at">Value=</span>value)</span>
<span id="cb49-680"><a href="#cb49-680" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-681"><a href="#cb49-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-682"><a href="#cb49-682" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: overall relative abundance</span></span>
<span id="cb49-683"><a href="#cb49-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-686"><a href="#cb49-686" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-687"><a href="#cb49-687" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb49-688"><a href="#cb49-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-689"><a href="#cb49-689" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb49-690"><a href="#cb49-690" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb49-691"><a href="#cb49-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb49-692"><a href="#cb49-692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-693"><a href="#cb49-693" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb49-694"><a href="#cb49-694" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> highscore) <span class="sc">%&gt;%</span></span>
<span id="cb49-695"><a href="#cb49-695" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">taxid_all =</span> taxid, <span class="at">taxid =</span> taxid_best)</span>
<span id="cb49-696"><a href="#cb49-696" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb49-697"><a href="#cb49-697" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb49-698"><a href="#cb49-698" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-699"><a href="#cb49-699" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-700"><a href="#cb49-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb49-701"><a href="#cb49-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-702"><a href="#cb49-702" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb49-703"><a href="#cb49-703" aria-hidden="true" tabindex="-1"></a>read_counts_grp <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(country) <span class="sc">%&gt;%</span></span>
<span id="cb49-704"><a href="#cb49-704" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb49-705"><a href="#cb49-705" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), </span>
<span id="cb49-706"><a href="#cb49-706" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_samples =</span> <span class="fu">n</span>(), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-707"><a href="#cb49-707" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span> <span class="st">"All samples"</span>)</span>
<span id="cb49-708"><a href="#cb49-708" aria-hidden="true" tabindex="-1"></a>read_counts_tot <span class="ot">&lt;-</span> read_counts_grp <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-709"><a href="#cb49-709" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb49-710"><a href="#cb49-710" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-711"><a href="#cb49-711" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">country=</span> <span class="st">"All countries"</span>)</span>
<span id="cb49-712"><a href="#cb49-712" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(read_counts_grp, read_counts_tot) <span class="sc">%&gt;%</span></span>
<span id="cb49-713"><a href="#cb49-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb49-714"><a href="#cb49-714" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">factor</span>(sample, <span class="at">levels=</span><span class="fu">c</span>(<span class="fu">levels</span>(libraries<span class="sc">$</span>sample), <span class="st">"All samples"</span>)))</span>
<span id="cb49-715"><a href="#cb49-715" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-716"><a href="#cb49-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-717"><a href="#cb49-717" aria-hidden="true" tabindex="-1"></a>Applying a disjunctive cutoff at S=20 identifies 325,390 read pairs as human-viral. This gives an overall relative HV abundance of $8.19 \times 10^{-6}$; higher than any other DNA WW dataset I've analyzed and competitive with many RNA datasets:</span>
<span id="cb49-718"><a href="#cb49-718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-721"><a href="#cb49-721" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-722"><a href="#cb49-722" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-ra</span></span>
<span id="cb49-723"><a href="#cb49-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-724"><a href="#cb49-724" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb49-725"><a href="#cb49-725" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb49-726"><a href="#cb49-726" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>country)) <span class="sc">+</span></span>
<span id="cb49-727"><a href="#cb49-727" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_hv)) <span class="sc">+</span></span>
<span id="cb49-728"><a href="#cb49-728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb49-729"><a href="#cb49-729" aria-hidden="true" tabindex="-1"></a>  theme_kit <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)))</span>
<span id="cb49-730"><a href="#cb49-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-731"><a href="#cb49-731" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb49-732"><a href="#cb49-732" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-733"><a href="#cb49-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-736"><a href="#cb49-736" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-737"><a href="#cb49-737" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-hv-past</span></span>
<span id="cb49-738"><a href="#cb49-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-739"><a href="#cb49-739" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate past RA values</span></span>
<span id="cb49-740"><a href="#cb49-740" aria-hidden="true" tabindex="-1"></a>ra_past <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb49-741"><a href="#cb49-741" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-742"><a href="#cb49-742" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-743"><a href="#cb49-743" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-744"><a href="#cb49-744" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-745"><a href="#cb49-745" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-746"><a href="#cb49-746" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb49-747"><a href="#cb49-747" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-748"><a href="#cb49-748" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb49-749"><a href="#cb49-749" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-750"><a href="#cb49-750" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-751"><a href="#cb49-751" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-752"><a href="#cb49-752" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-753"><a href="#cb49-753" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Leung"</span>, <span class="fl">1.73e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-754"><a href="#cb49-754" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brinch"</span>, <span class="fl">3.88e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-755"><a href="#cb49-755" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Bengtsson-Palme"</span>, <span class="fl">8.86e-8</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-756"><a href="#cb49-756" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Ng"</span>, <span class="fl">2.90e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb49-757"><a href="#cb49-757" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Maritz"</span>, <span class="fl">9.42e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span id="cb49-758"><a href="#cb49-758" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-759"><a href="#cb49-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-760"><a href="#cb49-760" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate new RA values</span></span>
<span id="cb49-761"><a href="#cb49-761" aria-hidden="true" tabindex="-1"></a>ra_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb49-762"><a href="#cb49-762" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Munk"</span>, <span class="fl">8.19e-6</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>)</span>
<span id="cb49-763"><a href="#cb49-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-764"><a href="#cb49-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-765"><a href="#cb49-765" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-766"><a href="#cb49-766" aria-hidden="true" tabindex="-1"></a>scale_color_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb49-767"><a href="#cb49-767" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb49-768"><a href="#cb49-768" aria-hidden="true" tabindex="-1"></a>ra_comp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ra_past, ra_new) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">fct_inorder</span>(dataset))</span>
<span id="cb49-769"><a href="#cb49-769" aria-hidden="true" tabindex="-1"></a>g_ra_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ra_comp, <span class="fu">aes</span>(<span class="at">y=</span>dataset, <span class="at">x=</span>ra, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb49-770"><a href="#cb49-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb49-771"><a href="#cb49-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb49-772"><a href="#cb49-772" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb49-773"><a href="#cb49-773" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb49-774"><a href="#cb49-774" aria-hidden="true" tabindex="-1"></a>g_ra_comp</span>
<span id="cb49-775"><a href="#cb49-775" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-776"><a href="#cb49-776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-777"><a href="#cb49-777" aria-hidden="true" tabindex="-1"></a>One potential explanation for the higher HV fraction in the Munk data compared to other DNA WW datasets is the sample location: whereas Brinch, Maritz, Bengtsson-Palme and Ng are all from highly developed economies with good sanitation, Munk includes samples from numerous countries including many with much lower incomes and development scores. To quickly test this, I took the most recent Human Development Index dataset from the UN (2022<span class="ot">[^1]</span>) and GDP per capita dataset from the World Bank (PPP, 2019). In both cases, there was a weak negative correlation between the development metric and measured human-viral load:</span>
<span id="cb49-778"><a href="#cb49-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-779"><a href="#cb49-779" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>I wasn't able to quickly find any HDI datasets other than the most recent one, and it didn't seem worth doing serious digging for this quick analysis.</span>
<span id="cb49-780"><a href="#cb49-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-783"><a href="#cb49-783" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-784"><a href="#cb49-784" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: dev-metrics-linear</span></span>
<span id="cb49-785"><a href="#cb49-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-786"><a href="#cb49-786" aria-hidden="true" tabindex="-1"></a><span class="co"># HDI</span></span>
<span id="cb49-787"><a href="#cb49-787" aria-hidden="true" tabindex="-1"></a>hdi_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"hdi.csv"</span>)</span>
<span id="cb49-788"><a href="#cb49-788" aria-hidden="true" tabindex="-1"></a>hdi <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(hdi_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-789"><a href="#cb49-789" aria-hidden="true" tabindex="-1"></a>read_counts_hdi <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(read_counts_grp, hdi, <span class="at">by=</span><span class="st">"country"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-790"><a href="#cb49-790" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb49-791"><a href="#cb49-791" aria-hidden="true" tabindex="-1"></a>         <span class="at">log_p =</span> <span class="fu">log10</span>(p_reads_hv))</span>
<span id="cb49-792"><a href="#cb49-792" aria-hidden="true" tabindex="-1"></a>g_hdi <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(read_counts_hdi, <span class="at">x=</span><span class="st">"HDI"</span>, <span class="at">y=</span><span class="st">"p_reads_hv"</span>,</span>
<span id="cb49-793"><a href="#cb49-793" aria-hidden="true" tabindex="-1"></a>                   <span class="at">add =</span> <span class="st">"reg.line"</span>) <span class="sc">+</span></span>
<span id="cb49-794"><a href="#cb49-794" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">method=</span><span class="st">"pearson"</span>) <span class="sc">+</span></span>
<span id="cb49-795"><a href="#cb49-795" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb49-796"><a href="#cb49-796" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="st">"HDI (2022)"</span>) <span class="sc">+</span></span>
<span id="cb49-797"><a href="#cb49-797" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"HV RA"</span>) <span class="sc">+</span></span>
<span id="cb49-798"><a href="#cb49-798" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb49-799"><a href="#cb49-799" aria-hidden="true" tabindex="-1"></a>g_hdi</span>
<span id="cb49-800"><a href="#cb49-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-801"><a href="#cb49-801" aria-hidden="true" tabindex="-1"></a><span class="co"># GDP</span></span>
<span id="cb49-802"><a href="#cb49-802" aria-hidden="true" tabindex="-1"></a>gdp_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir_base, <span class="st">"gdp.csv"</span>)</span>
<span id="cb49-803"><a href="#cb49-803" aria-hidden="true" tabindex="-1"></a>gdp <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(gdp_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-804"><a href="#cb49-804" aria-hidden="true" tabindex="-1"></a>read_counts_gdp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(read_counts_grp, gdp, <span class="at">by=</span><span class="st">"country"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-805"><a href="#cb49-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw,</span>
<span id="cb49-806"><a href="#cb49-806" aria-hidden="true" tabindex="-1"></a>         <span class="at">log_p =</span> <span class="fu">log10</span>(p_reads_hv),</span>
<span id="cb49-807"><a href="#cb49-807" aria-hidden="true" tabindex="-1"></a>         <span class="at">log_gdp =</span> <span class="fu">log10</span>(gdp_per_capita_ppp))</span>
<span id="cb49-808"><a href="#cb49-808" aria-hidden="true" tabindex="-1"></a>g_gdp <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(read_counts_gdp, <span class="at">x=</span><span class="st">"log_gdp"</span>, <span class="at">y=</span><span class="st">"p_reads_hv"</span>,</span>
<span id="cb49-809"><a href="#cb49-809" aria-hidden="true" tabindex="-1"></a>                   <span class="at">add =</span> <span class="st">"reg.line"</span>) <span class="sc">+</span></span>
<span id="cb49-810"><a href="#cb49-810" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">method =</span> <span class="st">"pearson"</span>) <span class="sc">+</span></span>
<span id="cb49-811"><a href="#cb49-811" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="st">"Log GDP per Capita (PPP, Int$, 2019)"</span>, <span class="at">labels =</span> <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">"1e+"</span>, x)) <span class="sc">+</span></span>
<span id="cb49-812"><a href="#cb49-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb49-813"><a href="#cb49-813" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb49-814"><a href="#cb49-814" aria-hidden="true" tabindex="-1"></a>g_gdp</span>
<span id="cb49-815"><a href="#cb49-815" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-816"><a href="#cb49-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-817"><a href="#cb49-817" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: taxonomy and composition</span></span>
<span id="cb49-818"><a href="#cb49-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-819"><a href="#cb49-819" aria-hidden="true" tabindex="-1"></a>In investigating the taxonomy of human-infecting virus reads, I restricted my analysis to samples with more than 5 HV read pairs total across all viruses, to reduce noise arising from extremely low HV read counts in some samples. 1,129 samples met this criterion.</span>
<span id="cb49-820"><a href="#cb49-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-821"><a href="#cb49-821" aria-hidden="true" tabindex="-1"></a>As usual, at the family level, most samples were dominated by *Adenoviridae*, *Polyomaviridae* and *Papillomaviridae.* Three other families, *Parvoviridae*, *Circoviridae* and *Herpesviridae*, also showed substantial prevalence.</span>
<span id="cb49-822"><a href="#cb49-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-825"><a href="#cb49-825" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-826"><a href="#cb49-826" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb49-827"><a href="#cb49-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-828"><a href="#cb49-828" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb49-829"><a href="#cb49-829" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb49-830"><a href="#cb49-830" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb49-831"><a href="#cb49-831" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb49-832"><a href="#cb49-832" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">585893</span>] <span class="ot">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span id="cb49-833"><a href="#cb49-833" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333922</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span id="cb49-834"><a href="#cb49-834" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334207</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span id="cb49-835"><a href="#cb49-835" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">369960</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span id="cb49-836"><a href="#cb49-836" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333924</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span id="cb49-837"><a href="#cb49-837" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687329</span>] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb49-838"><a href="#cb49-838" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">325455</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span id="cb49-839"><a href="#cb49-839" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333750</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span id="cb49-840"><a href="#cb49-840" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb49-841"><a href="#cb49-841" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334202</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span id="cb49-842"><a href="#cb49-842" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">197911</span>] <span class="ot">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span id="cb49-843"><a href="#cb49-843" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">186938</span>] <span class="ot">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span id="cb49-844"><a href="#cb49-844" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333926</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span id="cb49-845"><a href="#cb49-845" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337051</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span id="cb49-846"><a href="#cb49-846" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337043</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span id="cb49-847"><a href="#cb49-847" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb49-848"><a href="#cb49-848" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334204</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span id="cb49-849"><a href="#cb49-849" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334208</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span id="cb49-850"><a href="#cb49-850" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333928</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span id="cb49-851"><a href="#cb49-851" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337039</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span id="cb49-852"><a href="#cb49-852" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333929</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 3"</span></span>
<span id="cb49-853"><a href="#cb49-853" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337042</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 7"</span></span>
<span id="cb49-854"><a href="#cb49-854" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334203</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 1"</span></span>
<span id="cb49-855"><a href="#cb49-855" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333757</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 8"</span></span>
<span id="cb49-856"><a href="#cb49-856" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337050</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 6"</span></span>
<span id="cb49-857"><a href="#cb49-857" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333767</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 3"</span></span>
<span id="cb49-858"><a href="#cb49-858" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333754</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 10"</span></span>
<span id="cb49-859"><a href="#cb49-859" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687363</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 24"</span></span>
<span id="cb49-860"><a href="#cb49-860" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687342</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 3"</span></span>
<span id="cb49-861"><a href="#cb49-861" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687359</span>] <span class="ot">&lt;-</span> <span class="st">"Torque teno virus 20"</span></span>
<span id="cb49-862"><a href="#cb49-862" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194441</span>] <span class="ot">&lt;-</span> <span class="st">"Primate T-lymphotropic virus 2"</span></span>
<span id="cb49-863"><a href="#cb49-863" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334209</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 5"</span></span>
<span id="cb49-864"><a href="#cb49-864" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194965</span>] <span class="ot">&lt;-</span> <span class="st">"Aichivirus B"</span></span>
<span id="cb49-865"><a href="#cb49-865" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333930</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 4"</span></span>
<span id="cb49-866"><a href="#cb49-866" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337048</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 1"</span></span>
<span id="cb49-867"><a href="#cb49-867" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337041</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 9"</span></span>
<span id="cb49-868"><a href="#cb49-868" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337049</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 11"</span></span>
<span id="cb49-869"><a href="#cb49-869" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337044</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 5"</span></span>
<span id="cb49-870"><a href="#cb49-870" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-871"><a href="#cb49-871" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter samples and add viral taxa information</span></span>
<span id="cb49-872"><a href="#cb49-872" aria-hidden="true" tabindex="-1"></a>samples_keep <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n_reads_hv <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(sample)</span>
<span id="cb49-873"><a href="#cb49-873" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_keep, hv_status) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) </span>
<span id="cb49-874"><a href="#cb49-874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-875"><a href="#cb49-875" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb49-876"><a href="#cb49-876" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb49-877"><a href="#cb49-877" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb49-878"><a href="#cb49-878" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb49-879"><a href="#cb49-879" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb49-880"><a href="#cb49-880" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb49-881"><a href="#cb49-881" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb49-882"><a href="#cb49-882" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb49-883"><a href="#cb49-883" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb49-884"><a href="#cb49-884" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb49-885"><a href="#cb49-885" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb49-886"><a href="#cb49-886" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb49-887"><a href="#cb49-887" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb49-888"><a href="#cb49-888" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb49-889"><a href="#cb49-889" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb49-890"><a href="#cb49-890" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-891"><a href="#cb49-891" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb49-892"><a href="#cb49-892" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb49-893"><a href="#cb49-893" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb49-894"><a href="#cb49-894" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb49-895"><a href="#cb49-895" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb49-896"><a href="#cb49-896" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb49-897"><a href="#cb49-897" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb49-898"><a href="#cb49-898" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb49-899"><a href="#cb49-899" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb49-900"><a href="#cb49-900" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb49-901"><a href="#cb49-901" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb49-902"><a href="#cb49-902" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb49-903"><a href="#cb49-903" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb49-904"><a href="#cb49-904" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb49-905"><a href="#cb49-905" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb49-906"><a href="#cb49-906" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb49-907"><a href="#cb49-907" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb49-908"><a href="#cb49-908" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-909"><a href="#cb49-909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-912"><a href="#cb49-912" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-913"><a href="#cb49-913" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb49-914"><a href="#cb49-914" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-915"><a href="#cb49-915" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-916"><a href="#cb49-916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-917"><a href="#cb49-917" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.02</span></span>
<span id="cb49-918"><a href="#cb49-918" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-919"><a href="#cb49-919" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb49-920"><a href="#cb49-920" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-921"><a href="#cb49-921" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-922"><a href="#cb49-922" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-923"><a href="#cb49-923" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-924"><a href="#cb49-924" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-925"><a href="#cb49-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-926"><a href="#cb49-926" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-927"><a href="#cb49-927" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-928"><a href="#cb49-928" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-929"><a href="#cb49-929" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb49-930"><a href="#cb49-930" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-931"><a href="#cb49-931" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-932"><a href="#cb49-932" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-933"><a href="#cb49-933" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb49-934"><a href="#cb49-934" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-935"><a href="#cb49-935" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-936"><a href="#cb49-936" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-937"><a href="#cb49-937" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-938"><a href="#cb49-938" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb49-939"><a href="#cb49-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-940"><a href="#cb49-940" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-941"><a href="#cb49-941" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-942"><a href="#cb49-942" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-943"><a href="#cb49-943" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-944"><a href="#cb49-944" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-945"><a href="#cb49-945" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-946"><a href="#cb49-946" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral family"</span>) <span class="sc">+</span></span>
<span id="cb49-947"><a href="#cb49-947" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Family composition of human-viral reads"</span>) <span class="sc">+</span></span>
<span id="cb49-948"><a href="#cb49-948" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">4</span>)) <span class="sc">+</span></span>
<span id="cb49-949"><a href="#cb49-949" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-950"><a href="#cb49-950" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb49-951"><a href="#cb49-951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-952"><a href="#cb49-952" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent families for text</span></span>
<span id="cb49-953"><a href="#cb49-953" aria-hidden="true" tabindex="-1"></a>hv_family_collate <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-954"><a href="#cb49-954" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-955"><a href="#cb49-955" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_max =</span> <span class="fu">max</span>(p_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-956"><a href="#cb49-956" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-957"><a href="#cb49-957" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-958"><a href="#cb49-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-959"><a href="#cb49-959" aria-hidden="true" tabindex="-1"></a>In investigating individual viral families, to avoid distortions from a few rare reads, I restricted myself to samples where that family made up at least 10% of human-viral reads:</span>
<span id="cb49-960"><a href="#cb49-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-963"><a href="#cb49-963" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-964"><a href="#cb49-964" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-adeno</span></span>
<span id="cb49-965"><a href="#cb49-965" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-966"><a href="#cb49-966" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-967"><a href="#cb49-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-968"><a href="#cb49-968" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb49-969"><a href="#cb49-969" aria-hidden="true" tabindex="-1"></a>taxid_adeno <span class="ot">&lt;-</span> <span class="dv">10508</span></span>
<span id="cb49-970"><a href="#cb49-970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-971"><a href="#cb49-971" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of adenoviridae reads</span></span>
<span id="cb49-972"><a href="#cb49-972" aria-hidden="true" tabindex="-1"></a>adeno_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno) <span class="sc">%&gt;%</span></span>
<span id="cb49-973"><a href="#cb49-973" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-974"><a href="#cb49-974" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-975"><a href="#cb49-975" aria-hidden="true" tabindex="-1"></a>adeno_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-976"><a href="#cb49-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_adeno, sample <span class="sc">%in%</span> adeno_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-977"><a href="#cb49-977" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-978"><a href="#cb49-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-979"><a href="#cb49-979" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each adenoviridae species</span></span>
<span id="cb49-980"><a href="#cb49-980" aria-hidden="true" tabindex="-1"></a>adeno_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-981"><a href="#cb49-981" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> adeno_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-982"><a href="#cb49-982" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-983"><a href="#cb49-983" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-984"><a href="#cb49-984" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-985"><a href="#cb49-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_adeno =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-986"><a href="#cb49-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-987"><a href="#cb49-987" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-988"><a href="#cb49-988" aria-hidden="true" tabindex="-1"></a>adeno_species_major_tab <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-989"><a href="#cb49-989" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">==</span> <span class="fu">max</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-990"><a href="#cb49-990" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-991"><a href="#cb49-991" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_adeno)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-992"><a href="#cb49-992" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_adeno <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-993"><a href="#cb49-993" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_major <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-994"><a href="#cb49-994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> adeno_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-995"><a href="#cb49-995" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-996"><a href="#cb49-996" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-997"><a href="#cb49-997" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_adeno =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-998"><a href="#cb49-998" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_adeno =</span> <span class="fu">sum</span>(p_reads_adeno), </span>
<span id="cb49-999"><a href="#cb49-999" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1000"><a href="#cb49-1000" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1001"><a href="#cb49-1001" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(adeno_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1002"><a href="#cb49-1002" aria-hidden="true" tabindex="-1"></a>adeno_species_counts_display <span class="ot">&lt;-</span> adeno_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1003"><a href="#cb49-1003" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_adeno, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1004"><a href="#cb49-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1005"><a href="#cb49-1005" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1006"><a href="#cb49-1006" aria-hidden="true" tabindex="-1"></a>g_adeno_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1007"><a href="#cb49-1007" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>adeno_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1008"><a href="#cb49-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adenoviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1009"><a href="#cb49-1009" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1010"><a href="#cb49-1010" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1011"><a href="#cb49-1011" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1012"><a href="#cb49-1012" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Adenoviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1013"><a href="#cb49-1013" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1014"><a href="#cb49-1014" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1015"><a href="#cb49-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1016"><a href="#cb49-1016" aria-hidden="true" tabindex="-1"></a>g_adeno_species</span>
<span id="cb49-1017"><a href="#cb49-1017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1018"><a href="#cb49-1018" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1019"><a href="#cb49-1019" aria-hidden="true" tabindex="-1"></a>adeno_species_collate <span class="ot">&lt;-</span> adeno_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1020"><a href="#cb49-1020" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_adeno), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1021"><a href="#cb49-1021" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1022"><a href="#cb49-1022" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1023"><a href="#cb49-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1026"><a href="#cb49-1026" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1027"><a href="#cb49-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-polyoma</span></span>
<span id="cb49-1028"><a href="#cb49-1028" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1029"><a href="#cb49-1029" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-1030"><a href="#cb49-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1031"><a href="#cb49-1031" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb49-1032"><a href="#cb49-1032" aria-hidden="true" tabindex="-1"></a>taxid_polyoma <span class="ot">&lt;-</span> <span class="dv">151341</span></span>
<span id="cb49-1033"><a href="#cb49-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1034"><a href="#cb49-1034" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of polyomaviridae reads</span></span>
<span id="cb49-1035"><a href="#cb49-1035" aria-hidden="true" tabindex="-1"></a>polyoma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma) <span class="sc">%&gt;%</span></span>
<span id="cb49-1036"><a href="#cb49-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1037"><a href="#cb49-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-1038"><a href="#cb49-1038" aria-hidden="true" tabindex="-1"></a>polyoma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-1039"><a href="#cb49-1039" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_polyoma, sample <span class="sc">%in%</span> polyoma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-1040"><a href="#cb49-1040" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-1041"><a href="#cb49-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1042"><a href="#cb49-1042" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each polyomaviridae species</span></span>
<span id="cb49-1043"><a href="#cb49-1043" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-1044"><a href="#cb49-1044" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> polyoma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-1045"><a href="#cb49-1045" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-1046"><a href="#cb49-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1047"><a href="#cb49-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-1048"><a href="#cb49-1048" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_polyoma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-1049"><a href="#cb49-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1050"><a href="#cb49-1050" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-1051"><a href="#cb49-1051" aria-hidden="true" tabindex="-1"></a>polyoma_species_major_tab <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1052"><a href="#cb49-1052" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">==</span> <span class="fu">max</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1053"><a href="#cb49-1053" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1054"><a href="#cb49-1054" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_polyoma)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1055"><a href="#cb49-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_polyoma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-1056"><a href="#cb49-1056" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_major <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-1057"><a href="#cb49-1057" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> polyoma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-1058"><a href="#cb49-1058" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1059"><a href="#cb49-1059" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-1060"><a href="#cb49-1060" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_polyoma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-1061"><a href="#cb49-1061" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_polyoma =</span> <span class="fu">sum</span>(p_reads_polyoma), </span>
<span id="cb49-1062"><a href="#cb49-1062" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1063"><a href="#cb49-1063" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1064"><a href="#cb49-1064" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(polyoma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1065"><a href="#cb49-1065" aria-hidden="true" tabindex="-1"></a>polyoma_species_counts_display <span class="ot">&lt;-</span> polyoma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1066"><a href="#cb49-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_polyoma, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1067"><a href="#cb49-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1068"><a href="#cb49-1068" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1069"><a href="#cb49-1069" aria-hidden="true" tabindex="-1"></a>g_polyoma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1070"><a href="#cb49-1070" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>polyoma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1071"><a href="#cb49-1071" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Polyomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1072"><a href="#cb49-1072" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1073"><a href="#cb49-1073" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1074"><a href="#cb49-1074" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1075"><a href="#cb49-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Polyomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1076"><a href="#cb49-1076" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1077"><a href="#cb49-1077" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1078"><a href="#cb49-1078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1079"><a href="#cb49-1079" aria-hidden="true" tabindex="-1"></a>g_polyoma_species</span>
<span id="cb49-1080"><a href="#cb49-1080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1081"><a href="#cb49-1081" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1082"><a href="#cb49-1082" aria-hidden="true" tabindex="-1"></a>polyoma_species_collate <span class="ot">&lt;-</span> polyoma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1083"><a href="#cb49-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_polyoma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1084"><a href="#cb49-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1085"><a href="#cb49-1085" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1086"><a href="#cb49-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1089"><a href="#cb49-1089" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1090"><a href="#cb49-1090" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-papilloma</span></span>
<span id="cb49-1091"><a href="#cb49-1091" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1092"><a href="#cb49-1092" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-1093"><a href="#cb49-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1094"><a href="#cb49-1094" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb49-1095"><a href="#cb49-1095" aria-hidden="true" tabindex="-1"></a>taxid_papilloma <span class="ot">&lt;-</span> <span class="dv">151340</span></span>
<span id="cb49-1096"><a href="#cb49-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1097"><a href="#cb49-1097" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of papillomaviridae reads</span></span>
<span id="cb49-1098"><a href="#cb49-1098" aria-hidden="true" tabindex="-1"></a>papilloma_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma) <span class="sc">%&gt;%</span></span>
<span id="cb49-1099"><a href="#cb49-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1100"><a href="#cb49-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-1101"><a href="#cb49-1101" aria-hidden="true" tabindex="-1"></a>papilloma_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-1102"><a href="#cb49-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_papilloma, sample <span class="sc">%in%</span> papilloma_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-1103"><a href="#cb49-1103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-1104"><a href="#cb49-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1105"><a href="#cb49-1105" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each papillomaviridae species</span></span>
<span id="cb49-1106"><a href="#cb49-1106" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-1107"><a href="#cb49-1107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> papilloma_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-1108"><a href="#cb49-1108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-1109"><a href="#cb49-1109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1110"><a href="#cb49-1110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-1111"><a href="#cb49-1111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_papilloma =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-1112"><a href="#cb49-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1113"><a href="#cb49-1113" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-1114"><a href="#cb49-1114" aria-hidden="true" tabindex="-1"></a>papilloma_species_major_tab <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1115"><a href="#cb49-1115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">==</span> <span class="fu">max</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1116"><a href="#cb49-1116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1117"><a href="#cb49-1117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_papilloma)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1118"><a href="#cb49-1118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papilloma <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-1119"><a href="#cb49-1119" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_major <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-1120"><a href="#cb49-1120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> papilloma_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-1121"><a href="#cb49-1121" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1122"><a href="#cb49-1122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-1123"><a href="#cb49-1123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_papilloma =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-1124"><a href="#cb49-1124" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_papilloma =</span> <span class="fu">sum</span>(p_reads_papilloma), </span>
<span id="cb49-1125"><a href="#cb49-1125" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1126"><a href="#cb49-1126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1127"><a href="#cb49-1127" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(papilloma_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1128"><a href="#cb49-1128" aria-hidden="true" tabindex="-1"></a>papilloma_species_counts_display <span class="ot">&lt;-</span> papilloma_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1129"><a href="#cb49-1129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_papilloma, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1130"><a href="#cb49-1130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1131"><a href="#cb49-1131" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1132"><a href="#cb49-1132" aria-hidden="true" tabindex="-1"></a>g_papilloma_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1133"><a href="#cb49-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>papilloma_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1134"><a href="#cb49-1134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Papillomaviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1135"><a href="#cb49-1135" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1136"><a href="#cb49-1136" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1137"><a href="#cb49-1137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1138"><a href="#cb49-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Papillomaviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1139"><a href="#cb49-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1140"><a href="#cb49-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1141"><a href="#cb49-1141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1142"><a href="#cb49-1142" aria-hidden="true" tabindex="-1"></a>g_papilloma_species</span>
<span id="cb49-1143"><a href="#cb49-1143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1144"><a href="#cb49-1144" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1145"><a href="#cb49-1145" aria-hidden="true" tabindex="-1"></a>papilloma_species_collate <span class="ot">&lt;-</span> papilloma_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1146"><a href="#cb49-1146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_papilloma), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1147"><a href="#cb49-1147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1148"><a href="#cb49-1148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1149"><a href="#cb49-1149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1152"><a href="#cb49-1152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1153"><a href="#cb49-1153" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-parvo</span></span>
<span id="cb49-1154"><a href="#cb49-1154" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1155"><a href="#cb49-1155" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-1156"><a href="#cb49-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1157"><a href="#cb49-1157" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb49-1158"><a href="#cb49-1158" aria-hidden="true" tabindex="-1"></a>taxid_parvo <span class="ot">&lt;-</span> <span class="dv">10780</span></span>
<span id="cb49-1159"><a href="#cb49-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1160"><a href="#cb49-1160" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of parvoviridae reads</span></span>
<span id="cb49-1161"><a href="#cb49-1161" aria-hidden="true" tabindex="-1"></a>parvo_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_parvo) <span class="sc">%&gt;%</span></span>
<span id="cb49-1162"><a href="#cb49-1162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1163"><a href="#cb49-1163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-1164"><a href="#cb49-1164" aria-hidden="true" tabindex="-1"></a>parvo_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-1165"><a href="#cb49-1165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_parvo, sample <span class="sc">%in%</span> parvo_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-1166"><a href="#cb49-1166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-1167"><a href="#cb49-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1168"><a href="#cb49-1168" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each parvoviridae species</span></span>
<span id="cb49-1169"><a href="#cb49-1169" aria-hidden="true" tabindex="-1"></a>parvo_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-1170"><a href="#cb49-1170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> parvo_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-1171"><a href="#cb49-1171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-1172"><a href="#cb49-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1173"><a href="#cb49-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-1174"><a href="#cb49-1174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_parvo =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-1175"><a href="#cb49-1175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1176"><a href="#cb49-1176" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-1177"><a href="#cb49-1177" aria-hidden="true" tabindex="-1"></a>parvo_species_major_tab <span class="ot">&lt;-</span> parvo_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1178"><a href="#cb49-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_parvo <span class="sc">==</span> <span class="fu">max</span>(p_reads_parvo)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1179"><a href="#cb49-1179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1180"><a href="#cb49-1180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_parvo)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1181"><a href="#cb49-1181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_parvo <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-1182"><a href="#cb49-1182" aria-hidden="true" tabindex="-1"></a>parvo_species_counts_major <span class="ot">&lt;-</span> parvo_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-1183"><a href="#cb49-1183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> parvo_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-1184"><a href="#cb49-1184" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1185"><a href="#cb49-1185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-1186"><a href="#cb49-1186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_parvo =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-1187"><a href="#cb49-1187" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_parvo =</span> <span class="fu">sum</span>(p_reads_parvo), </span>
<span id="cb49-1188"><a href="#cb49-1188" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1189"><a href="#cb49-1189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1190"><a href="#cb49-1190" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(parvo_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1191"><a href="#cb49-1191" aria-hidden="true" tabindex="-1"></a>parvo_species_counts_display <span class="ot">&lt;-</span> parvo_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1192"><a href="#cb49-1192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_parvo, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1193"><a href="#cb49-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1194"><a href="#cb49-1194" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1195"><a href="#cb49-1195" aria-hidden="true" tabindex="-1"></a>g_parvo_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1196"><a href="#cb49-1196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>parvo_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1197"><a href="#cb49-1197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Parvoviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1198"><a href="#cb49-1198" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1199"><a href="#cb49-1199" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1200"><a href="#cb49-1200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1201"><a href="#cb49-1201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Parvoviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1202"><a href="#cb49-1202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1203"><a href="#cb49-1203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1204"><a href="#cb49-1204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1205"><a href="#cb49-1205" aria-hidden="true" tabindex="-1"></a>g_parvo_species</span>
<span id="cb49-1206"><a href="#cb49-1206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1207"><a href="#cb49-1207" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1208"><a href="#cb49-1208" aria-hidden="true" tabindex="-1"></a>parvo_species_collate <span class="ot">&lt;-</span> parvo_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1209"><a href="#cb49-1209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_parvo), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1210"><a href="#cb49-1210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1211"><a href="#cb49-1211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1212"><a href="#cb49-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1215"><a href="#cb49-1215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1216"><a href="#cb49-1216" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-circo</span></span>
<span id="cb49-1217"><a href="#cb49-1217" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1218"><a href="#cb49-1218" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-1219"><a href="#cb49-1219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1220"><a href="#cb49-1220" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb49-1221"><a href="#cb49-1221" aria-hidden="true" tabindex="-1"></a>taxid_circo <span class="ot">&lt;-</span> <span class="dv">39724</span></span>
<span id="cb49-1222"><a href="#cb49-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1223"><a href="#cb49-1223" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of circoviridae reads</span></span>
<span id="cb49-1224"><a href="#cb49-1224" aria-hidden="true" tabindex="-1"></a>circo_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_circo) <span class="sc">%&gt;%</span></span>
<span id="cb49-1225"><a href="#cb49-1225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1226"><a href="#cb49-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-1227"><a href="#cb49-1227" aria-hidden="true" tabindex="-1"></a>circo_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-1228"><a href="#cb49-1228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_circo, sample <span class="sc">%in%</span> circo_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-1229"><a href="#cb49-1229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-1230"><a href="#cb49-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1231"><a href="#cb49-1231" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each circoviridae species</span></span>
<span id="cb49-1232"><a href="#cb49-1232" aria-hidden="true" tabindex="-1"></a>circo_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-1233"><a href="#cb49-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> circo_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-1234"><a href="#cb49-1234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-1235"><a href="#cb49-1235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1236"><a href="#cb49-1236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-1237"><a href="#cb49-1237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_circo =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-1238"><a href="#cb49-1238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1239"><a href="#cb49-1239" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-1240"><a href="#cb49-1240" aria-hidden="true" tabindex="-1"></a>circo_species_major_tab <span class="ot">&lt;-</span> circo_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1241"><a href="#cb49-1241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_circo <span class="sc">==</span> <span class="fu">max</span>(p_reads_circo)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1242"><a href="#cb49-1242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1243"><a href="#cb49-1243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_circo)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1244"><a href="#cb49-1244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_circo <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-1245"><a href="#cb49-1245" aria-hidden="true" tabindex="-1"></a>circo_species_counts_major <span class="ot">&lt;-</span> circo_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-1246"><a href="#cb49-1246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> circo_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-1247"><a href="#cb49-1247" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1248"><a href="#cb49-1248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-1249"><a href="#cb49-1249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_circo =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-1250"><a href="#cb49-1250" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_circo =</span> <span class="fu">sum</span>(p_reads_circo), </span>
<span id="cb49-1251"><a href="#cb49-1251" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1252"><a href="#cb49-1252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1253"><a href="#cb49-1253" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(circo_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1254"><a href="#cb49-1254" aria-hidden="true" tabindex="-1"></a>circo_species_counts_display <span class="ot">&lt;-</span> circo_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1255"><a href="#cb49-1255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_circo, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1256"><a href="#cb49-1256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1257"><a href="#cb49-1257" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1258"><a href="#cb49-1258" aria-hidden="true" tabindex="-1"></a>g_circo_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1259"><a href="#cb49-1259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>circo_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1260"><a href="#cb49-1260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Circoviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1261"><a href="#cb49-1261" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1262"><a href="#cb49-1262" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1263"><a href="#cb49-1263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1264"><a href="#cb49-1264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Circoviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1265"><a href="#cb49-1265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1266"><a href="#cb49-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1267"><a href="#cb49-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1268"><a href="#cb49-1268" aria-hidden="true" tabindex="-1"></a>g_circo_species</span>
<span id="cb49-1269"><a href="#cb49-1269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1270"><a href="#cb49-1270" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1271"><a href="#cb49-1271" aria-hidden="true" tabindex="-1"></a>circo_species_collate <span class="ot">&lt;-</span> circo_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1272"><a href="#cb49-1272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_circo), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1273"><a href="#cb49-1273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1274"><a href="#cb49-1274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1275"><a href="#cb49-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1278"><a href="#cb49-1278" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1279"><a href="#cb49-1279" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species-herpes</span></span>
<span id="cb49-1280"><a href="#cb49-1280" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1281"><a href="#cb49-1281" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb49-1282"><a href="#cb49-1282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1283"><a href="#cb49-1283" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb49-1284"><a href="#cb49-1284" aria-hidden="true" tabindex="-1"></a>taxid_herpes <span class="ot">&lt;-</span> <span class="dv">10292</span></span>
<span id="cb49-1285"><a href="#cb49-1285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1286"><a href="#cb49-1286" aria-hidden="true" tabindex="-1"></a><span class="co"># Get set of herpesviridae reads</span></span>
<span id="cb49-1287"><a href="#cb49-1287" aria-hidden="true" tabindex="-1"></a>herpes_samples <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes) <span class="sc">%&gt;%</span></span>
<span id="cb49-1288"><a href="#cb49-1288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1289"><a href="#cb49-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb49-1290"><a href="#cb49-1290" aria-hidden="true" tabindex="-1"></a>herpes_ids <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb49-1291"><a href="#cb49-1291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">==</span> taxid_herpes, sample <span class="sc">%in%</span> herpes_samples) <span class="sc">%&gt;%</span></span>
<span id="cb49-1292"><a href="#cb49-1292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(seq_id)</span>
<span id="cb49-1293"><a href="#cb49-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1294"><a href="#cb49-1294" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each herpesviridae species</span></span>
<span id="cb49-1295"><a href="#cb49-1295" aria-hidden="true" tabindex="-1"></a>herpes_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span></span>
<span id="cb49-1296"><a href="#cb49-1296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(seq_id <span class="sc">%in%</span> herpes_ids) <span class="sc">%&gt;%</span></span>
<span id="cb49-1297"><a href="#cb49-1297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb49-1298"><a href="#cb49-1298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1299"><a href="#cb49-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb49-1300"><a href="#cb49-1300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_herpes =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb49-1301"><a href="#cb49-1301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1302"><a href="#cb49-1302" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb49-1303"><a href="#cb49-1303" aria-hidden="true" tabindex="-1"></a>herpes_species_major_tab <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1304"><a href="#cb49-1304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">==</span> <span class="fu">max</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1305"><a href="#cb49-1305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1306"><a href="#cb49-1306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_herpes)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1307"><a href="#cb49-1307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_herpes <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb49-1308"><a href="#cb49-1308" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_major <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb49-1309"><a href="#cb49-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> herpes_species_major_tab<span class="sc">$</span>name, </span>
<span id="cb49-1310"><a href="#cb49-1310" aria-hidden="true" tabindex="-1"></a>                               name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1311"><a href="#cb49-1311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb49-1312"><a href="#cb49-1312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_herpes =</span> <span class="fu">sum</span>(n_reads_hv),</span>
<span id="cb49-1313"><a href="#cb49-1313" aria-hidden="true" tabindex="-1"></a>            <span class="at">p_reads_herpes =</span> <span class="fu">sum</span>(p_reads_herpes), </span>
<span id="cb49-1314"><a href="#cb49-1314" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1315"><a href="#cb49-1315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb49-1316"><a href="#cb49-1316" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(herpes_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb49-1317"><a href="#cb49-1317" aria-hidden="true" tabindex="-1"></a>herpes_species_counts_display <span class="ot">&lt;-</span> herpes_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb49-1318"><a href="#cb49-1318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_herpes, <span class="at">classification =</span> name_display)</span>
<span id="cb49-1319"><a href="#cb49-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1320"><a href="#cb49-1320" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1321"><a href="#cb49-1321" aria-hidden="true" tabindex="-1"></a>g_herpes_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb49-1322"><a href="#cb49-1322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>herpes_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">width=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-1323"><a href="#cb49-1323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Herpesviridae Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb49-1324"><a href="#cb49-1324" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb49-1325"><a href="#cb49-1325" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb49-1326"><a href="#cb49-1326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb49-1327"><a href="#cb49-1327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Species composition of Herpesviridae reads"</span>) <span class="sc">+</span></span>
<span id="cb49-1328"><a href="#cb49-1328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb49-1329"><a href="#cb49-1329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb49-1330"><a href="#cb49-1330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1331"><a href="#cb49-1331" aria-hidden="true" tabindex="-1"></a>g_herpes_species</span>
<span id="cb49-1332"><a href="#cb49-1332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1333"><a href="#cb49-1333" aria-hidden="true" tabindex="-1"></a><span class="co"># Get most prominent species for text</span></span>
<span id="cb49-1334"><a href="#cb49-1334" aria-hidden="true" tabindex="-1"></a>herpes_species_collate <span class="ot">&lt;-</span> herpes_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1335"><a href="#cb49-1335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_tot =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_mean =</span> <span class="fu">mean</span>(p_reads_herpes), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1336"><a href="#cb49-1336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_reads_tot))</span>
<span id="cb49-1337"><a href="#cb49-1337" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1338"><a href="#cb49-1338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1339"><a href="#cb49-1339" aria-hidden="true" tabindex="-1"></a>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</span>
<span id="cb49-1340"><a href="#cb49-1340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1343"><a href="#cb49-1343" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb49-1344"><a href="#cb49-1344" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb49-1345"><a href="#cb49-1345" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-genera</span></span>
<span id="cb49-1346"><a href="#cb49-1346" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb49-1347"><a href="#cb49-1347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1348"><a href="#cb49-1348" aria-hidden="true" tabindex="-1"></a><span class="co"># Define reference genera</span></span>
<span id="cb49-1349"><a href="#cb49-1349" aria-hidden="true" tabindex="-1"></a>path_genera_rna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span>)</span>
<span id="cb49-1350"><a href="#cb49-1350" aria-hidden="true" tabindex="-1"></a>path_genera_dna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span id="cb49-1351"><a href="#cb49-1351" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span>)</span>
<span id="cb49-1352"><a href="#cb49-1352" aria-hidden="true" tabindex="-1"></a>path_genera <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">tibble</span>(<span class="at">name=</span>path_genera_rna, <span class="at">genome_type=</span><span class="st">"RNA genome"</span>),</span>
<span id="cb49-1353"><a href="#cb49-1353" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">tibble</span>(<span class="at">name=</span>path_genera_dna, <span class="at">genome_type=</span><span class="st">"DNA genome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1354"><a href="#cb49-1354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb49-1355"><a href="#cb49-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1356"><a href="#cb49-1356" aria-hidden="true" tabindex="-1"></a><span class="co"># Count in each sample</span></span>
<span id="cb49-1357"><a href="#cb49-1357" aria-hidden="true" tabindex="-1"></a>mrg_hv_named_all <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb49-1358"><a href="#cb49-1358" aria-hidden="true" tabindex="-1"></a>hv_reads_genus_all <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named_all, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb49-1359"><a href="#cb49-1359" aria-hidden="true" tabindex="-1"></a>n_path_genera <span class="ot">&lt;-</span> hv_reads_genus_all <span class="sc">%&gt;%</span> </span>
<span id="cb49-1360"><a href="#cb49-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1361"><a href="#cb49-1361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-1362"><a href="#cb49-1362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(path_genera, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1363"><a href="#cb49-1363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-1364"><a href="#cb49-1364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb49-1365"><a href="#cb49-1365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1366"><a href="#cb49-1366" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot out and back to add zero lines</span></span>
<span id="cb49-1367"><a href="#cb49-1367" aria-hidden="true" tabindex="-1"></a>n_path_genera_out <span class="ot">&lt;-</span> n_path_genera <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, name, n_reads_viral) <span class="sc">%&gt;%</span></span>
<span id="cb49-1368"><a href="#cb49-1368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"name"</span>, <span class="at">values_from=</span><span class="st">"n_reads_viral"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1369"><a href="#cb49-1369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to=</span><span class="st">"name"</span>, <span class="at">values_to=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1370"><a href="#cb49-1370" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1371"><a href="#cb49-1371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(path_genera, <span class="at">by=</span><span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1372"><a href="#cb49-1372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb49-1373"><a href="#cb49-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1374"><a href="#cb49-1374" aria-hidden="true" tabindex="-1"></a><span class="do">## Aggregate across dates</span></span>
<span id="cb49-1375"><a href="#cb49-1375" aria-hidden="true" tabindex="-1"></a>n_path_genera_stype <span class="ot">&lt;-</span> n_path_genera_out <span class="sc">%&gt;%</span> </span>
<span id="cb49-1376"><a href="#cb49-1376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid, genome_type) <span class="sc">%&gt;%</span></span>
<span id="cb49-1377"><a href="#cb49-1377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb49-1378"><a href="#cb49-1378" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_viral =</span> <span class="fu">sum</span>(n_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-1379"><a href="#cb49-1379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="st">"All samples"</span>, <span class="at">location=</span><span class="st">"All locations"</span>,</span>
<span id="cb49-1380"><a href="#cb49-1380" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw,</span>
<span id="cb49-1381"><a href="#cb49-1381" aria-hidden="true" tabindex="-1"></a>         <span class="at">na_type =</span> <span class="st">"DNA"</span>)</span>
<span id="cb49-1382"><a href="#cb49-1382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1383"><a href="#cb49-1383" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb49-1384"><a href="#cb49-1384" aria-hidden="true" tabindex="-1"></a>g_path_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_path_genera_stype,</span>
<span id="cb49-1385"><a href="#cb49-1385" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y=</span>name, <span class="at">x=</span>p_reads_viral)) <span class="sc">+</span></span>
<span id="cb49-1386"><a href="#cb49-1386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb49-1387"><a href="#cb49-1387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb49-1388"><a href="#cb49-1388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genome_type<span class="sc">~</span>., <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb49-1389"><a href="#cb49-1389" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb49-1390"><a href="#cb49-1390" aria-hidden="true" tabindex="-1"></a>g_path_genera</span>
<span id="cb49-1391"><a href="#cb49-1391" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb49-1392"><a href="#cb49-1392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1393"><a href="#cb49-1393" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb49-1394"><a href="#cb49-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-1395"><a href="#cb49-1395" aria-hidden="true" tabindex="-1"></a>This is the final P2RA dataset I needed to analyze before we finish re-doing that analysis for publication, so I'm pretty happy to have it done. In terms of the results, things mostly look similar to other DNA WW datasets I've looked at, with the notable difference that the total fraction of human-infecting viruses is significantly higher. I'm still not sure what's causing this elevation; the methods used in this study don't seem any different from other studies that got much lower fractions, and the fact that this study sampled from developing countries seems like only a partial explanation.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>