<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-04-12">
<title>Will’s Public NAO Notebook - Workflow analysis of Rosario et al.&nbsp;(2018)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Rosario et al.&nbsp;(2018)</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Air sampling from a student dorm in Colorado.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 12, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>Continuing our look at air sampling datasets, we turn to <a href="https://pubs.acs.org/doi/10.1021/acs.est.7b04203">Rosario et al.&nbsp;(2018)</a>, another study of air filters, this time from HVAC filters from an undergraduate dorm building at the University of Colorado campus in Boulder. As in <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-04-12_prussin.html">Prussin</a>, samples were eluted from filters (in this case MERV-8, so less stringent than Prussin’s MERV-14 filters) and underwent both RNA and DNA sequencing – this time on an Illumina MiSeq with 2x250bp reads.</p>
<section id="the-raw-data" class="level1"><h1>The raw data</h1>
<p>The Rosario dataset comprised sequencing data from 12 individual dormitory rooms at the sampling site, as well as a pooled sample of eight additional rooms and a negative control.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Data input paths</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-04-12_rosario/"</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"sample-metadata.csv"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_adapter_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_base_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_quality_sequence_stats.tsv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span><span class="va">libraries_raw</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="va">libraries_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>na_type <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="va">room</span><span class="op">)</span>, <span class="st">"NC"</span>,</span>
<span>                              <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Pool"</span>, <span class="va">room</span><span class="op">)</span>, <span class="st">"Pool"</span>, <span class="st">"Single-Room"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         sample_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">sample_type</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Single-Room"</span>, <span class="st">"Pool"</span>, <span class="st">"NC"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         room <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="va">room</span><span class="op">)</span>, <span class="st">"Negative Control"</span>, <span class="va">room</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import QC data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adapter_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">adapter_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"_2$"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_base_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">quality_seq_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         read_pair <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to raw data</span></span>
<span><span class="va">basic_stats_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">adapter_stats_raw</span> <span class="op">&lt;-</span> <span class="va">adapter_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_base_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_base_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span><span class="va">quality_seq_stats_raw</span> <span class="op">&lt;-</span> <span class="va">quality_seq_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"raw_concat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get key values for readout</span></span>
<span><span class="va">raw_read_counts</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>rmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, rmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>,</span>
<span>            rmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">raw_read_totals</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_read_pairs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, </span>
<span>            n_bases_approx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_bases_approx</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">raw_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The 12 single-room samples yielded 0.57-0.74M (mean 0.67M) RNA-sequencing reads and 0.48M-0.84M (mean 0.67M) DNA-sequencing reads; the pooled and control samples had similar depth. In total, non-control samples yielded 8.0M RNA read pairs and 8.0M DNA read pairs (4 gigabases of sequence each).</p>
<p>Read qualities were so-so, in need of cleaning. Adapter levels were high. Inferred duplication levels were low (4.6-8.2%) in DNA reads but highly variable (3-83%) in RNA reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare data</span></span>
<span><span class="va">basic_stats_raw_metrics</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>,</span>
<span>         `# Read pairs` <span class="op">=</span> <span class="va">n_read_pairs</span>,</span>
<span>         `Total base pairs\n(approx)` <span class="op">=</span> <span class="va">n_bases_approx</span>,</span>
<span>         `% Duplicates\n(FASTQC)` <span class="op">=</span> <span class="va">percent_duplicates</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="op">(</span><span class="va">room</span><span class="op">:</span><span class="va">sample_type</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"metric"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>metric <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set up plot templates</span></span>
<span><span class="va">scale_fill_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_fill_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>, </span>
<span>                                name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">g_basic</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats_raw_metrics</span>, </span>
<span>                  <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">room</span>, y<span class="op">=</span><span class="va">value</span>, fill<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Collection Date"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">expand_limits</span><span class="op">(</span>y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">metric</span><span class="op">~</span><span class="va">sample_type</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space<span class="op">=</span><span class="st">"free_x"</span>, switch<span class="op">=</span><span class="st">"y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_rotate</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">g_basic</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-basic-stats-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set up plotting templates</span></span>
<span><span class="va">scale_color_na</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_color_brewer</span>, palette<span class="op">=</span><span class="st">"Set1"</span>,</span>
<span>                                 name<span class="op">=</span><span class="st">"Nucleic acid type"</span><span class="op">)</span></span>
<span><span class="va">g_qual_raw</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">na_type</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">sample_type</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-raw-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">sample_type</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-raw-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq_raw</span> <span class="op">&lt;-</span> <span class="va">g_qual_raw</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats_raw</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">sample_type</span><span class="op">~</span><span class="va">.</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-raw-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="preprocessing" class="level1"><h1>Preprocessing</h1>
<p>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Single-room samples lost an average of 52%/14% (RNA/DNA) of total read pairs during cleaning and deduplication, with a further loss of 4.3%/0.1% during ribodepletion. However, the fraction of reads lost, especially during cleaning and deduplication, was highly variable between samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">n_reads_rel</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">stage</span>, </span>
<span>         <span class="va">percent_duplicates</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span>, <span class="va">na_type</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_retained <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_read_pairs</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">n_read_pairs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         p_reads_lost <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">p_reads_retained</span>,</span>
<span>         p_reads_retained_abs <span class="op">=</span> <span class="va">n_read_pairs</span> <span class="op">/</span> <span class="va">n_read_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         p_reads_lost_abs <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_retained_abs</span>,</span>
<span>         p_reads_lost_abs_marginal <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">p_reads_lost_abs</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lag.html">lag</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span> <span class="op">&lt;-</span> <span class="va">n_reads_rel</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>Stage<span class="op">=</span><span class="va">stage</span>, `NA Type`<span class="op">=</span><span class="va">na_type</span>, `Sample Type`<span class="op">=</span><span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">`Sample Type`</span>, <span class="va">`NA Type`</span>, <span class="va">Stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>`% Total Reads Lost (Cumulative)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>,</span>
<span>            `% Total Reads Lost (Marginal)` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">" (mean "</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span>,<span class="fl">1</span><span class="op">)</span>, <span class="st">")"</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Stage</span> <span class="op">!=</span> <span class="st">"raw_concat"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>Stage <span class="op">=</span> <span class="va">Stage</span> <span class="op">%&gt;%</span> <span class="va">as.numeric</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n_reads_rel_display</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["Sample Type"],"name":[1],"type":["fct"],"align":["left"]},{"label":["NA Type"],"name":[2],"type":["fct"],"align":["left"]},{"label":["Stage"],"name":[3],"type":["fct"],"align":["left"]},{"label":["% Total Reads Lost (Cumulative)"],"name":[4],"type":["chr"],"align":["left"]},{"label":["% Total Reads Lost (Marginal)"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"Single-Room","2":"RNA","3":"Trimming & filtering","4":"5.2-52.6 (mean 27)","5":"5.2-52.6 (mean 27)"},{"1":"Single-Room","2":"RNA","3":"Deduplication","4":"6.9-86.5 (mean 52.1)","5":"1.7-49.3 (mean 25.1)"},{"1":"Single-Room","2":"RNA","3":"Initial ribodepletion","4":"7.1-86.8 (mean 56.1)","5":"0.2-24.9 (mean 4)"},{"1":"Single-Room","2":"RNA","3":"Secondary ribodepletion","4":"7.2-86.9 (mean 56.4)","5":"0-1 (mean 0.3)"},{"1":"Single-Room","2":"DNA","3":"Trimming & filtering","4":"6-11.5 (mean 7.8)","5":"6-11.5 (mean 7.8)"},{"1":"Single-Room","2":"DNA","3":"Deduplication","4":"10.5-19.5 (mean 14.1)","5":"3.3-11.6 (mean 6.3)"},{"1":"Single-Room","2":"DNA","3":"Initial ribodepletion","4":"10.6-19.5 (mean 14.1)","5":"0-0.1 (mean 0.1)"},{"1":"Single-Room","2":"DNA","3":"Secondary ribodepletion","4":"10.6-19.6 (mean 14.2)","5":"0-0.1 (mean 0)"},{"1":"Pool","2":"RNA","3":"Trimming & filtering","4":"31.6-31.6 (mean 31.6)","5":"31.6-31.6 (mean 31.6)"},{"1":"Pool","2":"RNA","3":"Deduplication","4":"62.5-62.5 (mean 62.5)","5":"30.9-30.9 (mean 30.9)"},{"1":"Pool","2":"RNA","3":"Initial ribodepletion","4":"63.3-63.3 (mean 63.3)","5":"0.7-0.7 (mean 0.7)"},{"1":"Pool","2":"RNA","3":"Secondary ribodepletion","4":"63.3-63.3 (mean 63.3)","5":"0.1-0.1 (mean 0.1)"},{"1":"Pool","2":"DNA","3":"Trimming & filtering","4":"7.4-7.4 (mean 7.4)","5":"7.4-7.4 (mean 7.4)"},{"1":"Pool","2":"DNA","3":"Deduplication","4":"11.3-11.3 (mean 11.3)","5":"3.9-3.9 (mean 3.9)"},{"1":"Pool","2":"DNA","3":"Initial ribodepletion","4":"11.3-11.3 (mean 11.3)","5":"0.1-0.1 (mean 0.1)"},{"1":"Pool","2":"DNA","3":"Secondary ribodepletion","4":"11.4-11.4 (mean 11.4)","5":"0-0 (mean 0)"},{"1":"NC","2":"RNA","3":"Trimming & filtering","4":"36.4-36.4 (mean 36.4)","5":"36.4-36.4 (mean 36.4)"},{"1":"NC","2":"RNA","3":"Deduplication","4":"87.6-87.6 (mean 87.6)","5":"51.2-51.2 (mean 51.2)"},{"1":"NC","2":"RNA","3":"Initial ribodepletion","4":"87.9-87.9 (mean 87.9)","5":"0.3-0.3 (mean 0.3)"},{"1":"NC","2":"RNA","3":"Secondary ribodepletion","4":"88-88 (mean 88)","5":"0.1-0.1 (mean 0.1)"},{"1":"NC","2":"DNA","3":"Trimming & filtering","4":"6.1-6.1 (mean 6.1)","5":"6.1-6.1 (mean 6.1)"},{"1":"NC","2":"DNA","3":"Deduplication","4":"8.1-8.1 (mean 8.1)","5":"2-2 (mean 2)"},{"1":"NC","2":"DNA","3":"Initial ribodepletion","4":"8.2-8.2 (mean 8.2)","5":"0.1-0.1 (mean 0.1)"},{"1":"NC","2":"DNA","3":"Secondary ribodepletion","4":"8.3-8.3 (mean 8.3)","5":"0-0 (mean 0)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_stage_trace</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">basic_stats</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">na_type</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">sample_type</span><span class="op">~</span><span class="va">na_type</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">2</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span></span>
<span><span class="co"># Plot reads over preprocessing</span></span>
<span><span class="va">g_reads_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">n_read_pairs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"# Read pairs"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_reads_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/preproc-figures-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot relative read losses during preprocessing</span></span>
<span><span class="va">g_reads_rel</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_reads_rel</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">stage</span>, color<span class="op">=</span><span class="va">na_type</span>, group<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_lost_abs_marginal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Total Reads Lost"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                     labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="va">sample_type</span><span class="op">~</span><span class="va">na_type</span>, scales<span class="op">=</span><span class="st">"free"</span>, ncol<span class="op">=</span><span class="fl">2</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_reads_rel</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/preproc-figures-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Data cleaning was very successful at improving read qualities, as well as removing most adapter sequences. The exception to the latter was terminal poly-A sequences, which remained at a substantial (though reduced) prevalence throughout the pipeline. (This doesn’t seem to have caused downstream problems that I can see.)</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_qual</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">na_type</span>, linetype<span class="op">=</span><span class="va">read_pair</span>, </span>
<span>                         group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/interaction.html">interaction</a></span><span class="op">(</span><span class="va">sample</span>,<span class="va">read_pair</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_linetype_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Read Pair"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>color<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>         linetype <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>nrow<span class="op">=</span><span class="fl">2</span>,byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span></span>
<span><span class="co"># Visualize adapters</span></span>
<span><span class="va">g_adapters</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">pc_adapters</span><span class="op">)</span>, data<span class="op">=</span><span class="va">adapter_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Adapters"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">20</span><span class="op">)</span>,</span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">50</span>,<span class="fl">10</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">adapter</span><span class="op">)</span></span>
<span><span class="va">g_adapters</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-quality-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize quality</span></span>
<span><span class="va">g_quality_base</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">position</span>, y<span class="op">=</span><span class="va">mean_phred_score</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_base_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>,<span class="fl">45</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Position"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>,</span>
<span>                     breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">140</span>,<span class="fl">20</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_base</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-quality-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_quality_seq</span> <span class="op">&lt;-</span> <span class="va">g_qual</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">25</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">30</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">mean_phred_score</span>, y<span class="op">=</span><span class="va">n_sequences</span><span class="op">)</span>, data<span class="op">=</span><span class="va">quality_seq_stats</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Mean Phred score"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Sequences"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">stage</span><span class="op">~</span><span class="va">.</span><span class="op">)</span></span>
<span><span class="va">g_quality_seq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-quality-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>According to FASTQC, deduplication was very effective at reducing measured duplicate levels, which for single-room samples fell from an average of 40% to 7.8% for RNA reads and from 5.5% to 0.3% for DNA reads.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">stage_dup</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">stage</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>dmin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, dmax<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>,</span>
<span>            dmean<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">percent_duplicates</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">g_dup_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_duplicates</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"% Duplicates"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_dup_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/preproc-dedup-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_readlen_stages</span> <span class="op">&lt;-</span> <span class="va">g_stage_trace</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_seq_len</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span><span class="st">"Mean read length (nt)"</span>, expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_readlen_stages</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/preproc-dedup-2.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section><section id="high-level-composition" class="level1"><h1>High-level composition</h1>
<p>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import Bracken data</span></span>
<span><span class="va">bracken_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"bracken_counts.tsv"</span><span class="op">)</span></span>
<span><span class="va">bracken</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">bracken_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">total_assigned</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"Total"</span>,</span>
<span>  kraken_assigned_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">kraken_assigned_reads</span><span class="op">)</span>,</span>
<span>  added_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">added_reads</span><span class="op">)</span>,</span>
<span>  new_est_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">new_est_reads</span><span class="op">)</span>,</span>
<span>  fraction_total_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">fraction_total_reads</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bracken_spread</span> <span class="op">&lt;-</span> <span class="va">bracken</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="st">"sample"</span>, names_from <span class="op">=</span> <span class="st">"name"</span>, </span>
<span>              values_from <span class="op">=</span> <span class="st">"new_est_reads"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count reads</span></span>
<span><span class="va">read_counts_preproc</span> <span class="op">&lt;-</span> <span class="va">basic_stats</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">stage</span>, <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"room"</span>, <span class="st">"na_type"</span>, <span class="st">"sample_type"</span><span class="op">)</span>,</span>
<span>              names_from<span class="op">=</span><span class="st">"stage"</span>, values_from<span class="op">=</span><span class="st">"n_read_pairs"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_preproc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">total_assigned</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">new_est_reads</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>assigned <span class="op">=</span> <span class="va">new_est_reads</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">bracken_spread</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assess composition</span></span>
<span><span class="va">read_comp</span> <span class="op">&lt;-</span> <span class="fu">transmute</span><span class="op">(</span><span class="va">read_counts</span>, sample<span class="op">=</span><span class="va">sample</span>, na_type<span class="op">=</span><span class="va">na_type</span>,</span>
<span>                       room<span class="op">=</span><span class="va">room</span>, sample_type<span class="op">=</span><span class="va">sample_type</span>,</span>
<span>                       n_filtered <span class="op">=</span> <span class="va">raw_concat</span><span class="op">-</span><span class="va">cleaned</span>,</span>
<span>                       n_duplicate <span class="op">=</span> <span class="va">cleaned</span><span class="op">-</span><span class="va">dedup</span>,</span>
<span>                       n_ribosomal <span class="op">=</span> <span class="op">(</span><span class="va">dedup</span><span class="op">-</span><span class="va">ribo_initial</span><span class="op">)</span> <span class="op">+</span> <span class="op">(</span><span class="va">ribo_initial</span><span class="op">-</span><span class="va">ribo_secondary</span><span class="op">)</span>,</span>
<span>                       n_unassigned <span class="op">=</span> <span class="va">ribo_secondary</span><span class="op">-</span><span class="va">assigned</span>,</span>
<span>                       n_bacterial <span class="op">=</span> <span class="va">bacteria</span>,</span>
<span>                       n_archaeal <span class="op">=</span> <span class="va">archaea</span>,</span>
<span>                       n_viral <span class="op">=</span> <span class="va">viruses</span>,</span>
<span>                       n_human <span class="op">=</span> <span class="va">eukaryota</span><span class="op">)</span></span>
<span><span class="va">read_comp_long</span> <span class="op">&lt;-</span> <span class="fu">pivot_longer</span><span class="op">(</span><span class="va">read_comp</span>, <span class="op">-</span><span class="op">(</span><span class="va">sample</span><span class="op">:</span><span class="va">sample_type</span><span class="op">)</span>, </span>
<span>                               names_to <span class="op">=</span> <span class="st">"classification"</span>,</span>
<span>                               names_prefix <span class="op">=</span> <span class="st">"n_"</span>, values_to <span class="op">=</span> <span class="st">"n_reads"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="fu">str_to_sentence</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize composition</span></span>
<span><span class="va">read_comp_summ</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">classification</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop_last"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads</span>,<span class="fl">0</span><span class="op">)</span>,</span>
<span>    p_reads <span class="op">=</span> <span class="va">n_reads</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads</span><span class="op">)</span>,</span>
<span>    pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare plotting templates</span></span>
<span><span class="va">g_comp_base</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>mapping<span class="op">=</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">room</span>, y<span class="op">=</span><span class="va">p_reads</span>, fill<span class="op">=</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Collection Date"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">sample_type</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_rotate</span></span>
<span><span class="va">scale_y_pc_reads</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">scale_y_continuous</span>, name <span class="op">=</span> <span class="st">"% Reads"</span>,</span>
<span>                                   expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot overall composition</span></span>
<span><span class="va">g_comp</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data <span class="op">=</span> <span class="va">read_comp_long</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-composition-all-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot composition of minor components</span></span>
<span><span class="va">read_comp_minor</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">palette_minor</span> <span class="op">&lt;-</span> <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">7</span>,<span class="fl">9</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">g_comp_minor</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">read_comp_minor</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_pc_reads</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_minor</span>, name <span class="op">=</span> <span class="st">"Classification"</span><span class="op">)</span></span>
<span><span class="va">g_comp_minor</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-composition-all-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_reads_summ_group</span> <span class="op">&lt;-</span> <span class="va">read_comp_long</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">classification</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span><span class="op">)</span>, <span class="st">"Excluded"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">sample</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">classification</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>pc_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, pc_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, </span>
<span>            pc_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">p_reads</span><span class="op">)</span><span class="op">*</span><span class="fl">100</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ_prep</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_group</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>classification <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">classification</span><span class="op">)</span>,</span>
<span>         pc_min <span class="op">=</span> <span class="va">pc_min</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_max <span class="op">=</span> <span class="va">pc_max</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         pc_mean <span class="op">=</span> <span class="va">pc_mean</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span>digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">format</span>, scientific<span class="op">=</span><span class="cn">FALSE</span>, trim<span class="op">=</span><span class="cn">TRUE</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>         display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">pc_min</span>, <span class="st">"-"</span>, <span class="va">pc_max</span>, <span class="st">"% (mean "</span>, <span class="va">pc_mean</span>, <span class="st">"%)"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span> <span class="op">&lt;-</span> <span class="va">p_reads_summ_prep</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample_type</span>, <span class="va">classification</span>, <span class="va">na_type</span>, <span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">na_type</span>, values_from <span class="op">=</span> <span class="va">display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">sample_type</span>, <span class="va">classification</span><span class="op">)</span></span>
<span><span class="va">p_reads_summ</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample_type"],"name":[1],"type":["fct"],"align":["left"]},{"label":["classification"],"name":[2],"type":["fct"],"align":["left"]},{"label":["RNA"],"name":[3],"type":["chr"],"align":["left"]},{"label":["DNA"],"name":[4],"type":["chr"],"align":["left"]}],"data":[{"1":"Single-Room","2":"Excluded","3":"16-91% (mean 70%)","4":"76-96% (mean 92%)"},{"1":"Single-Room","2":"Ribosomal","3":"0.21-26% (mean 4.3%)","4":"0.082-0.19% (mean 0.1%)"},{"1":"Single-Room","2":"Bacterial","3":"0.38-7.8% (mean 4.5%)","4":"2.9-5.6% (mean 3.7%)"},{"1":"Single-Room","2":"Archaeal","3":"0.00027-0.015% (mean 0.0027%)","4":"0.0062-0.011% (mean 0.0087%)"},{"1":"Single-Room","2":"Viral","3":"0.0016-0.38% (mean 0.084%)","4":"0.009-0.14% (mean 0.037%)"},{"1":"Single-Room","2":"Human","3":"5.7-84% (mean 21%)","4":"0.68-18% (mean 4%)"},{"1":"Pool","2":"Excluded","3":"81-81% (mean 81%)","4":"90-90% (mean 90%)"},{"1":"Pool","2":"Ribosomal","3":"0.83-0.83% (mean 0.83%)","4":"0.12-0.12% (mean 0.12%)"},{"1":"Pool","2":"Bacterial","3":"7.6-7.6% (mean 7.6%)","4":"4.8-4.8% (mean 4.8%)"},{"1":"Pool","2":"Archaeal","3":"0.0014-0.0014% (mean 0.0014%)","4":"0.0088-0.0088% (mean 0.0088%)"},{"1":"Pool","2":"Viral","3":"0.33-0.33% (mean 0.33%)","4":"0.068-0.068% (mean 0.068%)"},{"1":"Pool","2":"Human","3":"11-11% (mean 11%)","4":"5.4-5.4% (mean 5.4%)"},{"1":"NC","2":"Excluded","3":"90-90% (mean 90%)","4":"82-82% (mean 82%)"},{"1":"NC","2":"Ribosomal","3":"0.41-0.41% (mean 0.41%)","4":"0.14-0.14% (mean 0.14%)"},{"1":"NC","2":"Bacterial","3":"1.8-1.8% (mean 1.8%)","4":"12-12% (mean 12%)"},{"1":"NC","2":"Archaeal","3":"0.0011-0.0011% (mean 0.0011%)","4":"0.0072-0.0072% (mean 0.0072%)"},{"1":"NC","2":"Viral","3":"0.0063-0.0063% (mean 0.0063%)","4":"0.0084-0.0084% (mean 0.0084%)"},{"1":"NC","2":"Human","3":"7.9-7.9% (mean 7.9%)","4":"6.1-6.1% (mean 6.1%)"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>As with the Prussin data, human reads are very elevated compared to wastewater, making up an average of 21% of read pairs in RNA libraries and 4% for DNA libraries for single-room samples (in both cases, the maximum abundance of human reads across samples is much higher). Viral read abundances are higher than Prussin, averaging 0.08% for single-room RNA libraries and 0.04% for DNA libraries; for some reason, the pooled RNA library shows much higher viral prevalence, at 0.33%. The most obvious difference from the Prussin data is that DNA libraries in Rosario show a much higher fraction of non-ribosomal unassigned reads, with an average of 78% of all reads for single-room libraries.</p>
<p>As in Prussin, viral DNA reads were dominated by <em>Caudoviricetes</em> phages, with many unclassifiable below the class level. Other prominent viral classes in DNA reads included <em>Papovaviricetes</em> (which includes <em>Polyomaviridae</em> and <em>Papillomaviridae</em>), <em>Megaviridae</em> (giant viruses) and <em>Repensiviricetes</em> (plant and fungal viruses). Pooled samples also showed a significant level of <em>Quintoviricetes</em> (parvoviruses).</p>
<p>RNA libraries, meanwhile, were dominated by <em>Alsuviricetes</em> (mainly plant viruses) and, notably, <em>Pisoniviricetes</em> (which includes coronaviruses, picornaviruses including <em>Enterovirus</em>, and caliciviruses including <em>Norovirus</em>). <em>Caudoviricetes</em>, <em>Herviviricetes</em> (herpesviruses) and <em>Revtraviricetes</em> (retroviruses + Hep B) were also prominent in some samples.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxonomy</span></span>
<span><span class="va">viral_taxa_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"viral-taxids.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">viral_taxa</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">viral_taxa_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import Kraken reports &amp; extract viral taxa</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">basic_stats_raw</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span><span class="va">col_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"pc_reads_total"</span>, <span class="st">"n_reads_clade"</span>, <span class="st">"n_reads_direct"</span>,</span>
<span>               <span class="st">"rank"</span>, <span class="st">"taxid"</span>, <span class="st">"name"</span><span class="op">)</span></span>
<span><span class="va">report_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"kraken/"</span>, <span class="va">samples</span>, <span class="st">".report.gz"</span><span class="op">)</span></span>
<span><span class="va">kraken_reports</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> </span>
<span>  <span class="fu">read_tsv</span><span class="op">(</span><span class="va">report_paths</span><span class="op">[</span><span class="va">n</span><span class="op">]</span>, col_names <span class="op">=</span> <span class="va">col_names</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="va">samples</span><span class="op">[</span><span class="va">n</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span><span class="va">kraken_reports_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">kraken_reports</span>, <span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">kraken_reports_viral_cleaned</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">pc_reads_total</span>, <span class="op">-</span><span class="va">n_reads_direct</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">p_reads_viral</span>, <span class="va">n_reads_clade</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">viral_classes</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span></span>
<span><span class="va">viral_families</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"F"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Identify major viral classes</span></span>
<span><span class="va">viral_classes_major_tab</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_viral_max</span> <span class="op">&gt;=</span> <span class="fl">0.04</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major_list</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">viral_classes_major</span> <span class="op">&lt;-</span> <span class="va">viral_classes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_classes_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_minor</span> <span class="op">&lt;-</span> <span class="va">viral_classes_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_viral_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_viral <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_viral_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_viral</span><span class="op">)</span></span>
<span><span class="va">viral_classes_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">viral_classes_major</span>, <span class="va">viral_classes_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_viral</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">viral_classes_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">p_reads_viral</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_viral</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">palette_viral</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">12</span>, <span class="st">"Set3"</span><span class="op">)</span>, <span class="fu">brewer.pal</span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Dark2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_classes</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">viral_classes_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Viral Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral class"</span><span class="op">)</span></span>
<span><span class="va">g_classes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/viral-class-composition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Of these, the strong presence of <em>Papovaviricetes</em> and <em>Pisoniviricetes</em> across multiple samples is the most interesting, as both contain important human pathogens. The presence of <em>Quintoviricetes</em>, <em>Herviviricetes</em> and <em>Revtraviricetes</em> in a smaller number of samples is also notable.</p>
<p>In DNA reads, <em>Papovaviricetes</em> are quite heterogeneous across samples, while mostly being dominated by one or two genera within samples. The three rooms with the largest <em>Papovaviricetes</em> fraction (Female room 1 &amp; 3 and Male room 4) are each dominated by a single genus, specifically <em>Dyothetapapillomavirus</em> (of which the sole member is <em>Dyothetapapillomavirus 1</em>, a cat pathogen), <em>Betapapillomavirus</em> (a genus of human wart-causing viruses), and <em>Alphapolyomavirus</em> (specifically Merkel cell polyomavirus).</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get all read counts in class</span></span>
<span><span class="va">papova_taxid</span> <span class="op">&lt;-</span> <span class="fl">2732421</span></span>
<span><span class="va">papova_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">papova_taxid</span></span>
<span><span class="va">papova_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">papova_desc_taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">papova_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">papova_desc_taxids_new</span></span>
<span>  <span class="va">papova_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">papova_counts</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_desc_taxids_new</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_papova <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genus composition</span></span>
<span><span class="va">papova_genera</span> <span class="op">&lt;-</span> <span class="va">papova_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"G"</span>, <span class="va">na_type</span> <span class="op">==</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major_tab</span> <span class="op">&lt;-</span> <span class="va">papova_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_papova_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_papova_max</span> <span class="op">&gt;=</span> <span class="fl">0.04</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major_list</span> <span class="op">&lt;-</span> <span class="va">papova_genera_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">papova_genera_major</span> <span class="op">&lt;-</span> <span class="va">papova_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">papova_genera_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_papova</span><span class="op">)</span></span>
<span><span class="va">papova_genera_minor</span> <span class="op">&lt;-</span> <span class="va">papova_genera_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_papova_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_papova <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_papova_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_papova</span><span class="op">)</span></span>
<span><span class="va">papova_genera_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">papova_genera_major</span>, <span class="va">papova_genera_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_papova</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">papova_genera_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_papova</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_papova_genera</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">papova_genera_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Papovaviricetes Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_papova_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/papovaviricetes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In RNA reads, <em>Pisoniviricetes</em> are conversely dominated by a single genus, <em>Sobemovirus</em>, across all samples, and specifically a single species. Disappointingly, this is Ryegrass mottle virus, a plant pathogen.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get all read counts in class</span></span>
<span><span class="va">pisoni_taxid</span> <span class="op">&lt;-</span> <span class="fl">2732506</span></span>
<span><span class="va">pisoni_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">pisoni_taxid</span></span>
<span><span class="va">pisoni_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pisoni_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pisoni_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">pisoni_desc_taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">pisoni_desc_taxids_old</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">pisoni_desc_taxids_old</span> <span class="op">&lt;-</span> <span class="va">pisoni_desc_taxids_new</span></span>
<span>  <span class="va">pisoni_desc_taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pisoni_desc_taxids_old</span>, <span class="va">viral_taxa</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pisoni_desc_taxids_old</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">pisoni_counts</span> <span class="op">&lt;-</span> <span class="va">kraken_reports_viral_cleaned</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pisoni_desc_taxids_new</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_pisoni <span class="op">=</span> <span class="va">n_reads_clade</span><span class="op">/</span><span class="va">n_reads_clade</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genus composition</span></span>
<span><span class="va">pisoni_genera</span> <span class="op">&lt;-</span> <span class="va">pisoni_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="st">"S"</span>, <span class="va">na_type</span> <span class="op">==</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">pisoni_genera_major_tab</span> <span class="op">&lt;-</span> <span class="va">pisoni_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_pisoni_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_pisoni</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_pisoni_max</span> <span class="op">&gt;=</span> <span class="fl">0.04</span><span class="op">)</span></span>
<span><span class="va">pisoni_genera_major_list</span> <span class="op">&lt;-</span> <span class="va">pisoni_genera_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="va">pisoni_genera_major</span> <span class="op">&lt;-</span> <span class="va">pisoni_genera</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">pisoni_genera_major_list</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_pisoni</span><span class="op">)</span></span>
<span><span class="va">pisoni_genera_minor</span> <span class="op">&lt;-</span> <span class="va">pisoni_genera_major</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>p_reads_pisoni_major <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_pisoni</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Other"</span>, taxid<span class="op">=</span><span class="cn">NA</span>, p_reads_pisoni <span class="op">=</span> <span class="fl">1</span><span class="op">-</span><span class="va">p_reads_pisoni_major</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">name</span>, <span class="va">taxid</span>, <span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">p_reads_pisoni</span><span class="op">)</span></span>
<span><span class="va">pisoni_genera_display</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">pisoni_genera_major</span>, <span class="va">pisoni_genera_minor</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_pisoni</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">pisoni_genera_major_list</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_pisoni</span>, classification<span class="op">=</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_pisoni_genera</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">pisoni_genera_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Pisoniviricetes Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_pisoni_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/pisoniviricetes-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-virus-reads-validation" class="level1"><h1>Human-infecting virus reads: validation</h1>
<p>Next, I investigated the human-infecting virus read content of these unenriched samples. Using the same workflow I used for Prussin et al, I identified 118 RNA read pairs and 1269 DNA read pairs as putatively human viral: 0.004% and 0.017% of surviving reads, respectively.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">hv_reads_filtered_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"ribo_initial"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>               <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_summ</span> <span class="op">&lt;-</span> <span class="va">n_hv_filtered</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_putative</span><span class="op">)</span>, n_total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_total</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>,</span>
<span>         adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span><span class="va">g_mrg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-hv-scores-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_hist_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">kraken_label</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-hv-scores-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I’ve used for previous datasets.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_fasta</span> <span class="op">&lt;-</span>  <span class="va">mrg</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">seq_id</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, </span>
<span>         header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mrg_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"blast/putative-viral.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import BLAST results</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"blast/putative-viral.blast.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                          col_names <span class="op">=</span> <span class="va">blast_cols</span>, col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># blast_results_path &lt;- file.path(data_dir, "blast/putative-viral-best.blast.gz")</span></span>
<span><span class="co"># blast_results &lt;- read_tsv(blast_results_path, show_col_types = FALSE)</span></span>
<span></span>
<span><span class="co"># Filter for best hit for each query/subject combination</span></span>
<span><span class="va">blast_results_best</span> <span class="op">&lt;-</span> <span class="va">blast_results</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each query and filter for high-ranking hits</span></span>
<span><span class="va">blast_results_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_best</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>read_pair <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, </span>
<span>         seq_id <span class="op">=</span> <span class="fu">str_split</span><span class="op">(</span><span class="va">qseqid</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">nth</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>,</span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_results_paired</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_id</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span>, <span class="va">adj_score_max</span>, <span class="va">na_type</span><span class="op">)</span></span>
<span><span class="va">blast_results_assign</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_viral</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">seq_id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span><span class="va">mrg_blast</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg</span>, <span class="va">blast_results_out</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_blast_rna</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">na_type</span> <span class="op">==</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">mrg_blast_dna</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">na_type</span> <span class="op">==</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot RNA</span></span>
<span><span class="va">g_mrg_blast_rna</span> <span class="op">&lt;-</span> <span class="va">mrg_blast_rna</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">viral_status_out</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span>, plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_mrg_blast_rna</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-blast-results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Plot DNA</span></span>
<span><span class="va">g_mrg_blast_dna</span> <span class="op">&lt;-</span> <span class="va">mrg_blast_dna</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">65</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">viral_status_out</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="st">"DNA"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span>, plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_mrg_blast_dna</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-blast-results-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">g_hist_1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_blast</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span>,position<span class="op">=</span><span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">viral_status_out</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_fill_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">geom_vline</span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">20</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">g_hist_1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-blast-histogram-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>These results look good on visual inspection, and indeed precision and sensitivity are both very high. For a disjunctive score threshold of 20, my updated workflow achieves an F1 score of 98.2% for RNA sequences and 97.9% for DNA sequences.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">test_sens_spec</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">tab</span>, <span class="va">score_threshold</span>, <span class="va">conjunctive</span>, <span class="va">include_special</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">include_special</span><span class="op">)</span> <span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">tab</span>, <span class="va">viral_status_out</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TRUE"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">tab_retained</span> <span class="op">&lt;-</span> <span class="va">tab</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span></span>
<span>    conjunctive <span class="op">=</span> <span class="va">conjunctive</span>,</span>
<span>    retain_score_conjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">&amp;</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>, </span>
<span>    retain_score_disjunctive <span class="op">=</span> <span class="op">(</span><span class="va">adj_score_fwd</span> <span class="op">&gt;</span> <span class="va">score_threshold</span> <span class="op">|</span> <span class="va">adj_score_rev</span> <span class="op">&gt;</span> <span class="va">score_threshold</span><span class="op">)</span>,</span>
<span>    retain_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="va">retain_score_conjunctive</span>, <span class="va">retain_score_disjunctive</span><span class="op">)</span>,</span>
<span>    retain <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">retain_score</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span></span>
<span>  <span class="va">pos_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">pos_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_tru</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">!=</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">neg_fls</span> <span class="op">&lt;-</span> <span class="va">tab_retained</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">viral_status_out</span> <span class="op">==</span> <span class="st">"TRUE"</span>, <span class="op">!</span><span class="va">retain</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sum</span></span>
<span>  <span class="va">sensitivity</span> <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">neg_fls</span><span class="op">)</span></span>
<span>  <span class="va">specificity</span> <span class="op">&lt;-</span> <span class="va">neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">neg_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">precision</span>   <span class="op">&lt;-</span> <span class="va">pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">pos_tru</span> <span class="op">+</span> <span class="va">pos_fls</span><span class="op">)</span></span>
<span>  <span class="va">f1</span> <span class="op">&lt;-</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="op">(</span>threshold<span class="op">=</span><span class="va">score_threshold</span>, include_special <span class="op">=</span> <span class="va">include_special</span>, </span>
<span>                conjunctive <span class="op">=</span> <span class="va">conjunctive</span>, sensitivity<span class="op">=</span><span class="va">sensitivity</span>, </span>
<span>                specificity<span class="op">=</span><span class="va">specificity</span>, precision<span class="op">=</span><span class="va">precision</span>, f1<span class="op">=</span><span class="va">f1</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">range_f1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">intab</span>, <span class="va">inc_special</span>, <span class="va">inrange</span><span class="op">=</span><span class="fl">15</span><span class="op">:</span><span class="fl">45</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">tss</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span><span class="op">(</span><span class="va">test_sens_spec</span>, tab<span class="op">=</span><span class="va">intab</span>, include_special<span class="op">=</span><span class="va">inc_special</span><span class="op">)</span></span>
<span>  <span class="va">stats_conj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_disj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">inrange</span>, <span class="va">tss</span>, conjunctive<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">bind_rows</span></span>
<span>  <span class="va">stats_all</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_conj</span>, <span class="va">stats_disj</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">threshold</span><span class="op">:</span><span class="va">conjunctive</span><span class="op">)</span>, names_to<span class="op">=</span><span class="st">"metric"</span>, values_to<span class="op">=</span><span class="st">"value"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>conj_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">conjunctive</span>, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">stats_all</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">inc_special</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">stats_rna</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast_rna</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>na_type <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span></span>
<span><span class="va">stats_dna</span> <span class="op">&lt;-</span> <span class="fu">range_f1</span><span class="op">(</span><span class="va">mrg_blast_dna</span>, <span class="va">inc_special</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>na_type <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span>
<span><span class="va">stats_0</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">stats_rna</span>, <span class="va">stats_dna</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>attempt<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">threshold_opt_0</span> <span class="op">&lt;-</span> <span class="va">stats_0</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">conj_label</span>,<span class="va">attempt</span>,<span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">metric</span> <span class="op">==</span> <span class="st">"f1"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">threshold</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_stats_0</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">stats_0</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">threshold</span>, y<span class="op">=</span><span class="va">value</span>, color<span class="op">=</span><span class="va">metric</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>data <span class="op">=</span> <span class="va">threshold_opt_0</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>xintercept<span class="op">=</span><span class="va">threshold</span><span class="op">)</span>,</span>
<span>             color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Value"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Threshold"</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette<span class="op">=</span><span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">conj_label</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_stats_0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-f1-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Looking into the composition of different read groups, the most notable observations for me are the predominance of (1) HIV reads among low-scoring RNA “true-positives”, and (2) human mastadenovirus C among RNA true-positives in general:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">211787</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">509154</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">493803</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">427343</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span></span>
<span><span class="va">major_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">viral_status_out</span>, </span>
<span>           highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">viral_status_out</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, <span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_tab</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_major_list</span> <span class="op">&lt;-</span> <span class="va">fp_major_tab</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span> <span class="op">%&gt;%</span> <span class="va">unique</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="va">fp_major</span> <span class="op">&lt;-</span> <span class="va">fp</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>major <span class="op">=</span> <span class="va">p</span> <span class="op">&gt;</span> <span class="va">major_threshold</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">major</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span>, p<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, levels <span class="op">=</span> <span class="va">fp_major_list</span><span class="op">)</span>,</span>
<span>         score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_fp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">score_display</span>, y<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">name_display</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"True positive?"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"% reads"</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral\ntaxon"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">status_display</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_kit</span></span>
<span><span class="va">g_fp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/fp-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There are only eight low-scoring true-positive RNA reads, three of which are HIV. These do seem to be real; at least, they don’t BLAST to anything else. Ironically the eight high-scoring “true-positive” HIV reads seem more likely to be fake; they also map to a range of cloning vectors:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxid_hiv</span> <span class="op">&lt;-</span> <span class="fl">11676</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span></span>
<span><span class="co"># Get taxon names</span></span>
<span><span class="va">tax_names_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"taxid-names.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">tax_names_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span><span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span>, <span class="va">tax_names_new</span><span class="op">)</span></span>
<span><span class="va">ref_name_hiv</span> <span class="op">&lt;-</span> <span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">ref_taxid_hiv</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get major matches</span></span>
<span><span class="va">fp_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">ref_taxid_hiv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count</span> <span class="op">&lt;-</span> <span class="va">fp_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">na_type</span>, </span>
<span>           <span class="va">taxid</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">na_type</span>, <span class="va">taxid</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count_major</span> <span class="op">&lt;-</span> <span class="va">fp_staxid_count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span><span class="op">&gt;</span><span class="fl">1</span>, <span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, </span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_staxid_count_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">status_display</span><span class="op">+</span><span class="va">score_display</span>, scales<span class="op">=</span><span class="st">"free"</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_hiv</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_hiv</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/hiv-blast-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>A similar pattern is seen for the Human mastadenovirus C hits: low-scoring true-positives only map to adenoviruses, but high-scoring ones also map to a range of cloning vectors.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Configure</span></span>
<span><span class="va">ref_taxid_masta</span> <span class="op">&lt;-</span> <span class="fl">129951</span></span>
<span><span class="va">p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="co"># Add missing names</span></span>
<span><span class="va">tax_names_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">staxid</span>, <span class="op">~</span><span class="va">name</span>,</span>
<span>                         <span class="fl">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span>                         <span class="fl">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span>                         <span class="fl">3033760</span>, <span class="st">"Human adenovirus 89"</span>,</span>
<span>                         <span class="fl">3043599</span>, <span class="st">"Human adenovirus 108"</span>,</span>
<span>                         <span class="fl">3088344</span>, <span class="st">"Human adenovirus C108"</span>,</span>
<span>                         <span class="fl">3102992</span>, <span class="st">"Cloning vector pBWH-C5"</span>,</span>
<span>                         <span class="fl">3102994</span>, <span class="st">"Cloning vector pBWH-C5-pIX-Kan"</span>,</span>
<span>                         <span class="fl">3102995</span>, <span class="st">"Cloning vector pBWH-C5-pIXZG"</span></span>
<span>                         <span class="op">)</span></span>
<span><span class="va">tax_names</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">tax_names</span>, <span class="va">tax_names_new</span><span class="op">)</span></span>
<span><span class="va">ref_name_masta</span> <span class="op">&lt;-</span> <span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">ref_taxid_masta</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get major matches</span></span>
<span><span class="va">fp_staxid</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">ref_taxid_masta</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>n_seq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count</span> <span class="op">&lt;-</span> <span class="va">fp_staxid</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">na_type</span>, </span>
<span>           <span class="va">taxid</span>, <span class="va">name</span>, <span class="va">staxid</span>, <span class="va">sname</span>, <span class="va">n_seq</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">highscore</span>, <span class="va">na_type</span>, <span class="va">taxid</span>, <span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="va">n_seq</span><span class="op">)</span></span>
<span><span class="va">fp_staxid_count_major</span> <span class="op">&lt;-</span> <span class="va">fp_staxid_count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p</span><span class="op">&gt;</span><span class="va">p_threshold</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>, </span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_staxid_count_major</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">p</span>, y<span class="op">=</span><span class="va">sname</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">status_display</span><span class="op">+</span><span class="va">score_display</span>, scales<span class="op">=</span><span class="st">"free"</span>,</span>
<span>             labeller <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span>multi_line <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% mapped reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_masta</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_masta</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/masta-blast-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The plots above only tell us the total number of reads that BLAST to each cloning vector; they don’t tell us how many reads map to any vector. In particular, they can’t distinguish between a situation where many reads map to a few cloning vectors, and one where a few reads map to many. This turns out to be about 35% of high-scoring HIV RNA reads and 55% of high-scoring mastadenovirus RNA reads:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fp_hiv_vec</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">ref_taxid_hiv</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span>vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"vector"</span>, <span class="va">sname</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_hiv_vec_match</span> <span class="op">&lt;-</span> <span class="va">fp_hiv_vec</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">seq_id</span>, <span class="va">na_type</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>vector_match <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">vector</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">fp_hiv_vec_count</span> <span class="op">&lt;-</span> <span class="va">fp_hiv_vec_match</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">na_type</span>, <span class="va">highscore</span>, <span class="va">vector_match</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">na_type</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>,</span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">g_hiv_vec_count</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_hiv_vec_count</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">score_display</span>, x<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">vector_match</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Mapped Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Maps to vector?"</span>, palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">status_display</span>, scales<span class="op">=</span><span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_hiv</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_hiv</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hiv_vec_count</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/hiv-vector-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fp_masta_vec</span> <span class="op">&lt;-</span> <span class="va">mrg_blast</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">taxid</span> <span class="op">==</span> <span class="va">ref_taxid_masta</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, by<span class="op">=</span><span class="st">"seq_id"</span><span class="op">)</span>  <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>staxid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">staxid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span>, by<span class="op">=</span><span class="st">"staxid"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>sname<span class="op">=</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">tax_names</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>taxid<span class="op">=</span><span class="va">staxid</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span>vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"vector"</span>, <span class="va">sname</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fp_masta_vec_match</span> <span class="op">&lt;-</span> <span class="va">fp_masta_vec</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">seq_id</span>, <span class="va">na_type</span>, highscore <span class="op">=</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>vector_match <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">vector</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">fp_masta_vec_count</span> <span class="op">&lt;-</span> <span class="va">fp_masta_vec_match</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">na_type</span>, <span class="va">highscore</span>, <span class="va">vector_match</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">viral_status_out</span>, <span class="va">na_type</span>, <span class="va">highscore</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p<span class="op">=</span><span class="va">n</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>score_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">highscore</span>, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span><span class="op">)</span>,</span>
<span>         status_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_out</span>,</span>
<span>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">g_masta_vec_count</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">fp_masta_vec_count</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">score_display</span>, x<span class="op">=</span><span class="va">p</span>, fill<span class="op">=</span><span class="va">vector_match</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% Mapped Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_brewer</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Maps to vector?"</span>, palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">na_type</span><span class="op">~</span><span class="va">status_display</span>, scales<span class="op">=</span><span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ref_name_masta</span>, <span class="st">" (taxid "</span>, <span class="va">ref_taxid_masta</span>, <span class="st">")"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fu">rel</span><span class="op">(</span><span class="fl">1.4</span><span class="op">)</span>, hjust<span class="op">=</span><span class="fl">0</span>, face<span class="op">=</span><span class="st">"plain"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_masta_vec_count</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/masta-vector-hits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This is more concerning for the mastadenovirus reads, which make up a nontrivial fraction of all high-scoring RNA “true-positives”. As in my last post, though, it’s hard to know what to take from this; many vectors are based on lentiviral or adenoviral backbones, and it’s not always easy to distinguish between true virus and virus-based vector. I’m going to stick with my pipeline as it is for now, but it’s worth keeping this in mind in interpreting the results.</p>
</section><section id="human-infecting-viruses-overall-relative-abundance" class="level1"><h1>Human-infecting viruses: overall relative abundance</h1>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get raw read counts</span></span>
<span><span class="va">read_counts_raw</span> <span class="op">&lt;-</span> <span class="va">basic_stats_raw</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, n_reads_raw <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get HV read counts</span></span>
<span><span class="va">mrg_hv</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>hv_status <span class="op">=</span> <span class="va">assigned_hv</span> <span class="op">|</span> <span class="va">hit_hv</span> <span class="op">|</span> <span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">read_counts_hv</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">hv_status</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_hv"</span><span class="op">)</span></span>
<span><span class="va">read_counts</span> <span class="op">&lt;-</span> <span class="va">read_counts_raw</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_hv</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_reads_hv</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate</span></span>
<span><span class="va">read_counts_total</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span> <span class="st">"All samples"</span>, room <span class="op">=</span> <span class="st">"All rooms"</span><span class="op">)</span></span>
<span><span class="va">read_counts_agg</span> <span class="op">&lt;-</span> <span class="va">read_counts</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">bind_rows</span><span class="op">(</span><span class="va">read_counts_total</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>,</span>
<span>         p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In non-control samples, applying a disjunctive cutoff at S=20 identifies 97 RNA reads and 1204 DNA reads as human-viral. This gives an overall relative HV abundance of <span class="math inline">\(1.21 \times 10^{-5}\)</span> for RNA reads and <span class="math inline">\(1.50 \times 10^{-4}\)</span> for DNA reads. For DNA reads, HV RA in the negative controls is about 10x lower than the non-control samples; however, for RNA reads, the HV RA in the negative controls is actually higher than in the non-controls. This, combined with the very low absolute read count, suggests that we shouldn’t take the HV results for RNA reads too seriously.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Visualize</span></span>
<span><span class="va">g_phv_agg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">read_counts_agg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">room</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_discrete</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Collection Date"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span><span class="op">~</span><span class="va">sample_type</span>, scales <span class="op">=</span> <span class="st">"free"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="va">theme_rotate</span></span>
<span><span class="va">g_phv_agg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/plot-hv-ra-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Collate past RA values</span></span>
<span><span class="va">ra_past</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Collate new RA values</span></span>
<span><span class="va">ra_new</span> <span class="op">&lt;-</span> <span class="fu">tribble</span><span class="op">(</span><span class="op">~</span><span class="va">dataset</span>, <span class="op">~</span><span class="va">ra</span>, <span class="op">~</span><span class="va">na_type</span>, <span class="op">~</span><span class="va">panel_enriched</span>,</span>
<span>                  <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span>                  <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">ra_comp</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">ra_past</span>, <span class="va">ra_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>dataset <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">ra_comp</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">dataset</span>, x<span class="op">=</span><span class="va">ra</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance of human virus reads"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_ra_comp</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/ra-hv-past-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="human-infecting-viruses-taxonomy-and-composition" class="level1"><h1>Human-infecting viruses: taxonomy and composition</h1>
<p>I’m going to focus on the DNA reads here, since as discussed above I expect the RNA reads to be noisy and not very informative – which seems to in fact be the case.</p>
<p>At the family level, as in Prussin, we see that <em>Papillomaviridae</em>, and <em>Polyomaviridae</em> dominate DNA reads in most samples, with <em>Poxviridae</em> and <em>Genomoviridae</em> showing significant presence in several samples. The first of these is primarily <em>Betapapillomavirus</em> and <em>Gammapapillomavirus</em>, with some <em>Alphapapillomavirus</em>; the second is primarily <em>Alphapolyomavirus</em> and <em>Deltapolyomavirus</em>. All of these are made up of several different viruses at the species level. <em>Genomoviridae</em> is represented primarily by <em>Gemykibivirus</em>, while <em>Poxviridae</em> is represented by small numbers of reads from several species, including vaccinia, cowpox and tanapox. (There’s also one variola read, which would be alarming, but BLASTN thinks it isn’t real.)</p>
<p>Compared to Prussin, the most striking difference is the absence of herpesvirus reads; whereas HCMV was dominant in the Prussin data, it barely shows up here except in the controls.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">249588</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">194960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">688449</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">585893</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333922</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334207</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">369960</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333924</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">687329</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">325455</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333750</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694002</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334202</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">197911</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">186938</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333926</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337051</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337043</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">694003</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334204</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">334208</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">333928</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">viral_taxa</span><span class="op">$</span><span class="va">taxid</span> <span class="op">==</span> <span class="fl">337039</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span></span>
<span></span>
<span><span class="va">mrg_hv_named</span> <span class="op">&lt;-</span> <span class="va">mrg_hv</span> <span class="op">%&gt;%</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span><span class="va">raise_rank</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">read_db</span>, <span class="va">taxid_db</span>, <span class="va">out_rank</span> <span class="op">=</span> <span class="st">"species"</span>, <span class="va">verbose</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># Get higher ranks than search rank</span></span>
<span>  <span class="va">ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span><span class="op">)</span></span>
<span>  <span class="va">rank_match</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="va">ranks</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="va">high_ranks</span> <span class="op">&lt;-</span> <span class="va">ranks</span><span class="op">[</span><span class="va">rank_match</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ranks</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="co"># Merge read DB and taxid DB</span></span>
<span>  <span class="va">reads</span> <span class="op">&lt;-</span> <span class="va">read_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span>  <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">reads</span>, <span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span></span>
<span>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span>  <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">while</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">reads_norank</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span> <span class="co"># As long as there are unclassified sequences...</span></span>
<span>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span>    <span class="va">reads_remaining</span> <span class="op">&lt;-</span> <span class="va">reads_norank</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">parent_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>    <span class="va">reads_rank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">==</span> <span class="va">out_rank</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_rank</span><span class="op">)</span></span>
<span>    <span class="va">reads_norank</span> <span class="op">&lt;-</span> <span class="va">reads_remaining</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">!=</span> <span class="va">out_rank</span>, <span class="op">!</span><span class="va">rank</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">high_ranks</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">taxid</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span>  <span class="va">reads_dropped</span> <span class="op">&lt;-</span> <span class="va">reads</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">seq_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_rank</span><span class="op">$</span><span class="va">seq_id</span><span class="op">)</span></span>
<span>  <span class="va">reads_out</span> <span class="op">&lt;-</span> <span class="va">reads_rank</span> <span class="op">%&gt;%</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">reads_dropped</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">parent_taxid</span>, <span class="op">-</span><span class="va">rank</span>, <span class="op">-</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">left_join</span><span class="op">(</span><span class="va">taxid_db</span>, by<span class="op">=</span><span class="st">"taxid"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">reads_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">hv_reads_species</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"species"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_genus</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_family</span> <span class="op">&lt;-</span> <span class="fu">raise_rank</span><span class="op">(</span><span class="va">mrg_hv_named</span>, <span class="va">viral_taxa</span>, <span class="st">"family"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_family</span> <span class="op">&lt;-</span> <span class="fl">0.06</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral family</span></span>
<span><span class="va">hv_family_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_family</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_family_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_family</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_family_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_family_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_family_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_family</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_family_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral family"</span><span class="op">)</span></span>
<span><span class="va">g_hv_family</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/hv-family-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_genus</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral genus</span></span>
<span><span class="va">hv_genus_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">na_type</span> <span class="op">==</span> <span class="st">"DNA"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_genus_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_genus_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_genus</span><span class="op">)</span></span>
<span><span class="va">hv_genus_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_genus_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_genus_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_genus_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_genus_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_genus_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_genus</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_genus_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral genus"</span><span class="op">)</span></span>
<span><span class="va">g_hv_genus</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/hv-genus-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold_major_species</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="co"># Count reads for each human-viral species</span></span>
<span><span class="va">hv_species_counts</span> <span class="op">&lt;-</span> <span class="va">hv_reads_species</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">na_type</span> <span class="op">==</span> <span class="st">"DNA"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">count</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"n_reads_hv"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_hv <span class="op">=</span> <span class="va">n_reads_hv</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify high-ranking families and group others</span></span>
<span><span class="va">hv_species_major_tab</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">p_reads_hv</span> <span class="op">&gt;</span> <span class="va">threshold_major_species</span><span class="op">)</span></span>
<span><span class="va">hv_species_counts_major</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">hv_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">sample_type</span>, <span class="va">na_type</span>, <span class="va">name_display</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_hv</span><span class="op">)</span>, p_reads_hv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">p_reads_hv</span><span class="op">)</span>, </span>
<span>            .groups<span class="op">=</span><span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>name_display <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">name_display</span>, </span>
<span>                               levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">hv_species_major_tab</span><span class="op">$</span><span class="va">name</span>, <span class="st">"Other"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_species_counts_display</span> <span class="op">&lt;-</span> <span class="va">hv_species_counts_major</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rename</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">p_reads_hv</span>, classification <span class="op">=</span> <span class="va">name_display</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_hv_species</span> <span class="op">&lt;-</span> <span class="va">g_comp_base</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_col</span><span class="op">(</span>data<span class="op">=</span><span class="va">hv_species_counts_display</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"% HV Reads"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1.01</span><span class="op">)</span>, </span>
<span>                     breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0.2</span><span class="op">)</span>,</span>
<span>                     expand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, labels <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="va">palette_viral</span>, name <span class="op">=</span> <span class="st">"Viral species"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">guides</span><span class="op">(</span>fill<span class="op">=</span><span class="fu">guide_legend</span><span class="op">(</span>ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hv_species</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/hv-species-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define reference genera</span></span>
<span><span class="va">path_genera_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera_dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span><span class="op">)</span></span>
<span><span class="va">path_genera</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_rna</span>, genome_type<span class="op">=</span><span class="st">"RNA genome"</span><span class="op">)</span>,</span>
<span>                         <span class="fu">tibble</span><span class="op">(</span>name<span class="op">=</span><span class="va">path_genera_dna</span>, genome_type<span class="op">=</span><span class="st">"DNA genome"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">viral_taxa</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count in each sample</span></span>
<span><span class="va">n_path_genera</span> <span class="op">&lt;-</span> <span class="va">hv_reads_genus</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">room</span>, <span class="va">na_type</span>, <span class="va">sample_type</span>, <span class="va">name</span>, <span class="va">taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">count</span><span class="op">(</span>name<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"taxid"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"room"</span>, <span class="st">"na_type"</span>, <span class="st">"sample_type"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot out and back to add zero lines</span></span>
<span><span class="va">n_path_genera_out</span> <span class="op">&lt;-</span> <span class="va">n_path_genera</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">name</span>, <span class="va">n_reads_viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"name"</span>, values_from<span class="op">=</span><span class="st">"n_reads_viral"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">-</span><span class="va">sample</span>, names_to<span class="op">=</span><span class="st">"name"</span>, values_to<span class="op">=</span><span class="st">"n_reads_viral"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">read_counts_raw</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">left_join</span><span class="op">(</span><span class="va">path_genera</span>, by<span class="op">=</span><span class="st">"name"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Aggregate across dates</span></span>
<span><span class="va">n_path_genera_stype</span> <span class="op">&lt;-</span> <span class="va">n_path_genera_out</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">na_type</span>, <span class="va">sample_type</span>,</span>
<span>           <span class="va">name</span>, <span class="va">taxid</span>, <span class="va">genome_type</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_reads_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_raw</span><span class="op">)</span>,</span>
<span>            n_reads_viral <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_reads_viral</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample<span class="op">=</span><span class="st">"All samples"</span>, date<span class="op">=</span><span class="st">"All dates"</span>,</span>
<span>         p_reads_viral <span class="op">=</span> <span class="va">n_reads_viral</span><span class="op">/</span><span class="va">n_reads_raw</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot</span></span>
<span><span class="va">g_path_genera</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">n_path_genera_stype</span>,</span>
<span>                        <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">name</span>, x<span class="op">=</span><span class="va">p_reads_viral</span>, color<span class="op">=</span><span class="va">na_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span>name<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_na</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">genome_type</span><span class="op">~</span><span class="va">sample_type</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.title.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_path_genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="2024-04-12_rosario_files/figure-html/ra-genera-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>One immediate observation, which I think helps demonstrate the unreliability of the RNA data here, is the almost complete absence of RNA-virus genera of interest. It’s very suspicious for putative HIV reads to be almost as abundant as <em>Enterovirus</em> reads (which, remember, includes rhinovirus), while coronaviruses and influenza are totally absent. The DNA reads seem much more similar to Prussin, though I’m still surprised HCMV and <em>Mastadenovirus</em> are so rare.</p>
</section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>This is the second, and by far the smaller, of the air-sampling datasets I’ve analyzed with this pipeline so far. Many of the high-level findings were similar to Prussin, including high relative abundance of human reads, low total viral reads, an absence of enteric viruses, and high abundance of papillomaviruses among human-infecting viruses. The biggest difference in the DNA data was the almost complete lack of cytomegalovirus, compared to its very high abundance in the Prussin data.</p>
<p>Given how small this dataset is (16M reads on a MiSeq, compared to ~450M for Prussin and 1.3B for the upcoming <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01044-7">Leung et al.</a> dataset), I don’t think too much weight should be placed on these findings, but given how rare air-MGS datasets are it’s nice to have another one.</p>


<!-- -->

</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb45" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Rosario et al. (2018)"</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Air sampling from a student dorm in Colorado."</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-04-12</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a>theme_rotate <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a>Continuing our look at air sampling datasets, we turn to <span class="co">[</span><span class="ot">Rosario et al. (2018)</span><span class="co">](https://pubs.acs.org/doi/10.1021/acs.est.7b04203)</span>, another study of air filters, this time from HVAC filters from an undergraduate dorm building at the University of Colorado campus in Boulder. As in <span class="co">[</span><span class="ot">Prussin</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-04-12_prussin.html)</span>, samples were eluted from filters (in this case MERV-8, so less stringent than Prussin's MERV-14 filters) and underwent both RNA and DNA sequencing – this time on an Illumina MiSeq with 2x250bp reads.</span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true" tabindex="-1"></a><span class="fu"># The raw data</span></span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true" tabindex="-1"></a>The Rosario dataset comprised sequencing data from 12 individual dormitory rooms at the sampling site, as well as a pooled sample of eight additional rooms and a negative control.</span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: read-qc-data</span></span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-04-12_rosario/"</span></span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"sample-metadata.csv"</span>)</span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats.tsv"</span>)</span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true" tabindex="-1"></a>adapter_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_adapter_stats.tsv"</span>)</span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true" tabindex="-1"></a>quality_base_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_base_stats.tsv"</span>)</span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_quality_sequence_stats.tsv"</span>)</span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries and extract metadata from sample names</span></span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true" tabindex="-1"></a>libraries_raw <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> libraries_raw <span class="sc">%&gt;%</span></span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(na_type)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">na_type =</span> <span class="fu">fct_inorder</span>(na_type)) <span class="sc">%&gt;%</span></span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_type =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Control"</span>, room), <span class="st">"NC"</span>,</span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Pool"</span>, room), <span class="st">"Pool"</span>, <span class="st">"Single-Room"</span>)),</span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample_type =</span> <span class="fu">factor</span>(sample_type, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Single-Room"</span>, <span class="st">"Pool"</span>, <span class="st">"NC"</span>)),</span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true" tabindex="-1"></a>         <span class="at">room =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Control"</span>, room), <span class="st">"Negative Control"</span>, room))</span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Import QC data</span></span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true" tabindex="-1"></a>adapter_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(adapter_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">sub</span>(<span class="st">"_2$"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true" tabindex="-1"></a>quality_base_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_base_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true" tabindex="-1"></a>quality_seq_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(quality_seq_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">read_pair =</span> <span class="fu">fct_inorder</span>(<span class="fu">as.character</span>(read_pair)))</span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to raw data</span></span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true" tabindex="-1"></a>basic_stats_raw <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb45-88"><a href="#cb45-88" aria-hidden="true" tabindex="-1"></a>adapter_stats_raw <span class="ot">&lt;-</span> adapter_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb45-89"><a href="#cb45-89" aria-hidden="true" tabindex="-1"></a>quality_base_stats_raw <span class="ot">&lt;-</span> quality_base_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb45-90"><a href="#cb45-90" aria-hidden="true" tabindex="-1"></a>quality_seq_stats_raw <span class="ot">&lt;-</span> quality_seq_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"raw_concat"</span>)</span>
<span id="cb45-91"><a href="#cb45-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-92"><a href="#cb45-92" aria-hidden="true" tabindex="-1"></a><span class="co"># Get key values for readout</span></span>
<span id="cb45-93"><a href="#cb45-93" aria-hidden="true" tabindex="-1"></a>raw_read_counts <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type, sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-94"><a href="#cb45-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">rmin =</span> <span class="fu">min</span>(n_read_pairs), <span class="at">rmax=</span><span class="fu">max</span>(n_read_pairs),</span>
<span id="cb45-95"><a href="#cb45-95" aria-hidden="true" tabindex="-1"></a>            <span class="at">rmean=</span><span class="fu">mean</span>(n_read_pairs), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-96"><a href="#cb45-96" aria-hidden="true" tabindex="-1"></a>raw_read_totals <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type, sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-97"><a href="#cb45-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_read_pairs =</span> <span class="fu">sum</span>(n_read_pairs), </span>
<span id="cb45-98"><a href="#cb45-98" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_bases_approx =</span> <span class="fu">sum</span>(n_bases_approx), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-99"><a href="#cb45-99" aria-hidden="true" tabindex="-1"></a>raw_dup <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type, sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-100"><a href="#cb45-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb45-101"><a href="#cb45-101" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-102"><a href="#cb45-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-103"><a href="#cb45-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-104"><a href="#cb45-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-105"><a href="#cb45-105" aria-hidden="true" tabindex="-1"></a>The 12 single-room samples yielded 0.57-0.74M (mean 0.67M) RNA-sequencing reads and 0.48M-0.84M (mean 0.67M) DNA-sequencing reads; the pooled and control samples had similar depth. In total, non-control samples yielded 8.0M RNA read pairs and 8.0M DNA read pairs (4 gigabases of sequence each).</span>
<span id="cb45-106"><a href="#cb45-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-107"><a href="#cb45-107" aria-hidden="true" tabindex="-1"></a>Read qualities were so-so, in need of cleaning. Adapter levels were high. Inferred duplication levels were low (4.6-8.2%) in DNA reads but highly variable (3-83%) in RNA reads.</span>
<span id="cb45-108"><a href="#cb45-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-111"><a href="#cb45-111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-112"><a href="#cb45-112" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb45-113"><a href="#cb45-113" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-114"><a href="#cb45-114" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-basic-stats</span></span>
<span id="cb45-115"><a href="#cb45-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-116"><a href="#cb45-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare data</span></span>
<span id="cb45-117"><a href="#cb45-117" aria-hidden="true" tabindex="-1"></a>basic_stats_raw_metrics <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb45-118"><a href="#cb45-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(room, na_type, sample_type,</span>
<span id="cb45-119"><a href="#cb45-119" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at"># Read pairs</span><span class="st">`</span> <span class="ot">=</span> n_read_pairs,</span>
<span id="cb45-120"><a href="#cb45-120" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">Total base pairs</span><span class="sc">\n</span><span class="at">(approx)</span><span class="st">`</span> <span class="ot">=</span> n_bases_approx,</span>
<span id="cb45-121"><a href="#cb45-121" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">% Duplicates</span><span class="sc">\n</span><span class="at">(FASTQC)</span><span class="st">`</span> <span class="ot">=</span> percent_duplicates) <span class="sc">%&gt;%</span></span>
<span id="cb45-122"><a href="#cb45-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>(room<span class="sc">:</span>sample_type), <span class="at">names_to =</span> <span class="st">"metric"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-123"><a href="#cb45-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">metric =</span> <span class="fu">fct_inorder</span>(metric))</span>
<span id="cb45-124"><a href="#cb45-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-125"><a href="#cb45-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plot templates</span></span>
<span id="cb45-126"><a href="#cb45-126" aria-hidden="true" tabindex="-1"></a>scale_fill_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_fill_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>, </span>
<span id="cb45-127"><a href="#cb45-127" aria-hidden="true" tabindex="-1"></a>                                <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb45-128"><a href="#cb45-128" aria-hidden="true" tabindex="-1"></a>g_basic <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats_raw_metrics, </span>
<span id="cb45-129"><a href="#cb45-129" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>room, <span class="at">y=</span>value, <span class="at">fill=</span>na_type)) <span class="sc">+</span></span>
<span id="cb45-130"><a href="#cb45-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb45-131"><a href="#cb45-131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Collection Date"</span>) <span class="sc">+</span></span>
<span id="cb45-132"><a href="#cb45-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-133"><a href="#cb45-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb45-134"><a href="#cb45-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_na</span>() <span class="sc">+</span> </span>
<span id="cb45-135"><a href="#cb45-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(metric<span class="sc">~</span>sample_type, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space=</span><span class="st">"free_x"</span>, <span class="at">switch=</span><span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb45-136"><a href="#cb45-136" aria-hidden="true" tabindex="-1"></a>  theme_rotate <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-137"><a href="#cb45-137" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-138"><a href="#cb45-138" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"plain"</span>)</span>
<span id="cb45-139"><a href="#cb45-139" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb45-140"><a href="#cb45-140" aria-hidden="true" tabindex="-1"></a>g_basic</span>
<span id="cb45-141"><a href="#cb45-141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-142"><a href="#cb45-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-145"><a href="#cb45-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-146"><a href="#cb45-146" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-raw-quality</span></span>
<span id="cb45-147"><a href="#cb45-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-148"><a href="#cb45-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up plotting templates</span></span>
<span id="cb45-149"><a href="#cb45-149" aria-hidden="true" tabindex="-1"></a>scale_color_na <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_color_brewer, <span class="at">palette=</span><span class="st">"Set1"</span>,</span>
<span id="cb45-150"><a href="#cb45-150" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">name=</span><span class="st">"Nucleic acid type"</span>)</span>
<span id="cb45-151"><a href="#cb45-151" aria-hidden="true" tabindex="-1"></a>g_qual_raw <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>na_type, <span class="at">linetype=</span>read_pair, </span>
<span id="cb45-152"><a href="#cb45-152" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb45-153"><a href="#cb45-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb45-154"><a href="#cb45-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb45-155"><a href="#cb45-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb45-156"><a href="#cb45-156" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb45-157"><a href="#cb45-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-158"><a href="#cb45-158" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb45-159"><a href="#cb45-159" aria-hidden="true" tabindex="-1"></a>g_adapters_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span> </span>
<span id="cb45-160"><a href="#cb45-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats_raw) <span class="sc">+</span></span>
<span id="cb45-161"><a href="#cb45-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb45-162"><a href="#cb45-162" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-163"><a href="#cb45-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb45-164"><a href="#cb45-164" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-165"><a href="#cb45-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(sample_type<span class="sc">~</span>adapter)</span>
<span id="cb45-166"><a href="#cb45-166" aria-hidden="true" tabindex="-1"></a>g_adapters_raw</span>
<span id="cb45-167"><a href="#cb45-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-168"><a href="#cb45-168" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb45-169"><a href="#cb45-169" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb45-170"><a href="#cb45-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-171"><a href="#cb45-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-172"><a href="#cb45-172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats_raw) <span class="sc">+</span></span>
<span id="cb45-173"><a href="#cb45-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb45-174"><a href="#cb45-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb45-175"><a href="#cb45-175" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-176"><a href="#cb45-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(sample_type<span class="sc">~</span>.)</span>
<span id="cb45-177"><a href="#cb45-177" aria-hidden="true" tabindex="-1"></a>g_quality_base_raw</span>
<span id="cb45-178"><a href="#cb45-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-179"><a href="#cb45-179" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw <span class="ot">&lt;-</span> g_qual_raw <span class="sc">+</span></span>
<span id="cb45-180"><a href="#cb45-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-181"><a href="#cb45-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-182"><a href="#cb45-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats_raw) <span class="sc">+</span></span>
<span id="cb45-183"><a href="#cb45-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-184"><a href="#cb45-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-185"><a href="#cb45-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(sample_type<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">"free_y"</span>)</span>
<span id="cb45-186"><a href="#cb45-186" aria-hidden="true" tabindex="-1"></a>g_quality_seq_raw</span>
<span id="cb45-187"><a href="#cb45-187" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-188"><a href="#cb45-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-189"><a href="#cb45-189" aria-hidden="true" tabindex="-1"></a><span class="fu"># Preprocessing</span></span>
<span id="cb45-190"><a href="#cb45-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-191"><a href="#cb45-191" aria-hidden="true" tabindex="-1"></a>The average fraction of reads lost at each stage in the preprocessing pipeline is shown in the following table. Single-room samples lost an average of 52%/14% (RNA/DNA) of total read pairs during cleaning and deduplication, with a further loss of 4.3%/0.1% during ribodepletion. However, the fraction of reads lost, especially during cleaning and deduplication, was highly variable between samples.</span>
<span id="cb45-192"><a href="#cb45-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-195"><a href="#cb45-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-196"><a href="#cb45-196" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-table</span></span>
<span id="cb45-197"><a href="#cb45-197" aria-hidden="true" tabindex="-1"></a>n_reads_rel <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb45-198"><a href="#cb45-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, na_type, sample_type, stage, </span>
<span id="cb45-199"><a href="#cb45-199" aria-hidden="true" tabindex="-1"></a>         percent_duplicates, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb45-200"><a href="#cb45-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, na_type) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample, na_type, stage) <span class="sc">%&gt;%</span></span>
<span id="cb45-201"><a href="#cb45-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_retained =</span> <span class="fu">replace_na</span>(n_read_pairs <span class="sc">/</span> <span class="fu">lag</span>(n_read_pairs), <span class="dv">0</span>),</span>
<span id="cb45-202"><a href="#cb45-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost =</span> <span class="dv">1</span> <span class="sc">-</span> p_reads_retained,</span>
<span id="cb45-203"><a href="#cb45-203" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_retained_abs =</span> n_read_pairs <span class="sc">/</span> n_read_pairs[<span class="dv">1</span>],</span>
<span id="cb45-204"><a href="#cb45-204" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_retained_abs,</span>
<span id="cb45-205"><a href="#cb45-205" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_lost_abs_marginal =</span> <span class="fu">replace_na</span>(p_reads_lost_abs <span class="sc">-</span> <span class="fu">lag</span>(p_reads_lost_abs), <span class="dv">0</span>))</span>
<span id="cb45-206"><a href="#cb45-206" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display <span class="ot">&lt;-</span> n_reads_rel <span class="sc">%&gt;%</span> </span>
<span id="cb45-207"><a href="#cb45-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Stage=</span>stage, <span class="st">`</span><span class="at">NA Type</span><span class="st">`</span><span class="ot">=</span>na_type, <span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span><span class="ot">=</span>sample_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-208"><a href="#cb45-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Sample Type</span><span class="st">`</span>, <span class="st">`</span><span class="at">NA Type</span><span class="st">`</span>, Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb45-209"><a href="#cb45-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="st">`</span><span class="at">% Total Reads Lost (Cumulative)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>),</span>
<span id="cb45-210"><a href="#cb45-210" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">% Total Reads Lost (Marginal)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">min</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">"-"</span>, <span class="fu">round</span>(<span class="fu">max</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">" (mean "</span>, <span class="fu">round</span>(<span class="fu">mean</span>(p_reads_lost_abs_marginal<span class="sc">*</span><span class="dv">100</span>),<span class="dv">1</span>), <span class="st">")"</span>), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-211"><a href="#cb45-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Stage <span class="sc">!=</span> <span class="st">"raw_concat"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-212"><a href="#cb45-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage =</span> Stage <span class="sc">%&gt;%</span> as.numeric <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Trimming &amp; filtering"</span>, <span class="st">"Deduplication"</span>, <span class="st">"Initial ribodepletion"</span>, <span class="st">"Secondary ribodepletion"</span>)))</span>
<span id="cb45-213"><a href="#cb45-213" aria-hidden="true" tabindex="-1"></a>n_reads_rel_display</span>
<span id="cb45-214"><a href="#cb45-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-215"><a href="#cb45-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-218"><a href="#cb45-218" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-219"><a href="#cb45-219" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-figures</span></span>
<span id="cb45-220"><a href="#cb45-220" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-221"><a href="#cb45-221" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb45-222"><a href="#cb45-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb45-223"><a href="#cb45-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-224"><a href="#cb45-224" aria-hidden="true" tabindex="-1"></a>g_stage_trace <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(basic_stats, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>na_type, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb45-225"><a href="#cb45-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb45-226"><a href="#cb45-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(sample_type<span class="sc">~</span>na_type, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb45-227"><a href="#cb45-227" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line=</span><span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb45-228"><a href="#cb45-228" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb45-229"><a href="#cb45-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-230"><a href="#cb45-230" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot reads over preprocessing</span></span>
<span id="cb45-231"><a href="#cb45-231" aria-hidden="true" tabindex="-1"></a>g_reads_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb45-232"><a href="#cb45-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>n_read_pairs)) <span class="sc">+</span></span>
<span id="cb45-233"><a href="#cb45-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"# Read pairs"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb45-234"><a href="#cb45-234" aria-hidden="true" tabindex="-1"></a>g_reads_stages</span>
<span id="cb45-235"><a href="#cb45-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-236"><a href="#cb45-236" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot relative read losses during preprocessing</span></span>
<span id="cb45-237"><a href="#cb45-237" aria-hidden="true" tabindex="-1"></a>g_reads_rel <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_reads_rel, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">color=</span>na_type, <span class="at">group=</span>sample)) <span class="sc">+</span></span>
<span id="cb45-238"><a href="#cb45-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_lost_abs_marginal)) <span class="sc">+</span></span>
<span id="cb45-239"><a href="#cb45-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Total Reads Lost"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), </span>
<span id="cb45-240"><a href="#cb45-240" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-241"><a href="#cb45-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb45-242"><a href="#cb45-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(sample_type<span class="sc">~</span>na_type, <span class="at">scales=</span><span class="st">"free"</span>, <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb45-243"><a href="#cb45-243" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line=</span><span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb45-244"><a href="#cb45-244" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb45-245"><a href="#cb45-245" aria-hidden="true" tabindex="-1"></a>g_reads_rel</span>
<span id="cb45-246"><a href="#cb45-246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-247"><a href="#cb45-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-248"><a href="#cb45-248" aria-hidden="true" tabindex="-1"></a>Data cleaning was very successful at improving read qualities, as well as removing most adapter sequences. The exception to the latter was terminal poly-A sequences, which remained at a substantial (though reduced) prevalence throughout the pipeline. (This doesn't seem to have caused downstream problems that I can see.)</span>
<span id="cb45-249"><a href="#cb45-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-252"><a href="#cb45-252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-253"><a href="#cb45-253" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-254"><a href="#cb45-254" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-quality</span></span>
<span id="cb45-255"><a href="#cb45-255" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb45-256"><a href="#cb45-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-257"><a href="#cb45-257" aria-hidden="true" tabindex="-1"></a>g_qual <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">color=</span>na_type, <span class="at">linetype=</span>read_pair, </span>
<span id="cb45-258"><a href="#cb45-258" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group=</span><span class="fu">interaction</span>(sample,read_pair))) <span class="sc">+</span> </span>
<span id="cb45-259"><a href="#cb45-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span> <span class="fu">scale_linetype_discrete</span>(<span class="at">name =</span> <span class="st">"Read Pair"</span>) <span class="sc">+</span></span>
<span id="cb45-260"><a href="#cb45-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>),</span>
<span id="cb45-261"><a href="#cb45-261" aria-hidden="true" tabindex="-1"></a>         <span class="at">linetype =</span> <span class="fu">guide_legend</span>(<span class="at">nrow=</span><span class="dv">2</span>,<span class="at">byrow=</span><span class="cn">TRUE</span>)) <span class="sc">+</span></span>
<span id="cb45-262"><a href="#cb45-262" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb45-263"><a href="#cb45-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-264"><a href="#cb45-264" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize adapters</span></span>
<span id="cb45-265"><a href="#cb45-265" aria-hidden="true" tabindex="-1"></a>g_adapters <span class="ot">&lt;-</span> g_qual <span class="sc">+</span> </span>
<span id="cb45-266"><a href="#cb45-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>pc_adapters), <span class="at">data=</span>adapter_stats) <span class="sc">+</span></span>
<span id="cb45-267"><a href="#cb45-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Adapters"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>),</span>
<span id="cb45-268"><a href="#cb45-268" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">50</span>,<span class="dv">10</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-269"><a href="#cb45-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb45-270"><a href="#cb45-270" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-271"><a href="#cb45-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>adapter)</span>
<span id="cb45-272"><a href="#cb45-272" aria-hidden="true" tabindex="-1"></a>g_adapters</span>
<span id="cb45-273"><a href="#cb45-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-274"><a href="#cb45-274" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize quality</span></span>
<span id="cb45-275"><a href="#cb45-275" aria-hidden="true" tabindex="-1"></a>g_quality_base <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb45-276"><a href="#cb45-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-277"><a href="#cb45-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-278"><a href="#cb45-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>position, <span class="at">y=</span>mean_phred_score), <span class="at">data=</span>quality_base_stats) <span class="sc">+</span></span>
<span id="cb45-279"><a href="#cb45-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb45-280"><a href="#cb45-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Position"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>),</span>
<span id="cb45-281"><a href="#cb45-281" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">140</span>,<span class="dv">20</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-282"><a href="#cb45-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb45-283"><a href="#cb45-283" aria-hidden="true" tabindex="-1"></a>g_quality_base</span>
<span id="cb45-284"><a href="#cb45-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-285"><a href="#cb45-285" aria-hidden="true" tabindex="-1"></a>g_quality_seq <span class="ot">&lt;-</span> g_qual <span class="sc">+</span></span>
<span id="cb45-286"><a href="#cb45-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">25</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-287"><a href="#cb45-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">30</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb45-288"><a href="#cb45-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_phred_score, <span class="at">y=</span>n_sequences), <span class="at">data=</span>quality_seq_stats) <span class="sc">+</span></span>
<span id="cb45-289"><a href="#cb45-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"Mean Phred score"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-290"><a href="#cb45-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Sequences"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-291"><a href="#cb45-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(stage<span class="sc">~</span>.)</span>
<span id="cb45-292"><a href="#cb45-292" aria-hidden="true" tabindex="-1"></a>g_quality_seq</span>
<span id="cb45-293"><a href="#cb45-293" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-294"><a href="#cb45-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-295"><a href="#cb45-295" aria-hidden="true" tabindex="-1"></a>According to FASTQC, deduplication was very effective at reducing measured duplicate levels, which for single-room samples fell from an average of 40% to 7.8% for RNA reads and from 5.5% to 0.3% for DNA reads.</span>
<span id="cb45-296"><a href="#cb45-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-299"><a href="#cb45-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-300"><a href="#cb45-300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: preproc-dedup</span></span>
<span id="cb45-301"><a href="#cb45-301" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb45-302"><a href="#cb45-302" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb45-303"><a href="#cb45-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-304"><a href="#cb45-304" aria-hidden="true" tabindex="-1"></a>stage_dup <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type, sample_type, stage) <span class="sc">%&gt;%</span> </span>
<span id="cb45-305"><a href="#cb45-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">dmin =</span> <span class="fu">min</span>(percent_duplicates), <span class="at">dmax=</span><span class="fu">max</span>(percent_duplicates),</span>
<span id="cb45-306"><a href="#cb45-306" aria-hidden="true" tabindex="-1"></a>            <span class="at">dmean=</span><span class="fu">mean</span>(percent_duplicates), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-307"><a href="#cb45-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-308"><a href="#cb45-308" aria-hidden="true" tabindex="-1"></a>g_dup_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span></span>
<span id="cb45-309"><a href="#cb45-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>percent_duplicates)) <span class="sc">+</span></span>
<span id="cb45-310"><a href="#cb45-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"% Duplicates"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>), <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb45-311"><a href="#cb45-311" aria-hidden="true" tabindex="-1"></a>g_dup_stages</span>
<span id="cb45-312"><a href="#cb45-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-313"><a href="#cb45-313" aria-hidden="true" tabindex="-1"></a>g_readlen_stages <span class="ot">&lt;-</span> g_stage_trace <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y=</span>mean_seq_len)) <span class="sc">+</span></span>
<span id="cb45-314"><a href="#cb45-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="st">"Mean read length (nt)"</span>, <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="cn">NA</span>))</span>
<span id="cb45-315"><a href="#cb45-315" aria-hidden="true" tabindex="-1"></a>g_readlen_stages</span>
<span id="cb45-316"><a href="#cb45-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-317"><a href="#cb45-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-318"><a href="#cb45-318" aria-hidden="true" tabindex="-1"></a><span class="fu"># High-level composition</span></span>
<span id="cb45-319"><a href="#cb45-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-320"><a href="#cb45-320" aria-hidden="true" tabindex="-1"></a>As before, to assess the high-level composition of the reads, I ran the ribodepleted files through Kraken (using the Standard 16 database) and summarized the results with Bracken. Combining these results with the read counts above gives us a breakdown of the inferred composition of the samples:</span>
<span id="cb45-321"><a href="#cb45-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-324"><a href="#cb45-324" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-325"><a href="#cb45-325" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: prepare-composition</span></span>
<span id="cb45-326"><a href="#cb45-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-327"><a href="#cb45-327" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bracken data</span></span>
<span id="cb45-328"><a href="#cb45-328" aria-hidden="true" tabindex="-1"></a>bracken_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"bracken_counts.tsv"</span>)</span>
<span id="cb45-329"><a href="#cb45-329" aria-hidden="true" tabindex="-1"></a>bracken <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(bracken_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-330"><a href="#cb45-330" aria-hidden="true" tabindex="-1"></a>total_assigned <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(</span>
<span id="cb45-331"><a href="#cb45-331" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"Total"</span>,</span>
<span id="cb45-332"><a href="#cb45-332" aria-hidden="true" tabindex="-1"></a>  <span class="at">kraken_assigned_reads =</span> <span class="fu">sum</span>(kraken_assigned_reads),</span>
<span id="cb45-333"><a href="#cb45-333" aria-hidden="true" tabindex="-1"></a>  <span class="at">added_reads =</span> <span class="fu">sum</span>(added_reads),</span>
<span id="cb45-334"><a href="#cb45-334" aria-hidden="true" tabindex="-1"></a>  <span class="at">new_est_reads =</span> <span class="fu">sum</span>(new_est_reads),</span>
<span id="cb45-335"><a href="#cb45-335" aria-hidden="true" tabindex="-1"></a>  <span class="at">fraction_total_reads =</span> <span class="fu">sum</span>(fraction_total_reads)</span>
<span id="cb45-336"><a href="#cb45-336" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-337"><a href="#cb45-337" aria-hidden="true" tabindex="-1"></a>bracken_spread <span class="ot">&lt;-</span> bracken <span class="sc">%&gt;%</span> <span class="fu">select</span>(name, sample, new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb45-338"><a href="#cb45-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">tolower</span>(name)) <span class="sc">%&gt;%</span></span>
<span id="cb45-339"><a href="#cb45-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"sample"</span>, <span class="at">names_from =</span> <span class="st">"name"</span>, </span>
<span id="cb45-340"><a href="#cb45-340" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="st">"new_est_reads"</span>)</span>
<span id="cb45-341"><a href="#cb45-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-342"><a href="#cb45-342" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads</span></span>
<span id="cb45-343"><a href="#cb45-343" aria-hidden="true" tabindex="-1"></a>read_counts_preproc <span class="ot">&lt;-</span> basic_stats <span class="sc">%&gt;%</span> </span>
<span id="cb45-344"><a href="#cb45-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, room, na_type, sample_type, stage, n_read_pairs) <span class="sc">%&gt;%</span></span>
<span id="cb45-345"><a href="#cb45-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"room"</span>, <span class="st">"na_type"</span>, <span class="st">"sample_type"</span>),</span>
<span id="cb45-346"><a href="#cb45-346" aria-hidden="true" tabindex="-1"></a>              <span class="at">names_from=</span><span class="st">"stage"</span>, <span class="at">values_from=</span><span class="st">"n_read_pairs"</span>)</span>
<span id="cb45-347"><a href="#cb45-347" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_preproc <span class="sc">%&gt;%</span></span>
<span id="cb45-348"><a href="#cb45-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(total_assigned <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, new_est_reads), <span class="at">by =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-349"><a href="#cb45-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">assigned =</span> new_est_reads) <span class="sc">%&gt;%</span></span>
<span id="cb45-350"><a href="#cb45-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(bracken_spread, <span class="at">by=</span><span class="st">"sample"</span>)</span>
<span id="cb45-351"><a href="#cb45-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-352"><a href="#cb45-352" aria-hidden="true" tabindex="-1"></a><span class="co"># Assess composition</span></span>
<span id="cb45-353"><a href="#cb45-353" aria-hidden="true" tabindex="-1"></a>read_comp <span class="ot">&lt;-</span> <span class="fu">transmute</span>(read_counts, <span class="at">sample=</span>sample, <span class="at">na_type=</span>na_type,</span>
<span id="cb45-354"><a href="#cb45-354" aria-hidden="true" tabindex="-1"></a>                       <span class="at">room=</span>room, <span class="at">sample_type=</span>sample_type,</span>
<span id="cb45-355"><a href="#cb45-355" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_filtered =</span> raw_concat<span class="sc">-</span>cleaned,</span>
<span id="cb45-356"><a href="#cb45-356" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_duplicate =</span> cleaned<span class="sc">-</span>dedup,</span>
<span id="cb45-357"><a href="#cb45-357" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_ribosomal =</span> (dedup<span class="sc">-</span>ribo_initial) <span class="sc">+</span> (ribo_initial<span class="sc">-</span>ribo_secondary),</span>
<span id="cb45-358"><a href="#cb45-358" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_unassigned =</span> ribo_secondary<span class="sc">-</span>assigned,</span>
<span id="cb45-359"><a href="#cb45-359" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_bacterial =</span> bacteria,</span>
<span id="cb45-360"><a href="#cb45-360" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_archaeal =</span> archaea,</span>
<span id="cb45-361"><a href="#cb45-361" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_viral =</span> viruses,</span>
<span id="cb45-362"><a href="#cb45-362" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n_human =</span> eukaryota)</span>
<span id="cb45-363"><a href="#cb45-363" aria-hidden="true" tabindex="-1"></a>read_comp_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(read_comp, <span class="sc">-</span>(sample<span class="sc">:</span>sample_type), </span>
<span id="cb45-364"><a href="#cb45-364" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_to =</span> <span class="st">"classification"</span>,</span>
<span id="cb45-365"><a href="#cb45-365" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_prefix =</span> <span class="st">"n_"</span>, <span class="at">values_to =</span> <span class="st">"n_reads"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-366"><a href="#cb45-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(<span class="fu">str_to_sentence</span>(classification))) <span class="sc">%&gt;%</span></span>
<span id="cb45-367"><a href="#cb45-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads))</span>
<span id="cb45-368"><a href="#cb45-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-369"><a href="#cb45-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize composition</span></span>
<span id="cb45-370"><a href="#cb45-370" aria-hidden="true" tabindex="-1"></a>read_comp_summ <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb45-371"><a href="#cb45-371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, sample_type, classification) <span class="sc">%&gt;%</span></span>
<span id="cb45-372"><a href="#cb45-372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads =</span> <span class="fu">sum</span>(n_reads), <span class="at">.groups =</span> <span class="st">"drop_last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-373"><a href="#cb45-373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads =</span> <span class="fu">replace_na</span>(n_reads,<span class="dv">0</span>),</span>
<span id="cb45-374"><a href="#cb45-374" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_reads =</span> n_reads<span class="sc">/</span><span class="fu">sum</span>(n_reads),</span>
<span id="cb45-375"><a href="#cb45-375" aria-hidden="true" tabindex="-1"></a>    <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb45-376"><a href="#cb45-376" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-377"><a href="#cb45-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-380"><a href="#cb45-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-381"><a href="#cb45-381" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-composition-all</span></span>
<span id="cb45-382"><a href="#cb45-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-383"><a href="#cb45-383" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare plotting templates</span></span>
<span id="cb45-384"><a href="#cb45-384" aria-hidden="true" tabindex="-1"></a>g_comp_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">x=</span>room, <span class="at">y=</span>p_reads, <span class="at">fill=</span>classification)) <span class="sc">+</span></span>
<span id="cb45-385"><a href="#cb45-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Collection Date"</span>) <span class="sc">+</span></span>
<span id="cb45-386"><a href="#cb45-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>sample_type, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb45-387"><a href="#cb45-387" aria-hidden="true" tabindex="-1"></a>  theme_rotate</span>
<span id="cb45-388"><a href="#cb45-388" aria-hidden="true" tabindex="-1"></a>scale_y_pc_reads <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(scale_y_continuous, <span class="at">name =</span> <span class="st">"% Reads"</span>,</span>
<span id="cb45-389"><a href="#cb45-389" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb45-390"><a href="#cb45-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-391"><a href="#cb45-391" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot overall composition</span></span>
<span id="cb45-392"><a href="#cb45-392" aria-hidden="true" tabindex="-1"></a>g_comp <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data =</span> read_comp_long, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-393"><a href="#cb45-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb45-394"><a href="#cb45-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb45-395"><a href="#cb45-395" aria-hidden="true" tabindex="-1"></a>g_comp</span>
<span id="cb45-396"><a href="#cb45-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-397"><a href="#cb45-397" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot composition of minor components</span></span>
<span id="cb45-398"><a href="#cb45-398" aria-hidden="true" tabindex="-1"></a>read_comp_minor <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span> </span>
<span id="cb45-399"><a href="#cb45-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Archaeal"</span>, <span class="st">"Viral"</span>, <span class="st">"Other"</span>))</span>
<span id="cb45-400"><a href="#cb45-400" aria-hidden="true" tabindex="-1"></a>palette_minor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">9</span>)]</span>
<span id="cb45-401"><a href="#cb45-401" aria-hidden="true" tabindex="-1"></a>g_comp_minor <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> <span class="fu">geom_col</span>(<span class="at">data=</span>read_comp_minor, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-402"><a href="#cb45-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_pc_reads</span>() <span class="sc">+</span></span>
<span id="cb45-403"><a href="#cb45-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_minor, <span class="at">name =</span> <span class="st">"Classification"</span>)</span>
<span id="cb45-404"><a href="#cb45-404" aria-hidden="true" tabindex="-1"></a>g_comp_minor</span>
<span id="cb45-405"><a href="#cb45-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-406"><a href="#cb45-406" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-407"><a href="#cb45-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-410"><a href="#cb45-410" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-411"><a href="#cb45-411" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: composition-summary</span></span>
<span id="cb45-412"><a href="#cb45-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-413"><a href="#cb45-413" aria-hidden="true" tabindex="-1"></a>p_reads_summ_group <span class="ot">&lt;-</span> read_comp_long <span class="sc">%&gt;%</span></span>
<span id="cb45-414"><a href="#cb45-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">ifelse</span>(classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Filtered"</span>, <span class="st">"Duplicate"</span>, <span class="st">"Unassigned"</span>), <span class="st">"Excluded"</span>, <span class="fu">as.character</span>(classification)),</span>
<span id="cb45-415"><a href="#cb45-415" aria-hidden="true" tabindex="-1"></a>         <span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification)) <span class="sc">%&gt;%</span></span>
<span id="cb45-416"><a href="#cb45-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, sample, na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-417"><a href="#cb45-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads =</span> <span class="fu">sum</span>(p_reads), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-418"><a href="#cb45-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(classification, na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-419"><a href="#cb45-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">pc_min =</span> <span class="fu">min</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">pc_max =</span> <span class="fu">max</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, </span>
<span id="cb45-420"><a href="#cb45-420" aria-hidden="true" tabindex="-1"></a>            <span class="at">pc_mean =</span> <span class="fu">mean</span>(p_reads)<span class="sc">*</span><span class="dv">100</span>, <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-421"><a href="#cb45-421" aria-hidden="true" tabindex="-1"></a>p_reads_summ_prep <span class="ot">&lt;-</span> p_reads_summ_group <span class="sc">%&gt;%</span></span>
<span id="cb45-422"><a href="#cb45-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">classification =</span> <span class="fu">fct_inorder</span>(classification),</span>
<span id="cb45-423"><a href="#cb45-423" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_min =</span> pc_min <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb45-424"><a href="#cb45-424" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_max =</span> pc_max <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb45-425"><a href="#cb45-425" aria-hidden="true" tabindex="-1"></a>         <span class="at">pc_mean =</span> pc_mean <span class="sc">%&gt;%</span> <span class="fu">signif</span>(<span class="at">digits=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(format, <span class="at">scientific=</span><span class="cn">FALSE</span>, <span class="at">trim=</span><span class="cn">TRUE</span>, <span class="at">digits=</span><span class="dv">2</span>),</span>
<span id="cb45-426"><a href="#cb45-426" aria-hidden="true" tabindex="-1"></a>         <span class="at">display =</span> <span class="fu">paste0</span>(pc_min, <span class="st">"-"</span>, pc_max, <span class="st">"% (mean "</span>, pc_mean, <span class="st">"%)"</span>))</span>
<span id="cb45-427"><a href="#cb45-427" aria-hidden="true" tabindex="-1"></a>p_reads_summ <span class="ot">&lt;-</span> p_reads_summ_prep <span class="sc">%&gt;%</span></span>
<span id="cb45-428"><a href="#cb45-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample_type, classification, na_type, display) <span class="sc">%&gt;%</span></span>
<span id="cb45-429"><a href="#cb45-429" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>na_type, <span class="at">values_from =</span> display) <span class="sc">%&gt;%</span> </span>
<span id="cb45-430"><a href="#cb45-430" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(sample_type, classification)</span>
<span id="cb45-431"><a href="#cb45-431" aria-hidden="true" tabindex="-1"></a>p_reads_summ</span>
<span id="cb45-432"><a href="#cb45-432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-433"><a href="#cb45-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-434"><a href="#cb45-434" aria-hidden="true" tabindex="-1"></a>As with the Prussin data, human reads are very elevated compared to wastewater, making up an average of 21% of read pairs in RNA libraries and 4% for DNA libraries for single-room samples (in both cases, the maximum abundance of human reads across samples is much higher). Viral read abundances are higher than Prussin, averaging 0.08% for single-room RNA libraries and 0.04% for DNA libraries; for some reason, the pooled RNA library shows much higher viral prevalence, at 0.33%. The most obvious difference from the Prussin data is that DNA libraries in Rosario show a much higher fraction of non-ribosomal unassigned reads, with an average of 78% of all reads for single-room libraries.</span>
<span id="cb45-435"><a href="#cb45-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-436"><a href="#cb45-436" aria-hidden="true" tabindex="-1"></a>As in Prussin, viral DNA reads were dominated by *Caudoviricetes* phages, with many unclassifiable below the class level. Other prominent viral classes in DNA reads included *Papovaviricetes* (which includes *Polyomaviridae* and *Papillomaviridae*), *Megaviridae* (giant viruses) and *Repensiviricetes* (plant and fungal viruses). Pooled samples also showed a significant level of *Quintoviricetes* (parvoviruses).</span>
<span id="cb45-437"><a href="#cb45-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-438"><a href="#cb45-438" aria-hidden="true" tabindex="-1"></a>RNA libraries, meanwhile, were dominated by *Alsuviricetes* (mainly plant viruses) and, notably, *Pisoniviricetes* (which includes coronaviruses, picornaviruses including *Enterovirus*, and caliciviruses including *Norovirus*). *Caudoviricetes*, *Herviviricetes* (herpesviruses) and *Revtraviricetes* (retroviruses + Hep B) were also prominent in some samples.</span>
<span id="cb45-439"><a href="#cb45-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-442"><a href="#cb45-442" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-443"><a href="#cb45-443" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: extract-viral-taxa</span></span>
<span id="cb45-444"><a href="#cb45-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-445"><a href="#cb45-445" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxonomy</span></span>
<span id="cb45-446"><a href="#cb45-446" aria-hidden="true" tabindex="-1"></a>viral_taxa_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"viral-taxids.tsv.gz"</span>)</span>
<span id="cb45-447"><a href="#cb45-447" aria-hidden="true" tabindex="-1"></a>viral_taxa <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(viral_taxa_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-448"><a href="#cb45-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-449"><a href="#cb45-449" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Kraken reports &amp; extract viral taxa</span></span>
<span id="cb45-450"><a href="#cb45-450" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">as.character</span>(basic_stats_raw<span class="sc">$</span>sample)</span>
<span id="cb45-451"><a href="#cb45-451" aria-hidden="true" tabindex="-1"></a>col_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"pc_reads_total"</span>, <span class="st">"n_reads_clade"</span>, <span class="st">"n_reads_direct"</span>,</span>
<span id="cb45-452"><a href="#cb45-452" aria-hidden="true" tabindex="-1"></a>               <span class="st">"rank"</span>, <span class="st">"taxid"</span>, <span class="st">"name"</span>)</span>
<span id="cb45-453"><a href="#cb45-453" aria-hidden="true" tabindex="-1"></a>report_paths <span class="ot">&lt;-</span> <span class="fu">paste0</span>(data_dir, <span class="st">"kraken/"</span>, samples, <span class="st">".report.gz"</span>)</span>
<span id="cb45-454"><a href="#cb45-454" aria-hidden="true" tabindex="-1"></a>kraken_reports <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(samples), <span class="cf">function</span>(n) </span>
<span id="cb45-455"><a href="#cb45-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_tsv</span>(report_paths[n], <span class="at">col_names =</span> col_names, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-456"><a href="#cb45-456" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample =</span> samples[n])) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb45-457"><a href="#cb45-457" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral <span class="ot">&lt;-</span> <span class="fu">filter</span>(kraken_reports, taxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-458"><a href="#cb45-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb45-459"><a href="#cb45-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb45-460"><a href="#cb45-460" aria-hidden="true" tabindex="-1"></a>kraken_reports_viral_cleaned <span class="ot">&lt;-</span> kraken_reports_viral <span class="sc">%&gt;%</span></span>
<span id="cb45-461"><a href="#cb45-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-462"><a href="#cb45-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pc_reads_total, <span class="sc">-</span>n_reads_direct) <span class="sc">%&gt;%</span></span>
<span id="cb45-463"><a href="#cb45-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, p_reads_viral, n_reads_clade, <span class="fu">everything</span>())</span>
<span id="cb45-464"><a href="#cb45-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-465"><a href="#cb45-465" aria-hidden="true" tabindex="-1"></a>viral_classes <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"C"</span>)</span>
<span id="cb45-466"><a href="#cb45-466" aria-hidden="true" tabindex="-1"></a>viral_families <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb45-467"><a href="#cb45-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-468"><a href="#cb45-468" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-469"><a href="#cb45-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-472"><a href="#cb45-472" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-473"><a href="#cb45-473" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: viral-class-composition</span></span>
<span id="cb45-474"><a href="#cb45-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-475"><a href="#cb45-475" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify major viral classes</span></span>
<span id="cb45-476"><a href="#cb45-476" aria-hidden="true" tabindex="-1"></a>viral_classes_major_tab <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb45-477"><a href="#cb45-477" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-478"><a href="#cb45-478" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_max =</span> <span class="fu">max</span>(p_reads_viral), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-479"><a href="#cb45-479" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_viral_max <span class="sc">&gt;=</span> <span class="fl">0.04</span>)</span>
<span id="cb45-480"><a href="#cb45-480" aria-hidden="true" tabindex="-1"></a>viral_classes_major_list <span class="ot">&lt;-</span> viral_classes_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb45-481"><a href="#cb45-481" aria-hidden="true" tabindex="-1"></a>viral_classes_major <span class="ot">&lt;-</span> viral_classes <span class="sc">%&gt;%</span> </span>
<span id="cb45-482"><a href="#cb45-482" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> viral_classes_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb45-483"><a href="#cb45-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_viral)</span>
<span id="cb45-484"><a href="#cb45-484" aria-hidden="true" tabindex="-1"></a>viral_classes_minor <span class="ot">&lt;-</span> viral_classes_major <span class="sc">%&gt;%</span> </span>
<span id="cb45-485"><a href="#cb45-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-486"><a href="#cb45-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_viral_major =</span> <span class="fu">sum</span>(p_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-487"><a href="#cb45-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_viral =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_viral_major) <span class="sc">%&gt;%</span></span>
<span id="cb45-488"><a href="#cb45-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_viral)</span>
<span id="cb45-489"><a href="#cb45-489" aria-hidden="true" tabindex="-1"></a>viral_classes_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(viral_classes_major, viral_classes_minor) <span class="sc">%&gt;%</span></span>
<span id="cb45-490"><a href="#cb45-490" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_viral)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-491"><a href="#cb45-491" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(viral_classes_major_list, <span class="st">"Other"</span>)),</span>
<span id="cb45-492"><a href="#cb45-492" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> <span class="fu">pmax</span>(p_reads_viral, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-493"><a href="#cb45-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_viral, <span class="at">classification=</span>name)</span>
<span id="cb45-494"><a href="#cb45-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-495"><a href="#cb45-495" aria-hidden="true" tabindex="-1"></a>palette_viral <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">12</span>, <span class="st">"Set3"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb45-496"><a href="#cb45-496" aria-hidden="true" tabindex="-1"></a>g_classes <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-497"><a href="#cb45-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>viral_classes_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-498"><a href="#cb45-498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Viral Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-499"><a href="#cb45-499" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-500"><a href="#cb45-500" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral class"</span>)</span>
<span id="cb45-501"><a href="#cb45-501" aria-hidden="true" tabindex="-1"></a>g_classes</span>
<span id="cb45-502"><a href="#cb45-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-503"><a href="#cb45-503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-504"><a href="#cb45-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-505"><a href="#cb45-505" aria-hidden="true" tabindex="-1"></a>Of these, the strong presence of *Papovaviricetes* and *Pisoniviricetes* across multiple samples is the most interesting, as both contain important human pathogens. The presence of *Quintoviricetes*, *Herviviricetes* and *Revtraviricetes* in a smaller number of samples is also notable.</span>
<span id="cb45-506"><a href="#cb45-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-507"><a href="#cb45-507" aria-hidden="true" tabindex="-1"></a>In DNA reads, *Papovaviricetes* are quite heterogeneous across samples, while mostly being dominated by one or two genera within samples. The three rooms with the largest *Papovaviricetes* fraction (Female room 1 &amp; 3 and Male room 4) are each dominated by a single genus, specifically *Dyothetapapillomavirus* (of which the sole member is *Dyothetapapillomavirus 1*, a cat pathogen), *Betapapillomavirus* (a genus of human wart-causing viruses), and *Alphapolyomavirus* (specifically Merkel cell polyomavirus).</span>
<span id="cb45-508"><a href="#cb45-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-511"><a href="#cb45-511" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-512"><a href="#cb45-512" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: papovaviricetes</span></span>
<span id="cb45-513"><a href="#cb45-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-514"><a href="#cb45-514" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all read counts in class</span></span>
<span id="cb45-515"><a href="#cb45-515" aria-hidden="true" tabindex="-1"></a>papova_taxid <span class="ot">&lt;-</span> <span class="dv">2732421</span></span>
<span id="cb45-516"><a href="#cb45-516" aria-hidden="true" tabindex="-1"></a>papova_desc_taxids_old <span class="ot">&lt;-</span> papova_taxid</span>
<span id="cb45-517"><a href="#cb45-517" aria-hidden="true" tabindex="-1"></a>papova_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(papova_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> papova_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb45-518"><a href="#cb45-518" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (<span class="fu">length</span>(papova_desc_taxids_new) <span class="sc">&gt;</span> <span class="fu">length</span>(papova_desc_taxids_old)){</span>
<span id="cb45-519"><a href="#cb45-519" aria-hidden="true" tabindex="-1"></a>  papova_desc_taxids_old <span class="ot">&lt;-</span> papova_desc_taxids_new</span>
<span id="cb45-520"><a href="#cb45-520" aria-hidden="true" tabindex="-1"></a>  papova_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(papova_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> papova_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb45-521"><a href="#cb45-521" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb45-522"><a href="#cb45-522" aria-hidden="true" tabindex="-1"></a>papova_counts <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb45-523"><a href="#cb45-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">%in%</span> papova_desc_taxids_new) <span class="sc">%&gt;%</span></span>
<span id="cb45-524"><a href="#cb45-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_papova =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb45-525"><a href="#cb45-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-526"><a href="#cb45-526" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genus composition</span></span>
<span id="cb45-527"><a href="#cb45-527" aria-hidden="true" tabindex="-1"></a>papova_genera <span class="ot">&lt;-</span> papova_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"G"</span>, na_type <span class="sc">==</span> <span class="st">"DNA"</span>)</span>
<span id="cb45-528"><a href="#cb45-528" aria-hidden="true" tabindex="-1"></a>papova_genera_major_tab <span class="ot">&lt;-</span> papova_genera <span class="sc">%&gt;%</span> </span>
<span id="cb45-529"><a href="#cb45-529" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-530"><a href="#cb45-530" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_papova_max =</span> <span class="fu">max</span>(p_reads_papova), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-531"><a href="#cb45-531" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_papova_max <span class="sc">&gt;=</span> <span class="fl">0.04</span>)</span>
<span id="cb45-532"><a href="#cb45-532" aria-hidden="true" tabindex="-1"></a>papova_genera_major_list <span class="ot">&lt;-</span> papova_genera_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb45-533"><a href="#cb45-533" aria-hidden="true" tabindex="-1"></a>papova_genera_major <span class="ot">&lt;-</span> papova_genera <span class="sc">%&gt;%</span> </span>
<span id="cb45-534"><a href="#cb45-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> papova_genera_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb45-535"><a href="#cb45-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_papova)</span>
<span id="cb45-536"><a href="#cb45-536" aria-hidden="true" tabindex="-1"></a>papova_genera_minor <span class="ot">&lt;-</span> papova_genera_major <span class="sc">%&gt;%</span> </span>
<span id="cb45-537"><a href="#cb45-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-538"><a href="#cb45-538" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_papova_major =</span> <span class="fu">sum</span>(p_reads_papova), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-539"><a href="#cb45-539" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_papova =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_papova_major) <span class="sc">%&gt;%</span></span>
<span id="cb45-540"><a href="#cb45-540" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_papova)</span>
<span id="cb45-541"><a href="#cb45-541" aria-hidden="true" tabindex="-1"></a>papova_genera_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(papova_genera_major, papova_genera_minor) <span class="sc">%&gt;%</span></span>
<span id="cb45-542"><a href="#cb45-542" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_papova)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-543"><a href="#cb45-543" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(papova_genera_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb45-544"><a href="#cb45-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_papova, <span class="at">classification=</span>name)</span>
<span id="cb45-545"><a href="#cb45-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-546"><a href="#cb45-546" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-547"><a href="#cb45-547" aria-hidden="true" tabindex="-1"></a>g_papova_genera <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-548"><a href="#cb45-548" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>papova_genera_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-549"><a href="#cb45-549" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Papovaviricetes Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-550"><a href="#cb45-550" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-551"><a href="#cb45-551" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-552"><a href="#cb45-552" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb45-553"><a href="#cb45-553" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb45-554"><a href="#cb45-554" aria-hidden="true" tabindex="-1"></a>g_papova_genera</span>
<span id="cb45-555"><a href="#cb45-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-556"><a href="#cb45-556" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-557"><a href="#cb45-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-558"><a href="#cb45-558" aria-hidden="true" tabindex="-1"></a>In RNA reads, *Pisoniviricetes* are conversely dominated by a single genus, *Sobemovirus*, across all samples, and specifically a single species. Disappointingly, this is Ryegrass mottle virus, a plant pathogen.</span>
<span id="cb45-559"><a href="#cb45-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-562"><a href="#cb45-562" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-563"><a href="#cb45-563" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: pisoniviricetes</span></span>
<span id="cb45-564"><a href="#cb45-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-565"><a href="#cb45-565" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all read counts in class</span></span>
<span id="cb45-566"><a href="#cb45-566" aria-hidden="true" tabindex="-1"></a>pisoni_taxid <span class="ot">&lt;-</span> <span class="dv">2732506</span></span>
<span id="cb45-567"><a href="#cb45-567" aria-hidden="true" tabindex="-1"></a>pisoni_desc_taxids_old <span class="ot">&lt;-</span> pisoni_taxid</span>
<span id="cb45-568"><a href="#cb45-568" aria-hidden="true" tabindex="-1"></a>pisoni_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(pisoni_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> pisoni_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb45-569"><a href="#cb45-569" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (<span class="fu">length</span>(pisoni_desc_taxids_new) <span class="sc">&gt;</span> <span class="fu">length</span>(pisoni_desc_taxids_old)){</span>
<span id="cb45-570"><a href="#cb45-570" aria-hidden="true" tabindex="-1"></a>  pisoni_desc_taxids_old <span class="ot">&lt;-</span> pisoni_desc_taxids_new</span>
<span id="cb45-571"><a href="#cb45-571" aria-hidden="true" tabindex="-1"></a>  pisoni_desc_taxids_new <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(pisoni_desc_taxids_old, viral_taxa <span class="sc">%&gt;%</span> <span class="fu">filter</span>(parent_taxid <span class="sc">%in%</span> pisoni_desc_taxids_old) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(taxid)))</span>
<span id="cb45-572"><a href="#cb45-572" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb45-573"><a href="#cb45-573" aria-hidden="true" tabindex="-1"></a>pisoni_counts <span class="ot">&lt;-</span> kraken_reports_viral_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb45-574"><a href="#cb45-574" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxid <span class="sc">%in%</span> pisoni_desc_taxids_new) <span class="sc">%&gt;%</span></span>
<span id="cb45-575"><a href="#cb45-575" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_pisoni =</span> n_reads_clade<span class="sc">/</span>n_reads_clade[<span class="dv">1</span>])</span>
<span id="cb45-576"><a href="#cb45-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-577"><a href="#cb45-577" aria-hidden="true" tabindex="-1"></a><span class="co"># Get genus composition</span></span>
<span id="cb45-578"><a href="#cb45-578" aria-hidden="true" tabindex="-1"></a>pisoni_genera <span class="ot">&lt;-</span> pisoni_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> <span class="st">"S"</span>, na_type <span class="sc">==</span> <span class="st">"RNA"</span>)</span>
<span id="cb45-579"><a href="#cb45-579" aria-hidden="true" tabindex="-1"></a>pisoni_genera_major_tab <span class="ot">&lt;-</span> pisoni_genera <span class="sc">%&gt;%</span> </span>
<span id="cb45-580"><a href="#cb45-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-581"><a href="#cb45-581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_pisoni_max =</span> <span class="fu">max</span>(p_reads_pisoni), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-582"><a href="#cb45-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_pisoni_max <span class="sc">&gt;=</span> <span class="fl">0.04</span>)</span>
<span id="cb45-583"><a href="#cb45-583" aria-hidden="true" tabindex="-1"></a>pisoni_genera_major_list <span class="ot">&lt;-</span> pisoni_genera_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb45-584"><a href="#cb45-584" aria-hidden="true" tabindex="-1"></a>pisoni_genera_major <span class="ot">&lt;-</span> pisoni_genera <span class="sc">%&gt;%</span> </span>
<span id="cb45-585"><a href="#cb45-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">%in%</span> pisoni_genera_major_list) <span class="sc">%&gt;%</span></span>
<span id="cb45-586"><a href="#cb45-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_pisoni)</span>
<span id="cb45-587"><a href="#cb45-587" aria-hidden="true" tabindex="-1"></a>pisoni_genera_minor <span class="ot">&lt;-</span> pisoni_genera_major <span class="sc">%&gt;%</span> </span>
<span id="cb45-588"><a href="#cb45-588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-589"><a href="#cb45-589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">p_reads_pisoni_major =</span> <span class="fu">sum</span>(p_reads_pisoni), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-590"><a href="#cb45-590" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="st">"Other"</span>, <span class="at">taxid=</span><span class="cn">NA</span>, <span class="at">p_reads_pisoni =</span> <span class="dv">1</span><span class="sc">-</span>p_reads_pisoni_major) <span class="sc">%&gt;%</span></span>
<span id="cb45-591"><a href="#cb45-591" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(name, taxid, sample, room, na_type, sample_type, p_reads_pisoni)</span>
<span id="cb45-592"><a href="#cb45-592" aria-hidden="true" tabindex="-1"></a>pisoni_genera_display <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(pisoni_genera_major, pisoni_genera_minor) <span class="sc">%&gt;%</span></span>
<span id="cb45-593"><a href="#cb45-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_pisoni)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-594"><a href="#cb45-594" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels=</span><span class="fu">c</span>(pisoni_genera_major_list, <span class="st">"Other"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb45-595"><a href="#cb45-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_pisoni, <span class="at">classification=</span>name)</span>
<span id="cb45-596"><a href="#cb45-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-597"><a href="#cb45-597" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-598"><a href="#cb45-598" aria-hidden="true" tabindex="-1"></a>g_pisoni_genera <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-599"><a href="#cb45-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>pisoni_genera_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-600"><a href="#cb45-600" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% Pisoniviricetes Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-601"><a href="#cb45-601" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-602"><a href="#cb45-602" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-603"><a href="#cb45-603" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>) <span class="sc">+</span></span>
<span id="cb45-604"><a href="#cb45-604" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb45-605"><a href="#cb45-605" aria-hidden="true" tabindex="-1"></a>g_pisoni_genera</span>
<span id="cb45-606"><a href="#cb45-606" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-607"><a href="#cb45-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-608"><a href="#cb45-608" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting virus reads: validation</span></span>
<span id="cb45-609"><a href="#cb45-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-610"><a href="#cb45-610" aria-hidden="true" tabindex="-1"></a>Next, I investigated the human-infecting virus read content of these unenriched samples. Using the same workflow I used for Prussin et al, I identified 118 RNA read pairs and 1269 DNA read pairs as putatively human viral: 0.004% and 0.017% of surviving reads, respectively.</span>
<span id="cb45-611"><a href="#cb45-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-614"><a href="#cb45-614" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-615"><a href="#cb45-615" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-read-counts</span></span>
<span id="cb45-616"><a href="#cb45-616" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb45-617"><a href="#cb45-617" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-618"><a href="#cb45-618" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-619"><a href="#cb45-619" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(na_type, sample_type)</span>
<span id="cb45-620"><a href="#cb45-620" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> </span>
<span id="cb45-621"><a href="#cb45-621" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, na_type, sample_type) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb45-622"><a href="#cb45-622" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"ribo_initial"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-623"><a href="#cb45-623" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-624"><a href="#cb45-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> </span>
<span id="cb45-625"><a href="#cb45-625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb45-626"><a href="#cb45-626" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_summ <span class="ot">&lt;-</span> n_hv_filtered <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-627"><a href="#cb45-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_putative =</span> <span class="fu">sum</span>(n_putative), <span class="at">n_total =</span> <span class="fu">sum</span>(n_total), </span>
<span id="cb45-628"><a href="#cb45-628" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-629"><a href="#cb45-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb45-630"><a href="#cb45-630" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-631"><a href="#cb45-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-634"><a href="#cb45-634" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-635"><a href="#cb45-635" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-scores</span></span>
<span id="cb45-636"><a href="#cb45-636" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-637"><a href="#cb45-637" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb45-638"><a href="#cb45-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-639"><a href="#cb45-639" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb45-640"><a href="#cb45-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb45-641"><a href="#cb45-641" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb45-642"><a href="#cb45-642" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb45-643"><a href="#cb45-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, na_type) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb45-644"><a href="#cb45-644" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>(),</span>
<span id="cb45-645"><a href="#cb45-645" aria-hidden="true" tabindex="-1"></a>         <span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb45-646"><a href="#cb45-646" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span id="cb45-647"><a href="#cb45-647" aria-hidden="true" tabindex="-1"></a>g_mrg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev)) <span class="sc">+</span></span>
<span id="cb45-648"><a href="#cb45-648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb45-649"><a href="#cb45-649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-650"><a href="#cb45-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-651"><a href="#cb45-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb45-652"><a href="#cb45-652" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb45-653"><a href="#cb45-653" aria-hidden="true" tabindex="-1"></a>g_mrg</span>
<span id="cb45-654"><a href="#cb45-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-655"><a href="#cb45-655" aria-hidden="true" tabindex="-1"></a>g_hist_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>kraken_label, <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb45-656"><a href="#cb45-656" aria-hidden="true" tabindex="-1"></a>g_hist_0</span>
<span id="cb45-657"><a href="#cb45-657" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-658"><a href="#cb45-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-659"><a href="#cb45-659" aria-hidden="true" tabindex="-1"></a>As previously described, I ran BLASTN on these reads via a dedicated EC2 instance, using the same parameters I've used for previous datasets.</span>
<span id="cb45-660"><a href="#cb45-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-663"><a href="#cb45-663" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-664"><a href="#cb45-664" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: make-blast-fasta</span></span>
<span id="cb45-665"><a href="#cb45-665" aria-hidden="true" tabindex="-1"></a>mrg_fasta <span class="ot">&lt;-</span>  mrg <span class="sc">%&gt;%</span></span>
<span id="cb45-666"><a href="#cb45-666" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, seq_id)) <span class="sc">%&gt;%</span></span>
<span id="cb45-667"><a href="#cb45-667" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb45-668"><a href="#cb45-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, </span>
<span id="cb45-669"><a href="#cb45-669" aria-hidden="true" tabindex="-1"></a>         <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb45-670"><a href="#cb45-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb45-671"><a href="#cb45-671" aria-hidden="true" tabindex="-1"></a>mrg_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb45-672"><a href="#cb45-672" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mrg_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"blast/putative-viral.fasta"</span>))</span>
<span id="cb45-673"><a href="#cb45-673" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-674"><a href="#cb45-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-677"><a href="#cb45-677" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-678"><a href="#cb45-678" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: process-blast-data</span></span>
<span id="cb45-679"><a href="#cb45-679" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-680"><a href="#cb45-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-681"><a href="#cb45-681" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb45-682"><a href="#cb45-682" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"blast/putative-viral.blast.gz"</span>)</span>
<span id="cb45-683"><a href="#cb45-683" aria-hidden="true" tabindex="-1"></a>blast_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span>)</span>
<span id="cb45-684"><a href="#cb45-684" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>,</span>
<span id="cb45-685"><a href="#cb45-685" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col_names =</span> blast_cols, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb45-686"><a href="#cb45-686" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results_path &lt;- file.path(data_dir, "blast/putative-viral-best.blast.gz")</span></span>
<span id="cb45-687"><a href="#cb45-687" aria-hidden="true" tabindex="-1"></a><span class="co"># blast_results &lt;- read_tsv(blast_results_path, show_col_types = FALSE)</span></span>
<span id="cb45-688"><a href="#cb45-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-689"><a href="#cb45-689" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for best hit for each query/subject combination</span></span>
<span id="cb45-690"><a href="#cb45-690" aria-hidden="true" tabindex="-1"></a>blast_results_best <span class="ot">&lt;-</span> blast_results <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> </span>
<span id="cb45-691"><a href="#cb45-691" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb45-692"><a href="#cb45-692" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb45-693"><a href="#cb45-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-694"><a href="#cb45-694" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each query and filter for high-ranking hits</span></span>
<span id="cb45-695"><a href="#cb45-695" aria-hidden="true" tabindex="-1"></a>blast_results_ranked <span class="ot">&lt;-</span> blast_results_best <span class="sc">%&gt;%</span> </span>
<span id="cb45-696"><a href="#cb45-696" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb45-697"><a href="#cb45-697" aria-hidden="true" tabindex="-1"></a>blast_results_highrank <span class="ot">&lt;-</span> blast_results_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-698"><a href="#cb45-698" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">read_pair =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>), </span>
<span id="cb45-699"><a href="#cb45-699" aria-hidden="true" tabindex="-1"></a>         <span class="at">seq_id =</span> <span class="fu">str_split</span>(qseqid, <span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">sapply</span>(nth, <span class="at">n=</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-700"><a href="#cb45-700" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore))</span>
<span id="cb45-701"><a href="#cb45-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-702"><a href="#cb45-702" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb45-703"><a href="#cb45-703" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb45-704"><a href="#cb45-704" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-705"><a href="#cb45-705" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore),</span>
<span id="cb45-706"><a href="#cb45-706" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-707"><a href="#cb45-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-708"><a href="#cb45-708" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb45-709"><a href="#cb45-709" aria-hidden="true" tabindex="-1"></a>blast_results_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_results_paired, <span class="at">viral =</span> staxid <span class="sc">%in%</span> viral_taxa<span class="sc">$</span>taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-710"><a href="#cb45-710" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb45-711"><a href="#cb45-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-712"><a href="#cb45-712" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb45-713"><a href="#cb45-713" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_id, taxid, assigned_taxid, adj_score_max, na_type)</span>
<span id="cb45-714"><a href="#cb45-714" aria-hidden="true" tabindex="-1"></a>blast_results_assign <span class="ot">&lt;-</span> <span class="fu">left_join</span>(blast_results_viral, mrg_assign, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-715"><a href="#cb45-715" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb45-716"><a href="#cb45-716" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb45-717"><a href="#cb45-717" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb45-718"><a href="#cb45-718" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_assign <span class="sc">%&gt;%</span></span>
<span id="cb45-719"><a href="#cb45-719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(seq_id) <span class="sc">%&gt;%</span></span>
<span id="cb45-720"><a href="#cb45-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb45-721"><a href="#cb45-721" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb45-722"><a href="#cb45-722" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb45-723"><a href="#cb45-723" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-724"><a href="#cb45-724" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-725"><a href="#cb45-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-728"><a href="#cb45-728" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-729"><a href="#cb45-729" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-results</span></span>
<span id="cb45-730"><a href="#cb45-730" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb45-731"><a href="#cb45-731" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-732"><a href="#cb45-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-733"><a href="#cb45-733" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge BLAST results with unenriched read data</span></span>
<span id="cb45-734"><a href="#cb45-734" aria-hidden="true" tabindex="-1"></a>mrg_blast <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg, blast_results_out, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-735"><a href="#cb45-735" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>),</span>
<span id="cb45-736"><a href="#cb45-736" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb45-737"><a href="#cb45-737" aria-hidden="true" tabindex="-1"></a>mrg_blast_rna <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(na_type <span class="sc">==</span> <span class="st">"RNA"</span>)</span>
<span id="cb45-738"><a href="#cb45-738" aria-hidden="true" tabindex="-1"></a>mrg_blast_dna <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(na_type <span class="sc">==</span> <span class="st">"DNA"</span>)</span>
<span id="cb45-739"><a href="#cb45-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-740"><a href="#cb45-740" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot RNA</span></span>
<span id="cb45-741"><a href="#cb45-741" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_rna <span class="ot">&lt;-</span> mrg_blast_rna <span class="sc">%&gt;%</span></span>
<span id="cb45-742"><a href="#cb45-742" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb45-743"><a href="#cb45-743" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb45-744"><a href="#cb45-744" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-745"><a href="#cb45-745" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-746"><a href="#cb45-746" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb45-747"><a href="#cb45-747" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(viral_status_out<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb45-748"><a href="#cb45-748" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"RNA"</span>) <span class="sc">+</span> </span>
<span id="cb45-749"><a href="#cb45-749" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="dv">2</span>), <span class="at">hjust=</span><span class="dv">0</span>))</span>
<span id="cb45-750"><a href="#cb45-750" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_rna</span>
<span id="cb45-751"><a href="#cb45-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-752"><a href="#cb45-752" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot DNA</span></span>
<span id="cb45-753"><a href="#cb45-753" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_dna <span class="ot">&lt;-</span> mrg_blast_dna <span class="sc">%&gt;%</span></span>
<span id="cb45-754"><a href="#cb45-754" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb45-755"><a href="#cb45-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb45-756"><a href="#cb45-756" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-757"><a href="#cb45-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">65</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-758"><a href="#cb45-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb45-759"><a href="#cb45-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(viral_status_out<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb45-760"><a href="#cb45-760" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"DNA"</span>) <span class="sc">+</span> </span>
<span id="cb45-761"><a href="#cb45-761" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="dv">2</span>), <span class="at">hjust=</span><span class="dv">0</span>))</span>
<span id="cb45-762"><a href="#cb45-762" aria-hidden="true" tabindex="-1"></a>g_mrg_blast_dna</span>
<span id="cb45-763"><a href="#cb45-763" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-764"><a href="#cb45-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-767"><a href="#cb45-767" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-768"><a href="#cb45-768" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-blast-histogram</span></span>
<span id="cb45-769"><a href="#cb45-769" aria-hidden="true" tabindex="-1"></a>g_hist_1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_blast, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>,<span class="at">position=</span><span class="st">"dodge"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>viral_status_out, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span> <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span> theme_base <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">20</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"red"</span>)</span>
<span id="cb45-770"><a href="#cb45-770" aria-hidden="true" tabindex="-1"></a>g_hist_1</span>
<span id="cb45-771"><a href="#cb45-771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-772"><a href="#cb45-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-773"><a href="#cb45-773" aria-hidden="true" tabindex="-1"></a>These results look good on visual inspection, and indeed precision and sensitivity are both very high. For a disjunctive score threshold of 20, my updated workflow achieves an F1 score of 98.2% for RNA sequences and 97.9% for DNA sequences.</span>
<span id="cb45-774"><a href="#cb45-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-777"><a href="#cb45-777" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-778"><a href="#cb45-778" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-f1</span></span>
<span id="cb45-779"><a href="#cb45-779" aria-hidden="true" tabindex="-1"></a>test_sens_spec <span class="ot">&lt;-</span> <span class="cf">function</span>(tab, score_threshold, conjunctive, include_special){</span>
<span id="cb45-780"><a href="#cb45-780" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>include_special) tab <span class="ot">&lt;-</span> <span class="fu">filter</span>(tab, viral_status_out <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>))</span>
<span id="cb45-781"><a href="#cb45-781" aria-hidden="true" tabindex="-1"></a>  tab_retained <span class="ot">&lt;-</span> tab <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb45-782"><a href="#cb45-782" aria-hidden="true" tabindex="-1"></a>    <span class="at">conjunctive =</span> conjunctive,</span>
<span id="cb45-783"><a href="#cb45-783" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_conjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">&amp;</span> adj_score_rev <span class="sc">&gt;</span> score_threshold), </span>
<span id="cb45-784"><a href="#cb45-784" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score_disjunctive =</span> (adj_score_fwd <span class="sc">&gt;</span> score_threshold <span class="sc">|</span> adj_score_rev <span class="sc">&gt;</span> score_threshold),</span>
<span id="cb45-785"><a href="#cb45-785" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain_score =</span> <span class="fu">ifelse</span>(conjunctive, retain_score_conjunctive, retain_score_disjunctive),</span>
<span id="cb45-786"><a href="#cb45-786" aria-hidden="true" tabindex="-1"></a>    <span class="at">retain =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> retain_score) <span class="sc">%&gt;%</span></span>
<span id="cb45-787"><a href="#cb45-787" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(viral_status_out, retain) <span class="sc">%&gt;%</span> count</span>
<span id="cb45-788"><a href="#cb45-788" aria-hidden="true" tabindex="-1"></a>  pos_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb45-789"><a href="#cb45-789" aria-hidden="true" tabindex="-1"></a>  pos_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb45-790"><a href="#cb45-790" aria-hidden="true" tabindex="-1"></a>  neg_tru <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">!=</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb45-791"><a href="#cb45-791" aria-hidden="true" tabindex="-1"></a>  neg_fls <span class="ot">&lt;-</span> tab_retained <span class="sc">%&gt;%</span> <span class="fu">filter</span>(viral_status_out <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="sc">!</span>retain) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(n) <span class="sc">%&gt;%</span> sum</span>
<span id="cb45-792"><a href="#cb45-792" aria-hidden="true" tabindex="-1"></a>  sensitivity <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> neg_fls)</span>
<span id="cb45-793"><a href="#cb45-793" aria-hidden="true" tabindex="-1"></a>  specificity <span class="ot">&lt;-</span> neg_tru <span class="sc">/</span> (neg_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb45-794"><a href="#cb45-794" aria-hidden="true" tabindex="-1"></a>  precision   <span class="ot">&lt;-</span> pos_tru <span class="sc">/</span> (pos_tru <span class="sc">+</span> pos_fls)</span>
<span id="cb45-795"><a href="#cb45-795" aria-hidden="true" tabindex="-1"></a>  f1 <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)</span>
<span id="cb45-796"><a href="#cb45-796" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">threshold=</span>score_threshold, <span class="at">include_special =</span> include_special, </span>
<span id="cb45-797"><a href="#cb45-797" aria-hidden="true" tabindex="-1"></a>                <span class="at">conjunctive =</span> conjunctive, <span class="at">sensitivity=</span>sensitivity, </span>
<span id="cb45-798"><a href="#cb45-798" aria-hidden="true" tabindex="-1"></a>                <span class="at">specificity=</span>specificity, <span class="at">precision=</span>precision, <span class="at">f1=</span>f1)</span>
<span id="cb45-799"><a href="#cb45-799" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb45-800"><a href="#cb45-800" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb45-801"><a href="#cb45-801" aria-hidden="true" tabindex="-1"></a>range_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(intab, inc_special, <span class="at">inrange=</span><span class="dv">15</span><span class="sc">:</span><span class="dv">45</span>){</span>
<span id="cb45-802"><a href="#cb45-802" aria-hidden="true" tabindex="-1"></a>  tss <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">partial</span>(test_sens_spec, <span class="at">tab=</span>intab, <span class="at">include_special=</span>inc_special)</span>
<span id="cb45-803"><a href="#cb45-803" aria-hidden="true" tabindex="-1"></a>  stats_conj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb45-804"><a href="#cb45-804" aria-hidden="true" tabindex="-1"></a>  stats_disj <span class="ot">&lt;-</span> <span class="fu">lapply</span>(inrange, tss, <span class="at">conjunctive=</span><span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> bind_rows</span>
<span id="cb45-805"><a href="#cb45-805" aria-hidden="true" tabindex="-1"></a>  stats_all <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_conj, stats_disj) <span class="sc">%&gt;%</span></span>
<span id="cb45-806"><a href="#cb45-806" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span>(threshold<span class="sc">:</span>conjunctive), <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-807"><a href="#cb45-807" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">conj_label =</span> <span class="fu">ifelse</span>(conjunctive, <span class="st">"Conjunctive"</span>, <span class="st">"Disjunctive"</span>))</span>
<span id="cb45-808"><a href="#cb45-808" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(stats_all)</span>
<span id="cb45-809"><a href="#cb45-809" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb45-810"><a href="#cb45-810" aria-hidden="true" tabindex="-1"></a>inc_special <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb45-811"><a href="#cb45-811" aria-hidden="true" tabindex="-1"></a>stats_rna <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast_rna, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">na_type =</span> <span class="st">"RNA"</span>)</span>
<span id="cb45-812"><a href="#cb45-812" aria-hidden="true" tabindex="-1"></a>stats_dna <span class="ot">&lt;-</span> <span class="fu">range_f1</span>(mrg_blast_dna, inc_special) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">na_type =</span> <span class="st">"DNA"</span>)</span>
<span id="cb45-813"><a href="#cb45-813" aria-hidden="true" tabindex="-1"></a>stats_0 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(stats_rna, stats_dna) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">attempt=</span><span class="dv">0</span>)</span>
<span id="cb45-814"><a href="#cb45-814" aria-hidden="true" tabindex="-1"></a>threshold_opt_0 <span class="ot">&lt;-</span> stats_0 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(conj_label,attempt,na_type) <span class="sc">%&gt;%</span> </span>
<span id="cb45-815"><a href="#cb45-815" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(metric <span class="sc">==</span> <span class="st">"f1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-816"><a href="#cb45-816" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(value <span class="sc">==</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(threshold <span class="sc">==</span> <span class="fu">min</span>(threshold))</span>
<span id="cb45-817"><a href="#cb45-817" aria-hidden="true" tabindex="-1"></a>g_stats_0 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(stats_0, <span class="fu">aes</span>(<span class="at">x=</span>threshold, <span class="at">y=</span>value, <span class="at">color=</span>metric)) <span class="sc">+</span></span>
<span id="cb45-818"><a href="#cb45-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">data =</span> threshold_opt_0, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">xintercept=</span>threshold),</span>
<span id="cb45-819"><a href="#cb45-819" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb45-820"><a href="#cb45-820" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb45-821"><a href="#cb45-821" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Value"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-822"><a href="#cb45-822" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Threshold"</span>, <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-823"><a href="#cb45-823" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set3"</span>) <span class="sc">+</span></span>
<span id="cb45-824"><a href="#cb45-824" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>conj_label) <span class="sc">+</span></span>
<span id="cb45-825"><a href="#cb45-825" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb45-826"><a href="#cb45-826" aria-hidden="true" tabindex="-1"></a>g_stats_0</span>
<span id="cb45-827"><a href="#cb45-827" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-828"><a href="#cb45-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-829"><a href="#cb45-829" aria-hidden="true" tabindex="-1"></a>Looking into the composition of different read groups, the most notable observations for me are the predominance of (1) HIV reads among low-scoring RNA "true-positives", and (2) human mastadenovirus C among RNA true-positives in general:</span>
<span id="cb45-830"><a href="#cb45-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-833"><a href="#cb45-833" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-834"><a href="#cb45-834" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fp</span></span>
<span id="cb45-835"><a href="#cb45-835" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb45-836"><a href="#cb45-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-837"><a href="#cb45-837" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">211787</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus type 92"</span></span>
<span id="cb45-838"><a href="#cb45-838" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">509154</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine endogenous retrovirus C"</span></span>
<span id="cb45-839"><a href="#cb45-839" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">493803</span>] <span class="ot">&lt;-</span> <span class="st">"Merkel cell polyomavirus"</span></span>
<span id="cb45-840"><a href="#cb45-840" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">427343</span>] <span class="ot">&lt;-</span> <span class="st">"Human papillomavirus 107"</span></span>
<span id="cb45-841"><a href="#cb45-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-842"><a href="#cb45-842" aria-hidden="true" tabindex="-1"></a>major_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb45-843"><a href="#cb45-843" aria-hidden="true" tabindex="-1"></a>fp <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> </span>
<span id="cb45-844"><a href="#cb45-844" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, viral_status_out, </span>
<span id="cb45-845"><a href="#cb45-845" aria-hidden="true" tabindex="-1"></a>           <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, taxid) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> </span>
<span id="cb45-846"><a href="#cb45-846" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, viral_status_out, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-847"><a href="#cb45-847" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-848"><a href="#cb45-848" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p)) <span class="sc">%&gt;%</span></span>
<span id="cb45-849"><a href="#cb45-849" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name =</span> <span class="fu">ifelse</span>(taxid <span class="sc">==</span> <span class="dv">194958</span>, <span class="st">"Porcine endogenous retrovirus A"</span>, name))</span>
<span id="cb45-850"><a href="#cb45-850" aria-hidden="true" tabindex="-1"></a>fp_major_tab <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(p))</span>
<span id="cb45-851"><a href="#cb45-851" aria-hidden="true" tabindex="-1"></a>fp_major_list <span class="ot">&lt;-</span> fp_major_tab <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name) <span class="sc">%&gt;%</span> sort <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> <span class="fu">c</span>(., <span class="st">"Other"</span>)</span>
<span id="cb45-852"><a href="#cb45-852" aria-hidden="true" tabindex="-1"></a>fp_major <span class="ot">&lt;-</span> fp <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">major =</span> p <span class="sc">&gt;</span> major_threshold) <span class="sc">%&gt;%</span> </span>
<span id="cb45-853"><a href="#cb45-853" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(major, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-854"><a href="#cb45-854" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, viral_status_out, highscore, name_display) <span class="sc">%&gt;%</span> </span>
<span id="cb45-855"><a href="#cb45-855" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n=</span><span class="fu">sum</span>(n), <span class="at">p=</span><span class="fu">sum</span>(p), <span class="at">.groups =</span> <span class="st">"drop"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb45-856"><a href="#cb45-856" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, <span class="at">levels =</span> fp_major_list),</span>
<span id="cb45-857"><a href="#cb45-857" aria-hidden="true" tabindex="-1"></a>         <span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb45-858"><a href="#cb45-858" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb45-859"><a href="#cb45-859" aria-hidden="true" tabindex="-1"></a>g_fp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_major, <span class="fu">aes</span>(<span class="at">x=</span>score_display, <span class="at">y=</span>p, <span class="at">fill=</span>name_display)) <span class="sc">+</span></span>
<span id="cb45-860"><a href="#cb45-860" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-861"><a href="#cb45-861" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"True positive?"</span>) <span class="sc">+</span></span>
<span id="cb45-862"><a href="#cb45-862" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"% reads"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-863"><a href="#cb45-863" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-864"><a href="#cb45-864" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette_viral, <span class="at">name =</span> <span class="st">"Viral</span><span class="sc">\n</span><span class="st">taxon"</span>) <span class="sc">+</span></span>
<span id="cb45-865"><a href="#cb45-865" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>status_display) <span class="sc">+</span></span>
<span id="cb45-866"><a href="#cb45-866" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb45-867"><a href="#cb45-867" aria-hidden="true" tabindex="-1"></a>  theme_kit</span>
<span id="cb45-868"><a href="#cb45-868" aria-hidden="true" tabindex="-1"></a>g_fp</span>
<span id="cb45-869"><a href="#cb45-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-870"><a href="#cb45-870" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-871"><a href="#cb45-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-872"><a href="#cb45-872" aria-hidden="true" tabindex="-1"></a>There are only eight low-scoring true-positive RNA reads, three of which are HIV. These do seem to be real; at least, they don't BLAST to anything else. Ironically the eight high-scoring "true-positive" HIV reads seem more likely to be fake; they also map to a range of cloning vectors:</span>
<span id="cb45-873"><a href="#cb45-873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-876"><a href="#cb45-876" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-877"><a href="#cb45-877" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hiv-blast-hits</span></span>
<span id="cb45-878"><a href="#cb45-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-879"><a href="#cb45-879" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb45-880"><a href="#cb45-880" aria-hidden="true" tabindex="-1"></a>ref_taxid_hiv <span class="ot">&lt;-</span> <span class="dv">11676</span></span>
<span id="cb45-881"><a href="#cb45-881" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.3</span></span>
<span id="cb45-882"><a href="#cb45-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-883"><a href="#cb45-883" aria-hidden="true" tabindex="-1"></a><span class="co"># Get taxon names</span></span>
<span id="cb45-884"><a href="#cb45-884" aria-hidden="true" tabindex="-1"></a>tax_names_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"taxid-names.tsv.gz"</span>)</span>
<span id="cb45-885"><a href="#cb45-885" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(tax_names_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-886"><a href="#cb45-886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-887"><a href="#cb45-887" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb45-888"><a href="#cb45-888" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb45-889"><a href="#cb45-889" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb45-890"><a href="#cb45-890" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>)</span>
<span id="cb45-891"><a href="#cb45-891" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(tax_names, tax_names_new)</span>
<span id="cb45-892"><a href="#cb45-892" aria-hidden="true" tabindex="-1"></a>ref_name_hiv <span class="ot">&lt;-</span> tax_names <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> ref_taxid_hiv) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb45-893"><a href="#cb45-893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-894"><a href="#cb45-894" aria-hidden="true" tabindex="-1"></a><span class="co"># Get major matches</span></span>
<span id="cb45-895"><a href="#cb45-895" aria-hidden="true" tabindex="-1"></a>fp_staxid <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> ref_taxid_hiv) <span class="sc">%&gt;%</span></span>
<span id="cb45-896"><a href="#cb45-896" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb45-897"><a href="#cb45-897" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-898"><a href="#cb45-898" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-899"><a href="#cb45-899" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-900"><a href="#cb45-900" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid=</span>staxid), <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-901"><a href="#cb45-901" aria-hidden="true" tabindex="-1"></a>fp_staxid_count <span class="ot">&lt;-</span> fp_staxid <span class="sc">%&gt;%</span></span>
<span id="cb45-902"><a href="#cb45-902" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, na_type, </span>
<span id="cb45-903"><a href="#cb45-903" aria-hidden="true" tabindex="-1"></a>           taxid, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb45-904"><a href="#cb45-904" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span></span>
<span id="cb45-905"><a href="#cb45-905" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, na_type, taxid, name) <span class="sc">%&gt;%</span></span>
<span id="cb45-906"><a href="#cb45-906" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb45-907"><a href="#cb45-907" aria-hidden="true" tabindex="-1"></a>fp_staxid_count_major <span class="ot">&lt;-</span> fp_staxid_count <span class="sc">%&gt;%</span></span>
<span id="cb45-908"><a href="#cb45-908" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n<span class="sc">&gt;</span><span class="dv">1</span>, p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-909"><a href="#cb45-909" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb45-910"><a href="#cb45-910" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, </span>
<span id="cb45-911"><a href="#cb45-911" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb45-912"><a href="#cb45-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-913"><a href="#cb45-913" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-914"><a href="#cb45-914" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_staxid_count_major, <span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span> </span>
<span id="cb45-915"><a href="#cb45-915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb45-916"><a href="#cb45-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>status_display<span class="sc">+</span>score_display, <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb45-917"><a href="#cb45-917" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb45-918"><a href="#cb45-918" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-919"><a href="#cb45-919" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-920"><a href="#cb45-920" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_hiv, <span class="st">" (taxid "</span>, ref_taxid_hiv, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb45-921"><a href="#cb45-921" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-922"><a href="#cb45-922" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-923"><a href="#cb45-923" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb45-924"><a href="#cb45-924" aria-hidden="true" tabindex="-1"></a>g</span>
<span id="cb45-925"><a href="#cb45-925" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-926"><a href="#cb45-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-927"><a href="#cb45-927" aria-hidden="true" tabindex="-1"></a>A similar pattern is seen for the Human mastadenovirus C hits: low-scoring true-positives only map to adenoviruses, but high-scoring ones also map to a range of cloning vectors.</span>
<span id="cb45-928"><a href="#cb45-928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-931"><a href="#cb45-931" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-932"><a href="#cb45-932" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: masta-blast-hits</span></span>
<span id="cb45-933"><a href="#cb45-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-934"><a href="#cb45-934" aria-hidden="true" tabindex="-1"></a><span class="co"># Configure</span></span>
<span id="cb45-935"><a href="#cb45-935" aria-hidden="true" tabindex="-1"></a>ref_taxid_masta <span class="ot">&lt;-</span> <span class="dv">129951</span></span>
<span id="cb45-936"><a href="#cb45-936" aria-hidden="true" tabindex="-1"></a>p_threshold <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb45-937"><a href="#cb45-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-938"><a href="#cb45-938" aria-hidden="true" tabindex="-1"></a><span class="co"># Add missing names</span></span>
<span id="cb45-939"><a href="#cb45-939" aria-hidden="true" tabindex="-1"></a>tax_names_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>staxid, <span class="sc">~</span>name,</span>
<span id="cb45-940"><a href="#cb45-940" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3050295</span>, <span class="st">"Cytomegalovirus humanbeta5"</span>,</span>
<span id="cb45-941"><a href="#cb45-941" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">459231</span>, <span class="st">"FLAG-tagging vector pFLAG97-TSR"</span>,</span>
<span id="cb45-942"><a href="#cb45-942" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3033760</span>, <span class="st">"Human adenovirus 89"</span>,</span>
<span id="cb45-943"><a href="#cb45-943" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3043599</span>, <span class="st">"Human adenovirus 108"</span>,</span>
<span id="cb45-944"><a href="#cb45-944" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3088344</span>, <span class="st">"Human adenovirus C108"</span>,</span>
<span id="cb45-945"><a href="#cb45-945" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3102992</span>, <span class="st">"Cloning vector pBWH-C5"</span>,</span>
<span id="cb45-946"><a href="#cb45-946" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3102994</span>, <span class="st">"Cloning vector pBWH-C5-pIX-Kan"</span>,</span>
<span id="cb45-947"><a href="#cb45-947" aria-hidden="true" tabindex="-1"></a>                         <span class="dv">3102995</span>, <span class="st">"Cloning vector pBWH-C5-pIXZG"</span></span>
<span id="cb45-948"><a href="#cb45-948" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb45-949"><a href="#cb45-949" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(tax_names, tax_names_new)</span>
<span id="cb45-950"><a href="#cb45-950" aria-hidden="true" tabindex="-1"></a>ref_name_masta <span class="ot">&lt;-</span> tax_names <span class="sc">%&gt;%</span> <span class="fu">filter</span>(staxid <span class="sc">==</span> ref_taxid_masta) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(name)</span>
<span id="cb45-951"><a href="#cb45-951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-952"><a href="#cb45-952" aria-hidden="true" tabindex="-1"></a><span class="co"># Get major matches</span></span>
<span id="cb45-953"><a href="#cb45-953" aria-hidden="true" tabindex="-1"></a>fp_staxid <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> ref_taxid_masta) <span class="sc">%&gt;%</span></span>
<span id="cb45-954"><a href="#cb45-954" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_seq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb45-955"><a href="#cb45-955" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-956"><a href="#cb45-956" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-957"><a href="#cb45-957" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-958"><a href="#cb45-958" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid=</span>staxid), <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-959"><a href="#cb45-959" aria-hidden="true" tabindex="-1"></a>fp_staxid_count <span class="ot">&lt;-</span> fp_staxid <span class="sc">%&gt;%</span></span>
<span id="cb45-960"><a href="#cb45-960" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, na_type, </span>
<span id="cb45-961"><a href="#cb45-961" aria-hidden="true" tabindex="-1"></a>           taxid, name, staxid, sname, n_seq) <span class="sc">%&gt;%</span></span>
<span id="cb45-962"><a href="#cb45-962" aria-hidden="true" tabindex="-1"></a>  count <span class="sc">%&gt;%</span></span>
<span id="cb45-963"><a href="#cb45-963" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, highscore, na_type, taxid, name) <span class="sc">%&gt;%</span></span>
<span id="cb45-964"><a href="#cb45-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span>n_seq)</span>
<span id="cb45-965"><a href="#cb45-965" aria-hidden="true" tabindex="-1"></a>fp_staxid_count_major <span class="ot">&lt;-</span> fp_staxid_count <span class="sc">%&gt;%</span></span>
<span id="cb45-966"><a href="#cb45-966" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p<span class="sc">&gt;</span>p_threshold, <span class="sc">!</span><span class="fu">is.na</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-967"><a href="#cb45-967" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb45-968"><a href="#cb45-968" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out, </span>
<span id="cb45-969"><a href="#cb45-969" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb45-970"><a href="#cb45-970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-971"><a href="#cb45-971" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-972"><a href="#cb45-972" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_staxid_count_major, <span class="fu">aes</span>(<span class="at">x=</span>p, <span class="at">y=</span>sname)) <span class="sc">+</span> </span>
<span id="cb45-973"><a href="#cb45-973" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span> </span>
<span id="cb45-974"><a href="#cb45-974" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>status_display<span class="sc">+</span>score_display, <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb45-975"><a href="#cb45-975" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">label_wrap_gen</span>(<span class="at">multi_line =</span> <span class="cn">FALSE</span>)) <span class="sc">+</span></span>
<span id="cb45-976"><a href="#cb45-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% mapped reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-977"><a href="#cb45-977" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-978"><a href="#cb45-978" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_masta, <span class="st">" (taxid "</span>, ref_taxid_masta, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb45-979"><a href="#cb45-979" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-980"><a href="#cb45-980" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-981"><a href="#cb45-981" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.5</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb45-982"><a href="#cb45-982" aria-hidden="true" tabindex="-1"></a>g</span>
<span id="cb45-983"><a href="#cb45-983" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-984"><a href="#cb45-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-985"><a href="#cb45-985" aria-hidden="true" tabindex="-1"></a>The plots above only tell us the total number of reads that BLAST to each cloning vector; they don't tell us how many reads map to any vector. In particular, they can't distinguish between a situation where many reads map to a few cloning vectors, and one where a few reads map to many. This turns out to be about 35% of high-scoring HIV RNA reads and 55% of high-scoring mastadenovirus RNA reads:</span>
<span id="cb45-986"><a href="#cb45-986" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-989"><a href="#cb45-989" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-990"><a href="#cb45-990" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hiv-vector-hits</span></span>
<span id="cb45-991"><a href="#cb45-991" aria-hidden="true" tabindex="-1"></a>fp_hiv_vec <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> ref_taxid_hiv) <span class="sc">%&gt;%</span></span>
<span id="cb45-992"><a href="#cb45-992" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb45-993"><a href="#cb45-993" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-994"><a href="#cb45-994" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-995"><a href="#cb45-995" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid=</span>staxid), <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-996"><a href="#cb45-996" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="at">vector =</span> <span class="fu">grepl</span>(<span class="st">"vector"</span>, sname))</span>
<span id="cb45-997"><a href="#cb45-997" aria-hidden="true" tabindex="-1"></a>fp_hiv_vec_match <span class="ot">&lt;-</span> fp_hiv_vec <span class="sc">%&gt;%</span></span>
<span id="cb45-998"><a href="#cb45-998" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, seq_id, na_type, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-999"><a href="#cb45-999" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">vector_match =</span> <span class="fu">any</span>(vector), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-1000"><a href="#cb45-1000" aria-hidden="true" tabindex="-1"></a>fp_hiv_vec_count <span class="ot">&lt;-</span> fp_hiv_vec_match <span class="sc">%&gt;%</span></span>
<span id="cb45-1001"><a href="#cb45-1001" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, na_type, highscore, vector_match) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb45-1002"><a href="#cb45-1002" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, na_type, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1003"><a href="#cb45-1003" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb45-1004"><a href="#cb45-1004" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out,</span>
<span id="cb45-1005"><a href="#cb45-1005" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb45-1006"><a href="#cb45-1006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1007"><a href="#cb45-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1008"><a href="#cb45-1008" aria-hidden="true" tabindex="-1"></a>g_hiv_vec_count <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_hiv_vec_count, <span class="fu">aes</span>(<span class="at">y=</span>score_display, <span class="at">x=</span>p, <span class="at">fill=</span>vector_match)) <span class="sc">+</span></span>
<span id="cb45-1009"><a href="#cb45-1009" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb45-1010"><a href="#cb45-1010" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% Mapped Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-1011"><a href="#cb45-1011" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-1012"><a href="#cb45-1012" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">name=</span><span class="st">"Maps to vector?"</span>, <span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb45-1013"><a href="#cb45-1013" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>status_display, <span class="at">scales=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb45-1014"><a href="#cb45-1014" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_hiv, <span class="st">" (taxid "</span>, ref_taxid_hiv, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb45-1015"><a href="#cb45-1015" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-1016"><a href="#cb45-1016" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-1017"><a href="#cb45-1017" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb45-1018"><a href="#cb45-1018" aria-hidden="true" tabindex="-1"></a>g_hiv_vec_count</span>
<span id="cb45-1019"><a href="#cb45-1019" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1020"><a href="#cb45-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1023"><a href="#cb45-1023" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1024"><a href="#cb45-1024" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: masta-vector-hits</span></span>
<span id="cb45-1025"><a href="#cb45-1025" aria-hidden="true" tabindex="-1"></a>fp_masta_vec <span class="ot">&lt;-</span> mrg_blast <span class="sc">%&gt;%</span> <span class="fu">filter</span>(taxid <span class="sc">==</span> ref_taxid_masta) <span class="sc">%&gt;%</span></span>
<span id="cb45-1026"><a href="#cb45-1026" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(blast_results_paired, <span class="at">by=</span><span class="st">"seq_id"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb45-1027"><a href="#cb45-1027" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">staxid =</span> <span class="fu">as.integer</span>(staxid)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1028"><a href="#cb45-1028" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names, <span class="at">by=</span><span class="st">"staxid"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">sname=</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-1029"><a href="#cb45-1029" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(tax_names <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">taxid=</span>staxid), <span class="at">by=</span><span class="st">"taxid"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1030"><a href="#cb45-1030" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="at">vector =</span> <span class="fu">grepl</span>(<span class="st">"vector"</span>, sname))</span>
<span id="cb45-1031"><a href="#cb45-1031" aria-hidden="true" tabindex="-1"></a>fp_masta_vec_match <span class="ot">&lt;-</span> fp_masta_vec <span class="sc">%&gt;%</span></span>
<span id="cb45-1032"><a href="#cb45-1032" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, seq_id, na_type, <span class="at">highscore =</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1033"><a href="#cb45-1033" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">vector_match =</span> <span class="fu">any</span>(vector), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb45-1034"><a href="#cb45-1034" aria-hidden="true" tabindex="-1"></a>fp_masta_vec_count <span class="ot">&lt;-</span> fp_masta_vec_match <span class="sc">%&gt;%</span></span>
<span id="cb45-1035"><a href="#cb45-1035" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, na_type, highscore, vector_match) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span></span>
<span id="cb45-1036"><a href="#cb45-1036" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(viral_status_out, na_type, highscore) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p=</span>n<span class="sc">/</span><span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1037"><a href="#cb45-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">score_display =</span> <span class="fu">ifelse</span>(highscore, <span class="st">"S &gt;= 20"</span>, <span class="st">"S &lt; 20"</span>),</span>
<span id="cb45-1038"><a href="#cb45-1038" aria-hidden="true" tabindex="-1"></a>         <span class="at">status_display =</span> <span class="fu">ifelse</span>(viral_status_out,</span>
<span id="cb45-1039"><a href="#cb45-1039" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"True positive"</span>, <span class="st">"False positive"</span>))</span>
<span id="cb45-1040"><a href="#cb45-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1041"><a href="#cb45-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1042"><a href="#cb45-1042" aria-hidden="true" tabindex="-1"></a>g_masta_vec_count <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fp_masta_vec_count, <span class="fu">aes</span>(<span class="at">y=</span>score_display, <span class="at">x=</span>p, <span class="at">fill=</span>vector_match)) <span class="sc">+</span></span>
<span id="cb45-1043"><a href="#cb45-1043" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb45-1044"><a href="#cb45-1044" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"% Mapped Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-1045"><a href="#cb45-1045" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb45-1046"><a href="#cb45-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">name=</span><span class="st">"Maps to vector?"</span>, <span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb45-1047"><a href="#cb45-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(na_type<span class="sc">~</span>status_display, <span class="at">scales=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb45-1048"><a href="#cb45-1048" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(ref_name_masta, <span class="st">" (taxid "</span>, ref_taxid_masta, <span class="st">")"</span>)) <span class="sc">+</span></span>
<span id="cb45-1049"><a href="#cb45-1049" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb45-1050"><a href="#cb45-1050" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-1051"><a href="#cb45-1051" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">1.4</span>), <span class="at">hjust=</span><span class="dv">0</span>, <span class="at">face=</span><span class="st">"plain"</span>))</span>
<span id="cb45-1052"><a href="#cb45-1052" aria-hidden="true" tabindex="-1"></a>g_masta_vec_count</span>
<span id="cb45-1053"><a href="#cb45-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1054"><a href="#cb45-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1055"><a href="#cb45-1055" aria-hidden="true" tabindex="-1"></a>This is more concerning for the mastadenovirus reads, which make up a nontrivial fraction of all high-scoring RNA "true-positives". As in my last post, though, it's hard to know what to take from this; many vectors are based on lentiviral or adenoviral backbones, and it's not always easy to distinguish between true virus and virus-based vector. I'm going to stick with my pipeline as it is for now, but it's worth keeping this in mind in interpreting the results.</span>
<span id="cb45-1056"><a href="#cb45-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1057"><a href="#cb45-1057" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: overall relative abundance</span></span>
<span id="cb45-1058"><a href="#cb45-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1061"><a href="#cb45-1061" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1062"><a href="#cb45-1062" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: count-hv-reads</span></span>
<span id="cb45-1063"><a href="#cb45-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1064"><a href="#cb45-1064" aria-hidden="true" tabindex="-1"></a><span class="co"># Get raw read counts</span></span>
<span id="cb45-1065"><a href="#cb45-1065" aria-hidden="true" tabindex="-1"></a>read_counts_raw <span class="ot">&lt;-</span> basic_stats_raw <span class="sc">%&gt;%</span></span>
<span id="cb45-1066"><a href="#cb45-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, room, na_type, sample_type, <span class="at">n_reads_raw =</span> n_read_pairs)</span>
<span id="cb45-1067"><a href="#cb45-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1068"><a href="#cb45-1068" aria-hidden="true" tabindex="-1"></a><span class="co"># Get HV read counts</span></span>
<span id="cb45-1069"><a href="#cb45-1069" aria-hidden="true" tabindex="-1"></a>mrg_hv <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">hv_status =</span> assigned_hv <span class="sc">|</span> hit_hv <span class="sc">|</span> adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>)</span>
<span id="cb45-1070"><a href="#cb45-1070" aria-hidden="true" tabindex="-1"></a>read_counts_hv <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">filter</span>(hv_status) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1071"><a href="#cb45-1071" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_hv"</span>)</span>
<span id="cb45-1072"><a href="#cb45-1072" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> read_counts_raw <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(read_counts_hv, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1073"><a href="#cb45-1073" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_reads_hv =</span> <span class="fu">replace_na</span>(n_reads_hv, <span class="dv">0</span>))</span>
<span id="cb45-1074"><a href="#cb45-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1075"><a href="#cb45-1075" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate</span></span>
<span id="cb45-1076"><a href="#cb45-1076" aria-hidden="true" tabindex="-1"></a>read_counts_total <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(na_type, sample_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-1077"><a href="#cb45-1077" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">n</span>() <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1078"><a href="#cb45-1078" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb45-1079"><a href="#cb45-1079" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1080"><a href="#cb45-1080" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span> <span class="st">"All samples"</span>, <span class="at">room =</span> <span class="st">"All rooms"</span>)</span>
<span id="cb45-1081"><a href="#cb45-1081" aria-hidden="true" tabindex="-1"></a>read_counts_agg <span class="ot">&lt;-</span> read_counts <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb45-1082"><a href="#cb45-1082" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(read_counts_total) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1083"><a href="#cb45-1083" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample),</span>
<span id="cb45-1084"><a href="#cb45-1084" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span>n_reads_raw)</span>
<span id="cb45-1085"><a href="#cb45-1085" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1086"><a href="#cb45-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1087"><a href="#cb45-1087" aria-hidden="true" tabindex="-1"></a>In non-control samples, applying a disjunctive cutoff at S=20 identifies 97 RNA reads and 1204 DNA reads as human-viral. This gives an overall relative HV abundance of $1.21 \times 10^{-5}$ for RNA reads and $1.50 \times 10^{-4}$ for DNA reads. For DNA reads, HV RA in the negative controls is about 10x lower than the non-control samples; however, for RNA reads, the HV RA in the negative controls is actually higher than in the non-controls. This, combined with the very low absolute read count, suggests that we shouldn't take the HV results for RNA reads too seriously.</span>
<span id="cb45-1088"><a href="#cb45-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1091"><a href="#cb45-1091" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1092"><a href="#cb45-1092" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot-hv-ra</span></span>
<span id="cb45-1093"><a href="#cb45-1093" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb45-1094"><a href="#cb45-1094" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize</span></span>
<span id="cb45-1095"><a href="#cb45-1095" aria-hidden="true" tabindex="-1"></a>g_phv_agg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(read_counts_agg, <span class="fu">aes</span>(<span class="at">x=</span>room, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb45-1096"><a href="#cb45-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>p_reads_hv)) <span class="sc">+</span></span>
<span id="cb45-1097"><a href="#cb45-1097" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb45-1098"><a href="#cb45-1098" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">"Collection Date"</span>) <span class="sc">+</span></span>
<span id="cb45-1099"><a href="#cb45-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>sample_type, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb45-1100"><a href="#cb45-1100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span> theme_rotate</span>
<span id="cb45-1101"><a href="#cb45-1101" aria-hidden="true" tabindex="-1"></a>g_phv_agg</span>
<span id="cb45-1102"><a href="#cb45-1102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1103"><a href="#cb45-1103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1106"><a href="#cb45-1106" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1107"><a href="#cb45-1107" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-hv-past</span></span>
<span id="cb45-1108"><a href="#cb45-1108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1109"><a href="#cb45-1109" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate past RA values</span></span>
<span id="cb45-1110"><a href="#cb45-1110" aria-hidden="true" tabindex="-1"></a>ra_past <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb45-1111"><a href="#cb45-1111" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">5e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1112"><a href="#cb45-1112" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Brumfield"</span>, <span class="fl">3.66e-7</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1113"><a href="#cb45-1113" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Spurbeck"</span>, <span class="fl">5.44e-6</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1114"><a href="#cb45-1114" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Yang"</span>, <span class="fl">3.62e-4</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1115"><a href="#cb45-1115" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (unenriched)"</span>, <span class="fl">1.87e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1116"><a href="#cb45-1116" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Rothman (panel-enriched)"</span>, <span class="fl">3.3e-5</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb45-1117"><a href="#cb45-1117" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (unenriched)"</span>, <span class="fl">1.37e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1118"><a href="#cb45-1118" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Crits-Christoph (panel-enriched)"</span>, <span class="fl">1.26e-2</span>, <span class="st">"RNA"</span>, <span class="cn">TRUE</span>,</span>
<span id="cb45-1119"><a href="#cb45-1119" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">1.63e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1120"><a href="#cb45-1120" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Prussin (non-control)"</span>, <span class="fl">4.16e-5</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span></span>
<span id="cb45-1121"><a href="#cb45-1121" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb45-1122"><a href="#cb45-1122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1123"><a href="#cb45-1123" aria-hidden="true" tabindex="-1"></a><span class="co"># Collate new RA values</span></span>
<span id="cb45-1124"><a href="#cb45-1124" aria-hidden="true" tabindex="-1"></a>ra_new <span class="ot">&lt;-</span> <span class="fu">tribble</span>(<span class="sc">~</span>dataset, <span class="sc">~</span>ra, <span class="sc">~</span>na_type, <span class="sc">~</span>panel_enriched,</span>
<span id="cb45-1125"><a href="#cb45-1125" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.21e-5</span>, <span class="st">"RNA"</span>, <span class="cn">FALSE</span>,</span>
<span id="cb45-1126"><a href="#cb45-1126" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"Rosario (non-control)"</span>, <span class="fl">1.50e-4</span>, <span class="st">"DNA"</span>, <span class="cn">FALSE</span>)</span>
<span id="cb45-1127"><a href="#cb45-1127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1128"><a href="#cb45-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1129"><a href="#cb45-1129" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-1130"><a href="#cb45-1130" aria-hidden="true" tabindex="-1"></a>ra_comp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(ra_past, ra_new) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">dataset =</span> <span class="fu">fct_inorder</span>(dataset))</span>
<span id="cb45-1131"><a href="#cb45-1131" aria-hidden="true" tabindex="-1"></a>g_ra_comp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ra_comp, <span class="fu">aes</span>(<span class="at">y=</span>dataset, <span class="at">x=</span>ra, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb45-1132"><a href="#cb45-1132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb45-1133"><a href="#cb45-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb45-1134"><a href="#cb45-1134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance of human virus reads"</span>) <span class="sc">+</span></span>
<span id="cb45-1135"><a href="#cb45-1135" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb45-1136"><a href="#cb45-1136" aria-hidden="true" tabindex="-1"></a>g_ra_comp</span>
<span id="cb45-1137"><a href="#cb45-1137" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1138"><a href="#cb45-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1139"><a href="#cb45-1139" aria-hidden="true" tabindex="-1"></a><span class="fu"># Human-infecting viruses: taxonomy and composition</span></span>
<span id="cb45-1140"><a href="#cb45-1140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1141"><a href="#cb45-1141" aria-hidden="true" tabindex="-1"></a>I'm going to focus on the DNA reads here, since as discussed above I expect the RNA reads to be noisy and not very informative – which seems to in fact be the case.</span>
<span id="cb45-1142"><a href="#cb45-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1143"><a href="#cb45-1143" aria-hidden="true" tabindex="-1"></a>At the family level, as in Prussin, we see that *Papillomaviridae*, and *Polyomaviridae* dominate DNA reads in most samples, with *Poxviridae* and *Genomoviridae* showing significant presence in several samples. The first of these is primarily *Betapapillomavirus* and *Gammapapillomavirus*, with some *Alphapapillomavirus*; the second is primarily *Alphapolyomavirus* and *Deltapolyomavirus*. All of these are made up of several different viruses at the species level. *Genomoviridae* is represented primarily by *Gemykibivirus*, while *Poxviridae* is represented by small numbers of reads from several species, including vaccinia, cowpox and tanapox. (There's also one variola read, which would be alarming, but BLASTN thinks it isn't real.)</span>
<span id="cb45-1144"><a href="#cb45-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1145"><a href="#cb45-1145" aria-hidden="true" tabindex="-1"></a>Compared to Prussin, the most striking difference is the absence of herpesvirus reads; whereas HCMV was dominant in the Prussin data, it barely shows up here except in the controls.</span>
<span id="cb45-1146"><a href="#cb45-1146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1149"><a href="#cb45-1149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1150"><a href="#cb45-1150" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: raise-hv-taxa</span></span>
<span id="cb45-1151"><a href="#cb45-1151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1152"><a href="#cb45-1152" aria-hidden="true" tabindex="-1"></a><span class="co"># Get viral taxon names for putative HV reads</span></span>
<span id="cb45-1153"><a href="#cb45-1153" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">249588</span>] <span class="ot">&lt;-</span> <span class="st">"Mamastrovirus"</span></span>
<span id="cb45-1154"><a href="#cb45-1154" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">194960</span>] <span class="ot">&lt;-</span> <span class="st">"Kobuvirus"</span></span>
<span id="cb45-1155"><a href="#cb45-1155" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">688449</span>] <span class="ot">&lt;-</span> <span class="st">"Salivirus"</span></span>
<span id="cb45-1156"><a href="#cb45-1156" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">585893</span>] <span class="ot">&lt;-</span> <span class="st">"Picobirnaviridae"</span></span>
<span id="cb45-1157"><a href="#cb45-1157" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333922</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus"</span></span>
<span id="cb45-1158"><a href="#cb45-1158" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334207</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 3"</span></span>
<span id="cb45-1159"><a href="#cb45-1159" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">369960</span>] <span class="ot">&lt;-</span> <span class="st">"Porcine type-C oncovirus"</span></span>
<span id="cb45-1160"><a href="#cb45-1160" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333924</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 2"</span></span>
<span id="cb45-1161"><a href="#cb45-1161" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">687329</span>] <span class="ot">&lt;-</span> <span class="st">"Anelloviridae"</span></span>
<span id="cb45-1162"><a href="#cb45-1162" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">325455</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus"</span></span>
<span id="cb45-1163"><a href="#cb45-1163" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333750</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus"</span></span>
<span id="cb45-1164"><a href="#cb45-1164" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694002</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus"</span></span>
<span id="cb45-1165"><a href="#cb45-1165" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334202</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus"</span></span>
<span id="cb45-1166"><a href="#cb45-1166" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">197911</span>] <span class="ot">&lt;-</span> <span class="st">"Alphainfluenzavirus"</span></span>
<span id="cb45-1167"><a href="#cb45-1167" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">186938</span>] <span class="ot">&lt;-</span> <span class="st">"Respirovirus"</span></span>
<span id="cb45-1168"><a href="#cb45-1168" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333926</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 1"</span></span>
<span id="cb45-1169"><a href="#cb45-1169" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337051</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 1"</span></span>
<span id="cb45-1170"><a href="#cb45-1170" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337043</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 4"</span></span>
<span id="cb45-1171"><a href="#cb45-1171" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">694003</span>] <span class="ot">&lt;-</span> <span class="st">"Betacoronavirus 1"</span></span>
<span id="cb45-1172"><a href="#cb45-1172" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334204</span>] <span class="ot">&lt;-</span> <span class="st">"Mupapillomavirus 2"</span></span>
<span id="cb45-1173"><a href="#cb45-1173" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">334208</span>] <span class="ot">&lt;-</span> <span class="st">"Betapapillomavirus 4"</span></span>
<span id="cb45-1174"><a href="#cb45-1174" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">333928</span>] <span class="ot">&lt;-</span> <span class="st">"Gammapapillomavirus 2"</span></span>
<span id="cb45-1175"><a href="#cb45-1175" aria-hidden="true" tabindex="-1"></a>viral_taxa<span class="sc">$</span>name[viral_taxa<span class="sc">$</span>taxid <span class="sc">==</span> <span class="dv">337039</span>] <span class="ot">&lt;-</span> <span class="st">"Alphapapillomavirus 2"</span></span>
<span id="cb45-1176"><a href="#cb45-1176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1177"><a href="#cb45-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1178"><a href="#cb45-1178" aria-hidden="true" tabindex="-1"></a>mrg_hv_named <span class="ot">&lt;-</span> mrg_hv <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-1179"><a href="#cb45-1179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1180"><a href="#cb45-1180" aria-hidden="true" tabindex="-1"></a><span class="co"># Discover viral species &amp; genera for HV reads</span></span>
<span id="cb45-1181"><a href="#cb45-1181" aria-hidden="true" tabindex="-1"></a>raise_rank <span class="ot">&lt;-</span> <span class="cf">function</span>(read_db, taxid_db, <span class="at">out_rank =</span> <span class="st">"species"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>){</span>
<span id="cb45-1182"><a href="#cb45-1182" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get higher ranks than search rank</span></span>
<span id="cb45-1183"><a href="#cb45-1183" aria-hidden="true" tabindex="-1"></a>  ranks <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subspecies"</span>, <span class="st">"species"</span>, <span class="st">"subgenus"</span>, <span class="st">"genus"</span>, <span class="st">"subfamily"</span>, <span class="st">"family"</span>, <span class="st">"suborder"</span>, <span class="st">"order"</span>, <span class="st">"class"</span>, <span class="st">"subphylum"</span>, <span class="st">"phylum"</span>, <span class="st">"kingdom"</span>, <span class="st">"superkingdom"</span>)</span>
<span id="cb45-1184"><a href="#cb45-1184" aria-hidden="true" tabindex="-1"></a>  rank_match <span class="ot">&lt;-</span> <span class="fu">which.max</span>(ranks <span class="sc">==</span> out_rank)</span>
<span id="cb45-1185"><a href="#cb45-1185" aria-hidden="true" tabindex="-1"></a>  high_ranks <span class="ot">&lt;-</span> ranks[rank_match<span class="sc">:</span><span class="fu">length</span>(ranks)]</span>
<span id="cb45-1186"><a href="#cb45-1186" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge read DB and taxid DB</span></span>
<span id="cb45-1187"><a href="#cb45-1187" aria-hidden="true" tabindex="-1"></a>  reads <span class="ot">&lt;-</span> read_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-1188"><a href="#cb45-1188" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-1189"><a href="#cb45-1189" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sequences that are already at appropriate rank</span></span>
<span id="cb45-1190"><a href="#cb45-1190" aria-hidden="true" tabindex="-1"></a>  reads_rank <span class="ot">&lt;-</span> <span class="fu">filter</span>(reads, rank <span class="sc">==</span> out_rank)</span>
<span id="cb45-1191"><a href="#cb45-1191" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop sequences at a higher rank and return unclassified sequences</span></span>
<span id="cb45-1192"><a href="#cb45-1192" aria-hidden="true" tabindex="-1"></a>  reads_norank <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb45-1193"><a href="#cb45-1193" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">nrow</span>(reads_norank) <span class="sc">&gt;</span> <span class="dv">0</span>){ <span class="co"># As long as there are unclassified sequences...</span></span>
<span id="cb45-1194"><a href="#cb45-1194" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Promote read taxids and re-merge with taxid DB, then re-classify and filter</span></span>
<span id="cb45-1195"><a href="#cb45-1195" aria-hidden="true" tabindex="-1"></a>    reads_remaining <span class="ot">&lt;-</span> reads_norank <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">taxid =</span> parent_taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-1196"><a href="#cb45-1196" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-1197"><a href="#cb45-1197" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-1198"><a href="#cb45-1198" aria-hidden="true" tabindex="-1"></a>    reads_rank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">==</span> out_rank) <span class="sc">%&gt;%</span></span>
<span id="cb45-1199"><a href="#cb45-1199" aria-hidden="true" tabindex="-1"></a>      <span class="fu">bind_rows</span>(reads_rank)</span>
<span id="cb45-1200"><a href="#cb45-1200" aria-hidden="true" tabindex="-1"></a>    reads_norank <span class="ot">&lt;-</span> reads_remaining <span class="sc">%&gt;%</span></span>
<span id="cb45-1201"><a href="#cb45-1201" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(rank <span class="sc">!=</span> out_rank, <span class="sc">!</span>rank <span class="sc">%in%</span> high_ranks, <span class="sc">!</span><span class="fu">is.na</span>(taxid))</span>
<span id="cb45-1202"><a href="#cb45-1202" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb45-1203"><a href="#cb45-1203" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Finally, extract and append reads that were excluded during the process</span></span>
<span id="cb45-1204"><a href="#cb45-1204" aria-hidden="true" tabindex="-1"></a>  reads_dropped <span class="ot">&lt;-</span> reads <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>seq_id <span class="sc">%in%</span> reads_rank<span class="sc">$</span>seq_id)</span>
<span id="cb45-1205"><a href="#cb45-1205" aria-hidden="true" tabindex="-1"></a>  reads_out <span class="ot">&lt;-</span> reads_rank <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>(reads_dropped) <span class="sc">%&gt;%</span></span>
<span id="cb45-1206"><a href="#cb45-1206" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>parent_taxid, <span class="sc">-</span>rank, <span class="sc">-</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb45-1207"><a href="#cb45-1207" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(taxid_db, <span class="at">by=</span><span class="st">"taxid"</span>)</span>
<span id="cb45-1208"><a href="#cb45-1208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(reads_out)</span>
<span id="cb45-1209"><a href="#cb45-1209" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb45-1210"><a href="#cb45-1210" aria-hidden="true" tabindex="-1"></a>hv_reads_species <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"species"</span>)</span>
<span id="cb45-1211"><a href="#cb45-1211" aria-hidden="true" tabindex="-1"></a>hv_reads_genus <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"genus"</span>)</span>
<span id="cb45-1212"><a href="#cb45-1212" aria-hidden="true" tabindex="-1"></a>hv_reads_family <span class="ot">&lt;-</span> <span class="fu">raise_rank</span>(mrg_hv_named, viral_taxa, <span class="st">"family"</span>)</span>
<span id="cb45-1213"><a href="#cb45-1213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1214"><a href="#cb45-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1217"><a href="#cb45-1217" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1218"><a href="#cb45-1218" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-family</span></span>
<span id="cb45-1219"><a href="#cb45-1219" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb45-1220"><a href="#cb45-1220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1221"><a href="#cb45-1221" aria-hidden="true" tabindex="-1"></a>threshold_major_family <span class="ot">&lt;-</span> <span class="fl">0.06</span></span>
<span id="cb45-1222"><a href="#cb45-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1223"><a href="#cb45-1223" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral family</span></span>
<span id="cb45-1224"><a href="#cb45-1224" aria-hidden="true" tabindex="-1"></a>hv_family_counts <span class="ot">&lt;-</span> hv_reads_family <span class="sc">%&gt;%</span> </span>
<span id="cb45-1225"><a href="#cb45-1225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-1226"><a href="#cb45-1226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1227"><a href="#cb45-1227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-1228"><a href="#cb45-1228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb45-1229"><a href="#cb45-1229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1230"><a href="#cb45-1230" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb45-1231"><a href="#cb45-1231" aria-hidden="true" tabindex="-1"></a>hv_family_major_tab <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1232"><a href="#cb45-1232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1233"><a href="#cb45-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_family)</span>
<span id="cb45-1234"><a href="#cb45-1234" aria-hidden="true" tabindex="-1"></a>hv_family_counts_major <span class="ot">&lt;-</span> hv_family_counts <span class="sc">%&gt;%</span></span>
<span id="cb45-1235"><a href="#cb45-1235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_family_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1236"><a href="#cb45-1236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb45-1237"><a href="#cb45-1237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb45-1238"><a href="#cb45-1238" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1239"><a href="#cb45-1239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb45-1240"><a href="#cb45-1240" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_family_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb45-1241"><a href="#cb45-1241" aria-hidden="true" tabindex="-1"></a>hv_family_counts_display <span class="ot">&lt;-</span> hv_family_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb45-1242"><a href="#cb45-1242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb45-1243"><a href="#cb45-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1244"><a href="#cb45-1244" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-1245"><a href="#cb45-1245" aria-hidden="true" tabindex="-1"></a>g_hv_family <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-1246"><a href="#cb45-1246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_family_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-1247"><a href="#cb45-1247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-1248"><a href="#cb45-1248" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-1249"><a href="#cb45-1249" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-1250"><a href="#cb45-1250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral family"</span>)</span>
<span id="cb45-1251"><a href="#cb45-1251" aria-hidden="true" tabindex="-1"></a>g_hv_family</span>
<span id="cb45-1252"><a href="#cb45-1252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1253"><a href="#cb45-1253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1256"><a href="#cb45-1256" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1257"><a href="#cb45-1257" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-genus</span></span>
<span id="cb45-1258"><a href="#cb45-1258" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb45-1259"><a href="#cb45-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1260"><a href="#cb45-1260" aria-hidden="true" tabindex="-1"></a>threshold_major_genus <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb45-1261"><a href="#cb45-1261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1262"><a href="#cb45-1262" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral genus</span></span>
<span id="cb45-1263"><a href="#cb45-1263" aria-hidden="true" tabindex="-1"></a>hv_genus_counts <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> <span class="fu">filter</span>(na_type <span class="sc">==</span> <span class="st">"DNA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1264"><a href="#cb45-1264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-1265"><a href="#cb45-1265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1266"><a href="#cb45-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-1267"><a href="#cb45-1267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb45-1268"><a href="#cb45-1268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1269"><a href="#cb45-1269" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb45-1270"><a href="#cb45-1270" aria-hidden="true" tabindex="-1"></a>hv_genus_major_tab <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1271"><a href="#cb45-1271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1272"><a href="#cb45-1272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_genus)</span>
<span id="cb45-1273"><a href="#cb45-1273" aria-hidden="true" tabindex="-1"></a>hv_genus_counts_major <span class="ot">&lt;-</span> hv_genus_counts <span class="sc">%&gt;%</span></span>
<span id="cb45-1274"><a href="#cb45-1274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_genus_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1275"><a href="#cb45-1275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb45-1276"><a href="#cb45-1276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb45-1277"><a href="#cb45-1277" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1278"><a href="#cb45-1278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb45-1279"><a href="#cb45-1279" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_genus_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb45-1280"><a href="#cb45-1280" aria-hidden="true" tabindex="-1"></a>hv_genus_counts_display <span class="ot">&lt;-</span> hv_genus_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb45-1281"><a href="#cb45-1281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb45-1282"><a href="#cb45-1282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1283"><a href="#cb45-1283" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-1284"><a href="#cb45-1284" aria-hidden="true" tabindex="-1"></a>g_hv_genus <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-1285"><a href="#cb45-1285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_genus_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-1286"><a href="#cb45-1286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-1287"><a href="#cb45-1287" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-1288"><a href="#cb45-1288" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-1289"><a href="#cb45-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral genus"</span>)</span>
<span id="cb45-1290"><a href="#cb45-1290" aria-hidden="true" tabindex="-1"></a>g_hv_genus</span>
<span id="cb45-1291"><a href="#cb45-1291" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1292"><a href="#cb45-1292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1295"><a href="#cb45-1295" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1296"><a href="#cb45-1296" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: hv-species</span></span>
<span id="cb45-1297"><a href="#cb45-1297" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb45-1298"><a href="#cb45-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1299"><a href="#cb45-1299" aria-hidden="true" tabindex="-1"></a>threshold_major_species <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb45-1300"><a href="#cb45-1300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1301"><a href="#cb45-1301" aria-hidden="true" tabindex="-1"></a><span class="co"># Count reads for each human-viral species</span></span>
<span id="cb45-1302"><a href="#cb45-1302" aria-hidden="true" tabindex="-1"></a>hv_species_counts <span class="ot">&lt;-</span> hv_reads_species <span class="sc">%&gt;%</span> <span class="fu">filter</span>(na_type <span class="sc">==</span> <span class="st">"DNA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1303"><a href="#cb45-1303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name, taxid) <span class="sc">%&gt;%</span></span>
<span id="cb45-1304"><a href="#cb45-1304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name =</span> <span class="st">"n_reads_hv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1305"><a href="#cb45-1305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-1306"><a href="#cb45-1306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_hv =</span> n_reads_hv<span class="sc">/</span><span class="fu">sum</span>(n_reads_hv))</span>
<span id="cb45-1307"><a href="#cb45-1307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1308"><a href="#cb45-1308" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify high-ranking families and group others</span></span>
<span id="cb45-1309"><a href="#cb45-1309" aria-hidden="true" tabindex="-1"></a>hv_species_major_tab <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(name) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1310"><a href="#cb45-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_reads_hv <span class="sc">==</span> <span class="fu">max</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1311"><a href="#cb45-1311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(p_reads_hv)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_reads_hv <span class="sc">&gt;</span> threshold_major_species)</span>
<span id="cb45-1312"><a href="#cb45-1312" aria-hidden="true" tabindex="-1"></a>hv_species_counts_major <span class="ot">&lt;-</span> hv_species_counts <span class="sc">%&gt;%</span></span>
<span id="cb45-1313"><a href="#cb45-1313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">ifelse</span>(name <span class="sc">%in%</span> hv_species_major_tab<span class="sc">$</span>name, name, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1314"><a href="#cb45-1314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, sample_type, na_type, name_display) <span class="sc">%&gt;%</span></span>
<span id="cb45-1315"><a href="#cb45-1315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_hv =</span> <span class="fu">sum</span>(n_reads_hv), <span class="at">p_reads_hv =</span> <span class="fu">sum</span>(p_reads_hv), </span>
<span id="cb45-1316"><a href="#cb45-1316" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1317"><a href="#cb45-1317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_display =</span> <span class="fu">factor</span>(name_display, </span>
<span id="cb45-1318"><a href="#cb45-1318" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(hv_species_major_tab<span class="sc">$</span>name, <span class="st">"Other"</span>)))</span>
<span id="cb45-1319"><a href="#cb45-1319" aria-hidden="true" tabindex="-1"></a>hv_species_counts_display <span class="ot">&lt;-</span> hv_species_counts_major <span class="sc">%&gt;%</span></span>
<span id="cb45-1320"><a href="#cb45-1320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">p_reads =</span> p_reads_hv, <span class="at">classification =</span> name_display)</span>
<span id="cb45-1321"><a href="#cb45-1321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1322"><a href="#cb45-1322" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-1323"><a href="#cb45-1323" aria-hidden="true" tabindex="-1"></a>g_hv_species <span class="ot">&lt;-</span> g_comp_base <span class="sc">+</span> </span>
<span id="cb45-1324"><a href="#cb45-1324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">data=</span>hv_species_counts_display, <span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb45-1325"><a href="#cb45-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"% HV Reads"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.01</span>), </span>
<span id="cb45-1326"><a href="#cb45-1326" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.2</span>),</span>
<span id="cb45-1327"><a href="#cb45-1327" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">labels =</span> <span class="cf">function</span>(y) y<span class="sc">*</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-1328"><a href="#cb45-1328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>palette_viral, <span class="at">name =</span> <span class="st">"Viral species"</span>) <span class="sc">+</span></span>
<span id="cb45-1329"><a href="#cb45-1329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">3</span>))</span>
<span id="cb45-1330"><a href="#cb45-1330" aria-hidden="true" tabindex="-1"></a>g_hv_species</span>
<span id="cb45-1331"><a href="#cb45-1331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1332"><a href="#cb45-1332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1333"><a href="#cb45-1333" aria-hidden="true" tabindex="-1"></a>Finally, here again are the overall relative abundances of the specific viral genera I picked out manually in my last entry:</span>
<span id="cb45-1334"><a href="#cb45-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1337"><a href="#cb45-1337" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb45-1338"><a href="#cb45-1338" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 5</span></span>
<span id="cb45-1339"><a href="#cb45-1339" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ra-genera</span></span>
<span id="cb45-1340"><a href="#cb45-1340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1341"><a href="#cb45-1341" aria-hidden="true" tabindex="-1"></a><span class="co"># Define reference genera</span></span>
<span id="cb45-1342"><a href="#cb45-1342" aria-hidden="true" tabindex="-1"></a>path_genera_rna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mamastrovirus"</span>, <span class="st">"Enterovirus"</span>, <span class="st">"Salivirus"</span>, <span class="st">"Kobuvirus"</span>, <span class="st">"Norovirus"</span>, <span class="st">"Sapovirus"</span>, <span class="st">"Rotavirus"</span>, <span class="st">"Alphacoronavirus"</span>, <span class="st">"Betacoronavirus"</span>, <span class="st">"Alphainfluenzavirus"</span>, <span class="st">"Betainfluenzavirus"</span>, <span class="st">"Lentivirus"</span>)</span>
<span id="cb45-1343"><a href="#cb45-1343" aria-hidden="true" tabindex="-1"></a>path_genera_dna <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mastadenovirus"</span>, <span class="st">"Alphapolyomavirus"</span>, <span class="st">"Betapolyomavirus"</span>, <span class="st">"Alphapapillomavirus"</span>, <span class="st">"Betapapillomavirus"</span>, <span class="st">"Gammapapillomavirus"</span>, <span class="st">"Orthopoxvirus"</span>, <span class="st">"Simplexvirus"</span>,</span>
<span id="cb45-1344"><a href="#cb45-1344" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Lymphocryptovirus"</span>, <span class="st">"Cytomegalovirus"</span>, <span class="st">"Dependoparvovirus"</span>)</span>
<span id="cb45-1345"><a href="#cb45-1345" aria-hidden="true" tabindex="-1"></a>path_genera <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">tibble</span>(<span class="at">name=</span>path_genera_rna, <span class="at">genome_type=</span><span class="st">"RNA genome"</span>),</span>
<span id="cb45-1346"><a href="#cb45-1346" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">tibble</span>(<span class="at">name=</span>path_genera_dna, <span class="at">genome_type=</span><span class="st">"DNA genome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1347"><a href="#cb45-1347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(viral_taxa, <span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb45-1348"><a href="#cb45-1348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1349"><a href="#cb45-1349" aria-hidden="true" tabindex="-1"></a><span class="co"># Count in each sample</span></span>
<span id="cb45-1350"><a href="#cb45-1350" aria-hidden="true" tabindex="-1"></a>n_path_genera <span class="ot">&lt;-</span> hv_reads_genus <span class="sc">%&gt;%</span> </span>
<span id="cb45-1351"><a href="#cb45-1351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, room, na_type, sample_type, name, taxid) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1352"><a href="#cb45-1352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(<span class="at">name=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-1353"><a href="#cb45-1353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(path_genera, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"name"</span>, <span class="st">"taxid"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1354"><a href="#cb45-1354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"room"</span>, <span class="st">"na_type"</span>, <span class="st">"sample_type"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-1355"><a href="#cb45-1355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb45-1356"><a href="#cb45-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1357"><a href="#cb45-1357" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot out and back to add zero lines</span></span>
<span id="cb45-1358"><a href="#cb45-1358" aria-hidden="true" tabindex="-1"></a>n_path_genera_out <span class="ot">&lt;-</span> n_path_genera <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, name, n_reads_viral) <span class="sc">%&gt;%</span></span>
<span id="cb45-1359"><a href="#cb45-1359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"name"</span>, <span class="at">values_from=</span><span class="st">"n_reads_viral"</span>, <span class="at">values_fill=</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1360"><a href="#cb45-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to=</span><span class="st">"name"</span>, <span class="at">values_to=</span><span class="st">"n_reads_viral"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1361"><a href="#cb45-1361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(read_counts_raw, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1362"><a href="#cb45-1362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(path_genera, <span class="at">by=</span><span class="st">"name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1363"><a href="#cb45-1363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb45-1364"><a href="#cb45-1364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1365"><a href="#cb45-1365" aria-hidden="true" tabindex="-1"></a><span class="do">## Aggregate across dates</span></span>
<span id="cb45-1366"><a href="#cb45-1366" aria-hidden="true" tabindex="-1"></a>n_path_genera_stype <span class="ot">&lt;-</span> n_path_genera_out <span class="sc">%&gt;%</span> </span>
<span id="cb45-1367"><a href="#cb45-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(na_type, sample_type,</span>
<span id="cb45-1368"><a href="#cb45-1368" aria-hidden="true" tabindex="-1"></a>           name, taxid, genome_type) <span class="sc">%&gt;%</span></span>
<span id="cb45-1369"><a href="#cb45-1369" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_reads_raw =</span> <span class="fu">sum</span>(n_reads_raw),</span>
<span id="cb45-1370"><a href="#cb45-1370" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads_viral =</span> <span class="fu">sum</span>(n_reads_viral), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-1371"><a href="#cb45-1371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample=</span><span class="st">"All samples"</span>, <span class="at">date=</span><span class="st">"All dates"</span>,</span>
<span id="cb45-1372"><a href="#cb45-1372" aria-hidden="true" tabindex="-1"></a>         <span class="at">p_reads_viral =</span> n_reads_viral<span class="sc">/</span>n_reads_raw)</span>
<span id="cb45-1373"><a href="#cb45-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1374"><a href="#cb45-1374" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb45-1375"><a href="#cb45-1375" aria-hidden="true" tabindex="-1"></a>g_path_genera <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(n_path_genera_stype,</span>
<span id="cb45-1376"><a href="#cb45-1376" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">y=</span>name, <span class="at">x=</span>p_reads_viral, <span class="at">color=</span>na_type)) <span class="sc">+</span></span>
<span id="cb45-1377"><a href="#cb45-1377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb45-1378"><a href="#cb45-1378" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>(<span class="at">name=</span><span class="st">"Relative abundance"</span>) <span class="sc">+</span></span>
<span id="cb45-1379"><a href="#cb45-1379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_na</span>() <span class="sc">+</span></span>
<span id="cb45-1380"><a href="#cb45-1380" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genome_type<span class="sc">~</span>sample_type, <span class="at">scales=</span><span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb45-1381"><a href="#cb45-1381" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb45-1382"><a href="#cb45-1382" aria-hidden="true" tabindex="-1"></a>g_path_genera</span>
<span id="cb45-1383"><a href="#cb45-1383" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-1384"><a href="#cb45-1384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1385"><a href="#cb45-1385" aria-hidden="true" tabindex="-1"></a>One immediate observation, which I think helps demonstrate the unreliability of the RNA data here, is the almost complete absence of RNA-virus genera of interest. It's very suspicious for putative HIV reads to be almost as abundant as *Enterovirus* reads (which, remember, includes rhinovirus), while coronaviruses and influenza are totally absent. The DNA reads seem much more similar to Prussin, though I'm still surprised HCMV and *Mastadenovirus* are so rare.</span>
<span id="cb45-1386"><a href="#cb45-1386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1387"><a href="#cb45-1387" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusion</span></span>
<span id="cb45-1388"><a href="#cb45-1388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1389"><a href="#cb45-1389" aria-hidden="true" tabindex="-1"></a>This is the second, and by far the smaller, of the air-sampling datasets I've analyzed with this pipeline so far. Many of the high-level findings were similar to Prussin, including high relative abundance of human reads, low total viral reads, an absence of enteric viruses, and high abundance of papillomaviruses among human-infecting viruses. The biggest difference in the DNA data was the almost complete lack of cytomegalovirus, compared to its very high abundance in the Prussin data.</span>
<span id="cb45-1390"><a href="#cb45-1390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-1391"><a href="#cb45-1391" aria-hidden="true" tabindex="-1"></a>Given how small this dataset is (16M reads on a MiSeq, compared to \~450M for Prussin and 1.3B for the upcoming <span class="co">[</span><span class="ot">Leung et al.</span><span class="co">](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01044-7)</span> dataset), I don't think too much weight should be placed on these findings, but given how rare air-MGS datasets are it's nice to have another one.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>