<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-01-23">
<title>Will’s Public NAO Notebook - Workflow analysis of Crits-Christoph et al.&nbsp;(2021), part 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Workflow analysis of Crits-Christoph et al.&nbsp;(2021), part 2</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Abundance and composition of human-infecting viruses.</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 23, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>In my last entry, I described the process of analyzing <a href="https://doi.org/10.1128%2FmBio.02703-20">Crits-Christoph et al.&nbsp;(2021)</a> data with my <a href="https://github.com/naobservatory/mgs-workflow">Nextflow workflow</a>, up to and including high-level taxonomic analysis. In this entry, I’ll look more deeply at the human-infecting-virus content of these data, and address some inadequacies in that pipeline that were causing issues in analyzing those reads.</p>
<p>To begin with, I analyzed the human-infecting virus reads in Crits-Christoph in an identical manner to my <a href="https://data.securebio.org/wills-public-notebook/notebooks/2023-12-22_bmc-rna-sequel.html">previous analysis</a> of our BMC data, making use of the automated BLAST characterization approach developed in a <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-01-30_blast-validation.html">previous entry</a>.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Data input paths</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-02-04_crits-christoph-2/"</span></span>
<span><span class="va">libraries_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"cc-libraries.txt"</span><span class="op">)</span></span>
<span><span class="va">basic_stats_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"qc_basic_stats.tsv"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_all_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_all.tsv.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import data</span></span>
<span><span class="va">stages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"remove_human"</span>, <span class="st">"remove_other"</span>, <span class="st">"ribo_secondary"</span><span class="op">)</span></span>
<span><span class="va">libraries</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">libraries_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>enrichment <span class="op">=</span> <span class="fu">str_to_title</span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">basic_stats</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">basic_stats_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>stage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">stage</span>, levels <span class="op">=</span> <span class="va">stages</span><span class="op">)</span>,</span>
<span>         sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_reads_all</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_all_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered_1</span> <span class="op">&lt;-</span> <span class="va">hv_reads_all</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">classified</span> <span class="op">|</span> <span class="va">assigned_hv</span><span class="op">)</span></span>
<span><span class="va">hv_reads_filtered</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_reads_filtered_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">libraries</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">enrichment</span>, <span class="va">location</span>, <span class="va">collection_date</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>sample <span class="op">=</span> <span class="fu">fct_inorder</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Count</span></span>
<span><span class="va">n_hv_reads_all</span> <span class="op">&lt;-</span> <span class="va">hv_reads_all</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">method</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"remove_other"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered_1</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered_1</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">method</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_filtered_1 <span class="op">=</span> <span class="va">n</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">n_hv_reads_all</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"method"</span>, <span class="st">"enrichment"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_filtered_1</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span><span class="op">*</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">n_hv_filtered</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">location</span>, <span class="va">method</span>, <span class="va">enrichment</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">count</span> <span class="op">%&gt;%</span> <span class="fu">inner_join</span><span class="op">(</span><span class="va">basic_stats</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"remove_other"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">n_read_pairs</span><span class="op">)</span>, by<span class="op">=</span><span class="st">"sample"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>n_putative <span class="op">=</span> <span class="va">n</span>, n_total <span class="op">=</span> <span class="va">n_read_pairs</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>p_reads <span class="op">=</span> <span class="va">n_putative</span><span class="op">/</span><span class="va">n_total</span>, pc_reads <span class="op">=</span> <span class="va">p_reads</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The process for identifying human-viral (HV) reads was similar to that described in my last two entries:</p>
<ol type="1">
<li>After removing human and livestock reads with BBMap, the remaining reads were aligned to a DB of human-infecting virus genomes with Bowtie2, using permissive alignment parameters. This retained roughly 0.3% to 0.4% of surviving reads (57k to 130k reads) for unenriched samples and 1% to 67% (650 to 25k reads) for enriched samples.</li>
<li>Reads that aligned successfully with Bowtie2 were run through Kraken2 (using the standard 16GB database) and reads that were assigned to any non-human-virus taxon were excluded. This retained roughly 0.17% to 0.29% of surviving reads (30k to 81k reads) for unenriched samples, and made little difference to the enriched samples.</li>
<li>For surviving reads, the length-adjusted alignment score <span class="math inline">\(S=\frac{\text{bowtie2 alignment score}}{\ln(\text{read length})}\)</span> was calculated, and reads were filtered out if they didn’t meet at least one of the following four criteria:
<ul>
<li>The read pair is <em>assigned</em> to a human-infecting virus by both Kraken and Bowtie2</li>
<li>The read pair contains a Kraken2 <em>hit</em> to the same taxid as it is assigned to by Bowtie2.</li>
<li>The read pair is unassigned by Kraken and <span class="math inline">\(S&gt;15\)</span> for the forward read</li>
<li>The read pair is unassigned by Kraken and <span class="math inline">\(S&gt;15\)</span> for the reverse read</li>
</ul>
</li>
</ol>
<p>Applying all of these filtering steps leaves a total of 7042 read pairs across all unenriched samples (0.004% of surviving reads) and &gt;110,000 read pairs across all enriched samples (4% of surviving reads):</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg</span> <span class="op">&lt;-</span> <span class="va">hv_reads_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>kraken_label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">assigned_hv</span>, <span class="st">"Kraken2 HV\nassignment"</span>,</span>
<span>                               <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">hit_hv</span>, <span class="st">"Kraken2 HV\nhit"</span>,</span>
<span>                                      <span class="st">"No hit or\nassignment"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_fwd</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span><span class="va">g_mrg</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_grid</span><span class="op">(</span><span class="va">kraken_label</span><span class="op">~</span><span class="va">enrichment</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-01-23_crits-christoph_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>I previously assessed potential criteria for designating a read as human-viral based on <span class="math inline">\(S\)</span> and Kraken assignment status, and found that a disjunctive threshold of <span class="math inline">\(S \geq 20\)</span> (i.e.&nbsp;passing a read pair if either the forward or reverse read had an adjusted alignment score of at least 20) achieved the best performance. To test whether this remains the case across datasets, I BLASTed the 7042 putative HV reads from the unenriched data against nt using <code>blastn -remote</code> as <a href="https://data.securebio.org/wills-public-notebook/notebooks/2024-01-30_blast-validation.html">described previously</a>. I then assigned “ground-truth” viral status to each read pair as follows: <strong>[TODO:update criteria]</strong></p>
<ol type="1">
<li>If both the forward and reverse reads from a read pair align to the <strong>same viral taxid</strong> (while passing filters on query coverage, percent identity, etc), classify that read pair as human viral.</li>
<li>If only one of the reads in a read pair aligns to the same viral taxid as that identified by Kraken/Bowtie, classify that read pair as human viral.</li>
<li>Otherwise, if only one of the reads in a read pair aligns to any given viral taxid, flag the read for manual inspection.</li>
<li>If neither read aligns to a viral taxid, classify that read pair as non-human-viral.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Import files for viral taxid search</span></span>
<span><span class="va">hv_taxids_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"human-virus-taxids-all.txt"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"nodes.dmp.gz"</span><span class="op">)</span></span>
<span><span class="va">hv_taxids</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_taxids_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes</span> <span class="op">&lt;-</span> <span class="fu">read_delim</span><span class="op">(</span><span class="va">tax_nodes_path</span>, delim <span class="op">=</span> <span class="st">"\t|\t"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">X1</span><span class="op">:</span><span class="va">X3</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>child_taxid <span class="op">=</span> <span class="va">X1</span>, parent_taxid <span class="op">=</span> <span class="va">X2</span>, rank <span class="op">=</span> <span class="va">X3</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define taxid search function</span></span>
<span><span class="va">expand_taxids</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxids_in</span>, <span class="va">nodes</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="va">taxids_in</span></span>
<span>  <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">taxids_out</span>, <span class="va">taxids_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>    <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, </span>
<span>                         <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">taxids_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">v_taxids</span> <span class="op">&lt;-</span> <span class="fu">expand_taxids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10239</span>, <span class="va">hv_taxids</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span>, <span class="va">tax_nodes</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import BLAST results</span></span>
<span><span class="co">#blast_results_path &lt;- file.path(data_dir, "cc-unenriched-out.blast.gz")</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"cat-cc-unenriched.blast.gz"</span><span class="op">)</span></span>
<span><span class="va">blast_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="va">blast_cols</span>,</span>
<span>                          col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add viral status</span></span>
<span><span class="va">blast_results_viral</span> <span class="op">&lt;-</span> <span class="fu">mutate</span><span class="op">(</span><span class="va">blast_results</span>, viral <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for the best hit for each sequence and taxid</span></span>
<span><span class="va">blast_results_best</span> <span class="op">&lt;-</span> <span class="va">blast_results_viral</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rank hits for each sequence</span></span>
<span><span class="va">blast_results_ranked</span> <span class="op">&lt;-</span> <span class="va">blast_results_best</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">qseqid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>rank <span class="op">=</span> <span class="fu">dense_rank</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter by rank</span></span>
<span><span class="va">blast_results_highrank</span> <span class="op">&lt;-</span> <span class="va">blast_results_ranked</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">rank</span> <span class="op">&lt;=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize by read pair and taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_highrank</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">qseqid</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span>, <span class="va">viral</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, </span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_full <span class="op">=</span> <span class="va">viral</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span><span class="va">mrg_assign</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">taxid</span>, <span class="va">assigned_taxid</span><span class="op">)</span></span>
<span><span class="va">blast_results_assign</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">blast_results_paired</span>, <span class="va">mrg_assign</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>taxid_match_bowtie <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_kraken <span class="op">=</span> <span class="op">(</span><span class="va">staxid</span> <span class="op">==</span> <span class="va">assigned_taxid</span><span class="op">)</span>,</span>
<span>           taxid_match_any <span class="op">=</span> <span class="va">taxid_match_bowtie</span> <span class="op">|</span> <span class="va">taxid_match_kraken</span><span class="op">)</span></span>
<span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_assign</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_full</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">taxid_match_any</span><span class="op">)</span>, <span class="fl">2</span>,</span>
<span>                                             <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge with unenriched read data</span></span>
<span><span class="va">mrg_unenriched</span> <span class="op">&lt;-</span> <span class="va">mrg</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">enrichment</span> <span class="op">==</span> <span class="st">"Unenriched"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">blast_results_out</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span><span class="va">mrg_unenriched_plot</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_mrg_v</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_v</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-01-23_crits-christoph_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Off the bat, it’s clear this approach isn’t working well. There are almost 1,000 “UNCLEAR” data points that need manual inspection; worse, there appear to be numerous nonviral sequences achieving very high normalized Bowtie2 alignment scores:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unenriched_debug</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>adj_score_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="va">adj_score_fwd</span>, <span class="va">adj_score_rev</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_hist</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">mrg_unenriched_debug</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span>binwidth<span class="op">=</span><span class="fl">5</span>,boundary<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">viral_status_out</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Maximum adjusted alignment score"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"# Read pairs"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span></span>
<span><span class="va">g_hist</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-01-23_crits-christoph_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Inspecting these high-scoring non-viral reads, we see that they are dominated by a few specific genome IDs: the top five genome IDs account for over 75% of high-scoring false-positives, and all of the top 10 overwhelmingly produce false rather than true positives.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">header_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"human-viral-headers.txt"</span><span class="op">)</span></span>
<span><span class="va">header_db</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">header_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                      col_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"genome_id"</span>, <span class="st">"genome_name"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mrg_unenriched_genomes</span> <span class="op">&lt;-</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">mrg_unenriched_debug</span>, <span class="va">header_db</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bad_genomes</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_genomes</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">viral_status_out</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">count</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from<span class="op">=</span><span class="va">viral_status_out</span>, values_from<span class="op">=</span><span class="va">n</span>, names_prefix <span class="op">=</span> <span class="st">"n_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">n_FALSE</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">genome_id</span>, <span class="va">genome_name</span>, <span class="va">n_FALSE</span>, <span class="va">n_TRUE</span>, <span class="va">n_UNCLEAR</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>n_TRUE <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_TRUE</span>, <span class="fl">0</span><span class="op">)</span>, n_UNCLEAR <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">n_UNCLEAR</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>p_FALSE <span class="op">=</span> <span class="va">n_FALSE</span><span class="op">/</span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">+</span><span class="va">n_TRUE</span><span class="op">+</span><span class="va">n_UNCLEAR</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["genome_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["genome_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_FALSE"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_TRUE"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_UNCLEAR"],"name":[5],"type":["int"],"align":["right"]},{"label":["p_FALSE"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"GU980198.1","2":"Human herpesvirus 5 transgenic strain CINCY+Towne, complete genome","3":"362","4":"27","5":"15","6":"0.896039604"},{"1":"KY633487.1","2":"UNVERIFIED: Vaccinia virus transgenic isolate MVA-CR19.GFP, complete genome","3":"134","4":"1","5":"4","6":"0.964028777"},{"1":"MH144178.1","2":"Measles virus genotype A transgenic strain vac2(GFP)H, complete genome","3":"131","4":"1","5":"2","6":"0.977611940"},{"1":"MT277585.1","2":"Mutant Human adenovirus 2 isolate HAdV-C2-dE3B-CMV-GFP, complete sequence","3":"118","4":"3","5":"3","6":"0.951612903"},{"1":"KY766069.1","2":"Zika virus isolate Pf13/251013-18, complete genome","3":"86","4":"0","5":"2","6":"0.977272727"},{"1":"FJ593289.1","2":"Human herpesvirus 1 transgenic strain 17, complete genome","3":"39","4":"0","5":"1","6":"0.975000000"},{"1":"AB618031.1","2":"Herpes simplex virus (type 1 /strain RH2) DNA, nearly complete genome","3":"23","4":"0","5":"2","6":"0.920000000"},{"1":"AC090446.27","2":"Baboon cytomegalovirus OCOM4-37 clone bacmv, WORKING DRAFT SEQUENCE, 22 unordered pieces","3":"20","4":"1","5":"1","6":"0.909090909"},{"1":"MN369532.1","2":"Vaccinia virus isolate VK01, complete genome","3":"20","4":"0","5":"0","6":"1.000000000"},{"1":"HM133903.1","2":"Orf virus strain D1701, complete genome","3":"11","4":"0","5":"0","6":"1.000000000"},{"1":"HQ540592.1","2":"Porcine endogenous retrovirus A clone 1347C1, complete genome","3":"11","4":"3","5":"0","6":"0.785714286"},{"1":"KF493877.1","2":"Human herpesvirus 5 transgenic isolate Towne-BAC-der, complete genome","3":"8","4":"24","5":"4","6":"0.222222222"},{"1":"KF493876.1","2":"Human herpesvirus 5 transgenic isolate Towne-BAC_UL96_Mutant, complete genome","3":"7","4":"19","5":"0","6":"0.269230769"},{"1":"GU179001.1","2":"Human herpesvirus 5 transgenic strain Merlin, complete genome","3":"6","4":"4","5":"0","6":"0.600000000"},{"1":"MF118167.1","2":"Human fecal virus Jorvi2, complete genome","3":"6","4":"41","5":"4","6":"0.117647059"},{"1":"HM011556.1","2":"Merkel cell polyomavirus isolate R17b, complete genome","3":"5","4":"313","5":"0","6":"0.015723270"},{"1":"D00835.1","2":"Human immunodeficiency virus 2 proviral DNA, complete genome","3":"4","4":"0","5":"0","6":"1.000000000"},{"1":"MT135283.1","2":"UNVERIFIED: Hudisavirus sp. isolate VIRES_AH01_2265035, complete genome","3":"4","4":"1","5":"0","6":"0.800000000"},{"1":"MG522853.1","2":"Hudisavirus sp. isolate ETH_P4_2016, complete genome","3":"3","4":"28","5":"0","6":"0.096774194"},{"1":"MK733609.1","2":"Human gammaherpesvirus 8 transgenic clone BAC16, complete genome","3":"3","4":"15","5":"0","6":"0.166666667"},{"1":"FJ804072.1","2":"Human papillomavirus type 116 strain HPV116, complete sequence","3":"2","4":"0","5":"0","6":"1.000000000"},{"1":"FM955841.1","2":"Human papillomavirus type 105, complete genome","3":"2","4":"18","5":"1","6":"0.095238095"},{"1":"JQ898342.1","2":"Kobuvirus sewage Kathmandu isolate KoV-SewKTM, complete genome","3":"2","4":"17","5":"0","6":"0.105263158"},{"1":"KU343137.1","2":"Gemycircularvirus HV-GcV2, complete genome","3":"2","4":"4","5":"0","6":"0.333333333"},{"1":"KX673221.1","2":"Husavirus sp. isolate 16370_59, complete genome","3":"2","4":"153","5":"0","6":"0.012903226"},{"1":"MF351515.1","2":"Hudisavirus sp. isolate P22, complete genome","3":"2","4":"4","5":"0","6":"0.333333333"},{"1":"MN900952.1","2":"Mutant Human betaherpesvirus 5 clone AD169-BAC2, complete genome","3":"2","4":"0","5":"0","6":"1.000000000"},{"1":"U31780.1","2":"Human papillomavirus type 22, complete genome","3":"2","4":"28","5":"1","6":"0.064516129"},{"1":"AB186897.1","2":"Human picobirnavirus RNA segment 1 genomic RNA, complete sequence","3":"1","4":"3","5":"0","6":"0.250000000"},{"1":"AB186898.1","2":"Human picobirnavirus RNA segment 2 genomic RNA, complete sequence","3":"1","4":"2","5":"0","6":"0.333333333"},{"1":"EU410348.1","2":"Human papillomavirus type 110, complete genome","3":"1","4":"349","5":"0","6":"0.002857143"},{"1":"FJ947080.1","2":"Human papillomavirus type 115 isolate GC02, complete genome","3":"1","4":"17","5":"0","6":"0.055555556"},{"1":"JF909601.1","2":"Senegalvirus SSV-A contig6 genomic sequence","3":"1","4":"2","5":"0","6":"0.333333333"},{"1":"JQ410350.1","2":"Ectromelia virus ERPV culture-collection ATCC:VR-1431, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"JX514942.1","2":"Human enterovirus C116 polyprotein gene, complete cds","3":"1","4":"1","5":"0","6":"0.500000000"},{"1":"KC443598.1","2":"Rotavirus A strain RVA/Human-wt/AUS/CK20039/2008/G1P[8] segment 1 RNA-dependent RNA polymerase VP1 (VP1) gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KF035107.1","2":"Human rotavirus A isolate RVA/Human-wt/BRB/2012821133/2012/G4P[14] VP1 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KF371874.1","2":"Human rotavirus A strain RVA/Human-wt/CHN/E2835/2011/G3P[8] segment 2, complete sequence","3":"1","4":"1","5":"0","6":"0.500000000"},{"1":"KF371999.1","2":"Human rotavirus A strain RVA/Human-wt/CHN/Z1557/2011/G3P[8] segment 8, complete sequence","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KJ412577.1","2":"Rotavirus A strain RVA/Human-wt/PRY/1471SR/2006/G12P[9] segment 3 RNA capping protein VP3 (VP3) gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KM023140.1","2":"Salivirus FHB, complete genome","3":"1","4":"9","5":"0","6":"0.100000000"},{"1":"KP641339.1","2":"Human mastadenovirus D isolate human/DEU/Leipzig/2014/70[P70H70F29], complete genome","3":"1","4":"1","5":"0","6":"0.500000000"},{"1":"KP882299.1","2":"Rotavirus A strain RVA/Human-wt/GHA/Ghan-002/2008/G2P[4] segment 1 VP1 (VP1) gene, partial cds","3":"1","4":"1","5":"0","6":"0.500000000"},{"1":"KR052736.1","2":"Rotavirus A strain RVA/Pig-hhp/USA/LS00009_RV0084/2011/G9P[13] segment 2 core capsid protein VP2 (VP2) gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KT600065.1","2":"Human feces pecovirus strain PeCV-PE, complete genome","3":"1","4":"27","5":"0","6":"0.035714286"},{"1":"KT600066.1","2":"Human feces pecovirus strain PeCV-NI, complete genome","3":"1","4":"47","5":"0","6":"0.020833333"},{"1":"KU059788.1","2":"Rotavirus A strain RVA/Human-wt/AUS/WAPC2016/2014/G3P[8] segment 1 RNA-dependent RNA polymerase VP1 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KU356575.1","2":"Human rotavirus A strain RVA/Human-wt/BGN/J253/2010/G2P[4] VP2 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KX079488.1","2":"Norovirus GII.21 strain NoV/Kor/JW, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"KX814961.1","2":"Rotavirus A isolate RVA/Bat-wt/CHN/YSSK5/2015/G3P[3] NSP3 gene, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC066150.1","2":"Human rotavirus A VP1 gene for VP1 protein, partial cds, strain: RVA/Human-wt/VNM/SP026/2012/G1P[8]","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC328210.1","2":"Rotavirus A RVA/Cat-tc/JPN/FRV303/1993/G3P[3] VP4 gene for viral protein 4, complete cds","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"LC708010.1","2":"Hudisavirus sp. CRESSV19-84-AMS-06 DNA, complete genome","3":"1","4":"4","5":"0","6":"0.200000000"},{"1":"LC708012.1","2":"Hudisavirus sp. CRESSV19-84-AMS-08 DNA, complete genome","3":"1","4":"5","5":"0","6":"0.166666667"},{"1":"LC708013.1","2":"Hudisavirus sp. CRESSV19-84-AMS-09 DNA, complete genome","3":"1","4":"35","5":"0","6":"0.027777778"},{"1":"LT960370.1","2":"Human polyomavirus 1 isolate CAMB-1055 genome assembly, chromosome: I","3":"1","4":"16","5":"0","6":"0.058823529"},{"1":"M81861.1","2":"Cardiovirus A1 strain Ruckert polyprotein gene, complete cds","3":"1","4":"0","5":"2","6":"0.333333333"},{"1":"MF351517.1","2":"Hudisavirus sp. isolate P29, complete genome","3":"1","4":"16","5":"0","6":"0.058823529"},{"1":"MF351520.1","2":"Hudisavirus sp. isolate P13_2, complete genome","3":"1","4":"5","5":"0","6":"0.166666667"},{"1":"MG522854.1","2":"Hudisavirus sp. isolate ETH_P7_2016, complete genome","3":"1","4":"5","5":"0","6":"0.166666667"},{"1":"MH121071.1","2":"UNVERIFIED_CONTAM: Human mastadenovirus C strain 3C2, partial genome","3":"1","4":"2","5":"4","6":"0.142857143"},{"1":"MH341447.1","2":"Vaccinia virus strain L-IVP_oncoQ, complete genome","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"MK059950.1","2":"Human astrovirus 2 strain Oxford, complete genome","3":"1","4":"3","5":"0","6":"0.250000000"},{"1":"ON549927.1","2":"Mutant Cowpox virus strain recombinant Brighton Red clone deltaCPXV014, complete genome","3":"1","4":"1","5":"0","6":"0.500000000"},{"1":"OR453678.1","2":"UNVERIFIED: Orf virus, complete sequence","3":"1","4":"0","5":"0","6":"1.000000000"},{"1":"X70829.1","2":"Human papillomavirus type 65 complete genome","3":"1","4":"5","5":"0","6":"0.166666667"},{"1":"X74468.1","2":"Human papillomavirus type 15 genomic DNA","3":"1","4":"33","5":"0","6":"0.029411765"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Inspecting these top genome IDs more closely, we notice something interesting: many of them are transgenic. When we try BLASTing read pairs mapping to these transgenic sequences against nt with NCBI’s online tool, we find that the best matches are all to bacterial plasmids, cloning vectors and synthetic constructs. This suggests a clear culprit for a large fraction of the false positives we are seeing here: contamination of the Bowtie2 reference database with non-viral sequences inserted into transgenic viruses.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unenriched_fasta</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_debug</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">adj_score_max</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">genome_id</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>nseq <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">nseq</span><span class="op">)</span>, <span class="fu">desc</span><span class="op">(</span><span class="va">adj_score_max</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>seq_num_gid <span class="op">=</span> <span class="fu">row_number</span><span class="op">(</span><span class="op">)</span>, seq_head <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"&gt;"</span>, <span class="va">genome_id</span>, <span class="st">"_"</span>, <span class="va">seq_num_gid</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span>header1<span class="op">=</span><span class="va">seq_head</span>, seq1<span class="op">=</span><span class="va">query_seq_fwd</span>, header2<span class="op">=</span><span class="va">seq_head</span>, seq2<span class="op">=</span><span class="va">query_seq_rev</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>header1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header1</span>, <span class="st">"_1"</span><span class="op">)</span>, header2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">header2</span>, <span class="st">"_2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mu_fasta_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mrg_unenriched_fasta</span>, sep<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/write.html">write</a></span><span class="op">(</span><span class="va">mu_fasta_out</span>, <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"cc-bad-gid.fasta"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Excluding “transgenic” and “mutant” genome IDs is predicted to remove 77% of the high-scoring false positives observed above. This left a small enough number of sequences to characterize manually using NCBI BLAST. When we do this, we see the following breakdown of causes:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Separate transgenic sequences</span></span>
<span><span class="va">bad_genomes_tr</span> <span class="op">&lt;-</span> <span class="va">bad_genomes</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>transgenic <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"transgenic"</span>, <span class="va">genome_name</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">|</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"mutant"</span>, <span class="va">genome_name</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bad_genomes_notr</span> <span class="op">&lt;-</span> <span class="va">bad_genomes_tr</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">transgenic</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">ungroup</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>pr_FALSE <span class="op">=</span> <span class="va">n_FALSE</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import manually annotated causes</span></span>
<span><span class="va">bg_causes</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"cc-bad-notr.tsv"</span><span class="op">)</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">bg_causes_out</span> <span class="op">&lt;-</span> <span class="va">bg_causes</span> <span class="op">%&gt;%</span> <span class="fu">group_by</span><span class="op">(</span><span class="va">cause</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">summarize</span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">n_FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bg_causes_out</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["cause"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"Human genome / eukaryotic synthetic construct","2":"130"},{"1":"Appears real","2":"37"},{"1":"No match","2":"27"},{"1":"E. coli chromosomal sequence","2":"23"},{"1":"Cow genome","2":"12"},{"1":"Pig genome","2":"11"},{"1":"Other bacterial genome","2":"2"},{"1":"Arabidopsis genome","2":"1"},{"1":"Cloning/expression vector","2":"1"},{"1":"Other eukaryotic genome","2":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>By far the most important attributable causes for misidentification of viral reads are:</p>
<ol type="1">
<li>Mapping to the human genome or human-derived synthetic constructs</li>
<li>Mapping to the E. coli genome</li>
<li>Mapping to cow or pig genomes</li>
</ol>
<p>This suggests (1) that the current filtering to remove human and cow/pig sequences is insufficiently stringent, and (2) that adding <em>E. coli</em> and possibly one or more eukaryotic synthetic constructs to the database of contaminants to screen for could go a long way toward removing remaining false positives.</p>
<p>Finally, it’s worth noting that a significant fraction of these sequences appear to be Bowtie true-positives that were falsely annotated as negatives by my automated BLAST analysis above. I’m not yet sure what’s underlying this, but it gives me some confidence that, if I can fix the issues outlined above, the general approach of this pipeline is still viable.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mrg_unenriched_plot_2</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_debug</span> <span class="op">%&gt;%</span> <span class="fu">full_join</span><span class="op">(</span><span class="va">bg_causes</span>, by<span class="op">=</span><span class="st">"genome_id"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>cause <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">cause</span>, <span class="st">"NA"</span><span class="op">)</span>,</span>
<span>         viral_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"Appears real"</span>, <span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">cause</span> <span class="op">==</span> <span class="st">"No match"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmax</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">viral_status</span><span class="op">)</span>, <span class="va">viral_status</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"FALSE"</span>,</span>
<span>                                   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status</span> <span class="op">==</span> <span class="fl">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">viral_status_out</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">g_mrg_v2</span> <span class="op">&lt;-</span> <span class="va">mrg_unenriched_plot_2</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">adj_score_fwd</span>, y<span class="op">=</span><span class="va">adj_score_rev</span>, color<span class="op">=</span><span class="va">viral_status_out</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, shape<span class="op">=</span><span class="fl">16</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(forward read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">"S(reverse read)"</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">40</span><span class="op">)</span>, breaks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">100</span>,<span class="fl">10</span><span class="op">)</span>, expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span>, name <span class="op">=</span> <span class="st">"Viral status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">kraken_label</span>, labeller <span class="op">=</span> <span class="fu">labeller</span><span class="op">(</span>kit <span class="op">=</span> <span class="fu">label_wrap_gen</span><span class="op">(</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="va">theme_base</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">g_mrg_v2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="2024-01-23_crits-christoph_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>


<!-- -->


</main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb11" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workflow analysis of Crits-Christoph et al. (2021), part 2"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Abundance and composition of human-infecting viruses."</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-01-23</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> true</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>theme_base <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="cn">NULL</span>)</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>tnl <span class="ot">&lt;-</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>In my last entry, I described the process of analyzing <span class="co">[</span><span class="ot">Crits-Christoph et al. (2021)</span><span class="co">](https://doi.org/10.1128%2FmBio.02703-20)</span> data with my <span class="co">[</span><span class="ot">Nextflow workflow</span><span class="co">](https://github.com/naobservatory/mgs-workflow)</span>, up to and including high-level taxonomic analysis. In this entry, I'll look more deeply at the human-infecting-virus content of these data, and address some inadequacies in that pipeline that were causing issues in analyzing those reads.</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>To begin with, I analyzed the human-infecting virus reads in Crits-Christoph in an identical manner to my <span class="co">[</span><span class="ot">previous analysis</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2023-12-22_bmc-rna-sequel.html)</span> of our BMC data, making use of the automated BLAST characterization approach developed in a <span class="co">[</span><span class="ot">previous entry</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-01-30_blast-validation.html)</span>.</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Data input paths</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-02-04_crits-christoph-2/"</span></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>libraries_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"cc-libraries.txt"</span>)</span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>basic_stats_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"qc_basic_stats.tsv"</span>)</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>hv_reads_all_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_all.tsv.gz"</span>)</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"hv_hits_putative_filtered.tsv.gz"</span>)</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>stages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"raw_concat"</span>, <span class="st">"cleaned"</span>, <span class="st">"dedup"</span>, <span class="st">"ribo_initial"</span>, <span class="st">"remove_human"</span>, <span class="st">"remove_other"</span>, <span class="st">"ribo_secondary"</span>)</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>libraries <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(libraries_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">enrichment =</span> <span class="fu">str_to_title</span>(enrichment))</span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>basic_stats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(basic_stats_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stage =</span> <span class="fu">factor</span>(stage, <span class="at">levels =</span> stages),</span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>hv_reads_all <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_all_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered_1 <span class="ot">&lt;-</span> hv_reads_all <span class="sc">%&gt;%</span> </span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>classified <span class="sc">|</span> assigned_hv)</span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>hv_reads_filtered <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_reads_filtered_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(libraries, <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(enrichment, location, collection_date) <span class="sc">%&gt;%</span></span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">fct_inorder</span>(sample))</span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Count</span></span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a>n_hv_reads_all <span class="ot">&lt;-</span> hv_reads_all <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, location, method, enrichment) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"remove_other"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a>n_hv_filtered_1 <span class="ot">&lt;-</span> hv_reads_filtered_1 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, location, method, enrichment) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_filtered_1 =</span> n) <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(n_hv_reads_all, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"location"</span>, <span class="st">"method"</span>, <span class="st">"enrichment"</span>)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_filtered_1<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>n_hv_filtered <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample, location, method, enrichment) <span class="sc">%&gt;%</span> count <span class="sc">%&gt;%</span> <span class="fu">inner_join</span>(basic_stats <span class="sc">%&gt;%</span> <span class="fu">filter</span>(stage <span class="sc">==</span> <span class="st">"remove_other"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, n_read_pairs), <span class="at">by=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">n_putative =</span> n, <span class="at">n_total =</span> n_read_pairs) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">p_reads =</span> n_putative<span class="sc">/</span>n_total, <span class="at">pc_reads =</span> p_reads <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>The process for identifying human-viral (HV) reads was similar to that described in my last two entries:</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>After removing human and livestock reads with BBMap, the remaining reads were aligned to a DB of human-infecting virus genomes with Bowtie2, using permissive alignment parameters. This retained roughly 0.3% to 0.4% of surviving reads (57k to 130k reads) for unenriched samples and 1% to 67% (650 to 25k reads) for enriched samples.</span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Reads that aligned successfully with Bowtie2 were run through Kraken2 (using the standard 16GB database) and reads that were assigned to any non-human-virus taxon were excluded. This retained roughly 0.17% to 0.29% of surviving reads (30k to 81k reads) for unenriched samples, and made little difference to the enriched samples.</span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>For surviving reads, the length-adjusted alignment score $S=\frac{\text{bowtie2 alignment score}}{\ln(\text{read length})}$ was calculated, and reads were filtered out if they didn't meet at least one of the following four criteria:</span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is *assigned* to a human-infecting virus by both Kraken and Bowtie2</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair contains a Kraken2 *hit* to the same taxid as it is assigned to by Bowtie2.</span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is unassigned by Kraken and $S&gt;15$ for the forward read</span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>The read pair is unassigned by Kraken and $S&gt;15$ for the reverse read</span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>Applying all of these filtering steps leaves a total of 7042 read pairs across all unenriched samples (0.004% of surviving reads) and <span class="sc">\&gt;</span>110,000 read pairs across all enriched samples (4% of surviving reads):</span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>mrg <span class="ot">&lt;-</span> hv_reads_filtered <span class="sc">%&gt;%</span></span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">kraken_label =</span> <span class="fu">ifelse</span>(assigned_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">assignment"</span>,</span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">ifelse</span>(hit_hv, <span class="st">"Kraken2 HV</span><span class="sc">\n</span><span class="st">hit"</span>,</span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"No hit or</span><span class="sc">\n</span><span class="st">assignment"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(adj_score_fwd), <span class="fu">desc</span>(adj_score_rev)) <span class="sc">%&gt;%</span></span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num =</span> <span class="fu">row_number</span>())</span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>g_mrg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev)) <span class="sc">+</span></span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(kraken_label<span class="sc">~</span>enrichment, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a>g_mrg</span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a>I previously assessed potential criteria for designating a read as human-viral based on $S$ and Kraken assignment status, and found that a disjunctive threshold of $S \geq 20$ (i.e. passing a read pair if either the forward or reverse read had an adjusted alignment score of at least 20) achieved the best performance. To test whether this remains the case across datasets, I BLASTed the 7042 putative HV reads from the unenriched data against nt using <span class="in">`blastn -remote`</span> as <span class="co">[</span><span class="ot">described previously</span><span class="co">](https://data.securebio.org/wills-public-notebook/notebooks/2024-01-30_blast-validation.html)</span>. I then assigned "ground-truth" viral status to each read pair as follows: **\[TODO:update criteria\]**</span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>If both the forward and reverse reads from a read pair align to the **same viral taxid** (while passing filters on query coverage, percent identity, etc), classify that read pair as human viral.</span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>If only one of the reads in a read pair aligns to the same viral taxid as that identified by Kraken/Bowtie, classify that read pair as human viral.</span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>Otherwise, if only one of the reads in a read pair aligns to any given viral taxid, flag the read for manual inspection.</span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a><span class="ss">4.  </span>If neither read aligns to a viral taxid, classify that read pair as non-human-viral.</span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a><span class="co"># Import files for viral taxid search</span></span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a>hv_taxids_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"human-virus-taxids-all.txt"</span>)</span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a>tax_nodes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"nodes.dmp.gz"</span>)</span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a>hv_taxids <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_taxids_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> <span class="st">"taxid"</span>)</span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a>tax_nodes <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(tax_nodes_path, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">|</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X1<span class="sc">:</span>X3) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">child_taxid =</span> X1, <span class="at">parent_taxid =</span> X2, <span class="at">rank =</span> X3)</span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a><span class="co"># Define taxid search function</span></span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a>expand_taxids <span class="ot">&lt;-</span> <span class="cf">function</span>(taxids_in, nodes){</span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a>  taxids_out <span class="ot">&lt;-</span> taxids_in</span>
<span id="cb11-132"><a href="#cb11-132" aria-hidden="true" tabindex="-1"></a>  taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb11-133"><a href="#cb11-133" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="fu">length</span>(taxids_new) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a>    taxids_out <span class="ot">&lt;-</span> <span class="fu">c</span>(taxids_out, taxids_new) <span class="sc">%&gt;%</span> sort</span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a>    taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, </span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a>                         <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-140"><a href="#cb11-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(taxids_out)</span>
<span id="cb11-141"><a href="#cb11-141" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-142"><a href="#cb11-142" aria-hidden="true" tabindex="-1"></a>v_taxids <span class="ot">&lt;-</span> <span class="fu">expand_taxids</span>(<span class="fu">c</span>(<span class="dv">10239</span>, hv_taxids<span class="sc">$</span>taxid), tax_nodes)</span>
<span id="cb11-143"><a href="#cb11-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-144"><a href="#cb11-144" aria-hidden="true" tabindex="-1"></a><span class="co"># Import BLAST results</span></span>
<span id="cb11-145"><a href="#cb11-145" aria-hidden="true" tabindex="-1"></a><span class="co">#blast_results_path &lt;- file.path(data_dir, "cc-unenriched-out.blast.gz")</span></span>
<span id="cb11-146"><a href="#cb11-146" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"cat-cc-unenriched.blast.gz"</span>)</span>
<span id="cb11-147"><a href="#cb11-147" aria-hidden="true" tabindex="-1"></a>blast_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span>)</span>
<span id="cb11-148"><a href="#cb11-148" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> blast_cols,</span>
<span id="cb11-149"><a href="#cb11-149" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb11-150"><a href="#cb11-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-151"><a href="#cb11-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Add viral status</span></span>
<span id="cb11-152"><a href="#cb11-152" aria-hidden="true" tabindex="-1"></a>blast_results_viral <span class="ot">&lt;-</span> <span class="fu">mutate</span>(blast_results, <span class="at">viral =</span> staxid <span class="sc">%in%</span> v_taxids)</span>
<span id="cb11-153"><a href="#cb11-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-154"><a href="#cb11-154" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for the best hit for each sequence and taxid</span></span>
<span id="cb11-155"><a href="#cb11-155" aria-hidden="true" tabindex="-1"></a>blast_results_best <span class="ot">&lt;-</span> blast_results_viral <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid, staxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb11-156"><a href="#cb11-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb11-157"><a href="#cb11-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-158"><a href="#cb11-158" aria-hidden="true" tabindex="-1"></a><span class="co"># Rank hits for each sequence</span></span>
<span id="cb11-159"><a href="#cb11-159" aria-hidden="true" tabindex="-1"></a>blast_results_ranked <span class="ot">&lt;-</span> blast_results_best <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(qseqid) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">dense_rank</span>(<span class="fu">desc</span>(bitscore)))</span>
<span id="cb11-160"><a href="#cb11-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-161"><a href="#cb11-161" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter by rank</span></span>
<span id="cb11-162"><a href="#cb11-162" aria-hidden="true" tabindex="-1"></a>blast_results_highrank <span class="ot">&lt;-</span> blast_results_ranked <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rank <span class="sc">&lt;=</span> <span class="dv">5</span>)</span>
<span id="cb11-163"><a href="#cb11-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-164"><a href="#cb11-164" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize by read pair and taxid</span></span>
<span id="cb11-165"><a href="#cb11-165" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_highrank <span class="sc">%&gt;%</span></span>
<span id="cb11-166"><a href="#cb11-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(qseqid, <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span>), <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-167"><a href="#cb11-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid, viral) <span class="sc">%&gt;%</span></span>
<span id="cb11-168"><a href="#cb11-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num)) <span class="sc">%&gt;%</span></span>
<span id="cb11-169"><a href="#cb11-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore), </span>
<span id="cb11-170"><a href="#cb11-170" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-171"><a href="#cb11-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_full =</span> viral <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb11-172"><a href="#cb11-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-173"><a href="#cb11-173" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare to Kraken &amp; Bowtie assignments</span></span>
<span id="cb11-174"><a href="#cb11-174" aria-hidden="true" tabindex="-1"></a>mrg_assign <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-175"><a href="#cb11-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample, seq_num, taxid, assigned_taxid)</span>
<span id="cb11-176"><a href="#cb11-176" aria-hidden="true" tabindex="-1"></a>blast_results_assign <span class="ot">&lt;-</span> <span class="fu">left_join</span>(blast_results_paired, mrg_assign, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-177"><a href="#cb11-177" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">taxid_match_bowtie =</span> (staxid <span class="sc">==</span> taxid),</span>
<span id="cb11-178"><a href="#cb11-178" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_kraken =</span> (staxid <span class="sc">==</span> assigned_taxid),</span>
<span id="cb11-179"><a href="#cb11-179" aria-hidden="true" tabindex="-1"></a>           <span class="at">taxid_match_any =</span> taxid_match_bowtie <span class="sc">|</span> taxid_match_kraken)</span>
<span id="cb11-180"><a href="#cb11-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-181"><a href="#cb11-181" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_assign <span class="sc">%&gt;%</span></span>
<span id="cb11-182"><a href="#cb11-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb11-183"><a href="#cb11-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_full), <span class="dv">2</span>,</span>
<span id="cb11-184"><a href="#cb11-184" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(taxid_match_any), <span class="dv">2</span>,</span>
<span id="cb11-185"><a href="#cb11-185" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">ifelse</span>(<span class="fu">any</span>(viral), <span class="dv">1</span>, <span class="dv">0</span>))),</span>
<span id="cb11-186"><a href="#cb11-186" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb11-187"><a href="#cb11-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-188"><a href="#cb11-188" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with unenriched read data</span></span>
<span id="cb11-189"><a href="#cb11-189" aria-hidden="true" tabindex="-1"></a>mrg_unenriched <span class="ot">&lt;-</span> mrg <span class="sc">%&gt;%</span> <span class="fu">filter</span>(enrichment <span class="sc">==</span> <span class="st">"Unenriched"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-190"><a href="#cb11-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(blast_results_out, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-191"><a href="#cb11-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status =</span> <span class="fu">replace_na</span>(viral_status, <span class="dv">0</span>))</span>
<span id="cb11-192"><a href="#cb11-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-193"><a href="#cb11-193" aria-hidden="true" tabindex="-1"></a><span class="co"># Import Bowtie2/Kraken data and perform filtering steps</span></span>
<span id="cb11-194"><a href="#cb11-194" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_plot <span class="ot">&lt;-</span> mrg_unenriched <span class="sc">%&gt;%</span></span>
<span id="cb11-195"><a href="#cb11-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb11-196"><a href="#cb11-196" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb11-197"><a href="#cb11-197" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb11-198"><a href="#cb11-198" aria-hidden="true" tabindex="-1"></a>g_mrg_v <span class="ot">&lt;-</span> mrg_unenriched_plot <span class="sc">%&gt;%</span></span>
<span id="cb11-199"><a href="#cb11-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb11-200"><a href="#cb11-200" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb11-201"><a href="#cb11-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-202"><a href="#cb11-202" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-203"><a href="#cb11-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb11-204"><a href="#cb11-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb11-205"><a href="#cb11-205" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb11-206"><a href="#cb11-206" aria-hidden="true" tabindex="-1"></a>g_mrg_v</span>
<span id="cb11-207"><a href="#cb11-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-208"><a href="#cb11-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-209"><a href="#cb11-209" aria-hidden="true" tabindex="-1"></a>Off the bat, it's clear this approach isn't working well. There are almost 1,000 "UNCLEAR" data points that need manual inspection; worse, there appear to be numerous nonviral sequences achieving very high normalized Bowtie2 alignment scores:</span>
<span id="cb11-210"><a href="#cb11-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-213"><a href="#cb11-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-214"><a href="#cb11-214" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_debug <span class="ot">&lt;-</span> mrg_unenriched_plot <span class="sc">%&gt;%</span></span>
<span id="cb11-215"><a href="#cb11-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_score_max =</span> <span class="fu">pmax</span>(adj_score_fwd, adj_score_rev))</span>
<span id="cb11-216"><a href="#cb11-216" aria-hidden="true" tabindex="-1"></a>g_hist <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(mrg_unenriched_debug, <span class="fu">aes</span>(<span class="at">x=</span>adj_score_max)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">5</span>,<span class="at">boundary=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb11-217"><a href="#cb11-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>viral_status_out) <span class="sc">+</span></span>
<span id="cb11-218"><a href="#cb11-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">"Maximum adjusted alignment score"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"# Read pairs"</span>) <span class="sc">+</span></span>
<span id="cb11-219"><a href="#cb11-219" aria-hidden="true" tabindex="-1"></a>  theme_base</span>
<span id="cb11-220"><a href="#cb11-220" aria-hidden="true" tabindex="-1"></a>g_hist</span>
<span id="cb11-221"><a href="#cb11-221" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-222"><a href="#cb11-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-223"><a href="#cb11-223" aria-hidden="true" tabindex="-1"></a>Inspecting these high-scoring non-viral reads, we see that they are dominated by a few specific genome IDs: the top five genome IDs account for over 75% of high-scoring false-positives, and all of the top 10 overwhelmingly produce false rather than true positives.</span>
<span id="cb11-224"><a href="#cb11-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-227"><a href="#cb11-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-228"><a href="#cb11-228" aria-hidden="true" tabindex="-1"></a>header_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"human-viral-headers.txt"</span>)</span>
<span id="cb11-229"><a href="#cb11-229" aria-hidden="true" tabindex="-1"></a>header_db <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(header_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>,</span>
<span id="cb11-230"><a href="#cb11-230" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"genome_id"</span>, <span class="st">"genome_name"</span>))</span>
<span id="cb11-231"><a href="#cb11-231" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_genomes <span class="ot">&lt;-</span> <span class="fu">full_join</span>(mrg_unenriched_debug, header_db, <span class="at">by=</span><span class="st">"genome_id"</span>)</span>
<span id="cb11-232"><a href="#cb11-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-233"><a href="#cb11-233" aria-hidden="true" tabindex="-1"></a>bad_genomes <span class="ot">&lt;-</span> mrg_unenriched_genomes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-234"><a href="#cb11-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genome_id, genome_name, viral_status_out) <span class="sc">%&gt;%</span> <span class="fu">count</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-235"><a href="#cb11-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>viral_status_out, <span class="at">values_from=</span>n, <span class="at">names_prefix =</span> <span class="st">"n_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-236"><a href="#cb11-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_FALSE <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n_FALSE)) <span class="sc">%&gt;%</span></span>
<span id="cb11-237"><a href="#cb11-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genome_id, genome_name, n_FALSE, n_TRUE, n_UNCLEAR) <span class="sc">%&gt;%</span></span>
<span id="cb11-238"><a href="#cb11-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_TRUE =</span> <span class="fu">replace_na</span>(n_TRUE, <span class="dv">0</span>), <span class="at">n_UNCLEAR =</span> <span class="fu">replace_na</span>(n_UNCLEAR, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-239"><a href="#cb11-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p_FALSE =</span> n_FALSE<span class="sc">/</span>(n_FALSE<span class="sc">+</span>n_TRUE<span class="sc">+</span>n_UNCLEAR))</span>
<span id="cb11-240"><a href="#cb11-240" aria-hidden="true" tabindex="-1"></a>bad_genomes</span>
<span id="cb11-241"><a href="#cb11-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-242"><a href="#cb11-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-243"><a href="#cb11-243" aria-hidden="true" tabindex="-1"></a>Inspecting these top genome IDs more closely, we notice something interesting: many of them are transgenic. When we try BLASTing read pairs mapping to these transgenic sequences against nt with NCBI's online tool, we find that the best matches are all to bacterial plasmids, cloning vectors and synthetic constructs. This suggests a clear culprit for a large fraction of the false positives we are seeing here: contamination of the Bowtie2 reference database with non-viral sequences inserted into transgenic viruses.</span>
<span id="cb11-244"><a href="#cb11-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-247"><a href="#cb11-247" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-248"><a href="#cb11-248" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_fasta <span class="ot">&lt;-</span> mrg_unenriched_debug <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj_score_max <span class="sc">&gt;=</span> <span class="dv">20</span>, viral_status <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(genome_id) <span class="sc">%&gt;%</span> </span>
<span id="cb11-249"><a href="#cb11-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">nseq =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(nseq), <span class="fu">desc</span>(adj_score_max)) <span class="sc">%&gt;%</span></span>
<span id="cb11-250"><a href="#cb11-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-251"><a href="#cb11-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">seq_num_gid =</span> <span class="fu">row_number</span>(), <span class="at">seq_head =</span> <span class="fu">paste0</span>(<span class="st">"&gt;"</span>, genome_id, <span class="st">"_"</span>, seq_num_gid)) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span></span>
<span id="cb11-252"><a href="#cb11-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">header1=</span>seq_head, <span class="at">seq1=</span>query_seq_fwd, <span class="at">header2=</span>seq_head, <span class="at">seq2=</span>query_seq_rev) <span class="sc">%&gt;%</span></span>
<span id="cb11-253"><a href="#cb11-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">header1=</span><span class="fu">paste0</span>(header1, <span class="st">"_1"</span>), <span class="at">header2=</span><span class="fu">paste0</span>(header2, <span class="st">"_2"</span>))</span>
<span id="cb11-254"><a href="#cb11-254" aria-hidden="true" tabindex="-1"></a>mu_fasta_out <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(mrg_unenriched_fasta, <span class="at">sep=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)) <span class="sc">%&gt;%</span> <span class="fu">paste</span>(<span class="at">collapse=</span><span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb11-255"><a href="#cb11-255" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(mu_fasta_out, <span class="fu">file.path</span>(data_dir, <span class="st">"cc-bad-gid.fasta"</span>))</span>
<span id="cb11-256"><a href="#cb11-256" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-257"><a href="#cb11-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-258"><a href="#cb11-258" aria-hidden="true" tabindex="-1"></a>Excluding "transgenic" and "mutant" genome IDs is predicted to remove 77% of the high-scoring false positives observed above. This left a small enough number of sequences to characterize manually using NCBI BLAST. When we do this, we see the following breakdown of causes:</span>
<span id="cb11-259"><a href="#cb11-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-262"><a href="#cb11-262" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-263"><a href="#cb11-263" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate transgenic sequences</span></span>
<span id="cb11-264"><a href="#cb11-264" aria-hidden="true" tabindex="-1"></a>bad_genomes_tr <span class="ot">&lt;-</span> bad_genomes <span class="sc">%&gt;%</span> </span>
<span id="cb11-265"><a href="#cb11-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">transgenic =</span> <span class="fu">grepl</span>(<span class="st">"transgenic"</span>, genome_name, <span class="at">ignore.case=</span><span class="cn">TRUE</span>)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">"mutant"</span>, genome_name, <span class="at">ignore.case=</span><span class="cn">TRUE</span>))</span>
<span id="cb11-266"><a href="#cb11-266" aria-hidden="true" tabindex="-1"></a>bad_genomes_notr <span class="ot">&lt;-</span> bad_genomes_tr <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>transgenic) <span class="sc">%&gt;%</span> ungroup <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">pr_FALSE =</span> n_FALSE<span class="sc">/</span><span class="fu">sum</span>(n_FALSE))</span>
<span id="cb11-267"><a href="#cb11-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-268"><a href="#cb11-268" aria-hidden="true" tabindex="-1"></a><span class="co"># Import manually annotated causes</span></span>
<span id="cb11-269"><a href="#cb11-269" aria-hidden="true" tabindex="-1"></a>bg_causes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">file.path</span>(data_dir, <span class="st">"cc-bad-notr.tsv"</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-270"><a href="#cb11-270" aria-hidden="true" tabindex="-1"></a>bg_causes_out <span class="ot">&lt;-</span> bg_causes <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cause) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(n_FALSE)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(n))</span>
<span id="cb11-271"><a href="#cb11-271" aria-hidden="true" tabindex="-1"></a>bg_causes_out</span>
<span id="cb11-272"><a href="#cb11-272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb11-273"><a href="#cb11-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-274"><a href="#cb11-274" aria-hidden="true" tabindex="-1"></a>By far the most important attributable causes for misidentification of viral reads are:</span>
<span id="cb11-275"><a href="#cb11-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-276"><a href="#cb11-276" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Mapping to the human genome or human-derived synthetic constructs</span>
<span id="cb11-277"><a href="#cb11-277" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Mapping to the E. coli genome</span>
<span id="cb11-278"><a href="#cb11-278" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>Mapping to cow or pig genomes</span>
<span id="cb11-279"><a href="#cb11-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-280"><a href="#cb11-280" aria-hidden="true" tabindex="-1"></a>This suggests (1) that the current filtering to remove human and cow/pig sequences is insufficiently stringent, and (2) that adding *E. coli* and possibly one or more eukaryotic synthetic constructs to the database of contaminants to screen for could go a long way toward removing remaining false positives.</span>
<span id="cb11-281"><a href="#cb11-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-282"><a href="#cb11-282" aria-hidden="true" tabindex="-1"></a>Finally, it's worth noting that a significant fraction of these sequences appear to be Bowtie true-positives that were falsely annotated as negatives by my automated BLAST analysis above. I'm not yet sure what's underlying this, but it gives me some confidence that, if I can fix the issues outlined above, the general approach of this pipeline is still viable.</span>
<span id="cb11-283"><a href="#cb11-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-286"><a href="#cb11-286" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb11-287"><a href="#cb11-287" aria-hidden="true" tabindex="-1"></a>mrg_unenriched_plot_2 <span class="ot">&lt;-</span> mrg_unenriched_debug <span class="sc">%&gt;%</span> <span class="fu">full_join</span>(bg_causes, <span class="at">by=</span><span class="st">"genome_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-288"><a href="#cb11-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cause =</span> <span class="fu">replace_na</span>(cause, <span class="st">"NA"</span>),</span>
<span id="cb11-289"><a href="#cb11-289" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status =</span> <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"Appears real"</span>, <span class="dv">2</span>, <span class="fu">ifelse</span>(cause <span class="sc">==</span> <span class="st">"No match"</span>, <span class="fu">pmax</span>(<span class="dv">1</span>, viral_status), viral_status))) <span class="sc">%&gt;%</span></span>
<span id="cb11-290"><a href="#cb11-290" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_out =</span> <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"FALSE"</span>,</span>
<span id="cb11-291"><a href="#cb11-291" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(viral_status <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)),</span>
<span id="cb11-292"><a href="#cb11-292" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_out =</span> <span class="fu">factor</span>(viral_status_out, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"FALSE"</span>, <span class="st">"UNCLEAR"</span>, <span class="st">"TRUE"</span>)))</span>
<span id="cb11-293"><a href="#cb11-293" aria-hidden="true" tabindex="-1"></a>g_mrg_v2 <span class="ot">&lt;-</span> mrg_unenriched_plot_2 <span class="sc">%&gt;%</span></span>
<span id="cb11-294"><a href="#cb11-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>adj_score_fwd, <span class="at">y=</span>adj_score_rev, <span class="at">color=</span>viral_status_out)) <span class="sc">+</span></span>
<span id="cb11-295"><a href="#cb11-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb11-296"><a href="#cb11-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name=</span><span class="st">"S(forward read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-297"><a href="#cb11-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">"S(reverse read)"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">40</span>), <span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb11-298"><a href="#cb11-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>, <span class="at">name =</span> <span class="st">"Viral status"</span>) <span class="sc">+</span></span>
<span id="cb11-299"><a href="#cb11-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>kraken_label, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">kit =</span> <span class="fu">label_wrap_gen</span>(<span class="dv">20</span>))) <span class="sc">+</span></span>
<span id="cb11-300"><a href="#cb11-300" aria-hidden="true" tabindex="-1"></a>  theme_base <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb11-301"><a href="#cb11-301" aria-hidden="true" tabindex="-1"></a>g_mrg_v2</span>
<span id="cb11-302"><a href="#cb11-302" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>