<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Will Bradshaw">
<meta name="dcterms.date" content="2024-01-30">
<title>Will’s Public NAO Notebook - Automating BLAST validation of human viral read assignment</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><style>

      .quarto-title-block .quarto-title-banner {
        background: black;
      }
</style>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>
</head>
<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Will’s Public NAO Notebook</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Automating BLAST validation of human viral read assignment</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">Experiments with BLASTN remote mode</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Will Bradshaw </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">January 30, 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><p>In previous entries, I presented the results of my <a href="https://github.com/naobservatory/mgs-workflow">Nextflow workflow</a> on recent BMC RNA-seq data. To validate and calibrate my Bowtie2/Kraken2 pipeline for identifying human-viral reads, I manually BLASTed putative viral reads against NCBI’s nt database using their online BLAST tool, then assessed the results by eye to determine if each read likely came from a human virus. This approach was effective, but slow and manual, and thus hard to scale to other datasets I wanted to analyze with a higher relative abundance of viral sequences. I thus investigated options for automating the process to generate custom tabular BLAST results on the command line, which could then be read and processed programmatically.</p>
<p>Among the several options I investigated, the most promising were (1) <a href="https://blast.ncbi.nlm.nih.gov/doc/elastic-blast/">Elastic-BLAST</a>, NCBI’s own cloud-based BLAST tool, and (2) running regular command-line blast with the <code>-remote</code> option enabled to query NCBI’s online databases. I don’t currently have the AWS permissions needed to run Elastic-BLAST on our AWS service, and <code>blastn -remote</code> initially seemed too slow to be useful when run on an EC2 instance. However, I discovered that the latter option appears to run much more quickly when run on my local machine, making this a much more attractive option, at least until I’m able to run Elastic-BLAST.</p>
<p>To evaluate the potential for this approach to semi-automate the process of HV read validation, I ran BLASTN on the 227 putative HV reads from my previous BMC entry, using the following command:</p>
<pre><code>blastn -query &lt;input_path&gt; -out &lt;output_path&gt; -db nt -remote -perc_identity 60 -max_hsps 5 -num_alignments 50 -qcov_hsp_perc 30 -outfmt "6 qseqid sseqid sgi staxid qlen evalue bitscore qcovs length pident mismatch gapopen sstrand qstart qend sstart send"</code></pre>
<p>To begin with, I first needed to identify a list of taxids I could use to designate a read as human-viral. To do this, I took the list of HV taxids generated by my Nextflow workflow and expanded it to include any missing descendent taxids using the NCBI taxid tree structure. I also generated another list of taxids that included any taxid descended from any of these taxids or the overall virus taxid (10239); this latter approach decreases the risk of false negatives at the cost of potentially increasing the risk of false positives from phage sequences. The two approaches generated lists of 28,105 and 234,499 taxids, respectively.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Set file paths</span></span>
<span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="st">"../data/2024-01-30_blast/"</span></span>
<span><span class="va">reads_db_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"bmc-hv-putative.tsv"</span><span class="op">)</span></span>
<span><span class="va">blast_results_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"bmc-hv-putative.blast"</span><span class="op">)</span></span>
<span><span class="va">hv_taxids_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"human-virus-taxids-all.txt"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"nodes.dmp.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import files for viral taxid search</span></span>
<span><span class="va">hv_taxids</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">hv_taxids_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="st">"taxid"</span><span class="op">)</span></span>
<span><span class="va">tax_nodes</span> <span class="op">&lt;-</span> <span class="fu">read_delim</span><span class="op">(</span><span class="va">tax_nodes_path</span>, delim <span class="op">=</span> <span class="st">"\t|\t"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">X1</span><span class="op">:</span><span class="va">X3</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">rename</span><span class="op">(</span>child_taxid <span class="op">=</span> <span class="va">X1</span>, parent_taxid <span class="op">=</span> <span class="va">X2</span>, rank <span class="op">=</span> <span class="va">X3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define taxid search function</span></span>
<span><span class="va">expand_taxids</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">taxids_in</span>, <span class="va">nodes</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="va">taxids_in</span></span>
<span>  <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="kw">while</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">taxids_new</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">taxids_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">taxids_out</span>, <span class="va">taxids_new</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>    <span class="va">taxids_new</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">parent_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span>, </span>
<span>                         <span class="op">!</span><span class="va">child_taxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">taxids_out</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">pull</span><span class="op">(</span><span class="va">child_taxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="va">sort</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">taxids_out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">v_taxids_1</span> <span class="op">&lt;-</span> <span class="fu">expand_taxids</span><span class="op">(</span><span class="va">hv_taxids</span><span class="op">$</span><span class="va">taxid</span>, <span class="va">tax_nodes</span><span class="op">)</span></span>
<span><span class="va">v_taxids_2</span> <span class="op">&lt;-</span> <span class="fu">expand_taxids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10239</span>, <span class="va">hv_taxids</span><span class="op">$</span><span class="va">taxid</span><span class="op">)</span>, <span class="va">tax_nodes</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>I next imported the BLAST alignment results and performed exploratory data analysis. Following visual inspection, it appeared that the following process would generate good results:</p>
<ol type="1">
<li>If both the forward and reverse reads from a read pair align to the same viral taxid (given the specified filters on query coverage, percent identity, etc), classify that read pair as human viral.</li>
<li>If only one of the reads in a read pair aligns to any given viral taxid, flag the read for manual inspection.</li>
<li>If neither read aligns to a viral taxid, classify that read pair as non-human-viral.</li>
</ol>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare and import BLAST results</span></span>
<span><span class="va">reads_db</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">reads_db_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">blast_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span><span class="op">)</span></span>
<span><span class="va">blast_results</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="va">blast_results_path</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span>, col_names <span class="op">=</span> <span class="va">blast_cols</span>,</span>
<span>                          col_types <span class="op">=</span> <span class="fu">cols</span><span class="op">(</span>.default<span class="op">=</span><span class="st">"c"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter BLAST results by read ID</span></span>
<span><span class="va">reads_db_seqs</span> <span class="op">&lt;-</span> <span class="va">reads_db</span> <span class="op">%&gt;%</span> <span class="fu">mutate</span><span class="op">(</span>read_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">pull</span><span class="op">(</span><span class="va">read_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"1"</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"2"</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">blast_results_filtered</span> <span class="op">&lt;-</span> <span class="va">blast_results</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">qseqid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">reads_db_seqs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize BLAST results by read pair and subject taxid</span></span>
<span><span class="va">blast_results_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_filtered</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">qseqid</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">read_pair</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, </span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign viral status to BLAST results</span></span>
<span><span class="va">blast_results_hviral</span> <span class="op">&lt;-</span> <span class="va">blast_results_paired</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_1 <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids_1</span>,</span>
<span>         viral_2 <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids_2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_1_full <span class="op">=</span> <span class="va">viral_1</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span>,</span>
<span>         viral_2_full <span class="op">=</span> <span class="va">viral_2</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign putative viral status to read pairs</span></span>
<span><span class="va">blast_results_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_hviral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status_1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_1_full</span><span class="op">)</span>, <span class="st">"TRUE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_1</span><span class="op">)</span>, <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            viral_status_2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_2_full</span><span class="op">)</span>, <span class="st">"TRUE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_2</span><span class="op">)</span>, <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">reads_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">hv_blast</span><span class="op">)</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assume manual inspection produces same results as previous</span></span>
<span><span class="va">blast_results_inspected</span> <span class="op">&lt;-</span> <span class="va">blast_results_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_1</span> <span class="op">==</span> <span class="st">"INSPECT"</span>, <span class="va">hv_blast</span>, <span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span><span class="op">(</span><span class="va">viral_status_1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_2</span> <span class="op">==</span> <span class="st">"INSPECT"</span>, <span class="va">hv_blast</span>, <span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span><span class="op">(</span><span class="va">viral_status_2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Evaluate performance vs fully manual inspection</span></span>
<span><span class="va">blast_status</span> <span class="op">&lt;-</span> <span class="va">blast_results_inspected</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>pos_tru_1 <span class="op">=</span> <span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_fls_1 <span class="op">=</span> <span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_tru_1 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_fls_1 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_tru_2 <span class="op">=</span> <span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_fls_2 <span class="op">=</span> <span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_tru_2 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_fls_2 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="va">hv_blast</span><span class="op">)</span></span>
<span><span class="va">blast_performance_1</span> <span class="op">&lt;-</span> <span class="va">blast_status</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_pos_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_tru_1</span><span class="op">)</span>,</span>
<span>            n_pos_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_fls_1</span><span class="op">)</span>,</span>
<span>            n_neg_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_tru_1</span><span class="op">)</span>,</span>
<span>            n_neg_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_fls_1</span><span class="op">)</span>,</span>
<span>            precision <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            sensitivity <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_neg_fls</span><span class="op">)</span>,</span>
<span>            specificity <span class="op">=</span> <span class="va">n_neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_neg_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            f1 <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ref_taxids <span class="op">=</span> <span class="st">"Expanded human viral"</span><span class="op">)</span></span>
<span><span class="va">blast_performance_2</span> <span class="op">&lt;-</span> <span class="va">blast_status</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_pos_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_tru_2</span><span class="op">)</span>,</span>
<span>            n_pos_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_fls_2</span><span class="op">)</span>,</span>
<span>            n_neg_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_tru_2</span><span class="op">)</span>,</span>
<span>            n_neg_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_fls_2</span><span class="op">)</span>,</span>
<span>            precision <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            sensitivity <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_neg_fls</span><span class="op">)</span>,</span>
<span>            specificity <span class="op">=</span> <span class="va">n_neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_neg_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            f1 <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ref_taxids <span class="op">=</span> <span class="st">"All viral"</span><span class="op">)</span></span>
<span><span class="va">blast_performance</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">blast_performance_1</span>, <span class="va">blast_performance_2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">ref_taxids</span>, <span class="va">n_pos_tru</span><span class="op">:</span><span class="va">f1</span><span class="op">)</span></span>
<span><span class="va">blast_performance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["ref_taxids"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n_pos_tru"],"name":[2],"type":["int"],"align":["right"]},{"label":["n_pos_fls"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_neg_tru"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_neg_fls"],"name":[5],"type":["int"],"align":["right"]},{"label":["precision"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["sensitivity"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["specificity"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["f1"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"Expanded human viral","2":"84","3":"1","4":"79","5":"63","6":"0.9882353","7":"0.5714286","8":"0.9875","9":"0.7241379"},{"1":"All viral","2":"147","3":"1","4":"79","5":"0","6":"0.9932432","7":"1.0000000","8":"0.9875","9":"0.9966102"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Using the inferred list of human-virus taxids results in high precision and specificity, but low sensitivity, missing many reads flagged as human-viral by manual inspection. Conversely, using a list of all viral taxids as the reference achieves perfect sensitivity and near-perfect precision and specificity, with only one false positive result, resulting in an F1 score of over 99%.</p>
<p>That one false positive turned out to be a sequence with a high-identity but low-query-coverage human-viral match (to human enterovirus 71) which was excluded during my previous manual assessment; in my opinion, it’s unclear whether this should really be classed as a false positive. If we want to class this sequence as non-human-viral, increasing the query coverage threshold from 30% to 35% successfully excludes it without losing any true positives, resulting in perfect precision relative to manual assignments:</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Filter BLAST results by query coverage</span></span>
<span><span class="va">blast_results_qcov</span> <span class="op">&lt;-</span> <span class="va">blast_results_filtered</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">qcovs</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">35</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summarize BLAST results by read pair and subject taxid</span></span>
<span><span class="va">blast_results_qcov_paired</span> <span class="op">&lt;-</span> <span class="va">blast_results_qcov</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">separate</span><span class="op">(</span><span class="va">qseqid</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">read_pair</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">bitscore</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">length</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="fu">row_number</span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">staxid</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>bitscore <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, seq_num <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">seq_num</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>bitscore_max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, bitscore_min <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">bitscore</span><span class="op">)</span>, </span>
<span>            n_reads <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign viral status to BLAST results</span></span>
<span><span class="va">blast_results_qcov_hviral</span> <span class="op">&lt;-</span> <span class="va">blast_results_qcov_paired</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_1 <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids_1</span>,</span>
<span>         viral_2 <span class="op">=</span> <span class="va">staxid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">v_taxids_2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_1_full <span class="op">=</span> <span class="va">viral_1</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span>,</span>
<span>         viral_2_full <span class="op">=</span> <span class="va">viral_2</span> <span class="op">&amp;</span> <span class="va">n_reads</span> <span class="op">==</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign putative viral status to read pairs</span></span>
<span><span class="va">blast_results_qcov_out</span> <span class="op">&lt;-</span> <span class="va">blast_results_qcov_hviral</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>viral_status_1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_1_full</span><span class="op">)</span>, <span class="st">"TRUE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_1</span><span class="op">)</span>, <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            viral_status_2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_2_full</span><span class="op">)</span>, <span class="st">"TRUE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">viral_2</span><span class="op">)</span>, <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">reads_db</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">sample</span>, <span class="va">seq_num</span>, <span class="va">hv_blast</span><span class="op">)</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"seq_num"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_1 <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_1</span>, <span class="st">"FALSE"</span><span class="op">)</span>,</span>
<span>         viral_status_2 <span class="op">=</span> <span class="fu">replace_na</span><span class="op">(</span><span class="va">viral_status_2</span>, <span class="st">"FALSE"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assume manual inspection produces same results as previous</span></span>
<span><span class="va">blast_results_qcov_inspected</span> <span class="op">&lt;-</span> <span class="va">blast_results_qcov_out</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>viral_status_1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_1</span> <span class="op">==</span> <span class="st">"INSPECT"</span>, <span class="va">hv_blast</span>, <span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span><span class="op">(</span><span class="va">viral_status_1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>         viral_status_2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">viral_status_2</span> <span class="op">==</span> <span class="st">"INSPECT"</span>, <span class="va">hv_blast</span>, <span class="fu"><a href="https://rdrr.io/r/base/logical.html">as.logical</a></span><span class="op">(</span><span class="va">viral_status_2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Evaluate performance vs fully manual inspection</span></span>
<span><span class="va">blast_status_qcov</span> <span class="op">&lt;-</span> <span class="va">blast_results_qcov_inspected</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>pos_tru_1 <span class="op">=</span> <span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_fls_1 <span class="op">=</span> <span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_tru_1 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_fls_1 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_1</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_tru_2 <span class="op">=</span> <span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="va">hv_blast</span>,</span>
<span>           pos_fls_2 <span class="op">=</span> <span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_tru_2 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">hv_blast</span>,</span>
<span>           neg_fls_2 <span class="op">=</span> <span class="op">!</span><span class="va">viral_status_2</span> <span class="op">&amp;</span> <span class="va">hv_blast</span><span class="op">)</span></span>
<span><span class="va">blast_performance_qcov_1</span> <span class="op">&lt;-</span> <span class="va">blast_status_qcov</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_pos_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_tru_1</span><span class="op">)</span>,</span>
<span>            n_pos_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_fls_1</span><span class="op">)</span>,</span>
<span>            n_neg_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_tru_1</span><span class="op">)</span>,</span>
<span>            n_neg_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_fls_1</span><span class="op">)</span>,</span>
<span>            precision <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            sensitivity <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_neg_fls</span><span class="op">)</span>,</span>
<span>            specificity <span class="op">=</span> <span class="va">n_neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_neg_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            f1 <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ref_taxids <span class="op">=</span> <span class="st">"Expanded human viral"</span><span class="op">)</span></span>
<span><span class="va">blast_performance_qcov_2</span> <span class="op">&lt;-</span> <span class="va">blast_status_qcov</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">summarize</span><span class="op">(</span>n_pos_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_tru_2</span><span class="op">)</span>,</span>
<span>            n_pos_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">pos_fls_2</span><span class="op">)</span>,</span>
<span>            n_neg_tru <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_tru_2</span><span class="op">)</span>,</span>
<span>            n_neg_fls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">neg_fls_2</span><span class="op">)</span>,</span>
<span>            precision <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            sensitivity <span class="op">=</span> <span class="va">n_pos_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_pos_tru</span> <span class="op">+</span> <span class="va">n_neg_fls</span><span class="op">)</span>,</span>
<span>            specificity <span class="op">=</span> <span class="va">n_neg_tru</span> <span class="op">/</span> <span class="op">(</span><span class="va">n_neg_tru</span> <span class="op">+</span> <span class="va">n_pos_fls</span><span class="op">)</span>,</span>
<span>            f1 <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">precision</span> <span class="op">*</span> <span class="va">sensitivity</span> <span class="op">/</span> <span class="op">(</span><span class="va">precision</span> <span class="op">+</span> <span class="va">sensitivity</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>ref_taxids <span class="op">=</span> <span class="st">"All viral"</span><span class="op">)</span></span>
<span><span class="va">blast_performance_qcov</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">blast_performance_qcov_1</span>, <span class="va">blast_performance_qcov_2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="va">ref_taxids</span>, <span class="va">n_pos_tru</span><span class="op">:</span><span class="va">f1</span><span class="op">)</span></span>
<span><span class="va">blast_performance_qcov</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["ref_taxids"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n_pos_tru"],"name":[2],"type":["int"],"align":["right"]},{"label":["n_pos_fls"],"name":[3],"type":["int"],"align":["right"]},{"label":["n_neg_tru"],"name":[4],"type":["int"],"align":["right"]},{"label":["n_neg_fls"],"name":[5],"type":["int"],"align":["right"]},{"label":["precision"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["sensitivity"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["specificity"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["f1"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"Expanded human viral","2":"84","3":"0","4":"80","5":"63","6":"1","7":"0.5714286","8":"1","9":"0.7272727"},{"1":"All viral","2":"147","3":"0","4":"80","5":"0","6":"1","7":"1.0000000","8":"1","9":"1.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>In summary, using command-line <code>blastn -remote,</code> combined with tabular processing in R against a reference class of all viral taxids, works well as a substitute for manual inspection of online BLAST results for validating human-viral read assignments. It’s too slow and failure-prone to be added to the pipeline as a replacement or supplement to the Bowtie/Kraken automated approach, but will be my go-to approach in future for manual validation and refinement of human-viral assignment criteria – at least until I’m able to try out Elastic-BLAST.</p>


<!-- -->


</main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb5" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Automating BLAST validation of human viral read assignment"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Experiments with BLASTN remote mode"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Will Bradshaw"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2024-01-30</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-link: true</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: paged</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="an">title-block-banner:</span><span class="co"> black</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load-packages</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastqcr)</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../scripts/aux_plot-theme.R"</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>theme_kit <span class="ot">&lt;-</span> theme_base <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">aspect.ratio =</span> <span class="dv">1</span>,</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">angle =</span> <span class="dv">45</span>),</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>In previous entries, I presented the results of my <span class="co">[</span><span class="ot">Nextflow workflow</span><span class="co">](https://github.com/naobservatory/mgs-workflow)</span> on recent BMC RNA-seq data. To validate and calibrate my Bowtie2/Kraken2 pipeline for identifying human-viral reads, I manually BLASTed putative viral reads against NCBI's nt database using their online BLAST tool, then assessed the results by eye to determine if each read likely came from a human virus. This approach was effective, but slow and manual, and thus hard to scale to other datasets I wanted to analyze with a higher relative abundance of viral sequences. I thus investigated options for automating the process to generate custom tabular BLAST results on the command line, which could then be read and processed programmatically.</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>Among the several options I investigated, the most promising were (1) <span class="co">[</span><span class="ot">Elastic-BLAST</span><span class="co">](https://blast.ncbi.nlm.nih.gov/doc/elastic-blast/)</span>, NCBI's own cloud-based BLAST tool, and (2) running regular command-line blast with the <span class="in">`-remote`</span> option enabled to query NCBI's online databases. I don't currently have the AWS permissions needed to run Elastic-BLAST on our AWS service, and <span class="in">`blastn -remote`</span> initially seemed too slow to be useful when run on an EC2 instance. However, I discovered that the latter option appears to run much more quickly when run on my local machine, making this a much more attractive option, at least until I'm able to run Elastic-BLAST.</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>To evaluate the potential for this approach to semi-automate the process of HV read validation, I ran BLASTN on the 227 putative HV reads from my previous BMC entry, using the following command:</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="in">blastn -query &lt;input_path&gt; -out &lt;output_path&gt; -db nt -remote -perc_identity 60 -max_hsps 5 -num_alignments 50 -qcov_hsp_perc 30 -outfmt "6 qseqid sseqid sgi staxid qlen evalue bitscore qcovs length pident mismatch gapopen sstrand qstart qend sstart send"</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>To begin with, I first needed to identify a list of taxids I could use to designate a read as human-viral. To do this, I took the list of HV taxids generated by my Nextflow workflow and expanded it to include any missing descendent taxids using the NCBI taxid tree structure. I also generated another list of taxids that included any taxid descended from any of these taxids or the overall virus taxid (10239); this latter approach decreases the risk of false negatives at the cost of potentially increasing the risk of false positives from phage sequences. The two approaches generated lists of 28,105 and 234,499 taxids, respectively.</span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Set file paths</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data/2024-01-30_blast/"</span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>reads_db_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"bmc-hv-putative.tsv"</span>)</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>blast_results_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"bmc-hv-putative.blast"</span>)</span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>hv_taxids_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"human-virus-taxids-all.txt"</span>)</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>tax_nodes_path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_dir, <span class="st">"nodes.dmp.gz"</span>)</span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Import files for viral taxid search</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>hv_taxids <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(hv_taxids_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> <span class="st">"taxid"</span>)</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>tax_nodes <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(tax_nodes_path, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">|</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X1<span class="sc">:</span>X3) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">child_taxid =</span> X1, <span class="at">parent_taxid =</span> X2, <span class="at">rank =</span> X3)</span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Define taxid search function</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>expand_taxids <span class="ot">&lt;-</span> <span class="cf">function</span>(taxids_in, nodes){</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>  taxids_out <span class="ot">&lt;-</span> taxids_in</span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>  taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> (<span class="fu">length</span>(taxids_new) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>    taxids_out <span class="ot">&lt;-</span> <span class="fu">c</span>(taxids_out, taxids_new) <span class="sc">%&gt;%</span> sort</span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>    taxids_new <span class="ot">&lt;-</span> <span class="fu">filter</span>(nodes, parent_taxid <span class="sc">%in%</span> taxids_out, </span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>                         <span class="sc">!</span>child_taxid <span class="sc">%in%</span> taxids_out) <span class="sc">%&gt;%</span></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pull</span>(child_taxid) <span class="sc">%&gt;%</span> sort</span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(taxids_out)</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>v_taxids_1 <span class="ot">&lt;-</span> <span class="fu">expand_taxids</span>(hv_taxids<span class="sc">$</span>taxid, tax_nodes)</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>v_taxids_2 <span class="ot">&lt;-</span> <span class="fu">expand_taxids</span>(<span class="fu">c</span>(<span class="dv">10239</span>, hv_taxids<span class="sc">$</span>taxid), tax_nodes)</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>I next imported the BLAST alignment results and performed exploratory data analysis. Following visual inspection, it appeared that the following process would generate good results:</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>If both the forward and reverse reads from a read pair align to the same viral taxid (given the specified filters on query coverage, percent identity, etc), classify that read pair as human viral.</span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>If only one of the reads in a read pair aligns to any given viral taxid, flag the read for manual inspection.</span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>If neither read aligns to a viral taxid, classify that read pair as non-human-viral.</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare and import BLAST results</span></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>reads_db <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(reads_db_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>blast_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"qseqid"</span>, <span class="st">"sseqid"</span>, <span class="st">"sgi"</span>, <span class="st">"staxid"</span>, <span class="st">"qlen"</span>, <span class="st">"evalue"</span>, <span class="st">"bitscore"</span>, <span class="st">"qcovs"</span>, <span class="st">"length"</span>, <span class="st">"pident"</span>, <span class="st">"mismatch"</span>, <span class="st">"gapopen"</span>, <span class="st">"sstrand"</span>, <span class="st">"qstart"</span>, <span class="st">"qend"</span>, <span class="st">"sstart"</span>, <span class="st">"send"</span>)</span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>blast_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(blast_results_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> blast_cols,</span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default=</span><span class="st">"c"</span>))</span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter BLAST results by read ID</span></span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>reads_db_seqs <span class="ot">&lt;-</span> reads_db <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">read_id =</span> <span class="fu">paste</span>(sample, seq_num, <span class="at">sep=</span><span class="st">"_"</span>)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(read_id) <span class="sc">%&gt;%</span> <span class="fu">c</span>(<span class="fu">paste</span>(., <span class="st">"1"</span>, <span class="at">sep=</span><span class="st">"_"</span>), <span class="fu">paste</span>(., <span class="st">"2"</span>, <span class="at">sep=</span><span class="st">"_"</span>))</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>blast_results_filtered <span class="ot">&lt;-</span> blast_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(qseqid <span class="sc">%in%</span> reads_db_seqs)</span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize BLAST results by read pair and subject taxid</span></span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>blast_results_paired <span class="ot">&lt;-</span> blast_results_filtered <span class="sc">%&gt;%</span></span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(qseqid, <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span>), <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, read_pair, staxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num)) <span class="sc">%&gt;%</span></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore), </span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign viral status to BLAST results</span></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a>blast_results_hviral <span class="ot">&lt;-</span> blast_results_paired <span class="sc">%&gt;%</span></span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_1 =</span> staxid <span class="sc">%in%</span> v_taxids_1,</span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_2 =</span> staxid <span class="sc">%in%</span> v_taxids_2) <span class="sc">%&gt;%</span></span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_1_full =</span> viral_1 <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>,</span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_2_full =</span> viral_2 <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign putative viral status to read pairs</span></span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>blast_results_out <span class="ot">&lt;-</span> blast_results_hviral <span class="sc">%&gt;%</span></span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status_1 =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_1_full), <span class="st">"TRUE"</span>, <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_1), <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span>)),</span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a>            <span class="at">viral_status_2 =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_2_full), <span class="st">"TRUE"</span>, <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_2), <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span>)),</span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(reads_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, hv_blast), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>))</span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Assume manual inspection produces same results as previous</span></span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>blast_results_inspected <span class="ot">&lt;-</span> blast_results_out <span class="sc">%&gt;%</span></span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_1 =</span> <span class="fu">ifelse</span>(viral_status_1 <span class="sc">==</span> <span class="st">"INSPECT"</span>, hv_blast, <span class="fu">as.logical</span>(viral_status_1)),</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_2 =</span> <span class="fu">ifelse</span>(viral_status_2 <span class="sc">==</span> <span class="st">"INSPECT"</span>, hv_blast, <span class="fu">as.logical</span>(viral_status_2)))</span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a><span class="co"># Evaluate performance vs fully manual inspection</span></span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a>blast_status <span class="ot">&lt;-</span> blast_results_inspected <span class="sc">%&gt;%</span></span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pos_tru_1 =</span> viral_status_1 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_fls_1 =</span> viral_status_1 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_tru_1 =</span> <span class="sc">!</span>viral_status_1 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_fls_1 =</span> <span class="sc">!</span>viral_status_1 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_tru_2 =</span> viral_status_2 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_fls_2 =</span> viral_status_2 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_tru_2 =</span> <span class="sc">!</span>viral_status_2 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_fls_2 =</span> <span class="sc">!</span>viral_status_2 <span class="sc">&amp;</span> hv_blast)</span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a>blast_performance_1 <span class="ot">&lt;-</span> blast_status <span class="sc">%&gt;%</span></span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_pos_tru =</span> <span class="fu">sum</span>(pos_tru_1),</span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_pos_fls =</span> <span class="fu">sum</span>(pos_fls_1),</span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_tru =</span> <span class="fu">sum</span>(neg_tru_1),</span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_fls =</span> <span class="fu">sum</span>(neg_fls_1),</span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>            <span class="at">precision =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>            <span class="at">sensitivity =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_neg_fls),</span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a>            <span class="at">specificity =</span> n_neg_tru <span class="sc">/</span> (n_neg_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>            <span class="at">f1 =</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)) <span class="sc">%&gt;%</span></span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ref_taxids =</span> <span class="st">"Expanded human viral"</span>)</span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a>blast_performance_2 <span class="ot">&lt;-</span> blast_status <span class="sc">%&gt;%</span></span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_pos_tru =</span> <span class="fu">sum</span>(pos_tru_2),</span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_pos_fls =</span> <span class="fu">sum</span>(pos_fls_2),</span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_tru =</span> <span class="fu">sum</span>(neg_tru_2),</span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_fls =</span> <span class="fu">sum</span>(neg_fls_2),</span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>            <span class="at">precision =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a>            <span class="at">sensitivity =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_neg_fls),</span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a>            <span class="at">specificity =</span> n_neg_tru <span class="sc">/</span> (n_neg_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>            <span class="at">f1 =</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)) <span class="sc">%&gt;%</span></span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ref_taxids =</span> <span class="st">"All viral"</span>)</span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>blast_performance <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(blast_performance_1, blast_performance_2) <span class="sc">%&gt;%</span> <span class="fu">select</span>(ref_taxids, n_pos_tru<span class="sc">:</span>f1)</span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a>blast_performance</span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a>Using the inferred list of human-virus taxids results in high precision and specificity, but low sensitivity, missing many reads flagged as human-viral by manual inspection. Conversely, using a list of all viral taxids as the reference achieves perfect sensitivity and near-perfect precision and specificity, with only one false positive result, resulting in an F1 score of over 99%.</span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>That one false positive turned out to be a sequence with a high-identity but low-query-coverage human-viral match (to human enterovirus 71) which was excluded during my previous manual assessment; in my opinion, it's unclear whether this should really be classed as a false positive. If we want to class this sequence as non-human-viral, increasing the query coverage threshold from 30% to 35% successfully excludes it without losing any true positives, resulting in perfect precision relative to manual assignments:</span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter BLAST results by query coverage</span></span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>blast_results_qcov <span class="ot">&lt;-</span> blast_results_filtered <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(qcovs) <span class="sc">&gt;=</span> <span class="dv">35</span>)</span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-172"><a href="#cb5-172" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize BLAST results by read pair and subject taxid</span></span>
<span id="cb5-173"><a href="#cb5-173" aria-hidden="true" tabindex="-1"></a>blast_results_qcov_paired <span class="ot">&lt;-</span> blast_results_qcov <span class="sc">%&gt;%</span></span>
<span id="cb5-174"><a href="#cb5-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(qseqid, <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>, <span class="st">"read_pair"</span>), <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-175"><a href="#cb5-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, read_pair, staxid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(bitscore <span class="sc">==</span> <span class="fu">max</span>(bitscore)) <span class="sc">%&gt;%</span></span>
<span id="cb5-176"><a href="#cb5-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(length <span class="sc">==</span> <span class="fu">max</span>(length)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-177"><a href="#cb5-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num, staxid) <span class="sc">%&gt;%</span></span>
<span id="cb5-178"><a href="#cb5-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bitscore =</span> <span class="fu">as.numeric</span>(bitscore), <span class="at">seq_num =</span> <span class="fu">as.numeric</span>(seq_num)) <span class="sc">%&gt;%</span></span>
<span id="cb5-179"><a href="#cb5-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">bitscore_max =</span> <span class="fu">max</span>(bitscore), <span class="at">bitscore_min =</span> <span class="fu">min</span>(bitscore), </span>
<span id="cb5-180"><a href="#cb5-180" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_reads =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb5-181"><a href="#cb5-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-182"><a href="#cb5-182" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign viral status to BLAST results</span></span>
<span id="cb5-183"><a href="#cb5-183" aria-hidden="true" tabindex="-1"></a>blast_results_qcov_hviral <span class="ot">&lt;-</span> blast_results_qcov_paired <span class="sc">%&gt;%</span></span>
<span id="cb5-184"><a href="#cb5-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_1 =</span> staxid <span class="sc">%in%</span> v_taxids_1,</span>
<span id="cb5-185"><a href="#cb5-185" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_2 =</span> staxid <span class="sc">%in%</span> v_taxids_2) <span class="sc">%&gt;%</span></span>
<span id="cb5-186"><a href="#cb5-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_1_full =</span> viral_1 <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>,</span>
<span id="cb5-187"><a href="#cb5-187" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_2_full =</span> viral_2 <span class="sc">&amp;</span> n_reads <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb5-188"><a href="#cb5-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-189"><a href="#cb5-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign putative viral status to read pairs</span></span>
<span id="cb5-190"><a href="#cb5-190" aria-hidden="true" tabindex="-1"></a>blast_results_qcov_out <span class="ot">&lt;-</span> blast_results_qcov_hviral <span class="sc">%&gt;%</span></span>
<span id="cb5-191"><a href="#cb5-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample, seq_num) <span class="sc">%&gt;%</span></span>
<span id="cb5-192"><a href="#cb5-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">viral_status_1 =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_1_full), <span class="st">"TRUE"</span>, <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_1), <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span>)),</span>
<span id="cb5-193"><a href="#cb5-193" aria-hidden="true" tabindex="-1"></a>            <span class="at">viral_status_2 =</span> <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_2_full), <span class="st">"TRUE"</span>, <span class="fu">ifelse</span>(<span class="fu">any</span>(viral_2), <span class="st">"INSPECT"</span>, <span class="st">"FALSE"</span>)),</span>
<span id="cb5-194"><a href="#cb5-194" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-195"><a href="#cb5-195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(reads_db <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample, seq_num, hv_blast), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"seq_num"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-196"><a href="#cb5-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_1 =</span> <span class="fu">replace_na</span>(viral_status_1, <span class="st">"FALSE"</span>),</span>
<span id="cb5-197"><a href="#cb5-197" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_2 =</span> <span class="fu">replace_na</span>(viral_status_2, <span class="st">"FALSE"</span>))</span>
<span id="cb5-198"><a href="#cb5-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-199"><a href="#cb5-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Assume manual inspection produces same results as previous</span></span>
<span id="cb5-200"><a href="#cb5-200" aria-hidden="true" tabindex="-1"></a>blast_results_qcov_inspected <span class="ot">&lt;-</span> blast_results_qcov_out <span class="sc">%&gt;%</span></span>
<span id="cb5-201"><a href="#cb5-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">viral_status_1 =</span> <span class="fu">ifelse</span>(viral_status_1 <span class="sc">==</span> <span class="st">"INSPECT"</span>, hv_blast, <span class="fu">as.logical</span>(viral_status_1)),</span>
<span id="cb5-202"><a href="#cb5-202" aria-hidden="true" tabindex="-1"></a>         <span class="at">viral_status_2 =</span> <span class="fu">ifelse</span>(viral_status_2 <span class="sc">==</span> <span class="st">"INSPECT"</span>, hv_blast, <span class="fu">as.logical</span>(viral_status_2)))</span>
<span id="cb5-203"><a href="#cb5-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-204"><a href="#cb5-204" aria-hidden="true" tabindex="-1"></a><span class="co"># Evaluate performance vs fully manual inspection</span></span>
<span id="cb5-205"><a href="#cb5-205" aria-hidden="true" tabindex="-1"></a>blast_status_qcov <span class="ot">&lt;-</span> blast_results_qcov_inspected <span class="sc">%&gt;%</span></span>
<span id="cb5-206"><a href="#cb5-206" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">pos_tru_1 =</span> viral_status_1 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-207"><a href="#cb5-207" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_fls_1 =</span> viral_status_1 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-208"><a href="#cb5-208" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_tru_1 =</span> <span class="sc">!</span>viral_status_1 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-209"><a href="#cb5-209" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_fls_1 =</span> <span class="sc">!</span>viral_status_1 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-210"><a href="#cb5-210" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_tru_2 =</span> viral_status_2 <span class="sc">&amp;</span> hv_blast,</span>
<span id="cb5-211"><a href="#cb5-211" aria-hidden="true" tabindex="-1"></a>           <span class="at">pos_fls_2 =</span> viral_status_2 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-212"><a href="#cb5-212" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_tru_2 =</span> <span class="sc">!</span>viral_status_2 <span class="sc">&amp;</span> <span class="sc">!</span>hv_blast,</span>
<span id="cb5-213"><a href="#cb5-213" aria-hidden="true" tabindex="-1"></a>           <span class="at">neg_fls_2 =</span> <span class="sc">!</span>viral_status_2 <span class="sc">&amp;</span> hv_blast)</span>
<span id="cb5-214"><a href="#cb5-214" aria-hidden="true" tabindex="-1"></a>blast_performance_qcov_1 <span class="ot">&lt;-</span> blast_status_qcov <span class="sc">%&gt;%</span></span>
<span id="cb5-215"><a href="#cb5-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_pos_tru =</span> <span class="fu">sum</span>(pos_tru_1),</span>
<span id="cb5-216"><a href="#cb5-216" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_pos_fls =</span> <span class="fu">sum</span>(pos_fls_1),</span>
<span id="cb5-217"><a href="#cb5-217" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_tru =</span> <span class="fu">sum</span>(neg_tru_1),</span>
<span id="cb5-218"><a href="#cb5-218" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_fls =</span> <span class="fu">sum</span>(neg_fls_1),</span>
<span id="cb5-219"><a href="#cb5-219" aria-hidden="true" tabindex="-1"></a>            <span class="at">precision =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-220"><a href="#cb5-220" aria-hidden="true" tabindex="-1"></a>            <span class="at">sensitivity =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_neg_fls),</span>
<span id="cb5-221"><a href="#cb5-221" aria-hidden="true" tabindex="-1"></a>            <span class="at">specificity =</span> n_neg_tru <span class="sc">/</span> (n_neg_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-222"><a href="#cb5-222" aria-hidden="true" tabindex="-1"></a>            <span class="at">f1 =</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)) <span class="sc">%&gt;%</span></span>
<span id="cb5-223"><a href="#cb5-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ref_taxids =</span> <span class="st">"Expanded human viral"</span>)</span>
<span id="cb5-224"><a href="#cb5-224" aria-hidden="true" tabindex="-1"></a>blast_performance_qcov_2 <span class="ot">&lt;-</span> blast_status_qcov <span class="sc">%&gt;%</span></span>
<span id="cb5-225"><a href="#cb5-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">n_pos_tru =</span> <span class="fu">sum</span>(pos_tru_2),</span>
<span id="cb5-226"><a href="#cb5-226" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_pos_fls =</span> <span class="fu">sum</span>(pos_fls_2),</span>
<span id="cb5-227"><a href="#cb5-227" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_tru =</span> <span class="fu">sum</span>(neg_tru_2),</span>
<span id="cb5-228"><a href="#cb5-228" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_neg_fls =</span> <span class="fu">sum</span>(neg_fls_2),</span>
<span id="cb5-229"><a href="#cb5-229" aria-hidden="true" tabindex="-1"></a>            <span class="at">precision =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-230"><a href="#cb5-230" aria-hidden="true" tabindex="-1"></a>            <span class="at">sensitivity =</span> n_pos_tru <span class="sc">/</span> (n_pos_tru <span class="sc">+</span> n_neg_fls),</span>
<span id="cb5-231"><a href="#cb5-231" aria-hidden="true" tabindex="-1"></a>            <span class="at">specificity =</span> n_neg_tru <span class="sc">/</span> (n_neg_tru <span class="sc">+</span> n_pos_fls),</span>
<span id="cb5-232"><a href="#cb5-232" aria-hidden="true" tabindex="-1"></a>            <span class="at">f1 =</span> <span class="dv">2</span> <span class="sc">*</span> precision <span class="sc">*</span> sensitivity <span class="sc">/</span> (precision <span class="sc">+</span> sensitivity)) <span class="sc">%&gt;%</span></span>
<span id="cb5-233"><a href="#cb5-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ref_taxids =</span> <span class="st">"All viral"</span>)</span>
<span id="cb5-234"><a href="#cb5-234" aria-hidden="true" tabindex="-1"></a>blast_performance_qcov <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(blast_performance_qcov_1, blast_performance_qcov_2) <span class="sc">%&gt;%</span> <span class="fu">select</span>(ref_taxids, n_pos_tru<span class="sc">:</span>f1)</span>
<span id="cb5-235"><a href="#cb5-235" aria-hidden="true" tabindex="-1"></a>blast_performance_qcov</span>
<span id="cb5-236"><a href="#cb5-236" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb5-237"><a href="#cb5-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-238"><a href="#cb5-238" aria-hidden="true" tabindex="-1"></a>In summary, using command-line <span class="in">`blastn -remote,`</span> combined with tabular processing in R against a reference class of all viral taxids, works well as a substitute for manual inspection of online BLAST results for validating human-viral read assignments. It's too slow and failure-prone to be added to the pipeline as a replacement or supplement to the Bowtie/Kraken automated approach, but will be my go-to approach in future for manual validation and refinement of human-viral assignment criteria – at least until I'm able to try out Elastic-BLAST.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>